{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from pandas import *\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "from matplotlib import pyplot\n",
    "import random\n",
    "from scipy.stats import norm\n",
    "import os\n",
    "import pybedtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_JASPAR_transfac_pfms(infile):\n",
    "    pfms_file = open(infile,'r')\n",
    "    pfms_info = {}\n",
    "    seq_len = 0\n",
    "    for line in pfms_file:\n",
    "        line = line.split('\\n')[0]\n",
    "        if(len(line.split(' ')) > 1):\n",
    "            line = line.split(' ')\n",
    "            if(line[0] == 'DE'):\n",
    "                pfms_info['Matrix_ID'] = line[1]\n",
    "                pfms_info['Matrix_Name'] = line[2]\n",
    "                seq_len = 0\n",
    "            elif(line[0] == 'CC'):\n",
    "                temp = line[1].split(':')\n",
    "                pfms_info[temp[0]] = temp[1]\n",
    "                if(seq_len > 0):\n",
    "                    pfms_info['TF_len'] = seq_len\n",
    "        elif(len(line.split('\\t')) > 1):\n",
    "            line = line.split('\\t')\n",
    "            if(line[0] == 'PO'):\n",
    "                curr_matorder = line[1:]\n",
    "            else:\n",
    "                curr_vals = {}\n",
    "                for n,v in enumerate(line[1:]):\n",
    "                    curr_vals[curr_matorder[n]] = float(v)+1\n",
    "                pfms_info[int(line[0])] = curr_vals\n",
    "                seq_len = int(line[0])\n",
    "        else:\n",
    "            pass\n",
    "#             print(line)\n",
    "    pfms_file.close()\n",
    "    return pfms_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_matrix_byTFName(tfname,info_dicts,seq): #old\n",
    "    seq = seq.upper()\n",
    "    matrix_dict_touse = None\n",
    "    for i in info_dicts:\n",
    "        if(i['Matrix_Name'] == tfname):\n",
    "            matrix_dict_touse = i\n",
    "            break\n",
    "    if(matrix_dict_touse == None):\n",
    "        print('Could not find a PWM for Transcription Factor {0}'.format(tfname))\n",
    "        return None\n",
    "    seq_val = 0\n",
    "    for n,b in enumerate(seq):\n",
    "        try:\n",
    "            seq_val += float(matrix_dict_touse[n+1][b])\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "            else:\n",
    "                print('Sequence of length {0} is too long'.format(len(seq)))\n",
    "            return None\n",
    "    return seq_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_matrix_byTF(tfname,info_dicts):\n",
    "    matrix_dict_touse = None\n",
    "    for i in info_dicts:\n",
    "        if(i['Matrix_Name'] == tfname):\n",
    "            matrix_dict_touse = i\n",
    "            break\n",
    "    if(matrix_dict_touse == None):\n",
    "        print('Could not find a PWM for Transcription Factor {0}'.format(tfname))\n",
    "        return None\n",
    "    return matrix_dict_touse\n",
    "\n",
    "def get_lnPWM_from_matrixdict(matrix_dict):\n",
    "    lnPWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        for b in 'ACTG':\n",
    "            f = float(matrix_dict[en][b])\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = np.log(1)\n",
    "            else:\n",
    "                temp_matrix[b] = np.log(f)\n",
    "        lnPWM_dict[en] = temp_matrix\n",
    "    return lnPWM_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fraclnPWM_from_matrixdict(matrix_dict):\n",
    "    lnPWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in matrix_dict[en].values()])\n",
    "        for b in 'ACTG':\n",
    "            f = float(matrix_dict[en][b])/curr_totcount\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = np.log(1)\n",
    "            else:\n",
    "                temp_matrix[b] = np.log(f)\n",
    "        lnPWM_dict[en] = temp_matrix\n",
    "    return lnPWM_dict\n",
    "\n",
    "def get_fracPWM_from_matrixdict(matrix_dict):\n",
    "    PWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in matrix_dict[en].values()])\n",
    "        for b in 'ACTG':\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = 1/curr_totcount\n",
    "            else:\n",
    "                temp_matrix[b] = float(matrix_dict[en][b])/curr_totcount\n",
    "        PWM_dict[en] = temp_matrix\n",
    "    return PWM_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def get_frac_matrix_scores(tfname,pwm,ref_al,alt_al,position,reference_seq):\n",
    "    ref_al = ref_al.upper()\n",
    "    alt_al = alt_al.upper()\n",
    "    reference_seq = reference_seq.upper()\n",
    "    ref_seqval_list,alt_seqval_list = [],[]\n",
    "    for n,b in enumerate(reference_seq):\n",
    "        try:\n",
    "            if(n == position):\n",
    "                ref_seqval_list.append(float(pwm[n+1][ref_al]))\n",
    "                alt_seqval_list.append(float(pwm[n+1][alt_al]))\n",
    "            else:\n",
    "                ref_seqval_list.append(float(pwm[n+1][b]))\n",
    "                alt_seqval_list.append(float(pwm[n+1][b]))\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                continue\n",
    "\n",
    "    return ref_seqval_list,alt_seqval_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "test_fPWM = get_fracPWM_from_matrixdict(test)\n",
    "test_flnPWM = get_fraclnPWM_from_matrixdict(test)\n",
    "test_pos_counts = get_matrix_counts(test,test['TF_len'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\x01'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[-1.6558171380122764,\n",
       " -10.551088053140093,\n",
       " -10.551114217794265,\n",
       " -2.9491337619027305,\n",
       " -0.8244308526280849,\n",
       " -0.7452422924433473]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get_matrix_counts(test,test['TF_len'])\n",
    "# test\n",
    "get_matrix_score('DUX4',test_flnPWM,'ACCGTT')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_lnmatrix_scores(tfname,matrix_dict,ref_al,alt_al,position,reference_seq):\n",
    "    ref_al = ref_al.upper()\n",
    "    alt_al = alt_al.upper()\n",
    "    reference_seq = reference_seq.upper()\n",
    "    ref_seqval,alt_seqval = 0,0\n",
    "    seqpos_counts = []\n",
    "    seqpos_counts_bybase = {'A':[],'C':[],'G':[],'T':[],}\n",
    "    for n,b in enumerate(reference_seq):\n",
    "        try:\n",
    "            if(n == position):\n",
    "                ref_seqval += float(matrix_dict[n+1][ref_al])\n",
    "                alt_seqval += float(matrix_dict[n+1][alt_al])\n",
    "            else:\n",
    "                ref_seqval += float(matrix_dict[n+1][b])\n",
    "                alt_seqval += float(matrix_dict[n+1][b])\n",
    "            curr_vals = [float(x) for x in matrix_dict[n+1].values()]\n",
    "            seqpos_counts.append(sum(curr_vals))\n",
    "            seqpos_counts_bybase['A'].append(matrix_dict[n+1]['A'])\n",
    "            seqpos_counts_bybase['C'].append(matrix_dict[n+1]['C'])\n",
    "            seqpos_counts_bybase['G'].append(matrix_dict[n+1]['G'])\n",
    "            seqpos_counts_bybase['T'].append(matrix_dict[n+1]['T'])\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                continue\n",
    "        \n",
    "#     total_count_fraction = seqpos_counts/len(seqpos_counts)\n",
    "    refseq_fraction = ref_seqval/sum(seqpos_counts)\n",
    "    altseq_fraction = alt_seqval/sum(seqpos_counts)\n",
    "    return ref_seqval,alt_seqval,seqpos_counts_bybase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_numbp_forseq(seq):\n",
    "    bpcounts = {'A':0,'C':0,'G':0,'T':0}\n",
    "    for b in seq:\n",
    "        try:\n",
    "            bpcounts[b] += 1\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T!'.format(b))\n",
    "    bpc_list = []\n",
    "    for o in 'ACGT':\n",
    "        bpc_list.append(bpcounts[o])\n",
    "    return bpc_list\n",
    "            \n",
    "def compute_Y(seq,bg_props):\n",
    "#     seq_bpcounts = get_numbp_forseq(seq)\n",
    "    bpcounts = {'A':0,'C':0,'G':0,'T':0}\n",
    "    for b in seq:\n",
    "        try:\n",
    "            bpcounts[b] += 1\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T!'.format(b))\n",
    "    bpc_list = []\n",
    "    for o in 'ACGT':\n",
    "        bpc_list.append(bpcounts[o])\n",
    "    Y = np.dot(np.log(bg_props),bpc_list)\n",
    "    return Y\n",
    "\n",
    "def make_seq(fullseq,position,allele):\n",
    "    new_seq = ''\n",
    "    for n,b in enumerate(fullseq):\n",
    "        try:\n",
    "            if(n==position):\n",
    "                new_seq = ''.join([new_seq,allele])\n",
    "            else:\n",
    "                new_seq = ''.join([new_seq,b])\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                print(b)\n",
    "                continue\n",
    "    return new_seq.upper()\n",
    "\n",
    "def get_matrix_counts(pwm,seqlen):\n",
    "    pos_counts = []\n",
    "    for n in range(1,seqlen+1):\n",
    "        temp = [float(x) for x in pwm[n].values()]\n",
    "        pos_counts.append(sum(temp))\n",
    "    return pos_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def get_lnPWM_from_fracPWM(fracPWM,bgfreqs):\n",
    "    lnPWM_dict = {}\n",
    "    for en in range(1,len(fracPWM)+1):\n",
    "        temp_matrix = {}\n",
    "        for b in 'ACTG':\n",
    "            f = float(fracPWM[en][b])/bgfreqs['frac_{0}'.format(b)].values[0]\n",
    "            if(f == 0.0):\n",
    "                print('fraction is 0')\n",
    "                temp_matrix[b] = np.log(1)\n",
    "            else:\n",
    "                temp_matrix[b] = -np.log(f)\n",
    "        lnPWM_dict[en] = temp_matrix\n",
    "    return lnPWM_dict\n",
    "\n",
    "def get_fracPWM_from_matrixdict(matrix_dict):\n",
    "    PWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in matrix_dict[en].values()])\n",
    "        for b in 'ACTG':\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = 1/curr_totcount\n",
    "            else:\n",
    "                temp_matrix[b] = float(matrix_dict[en][b])/curr_totcount\n",
    "        PWM_dict[en] = temp_matrix\n",
    "    return PWM_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def get_matrix_scores(pwm,seq):\n",
    "    seqval_list = []\n",
    "    for n,b in enumerate(seq):\n",
    "        try:\n",
    "            seqval_list.append(float(pwm[n+1][b]))\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                continue\n",
    "    return seqval_list\n",
    "def get_complseq(seq):\n",
    "    new_seq = []\n",
    "    for b in seq:\n",
    "        if(b == 'A'):\n",
    "            new_seq.append('T')\n",
    "        elif(b == 'T'):\n",
    "            new_seq.append('A')\n",
    "        elif(b == 'G'):\n",
    "            new_seq.append('C')\n",
    "        elif(b == 'C'):\n",
    "            new_seq.append('G')\n",
    "        else:\n",
    "            print('Base pair not A/C/T/G! {0}'.format(b))\n",
    "    return ''.join(new_seq)\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'infodicts_list' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-e1eaf01d5f37>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtest\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_byTF\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'DUX4'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfodicts_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mln_test\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_lnPWM_from_matrixdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mtest_counts\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_counts\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtest\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m11\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mtest_matscores\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_scores\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'DUX4'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mtest\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'ACCTGGACATG'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'infodicts_list' is not defined"
     ]
    }
   ],
   "source": [
    "test = get_matrix_byTF('DUX4',infodicts_list)\n",
    "ln_test = get_lnPWM_from_matrixdict(test)\n",
    "test_counts = get_matrix_counts(test,11)\n",
    "test_matscores = get_matrix_scores('DUX4',test,'ACCTGGACATG')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# test\n",
    "# r,a,c = get_lnmatrix_scores('DUX4',ln_test,'C','A',2,'ACCTG')\n",
    "test_ln_fracPWM = get_fracPWM_from_matrixdict(test)\n",
    "ln_test = get_lnPWM_from_matrixdict(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: {'A': 0.4344962438862845,\n",
       "  'C': 1.1435191821247397,\n",
       "  'T': -0.8888460955971449,\n",
       "  'G': 2.0788455465242905},\n",
       " 2: {'A': -1.1888753833368704,\n",
       "  'C': 8.965963911614057,\n",
       "  'T': 9.3311660557505,\n",
       "  'G': 1.8622540252013808},\n",
       " 3: {'A': -1.2212100269186392,\n",
       "  'C': 8.965963911614057,\n",
       "  'T': 9.3311660557505,\n",
       "  'G': 8.966398118188907},\n",
       " 4: {'A': 0.7678765266945468,\n",
       "  'C': -0.39932682035994754,\n",
       "  'T': -0.5373676675220993,\n",
       "  'G': 1.3639967825230899},\n",
       " 5: {'A': 9.329851860882677,\n",
       "  'C': -0.7438782103966722,\n",
       "  'T': -0.3955246506343733,\n",
       "  'G': 0.45281142236678246},\n",
       " 6: {'A': 9.329851860882677,\n",
       "  'C': -0.5774142345347029,\n",
       "  'T': -0.47470866791862754,\n",
       "  'G': 0.24577440676848086},\n",
       " 7: {'A': -1.1411913041651307,\n",
       "  'C': 0.9969521305075801,\n",
       "  'T': 5.399340423026175,\n",
       "  'G': 8.966398118188907},\n",
       " 8: {'A': -1.2212100269186392,\n",
       "  'C': 8.965963911614057,\n",
       "  'T': 9.3311660557505,\n",
       "  'G': 8.966398118188907},\n",
       " 9: {'A': 9.329851860882677,\n",
       "  'C': 8.965963911614057,\n",
       "  'T': -1.2198958320508166,\n",
       "  'G': 8.966398118188907},\n",
       " 10: {'A': 9.329851860882677,\n",
       "  'C': -1.563702684550223,\n",
       "  'T': 2.6341318080840157,\n",
       "  'G': 8.966398118188907},\n",
       " 11: {'A': -1.098334634351948,\n",
       "  'C': 8.965963911614057,\n",
       "  'T': 0.9372710806787562,\n",
       "  'G': 8.966398118188907}}"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_lnPWM_from_fracPWM(test_ln_fracPWM,test_bgfreqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_seq = 'ACCTGG'\n",
    "test_bgfreqs1 = np.array(bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'][['frac_A','frac_C','frac_G','frac_T']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'][['frac_A','frac_C','frac_G','frac_T']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-42-86c7a2185d87>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtest_bgfreqs1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'A'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices"
     ]
    }
   ],
   "source": [
    "test_bgfreqs1['A']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] File b'/local3/jake/TF_binding/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes' does not exist: b'/local3/jake/TF_binding/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-12-5bab6eab3699>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mgene_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/local3/jake/TF_binding/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdelimiter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mgene_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'pos_start'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'pos_end'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mposition\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m61768012\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mchromosome\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m8\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mref_pos_end\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgene_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos_end'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    683\u001b[0m         )\n\u001b[1;32m    684\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 685\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    686\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    687\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    455\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    456\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 457\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfp_or_buf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    458\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    459\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    893\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1133\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"c\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1134\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"c\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1135\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1136\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1137\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"python\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1915\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"usecols\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1916\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1917\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1918\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1919\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] File b'/local3/jake/TF_binding/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes' does not exist: b'/local3/jake/TF_binding/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes'"
     ]
    }
   ],
   "source": [
    "gene_df = read_csv('/local3/jake/TF_binding/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes',header=None,delimiter='\\t')\n",
    "gene_df.columns = ['pos_start','pos_end','tf_name']\n",
    "position = 61768012\n",
    "chromosome = 8\n",
    "ref_pos_end = max(gene_df['pos_end'])\n",
    "ref_pos_start = min(gene_df['pos_start'])\n",
    "gene_df['relative_start'] =  gene_df['pos_start']-ref_pos_start\n",
    "gene_df['relative_end'] = gene_df['pos_end']-ref_pos_start\n",
    "\n",
    "matrix_loc = '/local3/jake/TF_binding/JASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac/'\n",
    "# matrix_loc = 'gene_dfJASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac'\n",
    "transfac_matrix_list = os.listdir(matrix_loc)\n",
    "infodicts_list = []\n",
    "for f in transfac_matrix_list:\n",
    "    curr_infodict = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    infodicts_list.append(curr_infodict)\n",
    "    \n",
    "# for i,g in gene_df.iterrows():\n",
    "#     curr_score = get_matrix_byTFName(g[2],infodicts_list,'ACT')\n",
    "#     print('score for {0} is {1}'.format(g[2],curr_score))\n",
    "    \n",
    "\n",
    "# ref_full_seq = get_hg19reference_sequence(ref_pos_start,ref_pos_end,'/local3/jake/TF_binding/chr4.fa')\n",
    "ref_full_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(8,ref_pos_start,(ref_pos_end)),'/local3/jake/TF_binding/chr8.fa')\n",
    "\n",
    "updated_pos = (position-ref_pos_start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "bgfrac_df = read_csv('/local3/jake/TF_binding/ACTG_count.all_chrms.fractions.v2.txt',delimiter='\\t')\n",
    "score_dict_bytf ={}\n",
    "for i,g in gene_df.iterrows():\n",
    "    curr_relative_pos = abs(updated_pos-g['relative_start'])\n",
    "    curr_refseq = make_seq(ref_full_seq[g['relative_start']:(g['relative_end']+1)],curr_relative_pos,'G')\n",
    "    curr_altseq = make_seq(ref_full_seq[g['relative_start']:(g['relative_end']+1)],curr_relative_pos,'A')\n",
    "    curr_matrix = get_matrix_byTF(g['tf_name'],infodicts_list)\n",
    "    curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "    curr_lnfracPWM = get_fraclnPWM_from_matrixdict(curr_matrix)\n",
    "    # curr_refscore,curr_altscore,curr_counts = get_matrix_scores(tfname=g['tf_name'],matrix_dict=curr_matrix,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    # curr_totcount = sum(curr_counts)\n",
    "    refscore_list = get_matrix_scores(curr_matrix,curr_refseq)\n",
    "    altscore_list = get_matrix_scores(curr_matrix,curr_altseq)\n",
    "    pos_counts = get_matrix_counts(curr_matrix,curr_matrix['TF_len'])\n",
    "    tot_count = sum(pos_counts)\n",
    "    curr_scoredict = {'ref_score':sum(refscore_list),'alt_score':sum(altscore_list),'tf_len':curr_matrix['TF_len'],'counts_perpos':min(pos_counts)}\n",
    "    curr_scoredict['ref_fraction_score'] = sum([refscore_list[x]/pos_counts[x] for x in range(len(pos_counts))])\n",
    "    curr_scoredict['alt_fraction_score'] = sum([altscore_list[x]/pos_counts[x] for x in range(len(pos_counts))])\n",
    "\n",
    "    refscore_ln = get_matrix_scores(curr_lnfracPWM,curr_refseq)\n",
    "    altscore_ln = get_matrix_scores(curr_lnfracPWM,curr_altseq)\n",
    "    curr_scoredict['ref_fraction_ln_score'] = sum(refscore_ln)\n",
    "    curr_scoredict['alt_fraction_ln_score'] = sum(altscore_ln)\n",
    "\n",
    "    # frac_refvals,frac_altvals = get_frac_matrix_scores(tfname=g['tf_name'],pwm=curr_fracPWM,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    # fracln_refvals,fracln_altvals = get_frac_matrix_scores(tfname=g['tf_name'],pwm=curr_lnfracPWM,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    bgfreqs = np.array(bgfrac_df.loc[bgfrac_df['Chrm'] == str(chromosome)][['frac_A','frac_C','frac_G','frac_T']])\n",
    "    ref_Y = compute_Y(curr_refseq,bgfreqs)\n",
    "    alt_Y = compute_Y(curr_altseq,bgfreqs)\n",
    "    ref_H = refscore_ln - ref_Y\n",
    "    alt_H = altscore_ln - alt_Y\n",
    "    curr_scoredict['H'] = np.sum(ref_H)\n",
    "    curr_scoredict['Hprime'] = np.sum(alt_H)\n",
    "\n",
    "    score_dict_bytf[g['tf_name']] = curr_scoredict\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "seqA = 'AAAAAAAAAAAAAA'\n",
    "seqB = 'GAAAAAAAAAAAAA'\n",
    "curr_matrix = read_JASPAR_transfac_pfms('/local3/jake/TF_binding/TF_1.transfac')\n",
    "bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'][['frac_A','frac_C','frac_G','frac_T']]\n",
    "# curr_lnfracPWM = get_fraclnPWM_from_matrixdict(curr_matrix)\n",
    "curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "curr_lnfracPWM = get_lnPWM_from_fracPWM(curr_fracPWM,bgfreqs)\n",
    "Ascore_ln = get_matrix_scores(curr_lnfracPWM,seqA)\n",
    "Bscore_ln = get_matrix_scores(curr_lnfracPWM,seqB)\n",
    "\n",
    "A_Y = compute_Y(seqA,np.array(bgfreqs))\n",
    "B_Y = compute_Y(seqB,np.array(bgfreqs))\n",
    "A_H = Ascore_ln - A_Y\n",
    "B_H = Bscore_ln - B_Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-8.983879845730163"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.sum(Ascore_ln)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-7.158948847656853"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.sum(Bscore_ln)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_Y(seq,bg_props):\n",
    "#     seq_bpcounts = get_numbp_forseq(seq)\n",
    "    bpcounts = {'A':0,'C':0,'G':0,'T':0}\n",
    "    for b in seq:\n",
    "        try:\n",
    "            bpcounts[b] += 1\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T!'.format(b))\n",
    "    bpc_list = []\n",
    "    for o in 'ACGT':\n",
    "        bpc_list.append(bpcounts[o])\n",
    "    Y = np.dot(np.log(bg_props),bpc_list)\n",
    "    return Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([], shape=(0, 4), dtype=float64)"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sum_A = np.sum(bgfrac_df['count_A'])\n",
    "# sum_C = np.sum(bgfrac_df['count_C'])\n",
    "# sum_G = np.sum(bgfrac_df['count_G'])\n",
    "# sum_T = np.sum(bgfrac_df['count_T'])\n",
    "# sum_total = np.sum(bgfrac_df['total'])\n",
    "# bgfrac_df = bgfrac_df.append({'Chrm':'Total','count_A':sum_A,'count_C':sum_C,'count_G':sum_G,'count_T':sum_T,'total':sum_total,'frac_A':(sum_A/sum_total),'frac_C':(sum_C/sum_total),'frac_G':(sum_G/sum_total),'frac_T':(sum_T/sum_total)},ignore_index=True)\n",
    "np.array(bgfrac_df.loc[bgfrac_df['Chrm'] == 8][['frac_A','frac_C','frac_G','frac_T']])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'TTTG'"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_complseq('AAAC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "bgfrac_df.to_csv('/local3/jake/TF_binding/ACTG_count.all_chrms.fractions.v2.txt',sep='\\t',index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Beta'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4735\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4736\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mlibindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value_box\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4737\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mIndexError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.get_value_box\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.get_value_at\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/util.pxd\u001b[0m in \u001b[0;36mpandas._libs.util.get_value_at\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/util.pxd\u001b[0m in \u001b[0;36mpandas._libs.util.validate_indexer\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'str' object cannot be interpreted as an integer",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-fbc78aa974ec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     17\u001b[0m         \u001b[0mtemp_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Pos'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m         \u001b[0mtemp_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'snp_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'VID'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 19\u001b[0;31m         \u001b[0mtemp_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'beta'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Beta'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     20\u001b[0m         \u001b[0mtemp_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pvalue'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pval'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mtemp_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ref_allele'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Ref'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1066\u001b[0m         \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1067\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1068\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1069\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1070\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4742\u001b[0m                     \u001b[0;32mraise\u001b[0m \u001b[0mInvalidIndexError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4743\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4744\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0me1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4745\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# pragma: no cover\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4746\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0me1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/msprime-env/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4728\u001b[0m         \u001b[0mk\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_convert_scalar_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkind\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"getitem\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4729\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4730\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtz\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"tz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4731\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4732\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mholds_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_boolean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_value\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Beta'"
     ]
    }
   ],
   "source": [
    "results_loc = '/local3/jake/TF_binding/tophit_SNPs/output_v2/'\n",
    "pwmresults_list = os.listdir(results_loc)\n",
    "\n",
    "assocfile_loc = '/local3/mhansen/Projects/CTA/Analysis/GWAS/All_Inds/Results/5M_Filtered'\n",
    "infofile_loc = '/local3/jake/TF_binding/tophit_SNPs/'\n",
    "infofile_list = os.listdir(infofile_loc)\n",
    "\n",
    "final_df_dictlist = []\n",
    "for i in infofile_list:\n",
    "    if('.full_assoc' not in i):\n",
    "        continue\n",
    "    curr_df = read_csv('{0}{1}'.format(infofile_loc,i),delimiter='\\t')\n",
    "    curr_prefix = '.'.join(i.split('.')[:-1])\n",
    "    temp_dict = {}\n",
    "    for num,snp in curr_df.iterrows():\n",
    "        temp_dict['chr'] = snp['#Chr']\n",
    "        temp_dict['pos'] = snp['Pos']\n",
    "        temp_dict['snp_id'] = snp['VID']\n",
    "        temp_dict['beta'] = snp['Beta']\n",
    "        temp_dict['pvalue'] = snp['pval']\n",
    "        temp_dict['ref_allele'] = snp['Ref']\n",
    "        temp_dict['alt_allele'] = snp['Alt']\n",
    "        temp_dict['trait_num'] = curr_prefix.split('.')[0]\n",
    "        curr_pwmfile = '{0}{1}.chrm{2}.pos{3}.PWM_scores'.format(results_loc,curr_prefix,snp['#Chr'],snp['Pos'])\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'args' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-ae9873ab5b8e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0mgene_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/local3/jake/TF_binding/tophit_SNPs/output/24.5M.TopVars.LE.chrm4.pos165034923.TF_genes'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdelimiter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0mgene_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'pos_start'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'pos_end'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m \u001b[0mtransfac_matrix_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmatrix_loc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m \u001b[0minfodicts_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtransfac_matrix_list\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'args' is not defined"
     ]
    }
   ],
   "source": [
    "test_pwmfile = '24.5M.TopVars.LE.chrm4.pos165034923.PWM_scores'\n",
    "# position = int(args.position.split(':')[1])\n",
    "position = 165034923\n",
    "# chromosome = int(args.position.split(':')[0])\n",
    "chromosome = 4\n",
    "\n",
    "# gene_df = read_csv('{0}'.format(args.input_gene_file),header=None,delimiter='\\t')\n",
    "gene_df = read_csv('/local3/jake/TF_binding/tophit_SNPs/output/24.5M.TopVars.LE.chrm4.pos165034923.TF_genes',header=None,delimiter='\\t')\n",
    "gene_df.columns = ['pos_start','pos_end','tf_name']\n",
    "# transfac_matrix_list = os.listdir(args.matrix_loc)\n",
    "matrix_loc = '/local3/jake/TF_binding/JASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac/'\n",
    "# matrix_loc = 'gene_dfJASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac'\n",
    "transfac_matrix_list = os.listdir(matrix_loc)\n",
    "infodicts_list = []\n",
    "for f in transfac_matrix_list:\n",
    "    curr_infodict = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    infodicts_list.append(curr_infodict)\n",
    "\n",
    "ref_pos_end = max(gene_df['pos_end'])\n",
    "ref_pos_start = min(gene_df['pos_start'])\n",
    "# ref_full_seq = get_hg19reference_sequence(ref_pos_start,ref_pos_end,args.ref_fasta_file)\n",
    "ref_full_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(chromosome,ref_pos_start,ref_pos_end),args.ref_fasta_file)\n",
    "updated_pos = (position-ref_pos_start)\n",
    "gene_df['relative_start'] =  gene_df['pos_start']-ref_pos_start\n",
    "gene_df['relative_end'] = gene_df['pos_end']-ref_pos_start\n",
    "\n",
    "score_dict_bytf ={}\n",
    "for i,g in gene_df.iterrows():\n",
    "    curr_relative_pos = abs(updated_pos-g['relative_start'])\n",
    "    curr_seq = ref_full_seq[g['relative_start']:(g['relative_end']+1)]\n",
    "    curr_matrix = get_matrix_byTF(g['tf_name'],infodicts_list)\n",
    "    curr_refscore,curr_altscore,curr_counts = get_matrix_scores(tfname=g['tf_name'],matrix_dict=curr_matrix,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    curr_totcount = sum(curr_counts)\n",
    "    score_dict_bytf[g['tf_name']] = {'ref_score':curr_refscore,'alt_score':curr_altscore,'ref_fraction_score':(curr_refscore/curr_totcount),'alt_fraction_score':(curr_altscore/curr_totcount),'tf_len':len(curr_seq),'counts_perpos':curr_counts[0]}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "results_loc = '/local3/jake/TF_binding/tophit_SNPs/output_v3/'\n",
    "pwmresults_list = os.listdir(results_loc)\n",
    "\n",
    "assocfile_loc = '/local3/mhansen/Projects/CTA/Analysis/GWAS/All_Inds/Results/5M_Filtered'\n",
    "infofile_loc = '/local3/jake/TF_binding/tophit_SNPs/'\n",
    "infofile_list = os.listdir(infofile_loc)\n",
    "\n",
    "final_df_dictlist = []\n",
    "for i in infofile_list:\n",
    "    if('.full_assoc' not in i):\n",
    "        continue\n",
    "    elif(i == '8.1M_5M.TopVars.LE.5M.full_assoc' or i == '7.1M_5M.TopVars.LE.5M.full_assoc'):\n",
    "        continue\n",
    "    curr_df = read_csv('{0}{1}'.format(infofile_loc,i),delimiter='\\t')\n",
    "    curr_prefix = '.'.join(i.split('.')[:-1])\n",
    "    temp_dict = {}\n",
    "    for num,snp in curr_df.iterrows():\n",
    "        temp_dict['chr'] = snp['#Chr']\n",
    "        temp_dict['pos'] = snp['Pos']\n",
    "        temp_dict['snp_id'] = snp['VID']\n",
    "        if(i == '8.1M_5M.TopVars.LE.full_assoc' or i == '7.1M_5M.TopVars.LE.full_assoc' ):\n",
    "            temp_dict['beta'] = snp['Beta_FE']\n",
    "            temp_dict['pvalue'] = snp['pval_FE']\n",
    "            if(i == '8.1M_5M.TopVars.LE.full_assoc'):\n",
    "                d = read_csv('{0}{1}'.format(infofile_loc,'8.1M_5M.TopVars.LE.5M.full_assoc'),delimiter='\\t')\n",
    "                temp_dict['ref_allele'] = d.loc[num]['Ref']\n",
    "                temp_dict['alt_allele'] = d.loc[num]['Alt']\n",
    "            if(i == '7.1M_5M.TopVars.LE.full_assoc'):\n",
    "                d = read_csv('{0}{1}'.format(infofile_loc,'7.1M_5M.TopVars.LE.5M.full_assoc'),delimiter='\\t')\n",
    "                temp_dict['ref_allele'] = d.loc[num]['Ref']\n",
    "                temp_dict['alt_allele'] = d.loc[num]['Alt']\n",
    "        else:\n",
    "            temp_dict['pvalue'] = snp['pval']\n",
    "            temp_dict['beta'] = snp['Beta']\n",
    "            temp_dict['ref_allele'] = snp['Ref']\n",
    "            temp_dict['alt_allele'] = snp['Alt']\n",
    "        temp_dict['trait_num'] = curr_prefix.split('.')[0]\n",
    "        curr_pwmfile = '{0}{1}.chrm{2}.pos{3}.PWM_scores'.format(results_loc,curr_prefix,snp['#Chr'],snp['Pos'])\n",
    "        curr_pwm_df = read_csv(curr_pwmfile,skiprows=1,delimiter='\\t')\n",
    "        for index,tf in curr_pwm_df.iterrows():\n",
    "            curr_dict = temp_dict.copy()\n",
    "            curr_dict['TF_name'] = tf['TF_Name']\n",
    "            curr_dict['PWM_fracscore_ref'] = \"{0:.5f}\".format(tf['PWM Fraction Score (REF allele)'])\n",
    "            curr_dict['PWM_fracscore_alt'] = \"{0:.5f}\".format(tf['PWM Fraction Score (ALT allele)'])\n",
    "            curr_dict['TF_len'] = tf['TF Length']\n",
    "            curr_dict['TF_count'] = tf['TF Counts per position']\n",
    "            curr_dict['PWM_ln_fracscore_ref'] = \"{0:.5f}\".format(tf['REF Log(PWM Fraction Score)'])\n",
    "            curr_dict['PWM_ln_fracscore_alt'] = \"{0:.5f}\".format(tf['ALT Log(PWM Fraction Score)'])\n",
    "            curr_dict['H (REF)'] = \"{0:.5f}\".format(tf['H (REF)'])\n",
    "            curr_dict['Hprime (ALT)'] = \"{0:.5f}\".format(tf['Hprime (ALT)'])\n",
    "            final_df_dictlist.append(curr_dict)\n",
    "\n",
    "final_df = DataFrame(final_df_dictlist)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7.1M_5M.TopVars.LE.full_assoc\n",
      "8.1M_5M.TopVars.LE.full_assoc\n"
     ]
    }
   ],
   "source": [
    "for i in infofile_list:\n",
    "    if('.full_assoc' not in i):\n",
    "        continue\n",
    "    elif(i == '8.1M_5M.TopVars.LE.5M.full_assoc' or i == '7.1M_5M.TopVars.LE.5M.full_assoc'):\n",
    "        continue\n",
    "    if(i == '8.1M_5M.TopVars.LE.full_assoc' or i == '7.1M_5M.TopVars.LE.full_assoc' ):\n",
    "        print(i)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('/local3/jake/TF_binding/tophit_SNPs/final_PWM.alltraits.allsnps.output_v3',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>beta</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>trait_num</th>\n",
       "      <th>TF_name</th>\n",
       "      <th>PWM_fracscore_ref</th>\n",
       "      <th>PWM_fracscore_alt</th>\n",
       "      <th>TF_len</th>\n",
       "      <th>TF_count</th>\n",
       "      <th>PWM_ln_fracscore_ref</th>\n",
       "      <th>PWM_ln_fracscore_alt</th>\n",
       "      <th>H (REF)</th>\n",
       "      <th>Hprime (ALT)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>13308152</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>1.252538e-11</td>\n",
       "      <td>6.312838</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>TFAP2A(var.2)</td>\n",
       "      <td>4.92927</td>\n",
       "      <td>5.30035</td>\n",
       "      <td>12</td>\n",
       "      <td>4009.0</td>\n",
       "      <td>-18.36669</td>\n",
       "      <td>-16.88040</td>\n",
       "      <td>213.84023</td>\n",
       "      <td>210.31008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>13308152</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>1.252538e-11</td>\n",
       "      <td>6.312838</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>BARX2</td>\n",
       "      <td>3.84562</td>\n",
       "      <td>4.58265</td>\n",
       "      <td>12</td>\n",
       "      <td>28391.0</td>\n",
       "      <td>-32.67471</td>\n",
       "      <td>-26.33297</td>\n",
       "      <td>194.51138</td>\n",
       "      <td>195.83667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>13308152</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>1.252538e-11</td>\n",
       "      <td>6.312838</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>4.02331</td>\n",
       "      <td>4.07698</td>\n",
       "      <td>12</td>\n",
       "      <td>1416.0</td>\n",
       "      <td>-25.29941</td>\n",
       "      <td>-24.16443</td>\n",
       "      <td>196.86585</td>\n",
       "      <td>192.98438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>13308152</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>1.252538e-11</td>\n",
       "      <td>6.312838</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>ESRRA</td>\n",
       "      <td>3.95458</td>\n",
       "      <td>3.75291</td>\n",
       "      <td>13</td>\n",
       "      <td>25999.0</td>\n",
       "      <td>-29.14707</td>\n",
       "      <td>-30.05458</td>\n",
       "      <td>227.12275</td>\n",
       "      <td>220.78075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>13308152</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>1.252538e-11</td>\n",
       "      <td>6.312838</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>RHOXF1</td>\n",
       "      <td>3.59007</td>\n",
       "      <td>3.56901</td>\n",
       "      <td>8</td>\n",
       "      <td>10920.0</td>\n",
       "      <td>-22.44783</td>\n",
       "      <td>-22.51674</td>\n",
       "      <td>83.95005</td>\n",
       "      <td>80.53684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4213</td>\n",
       "      <td>6</td>\n",
       "      <td>4907848</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>2.908110e-08</td>\n",
       "      <td>2.545850</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>ZBTB6</td>\n",
       "      <td>1.48991</td>\n",
       "      <td>1.48991</td>\n",
       "      <td>13</td>\n",
       "      <td>2231.0</td>\n",
       "      <td>-38.22970</td>\n",
       "      <td>-38.22970</td>\n",
       "      <td>218.08801</td>\n",
       "      <td>223.56885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4214</td>\n",
       "      <td>6</td>\n",
       "      <td>4907848</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>2.908110e-08</td>\n",
       "      <td>2.545850</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>Dux</td>\n",
       "      <td>1.46315</td>\n",
       "      <td>1.46315</td>\n",
       "      <td>8</td>\n",
       "      <td>1003.0</td>\n",
       "      <td>-37.49455</td>\n",
       "      <td>-37.49455</td>\n",
       "      <td>58.84941</td>\n",
       "      <td>62.22224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4215</td>\n",
       "      <td>6</td>\n",
       "      <td>4907848</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>2.908110e-08</td>\n",
       "      <td>2.545850</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>DUX4</td>\n",
       "      <td>1.45930</td>\n",
       "      <td>1.45930</td>\n",
       "      <td>11</td>\n",
       "      <td>38221.0</td>\n",
       "      <td>-65.74039</td>\n",
       "      <td>-65.74039</td>\n",
       "      <td>115.53229</td>\n",
       "      <td>120.16992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4216</td>\n",
       "      <td>6</td>\n",
       "      <td>4907848</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>2.908110e-08</td>\n",
       "      <td>2.545850</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>Sox5</td>\n",
       "      <td>1.44444</td>\n",
       "      <td>1.29630</td>\n",
       "      <td>7</td>\n",
       "      <td>27.0</td>\n",
       "      <td>-16.35184</td>\n",
       "      <td>-16.93963</td>\n",
       "      <td>59.56230</td>\n",
       "      <td>61.92574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4217</td>\n",
       "      <td>6</td>\n",
       "      <td>4907848</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>2.908110e-08</td>\n",
       "      <td>2.545850</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>ETV4</td>\n",
       "      <td>1.06747</td>\n",
       "      <td>1.15419</td>\n",
       "      <td>10</td>\n",
       "      <td>8418.0</td>\n",
       "      <td>-39.91008</td>\n",
       "      <td>-39.43708</td>\n",
       "      <td>112.91020</td>\n",
       "      <td>117.59924</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4218 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr       pos      snp_id        pvalue      beta ref_allele alt_allele  \\\n",
       "0       3  13308152  3:13308152  1.252538e-11  6.312838          C          A   \n",
       "1       3  13308152  3:13308152  1.252538e-11  6.312838          C          A   \n",
       "2       3  13308152  3:13308152  1.252538e-11  6.312838          C          A   \n",
       "3       3  13308152  3:13308152  1.252538e-11  6.312838          C          A   \n",
       "4       3  13308152  3:13308152  1.252538e-11  6.312838          C          A   \n",
       "...   ...       ...         ...           ...       ...        ...        ...   \n",
       "4213    6   4907848   6:4907848  2.908110e-08  2.545850          A          G   \n",
       "4214    6   4907848   6:4907848  2.908110e-08  2.545850          A          G   \n",
       "4215    6   4907848   6:4907848  2.908110e-08  2.545850          A          G   \n",
       "4216    6   4907848   6:4907848  2.908110e-08  2.545850          A          G   \n",
       "4217    6   4907848   6:4907848  2.908110e-08  2.545850          A          G   \n",
       "\n",
       "     trait_num        TF_name PWM_fracscore_ref PWM_fracscore_alt  TF_len  \\\n",
       "0            6  TFAP2A(var.2)           4.92927           5.30035      12   \n",
       "1            6          BARX2           3.84562           4.58265      12   \n",
       "2            6           PAX5           4.02331           4.07698      12   \n",
       "3            6          ESRRA           3.95458           3.75291      13   \n",
       "4            6         RHOXF1           3.59007           3.56901       8   \n",
       "...        ...            ...               ...               ...     ...   \n",
       "4213         8          ZBTB6           1.48991           1.48991      13   \n",
       "4214         8            Dux           1.46315           1.46315       8   \n",
       "4215         8           DUX4           1.45930           1.45930      11   \n",
       "4216         8           Sox5           1.44444           1.29630       7   \n",
       "4217         8           ETV4           1.06747           1.15419      10   \n",
       "\n",
       "      TF_count PWM_ln_fracscore_ref PWM_ln_fracscore_alt    H (REF)  \\\n",
       "0       4009.0            -18.36669            -16.88040  213.84023   \n",
       "1      28391.0            -32.67471            -26.33297  194.51138   \n",
       "2       1416.0            -25.29941            -24.16443  196.86585   \n",
       "3      25999.0            -29.14707            -30.05458  227.12275   \n",
       "4      10920.0            -22.44783            -22.51674   83.95005   \n",
       "...        ...                  ...                  ...        ...   \n",
       "4213    2231.0            -38.22970            -38.22970  218.08801   \n",
       "4214    1003.0            -37.49455            -37.49455   58.84941   \n",
       "4215   38221.0            -65.74039            -65.74039  115.53229   \n",
       "4216      27.0            -16.35184            -16.93963   59.56230   \n",
       "4217    8418.0            -39.91008            -39.43708  112.91020   \n",
       "\n",
       "     Hprime (ALT)  \n",
       "0       210.31008  \n",
       "1       195.83667  \n",
       "2       192.98438  \n",
       "3       220.78075  \n",
       "4        80.53684  \n",
       "...           ...  \n",
       "4213    223.56885  \n",
       "4214     62.22224  \n",
       "4215    120.16992  \n",
       "4216     61.92574  \n",
       "4217    117.59924  \n",
       "\n",
       "[4218 rows x 17 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomesize_file = open('/local3/jake/TF_binding/genomesize.all_chrms.hg19.txt','w')\n",
    "for row,c in bgfrac_df.iterrows():\n",
    "    genomesize_file.write('chr{0}\\t{1}\\n'.format(int(c['Chrm']),int(c['total'])))\n",
    "\n",
    "genomesize_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_full_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(chromosome,ref_pos_start,ref_pos_end),'/local3/jake/TF_binding/chr8.fa')\n",
    "tfname_list = list(set(final_df['TF_name']))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "decorated() missing 1 required positional argument: 'self'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-51-c28ff6094562>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# pybedtools.BedTool.random(genome='/local3/jake/TF_binding/genomesize.all_chrms.hg19.txt',l=10, n=10)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;31m# x = pybedtools.BedTool()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0my\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpybedtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mBedTool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrandom\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ml\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenome\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'hg19'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: decorated() missing 1 required positional argument: 'self'"
     ]
    }
   ],
   "source": [
    "# pybedtools.BedTool.random(genome='/local3/jake/TF_binding/genomesize.all_chrms.hg19.txt',l=10, n=10)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def get_random_bgseqs(slen,reps,fastaloc):\n",
    "    x = pybedtools.BedTool()\n",
    "    \n",
    "    bgseq_list = []\n",
    "    nreps = 0\n",
    "    while(len(bgseq_list) < reps):\n",
    "        nreps += 1\n",
    "        y = x.random(l=slen, n=1, genome='hg19')\n",
    "        for s in y:\n",
    "            try:\n",
    "#                 curr_chrm = int(s[0].split('r')[1])\n",
    "                curr_chrm = 8\n",
    "                curr_start = int(s[1])\n",
    "                curr_end = int(s[2])\n",
    "                chrmfile = '{0}/chr{1}.fa'.format(fastaloc,curr_chrm)\n",
    "                bgseq_list.append({'chrm':curr_chrm,'start':curr_start,'end':curr_end,'seq':'AAAC'})\n",
    "            except:\n",
    "                continue\n",
    "#         try:\n",
    "#             curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(curr_chrm,curr_start,curr_end),chrmfile)\n",
    "#             bgseq_list.append({'chrm':curr_chrm,'start':curr_start,'end':curr_end,'seq':curr_seq.upper()})\n",
    "#         except:\n",
    "#             print('Could not find fasta file! Check the location and try again.')\n",
    "#             return -1\n",
    "        \n",
    "    return bgseq_list\n",
    "\n",
    "def get_random_bgseqs_v2(slen,reps,chrm,bgfrac_df,chrmfile):\n",
    "    csize = bgfrac_df.loc[bgfrac_df['Chrm'] == chrm]['total'].values[0]\n",
    "    bgseq_list = []\n",
    "    for s in range(reps):\n",
    "        curr_start = random.randint(0,(csize-slen))\n",
    "        curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(chrm,curr_start,(curr_start+slen)),chrmfile)\n",
    "        bgseq_list.append({'start':curr_start,'end':(curr_start+slen),'seq':curr_seq.upper()}) \n",
    "    return bgseq_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'chrm': 8, 'start': 158615095, 'end': 158615105, 'seq': 'TEST'},\n",
       " {'chrm': 8, 'start': 146653718, 'end': 146653728, 'seq': 'TEST'},\n",
       " {'chrm': 8, 'start': 128834204, 'end': 128834214, 'seq': 'TEST'},\n",
       " {'chrm': 8, 'start': 111433622, 'end': 111433632, 'seq': 'TEST'},\n",
       " {'chrm': 8, 'start': 44903868, 'end': 44903878, 'seq': 'TEST'}]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_random_bgseqs(10,5,'/local3/jake/TF_binding')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_bgH(seq,ln_pwm,bgfreqs):\n",
    "    currscore_ln = get_matrix_scores(ln_pwm,seq)\n",
    "    Y = compute_Y(seq,bgfreqs)\n",
    "    H = currscore_ln - Y\n",
    "    return np.sum(H)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "bgseqs = get_random_bgseqs(10,5,'/local3/jake/TF_binding')\n",
    "bg_H_list = []\n",
    "bg_H_by_TF = {}\n",
    "for s in bgseqs:\n",
    "    curr_seq = s['seq']\n",
    "    curr_matrix = get_matrix_byTF('TEF',infodicts_list)\n",
    "    bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == '8'][['frac_A','frac_C','frac_G','frac_T']]\n",
    "    curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "    curr_lnfracPWM = get_lnPWM_from_fracPWM(curr_fracPWM,bgfreqs)\n",
    "    curr_H = np.sum(get_matrix_scores(curr_lnfracPWM,curr_seq))\n",
    "    bg_H_list.append(curr_H)\n",
    "#     bg_H_list.append()\n",
    "bg_H_by_TF['TEST_TF'] = bg_H_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TEST_TF 46.56055507754986\n"
     ]
    }
   ],
   "source": [
    "for tf,h in bg_H_by_TF.items():\n",
    "    print(tf,np.sum(h))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "too many values to unpack (expected 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-111-9de1fd5dabe2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mcurr_matrix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_byTF\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfodicts_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m#     print(curr_matrix,curr_seq)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m     \u001b[0mcurr_refscore\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcurr_altscore\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_scores\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtfname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmatrix_dict\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcurr_matrix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mref_al\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'T'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0malt_al\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'G'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mposition\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcurr_relative_pos\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mreference_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcurr_seq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0;31m#     print(g['tf_name'],curr_refscore,curr_altscore)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;31m#     print((g['pos_start']-ref_pos_start),(g['pos_end']-ref_pos_start),\"\"(position-ref_pos_start))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: too many values to unpack (expected 2)"
     ]
    }
   ],
   "source": [
    "for i,g in gene_df.iterrows():\n",
    "    curr_relative_pos = abs(updated_pos-g['relative_start'])\n",
    "    curr_seq = ref_full_seq[g['relative_start']:(g['relative_end']+1)]\n",
    "    \n",
    "    curr_matrix = get_matrix_byTF(g['tf_name'],infodicts_list)\n",
    "#     print(curr_matrix,curr_seq)\n",
    "    curr_refscore,curr_altscore = get_matrix_scores(tfname=g['tf_name'],matrix_dict=curr_matrix,ref_al='T',alt_al='G',position=curr_relative_pos,reference_seq=curr_seq)\n",
    "#     print(g['tf_name'],curr_refscore,curr_altscore)\n",
    "#     print((g['pos_start']-ref_pos_start),(g['pos_end']-ref_pos_start),\"\"(position-ref_pos_start))\n",
    "#     print(g['pos_start'],g['pos_end'],position)\n",
    "#     print(g['relative_start'],g['relative_end'],curr_relative_pos,updated_pos)\n",
    "    print(curr_seq,curr_seq[curr_relative_pos])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OTX2 {'ref_score': 264488.0, 'alt_score': 148783.0}\n",
      "KLF5 {'ref_score': 72298.0, 'alt_score': 72429.0}\n",
      "NR2C2(var.2) {'ref_score': 28556.0, 'alt_score': 47184.0}\n",
      "PITX1 {'ref_score': 41886.0, 'alt_score': 31355.0}\n",
      "RHOXF1 {'ref_score': 40212.0, 'alt_score': 29296.0}\n",
      "NR2C1 {'ref_score': 16249.0, 'alt_score': 25871.0}\n",
      "PITX3 {'ref_score': 27003.0, 'alt_score': 20765.0}\n",
      "THRB {'ref_score': 14938.0, 'alt_score': 15644.0}\n",
      "HOXB6 {'ref_score': 12451.0, 'alt_score': 13489.0}\n",
      "GSC2 {'ref_score': 8967.0, 'alt_score': 6951.0}\n",
      "RARA::RXRG {'ref_score': 3092.0, 'alt_score': 4033.0}\n",
      "VDR {'ref_score': 1707.0, 'alt_score': 1538.0}\n",
      "SOX18 {'ref_score': 1374.0, 'alt_score': 925.0}\n",
      "Dmbx1 {'ref_score': 396.0, 'alt_score': 405.0}\n",
      "MZF1(var.2) {'ref_score': 17.0, 'alt_score': 28.0}\n"
     ]
    }
   ],
   "source": [
    "for tf,scores in sorted(score_dict_bytf.items(), key=lambda k_v: k_v[1]['alt_score'],reverse=True):\n",
    "    print(tf,scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "results_loc = '/local3/jake/TF_binding/tophit_SNPs/output/'\n",
    "results_files_list = os.listdir(results_loc)\n",
    "results_files_bytraitcode = {}\n",
    "for f in results_files_list:\n",
    "    if(f.split('.')[-1] != 'PWM_scores'):\n",
    "        continue\n",
    "    traitcode = int(f.split('.')[0]) #\".\".join(f.split('.')[:2])\n",
    "    if(traitcode not in results_files_bytraitcode.keys()):\n",
    "        results_files_bytraitcode[traitcode] = []\n",
    "    results_files_bytraitcode[traitcode].append(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrm</th>\n",
       "      <th>count_A</th>\n",
       "      <th>count_C</th>\n",
       "      <th>count_G</th>\n",
       "      <th>count_T</th>\n",
       "      <th>total</th>\n",
       "      <th>frac_A</th>\n",
       "      <th>frac_C</th>\n",
       "      <th>frac_G</th>\n",
       "      <th>frac_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65570891</td>\n",
       "      <td>47024412</td>\n",
       "      <td>47016562</td>\n",
       "      <td>65668756</td>\n",
       "      <td>225280621.0</td>\n",
       "      <td>0.291063</td>\n",
       "      <td>0.208737</td>\n",
       "      <td>0.208702</td>\n",
       "      <td>0.291498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>71102632</td>\n",
       "      <td>47915465</td>\n",
       "      <td>47947042</td>\n",
       "      <td>71239379</td>\n",
       "      <td>238204518.0</td>\n",
       "      <td>0.298494</td>\n",
       "      <td>0.201153</td>\n",
       "      <td>0.201285</td>\n",
       "      <td>0.299068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>58713343</td>\n",
       "      <td>38653197</td>\n",
       "      <td>38670110</td>\n",
       "      <td>58760485</td>\n",
       "      <td>194797135.0</td>\n",
       "      <td>0.301408</td>\n",
       "      <td>0.198428</td>\n",
       "      <td>0.198515</td>\n",
       "      <td>0.301650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>57932980</td>\n",
       "      <td>35885806</td>\n",
       "      <td>35890822</td>\n",
       "      <td>57952068</td>\n",
       "      <td>187661676.0</td>\n",
       "      <td>0.308710</td>\n",
       "      <td>0.191226</td>\n",
       "      <td>0.191253</td>\n",
       "      <td>0.308811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>53672554</td>\n",
       "      <td>35089383</td>\n",
       "      <td>35129186</td>\n",
       "      <td>53804137</td>\n",
       "      <td>177695260.0</td>\n",
       "      <td>0.302048</td>\n",
       "      <td>0.197469</td>\n",
       "      <td>0.197693</td>\n",
       "      <td>0.302789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>50554433</td>\n",
       "      <td>33143287</td>\n",
       "      <td>33163423</td>\n",
       "      <td>50533923</td>\n",
       "      <td>167395066.0</td>\n",
       "      <td>0.302007</td>\n",
       "      <td>0.197994</td>\n",
       "      <td>0.198115</td>\n",
       "      <td>0.301884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>45997757</td>\n",
       "      <td>31671670</td>\n",
       "      <td>31636979</td>\n",
       "      <td>46047257</td>\n",
       "      <td>155353663.0</td>\n",
       "      <td>0.296084</td>\n",
       "      <td>0.203868</td>\n",
       "      <td>0.203645</td>\n",
       "      <td>0.296403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>42767293</td>\n",
       "      <td>28703983</td>\n",
       "      <td>28702621</td>\n",
       "      <td>42715025</td>\n",
       "      <td>142888922.0</td>\n",
       "      <td>0.299304</td>\n",
       "      <td>0.200883</td>\n",
       "      <td>0.200874</td>\n",
       "      <td>0.298939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>35260078</td>\n",
       "      <td>24826212</td>\n",
       "      <td>24813259</td>\n",
       "      <td>35243882</td>\n",
       "      <td>120143431.0</td>\n",
       "      <td>0.293483</td>\n",
       "      <td>0.206638</td>\n",
       "      <td>0.206530</td>\n",
       "      <td>0.293348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>38330752</td>\n",
       "      <td>27308648</td>\n",
       "      <td>27298423</td>\n",
       "      <td>38376915</td>\n",
       "      <td>131314738.0</td>\n",
       "      <td>0.291900</td>\n",
       "      <td>0.207963</td>\n",
       "      <td>0.207885</td>\n",
       "      <td>0.292251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>38307244</td>\n",
       "      <td>27236798</td>\n",
       "      <td>27268038</td>\n",
       "      <td>38317436</td>\n",
       "      <td>131129516.0</td>\n",
       "      <td>0.292133</td>\n",
       "      <td>0.207709</td>\n",
       "      <td>0.207947</td>\n",
       "      <td>0.292211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>38604831</td>\n",
       "      <td>26634995</td>\n",
       "      <td>26617050</td>\n",
       "      <td>38624517</td>\n",
       "      <td>130481393.0</td>\n",
       "      <td>0.295865</td>\n",
       "      <td>0.204129</td>\n",
       "      <td>0.203991</td>\n",
       "      <td>0.296016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>29336945</td>\n",
       "      <td>18412698</td>\n",
       "      <td>18414776</td>\n",
       "      <td>29425459</td>\n",
       "      <td>95589878.0</td>\n",
       "      <td>0.306904</td>\n",
       "      <td>0.192622</td>\n",
       "      <td>0.192644</td>\n",
       "      <td>0.307830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>25992966</td>\n",
       "      <td>18027132</td>\n",
       "      <td>18071947</td>\n",
       "      <td>26197495</td>\n",
       "      <td>88289540.0</td>\n",
       "      <td>0.294406</td>\n",
       "      <td>0.204182</td>\n",
       "      <td>0.204690</td>\n",
       "      <td>0.296723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>23620876</td>\n",
       "      <td>17247582</td>\n",
       "      <td>17228387</td>\n",
       "      <td>23597921</td>\n",
       "      <td>81694766.0</td>\n",
       "      <td>0.289136</td>\n",
       "      <td>0.211122</td>\n",
       "      <td>0.210887</td>\n",
       "      <td>0.288855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>21724083</td>\n",
       "      <td>17630040</td>\n",
       "      <td>17701988</td>\n",
       "      <td>21828642</td>\n",
       "      <td>78884753.0</td>\n",
       "      <td>0.275390</td>\n",
       "      <td>0.223491</td>\n",
       "      <td>0.224403</td>\n",
       "      <td>0.276716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>21159933</td>\n",
       "      <td>17727956</td>\n",
       "      <td>17700340</td>\n",
       "      <td>21206981</td>\n",
       "      <td>77795210.0</td>\n",
       "      <td>0.271995</td>\n",
       "      <td>0.227880</td>\n",
       "      <td>0.227525</td>\n",
       "      <td>0.272600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>22465380</td>\n",
       "      <td>14838685</td>\n",
       "      <td>14863671</td>\n",
       "      <td>22489493</td>\n",
       "      <td>74657229.0</td>\n",
       "      <td>0.300914</td>\n",
       "      <td>0.198758</td>\n",
       "      <td>0.199092</td>\n",
       "      <td>0.301237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>14390632</td>\n",
       "      <td>13478255</td>\n",
       "      <td>13511145</td>\n",
       "      <td>14428951</td>\n",
       "      <td>55808983.0</td>\n",
       "      <td>0.257855</td>\n",
       "      <td>0.241507</td>\n",
       "      <td>0.242096</td>\n",
       "      <td>0.258542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>20</td>\n",
       "      <td>16523053</td>\n",
       "      <td>13107828</td>\n",
       "      <td>13149412</td>\n",
       "      <td>16725227</td>\n",
       "      <td>59505520.0</td>\n",
       "      <td>0.277673</td>\n",
       "      <td>0.220279</td>\n",
       "      <td>0.220978</td>\n",
       "      <td>0.281070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>10422924</td>\n",
       "      <td>7160212</td>\n",
       "      <td>7174721</td>\n",
       "      <td>10348785</td>\n",
       "      <td>35106642.0</td>\n",
       "      <td>0.296893</td>\n",
       "      <td>0.203956</td>\n",
       "      <td>0.204369</td>\n",
       "      <td>0.294781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "      <td>9094775</td>\n",
       "      <td>8375984</td>\n",
       "      <td>8369235</td>\n",
       "      <td>9054551</td>\n",
       "      <td>34894545.0</td>\n",
       "      <td>0.260636</td>\n",
       "      <td>0.240037</td>\n",
       "      <td>0.239844</td>\n",
       "      <td>0.259483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chrm   count_A   count_C   count_G   count_T        total    frac_A  \\\n",
       "0      1  65570891  47024412  47016562  65668756  225280621.0  0.291063   \n",
       "1      2  71102632  47915465  47947042  71239379  238204518.0  0.298494   \n",
       "2      3  58713343  38653197  38670110  58760485  194797135.0  0.301408   \n",
       "3      4  57932980  35885806  35890822  57952068  187661676.0  0.308710   \n",
       "4      5  53672554  35089383  35129186  53804137  177695260.0  0.302048   \n",
       "5      6  50554433  33143287  33163423  50533923  167395066.0  0.302007   \n",
       "6      7  45997757  31671670  31636979  46047257  155353663.0  0.296084   \n",
       "7      8  42767293  28703983  28702621  42715025  142888922.0  0.299304   \n",
       "8      9  35260078  24826212  24813259  35243882  120143431.0  0.293483   \n",
       "9     10  38330752  27308648  27298423  38376915  131314738.0  0.291900   \n",
       "10    11  38307244  27236798  27268038  38317436  131129516.0  0.292133   \n",
       "11    12  38604831  26634995  26617050  38624517  130481393.0  0.295865   \n",
       "12    13  29336945  18412698  18414776  29425459   95589878.0  0.306904   \n",
       "13    14  25992966  18027132  18071947  26197495   88289540.0  0.294406   \n",
       "14    15  23620876  17247582  17228387  23597921   81694766.0  0.289136   \n",
       "15    16  21724083  17630040  17701988  21828642   78884753.0  0.275390   \n",
       "16    17  21159933  17727956  17700340  21206981   77795210.0  0.271995   \n",
       "17    18  22465380  14838685  14863671  22489493   74657229.0  0.300914   \n",
       "18    19  14390632  13478255  13511145  14428951   55808983.0  0.257855   \n",
       "19    20  16523053  13107828  13149412  16725227   59505520.0  0.277673   \n",
       "20    21  10422924   7160212   7174721  10348785   35106642.0  0.296893   \n",
       "21    22   9094775   8375984   8369235   9054551   34894545.0  0.260636   \n",
       "\n",
       "      frac_C    frac_G    frac_T  \n",
       "0   0.208737  0.208702  0.291498  \n",
       "1   0.201153  0.201285  0.299068  \n",
       "2   0.198428  0.198515  0.301650  \n",
       "3   0.191226  0.191253  0.308811  \n",
       "4   0.197469  0.197693  0.302789  \n",
       "5   0.197994  0.198115  0.301884  \n",
       "6   0.203868  0.203645  0.296403  \n",
       "7   0.200883  0.200874  0.298939  \n",
       "8   0.206638  0.206530  0.293348  \n",
       "9   0.207963  0.207885  0.292251  \n",
       "10  0.207709  0.207947  0.292211  \n",
       "11  0.204129  0.203991  0.296016  \n",
       "12  0.192622  0.192644  0.307830  \n",
       "13  0.204182  0.204690  0.296723  \n",
       "14  0.211122  0.210887  0.288855  \n",
       "15  0.223491  0.224403  0.276716  \n",
       "16  0.227880  0.227525  0.272600  \n",
       "17  0.198758  0.199092  0.301237  \n",
       "18  0.241507  0.242096  0.258542  \n",
       "19  0.220279  0.220978  0.281070  \n",
       "20  0.203956  0.204369  0.294781  \n",
       "21  0.240037  0.239844  0.259483  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bgfrac_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
