{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "from pandas import *\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "from matplotlib import pyplot\n",
    "import random\n",
    "from scipy.stats import norm\n",
    "import scipy.stats as scs\n",
    "import os\n",
    "import pybedtools\n",
    "import math\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.mlab as mlab\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_JASPAR_transfac_pfms(infile):\n",
    "    pfms_file = open(infile,'r')\n",
    "    pfms_info = {}\n",
    "    seq_len = 0\n",
    "    for line in pfms_file:\n",
    "        line = line.split('\\n')[0]\n",
    "        if(len(line.split(' ')) > 1):\n",
    "            line = line.split(' ')\n",
    "            if(line[0] == 'DE'):\n",
    "                pfms_info['Matrix_ID'] = line[1]\n",
    "                pfms_info['Matrix_Name'] = line[2]\n",
    "                seq_len = 0\n",
    "            elif(line[0] == 'CC'):\n",
    "                temp = line[1].split(':')\n",
    "                pfms_info[temp[0]] = temp[1]\n",
    "                if(seq_len > 0):\n",
    "                    pfms_info['TF_len'] = seq_len\n",
    "        elif(len(line.split('\\t')) > 1):\n",
    "            line = line.split('\\t')\n",
    "            if(line[0] == 'PO'):\n",
    "                curr_matorder = line[1:]\n",
    "            else:\n",
    "                curr_vals = {}\n",
    "                for n,v in enumerate(line[1:]):\n",
    "                    curr_vals[curr_matorder[n]] = float(v)+1\n",
    "                pfms_info[int(line[0])] = curr_vals\n",
    "                seq_len = int(line[0])\n",
    "        else:\n",
    "            pass\n",
    "#             print(line)\n",
    "    pfms_file.close()\n",
    "    return pfms_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_matrix_byTFName(tfname,info_dicts,seq): #old\n",
    "    seq = seq.upper()\n",
    "    matrix_dict_touse = None\n",
    "    for i in info_dicts:\n",
    "        if(i['Matrix_Name'] == tfname):\n",
    "            matrix_dict_touse = i\n",
    "            break\n",
    "    if(matrix_dict_touse == None):\n",
    "        print('Could not find a PWM for Transcription Factor {0}'.format(tfname))\n",
    "        return None\n",
    "    seq_val = 0\n",
    "    for n,b in enumerate(seq):\n",
    "        try:\n",
    "            seq_val += float(matrix_dict_touse[n+1][b])\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "            else:\n",
    "                print('Sequence of length {0} is too long'.format(len(seq)))\n",
    "            return None\n",
    "    return seq_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_matrix_byTF(tfname,info_dicts):\n",
    "    matrix_dict_touse = None\n",
    "    for i in info_dicts:\n",
    "        if(i['Matrix_Name'] == tfname):\n",
    "            matrix_dict_touse = i\n",
    "            break\n",
    "    if(matrix_dict_touse == None):\n",
    "        print('Could not find a PWM for Transcription Factor {0}'.format(tfname))\n",
    "        return None\n",
    "    return matrix_dict_touse\n",
    "\n",
    "def get_lnPWM_from_matrixdict(matrix_dict):\n",
    "    lnPWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        for b in 'ACTG':\n",
    "            f = float(matrix_dict[en][b])\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = np.log(1)\n",
    "            else:\n",
    "                temp_matrix[b] = np.log(f)\n",
    "        lnPWM_dict[en] = temp_matrix\n",
    "    return lnPWM_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fraclnPWM_from_matrixdict(matrix_dict):\n",
    "    lnPWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in matrix_dict[en].values()])\n",
    "        for b in 'ACTG':\n",
    "            f = float(matrix_dict[en][b])/curr_totcount\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = np.log(1)\n",
    "            else:\n",
    "                temp_matrix[b] = np.log(f)\n",
    "        lnPWM_dict[en] = temp_matrix\n",
    "    return lnPWM_dict\n",
    "\n",
    "def get_fracPWM_from_matrixdict(matrix_dict):\n",
    "    PWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in matrix_dict[en].values()])\n",
    "        for b in 'ACTG':\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = 1/curr_totcount\n",
    "            else:\n",
    "                temp_matrix[b] = float(matrix_dict[en][b])/curr_totcount\n",
    "        PWM_dict[en] = temp_matrix\n",
    "    return PWM_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def get_frac_matrix_scores(tfname,pwm,ref_al,alt_al,position,reference_seq):\n",
    "    ref_al = ref_al.upper()\n",
    "    alt_al = alt_al.upper()\n",
    "    reference_seq = reference_seq.upper()\n",
    "    ref_seqval_list,alt_seqval_list = [],[]\n",
    "    for n,b in enumerate(reference_seq):\n",
    "        try:\n",
    "            if(n == position):\n",
    "                ref_seqval_list.append(float(pwm[n+1][ref_al]))\n",
    "                alt_seqval_list.append(float(pwm[n+1][alt_al]))\n",
    "            else:\n",
    "                ref_seqval_list.append(float(pwm[n+1][b]))\n",
    "                alt_seqval_list.append(float(pwm[n+1][b]))\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                continue\n",
    "\n",
    "    return ref_seqval_list,alt_seqval_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_lnmatrix_scores(tfname,matrix_dict,ref_al,alt_al,position,reference_seq):\n",
    "    ref_al = ref_al.upper()\n",
    "    alt_al = alt_al.upper()\n",
    "    reference_seq = reference_seq.upper()\n",
    "    ref_seqval,alt_seqval = 0,0\n",
    "    seqpos_counts = []\n",
    "    seqpos_counts_bybase = {'A':[],'C':[],'G':[],'T':[],}\n",
    "    for n,b in enumerate(reference_seq):\n",
    "        try:\n",
    "            if(n == position):\n",
    "                ref_seqval += float(matrix_dict[n+1][ref_al])\n",
    "                alt_seqval += float(matrix_dict[n+1][alt_al])\n",
    "            else:\n",
    "                ref_seqval += float(matrix_dict[n+1][b])\n",
    "                alt_seqval += float(matrix_dict[n+1][b])\n",
    "            curr_vals = [float(x) for x in matrix_dict[n+1].values()]\n",
    "            seqpos_counts.append(sum(curr_vals))\n",
    "            seqpos_counts_bybase['A'].append(matrix_dict[n+1]['A'])\n",
    "            seqpos_counts_bybase['C'].append(matrix_dict[n+1]['C'])\n",
    "            seqpos_counts_bybase['G'].append(matrix_dict[n+1]['G'])\n",
    "            seqpos_counts_bybase['T'].append(matrix_dict[n+1]['T'])\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                continue\n",
    "        \n",
    "#     total_count_fraction = seqpos_counts/len(seqpos_counts)\n",
    "    refseq_fraction = ref_seqval/sum(seqpos_counts)\n",
    "    altseq_fraction = alt_seqval/sum(seqpos_counts)\n",
    "    return ref_seqval,alt_seqval,seqpos_counts_bybase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_numbp_forseq(seq):\n",
    "    bpcounts = {'A':0,'C':0,'G':0,'T':0}\n",
    "    for b in seq:\n",
    "        try:\n",
    "            bpcounts[b] += 1\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T!'.format(b))\n",
    "    bpc_list = []\n",
    "    for o in 'ACGT':\n",
    "        bpc_list.append(bpcounts[o])\n",
    "    return bpc_list\n",
    "            \n",
    "def compute_Y(seq,bg_props):\n",
    "#     seq_bpcounts = get_numbp_forseq(seq)\n",
    "    bpcounts = {'A':0,'C':0,'G':0,'T':0}\n",
    "    for b in seq:\n",
    "        try:\n",
    "            bpcounts[b] += 1\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T!'.format(b))\n",
    "    bpc_list = []\n",
    "    for o in 'ACGT':\n",
    "        bpc_list.append(bpcounts[o])\n",
    "    Y = np.dot(np.log(bg_props),bpc_list)\n",
    "    return Y\n",
    "\n",
    "def make_seq(fullseq,position,allele):\n",
    "    new_seq = ''\n",
    "    for n,b in enumerate(fullseq):\n",
    "        try:\n",
    "            if(n==position):\n",
    "                new_seq = ''.join([new_seq,allele])\n",
    "            else:\n",
    "                new_seq = ''.join([new_seq,b])\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                print(b)\n",
    "                continue\n",
    "    return new_seq.upper()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def get_matrix_counts(pwm):\n",
    "    pos_counts = []\n",
    "    for en in pwm:\n",
    "        temp = [float(x) for x in en.values()]\n",
    "        pos_counts.append(sum(temp))\n",
    "    return pos_counts\n",
    "def get_complseq(seq):\n",
    "    new_seq = []\n",
    "    for b in seq:\n",
    "        if(b == 'A'):\n",
    "            new_seq.append('T')\n",
    "        elif(b == 'T'):\n",
    "            new_seq.append('A')\n",
    "        elif(b == 'G'):\n",
    "            new_seq.append('C')\n",
    "        elif(b == 'C'):\n",
    "            new_seq.append('G')\n",
    "        else:\n",
    "            print('Base pair not A/C/T/G! {0}'.format(b))\n",
    "    return ''.join(new_seq)\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_random_bgseqs(slen,reps,fastaloc,chrmsizes):\n",
    "    # x = pybedtools.BedTool()\n",
    "    chrms_touse = list(chrmsizes['chrom'])\n",
    "    bgseq_list = []\n",
    "    nrep = 0\n",
    "    while(len(bgseq_list) < int(reps)):\n",
    "#     for n in range(int(reps)):\n",
    "        is_valid_seq = True\n",
    "        curr_chrm = 8\n",
    "        curr_start = random.randint(1,chrmsizes.loc[chrmsizes['chrom'] == 'chr{0}'.format(curr_chrm)]['size'].values[0])\n",
    "        curr_end = curr_start + slen\n",
    "        chrmfile = '{0}/chr{1}.fa'.format(fastaloc,curr_chrm)\n",
    "        curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(curr_chrm,curr_start,curr_end),chrmfile)\n",
    "        for b in curr_seq:\n",
    "            if(b not in 'ACTG'):\n",
    "                is_valid_seq = False\n",
    "                continue\n",
    "        if(is_valid_seq):\n",
    "            bgseq_list.append({'chrm':curr_chrm,'start':curr_start,'end':curr_end,'seq':curr_seq.upper()})\n",
    "    return bgseq_list\n",
    "\n",
    "def get_random_bgseqs_v2(slen,reps,chrm,bgfrac_df,chrmfile):\n",
    "    csize = bgfrac_df.loc[bgfrac_df['Chrm'] == chrm]['total'].values[0]\n",
    "    bgseq_list = []\n",
    "    for s in range(reps):\n",
    "        curr_start = random.randint(0,(csize-slen))\n",
    "        curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(chrm,curr_start,(curr_start+slen)),chrmfile)\n",
    "        bgseq_list.append({'start':curr_start,'end':(curr_start+slen),'seq':curr_seq.upper()}) \n",
    "    return bgseq_list\n",
    "\n",
    "def generate_Z(H_list):\n",
    "    for tf,h in bg_H_by_TF.items():\n",
    "        curr_z = [pow(math.e,-x) for x in h]\n",
    "        bg_Z_byTF[tf] = sum(curr_z)\n",
    "    return\n",
    "def get_lnPWM_from_fracPWM(fracPWM,bgfreqs):\n",
    "    lnPWM_dict = {}\n",
    "    for en in range(1,len(fracPWM)+1):\n",
    "        temp_matrix = {}\n",
    "        for b in 'ACTG':\n",
    "            f = float(fracPWM[en][b])/bgfreqs['frac_{0}'.format(b)].values[0]\n",
    "            if(f == 0.0):\n",
    "                print('fraction is 0')\n",
    "                temp_matrix[b] = np.log(1)\n",
    "            else:\n",
    "                temp_matrix[b] = -np.log(f)\n",
    "        lnPWM_dict[en] = temp_matrix\n",
    "    return lnPWM_dict\n",
    "\n",
    "def get_fracPWM_from_matrixdict(matrix_dict):\n",
    "    PWM_dict = {}\n",
    "    for en in range(1,matrix_dict['TF_len']+1):\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in matrix_dict[en].values()])\n",
    "        for b in 'ACTG':\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = 1/curr_totcount\n",
    "            else:\n",
    "                temp_matrix[b] = float(matrix_dict[en][b])/curr_totcount\n",
    "        PWM_dict[en] = temp_matrix\n",
    "    return PWM_dict\n",
    "\n",
    "def calculate_bgH(seq,ln_pwm,bgfreqs):\n",
    "    currscore_ln = get_matrix_scores(ln_pwm,seq)\n",
    "    Y = compute_Y(seq,bgfreqs)\n",
    "    H = currscore_ln - Y\n",
    "    return np.sum(H)\n",
    "\n",
    "def calculate_bindingP(hscore,bg_z_df,tf_name):\n",
    "    try:\n",
    "        bgZ_touse = bg_z_df.loc[bg_z_df['TF_name'] == tf_name]['BG Z score'].values[0]\n",
    "    except:\n",
    "        print('Could not find background Z score for TF {0}!'.format(tf_name))\n",
    "        return -1\n",
    "    return math.exp(-hscore)/bgZ_touse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_Y(seq,bg_props):\n",
    "#     seq_bpcounts = get_numbp_forseq(seq)\n",
    "    bpcounts = {'A':0,'C':0,'G':0,'T':0}\n",
    "    for b in seq:\n",
    "        try:\n",
    "            bpcounts[b] += 1\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T!'.format(b))\n",
    "    bpc_list = []\n",
    "    for o in 'ACGT':\n",
    "        bpc_list.append(bpcounts[o])\n",
    "    Y = np.dot(np.log(bg_props),bpc_list)\n",
    "    return Y\n",
    "def get_matrix_byTF(tfname,info_dicts):\n",
    "    matrix_dict_touse = None\n",
    "    for i in info_dicts:\n",
    "        if(i['Matrix_Name'] == tfname):\n",
    "            matrix_dict_touse = i\n",
    "            break\n",
    "    if(matrix_dict_touse == None):\n",
    "        print('Could not find a PWM for Transcription Factor {0}'.format(tfname))\n",
    "        return None\n",
    "    matrix_aslist = []\n",
    "    for i in range(1,matrix_dict_touse['TF_len']+1):\n",
    "        matrix_aslist.append(matrix_dict_touse[i])\n",
    "    return matrix_aslist\n",
    "\n",
    "def get_fracPWM_from_matrix(pwm):\n",
    "    fracPWM = []\n",
    "    for en in pwm:\n",
    "        temp_matrix = {}\n",
    "        curr_totcount = sum([float(x) for x in en.values()])\n",
    "        for b in 'ACTG':\n",
    "            if(f == 0.0):\n",
    "                temp_matrix[b] = 1/curr_totcount\n",
    "            else:\n",
    "                temp_matrix[b] = float(en[b])/curr_totcount\n",
    "        fracPWM.append(temp_matrix)\n",
    "    return fracPWM\n",
    "def get_lnPWM_from_fracPWM(fracPWM,bgfreqs):\n",
    "    lnPWM = []\n",
    "    bgfreqs_dict = {x:bgfreqs['frac_{0}'.format(x)].values[0] for x in 'ACTG'}\n",
    "    for en in fracPWM:\n",
    "        temp_matrix = {}\n",
    "        for b in 'ACTG':\n",
    "            f = float(en[b])/bgfreqs_dict[b]\n",
    "            temp_matrix[b] = -np.log(f)\n",
    "        lnPWM.append(temp_matrix)\n",
    "    return lnPWM\n",
    "\n",
    "def get_matrix_scores(pwm,seq):\n",
    "    seqval_list = []\n",
    "    for n,b in enumerate(seq):\n",
    "        try:\n",
    "            seqval_list.append(float(pwm[n][b]))\n",
    "        except:\n",
    "            if(b not in 'ACTG'):\n",
    "                print('Sequence contains a letter, {0}, that is not A/C/G/T at position {1}'.format(b,n))\n",
    "                return None\n",
    "            else:\n",
    "                continue\n",
    "    return seqval_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_rev_seq(seq):\n",
    "    new_seq = ''\n",
    "    for b in reversed(seq):\n",
    "        new_seq = ''.join([new_seq,b])\n",
    "    return new_seq\n",
    "\n",
    "def get_seq_combos(seq,al):\n",
    "    seq_list = []\n",
    "    compl_seq_list = []\n",
    "    for pos in range(len(seq)):\n",
    "        temp_seq = ''\n",
    "        for n,b in enumerate(seq):\n",
    "            if(n == pos):\n",
    "                temp_seq = ''.join([temp_seq,al])\n",
    "            else:\n",
    "                temp_seq = ''.join([temp_seq,b])\n",
    "        compl_seq_list.append([get_complseq(temp_seq),pos])\n",
    "        seq_list.append([temp_seq,pos])\n",
    "    return seq_list,compl_seq_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "matrix_loc = '/local3/jake/TF_binding/JASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac/'\n",
    "bgfrac_df = read_csv('/local3/jake/TF_binding/ACTG_count.all_chrms.fractions.txt',delimiter='\\t')\n",
    "# matrix_loc = 'gene_dfJASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac'\n",
    "transfac_matrix_list = os.listdir(matrix_loc)\n",
    "infodicts_list = []\n",
    "for f in transfac_matrix_list:\n",
    "    curr_infodict = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    infodicts_list.append(curr_infodict)\n",
    "    \n",
    "chrmsizes_df = read_csv('/local3/jake/TF_binding/genomesize.all_chrms.hg19.txt',delimiter='\\t')\n",
    "ref_fasta_loc = '/local3/jake/TF_binding/testing'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_bgfracs = bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'][['frac_A','frac_C','frac_G','frac_T']]\n",
    "test_matrix = get_matrix_byTF('DUX4',infodicts_list)\n",
    "test_fracPWM = get_fracPWM_from_matrix(test_matrix)\n",
    "test_lnPWM = get_lnPWM_from_fracPWM(test_fracPWM,total_bgfracs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "time to read in info for tf 0.0005755424499511719\n",
      "time to read in info for tf 0.0004253387451171875\n",
      "time to read in info for tf 0.0011403560638427734\n",
      "time to read in info for tf 0.00047707557678222656\n",
      "time to read in info for tf 0.0005290508270263672\n",
      "time to read in info for tf 0.0005137920379638672\n",
      "time to read in info for tf 0.00032401084899902344\n",
      "time to read in info for tf 0.0003581047058105469\n",
      "time to read in info for tf 0.00035381317138671875\n",
      "time to read in info for tf 0.00035858154296875\n",
      "time to read in info for tf 0.0003616809844970703\n",
      "time to read in info for tf 0.0003879070281982422\n",
      "time to read in info for tf 0.00036215782165527344\n",
      "time to read in info for tf 0.0003578662872314453\n",
      "time to read in info for tf 0.0003979206085205078\n",
      "time to read in info for tf 0.00037384033203125\n",
      "time to read in info for tf 0.00038361549377441406\n",
      "time to read in info for tf 0.00039458274841308594\n",
      "time to read in info for tf 0.00035953521728515625\n",
      "time to read in info for tf 0.0003705024719238281\n",
      "time to read in info for tf 0.0003600120544433594\n",
      "time to read in info for tf 0.0003516674041748047\n",
      "time to read in info for tf 0.0003445148468017578\n",
      "time to read in info for tf 0.0004210472106933594\n",
      "time to read in info for tf 0.0003838539123535156\n",
      "time to read in info for tf 0.00037169456481933594\n",
      "time to read in info for tf 0.000377655029296875\n",
      "time to read in info for tf 0.0004055500030517578\n",
      "time to read in info for tf 0.0003795623779296875\n",
      "time to read in info for tf 0.00035953521728515625\n",
      "time to read in info for tf 0.0003578662872314453\n",
      "time to read in info for tf 0.0003643035888671875\n",
      "time to read in info for tf 0.0003638267517089844\n",
      "time to read in info for tf 0.00041222572326660156\n",
      "time to read in info for tf 0.00035881996154785156\n",
      "time to read in info for tf 0.0003859996795654297\n",
      "time to read in info for tf 0.00036597251892089844\n",
      "time to read in info for tf 0.0004124641418457031\n",
      "time to read in info for tf 0.0003833770751953125\n",
      "time to read in info for tf 0.0003807544708251953\n",
      "time to read in info for tf 0.0004413127899169922\n",
      "time to read in info for tf 0.00037550926208496094\n",
      "time to read in info for tf 0.00039124488830566406\n",
      "time to read in info for tf 0.0003688335418701172\n",
      "time to read in info for tf 0.00036787986755371094\n",
      "time to read in info for tf 0.00041365623474121094\n",
      "time to read in info for tf 0.0003757476806640625\n",
      "time to read in info for tf 0.00036406517028808594\n",
      "time to read in info for tf 0.00039124488830566406\n",
      "time to read in info for tf 0.00037169456481933594\n",
      "time to read in info for tf 0.0003898143768310547\n",
      "time to read in info for tf 0.000377655029296875\n",
      "time to read in info for tf 0.0003821849822998047\n",
      "time to read in info for tf 0.0003809928894042969\n",
      "time to read in info for tf 0.0004131793975830078\n",
      "time to read in info for tf 0.0003974437713623047\n",
      "time to read in info for tf 0.0003902912139892578\n",
      "time to read in info for tf 0.00041294097900390625\n",
      "time to read in info for tf 0.0004169940948486328\n",
      "time to read in info for tf 0.0003819465637207031\n",
      "time to read in info for tf 0.0003955364227294922\n",
      "time to read in info for tf 0.00040912628173828125\n",
      "time to read in info for tf 0.00039505958557128906\n",
      "time to read in info for tf 0.0003581047058105469\n",
      "time to read in info for tf 0.00037598609924316406\n",
      "time to read in info for tf 0.0003695487976074219\n",
      "time to read in info for tf 0.0004010200500488281\n",
      "time to read in info for tf 0.0003829002380371094\n",
      "time to read in info for tf 0.0003590583801269531\n",
      "time to read in info for tf 0.00043272972106933594\n",
      "time to read in info for tf 0.0003654956817626953\n",
      "time to read in info for tf 0.000408172607421875\n",
      "time to read in info for tf 0.0003516674041748047\n",
      "time to read in info for tf 0.00040602684020996094\n",
      "time to read in info for tf 0.0003693103790283203\n",
      "time to read in info for tf 0.00041103363037109375\n",
      "time to read in info for tf 0.00037479400634765625\n",
      "time to read in info for tf 0.0003802776336669922\n",
      "time to read in info for tf 0.0004239082336425781\n",
      "time to read in info for tf 0.0003826618194580078\n",
      "time to read in info for tf 0.0003807544708251953\n",
      "time to read in info for tf 0.0003571510314941406\n",
      "time to read in info for tf 0.00035762786865234375\n",
      "time to read in info for tf 0.0004076957702636719\n",
      "time to read in info for tf 0.0003745555877685547\n",
      "time to read in info for tf 0.0003762245178222656\n",
      "time to read in info for tf 0.0003685951232910156\n",
      "time to read in info for tf 0.0004024505615234375\n",
      "time to read in info for tf 0.0003762245178222656\n",
      "time to read in info for tf 0.0003871917724609375\n",
      "time to read in info for tf 0.00040268898010253906\n",
      "time to read in info for tf 0.00040841102600097656\n",
      "time to read in info for tf 0.000396728515625\n",
      "time to read in info for tf 0.00037932395935058594\n",
      "time to read in info for tf 0.0004203319549560547\n",
      "time to read in info for tf 0.00037026405334472656\n",
      "time to read in info for tf 0.00040984153747558594\n",
      "time to read in info for tf 0.0003819465637207031\n",
      "time to read in info for tf 0.00035834312438964844\n",
      "time to read in info for tf 0.0003707408905029297\n",
      "time to read in info for tf 0.00036978721618652344\n",
      "time to read in info for tf 0.000400543212890625\n",
      "time to read in info for tf 0.0003578662872314453\n",
      "time to read in info for tf 0.0003857612609863281\n",
      "time to read in info for tf 0.0003714561462402344\n",
      "time to read in info for tf 0.000385284423828125\n",
      "time to read in info for tf 0.00038361549377441406\n",
      "time to read in info for tf 0.0003886222839355469\n",
      "time to read in info for tf 0.00039005279541015625\n",
      "time to read in info for tf 0.0003933906555175781\n",
      "time to read in info for tf 0.0003707408905029297\n",
      "time to read in info for tf 0.0003714561462402344\n",
      "time to read in info for tf 0.0003848075866699219\n",
      "time to read in info for tf 0.00046062469482421875\n",
      "time to read in info for tf 0.00037741661071777344\n",
      "time to read in info for tf 0.0004067420959472656\n",
      "time to read in info for tf 0.00039577484130859375\n",
      "time to read in info for tf 0.00037169456481933594\n",
      "time to read in info for tf 0.00039076805114746094\n",
      "time to read in info for tf 0.0003771781921386719\n",
      "time to read in info for tf 0.00039887428283691406\n",
      "time to read in info for tf 0.0003802776336669922\n",
      "time to read in info for tf 0.0003952980041503906\n",
      "time to read in info for tf 0.0003714561462402344\n",
      "time to read in info for tf 0.0003910064697265625\n",
      "time to read in info for tf 0.0003745555877685547\n",
      "time to read in info for tf 0.00035953521728515625\n",
      "time to read in info for tf 0.0003597736358642578\n",
      "time to read in info for tf 0.0003979206085205078\n",
      "time to read in info for tf 0.0003764629364013672\n",
      "time to read in info for tf 0.00038313865661621094\n",
      "time to read in info for tf 0.0004131793975830078\n",
      "time to read in info for tf 0.0003819465637207031\n",
      "time to read in info for tf 0.0004100799560546875\n",
      "time to read in info for tf 0.00039839744567871094\n",
      "time to read in info for tf 0.0003695487976074219\n",
      "time to read in info for tf 0.00037026405334472656\n",
      "time to read in info for tf 0.0003705024719238281\n",
      "time to read in info for tf 0.00038933753967285156\n",
      "time to read in info for tf 0.000415802001953125\n",
      "time to read in info for tf 0.0004646778106689453\n",
      "time to read in info for tf 0.0003573894500732422\n",
      "time to read in info for tf 0.0003600120544433594\n",
      "time to read in info for tf 0.0003752708435058594\n",
      "time to read in info for tf 0.0003561973571777344\n",
      "time to read in info for tf 0.00039315223693847656\n",
      "time to read in info for tf 0.0003876686096191406\n",
      "time to read in info for tf 0.00035858154296875\n",
      "time to read in info for tf 0.0003821849822998047\n",
      "time to read in info for tf 0.00039887428283691406\n",
      "time to read in info for tf 0.00038051605224609375\n",
      "time to read in info for tf 0.00041222572326660156\n",
      "time to read in info for tf 0.0003857612609863281\n",
      "time to read in info for tf 0.0003485679626464844\n",
      "time to read in info for tf 0.0003924369812011719\n",
      "time to read in info for tf 0.0003581047058105469\n",
      "time to read in info for tf 0.0003883838653564453\n",
      "time to read in info for tf 0.0004010200500488281\n",
      "time to read in info for tf 0.00042057037353515625\n",
      "time to read in info for tf 0.0003712177276611328\n",
      "time to read in info for tf 0.0003879070281982422\n",
      "time to read in info for tf 0.00040602684020996094\n",
      "time to read in info for tf 0.00039005279541015625\n",
      "time to read in info for tf 0.0003933906555175781\n",
      "time to read in info for tf 0.00036978721618652344\n",
      "time to read in info for tf 0.0006189346313476562\n",
      "time to read in info for tf 0.00038242340087890625\n",
      "time to read in info for tf 0.00037360191345214844\n",
      "time to read in info for tf 0.0004093647003173828\n",
      "time to read in info for tf 0.0003764629364013672\n",
      "time to read in info for tf 0.00038242340087890625\n",
      "time to read in info for tf 0.00041985511779785156\n",
      "time to read in info for tf 0.0003960132598876953\n",
      "time to read in info for tf 0.00039196014404296875\n",
      "time to read in info for tf 0.0003657341003417969\n",
      "time to read in info for tf 0.0003910064697265625\n",
      "time to read in info for tf 0.0004589557647705078\n",
      "time to read in info for tf 0.0003895759582519531\n",
      "time to read in info for tf 0.0003540515899658203\n",
      "time to read in info for tf 0.000335693359375\n",
      "time to read in info for tf 0.00034356117248535156\n",
      "time to read in info for tf 0.0003132820129394531\n",
      "time to read in info for tf 0.00031113624572753906\n",
      "time to read in info for tf 0.0003039836883544922\n",
      "time to read in info for tf 0.0002994537353515625\n",
      "time to read in info for tf 0.0003216266632080078\n",
      "time to read in info for tf 0.0003349781036376953\n",
      "time to read in info for tf 0.00031065940856933594\n",
      "time to read in info for tf 0.0003643035888671875\n",
      "time to read in info for tf 0.0003216266632080078\n",
      "time to read in info for tf 0.0003046989440917969\n",
      "time to read in info for tf 0.0003306865692138672\n",
      "time to read in info for tf 0.0003108978271484375\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.0003421306610107422\n",
      "time to read in info for tf 0.0003428459167480469\n",
      "time to read in info for tf 0.00042819976806640625\n",
      "time to read in info for tf 0.0003592967987060547\n",
      "time to read in info for tf 0.00032830238342285156\n",
      "time to read in info for tf 0.00031828880310058594\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.00036525726318359375\n",
      "time to read in info for tf 0.00031685829162597656\n",
      "time to read in info for tf 0.0003418922424316406\n",
      "time to read in info for tf 0.0003483295440673828\n",
      "time to read in info for tf 0.0003139972686767578\n",
      "time to read in info for tf 0.0003638267517089844\n",
      "time to read in info for tf 0.0003361701965332031\n",
      "time to read in info for tf 0.00033211708068847656\n",
      "time to read in info for tf 0.00033664703369140625\n",
      "time to read in info for tf 0.0003552436828613281\n",
      "time to read in info for tf 0.0003554821014404297\n",
      "time to read in info for tf 0.00033545494079589844\n",
      "time to read in info for tf 0.00035071372985839844\n",
      "time to read in info for tf 0.00030803680419921875\n",
      "time to read in info for tf 0.0003421306610107422\n",
      "time to read in info for tf 0.00038695335388183594\n",
      "time to read in info for tf 0.0003733634948730469\n",
      "time to read in info for tf 0.000316619873046875\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.0003314018249511719\n",
      "time to read in info for tf 0.0005693435668945312\n",
      "time to read in info for tf 0.00033092498779296875\n",
      "time to read in info for tf 0.0003294944763183594\n",
      "time to read in info for tf 0.00036406517028808594\n",
      "time to read in info for tf 0.00034880638122558594\n",
      "time to read in info for tf 0.0003848075866699219\n",
      "time to read in info for tf 0.0003008842468261719\n",
      "time to read in info for tf 0.00031757354736328125\n",
      "time to read in info for tf 0.00035309791564941406\n",
      "time to read in info for tf 0.00031065940856933594\n",
      "time to read in info for tf 0.0003535747528076172\n",
      "time to read in info for tf 0.000347137451171875\n",
      "time to read in info for tf 0.0003006458282470703\n",
      "time to read in info for tf 0.0003113746643066406\n",
      "time to read in info for tf 0.0003781318664550781\n",
      "time to read in info for tf 0.00035452842712402344\n",
      "time to read in info for tf 0.00031828880310058594\n",
      "time to read in info for tf 0.0003323554992675781\n",
      "time to read in info for tf 0.0003104209899902344\n",
      "time to read in info for tf 0.0003108978271484375\n",
      "time to read in info for tf 0.000308990478515625\n",
      "time to read in info for tf 0.0003361701965332031\n",
      "time to read in info for tf 0.00034356117248535156\n",
      "time to read in info for tf 0.00029850006103515625\n",
      "time to read in info for tf 0.00032448768615722656\n",
      "time to read in info for tf 0.0003154277801513672\n",
      "time to read in info for tf 0.0004150867462158203\n",
      "time to read in info for tf 0.00034332275390625\n",
      "time to read in info for tf 0.0003170967102050781\n",
      "time to read in info for tf 0.0003178119659423828\n",
      "time to read in info for tf 0.0003757476806640625\n",
      "time to read in info for tf 0.0003566741943359375\n",
      "time to read in info for tf 0.00019073486328125\n",
      "time to read in info for tf 0.00019741058349609375\n",
      "time to read in info for tf 0.00039577484130859375\n",
      "time to read in info for tf 0.0003376007080078125\n",
      "time to read in info for tf 0.00033164024353027344\n",
      "time to read in info for tf 0.00032806396484375\n",
      "time to read in info for tf 0.000347137451171875\n",
      "time to read in info for tf 0.000335693359375\n",
      "time to read in info for tf 0.00031185150146484375\n",
      "time to read in info for tf 0.00034308433532714844\n",
      "time to read in info for tf 0.0003261566162109375\n",
      "time to read in info for tf 0.000335693359375\n",
      "time to read in info for tf 0.0003294944763183594\n",
      "time to read in info for tf 0.00034332275390625\n",
      "time to read in info for tf 0.0003204345703125\n",
      "time to read in info for tf 0.0003170967102050781\n",
      "time to read in info for tf 0.00032067298889160156\n",
      "time to read in info for tf 0.0003485679626464844\n",
      "time to read in info for tf 0.0003497600555419922\n",
      "time to read in info for tf 0.00034618377685546875\n",
      "time to read in info for tf 0.00033664703369140625\n",
      "time to read in info for tf 0.000370025634765625\n",
      "time to read in info for tf 0.00033092498779296875\n",
      "time to read in info for tf 0.0003001689910888672\n",
      "time to read in info for tf 0.0003044605255126953\n",
      "time to read in info for tf 0.000354766845703125\n",
      "time to read in info for tf 0.0003409385681152344\n",
      "time to read in info for tf 0.000354766845703125\n",
      "time to read in info for tf 0.00037217140197753906\n",
      "time to read in info for tf 0.0003237724304199219\n",
      "time to read in info for tf 0.0003352165222167969\n",
      "time to read in info for tf 0.0003173351287841797\n",
      "time to read in info for tf 0.0003325939178466797\n",
      "time to read in info for tf 0.0003094673156738281\n",
      "time to read in info for tf 0.0003211498260498047\n",
      "time to read in info for tf 0.00036334991455078125\n",
      "time to read in info for tf 0.0003476142883300781\n",
      "time to read in info for tf 0.0003256797790527344\n",
      "time to read in info for tf 0.0003044605255126953\n",
      "time to read in info for tf 0.0003113746643066406\n",
      "time to read in info for tf 0.0003635883331298828\n",
      "time to read in info for tf 0.0003330707550048828\n",
      "time to read in info for tf 0.00031280517578125\n",
      "time to read in info for tf 0.0003113746643066406\n",
      "time to read in info for tf 0.0003337860107421875\n",
      "time to read in info for tf 0.00035691261291503906\n",
      "time to read in info for tf 0.0003490447998046875\n",
      "time to read in info for tf 0.00032711029052734375\n",
      "time to read in info for tf 0.0003063678741455078\n",
      "time to read in info for tf 0.00032711029052734375\n",
      "time to read in info for tf 0.0003516674041748047\n",
      "time to read in info for tf 0.000308990478515625\n",
      "time to read in info for tf 0.00032830238342285156\n",
      "time to read in info for tf 0.0003101825714111328\n",
      "time to read in info for tf 0.0003407001495361328\n",
      "time to read in info for tf 0.000324249267578125\n",
      "time to read in info for tf 0.00037097930908203125\n",
      "time to read in info for tf 0.00031948089599609375\n",
      "time to read in info for tf 0.00030517578125\n",
      "time to read in info for tf 0.0003082752227783203\n",
      "time to read in info for tf 0.0003178119659423828\n",
      "time to read in info for tf 0.0003254413604736328\n",
      "time to read in info for tf 0.0002987384796142578\n",
      "time to read in info for tf 0.00035119056701660156\n",
      "time to read in info for tf 0.00033783912658691406\n",
      "time to read in info for tf 0.0003108978271484375\n",
      "time to read in info for tf 0.0003097057342529297\n",
      "time to read in info for tf 0.0003504753112792969\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.00031495094299316406\n",
      "time to read in info for tf 0.0003516674041748047\n",
      "time to read in info for tf 0.00032258033752441406\n",
      "time to read in info for tf 0.0003514289855957031\n",
      "time to read in info for tf 0.0003235340118408203\n",
      "time to read in info for tf 0.0003542900085449219\n",
      "time to read in info for tf 0.00032067298889160156\n",
      "time to read in info for tf 0.00032806396484375\n",
      "time to read in info for tf 0.0003654956817626953\n",
      "time to read in info for tf 0.00038504600524902344\n",
      "time to read in info for tf 0.00032210350036621094\n",
      "time to read in info for tf 0.0003540515899658203\n",
      "time to read in info for tf 0.0002989768981933594\n",
      "time to read in info for tf 0.00030422210693359375\n",
      "time to read in info for tf 0.00031757354736328125\n",
      "time to read in info for tf 0.00029921531677246094\n",
      "time to read in info for tf 0.0003180503845214844\n",
      "time to read in info for tf 0.00032210350036621094\n",
      "time to read in info for tf 0.0003266334533691406\n",
      "time to read in info for tf 0.0003082752227783203\n",
      "time to read in info for tf 0.00038886070251464844\n",
      "time to read in info for tf 0.0003464221954345703\n",
      "time to read in info for tf 0.00032401084899902344\n",
      "time to read in info for tf 0.0003170967102050781\n",
      "time to read in info for tf 0.00034809112548828125\n",
      "time to read in info for tf 0.0003097057342529297\n",
      "time to read in info for tf 0.0003192424774169922\n",
      "time to read in info for tf 0.0003173351287841797\n",
      "time to read in info for tf 0.0003147125244140625\n",
      "time to read in info for tf 0.00034236907958984375\n",
      "time to read in info for tf 0.00037026405334472656\n",
      "time to read in info for tf 0.0003521442413330078\n",
      "time to read in info for tf 0.0003287792205810547\n",
      "time to read in info for tf 0.0001647472381591797\n",
      "time to read in info for tf 0.00028586387634277344\n",
      "time to read in info for tf 0.00030517578125\n",
      "time to read in info for tf 0.0003094673156738281\n",
      "time to read in info for tf 0.0003135204315185547\n",
      "time to read in info for tf 0.0003578662872314453\n",
      "time to read in info for tf 0.000316619873046875\n",
      "time to read in info for tf 0.0003495216369628906\n",
      "time to read in info for tf 0.00034332275390625\n",
      "time to read in info for tf 0.0003249645233154297\n",
      "time to read in info for tf 0.00036263465881347656\n",
      "time to read in info for tf 0.00036215782165527344\n",
      "time to read in info for tf 0.00031685829162597656\n",
      "time to read in info for tf 0.0003104209899902344\n",
      "time to read in info for tf 0.00031113624572753906\n",
      "time to read in info for tf 0.00035643577575683594\n",
      "time to read in info for tf 0.00031757354736328125\n",
      "time to read in info for tf 0.00035500526428222656\n",
      "time to read in info for tf 0.00033664703369140625\n",
      "time to read in info for tf 0.0003180503845214844\n",
      "time to read in info for tf 0.00031757354736328125\n",
      "time to read in info for tf 0.0003037452697753906\n",
      "time to read in info for tf 0.0003120899200439453\n",
      "time to read in info for tf 0.00031685829162597656\n",
      "time to read in info for tf 0.00032448768615722656\n",
      "time to read in info for tf 0.0003056526184082031\n",
      "time to read in info for tf 0.0003287792205810547\n",
      "time to read in info for tf 0.0003371238708496094\n",
      "time to read in info for tf 0.00039196014404296875\n",
      "time to read in info for tf 0.0003409385681152344\n",
      "time to read in info for tf 0.00040984153747558594\n",
      "time to read in info for tf 0.00034356117248535156\n",
      "time to read in info for tf 0.00029969215393066406\n",
      "time to read in info for tf 0.00030803680419921875\n",
      "time to read in info for tf 0.00035572052001953125\n",
      "time to read in info for tf 0.00030684471130371094\n",
      "time to read in info for tf 0.00031256675720214844\n",
      "time to read in info for tf 0.00032329559326171875\n",
      "time to read in info for tf 0.0003228187561035156\n",
      "time to read in info for tf 0.00032639503479003906\n",
      "time to read in info for tf 0.0003314018249511719\n",
      "time to read in info for tf 0.0003592967987060547\n",
      "time to read in info for tf 0.0003952980041503906\n",
      "time to read in info for tf 0.0003516674041748047\n",
      "time to read in info for tf 0.00033545494079589844\n",
      "time to read in info for tf 0.0003180503845214844\n",
      "time to read in info for tf 0.00031876564025878906\n",
      "time to read in info for tf 0.0003619194030761719\n",
      "time to read in info for tf 0.0002989768981933594\n",
      "time to read in info for tf 0.00040435791015625\n",
      "time to read in info for tf 0.00030612945556640625\n",
      "time to read in info for tf 0.0004572868347167969\n",
      "time to read in info for tf 0.0005388259887695312\n",
      "time to read in info for tf 0.0004305839538574219\n",
      "time to read in info for tf 0.00045228004455566406\n",
      "time to read in info for tf 0.00045990943908691406\n",
      "time to read in info for tf 0.0004324913024902344\n",
      "time to read in info for tf 0.0004119873046875\n",
      "time to read in info for tf 0.0003719329833984375\n",
      "time to read in info for tf 0.00033092498779296875\n",
      "time to read in info for tf 0.0003478527069091797\n",
      "time to read in info for tf 0.00033211708068847656\n",
      "time to read in info for tf 0.0003364086151123047\n",
      "time to read in info for tf 0.0003592967987060547\n",
      "time to read in info for tf 0.00032806396484375\n",
      "time to read in info for tf 0.00031304359436035156\n",
      "time to read in info for tf 0.00031065940856933594\n",
      "time to read in info for tf 0.00035381317138671875\n",
      "time to read in info for tf 0.0003304481506347656\n",
      "time to read in info for tf 0.0003490447998046875\n",
      "time to read in info for tf 0.00032782554626464844\n",
      "time to read in info for tf 0.0003533363342285156\n",
      "time to read in info for tf 0.0003814697265625\n",
      "time to read in info for tf 0.0003333091735839844\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.0003173351287841797\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.0003349781036376953\n",
      "time to read in info for tf 0.00035691261291503906\n",
      "time to read in info for tf 0.00030422210693359375\n",
      "time to read in info for tf 0.00033974647521972656\n",
      "time to read in info for tf 0.00031256675720214844\n",
      "time to read in info for tf 0.0004031658172607422\n",
      "time to read in info for tf 0.00035309791564941406\n",
      "time to read in info for tf 0.00031256675720214844\n",
      "time to read in info for tf 0.0003192424774169922\n",
      "time to read in info for tf 0.00030350685119628906\n",
      "time to read in info for tf 0.0003180503845214844\n",
      "time to read in info for tf 0.0003643035888671875\n",
      "time to read in info for tf 0.0003123283386230469\n",
      "time to read in info for tf 0.0003170967102050781\n",
      "time to read in info for tf 0.00037026405334472656\n",
      "time to read in info for tf 0.0003337860107421875\n",
      "time to read in info for tf 0.0003819465637207031\n",
      "time to read in info for tf 0.00035119056701660156\n",
      "time to read in info for tf 0.0003573894500732422\n",
      "time to read in info for tf 0.00032258033752441406\n",
      "time to read in info for tf 0.0002980232238769531\n",
      "time to read in info for tf 0.00036144256591796875\n",
      "time to read in info for tf 0.00034165382385253906\n",
      "time to read in info for tf 0.0003457069396972656\n",
      "time to read in info for tf 0.00032258033752441406\n",
      "time to read in info for tf 0.0003170967102050781\n",
      "time to read in info for tf 0.0003440380096435547\n",
      "time to read in info for tf 0.00031685829162597656\n",
      "time to read in info for tf 0.00034236907958984375\n",
      "time to read in info for tf 0.00034809112548828125\n",
      "time to read in info for tf 0.0003578662872314453\n",
      "time to read in info for tf 0.00030684471130371094\n",
      "time to read in info for tf 0.00029921531677246094\n",
      "time to read in info for tf 0.00031065940856933594\n",
      "time to read in info for tf 0.0003502368927001953\n",
      "time to read in info for tf 0.0003211498260498047\n",
      "time to read in info for tf 0.0003504753112792969\n",
      "time to read in info for tf 0.00029921531677246094\n",
      "time to read in info for tf 0.0003337860107421875\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.0003306865692138672\n",
      "time to read in info for tf 0.0003063678741455078\n",
      "time to read in info for tf 0.00035262107849121094\n",
      "time to read in info for tf 0.0003299713134765625\n",
      "time to read in info for tf 0.0003540515899658203\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.00032806396484375\n",
      "time to read in info for tf 0.0003402233123779297\n",
      "time to read in info for tf 0.0003306865692138672\n",
      "time to read in info for tf 0.00032782554626464844\n",
      "time to read in info for tf 0.0003764629364013672\n",
      "time to read in info for tf 0.000335693359375\n",
      "time to read in info for tf 0.0003616809844970703\n",
      "time to read in info for tf 0.0003452301025390625\n",
      "time to read in info for tf 0.0003147125244140625\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "time to read in info for tf 0.0008966922760009766\n",
      "time to read in info for tf 0.0007107257843017578\n",
      "time to read in info for tf 0.0006330013275146484\n",
      "time to read in info for tf 0.0004417896270751953\n",
      "time to read in info for tf 0.0004475116729736328\n",
      "time to read in info for tf 0.0003256797790527344\n",
      "time to read in info for tf 0.00031876564025878906\n",
      "time to read in info for tf 0.0003414154052734375\n",
      "time to read in info for tf 0.0003597736358642578\n",
      "time to read in info for tf 0.00032329559326171875\n",
      "time to read in info for tf 0.00031876564025878906\n",
      "time to read in info for tf 0.0003237724304199219\n",
      "time to read in info for tf 0.00036787986755371094\n",
      "time to read in info for tf 0.0003552436828613281\n",
      "time to read in info for tf 0.0003504753112792969\n",
      "time to read in info for tf 0.00033211708068847656\n",
      "time to read in info for tf 0.00034236907958984375\n",
      "time to read in info for tf 0.0003325939178466797\n",
      "time to read in info for tf 0.0003616809844970703\n",
      "time to read in info for tf 0.00030112266540527344\n",
      "time to read in info for tf 0.00031566619873046875\n",
      "time to read in info for tf 0.0003361701965332031\n",
      "time to read in info for tf 0.0003104209899902344\n",
      "time to read in info for tf 0.00036263465881347656\n",
      "time to read in info for tf 0.00034880638122558594\n",
      "time to read in info for tf 0.0003249645233154297\n",
      "time to read in info for tf 0.0003421306610107422\n",
      "time to read in info for tf 0.00036406517028808594\n",
      "time to read in info for tf 0.0003352165222167969\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.0003688335418701172\n",
      "time to read in info for tf 0.00032591819763183594\n",
      "time to read in info for tf 0.0003352165222167969\n",
      "time to read in info for tf 0.00033020973205566406\n",
      "time to read in info for tf 0.0003447532653808594\n",
      "time to read in info for tf 0.0003311634063720703\n",
      "time to read in info for tf 0.0002932548522949219\n",
      "time to read in info for tf 0.0002994537353515625\n",
      "time to read in info for tf 0.0003070831298828125\n",
      "time to read in info for tf 0.00033736228942871094\n",
      "time to read in info for tf 0.00032520294189453125\n",
      "time to read in info for tf 0.00031375885009765625\n",
      "time to read in info for tf 0.0003554821014404297\n",
      "time to read in info for tf 0.0003085136413574219\n",
      "time to read in info for tf 0.00031495094299316406\n",
      "time to read in info for tf 0.00035643577575683594\n",
      "time to read in info for tf 0.00031948089599609375\n",
      "time to read in info for tf 0.00033593177795410156\n",
      "time to read in info for tf 0.0003001689910888672\n",
      "time to read in info for tf 0.00033545494079589844\n",
      "time to read in info for tf 0.0003440380096435547\n",
      "time to read in info for tf 0.00034928321838378906\n",
      "time to read in info for tf 0.0003154277801513672\n",
      "time to read in info for tf 0.00034809112548828125\n",
      "time to read in info for tf 0.0003185272216796875\n",
      "time to read in info for tf 0.0003333091735839844\n",
      "time to read in info for tf 0.00037932395935058594\n",
      "time to read in info for tf 0.00036835670471191406\n",
      "time to read in info for tf 0.0003409385681152344\n",
      "time to read in info for tf 0.00031638145446777344\n",
      "time to read in info for tf 0.00030517578125\n",
      "time to read in info for tf 0.0003173351287841797\n",
      "time to read in info for tf 0.00032329559326171875\n",
      "time to read in info for tf 0.0003497600555419922\n",
      "time to read in info for tf 0.00032019615173339844\n",
      "time to read in info for tf 0.0003445148468017578\n",
      "time to read in info for tf 0.0003294944763183594\n",
      "time to read in info for tf 0.0003407001495361328\n",
      "time to read in info for tf 0.0003542900085449219\n",
      "time to read in info for tf 0.00030684471130371094\n",
      "time to read in info for tf 0.0003209114074707031\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.00034999847412109375\n",
      "time to read in info for tf 0.00030612945556640625\n",
      "time to read in info for tf 0.000331878662109375\n",
      "time to read in info for tf 0.0003528594970703125\n",
      "time to read in info for tf 0.0003204345703125\n",
      "time to read in info for tf 0.0003025531768798828\n",
      "time to read in info for tf 0.00031566619873046875\n",
      "time to read in info for tf 0.0003113746643066406\n",
      "time to read in info for tf 0.0003299713134765625\n",
      "time to read in info for tf 0.0003504753112792969\n",
      "time to read in info for tf 0.00030541419982910156\n",
      "time to read in info for tf 0.0003085136413574219\n",
      "time to read in info for tf 0.00033164024353027344\n",
      "time to read in info for tf 0.00034332275390625\n",
      "time to read in info for tf 0.00035119056701660156\n",
      "time to read in info for tf 0.00038313865661621094\n",
      "time to read in info for tf 0.0003368854522705078\n",
      "time to read in info for tf 0.000331878662109375\n",
      "time to read in info for tf 0.0002987384796142578\n",
      "time to read in info for tf 0.00031876564025878906\n",
      "time to read in info for tf 0.0003650188446044922\n",
      "time to read in info for tf 0.0003509521484375\n",
      "time to read in info for tf 0.00030922889709472656\n",
      "time to read in info for tf 0.00033020973205566406\n",
      "time to read in info for tf 0.00029850006103515625\n",
      "time to read in info for tf 0.000316619873046875\n",
      "time to read in info for tf 0.00034737586975097656\n",
      "time to read in info for tf 0.0003268718719482422\n",
      "time to read in info for tf 0.00031113624572753906\n",
      "time to read in info for tf 0.0003199577331542969\n",
      "time to read in info for tf 0.0003185272216796875\n",
      "time to read in info for tf 0.0003685951232910156\n",
      "time to read in info for tf 0.00032639503479003906\n",
      "time to read in info for tf 0.0003292560577392578\n",
      "time to read in info for tf 0.00034165382385253906\n",
      "time to read in info for tf 0.00033974647521972656\n",
      "time to read in info for tf 0.0003113746643066406\n",
      "time to read in info for tf 0.00035309791564941406\n",
      "time to read in info for tf 0.0003285408020019531\n",
      "time to read in info for tf 0.0003757476806640625\n",
      "time to read in info for tf 0.0003371238708496094\n",
      "time to read in info for tf 0.00032329559326171875\n",
      "time to read in info for tf 0.0003235340118408203\n",
      "time to read in info for tf 0.0003101825714111328\n",
      "time to read in info for tf 0.00036787986755371094\n",
      "time to read in info for tf 0.0003440380096435547\n",
      "time to read in info for tf 0.0003628730773925781\n",
      "time to read in info for tf 0.0003223419189453125\n",
      "time to read in info for tf 0.00034546852111816406\n",
      "time to read in info for tf 0.0003364086151123047\n",
      "time to read in info for tf 0.00033283233642578125\n",
      "time to read in info for tf 0.0003390312194824219\n",
      "time to read in info for tf 0.0003120899200439453\n",
      "time to read in info for tf 0.0003414154052734375\n",
      "time to read in info for tf 0.00032067298889160156\n",
      "time to read in info for tf 0.0003428459167480469\n",
      "time to read in info for tf 0.00031566619873046875\n",
      "time to read in info for tf 0.0003383159637451172\n",
      "time to read in info for tf 0.0003235340118408203\n",
      "time to read in info for tf 0.0003871917724609375\n",
      "time to read in info for tf 0.00036144256591796875\n",
      "time to read in info for tf 0.00022220611572265625\n",
      "time to read in info for tf 0.00029969215393066406\n",
      "time to read in info for tf 0.0003705024719238281\n",
      "time to read in info for tf 0.0003578662872314453\n",
      "time to read in info for tf 0.0003314018249511719\n",
      "time to read in info for tf 0.0003197193145751953\n",
      "time to read in info for tf 0.0003914833068847656\n",
      "time to read in info for tf 0.0003268718719482422\n",
      "time to read in info for tf 0.0003693103790283203\n",
      "time to read in info for tf 0.0003490447998046875\n",
      "time to read in info for tf 0.0003418922424316406\n",
      "time to read in info for tf 0.0003306865692138672\n",
      "time to read in info for tf 0.0003178119659423828\n",
      "time to read in info for tf 0.00036787986755371094\n",
      "time to read in info for tf 0.00033020973205566406\n",
      "time to read in info for tf 0.0003094673156738281\n",
      "time to read in info for tf 0.00036334991455078125\n",
      "time to read in info for tf 0.00033783912658691406\n",
      "time to read in info for tf 0.0003304481506347656\n",
      "time to read in info for tf 0.00034689903259277344\n",
      "time to read in info for tf 0.0003077983856201172\n",
      "time to read in info for tf 0.00029969215393066406\n",
      "time to read in info for tf 0.00032019615173339844\n",
      "time to read in info for tf 0.0003628730773925781\n",
      "time to read in info for tf 0.0003437995910644531\n",
      "time to read in info for tf 0.0003573894500732422\n",
      "time to read in info for tf 0.0003361701965332031\n",
      "time to read in info for tf 0.0003151893615722656\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.00034999847412109375\n",
      "time to read in info for tf 0.0003333091735839844\n",
      "time to read in info for tf 0.00036406517028808594\n",
      "time to read in info for tf 0.0003161430358886719\n",
      "time to read in info for tf 0.0003235340118408203\n",
      "time to read in info for tf 0.0003390312194824219\n",
      "time to read in info for tf 0.0003123283386230469\n",
      "time to read in info for tf 0.00035572052001953125\n",
      "time to read in info for tf 0.0003650188446044922\n",
      "time to read in info for tf 0.00031757354736328125\n",
      "time to read in info for tf 0.00035071372985839844\n",
      "time to read in info for tf 0.0003058910369873047\n",
      "time to read in info for tf 0.00031828880310058594\n",
      "time to read in info for tf 0.0003895759582519531\n",
      "time to read in info for tf 0.0003590583801269531\n",
      "time to read in info for tf 0.000316619873046875\n",
      "time to read in info for tf 0.0003178119659423828\n",
      "time to read in info for tf 0.00038433074951171875\n",
      "time to read in info for tf 0.0003783702850341797\n",
      "time to read in info for tf 0.00034117698669433594\n",
      "time to read in info for tf 0.0003199577331542969\n",
      "time to read in info for tf 0.0003447532653808594\n",
      "time to read in info for tf 0.00031876564025878906\n",
      "time to read in info for tf 0.0003457069396972656\n",
      "time to read in info for tf 0.0003292560577392578\n",
      "time to read in info for tf 0.00040268898010253906\n",
      "time to read in info for tf 0.00030612945556640625\n",
      "time to read in info for tf 0.000316619873046875\n",
      "time to read in info for tf 0.0003190040588378906\n",
      "time to read in info for tf 0.00031113624572753906\n",
      "time to read in info for tf 0.00034618377685546875\n",
      "time to read in info for tf 0.0003380775451660156\n",
      "time to read in info for tf 0.0003428459167480469\n",
      "time to read in info for tf 0.00036263465881347656\n",
      "time to read in info for tf 0.0003037452697753906\n",
      "time to read in info for tf 0.0003476142883300781\n",
      "time to read in info for tf 0.00032448768615722656\n",
      "time to read in info for tf 0.0003349781036376953\n",
      "time to read in info for tf 0.00034999847412109375\n",
      "time to read in info for tf 0.00035500526428222656\n",
      "time to read in info for tf 0.0003490447998046875\n",
      "time to read in info for tf 0.0003113746643066406\n",
      "time to read in info for tf 0.00036454200744628906\n",
      "time to read in info for tf 0.00031638145446777344\n",
      "time to read in info for tf 0.00034236907958984375\n",
      "time to read in info for tf 0.00032258033752441406\n",
      "time to read in info for tf 0.0003142356872558594\n",
      "time to read in info for tf 0.00032210350036621094\n",
      "time to read in info for tf 0.00030922889709472656\n",
      "time to read in info for tf 0.0003483295440673828\n",
      "time to read in info for tf 0.0003218650817871094\n",
      "time to read in info for tf 0.00029850006103515625\n",
      "time to read in info for tf 0.00032639503479003906\n",
      "time to read in info for tf 0.00032067298889160156\n",
      "time to read in info for tf 0.0003528594970703125\n",
      "time to read in info for tf 0.00032806396484375\n",
      "time to read in info for tf 0.0003056526184082031\n",
      "time to read in info for tf 0.00038552284240722656\n",
      "time to read in info for tf 0.0003230571746826172\n",
      "time to read in info for tf 0.0003230571746826172\n",
      "time to read in info for tf 0.0003764629364013672\n",
      "time to read in info for tf 0.00032591819763183594\n",
      "time to read in info for tf 0.00034165382385253906\n",
      "time to read in info for tf 0.0003223419189453125\n",
      "time to read in info for tf 0.0003230571746826172\n",
      "time to read in info for tf 0.0003323554992675781\n",
      "time to read in info for tf 0.00033974647521972656\n",
      "time to read in info for tf 0.00030732154846191406\n",
      "time to read in info for tf 0.0003514289855957031\n",
      "time to read in info for tf 0.00036263465881347656\n",
      "time to read in info for tf 0.0003764629364013672\n",
      "time to read in info for tf 0.0003077983856201172\n",
      "time to read in info for tf 0.0003399848937988281\n",
      "time to read in info for tf 0.0003108978271484375\n",
      "time to read in info for tf 0.00032830238342285156\n",
      "time to read in info for tf 0.0003390312194824219\n",
      "time to read in info for tf 0.0003204345703125\n",
      "time to read in info for tf 0.0003669261932373047\n",
      "time to read in info for tf 0.0003292560577392578\n",
      "time to read in info for tf 0.00032973289489746094\n",
      "time to read in info for tf 0.00034308433532714844\n",
      "time to read in info for tf 0.00031638145446777344\n",
      "time to read in info for tf 0.0003497600555419922\n",
      "time to read in info for tf 0.00032782554626464844\n",
      "time to read in info for tf 0.00035262107849121094\n",
      "time to read in info for tf 0.00034308433532714844\n",
      "time to read in info for tf 0.0003235340118408203\n",
      "time to read in info for tf 0.00035572052001953125\n",
      "time to read in info for tf 0.000293731689453125\n",
      "time to read in info for tf 0.000335693359375\n",
      "time to read in info for tf 0.0003440380096435547\n",
      "time to read in info for tf 0.000354766845703125\n",
      "time to read in info for tf 0.0003466606140136719\n",
      "time to read in info for tf 0.00030303001403808594\n",
      "time to read in info for tf 0.00032210350036621094\n",
      "time to read in info for tf 0.0003685951232910156\n",
      "time to read in info for all tfs = 0.30228495597839355\n"
     ]
    }
   ],
   "source": [
    "infodicts_list = []\n",
    "total_start = time.time()\n",
    "for f in transfac_matrix_list:\n",
    "    start = time.time()\n",
    "    curr_infodict = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    infodicts_list.append(curr_infodict)\n",
    "    end = time.time()\n",
    "    print('time to read in info for tf {0}'.format(end - start))\n",
    "total_end = time.time()\n",
    "print('time to read in info for all tfs = {0}'.format(total_end - total_start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in infodicts_list:\n",
    "    curr_matrix = []\n",
    "    for n in range(1,i['TF_len']+1):\n",
    "        curr_matrix.append(i[n])\n",
    "    break\n",
    "    \n",
    "position = 61768012\n",
    "ref_full_seq = pybedtools.BedTool.seq('chr8:{0}-{1}'.format((position-len(test_matrix)),(position+len(test_matrix))),'/local3/jake/TF_binding/testing/chr8.fa')\n",
    "# ref_reverse_seq = pybedtools.BedTool.seq('chr8:{0}-{1}'.format((position-len(test_matrix)),position),'/local3/jake/TF_binding/testing/chr8.fa')\n",
    "tf_len = len(curr_matrix)\n",
    "ref_reverse_seq = get_rev_seq(ref_full_seq)\n",
    "\n",
    "bg_z_df = read_csv('/local3/jake/TF_binding/backgroundZ_forTFs.100reps.txt',delimiter='\\t',skiprows=1)\n",
    "\n",
    "\n",
    "finalres_df = read_csv('/local3/jake/TF_binding/testing/final_PWM.alltraits.allsnps.withPBinding100reps.output',delimiter='\\t')\n",
    "\n",
    "normed_pbind_dicts = []\n",
    "for row,i in finalres_df.iterrows():\n",
    "    temp_dict = {'snp':i['snp_id'],'TF':i['TF_name'],'H (REF)':i['H (REF)'],'H (ALT)':i['Hprime (ALT)'],'pbind (REF)':i['P binding (REF)'],'pbind (ALT)':i['P binding (ALT)']}\n",
    "    curr_bgz = float(bg_z_df.loc[bg_z_df['TF_name'] == i['TF_name']]['BG Z score'].values[0])\n",
    "    temp_dict['Norm_PBind (REF)'] = math.exp(-i['P binding (REF)'])/(curr_bgz+math.exp(-i['P binding (REF)']))\n",
    "    temp_dict['Norm_PBind (ALT)'] = math.exp(-i['P binding (ALT)'])/(curr_bgz+math.exp(-i['P binding (ALT)']))\n",
    "    normed_pbind_dicts.append(temp_dict)\n",
    "\n",
    "normed_pbind_df = DataFrame(normed_pbind_dicts)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_var_hist_v1(varname,plot_df):\n",
    "    fig,ax = plt.subplots()\n",
    "    x_ref = plot_df['{0} (REF)'.format(varname)]\n",
    "    x_alt = plot_df['{0} (ALT)'.format(varname)]\n",
    "\n",
    "    plotmin = min(x_ref)\n",
    "    if(min(x_alt) < plotmin):\n",
    "        plotmin = min(x_alt)\n",
    "    plotmax = max(x_ref)\n",
    "    if(max(x_alt) < plotmax):\n",
    "        plotmax = max(x_alt)\n",
    "        \n",
    "    result_ref = plt.hist(x_ref,50,color='blue',alpha=0.6)\n",
    "    mean_ref = np.mean(x_ref)\n",
    "    variance_ref = np.var(x_ref)\n",
    "    sigma_ref = np.sqrt(variance_ref)\n",
    "    xref = np.linspace(min(x_ref), max(x_ref),50)\n",
    "    dx_ref = result_ref[1][1] - result_ref[1][0]\n",
    "    scale_ref = len(x_ref)*dx_ref\n",
    "    plt.plot(xref, scs.norm.pdf(xref, mean_ref, sigma_ref)*scale_ref,color='blue')\n",
    "    \n",
    "    result_alt = plt.hist(x_alt,50,color='orange',alpha=0.6)\n",
    "    mean_alt = np.mean(x_alt)\n",
    "    variance_alt = np.var(x_alt)\n",
    "    sigma_alt = np.sqrt(variance_alt)\n",
    "    xalt = np.linspace(min(x_alt), max(x_alt),50)\n",
    "    dx_alt = result_alt[1][1] - result_alt[1][0]\n",
    "    scale_alt = len(x_alt)*dx_alt\n",
    "    plt.plot(xalt, scs.norm.pdf(xalt, mean_alt, sigma_alt)*scale_alt,color='orange')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def plot_var_hist(varname,plot_df):\n",
    "    fig,ax = plt.subplots()\n",
    "    x_ref = plot_df['{0} (REF)'.format(varname)]\n",
    "    x_alt = plot_df['{0} (ALT)'.format(varname)]\n",
    "    (mu_ref, sigma_ref) = norm.fit(x_ref)\n",
    "    plotmin = min(x_ref)\n",
    "    if(min(x_alt) < plotmin):\n",
    "        plotmin = min(x_alt)\n",
    "    plotmax = max(x_ref)\n",
    "    if(max(x_alt) < plotmax):\n",
    "        plotmax = max(x_alt)\n",
    "    # Empirical average and variance are computed\n",
    "    avg_ref = np.mean(x_ref)\n",
    "    var_ref = np.var(x_ref)\n",
    "    # From that, we know the shape of the fitted Gaussian.\n",
    "    pdf_x_ref = np.linspace(np.min(x_ref),np.max(x_ref),100)\n",
    "    pdf_y_ref = 1.0/np.sqrt(2*np.pi*var_ref)*np.exp(-0.5*(pdf_x_ref-avg_ref)**2/var_ref)\n",
    "    \n",
    "     # Empirical average and variance are computed\n",
    "    avg_alt = np.mean(x_alt)\n",
    "    var_alt = np.var(x_alt)\n",
    "    # From that, we know the shape of the fitted Gaussian.\n",
    "    pdf_x_alt = np.linspace(np.min(x_alt),np.max(x_alt),100)\n",
    "    pdf_y_alt = 1.0/np.sqrt(2*np.pi*var_alt)*np.exp(-0.5*(pdf_x_alt-avg_alt)**2/var_alt)\n",
    "\n",
    "    plt.hist([x_ref,x_alt],50)\n",
    "#     plt.plot(pdf_x_ref,pdf_y_ref,'k--', linewidth=2,color='blue')\n",
    "#     plt.plot(pdf_x_alt,pdf_y_alt,'k--', linewidth=2,color='orange')\n",
    "#     bins = np.linspace(plotmin,plotmax)\n",
    "#     plt.hist([x_ref,x_alt],bins=bins)\n",
    "    \n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX0AAAD4CAYAAAAAczaOAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAQkElEQVR4nO3df6zdd13H8efLlm38HJ27W0o7vMUUcCMSoI4JStBpNoaxM2FJnUAlM406EI2JdPzhlpgmMzEGiA7STKREwtKMxVUn6CwiGmCjg8HW1rnKsLuurhfQQTAZdrz943yHZ92967nnnHvuPf08H8nN+X4/5/O9n8+n9/Z1Pudzvt/vTVUhSWrDD610ByRJk2PoS1JDDH1JaoihL0kNMfQlqSFrV7oDp3LuuefW7OzsSndDkqbKPffc842qmjm5fNWH/uzsLAcOHFjpbkjSVEny7wuVu7wjSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNOa1Df3bnHczuvGOluyFJq8ZpHfqSpKda9ffeGYsbzu7bfmzl+iFJK8yZviQ15JShn+TDSY4nub+v7JwkdyZ5sHtc1/fcdUmOJHkgyWV95a9Jcl/33AeSZPzDkSQ9k0Fm+h8BLj+pbCewv6o2A/u7fZJcCGwDLuqOuSnJmu6YDwI7gM3d18nfU5K0zE4Z+lX1WeBbJxVvBfZ023uAK/vKb6mqx6vqIeAIcHGS9cALqurzVVXAR/uOkSRNyLBr+udX1TGA7vG8rnwD8HBfvbmubEO3fXL5gpLsSHIgyYH5+fkhuyhJOtm4P8hdaJ2+nqF8QVW1u6q2VNWWmZmn/bUvSdKQhg39R7slG7rH4135HHBBX72NwCNd+cYFyiVJEzRs6O8Dtnfb24Hb+8q3JTkzySZ6H9je3S0BfSfJJd1ZO2/vO0aSNCGnvDgryceBNwLnJpkDrgduBPYmuQY4ClwFUFUHk+wFDgEngGur6onuW/0GvTOBng18svuSJE3QKUO/qn55kacuXaT+LmDXAuUHgFcsqXeSpLHyilxJaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhoyUugn+Z0kB5Pcn+TjSc5Kck6SO5M82D2u66t/XZIjSR5Ictno3ZckLcXQoZ9kA/BbwJaqegWwBtgG7AT2V9VmYH+3T5ILu+cvAi4HbkqyZrTuS5KWYtTlnbXAs5OsBZ4DPAJsBfZ0z+8Bruy2twK3VNXjVfUQcAS4eMT2JUlLMHToV9V/AH8EHAWOAY9V1d8B51fVsa7OMeC87pANwMN932KuK3uaJDuSHEhyYH5+ftguSpJOMsryzjp6s/dNwIuA5yZ56zMdskBZLVSxqnZX1Zaq2jIzMzNsFyVJJxlleefngIeqar6q/he4DXgd8GiS9QDd4/Gu/hxwQd/xG+ktB0mSJmSU0D8KXJLkOUkCXAocBvYB27s624Hbu+19wLYkZybZBGwG7h6hfUnSEq0d9sCquivJrcCXgBPAl4HdwPOAvUmuoffCcFVX/2CSvcChrv61VfXEiP2XJC3B0KEPUFXXA9efVPw4vVn/QvV3AbtGaVOSNDyvyJWkhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSEjhX6SFya5Ncm/JDmc5CeTnJPkziQPdo/r+upfl+RIkgeSXDZ69yVJSzHqTP/9wKeq6uXAK4HDwE5gf1VtBvZ3+yS5ENgGXARcDtyUZM2I7UuSlmDo0E/yAuANwJ8BVNX3quq/ga3Anq7aHuDKbnsrcEtVPV5VDwFHgIuHbV+StHSjzPRfAswDf57ky0luTvJc4PyqOgbQPZ7X1d8APNx3/FxX9jRJdiQ5kOTA/Pz8CF2UJPUbJfTXAq8GPlhVrwK+S7eUs4gsUFYLVayq3VW1paq2zMzMjNBFSVK/UUJ/Dpirqru6/VvpvQg8mmQ9QPd4vK/+BX3HbwQeGaF9SdISDR36VfWfwMNJXtYVXQocAvYB27uy7cDt3fY+YFuSM5NsAjYDdw/bviRp6daOePy7gI8lOQP4GvAOei8ke5NcAxwFrgKoqoNJ9tJ7YTgBXFtVT4zYviRpCUYK/aq6F9iywFOXLlJ/F7BrlDYlScPzilxJaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNWbvSHVgJszvvAODrZ139/4U3PLZCvZGkyXGmL0kNMfQlqSGGviQ1xNCXpIaMHPpJ1iT5cpK/7vbPSXJnkge7x3V9da9LciTJA0kuG7VtSdLSjGOm/27gcN/+TmB/VW0G9nf7JLkQ2AZcBFwO3JRkzRjalyQNaKTQT7IReDNwc1/xVmBPt70HuLKv/JaqeryqHgKOABeP0r4kaWlGnem/D/g94Pt9ZedX1TGA7vG8rnwD8HBfvbmu7GmS7EhyIMmB+fn5EbsoSXrS0KGf5BeA41V1z6CHLFBWC1Wsqt1VtaWqtszMzAzbRUnSSUa5Ivf1wC8muQI4C3hBkr8AHk2yvqqOJVkPHO/qzwEX9B2/EXhkhPYlSUs09Ey/qq6rqo1VNUvvA9pPV9VbgX3A9q7aduD2bnsfsC3JmUk2AZuBu4fuuSRpyZbj3js3AnuTXAMcBa4CqKqDSfYCh4ATwLVV9cQytC9JWsRYQr+qPgN8ptv+JnDpIvV2AbvG0aYkaem8IleSGmLoS1JDDH1JaoihL0kNMfQlaTW54eze1zJp8s8lStJq8uSfcAX4+lnL25YzfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqyNChn+SCJP+Q5HCSg0ne3ZWfk+TOJA92j+v6jrkuyZEkDyS5bBwDWDY3nN37kqTTyNoRjj0B/G5VfSnJ84F7ktwJ/Cqwv6puTLIT2Am8J8mFwDbgIuBFwN8neWlVPTHaEMZnducdP9j++lkr2BFJWiZDz/Sr6lhVfanb/g5wGNgAbAX2dNX2AFd221uBW6rq8ap6CDgCXDxs+5KkpRvLmn6SWeBVwF3A+VV1DHovDMB5XbUNwMN9h811ZQt9vx1JDiQ5MD8/P44uSpIYQ+gneR7wCeC3q+rbz1R1gbJaqGJV7a6qLVW1ZWZmZtQuSpI6I4V+kmfRC/yPVdVtXfGjSdZ3z68Hjnflc8AFfYdvBB4ZpX1J0tKMcvZOgD8DDlfVH/c9tQ/Y3m1vB27vK9+W5Mwkm4DNwN3Dti9JWrpRzt55PfA24L4k93Zl7wVuBPYmuQY4ClwFUFUHk+wFDtE78+fa1XTmjiS1YOjQr6p/ZuF1eoBLFzlmF7Br2DYlSaPxilxJaoihL0kNMfQlqSGGviQ1ZJSzd5rxlHvy3PjmFeyJJI3Gmf5SefdNSVPM0Jekhhj6ktQQQ1+SGmLoj4tr/ZKmgGfvjGASf2nryTY8a0jSOBj606L/XcQNj61cPyRNNZd3JKkhhr4kNcTQl6SGuKYvSROyGm7p4kx/pXiKp9S2FcoAZ/oT9oNTMMd1iqdn9UhaAkN/lVjq+fhjf/GQ1ARDf7Vx5i5Npyf/767y/7eG/jJYDR/WaHFPvZL66t7GKv+PqtVpElflj5uhr8mZpncxUzJrk5bK0F9u0xQey9TXafn8YRpnbdJSGfqNWGxJY6lB59KIdGqr+f+J5+mfZmZ33vGUX7gV4TUI0qrlTP90tQLr5+NaxlmNt5Mee58W+/ms6M9tdc1IgQksOV7d19YqGvcyMvS1eq2iYHxa20O2u9gL47R87vFMhr3WBFxynCRDX1PjdAjGpZrK039X0Vlaq/pdzAox9HVqvsU+tUkE3TSdCXaaWY1LjsMy9LUgT18cTMvvPgZ9sR5XYE5iGeeUY1tF72KGZejr9LJKZsOrYmbY92+xKta9T4PAPB1MPPSTXA68H1gD3FxVN066Dzq9LPau5HQOunF9CKr2TDT0k6wB/hT4eWAO+GKSfVV1aJL9kHRqp9VnLvqBSV+cdTFwpKq+VlXfA24Btk64D5LUrFTV5BpL3gJcXlW/1u2/DXhtVb3zpHo7gB3d7suAB4Zo7lzgGyN0dxq1OGZoc9yOuQ2jjPlHqmrm5MJJr+lngbKnvepU1W5g90gNJQeqasso32PatDhmaHPcjrkNyzHmSS/vzAEX9O1vBB6ZcB8kqVmTDv0vApuTbEpyBrAN2DfhPkhSsya6vFNVJ5K8E/hbeqdsfriqDi5TcyMtD02pFscMbY7bMbdh7GOe6Ae5kqSV5f30Jakhhr4kNWTqQz/J5UkeSHIkyc4Fnk+SD3TPfzXJq1ein+M0wJh/pRvrV5N8LskrV6Kf43SqMffV+4kkT3TXhEy1Qcac5I1J7k1yMMk/TrqP4zbA7/bZSf4qyVe6Mb9jJfo5Tkk+nOR4kvsXeX68GVZVU/tF78PgfwNeApwBfAW48KQ6VwCfpHeNwCXAXSvd7wmM+XXAum77TS2Mua/ep4G/Ad6y0v2ewM/5hcAh4MXd/nkr3e8JjPm9wB922zPAt4AzVrrvI477DcCrgfsXeX6sGTbtM/1BbuuwFfho9XwBeGGS9ZPu6BidcsxV9bmq+q9u9wv0roeYZoPevuNdwCeA45Ps3DIZZMxXA7dV1VGAqpr2cQ8y5gKenyTA8+iF/onJdnO8quqz9MaxmLFm2LSH/gbg4b79ua5sqXWmyVLHcw29WcI0O+WYk2wAfgn40AT7tZwG+Tm/FFiX5DNJ7kny9on1bnkMMuY/AX6M3kWd9wHvrqrvT6Z7K2asGTbt99Mf5LYOA936YYoMPJ4kP0Mv9H9qWXu0/AYZ8/uA91TVE71J4NQbZMxrgdcAlwLPBj6f5AtV9a/L3bllMsiYLwPuBX4W+FHgziT/VFXfXu7OraCxZti0h/4gt3U43W79MNB4kvw4cDPwpqr65oT6tlwGGfMW4JYu8M8Frkhyoqr+cjJdHLtBf7e/UVXfBb6b5LPAK4FpDf1BxvwO4MbqLXYfSfIQ8HLg7sl0cUWMNcOmfXlnkNs67APe3n0CfgnwWFUdm3RHx+iUY07yYuA24G1TPOvrd8oxV9WmqpqtqlngVuA3pzjwYbDf7duBn06yNslzgNcChyfcz3EaZMxH6b2zIcn59O7C+7WJ9nLyxpphUz3Tr0Vu65Dk17vnP0TvTI4rgCPA/9CbKUytAcf8+8APAzd1M98TNcV3JxxwzKeVQcZcVYeTfAr4KvB9en+JbsHT/qbBgD/nPwA+kuQ+esse76mqqb7dcpKPA28Ezk0yB1wPPAuWJ8O8DYMkNWTal3ckSUtg6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SG/B+XL9+aQcihbQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot_var_hist('Norm_PBind',normed_pbind_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "curr_lnfracPWM_bychrm= []\n",
    "for n in range(1,23):\n",
    "    bgfreqs_n = bgfrac_df.loc[bgfrac_df['Chrm'] == str(n)][['frac_A','frac_C','frac_G','frac_T']]\n",
    "    curr_lnfracPWM_bychrm.append(get_lnPWM_from_fracPWM(test_fracPWM,bgfreqs_n))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def get_random_bgseqs(slen,reps,fastaloc,chrmsizes,seqinfo_file=None):\n",
    "    chrms_touse = list(chrmsizes['chrom'])\n",
    "    bgseq_list = []\n",
    "    if(seqinfo_file is not None):\n",
    "        genseqs_df = read_csv(seqinfo_file,delimiter='\\t')\n",
    "        for r,seq in genseqs_df.iterrows():\n",
    "            chrmfile = '{0}/chr{1}.fa'.format(fastaloc,seq['Chrm'])\n",
    "            curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(seq['Chrm'],seq['Start'],seq['End']),chrmfile)\n",
    "            bgseq_list.append([seq['Chrm'],seq['Start'],seq['End'],curr_seq.upper()])\n",
    "    while(len(bgseq_list) < int(reps)):\n",
    "        is_valid_seq = True\n",
    "        try:\n",
    "#             curr_chrm = random.randint(1,22)\n",
    "            curr_chrm = 8\n",
    "            curr_start = random.randint(1,chrmsizes.loc[chrmsizes['chrom'] == 'chr{0}'.format(curr_chrm)]['size'].values[0])\n",
    "            #TODO check for end being over the size of the chrm\n",
    "            curr_end = curr_start + slen\n",
    "            chrmfile = '{0}/chr{1}.fa'.format(fastaloc,curr_chrm)\n",
    "            curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(curr_chrm,curr_start,curr_end),chrmfile).upper()\n",
    "            for b in curr_seq:\n",
    "                if(b not in 'ACTG'):\n",
    "                    is_valid_seq = False\n",
    "                    continue\n",
    "            if(is_valid_seq):\n",
    "                bgseq_list.append([curr_chrm,curr_start,curr_end,curr_seq])\n",
    "        except:\n",
    "            continue\n",
    "            # bgseq_list.append({'chrm':curr_chrm,'start':curr_start,'end':curr_end,'seq':curr_seq.upper()})\n",
    "    return bgseq_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "bgseqs_test = get_random_bgseqs(10,1000,'/local3/jake/TF_binding/testing/',chrmsizes_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "time taken = 0.15946602821350098\n"
     ]
    }
   ],
   "source": [
    "start=time.time()\n",
    "outfile_currTF = open('/local3/jake/TF_binding/testing/test_output_seqs.outputfile','w')\n",
    "outfile_currTF.write('Chrm\\tStart\\tEnd\\n')\n",
    "for s in bgseqs_test:\n",
    "    curr_seq = s[3]\n",
    "    time.sleep(5.18288373947e-05)\n",
    "    outfile_currTF.write('{0}\\t{1}\\t{2}\\n'.format(s[0],s[1],s[2]))\n",
    "outfile_currTF.close()\n",
    "end=time.time()\n",
    "print('time taken = {0}'.format(end-start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "time taken = 0.06365561485290527\n"
     ]
    }
   ],
   "source": [
    "start=time.time()\n",
    "seqsdf_test = DataFrame(bgseqs_test,columns=['Chrm','Start','End','seq'])\n",
    "seqsdf_test.to_csv('/local3/jake/TF_binding/testing/test_output_seqs.outputfile_fromdf',sep='\\t',columns=['Chrm','Start','End'],index=False)\n",
    "end=time.time()\n",
    "print('time taken = {0}'.format(end-start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[8, 24455422, 24455432, 'TATGGCACATA'],\n",
       " [8, 85030719, 85030729, 'CTTTAACATGT'],\n",
       " [8, 87026425, 87026435, 'ACACAGGATAG'],\n",
       " [8, 141101110, 141101120, 'CTCAAGGACTG'],\n",
       " [8, 77052724, 77052734, 'TATTCTAATTA'],\n",
       " [8, 139516786, 139516796, 'TTAATTACAGC'],\n",
       " [8, 16079328, 16079338, 'GGCAGCCTAAG'],\n",
       " [8, 52129563, 52129573, 'CAGGCATACTT'],\n",
       " [8, 48282966, 48282976, 'AGATAAGCAGA'],\n",
       " [8, 80697033, 80697043, 'TCTGAGTTCTA'],\n",
       " [8, 137029103, 137029113, 'GAGAATGGTGT'],\n",
       " [8, 116871118, 116871128, 'TATGTTTGTTT'],\n",
       " [8, 41489368, 41489378, 'AAGCAATTCTC'],\n",
       " [8, 131625939, 131625949, 'GAGGCCATTAA'],\n",
       " [8, 35321313, 35321323, 'AAAAAAAAAAA'],\n",
       " [8, 17500543, 17500553, 'GTGTTCCTAGT'],\n",
       " [8, 2561457, 2561467, 'TTGCTCAGGCT'],\n",
       " [8, 28250412, 28250422, 'CCGCCTTGGCC'],\n",
       " [8, 23994670, 23994680, 'TGGACTTACAG'],\n",
       " [8, 67094216, 67094226, 'TTAAAACAGAG'],\n",
       " [8, 90388046, 90388056, 'AGGAGACATAA'],\n",
       " [8, 27511594, 27511604, 'TGCATCCCATC'],\n",
       " [8, 74915186, 74915196, 'TGTTCTCTTGA'],\n",
       " [8, 96961639, 96961649, 'GACTCGGACAG'],\n",
       " [8, 490041, 490051, 'AATCCAGTTTA'],\n",
       " [8, 49208397, 49208407, 'TATATTTCAGT'],\n",
       " [8, 30749563, 30749573, 'TTTGCTTTTGA'],\n",
       " [8, 8261457, 8261467, 'GGTGGCGGGCG'],\n",
       " [8, 79308776, 79308786, 'TAGTCTCCAGT'],\n",
       " [8, 41789590, 41789600, 'AAACCAAAGAA'],\n",
       " [8, 112323368, 112323378, 'AACTAACCAAT'],\n",
       " [8, 100531632, 100531642, 'GGGATCTAGAA'],\n",
       " [8, 62022189, 62022199, 'GGTAGAAACTT'],\n",
       " [8, 128693202, 128693212, 'AATATAGATAC'],\n",
       " [8, 85437893, 85437903, 'TTTTTTCAATG'],\n",
       " [8, 61750418, 61750428, 'ATTTATTTATT'],\n",
       " [8, 107148376, 107148386, 'ACTCCTGGCCT'],\n",
       " [8, 65466040, 65466050, 'AGCCACTGCGT'],\n",
       " [8, 121657535, 121657545, 'AGGTCTAACGT'],\n",
       " [8, 55680559, 55680569, 'GCTGATTGCAA'],\n",
       " [8, 112685346, 112685356, 'TTCTTAACTCT'],\n",
       " [8, 135472182, 135472192, 'GGCCCATGATC'],\n",
       " [8, 137351210, 137351220, 'TGAGAACAATG'],\n",
       " [8, 114156104, 114156114, 'TATGTAAATAT'],\n",
       " [8, 102203286, 102203296, 'TTTAAACTGCT'],\n",
       " [8, 42856197, 42856207, 'GCCCACAAGTT'],\n",
       " [8, 53411934, 53411944, 'TTATAATGAGA'],\n",
       " [8, 122714006, 122714016, 'TAGCTAGTGAT'],\n",
       " [8, 18567510, 18567520, 'GAAGTGATAGA'],\n",
       " [8, 25733405, 25733415, 'GACTTAGGAAT'],\n",
       " [8, 63788634, 63788644, 'CCATCTGACAA'],\n",
       " [8, 101766593, 101766603, 'CTCTCTCCCTA'],\n",
       " [8, 65206998, 65207008, 'ATACATTCTGA'],\n",
       " [8, 114172587, 114172597, 'GAGTGGAAATG'],\n",
       " [8, 61648242, 61648252, 'TTTAAAAGAAT'],\n",
       " [8, 108758662, 108758672, 'ACACATCAGAT'],\n",
       " [8, 23501073, 23501083, 'CCATCTCACAA'],\n",
       " [8, 117873547, 117873557, 'GGGAAACATAA'],\n",
       " [8, 104723779, 104723789, 'TTGCCATATAT'],\n",
       " [8, 135256778, 135256788, 'ATACCTGATAA'],\n",
       " [8, 29908879, 29908889, 'CTTTTGACTGG'],\n",
       " [8, 24245587, 24245597, 'AGTTCCAGGTC'],\n",
       " [8, 99838137, 99838147, 'GAGAAGGTGGA'],\n",
       " [8, 4813850, 4813860, 'ATTAACATCCA'],\n",
       " [8, 139870342, 139870352, 'TCCTTGCTGCC'],\n",
       " [8, 24007981, 24007991, 'ATTTTACTATT'],\n",
       " [8, 65079462, 65079472, 'AGAAAAATAAT'],\n",
       " [8, 68604771, 68604781, 'TTTTCCCAATT'],\n",
       " [8, 27524396, 27524406, 'AGGTCAGCTGT'],\n",
       " [8, 86471381, 86471391, 'CTAATAATGTA'],\n",
       " [8, 124955416, 124955426, 'CTCAATAAATG'],\n",
       " [8, 20759328, 20759338, 'CAACTCAGCTA'],\n",
       " [8, 101604788, 101604798, 'TGGCAGTTCCT'],\n",
       " [8, 127724798, 127724808, 'TGTGTTTTTTG'],\n",
       " [8, 72545408, 72545418, 'GATATTTTTTG'],\n",
       " [8, 15425807, 15425817, 'CTTGTTCTGGG'],\n",
       " [8, 75597018, 75597028, 'TTCAAGTGAGA'],\n",
       " [8, 112984006, 112984016, 'ACATGTTTACA'],\n",
       " [8, 109044434, 109044444, 'AGAGGAAAGGA'],\n",
       " [8, 76071005, 76071015, 'ATGCTGCAATA'],\n",
       " [8, 92937423, 92937433, 'TTGAAAATGCA'],\n",
       " [8, 39557068, 39557078, 'TGTCTTCTTAA'],\n",
       " [8, 126639119, 126639129, 'GAACAAGAGCA'],\n",
       " [8, 117906559, 117906569, 'AGCAAAAGTTG'],\n",
       " [8, 62580506, 62580516, 'ATCATAACAAA'],\n",
       " [8, 122074477, 122074487, 'CATTCTGGAAA'],\n",
       " [8, 101232676, 101232686, 'AGCTACCTAAA'],\n",
       " [8, 73515631, 73515641, 'AGCAGCTCTTT'],\n",
       " [8, 116154302, 116154312, 'GTATTGAAGAT'],\n",
       " [8, 89088679, 89088689, 'CCCAGCCACGT'],\n",
       " [8, 55023354, 55023364, 'CAAAATATTAT'],\n",
       " [8, 78153930, 78153940, 'TTTAGCACCAT'],\n",
       " [8, 16791496, 16791506, 'TAGCTCTCTGA'],\n",
       " [8, 94961944, 94961954, 'ATTTTCTAGTT'],\n",
       " [8, 86332381, 86332391, 'TTGTAGATGTG'],\n",
       " [8, 92677568, 92677578, 'AGGATTGCTTG'],\n",
       " [8, 15211034, 15211044, 'ATATTTACATA'],\n",
       " [8, 1679618, 1679628, 'ACAGCTGCAGC'],\n",
       " [8, 81679952, 81679962, 'GTTAATCAATA'],\n",
       " [8, 50635900, 50635910, 'TGTAGTCACGA'],\n",
       " [8, 66201143, 66201153, 'CCAGAATGTTC'],\n",
       " [8, 4211642, 4211652, 'GCGCCTGAAAG'],\n",
       " [8, 32437058, 32437068, 'CTGTTTCTTGA'],\n",
       " [8, 12531168, 12531178, 'AGGACCTGCAG'],\n",
       " [8, 24002622, 24002632, 'AGATCCCTAAG'],\n",
       " [8, 17409685, 17409695, 'GTTGAGTCAGT'],\n",
       " [8, 69248215, 69248225, 'GGGAGGCTGAG'],\n",
       " [8, 32122102, 32122112, 'TTGTTTGTATA'],\n",
       " [8, 2838899, 2838909, 'GCCAGTGAAGT'],\n",
       " [8, 85450095, 85450105, 'TTCCAATATTG'],\n",
       " [8, 32035359, 32035369, 'GAGAAGTTTGC'],\n",
       " [8, 7750018, 7750028, 'CCAGCTATGTT'],\n",
       " [8, 54437722, 54437732, 'CCTCACCTTGC'],\n",
       " [8, 133402622, 133402632, 'GCTCCTGGTGG'],\n",
       " [8, 127791983, 127791993, 'TGGGAGGCTGA'],\n",
       " [8, 59282405, 59282415, 'GAGGCTTGGCA'],\n",
       " [8, 103386672, 103386682, 'GTCACTCTAAT'],\n",
       " [8, 74296492, 74296502, 'CTAAGGCCCCA'],\n",
       " [8, 142646587, 142646597, 'GGCAAAGTGAC'],\n",
       " [8, 66716124, 66716134, 'GGAGAACAAAC'],\n",
       " [8, 142611811, 142611821, 'ACTAGCTAAAA'],\n",
       " [8, 22618408, 22618418, 'AGCCTTGCTGG'],\n",
       " [8, 60694155, 60694165, 'TAAAAAGAACA'],\n",
       " [8, 3918118, 3918128, 'CGGAACTGTCC'],\n",
       " [8, 87198348, 87198358, 'CTTCCACTCAT'],\n",
       " [8, 97969560, 97969570, 'GGGTTCTGGAA'],\n",
       " [8, 113409494, 113409504, 'GAATCATCACC'],\n",
       " [8, 108204248, 108204258, 'GAAAGAGCTGA'],\n",
       " [8, 26533096, 26533106, 'ACACTTGGGTT'],\n",
       " [8, 103948741, 103948751, 'ACATACATTTG'],\n",
       " [8, 13675715, 13675725, 'AACACGGTGAA'],\n",
       " [8, 78896990, 78897000, 'GCAATAGGGAC'],\n",
       " [8, 47641459, 47641469, 'TGAAGGACCTC'],\n",
       " [8, 93002300, 93002310, 'AAAGGTAAAAA'],\n",
       " [8, 106723615, 106723625, 'TCTGCCTTAAA'],\n",
       " [8, 36628986, 36628996, 'TTCCCAGAGAA'],\n",
       " [8, 28888060, 28888070, 'AATTAGTAAAT'],\n",
       " [8, 78023465, 78023475, 'AAAGGTGTGCT'],\n",
       " [8, 94958451, 94958461, 'TCTCCCTGTGT'],\n",
       " [8, 29510767, 29510777, 'TAGTTCTGCAT'],\n",
       " [8, 121213584, 121213594, 'GCCAGTTTGCA'],\n",
       " [8, 24495473, 24495483, 'TTTATAAATAA'],\n",
       " [8, 82159613, 82159623, 'TAATACTTAAA'],\n",
       " [8, 124502034, 124502044, 'TCTACTAAAAA'],\n",
       " [8, 23010989, 23010999, 'TGGTTAACATC'],\n",
       " [8, 10386763, 10386773, 'TCACAAATGCT'],\n",
       " [8, 81125794, 81125804, 'GCACTCTAGCC'],\n",
       " [8, 65184629, 65184639, 'CTTCTTAAAAG'],\n",
       " [8, 69709729, 69709739, 'CTACCATATTC'],\n",
       " [8, 127781727, 127781737, 'TTGGTGACATG'],\n",
       " [8, 67842025, 67842035, 'GAAAAATGCAA'],\n",
       " [8, 131491984, 131491994, 'ATACCTGAGAC'],\n",
       " [8, 81698128, 81698138, 'TCTTGGCTGAT'],\n",
       " [8, 91246039, 91246049, 'GTAATTTATAT'],\n",
       " [8, 101537785, 101537795, 'ATCCATGCTGT'],\n",
       " [8, 72250878, 72250888, 'TTCAGAAGAAA'],\n",
       " [8, 89857840, 89857850, 'AGATCATGACA'],\n",
       " [8, 12709989, 12709999, 'ATGTCTAAGTT'],\n",
       " [8, 79723663, 79723673, 'GGGTATCAGTC'],\n",
       " [8, 95074761, 95074771, 'AGCTCCTCACT'],\n",
       " [8, 48142993, 48143003, 'CTCAGACCACA'],\n",
       " [8, 124583014, 124583024, 'AACCCAGCTAA'],\n",
       " [8, 48506236, 48506246, 'GAGTGGAGTGG'],\n",
       " [8, 67696931, 67696941, 'ATCCTTTTTAG'],\n",
       " [8, 28884967, 28884977, 'AGCATTTTGGG'],\n",
       " [8, 40263783, 40263793, 'AGAAAGTAGAG'],\n",
       " [8, 61819552, 61819562, 'AGTTTACTGGC'],\n",
       " [8, 9796369, 9796379, 'CTAAAGATCAT'],\n",
       " [8, 136764729, 136764739, 'TCGTGGGTTCC'],\n",
       " [8, 58939258, 58939268, 'AAACAGCGCTT'],\n",
       " [8, 3655808, 3655818, 'ATTTCCTCTTA'],\n",
       " [8, 73710286, 73710296, 'AGTATTTTCAT'],\n",
       " [8, 102975773, 102975783, 'AATGGCAGATG'],\n",
       " [8, 105715728, 105715738, 'ATAATTGTACC'],\n",
       " [8, 62773261, 62773271, 'TAATTTTGCAA'],\n",
       " [8, 65124488, 65124498, 'CTAAAATTTCT'],\n",
       " [8, 6536766, 6536776, 'TGCTATTTCTG'],\n",
       " [8, 134768951, 134768961, 'TGGCTTAAAAC'],\n",
       " [8, 18491, 18501, 'CCAGCAGAGCC'],\n",
       " [8, 119798462, 119798472, 'GTTAACACTGA'],\n",
       " [8, 86931601, 86931611, 'CTACCTCACAA'],\n",
       " [8, 20862557, 20862567, 'TGATGGCACCA'],\n",
       " [8, 29476945, 29476955, 'ACTCCTAGGTC'],\n",
       " [8, 52356115, 52356125, 'AACTAAAACTG'],\n",
       " [8, 63304349, 63304359, 'AATGTGTGTGA'],\n",
       " [8, 135236297, 135236307, 'CACTCAAATGC'],\n",
       " [8, 93902939, 93902949, 'ATTTTTCTAAG'],\n",
       " [8, 51991513, 51991523, 'ATGCAGTTCAT'],\n",
       " [8, 48884362, 48884372, 'TCACTCTTGGC'],\n",
       " [8, 77648992, 77649002, 'ATTAATGGTTG'],\n",
       " [8, 47797185, 47797195, 'AACAGGTGGGC'],\n",
       " [8, 84144524, 84144534, 'CTCTCTGTTAG'],\n",
       " [8, 68944412, 68944422, 'AGTGCAAGAGA'],\n",
       " [8, 119292154, 119292164, 'CAACACACAAA'],\n",
       " [8, 36036258, 36036268, 'TGTAATTATAT'],\n",
       " [8, 61683074, 61683084, 'GTCATGGGGAC'],\n",
       " [8, 112242762, 112242772, 'ATTTATCCCTG'],\n",
       " [8, 128167908, 128167918, 'TATGACAAACC'],\n",
       " [8, 133427843, 133427853, 'ATTCCATTTCT'],\n",
       " [8, 115528888, 115528898, 'GAGTAAATATG'],\n",
       " [8, 123891856, 123891866, 'GATGAAGTAAG'],\n",
       " [8, 111162698, 111162708, 'ACAATGGTTGT'],\n",
       " [8, 2020778, 2020788, 'TGGAGAGGGAG'],\n",
       " [8, 37952065, 37952075, 'GCCTTGGAGGG'],\n",
       " [8, 117143540, 117143550, 'GATTCCCAAGG'],\n",
       " [8, 42325873, 42325883, 'AGAAAGGTAGG'],\n",
       " [8, 98916779, 98916789, 'ATTGACTCTTG'],\n",
       " [8, 56379753, 56379763, 'ATCTCCACAAT'],\n",
       " [8, 119628790, 119628800, 'CTCTTTACCCT'],\n",
       " [8, 98311195, 98311205, 'TAATCAATAGT'],\n",
       " [8, 124269358, 124269368, 'CACGTTGGCCA'],\n",
       " [8, 105246483, 105246493, 'ATCTGATTCTA'],\n",
       " [8, 24047175, 24047185, 'ACTTCACTGAG'],\n",
       " [8, 14337314, 14337324, 'GCATGATCTCG'],\n",
       " [8, 120486408, 120486418, 'ATCCATGTCCC'],\n",
       " [8, 110221247, 110221257, 'TTATGTTCACG'],\n",
       " [8, 134600046, 134600056, 'TGTGGATATCT'],\n",
       " [8, 12231237, 12231247, 'ATTTATAAGAT'],\n",
       " [8, 49739369, 49739379, 'CTCTCATTTCT'],\n",
       " [8, 27283731, 27283741, 'TGTTTAGAAGA'],\n",
       " [8, 126425288, 126425298, 'TAAGTCCTCAG'],\n",
       " [8, 51158350, 51158360, 'AACTTTCTTAT'],\n",
       " [8, 5132075, 5132085, 'TTTTCATTGTT'],\n",
       " [8, 53421698, 53421708, 'CAAATCAACAG'],\n",
       " [8, 9999037, 9999047, 'GAAACCCACTC'],\n",
       " [8, 19091220, 19091230, 'TGAAAAGGCAA'],\n",
       " [8, 42825257, 42825267, 'GAAGTAGGGAT'],\n",
       " [8, 31502442, 31502452, 'GATAAATGATA'],\n",
       " [8, 1008574, 1008584, 'AGATCAACTTG'],\n",
       " [8, 33776684, 33776694, 'GAAGGGATGTG'],\n",
       " [8, 9750636, 9750646, 'TTAAACACCAC'],\n",
       " [8, 39630800, 39630810, 'AGGTCAAGTTC'],\n",
       " [8, 6657293, 6657303, 'ACAGGCGCCTG'],\n",
       " [8, 62966145, 62966155, 'GAGACTTCCAT'],\n",
       " [8, 138874674, 138874684, 'ATGTTTCCAAG'],\n",
       " [8, 50539255, 50539265, 'ACATTCCTATA'],\n",
       " [8, 19802808, 19802818, 'CACTGAGACAT'],\n",
       " [8, 87669495, 87669505, 'GTCTTACCTTG'],\n",
       " [8, 114164712, 114164722, 'CAAAAAAAGTT'],\n",
       " [8, 62077766, 62077776, 'TCCGGAAGAAT'],\n",
       " [8, 117958640, 117958650, 'TCTTGTTCCAT'],\n",
       " [8, 80802058, 80802068, 'ATACATAAAAG'],\n",
       " [8, 47850830, 47850840, 'GCTGAGGCAGG'],\n",
       " [8, 37009803, 37009813, 'TTGGAAAGGGA'],\n",
       " [8, 120010588, 120010598, 'TTCAATTGGCT'],\n",
       " [8, 97917186, 97917196, 'GAGTGCTTTGA'],\n",
       " [8, 48015953, 48015963, 'TCTGGCACCTT'],\n",
       " [8, 35290705, 35290715, 'ACAAGAATCAA'],\n",
       " [8, 24247678, 24247688, 'AAATATTTGTG'],\n",
       " [8, 71206446, 71206456, 'ATTTTTCCCAG'],\n",
       " [8, 105637579, 105637589, 'GCTTTCTAAGT'],\n",
       " [8, 128188741, 128188751, 'TGGAATGTTTT'],\n",
       " [8, 132332092, 132332102, 'GCCTATAGTCC'],\n",
       " [8, 86979944, 86979954, 'ATAGGAACCTT'],\n",
       " [8, 137349518, 137349528, 'GTGAGGGAAGG'],\n",
       " [8, 4351890, 4351900, 'TAAAGTGAAAT'],\n",
       " [8, 109427907, 109427917, 'TTGGAATGTCA'],\n",
       " [8, 20843110, 20843120, 'TGTGAATCTGA'],\n",
       " [8, 110721634, 110721644, 'ATTAGCTTATA'],\n",
       " [8, 121165100, 121165110, 'AAACCTCGACG'],\n",
       " [8, 89630188, 89630198, 'CAGTGTCTCAG'],\n",
       " [8, 74039374, 74039384, 'CTAGTTACATC'],\n",
       " [8, 37128275, 37128285, 'AGCGTCCCGTG'],\n",
       " [8, 51733077, 51733087, 'TGTTGAGGTTT'],\n",
       " [8, 92952895, 92952905, 'TCTTACTCAGG'],\n",
       " [8, 5854564, 5854574, 'TCCACAGCCCT'],\n",
       " [8, 21026447, 21026457, 'CTGGTGTCCCC'],\n",
       " [8, 32726152, 32726162, 'GAGATTGAACT'],\n",
       " [8, 7206728, 7206738, 'CCTTCTGCATG'],\n",
       " [8, 12404268, 12404278, 'ACGTGGGTTTT'],\n",
       " [8, 14384533, 14384543, 'GAGACATCTCG'],\n",
       " [8, 121892665, 121892675, 'CTCAAAGGCTC'],\n",
       " [8, 23684563, 23684573, 'GGTTGTATTTA'],\n",
       " [8, 81885061, 81885071, 'CCAACAAAAGT'],\n",
       " [8, 75260330, 75260340, 'CATCCATGACT'],\n",
       " [8, 30680613, 30680623, 'AAGAAAAAAAA'],\n",
       " [8, 80984850, 80984860, 'ATGTACACATA'],\n",
       " [8, 100403897, 100403907, 'ATGCCAGCAGC'],\n",
       " [8, 9789579, 9789589, 'CTGGGAATCAG'],\n",
       " [8, 13888416, 13888426, 'AATTTAAACAG'],\n",
       " [8, 125601553, 125601563, 'TGAGCCACCGC'],\n",
       " [8, 128708117, 128708127, 'AGGTGGCAGTG'],\n",
       " [8, 98822262, 98822272, 'ATTCTCCCACC'],\n",
       " [8, 90887565, 90887575, 'TATTAATCACA'],\n",
       " [8, 73432997, 73433007, 'ACGAATGGATA'],\n",
       " [8, 47617235, 47617245, 'TTTTTTTTTTT'],\n",
       " [8, 128801000, 128801010, 'GTCTTGGCTTC'],\n",
       " [8, 1399888, 1399898, 'GCAACTCCATC'],\n",
       " [8, 37554528, 37554538, 'CCCTCGCTCGC'],\n",
       " [8, 38546195, 38546205, 'AACAGTGCTTT'],\n",
       " [8, 128705207, 128705217, 'TTTAGCAAACT'],\n",
       " [8, 60255975, 60255985, 'CCCGGGTTCAA'],\n",
       " [8, 38328558, 38328568, 'ACCAATTCTAC'],\n",
       " [8, 862220, 862230, 'CATGCGGGAGT'],\n",
       " [8, 11007827, 11007837, 'CATGTAACAAA'],\n",
       " [8, 67549806, 67549816, 'TGGCAGGTATA'],\n",
       " [8, 136706665, 136706675, 'TGTTAAAATTT'],\n",
       " [8, 80907984, 80907994, 'CCATACACTAA'],\n",
       " [8, 131786658, 131786668, 'AAAGCATTCAA'],\n",
       " [8, 27973541, 27973551, 'AGCAGTTCTTG'],\n",
       " [8, 58094088, 58094098, 'CTGGAGCCTGT'],\n",
       " [8, 90564186, 90564196, 'ATTAGGTCTTT'],\n",
       " [8, 113504829, 113504839, 'CTGTTGACCCT'],\n",
       " [8, 125926716, 125926726, 'TAAGGTCTGAA'],\n",
       " [8, 59426025, 59426035, 'CCCTTAGTGAT'],\n",
       " [8, 86997009, 86997019, 'GCTTGAATCCT'],\n",
       " [8, 38476290, 38476300, 'AAATTCTAGAA'],\n",
       " [8, 73323497, 73323507, 'CACACGTTTAC'],\n",
       " [8, 53530111, 53530121, 'AGCAAGCTGAA'],\n",
       " [8, 55748602, 55748612, 'TAGAGAGCATT'],\n",
       " [8, 74557876, 74557886, 'AATACCTGAGA'],\n",
       " [8, 84539481, 84539491, 'TCTTTGTTGGT'],\n",
       " [8, 21719652, 21719662, 'TGCTAATGATG'],\n",
       " [8, 75986226, 75986236, 'GTGAAACCACC'],\n",
       " [8, 105756682, 105756692, 'AACTTGGTTCC'],\n",
       " [8, 12670416, 12670426, 'AATGGCACACC'],\n",
       " [8, 63777166, 63777176, 'TCCACCTCGTT'],\n",
       " [8, 132690626, 132690636, 'CCTTAGACATG'],\n",
       " [8, 14589044, 14589054, 'AAAAATAATTT'],\n",
       " [8, 125878085, 125878095, 'TGCCATAAATC'],\n",
       " [8, 88522740, 88522750, 'AGGCTTTAAAT'],\n",
       " [8, 1023355, 1023365, 'TAGTTAATACG'],\n",
       " [8, 142546440, 142546450, 'CTTCCCTCACT'],\n",
       " [8, 32784789, 32784799, 'TAACTATCATT'],\n",
       " [8, 32435382, 32435392, 'GTGCCATCAGC'],\n",
       " [8, 90389218, 90389228, 'AGAACCAGAGA'],\n",
       " [8, 104066599, 104066609, 'GAATGGATTAG'],\n",
       " [8, 108045905, 108045915, 'GATCAGGAACA'],\n",
       " [8, 90155485, 90155495, 'AATAAATTTCC'],\n",
       " [8, 89275098, 89275108, 'AAAAAAATTAG'],\n",
       " [8, 118140111, 118140121, 'TTGGCAATGCT'],\n",
       " [8, 95916739, 95916749, 'TGTTTTTCTTC'],\n",
       " [8, 109971835, 109971845, 'GAGATAACTGT'],\n",
       " [8, 78796738, 78796748, 'CTCTTTCTTCA'],\n",
       " [8, 1394565, 1394575, 'TGTAGATATGT'],\n",
       " [8, 36651474, 36651484, 'AGGCAGGAGAA'],\n",
       " [8, 4256790, 4256800, 'CAACAAAGGAC'],\n",
       " [8, 5394599, 5394609, 'CACAATCAGGT'],\n",
       " [8, 15322603, 15322613, 'TGCCTTGGCTC'],\n",
       " [8, 130920966, 130920976, 'TGAGGATTCCT'],\n",
       " [8, 71277931, 71277941, 'CACACTAGAGA'],\n",
       " [8, 27062005, 27062015, 'GAACTCAACAC'],\n",
       " [8, 70232849, 70232859, 'TTACTCTCTAG'],\n",
       " [8, 102498749, 102498759, 'TATATTTTATT'],\n",
       " [8, 113725075, 113725085, 'AAACTGAACTT'],\n",
       " [8, 50020945, 50020955, 'GGCTAAAGAGA'],\n",
       " [8, 94161289, 94161299, 'AAGGTTTAAGA'],\n",
       " [8, 53431778, 53431788, 'CTTCCTCTGGA'],\n",
       " [8, 36553468, 36553478, 'GGCAAATAAGC'],\n",
       " [8, 119389809, 119389819, 'AACTGAATTCA'],\n",
       " [8, 124421517, 124421527, 'AATTCAGATGG'],\n",
       " [8, 77627592, 77627602, 'CTGTAGTCCCA'],\n",
       " [8, 40062937, 40062947, 'AGTCTGGGGTA'],\n",
       " [8, 56256208, 56256218, 'ATGCAGAGAAG'],\n",
       " [8, 93922702, 93922712, 'ACAAATGAAAA'],\n",
       " [8, 128055631, 128055641, 'TACACACAGGA'],\n",
       " [8, 50961207, 50961217, 'TTCACTTAATT'],\n",
       " [8, 19609244, 19609254, 'TGAGATCTGGT'],\n",
       " [8, 71352176, 71352186, 'AAGACAAGGTG'],\n",
       " [8, 116933748, 116933758, 'CTAATAGCAAA'],\n",
       " [8, 135997489, 135997499, 'GTAAACTGGAG'],\n",
       " [8, 39733243, 39733253, 'GCTCCTGCTGC'],\n",
       " [8, 88493917, 88493927, 'TCATTTTTGCA'],\n",
       " [8, 36821823, 36821833, 'GGTGATATCAT'],\n",
       " [8, 83892227, 83892237, 'TAATGAGTTCT'],\n",
       " [8, 70090800, 70090810, 'TGTGCCAGAGC'],\n",
       " [8, 135308708, 135308718, 'TCTGGTGAAGC'],\n",
       " [8, 106760953, 106760963, 'GAAGTACTAAA'],\n",
       " [8, 111967963, 111967973, 'CTAGAAAACCC'],\n",
       " [8, 986277, 986287, 'TAATGAACACT'],\n",
       " [8, 142189329, 142189339, 'TGCAAGGCCAG'],\n",
       " [8, 112999757, 112999767, 'GAGATTTTGTC'],\n",
       " [8, 39261448, 39261458, 'TTTATGGCCGC'],\n",
       " [8, 25504772, 25504782, 'CTCTGTGATAT'],\n",
       " [8, 55008239, 55008249, 'AAAAGGTATAA'],\n",
       " [8, 61138164, 61138174, 'GCCTGGTTCAG'],\n",
       " [8, 74945379, 74945389, 'GTGGGTTTTAC'],\n",
       " [8, 58123831, 58123841, 'AAGTATTGGGC'],\n",
       " [8, 123568767, 123568777, 'GGCAAGCATGG'],\n",
       " [8, 23855063, 23855073, 'AAACATAAGAT'],\n",
       " [8, 98378867, 98378877, 'ATATAAGAGAC'],\n",
       " [8, 43395133, 43395143, 'CTATATCGCCA'],\n",
       " [8, 50807198, 50807208, 'GGTAGATAGCC'],\n",
       " [8, 15406463, 15406473, 'TGTTTCTTTTA'],\n",
       " [8, 657954, 657964, 'AGACAATCCAA'],\n",
       " [8, 1778951, 1778961, 'ACACGGGGCTA'],\n",
       " [8, 20704397, 20704407, 'GATTTCAAAGT'],\n",
       " [8, 96288428, 96288438, 'ATAGCTTTTGA'],\n",
       " [8, 71763010, 71763020, 'ACGTATCTCAA'],\n",
       " [8, 133292709, 133292719, 'GTGCCTCATTT'],\n",
       " [8, 131535687, 131535697, 'ATCTCATCTTG'],\n",
       " [8, 98562651, 98562661, 'ATCAAATACCA'],\n",
       " [8, 8646210, 8646220, 'TGAGGAGACAT'],\n",
       " [8, 131917984, 131917994, 'TAGTCCCAGCT'],\n",
       " [8, 72416342, 72416352, 'GCTTCACCCTG'],\n",
       " [8, 9472857, 9472867, 'TCATGTTTATG'],\n",
       " [8, 104096378, 104096388, 'ACAACCTTTTA'],\n",
       " [8, 47522525, 47522535, 'CAGGAAAGATC'],\n",
       " [8, 38541563, 38541573, 'ACACACACAGA'],\n",
       " [8, 15792997, 15793007, 'AGGAGAAGGGC'],\n",
       " [8, 15796013, 15796023, 'TCTTCTGTACC'],\n",
       " [8, 140923653, 140923663, 'GGATAAGAGTA'],\n",
       " [8, 2918708, 2918718, 'CTCACCTGCAA'],\n",
       " [8, 110239310, 110239320, 'TTATACATTAC'],\n",
       " [8, 121320178, 121320188, 'AGGGATATGTA'],\n",
       " [8, 141679575, 141679585, 'CTTGGTTGTTC'],\n",
       " [8, 101814884, 101814894, 'TCAGCCTCCCA'],\n",
       " [8, 135365482, 135365492, 'ACTTCTGTCAA'],\n",
       " [8, 66639962, 66639972, 'GATTTCAATAA'],\n",
       " [8, 105561830, 105561840, 'CCGTGTTAGCC'],\n",
       " [8, 48568139, 48568149, 'TTTTTTTCTTG'],\n",
       " [8, 106770538, 106770548, 'TGGAGAATCTT'],\n",
       " [8, 8963065, 8963075, 'TACAGGCATTC'],\n",
       " [8, 84247179, 84247189, 'ATCTATGTCTA'],\n",
       " [8, 91007403, 91007413, 'TAGTATCAGGG'],\n",
       " [8, 274419, 274429, 'ATAAAATTTGC'],\n",
       " [8, 50465199, 50465209, 'ACCCCATAGTC'],\n",
       " [8, 62716207, 62716217, 'GAGAGAGAGAT'],\n",
       " [8, 140990456, 140990466, 'CCTCGGACACA'],\n",
       " [8, 6442796, 6442806, 'ACACGTGAAGC'],\n",
       " [8, 72173410, 72173420, 'TAAAAAACACA'],\n",
       " [8, 123706057, 123706067, 'CCATTAGGTCA'],\n",
       " [8, 7961377, 7961387, 'GTCTCTACTGA'],\n",
       " [8, 85496561, 85496571, 'CAGGCGTGGTG'],\n",
       " [8, 127005647, 127005657, 'TGGCATTCCAT'],\n",
       " [8, 14572863, 14572873, 'GGTAAAACCTA'],\n",
       " [8, 111009072, 111009082, 'GAGTGAAAGTT'],\n",
       " [8, 89606841, 89606851, 'GAAGCATGAAA'],\n",
       " [8, 100358019, 100358029, 'CTCAAGATCTG'],\n",
       " [8, 118792973, 118792983, 'CAGCCTGAGCG'],\n",
       " [8, 43360462, 43360472, 'ACAAGTAGAGT'],\n",
       " [8, 91223848, 91223858, 'GTGTCCCAACA'],\n",
       " [8, 9996933, 9996943, 'CATGGATCTGT'],\n",
       " [8, 117336389, 117336399, 'GAATACAATTC'],\n",
       " [8, 34846015, 34846025, 'ATCATAAACTG'],\n",
       " [8, 73905170, 73905180, 'CCAGCTACTTC'],\n",
       " [8, 24083522, 24083532, 'CCTAGCTCTCT'],\n",
       " [8, 41536702, 41536712, 'TGTCTGATTAC'],\n",
       " [8, 61244184, 61244194, 'CTGTAATCCAA'],\n",
       " [8, 79425877, 79425887, 'TTTCATGTAGT'],\n",
       " [8, 70142527, 70142537, 'ACTAAACAGTA'],\n",
       " [8, 113686516, 113686526, 'TAGGGCTGGGA'],\n",
       " [8, 86985696, 86985706, 'CCACATTCACC'],\n",
       " [8, 22028794, 22028804, 'GGAACCTAAAC'],\n",
       " [8, 97710737, 97710747, 'TAAGTCAAAAT'],\n",
       " [8, 97601419, 97601429, 'TCAATTTTCCA'],\n",
       " [8, 62798532, 62798542, 'TCATTCATTGT'],\n",
       " [8, 104387488, 104387498, 'AGAAGAAAAAG'],\n",
       " [8, 61873123, 61873133, 'CCCCTTGTGAA'],\n",
       " [8, 37670605, 37670615, 'GCCAGGCATGA'],\n",
       " [8, 112086859, 112086869, 'CTGGCCTGAAA'],\n",
       " [8, 14496433, 14496443, 'ATATGTAAACA'],\n",
       " [8, 21258630, 21258640, 'TTTGATTGTTG'],\n",
       " [8, 92754808, 92754818, 'CGGACAGATCT'],\n",
       " [8, 127035423, 127035433, 'TTGAAGGTCAC'],\n",
       " [8, 82485560, 82485570, 'AGTACCACGCT'],\n",
       " [8, 124121400, 124121410, 'GGAGAGAGGGG'],\n",
       " [8, 120704319, 120704329, 'GTTTCCACTTG'],\n",
       " [8, 65399123, 65399133, 'ACAAGAAGGAG'],\n",
       " [8, 103968954, 103968964, 'TCGCTCACTGC'],\n",
       " [8, 100316654, 100316664, 'ATAAGGCATTG'],\n",
       " [8, 108736695, 108736705, 'TTATCTTACAT'],\n",
       " [8, 41704499, 41704509, 'TCGCCTCTCAT'],\n",
       " [8, 47608016, 47608026, 'TTCTGTTATAA'],\n",
       " [8, 9536008, 9536018, 'TTAAACTTATA'],\n",
       " [8, 59998066, 59998076, 'GCCTCTCAAGT'],\n",
       " [8, 20274490, 20274500, 'GCCTCACTCCT'],\n",
       " [8, 135382177, 135382187, 'ATCTTTGTCTA'],\n",
       " [8, 122596656, 122596666, 'AATGTACATTT'],\n",
       " [8, 119336904, 119336914, 'ATTTGAATTGA'],\n",
       " [8, 61111636, 61111646, 'AAAAAAATCTT'],\n",
       " [8, 96400186, 96400196, 'AGTTTAAATAC'],\n",
       " [8, 120004903, 120004913, 'TTTTATCTCAT'],\n",
       " [8, 124116646, 124116656, 'TGAAGTCAAAA'],\n",
       " [8, 22091172, 22091182, 'ATGACTACGTC'],\n",
       " [8, 20638578, 20638588, 'GTATAAAATAG'],\n",
       " [8, 66192847, 66192857, 'CCTAAGGCAAC'],\n",
       " [8, 125085943, 125085953, 'CCCTGTTCTAT'],\n",
       " [8, 79078655, 79078665, 'TAAAATATTCT'],\n",
       " [8, 134377912, 134377922, 'TAGGACATCTT'],\n",
       " [8, 12464692, 12464702, 'CCAACTTCAAC'],\n",
       " [8, 24827895, 24827905, 'AAGACATTCTA'],\n",
       " [8, 85911109, 85911119, 'ATTTGACCCAG'],\n",
       " [8, 135409894, 135409904, 'TAAACAAGAAA'],\n",
       " [8, 58371218, 58371228, 'AGGACACATGG'],\n",
       " [8, 81060219, 81060229, 'GTTTAGAGACA'],\n",
       " [8, 94806613, 94806623, 'ATGTACCAGGC'],\n",
       " [8, 76975772, 76975782, 'TTGATCTTATA'],\n",
       " [8, 103770911, 103770921, 'ACACACCTCAA'],\n",
       " [8, 37249350, 37249360, 'CTCCATATCTG'],\n",
       " [8, 99400593, 99400603, 'GGGAGACTTCA'],\n",
       " [8, 122822277, 122822287, 'GTCAGGGTCAT'],\n",
       " [8, 39678007, 39678017, 'GGGCTATAAAA'],\n",
       " [8, 30628550, 30628560, 'AGGGGGTGAAT'],\n",
       " [8, 25979681, 25979691, 'AGAGGAGGGCA'],\n",
       " [8, 62798920, 62798930, 'ACAGGTTGTCT'],\n",
       " [8, 34207665, 34207675, 'CAAGTCATGAT'],\n",
       " [8, 136940382, 136940392, 'TTACAGGAGAC'],\n",
       " [8, 78289931, 78289941, 'TTTCGGGGAGC'],\n",
       " [8, 141830394, 141830404, 'TTCATCATGTT'],\n",
       " [8, 139479884, 139479894, 'AATTTGTATAA'],\n",
       " [8, 50710572, 50710582, 'ACTATTGTAAA'],\n",
       " [8, 362902, 362912, 'GTACACAGTAC'],\n",
       " [8, 141535150, 141535160, 'GCTGGGGGCTG'],\n",
       " [8, 15749523, 15749533, 'CGTGGGAGGAT'],\n",
       " [8, 21452535, 21452545, 'CTTTATCAACC'],\n",
       " [8, 121544089, 121544099, 'CAGCTCCTGTG'],\n",
       " [8, 39980331, 39980341, 'TTTTTATTTGA'],\n",
       " [8, 101493354, 101493364, 'TTCATTTTGGC'],\n",
       " [8, 55649065, 55649075, 'TGCCTGGGAAG'],\n",
       " [8, 29859468, 29859478, 'GGTTTTATCAT'],\n",
       " [8, 42496851, 42496861, 'TTCATTTTAAA'],\n",
       " [8, 109208879, 109208889, 'TTATAAGAAAG'],\n",
       " [8, 58892583, 58892593, 'ACCTCATCACT'],\n",
       " [8, 47494699, 47494709, 'CTGTACCTGGT'],\n",
       " [8, 131919201, 131919211, 'CCCAAATCATA'],\n",
       " [8, 98950203, 98950213, 'GTCTATTTCCG'],\n",
       " [8, 77800391, 77800401, 'TCTCTGAAAAA'],\n",
       " [8, 55919929, 55919939, 'CCATATTTTGT'],\n",
       " [8, 79576346, 79576356, 'AAAGCATAAAA'],\n",
       " [8, 66274258, 66274268, 'AAGAATATCTT'],\n",
       " [8, 141112347, 141112357, 'ACTCTGTGTTA'],\n",
       " [8, 81428358, 81428368, 'TCTGTCACCCA'],\n",
       " [8, 69296722, 69296732, 'AGACAAGATTA'],\n",
       " [8, 82377804, 82377814, 'GTCCTGATGCT'],\n",
       " [8, 109268371, 109268381, 'AAATATGGAAA'],\n",
       " [8, 81811585, 81811595, 'TGGCAGGCGCC'],\n",
       " [8, 117110479, 117110489, 'ACTTCCCAGTC'],\n",
       " [8, 23662343, 23662353, 'TATGTCTTGTC'],\n",
       " [8, 11912537, 11912547, 'AAGACCACTAG'],\n",
       " [8, 67276424, 67276434, 'GCTGAAGTATA'],\n",
       " [8, 18576911, 18576921, 'TTCTCAAGCCA'],\n",
       " [8, 138326919, 138326929, 'AAGTGTAGTTT'],\n",
       " [8, 142137065, 142137075, 'CCTGTAATCCC'],\n",
       " [8, 54573899, 54573909, 'AGTCTATGCTC'],\n",
       " [8, 87540712, 87540722, 'GCTTCTTAAAG'],\n",
       " [8, 108591941, 108591951, 'TGATATTACTC'],\n",
       " [8, 55768971, 55768981, 'ATTTTAAAACG'],\n",
       " [8, 101085303, 101085313, 'CTTTTTCTTGT'],\n",
       " [8, 96828323, 96828333, 'TATTAGTCTGT'],\n",
       " [8, 139249544, 139249554, 'CAGATGCCCGC'],\n",
       " [8, 103947043, 103947053, 'CAAGGCAGCAG'],\n",
       " [8, 85459353, 85459363, 'CTCTTGCCCTC'],\n",
       " [8, 18880288, 18880298, 'CATGTTCTTCT'],\n",
       " [8, 92856919, 92856929, 'TAAACTCATCT'],\n",
       " [8, 71708826, 71708836, 'TTATTGACTTG'],\n",
       " [8, 75075000, 75075010, 'ACCTGTTGCTA'],\n",
       " [8, 78122207, 78122217, 'CAAAGCCCAAT'],\n",
       " [8, 22596234, 22596244, 'CATACTTGTAG'],\n",
       " [8, 129743290, 129743300, 'TTTGGTTTTTG'],\n",
       " [8, 33422861, 33422871, 'TTAATCCAAGA'],\n",
       " [8, 65788500, 65788510, 'TCCTTTTTACT'],\n",
       " [8, 114488397, 114488407, 'ACTCATAGCAC'],\n",
       " [8, 109210870, 109210880, 'CATAAACACAT'],\n",
       " [8, 78392033, 78392043, 'TTTCTTCTATA'],\n",
       " [8, 81141404, 81141414, 'GCTTTTTTAGA'],\n",
       " [8, 105636909, 105636919, 'TTGTAACAAAC'],\n",
       " [8, 31963261, 31963271, 'TATATTATAAC'],\n",
       " [8, 106490576, 106490586, 'GCTATTTGGAT'],\n",
       " [8, 24204398, 24204408, 'TTTTGGTTATA'],\n",
       " [8, 126524224, 126524234, 'TGTGCCATGAA'],\n",
       " [8, 109689127, 109689137, 'GAAAGAATCAA'],\n",
       " [8, 90819955, 90819965, 'CCAAACCCGGC'],\n",
       " [8, 57109854, 57109864, 'TCCTGGCGTGA'],\n",
       " [8, 54147404, 54147414, 'TGCCTTCAAGG'],\n",
       " [8, 119379058, 119379068, 'AATAAAATAAA'],\n",
       " [8, 75659635, 75659645, 'ACTTTGAGAAA'],\n",
       " [8, 119236474, 119236484, 'CTTGACTAGTA'],\n",
       " [8, 90868831, 90868841, 'CTTATAGCAGG'],\n",
       " [8, 83783411, 83783421, 'TATGCAAATAT'],\n",
       " [8, 138831209, 138831219, 'GATATATAAAG'],\n",
       " [8, 96678289, 96678299, 'AAAGACGAATT'],\n",
       " [8, 129271436, 129271446, 'CCTGGAATTTT'],\n",
       " [8, 46962579, 46962589, 'TTGTAGATCTG'],\n",
       " [8, 27477554, 27477564, 'CGGCTTACTGT'],\n",
       " [8, 76469001, 76469011, 'TTTTTGTGCTA'],\n",
       " [8, 68997379, 68997389, 'TACCTGTGCCA'],\n",
       " [8, 8995042, 8995052, 'TGATGAGCTAT'],\n",
       " [8, 98901042, 98901052, 'AGGTTATAGGA'],\n",
       " [8, 74035158, 74035168, 'TTCAATTTTTA'],\n",
       " [8, 40999975, 40999985, 'CTGAAGAAACT'],\n",
       " [8, 121893861, 121893871, 'GATAGGTTGTT'],\n",
       " [8, 2399643, 2399653, 'GTGTCAACTTG'],\n",
       " [8, 132386174, 132386184, 'ACTCTGCTGCA'],\n",
       " [8, 95088948, 95088958, 'ATGTACTAAGC'],\n",
       " [8, 58646045, 58646055, 'TGTTCTTTCTT'],\n",
       " [8, 32665431, 32665441, 'AAACTGACCAT'],\n",
       " [8, 1781284, 1781294, 'AGAGAGACAGA'],\n",
       " [8, 86538035, 86538045, 'GTTTTAACTCC'],\n",
       " [8, 115117327, 115117337, 'CCTTCCTGTAA'],\n",
       " [8, 12356867, 12356877, 'AAAGATGACTA'],\n",
       " [8, 32863421, 32863431, 'TCAGAGGAAGA'],\n",
       " [8, 80413282, 80413292, 'AACATAGCACT'],\n",
       " [8, 75742977, 75742987, 'GGCGCATTTCC'],\n",
       " [8, 55402936, 55402946, 'TAGCAGGATAG'],\n",
       " [8, 27269509, 27269519, 'CCAAATGAAGA'],\n",
       " [8, 109333180, 109333190, 'TGTTTCCTTTT'],\n",
       " [8, 1624340, 1624350, 'TGCTGTACTTT'],\n",
       " [8, 112007818, 112007828, 'TTTTATAAAAA'],\n",
       " [8, 50758530, 50758540, 'TTTGGCTTCTG'],\n",
       " [8, 17957042, 17957052, 'CAGTCTTTCAT'],\n",
       " [8, 59919519, 59919529, 'AAAGAATAATG'],\n",
       " [8, 21725082, 21725092, 'GACCAGTCTGG'],\n",
       " [8, 13476063, 13476073, 'AAAGTGTTACA'],\n",
       " [8, 64786286, 64786296, 'ACCCAGGGCAG'],\n",
       " [8, 10490596, 10490606, 'ACGGCGACGTT'],\n",
       " [8, 43189670, 43189680, 'TCTGCCCACCT'],\n",
       " [8, 77901948, 77901958, 'CCTGTGTGCTT'],\n",
       " [8, 19166829, 19166839, 'CATTAGTATCC'],\n",
       " [8, 121746452, 121746462, 'GGTGGGCAGAT'],\n",
       " [8, 61149390, 61149400, 'TCTAATGCAGT'],\n",
       " [8, 116779016, 116779026, 'AGAAACCTGGG'],\n",
       " [8, 17035647, 17035657, 'TATGACTAAAA'],\n",
       " [8, 109549748, 109549758, 'GTGTGTGTGTG'],\n",
       " [8, 42966903, 42966913, 'CACCAATATCT'],\n",
       " [8, 25433658, 25433668, 'ATAAAACAAAA'],\n",
       " [8, 127500757, 127500767, 'CAGCAGTCACA'],\n",
       " [8, 83636985, 83636995, 'ACCTGTTTGTC'],\n",
       " [8, 76498297, 76498307, 'ATTTCCAATCT'],\n",
       " [8, 64548923, 64548933, 'ATGTTGGCTTC'],\n",
       " [8, 87826298, 87826308, 'TTGATTGAAGT'],\n",
       " [8, 31714922, 31714932, 'TAGGAAAGGGA'],\n",
       " [8, 49994675, 49994685, 'AAAAAGAAAGA'],\n",
       " [8, 105992133, 105992143, 'CTTTGCTATAT'],\n",
       " [8, 128148550, 128148560, 'AGTACTATGGC'],\n",
       " [8, 127748551, 127748561, 'GCCACATGTGC'],\n",
       " [8, 66851584, 66851594, 'TTGATTTTACT'],\n",
       " [8, 4101709, 4101719, 'TAGGCTTAGCT'],\n",
       " [8, 129248659, 129248669, 'CAGGAGCTGGG'],\n",
       " [8, 59647472, 59647482, 'ATTTAAACAAT'],\n",
       " [8, 24452126, 24452136, 'TTTTTGCAAAC'],\n",
       " [8, 15924098, 15924108, 'ATTCAACACAT'],\n",
       " [8, 32816138, 32816148, 'ATACTTGGGTT'],\n",
       " [8, 139064786, 139064796, 'AGCCTCCACCT'],\n",
       " [8, 118994713, 118994723, 'TATTTAGGGCT'],\n",
       " [8, 68766372, 68766382, 'GGGTAGACTCA'],\n",
       " [8, 8918869, 8918879, 'GACTCAACAAA'],\n",
       " [8, 141264734, 141264744, 'AAAGGTCCTCC'],\n",
       " [8, 3974776, 3974786, 'CTATAATCCCA'],\n",
       " [8, 6947798, 6947808, 'GTGTGCCTATG'],\n",
       " [8, 73096987, 73096997, 'GTGCAGTTTCA'],\n",
       " [8, 134236746, 134236756, 'GAAGCCTGTGT'],\n",
       " [8, 34874251, 34874261, 'TATTAGAGAAA'],\n",
       " [8, 67721127, 67721137, 'AGAGGCTCTCC'],\n",
       " [8, 130186420, 130186430, 'GCAGCCCATTC'],\n",
       " [8, 68626558, 68626568, 'GAAGGGTGGTG'],\n",
       " [8, 122417738, 122417748, 'AAAAGTTATGA'],\n",
       " [8, 9289703, 9289713, 'CTGCATTTTTA'],\n",
       " [8, 133679435, 133679445, 'AAAAGATAATG'],\n",
       " [8, 129012920, 129012930, 'TAGCAATACTT'],\n",
       " [8, 128958498, 128958508, 'TATATCAGACA'],\n",
       " [8, 60400494, 60400504, 'TTCGTTAGACA'],\n",
       " [8, 107654199, 107654209, 'CATATTGCCCT'],\n",
       " [8, 40265155, 40265165, 'CTTGAGGCTAG'],\n",
       " [8, 85206852, 85206862, 'AAGGAAGGTCG'],\n",
       " [8, 74825403, 74825413, 'TGCCTGGCATT'],\n",
       " [8, 91144594, 91144604, 'GTTTTAAGCTA'],\n",
       " [8, 119467627, 119467637, 'TTGGTAAAATC'],\n",
       " [8, 85363069, 85363079, 'TACTCAATATA'],\n",
       " [8, 20321319, 20321329, 'CTGGGATTACA'],\n",
       " [8, 21651837, 21651847, 'ACCACAATCTC'],\n",
       " [8, 10027452, 10027462, 'TTTCACTCTTG'],\n",
       " [8, 121281092, 121281102, 'AATTTTCATTT'],\n",
       " [8, 60916, 60926, 'TGGTGAAGATT'],\n",
       " [8, 134450876, 134450886, 'TTGTTTATTAT'],\n",
       " [8, 35596554, 35596564, 'CCCAGGTAACA'],\n",
       " [8, 139205438, 139205448, 'CCACTTACCAG'],\n",
       " [8, 23356304, 23356314, 'AACATGGTGAA'],\n",
       " [8, 96661422, 96661432, 'ATCAAAAATAG'],\n",
       " [8, 101908855, 101908865, 'TGTCAAAGGAT'],\n",
       " [8, 108084307, 108084317, 'TTGAAGTAGGT'],\n",
       " [8, 49625309, 49625319, 'CAGCACATTGG'],\n",
       " [8, 126415315, 126415325, 'AAAAAAAATTA'],\n",
       " [8, 61130307, 61130317, 'AGGGGAGGTCC'],\n",
       " [8, 114881790, 114881800, 'TAATAAACTTT'],\n",
       " [8, 138249121, 138249131, 'TTTTACATGTA'],\n",
       " [8, 124025788, 124025798, 'ACGTCATCAGC'],\n",
       " [8, 87989392, 87989402, 'ACCTATGGCCT'],\n",
       " [8, 107619866, 107619876, 'TGCTTAGCATT'],\n",
       " [8, 136859653, 136859663, 'GTGCCATGTTG'],\n",
       " [8, 85261312, 85261322, 'GGTCCAGGGAA'],\n",
       " [8, 105582002, 105582012, 'TCCTACTCAGT'],\n",
       " [8, 139047310, 139047320, 'TTTGGGCATCT'],\n",
       " [8, 105046603, 105046613, 'TGCTTTTTTTA'],\n",
       " [8, 98357028, 98357038, 'ATAAGGCTAAA'],\n",
       " [8, 17576920, 17576930, 'AAAGCAATATC'],\n",
       " [8, 40527709, 40527719, 'AAGACAAAGAG'],\n",
       " [8, 24493649, 24493659, 'TTTATTTGGAG'],\n",
       " [8, 60618426, 60618436, 'CCCTCAGAAAT'],\n",
       " [8, 98287777, 98287787, 'CTACTATGGCA'],\n",
       " [8, 114788656, 114788666, 'TTTACATTTCA'],\n",
       " [8, 103363504, 103363514, 'TAGAAGCTACC'],\n",
       " [8, 79141633, 79141643, 'TATGACAGTTT'],\n",
       " [8, 122758434, 122758444, 'TAATCCCAGCT'],\n",
       " [8, 105218724, 105218734, 'ATAATATAAAA'],\n",
       " [8, 119328732, 119328742, 'GGCACTGCACT'],\n",
       " [8, 126189512, 126189522, 'AATTTCTGGGT'],\n",
       " [8, 91231569, 91231579, 'ACGAACATTCA'],\n",
       " [8, 115468107, 115468117, 'ACATTTGTATC'],\n",
       " [8, 47115846, 47115856, 'GGAGTTTCCCA'],\n",
       " [8, 61645207, 61645217, 'ACATGGTGTTA'],\n",
       " [8, 93770805, 93770815, 'AAATAGGATGA'],\n",
       " [8, 47787678, 47787688, 'GTGCTTCTTTA'],\n",
       " [8, 76303662, 76303672, 'AAACAGTTTTC'],\n",
       " [8, 89672322, 89672332, 'TTTGCTAAATA'],\n",
       " [8, 85297126, 85297136, 'TATTTTTAAGG'],\n",
       " [8, 60575833, 60575843, 'TCTGCTCCCAT'],\n",
       " [8, 85439372, 85439382, 'AAGTCGGTGGT'],\n",
       " [8, 102107834, 102107844, 'TCAATCAGATT'],\n",
       " [8, 52393749, 52393759, 'GATAGTGACAT'],\n",
       " [8, 102291928, 102291938, 'GAGTCTTAAGA'],\n",
       " [8, 34596697, 34596707, 'CCTCCGAGCCA'],\n",
       " [8, 108605671, 108605681, 'TACGTTCCATC'],\n",
       " [8, 137715676, 137715686, 'CACATCACAGT'],\n",
       " [8, 31894764, 31894774, 'TCCCACACCTG'],\n",
       " [8, 68561343, 68561353, 'TTTTTTTTTCA'],\n",
       " [8, 27925829, 27925839, 'TGGTAGAAACA'],\n",
       " [8, 132448722, 132448732, 'GATGAGAATGC'],\n",
       " [8, 42140846, 42140856, 'TCCTAATCCCT'],\n",
       " [8, 73719888, 73719898, 'AAACTGCTCAG'],\n",
       " [8, 122694684, 122694694, 'CTATTCAGATC'],\n",
       " [8, 31709597, 31709607, 'GTCTTTATAGC'],\n",
       " [8, 97892786, 97892796, 'TATATATATAT'],\n",
       " [8, 19803611, 19803621, 'AAGGCTACCAA'],\n",
       " [8, 39583373, 39583383, 'TTTGAATTCAT'],\n",
       " [8, 82982147, 82982157, 'TCACAAATAAA'],\n",
       " [8, 61939049, 61939059, 'TTCCATTATTT'],\n",
       " [8, 73029418, 73029428, 'CTTTCTGTATT'],\n",
       " [8, 124657164, 124657174, 'GAGTCTGGGTT'],\n",
       " [8, 8182057, 8182067, 'AGTGGCTTTTT'],\n",
       " [8, 50807512, 50807522, 'CACTTTTTGAT'],\n",
       " [8, 141946220, 141946230, 'ATTGCCCTCCA'],\n",
       " [8, 43706408, 43706418, 'CTATCTGTGAA'],\n",
       " [8, 132820181, 132820191, 'ATTCTGTTCAC'],\n",
       " [8, 77018754, 77018764, 'CTCTCTTAGAT'],\n",
       " [8, 24024481, 24024491, 'CTATGTACTGC'],\n",
       " [8, 95138303, 95138313, 'TTTCAGGTACA'],\n",
       " [8, 123684204, 123684214, 'GGGCTAAAATA'],\n",
       " [8, 112112677, 112112687, 'AGCCACCCAGG'],\n",
       " [8, 43268144, 43268154, 'TTGTTTCCTGA'],\n",
       " [8, 95690578, 95690588, 'ATTACACAGCG'],\n",
       " [8, 70237683, 70237693, 'GCAGCTGGAGA'],\n",
       " [8, 116398951, 116398961, 'TTTATCACAGG'],\n",
       " [8, 80919741, 80919751, 'TGTATATTTAG'],\n",
       " [8, 21098286, 21098296, 'CTGTTTTGTTT'],\n",
       " [8, 105855308, 105855318, 'CAGGACCGTAC'],\n",
       " [8, 88464967, 88464977, 'CACACACACAC'],\n",
       " [8, 109503253, 109503263, 'GACTTAAAGAG'],\n",
       " [8, 111550265, 111550275, 'CTGTGGCAAAC'],\n",
       " [8, 130128571, 130128581, 'GCCTAACCTAG'],\n",
       " [8, 42291824, 42291834, 'TGAACGCCCGA'],\n",
       " [8, 48445070, 48445080, 'ATTTTAGAGCA'],\n",
       " [8, 39878944, 39878954, 'TTCCGTTGCTG'],\n",
       " [8, 114323470, 114323480, 'GAAATTATCTA'],\n",
       " [8, 29715453, 29715463, 'CAGTGAGCTAT'],\n",
       " [8, 89155080, 89155090, 'GGGATGTTAAA'],\n",
       " [8, 12770130, 12770140, 'TACTTGGTTAG'],\n",
       " [8, 132912479, 132912489, 'AGGTCATAGTC'],\n",
       " [8, 65980103, 65980113, 'TCTTAAATTCT'],\n",
       " [8, 14522596, 14522606, 'ATGCCATGTTG'],\n",
       " [8, 109905689, 109905699, 'ATCCACATAGA'],\n",
       " [8, 18527675, 18527685, 'GAGTGACCGGG'],\n",
       " [8, 103515552, 103515562, 'ACCAGAAGAAA'],\n",
       " [8, 102759615, 102759625, 'TTTGATGATAT'],\n",
       " [8, 30053394, 30053404, 'AATAAAGAACA'],\n",
       " [8, 50142580, 50142590, 'TGAGCGGAGAA'],\n",
       " [8, 27795571, 27795581, 'TGGGGTGGGCA'],\n",
       " [8, 20313982, 20313992, 'GGTCTCACTAT'],\n",
       " [8, 101016471, 101016481, 'CTGGTAAAGAA'],\n",
       " [8, 82495476, 82495486, 'TATCAAAAAAG'],\n",
       " [8, 10105528, 10105538, 'GCCTCTGGTGC'],\n",
       " [8, 62919732, 62919742, 'TTCTGACAGTT'],\n",
       " [8, 35116938, 35116948, 'GAATTGGTTCC'],\n",
       " [8, 137428723, 137428733, 'TTGCTCCTCGC'],\n",
       " [8, 15319467, 15319477, 'ATAAGTGCTTC'],\n",
       " [8, 34235332, 34235342, 'AAAATTAGCCG'],\n",
       " [8, 128512621, 128512631, 'TGAGTATGTAA'],\n",
       " [8, 98834834, 98834844, 'GTCTAAATTAG'],\n",
       " [8, 8788578, 8788588, 'TACAATCAAGT'],\n",
       " [8, 49477482, 49477492, 'AGGGGGAAGGG'],\n",
       " [8, 84941964, 84941974, 'GGTCCCCAGTG'],\n",
       " [8, 10812343, 10812353, 'GTATGATGTAA'],\n",
       " [8, 101401767, 101401777, 'TATTATTTGAA'],\n",
       " [8, 89545918, 89545928, 'ACCCATGAGCA'],\n",
       " [8, 125302281, 125302291, 'ATTGCATTTCA'],\n",
       " [8, 115069216, 115069226, 'TATAGACAAAA'],\n",
       " [8, 127330154, 127330164, 'CCATCAGCTCC'],\n",
       " [8, 6294448, 6294458, 'ACCACTGCACT'],\n",
       " [8, 135170444, 135170454, 'TTAAGTGGTCG'],\n",
       " [8, 138548728, 138548738, 'GAGTGAGACCC'],\n",
       " [8, 47922167, 47922177, 'TCATGTCCTCA'],\n",
       " [8, 98518532, 98518542, 'AGAAAAGGGGA'],\n",
       " [8, 102334636, 102334646, 'GGGACACAGAG'],\n",
       " [8, 11354538, 11354548, 'AAACCTGGCCC'],\n",
       " [8, 104501118, 104501128, 'CTGGGATACTG'],\n",
       " [8, 61475881, 61475891, 'AGCCTTGCTCC'],\n",
       " [8, 83707627, 83707637, 'TTATCTATTCA'],\n",
       " [8, 23626236, 23626246, 'CAATGTTTCTT'],\n",
       " [8, 82690127, 82690137, 'AACACAGGAAC'],\n",
       " [8, 114342665, 114342675, 'ATGACAAGAAT'],\n",
       " [8, 26726591, 26726601, 'GACATCCCAGA'],\n",
       " [8, 52320196, 52320206, 'AACTACCACAA'],\n",
       " [8, 6340376, 6340386, 'TAGACTCCTTG'],\n",
       " [8, 113866420, 113866430, 'GATTATATTTT'],\n",
       " [8, 121278156, 121278166, 'GTCAGTCTGCC'],\n",
       " [8, 112232698, 112232708, 'TCTAAAATAAA'],\n",
       " [8, 3910942, 3910952, 'CACCTGCTTTT'],\n",
       " [8, 104964994, 104965004, 'ACGAATTGACA'],\n",
       " [8, 139328271, 139328281, 'AGAAAAATTTA'],\n",
       " [8, 58480702, 58480712, 'GGTTCAACATA'],\n",
       " [8, 32053718, 32053728, 'GTTTGAGTGCC'],\n",
       " [8, 127142556, 127142566, 'ACCCAGTCTTG'],\n",
       " [8, 68932625, 68932635, 'TTTGAGAATAC'],\n",
       " [8, 94782892, 94782902, 'CATCACAACAA'],\n",
       " [8, 48434758, 48434768, 'AAAAAAGAAAC'],\n",
       " [8, 3476315, 3476325, 'CTCTCCCTTGT'],\n",
       " [8, 85809210, 85809220, 'TGAAATGTACC'],\n",
       " [8, 107543267, 107543277, 'CAAAATATTAC'],\n",
       " [8, 48356289, 48356299, 'TCTGTGTTCAT'],\n",
       " [8, 42093324, 42093334, 'TTTGTTTGTGC'],\n",
       " [8, 36594024, 36594034, 'CTATTAATTAT'],\n",
       " [8, 9851139, 9851149, 'CTACAGTAACC'],\n",
       " [8, 72007961, 72007971, 'TCTATTTGTTA'],\n",
       " [8, 97166212, 97166222, 'AGCTTCGGAGC'],\n",
       " [8, 22896807, 22896817, 'GGGGTCAGGGG'],\n",
       " [8, 5427628, 5427638, 'CTTGTATTATT'],\n",
       " [8, 69426647, 69426657, 'TATTATCATGC'],\n",
       " [8, 75454770, 75454780, 'ATTGAAAGGAA'],\n",
       " [8, 31092830, 31092840, 'TAGCTGGTTAT'],\n",
       " [8, 30115824, 30115834, 'CTGGCCTGTCT'],\n",
       " [8, 64680220, 64680230, 'AATTTTTTGGA'],\n",
       " [8, 133502005, 133502015, 'TGGTAACGGCA'],\n",
       " [8, 83631738, 83631748, 'ATGTGGGACCA'],\n",
       " [8, 95375730, 95375740, 'TACCTGGGAAA'],\n",
       " [8, 23793781, 23793791, 'TGTATCTACCT'],\n",
       " [8, 132232207, 132232217, 'CATTGTCTGTT'],\n",
       " [8, 83424843, 83424853, 'TATTTGCTGAC'],\n",
       " [8, 142761922, 142761932, 'TTTTTTTCTCC'],\n",
       " [8, 61035693, 61035703, 'ACTTTTATAAT'],\n",
       " [8, 111778464, 111778474, 'AGCTTAATTTA'],\n",
       " [8, 1647542, 1647552, 'CAGGTGGATCA'],\n",
       " [8, 47177643, 47177653, 'GATTTCTTGTT'],\n",
       " [8, 11174608, 11174618, 'AATTAGTCCAT'],\n",
       " [8, 69809254, 69809264, 'TATCTATTTTC'],\n",
       " [8, 32821271, 32821281, 'AAAACAGACTT'],\n",
       " [8, 119238729, 119238739, 'TTCTTTAACAC'],\n",
       " [8, 34893972, 34893982, 'CCATAACTACT'],\n",
       " [8, 111458013, 111458023, 'GTCCACGGGAT'],\n",
       " [8, 5598647, 5598657, 'TTCGAAAATTA'],\n",
       " [8, 124295876, 124295886, 'GCACATGCCTG'],\n",
       " [8, 109486512, 109486522, 'TGAATTTTTGG'],\n",
       " [8, 84600782, 84600792, 'AGGATCAAGAG'],\n",
       " [8, 84816473, 84816483, 'CAATTAATAAT'],\n",
       " [8, 65802201, 65802211, 'TTTAACTCTCT'],\n",
       " [8, 129584112, 129584122, 'TCTACAATTAT'],\n",
       " [8, 136639306, 136639316, 'TTTCCGAATGC'],\n",
       " [8, 4357236, 4357246, 'GAAGTTGATAT'],\n",
       " [8, 118079860, 118079870, 'ATCTTGCAAAT'],\n",
       " [8, 80155720, 80155730, 'TTATTGCTTGT'],\n",
       " [8, 4955423, 4955433, 'AGCTCTTTCAT'],\n",
       " [8, 126749331, 126749341, 'TGGAGCTGCTC'],\n",
       " [8, 57429637, 57429647, 'GTAAATCCTGA'],\n",
       " [8, 96691621, 96691631, 'GGGCCAAATTC'],\n",
       " [8, 99634983, 99634993, 'TATATTTTTAG'],\n",
       " [8, 58762413, 58762423, 'TGAATTTCAGA'],\n",
       " [8, 98451628, 98451638, 'ATAATTCTAAT'],\n",
       " [8, 140265898, 140265908, 'CATCCGGAGAT'],\n",
       " [8, 117000643, 117000653, 'GAAGAGGCCAG'],\n",
       " [8, 41396425, 41396435, 'AGGCTGATCTC'],\n",
       " [8, 16112465, 16112475, 'TCAAGTCCACA'],\n",
       " [8, 68351502, 68351512, 'TGGGTGACAGA'],\n",
       " [8, 124979564, 124979574, 'TTCACAAAACA'],\n",
       " [8, 99610389, 99610399, 'TGTAAACTGGT'],\n",
       " [8, 74299760, 74299770, 'ATATCTGGCAA'],\n",
       " [8, 84455620, 84455630, 'GAAGAGAGTTG'],\n",
       " [8, 114976098, 114976108, 'TTAAGCTAAAA'],\n",
       " [8, 80300301, 80300311, 'AGTATATGCGT'],\n",
       " [8, 32929632, 32929642, 'GAACTTTAATT'],\n",
       " [8, 17585344, 17585354, 'TTCTGTAATTA'],\n",
       " [8, 55864320, 55864330, 'ATTATTGTAAA'],\n",
       " [8, 70065488, 70065498, 'TGTCTTAAATA'],\n",
       " [8, 34594113, 34594123, 'ATGCTATATCT'],\n",
       " [8, 66563945, 66563955, 'GTATAATTTGA'],\n",
       " [8, 32908722, 32908732, 'CCACCGTTCCT'],\n",
       " [8, 57854723, 57854733, 'TGCCTTAAAAA'],\n",
       " [8, 9126276, 9126286, 'AGCCTCCTCTA'],\n",
       " [8, 30128615, 30128625, 'TGCCATGTTGT'],\n",
       " [8, 42781259, 42781269, 'TTAACCAAAAC'],\n",
       " [8, 104265430, 104265440, 'TGATAATCTTT'],\n",
       " [8, 6088489, 6088499, 'TATTTTCAACC'],\n",
       " [8, 85894976, 85894986, 'CAAAATAGGCT'],\n",
       " [8, 18945998, 18946008, 'CATCAGATCTT'],\n",
       " [8, 114878053, 114878063, 'CTAACTTTTAC'],\n",
       " [8, 89154802, 89154812, 'TATTGTTTATA'],\n",
       " [8, 39326205, 39326215, 'ATCCACCTTGA'],\n",
       " [8, 92427492, 92427502, 'ACGTAATCTTC'],\n",
       " [8, 48368222, 48368232, 'TTGATAGCGAA'],\n",
       " [8, 1068110, 1068120, 'GACAGAGGAGC'],\n",
       " [8, 90749477, 90749487, 'GAACCCTCACA'],\n",
       " [8, 130991658, 130991668, 'AGGTTCTTTTA'],\n",
       " [8, 41535919, 41535929, 'AAAAGTGCTGG'],\n",
       " [8, 31374711, 31374721, 'GGCTTCTCTAG'],\n",
       " [8, 16562661, 16562671, 'GAAACCAACTC'],\n",
       " [8, 27442790, 27442800, 'AGTTCCTTCCT'],\n",
       " [8, 20203536, 20203546, 'TGGCTGCAATC'],\n",
       " [8, 109266756, 109266766, 'AGTTTACATAA'],\n",
       " [8, 18829040, 18829050, 'TGATGGCCAGA'],\n",
       " [8, 55028147, 55028157, 'TGTGGAAGCAT'],\n",
       " [8, 34661163, 34661173, 'AGTCCCATTTT'],\n",
       " [8, 23923246, 23923256, 'CCTTGTCTGCT'],\n",
       " [8, 51196018, 51196028, 'CTGATGAATGT'],\n",
       " [8, 32672849, 32672859, 'ATATACAATAT'],\n",
       " [8, 14128914, 14128924, 'ATCTCCATGAA'],\n",
       " [8, 36230384, 36230394, 'AGATGAAATTT'],\n",
       " [8, 114910920, 114910930, 'AGTCTCTTTTG'],\n",
       " [8, 37867735, 37867745, 'CCACCATGATT'],\n",
       " [8, 110797909, 110797919, 'ACCTGCACATC'],\n",
       " [8, 111663211, 111663221, 'TTTTGGTTGAT'],\n",
       " [8, 139000264, 139000274, 'TGTTTTGTTTT'],\n",
       " [8, 77673249, 77673259, 'GGTGGGGTTCA'],\n",
       " [8, 35881607, 35881617, 'TCAGCTCACTG'],\n",
       " [8, 90992574, 90992584, 'TTGGAGAAAGA'],\n",
       " [8, 102007833, 102007843, 'GATAAAAATCC'],\n",
       " [8, 14259231, 14259241, 'AATGTGAATAA'],\n",
       " [8, 141144731, 141144741, 'GGAGTTCCACA'],\n",
       " [8, 98306608, 98306618, 'GGTGGTAAAAT'],\n",
       " [8, 21596217, 21596227, 'CTACTCCTCCT'],\n",
       " [8, 19956084, 19956094, 'AAAATTTCAAA'],\n",
       " [8, 19922923, 19922933, 'GGTTGGGTGTG'],\n",
       " [8, 136277339, 136277349, 'ACCCTAAAACT'],\n",
       " [8, 48585291, 48585301, 'CTGGTGTTACT'],\n",
       " [8, 27174407, 27174417, 'TTATGATCTTT'],\n",
       " [8, 106324696, 106324706, 'AACTTTCTCAG'],\n",
       " [8, 5588898, 5588908, 'GTAATAATCTA'],\n",
       " [8, 129423339, 129423349, 'AAGAGGGGAAG'],\n",
       " [8, 129274335, 129274345, 'TTAAGCACTTG'],\n",
       " [8, 75270068, 75270078, 'TTCTTGCTTTT'],\n",
       " [8, 3836646, 3836656, 'TGAAACATGGG'],\n",
       " [8, 56247664, 56247674, 'GGTCTAAACTC'],\n",
       " [8, 92291830, 92291840, 'CAGGCTGGACT'],\n",
       " [8, 66216488, 66216498, 'AAACTTGTTTT'],\n",
       " [8, 83128911, 83128921, 'AGTGAAGTGAA'],\n",
       " [8, 95792014, 95792024, 'ATGACTGGATC'],\n",
       " [8, 49909517, 49909527, 'TTCTTTTAATT'],\n",
       " [8, 50083524, 50083534, 'ATGTATTTTTT'],\n",
       " [8, 118548752, 118548762, 'GAAATTGTGCA'],\n",
       " [8, 43698977, 43698987, 'GGGCAAAACTG'],\n",
       " [8, 86522292, 86522302, 'AAGGGAAAAGA'],\n",
       " [8, 57379926, 57379936, 'CTTATTATTAC'],\n",
       " [8, 32266454, 32266464, 'GTCCATGTGTA'],\n",
       " [8, 71284982, 71284992, 'AAAAAGTGCCC'],\n",
       " [8, 92519006, 92519016, 'GGATATTTGAA'],\n",
       " [8, 27643976, 27643986, 'TGGGTGGCAGA'],\n",
       " [8, 134981039, 134981049, 'AGGTGAAGGGG'],\n",
       " [8, 117607809, 117607819, 'AAAAAAAAGCC'],\n",
       " [8, 30267815, 30267825, 'CCTGGGTTCAA'],\n",
       " [8, 103600299, 103600309, 'TCTCTGAGCAG'],\n",
       " [8, 113395266, 113395276, 'AACATCTTACA'],\n",
       " [8, 14859678, 14859688, 'GCTTCAGGAAT'],\n",
       " [8, 23055839, 23055849, 'GAGGAGTTCGT'],\n",
       " [8, 116286091, 116286101, 'ACACGTTGCTT'],\n",
       " [8, 109765723, 109765733, 'GCCCTTTGGAT'],\n",
       " [8, 41644982, 41644992, 'CCTGACAGGGG'],\n",
       " [8, 109215879, 109215889, 'ATTTGTTCATG'],\n",
       " [8, 52753308, 52753318, 'TAAAATACAAA'],\n",
       " [8, 57009863, 57009873, 'ACTTCCAAGAG'],\n",
       " [8, 73645436, 73645446, 'ACATAGTGTGT'],\n",
       " [8, 119097055, 119097065, 'TCCGTATTAAT'],\n",
       " [8, 69795242, 69795252, 'TAGACCAAATA'],\n",
       " [8, 20815684, 20815694, 'GAGAAATAAAA'],\n",
       " [8, 92256782, 92256792, 'AATGCAGTGGC'],\n",
       " [8, 139358064, 139358074, 'CTGCCTCAGTT'],\n",
       " [8, 119184124, 119184134, 'AATGATAGACT'],\n",
       " [8, 93461586, 93461596, 'ATTGTTCACAC'],\n",
       " [8, 46979961, 46979971, 'TTTTGTGGAAT'],\n",
       " [8, 119489602, 119489612, 'ACCTCTGCGGC'],\n",
       " [8, 142488379, 142488389, 'ACGATGGACGT'],\n",
       " [8, 100107869, 100107879, 'CAGCAATTTGG'],\n",
       " [8, 41057662, 41057672, 'AGAGGATTGCT'],\n",
       " [8, 36997415, 36997425, 'ATATCTCTAAT'],\n",
       " [8, 83635368, 83635378, 'CAGATGTTATA'],\n",
       " [8, 82284553, 82284563, 'AGAAAATGGGG'],\n",
       " [8, 101983169, 101983179, 'ATTGGGTTTTA'],\n",
       " [8, 124333185, 124333195, 'TCAGTTATCTT'],\n",
       " [8, 39556668, 39556678, 'CTTTGAAAAGA'],\n",
       " [8, 29090611, 29090621, 'TGCAGTGAGCA'],\n",
       " [8, 133491326, 133491336, 'AAGGCAGAGCT'],\n",
       " [8, 106401699, 106401709, 'CTCTATGCCTA'],\n",
       " [8, 76855795, 76855805, 'AAATCTGGGTA'],\n",
       " [8, 18678664, 18678674, 'CAAAACAAGGT'],\n",
       " [8, 137283022, 137283032, 'CAAGAGTTTGT'],\n",
       " [8, 78755267, 78755277, 'CTATTTTAGTA'],\n",
       " [8, 37581451, 37581461, 'ACGGATGAGTG'],\n",
       " [8, 98269047, 98269057, 'TTGCATACCTT'],\n",
       " [8, 22390872, 22390882, 'ATGATTAAGAG'],\n",
       " [8, 41851648, 41851658, 'TATTGTTCCCA'],\n",
       " [8, 125816255, 125816265, 'GCCTGTGCCTT'],\n",
       " [8, 31321767, 31321777, 'TGATGACAGTG'],\n",
       " [8, 26986667, 26986677, 'TGTTGAATTTT'],\n",
       " [8, 131013786, 131013796, 'TTTTTTTGAGA']]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bgseqs_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "([array([1045.,  442.,  251.,  105.,  118.,   68.,  139.,   60.,   88.,\n",
       "           85.,   54.,   37.,   56.,   41.,   42.,   58.,   50.,   22.,\n",
       "           57.,   23.,   16.,   68.,   42.,   46.,   34.,   19.,   38.,\n",
       "           38.,   52.,   15.,   16.,    9.,   33.,   32.,    5.,   55.,\n",
       "           11.,   54.,   40.,   18.,   40.,   36.,   38.,   35.,   25.,\n",
       "           59.,   68.,   84.,  351.]),\n",
       "  array([1034.,  443.,  257.,   99.,  119.,   63.,  140.,   48.,   89.,\n",
       "           90.,   48.,   42.,   59.,   40.,   46.,   62.,   47.,   24.,\n",
       "           60.,   27.,   15.,   66.,   41.,   49.,   35.,   20.,   35.,\n",
       "           42.,   51.,   15.,   17.,   11.,   35.,   31.,    6.,   49.,\n",
       "           11.,   52.,   42.,   19.,   41.,   34.,   42.,   33.,   23.,\n",
       "           64.,   65.,   79.,  358.])],\n",
       " array([0.        , 0.02040816, 0.04081633, 0.06122449, 0.08163265,\n",
       "        0.10204082, 0.12244898, 0.14285714, 0.16326531, 0.18367347,\n",
       "        0.20408163, 0.2244898 , 0.24489796, 0.26530612, 0.28571429,\n",
       "        0.30612245, 0.32653061, 0.34693878, 0.36734694, 0.3877551 ,\n",
       "        0.40816327, 0.42857143, 0.44897959, 0.46938776, 0.48979592,\n",
       "        0.51020408, 0.53061224, 0.55102041, 0.57142857, 0.59183673,\n",
       "        0.6122449 , 0.63265306, 0.65306122, 0.67346939, 0.69387755,\n",
       "        0.71428571, 0.73469388, 0.75510204, 0.7755102 , 0.79591837,\n",
       "        0.81632653, 0.83673469, 0.85714286, 0.87755102, 0.89795918,\n",
       "        0.91836735, 0.93877551, 0.95918367, 0.97959184, 1.        ]),\n",
       " <a list of 2 Lists of Patches objects>)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX0AAAD4CAYAAAAAczaOAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAQiElEQVR4nO3df6zdd13H8efLlo1fOjp3t8x22GIqsBEJUMcEJcg0K2DsTFhSEWjITKMORGMiHX+4JabJTIwBooM0AymR0DRjcdUJuhQRDbDRwWDr6lxl2F1X1wvoIJgMO97+cb6Ms+7e9dxzzj3nnn6ej6Q53/M5n+/5vj+7Z6/zOZ9zzvekqpAkteFHpl2AJGlyDH1JaoihL0kNMfQlqSGGviQ1ZO20Czid8847rzZu3DjtMiRpptx1113fqKq5U9tXfehv3LiRQ4cOTbsMSZopSf5jsXaXdySpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSFndOhv3HUbG3fdNu0yJGnVOKNDX5L0ZKv+3Dtjcf05fduPTq8OSZoyZ/qS1BBDX5IactrQT/LhJCeS3NvXdm6S25M80F2u67vt2iRHk9yf5Iq+9lckuae77f1JMv7hSJKeziAz/Y8AW09p2wUcrKrNwMHuOkkuBrYDl3T73JhkTbfPB4CdwObu36n3KUlaYacN/ar6LPCtU5q3AXu77b3AlX3t+6rqsap6EDgKXJrkQuDHqurzVVXAR/v2kSRNyLBr+hdU1XGA7vL8rn098FBfv/mubX23fWq7JGmCxv1G7mLr9PU07YvfSbIzyaEkhxYWFsZWnCS1btjQf6RbsqG7PNG1zwMX9fXbADzctW9YpH1RVbWnqrZU1Za5uaf8rq8kaUjDhv4BYEe3vQO4ta99e5Kzk2yi94btnd0S0HeSXNZ9audtfftIkibktN/ITfJx4LXAeUnmgeuAG4D9Sa4GjgFXAVTV4ST7gfuAk8A1VfV4d1e/Te+TQM8CPtn9kyRN0GlDv6p+fYmbLl+i/25g9yLth4CXLKs6SdJY+Y1cSWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaMlLoJ/n9JIeT3Jvk40memeTcJLcneaC7XNfX/9okR5Pcn+SK0cuXJC3H0KGfZD3wu8CWqnoJsAbYDuwCDlbVZuBgd50kF3e3XwJsBW5Msma08iVJyzHq8s5a4FlJ1gLPBh4GtgF7u9v3Ald229uAfVX1WFU9CBwFLh3x+JKkZRg69KvqP4E/BY4Bx4FHq+ofgAuq6njX5zhwfrfLeuChvruY79qeIsnOJIeSHFpYWBi2REnSKUZZ3llHb/a+CfgJ4DlJ3vJ0uyzSVot1rKo9VbWlqrbMzc0NW6Ik6RSjLO/8EvBgVS1U1f8BtwCvAh5JciFAd3mi6z8PXNS3/wZ6y0GSpAkZJfSPAZcleXaSAJcDR4ADwI6uzw7g1m77ALA9ydlJNgGbgTtHOL4kaZnWDrtjVd2R5GbgS8BJ4MvAHuC5wP4kV9N7Yriq6384yX7gvq7/NVX1+Ij1S5KWYejQB6iq64DrTml+jN6sf7H+u4HdoxxTkjQ8v5ErSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDRgr9JM9LcnOSf01yJMnPJTk3ye1JHugu1/X1vzbJ0ST3J7li9PIlScsx6kz/fcCnqupFwEuBI8Au4GBVbQYOdtdJcjGwHbgE2ArcmGTNiMeXJC3D0KGf5MeA1wAfAqiq71XV/wDbgL1dt73Ald32NmBfVT1WVQ8CR4FLhz2+JGn5RpnpvwBYAP4yyZeT3JTkOcAFVXUcoLs8v+u/Hniob//5ru0pkuxMcijJoYWFhRFKlCT1GyX01wIvBz5QVS8Dvku3lLOELNJWi3Wsqj1VtaWqtszNzY1QoiSp3yihPw/MV9Ud3fWb6T0JPJLkQoDu8kRf/4v69t8APDzC8SVJyzR06FfVfwEPJXlh13Q5cB9wANjRte0Abu22DwDbk5ydZBOwGbhz2ONLkpZv7Yj7vxP4WJKzgK8Bb6f3RLI/ydXAMeAqgKo6nGQ/vSeGk8A1VfX4iMeXJC3DSKFfVXcDWxa56fIl+u8Gdo9yTEnS8PxGriQ1xNCXpIYY+pLUEENfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1ZO20C5iGjbtue2L76898c2/j+kenVI0kTY4zfUlqiKEvSQ0x9CWpIYa+JDVk5NBPsibJl5P8bXf93CS3J3mgu1zX1/faJEeT3J/kilGPLUlannHM9N8FHOm7vgs4WFWbgYPddZJcDGwHLgG2AjcmWTOG40uSBjRS6CfZALwRuKmveRuwt9veC1zZ176vqh6rqgeBo8CloxxfkrQ8o8703wv8IfD9vrYLquo4QHd5fte+Hnior9981yZJmpChQz/JrwAnququQXdZpK2WuO+dSQ4lObSwsDBsiZKkU4wy03818KtJvg7sA16X5K+AR5JcCNBdnuj6zwMX9e2/AXh4sTuuqj1VtaWqtszNzY1QoiSp39ChX1XXVtWGqtpI7w3aT1fVW4ADwI6u2w7g1m77ALA9ydlJNgGbgTuHrlyStGwrce6dG4D9Sa4GjgFXAVTV4ST7gfuAk8A1VfX4ChxfkrSEsYR+VX0G+Ey3/U3g8iX67QZ2j+OYkqTl8xu5ktQQQ1+SGmLoS1JDDH1JWk2uP6f3b4U0+ctZkrSaPPnX/Fb2WM70Jakhhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqiKEvSQ0x9CWpIYa+JDXE0Jekhhj6ktQQQ1+SGmLoS1JDDH1JaoihL0kNMfQlqSGGviQ1xNBfygr/TqUkTYO/kdtnkr9TKUnT4Exfkhpi6EtSQwx9SWqIoS9JDTH0Jakhhr4kNWTo0E9yUZJ/THIkyeEk7+raz01ye5IHust1fftcm+RokvuTXDGOAUiSBjfKTP8k8AdV9WLgMuCaJBcDu4CDVbUZONhdp7ttO3AJsBW4McmaUYqXJC3P0KFfVcer6kvd9neAI8B6YBuwt+u2F7iy294G7Kuqx6rqQeAocOmwx5ckLd9Y1vSTbAReBtwBXFBVx6H3xACc33VbDzzUt9t817bY/e1McijJoYWFhXGUKEliDKGf5LnAJ4Dfq6pvP13XRdpqsY5VtaeqtlTVlrm5uVFLlCR1Rgr9JM+gF/gfq6pbuuZHklzY3X4hcKJrnwcu6tt9A/DwKMeXJC3PKJ/eCfAh4EhV/VnfTQeAHd32DuDWvvbtSc5OsgnYDNw57PElScs3ylk2Xw28Fbgnyd1d23uAG4D9Sa4GjgFXAVTV4ST7gfvoffLnmqp6fITjS5KWaejQr6p/YfF1eoDLl9hnN7B72GNKkkbjN3IlqSGGviQ1xNCXpIYY+pLUEENfkhpi6EtSQ0b5nL6GsHHXbU9sf/2Zb+5tXP/olKqR1Bpn+pLUEENfkhpi6EtSQ1zTH9EP1uifWJ8H1+glrVrO9CWpIYa+JDXE0B/Qxl23PenjlpI0i1zTX67rz+nbdu1e0mxxpi9JDXGmPwMG+oTQD16B+OpD0tMw9GfYk0/pMMVCJA3kSf/P3vDGqdTg8o4kTcP15zz5PcIJMfQlqSEu76yQ1fAyTj/0xPsi/i00Aav58WboT4Jvsq4efuRWK2Gpx9UqfLwZ+qvIqpgdjPggXbHfC/CJU6vQDz9ZN+VClsHQPwMNs7S0Gh+8LX86aVVMAAY02uPNExVOmqG/Go3zJeEqmiHP7P/o0/xvuAqXB57WKnq8aXGGfqtW6zLOhC01S534qwzD8skm8WQ3a0+oY2LoN2a5yzgzOzsfxgoF76p5YpkBwz4+e/sMNvlYjUuZk2ToS8s00hPhKprRP2UcXU2z9H7C01nu3+lMefV6Ooa+ZtsqCtEzTqPLH2c6Q19TN9qMbMXKkoAz7xWAoS+Nyap5/+MMfPVzpgXvNE089JNsBd4HrAFuqqobJl2DzlyrJnhXyFLj89WPBjXR0E+yBvgL4JeBeeCLSQ5U1X2TrENqxTRnyM7OV6dJz/QvBY5W1dcAkuwDtgGGvrTKnemvolqRqprcwZI3AVur6je7628FXllV7zil305gZ3f1hcD9QxzuPOAbI5Q7i1ocM7Q5bsfchlHG/JNVNXdq46Rn+lmk7SnPOlW1B9gz0oGSQ1W1ZZT7mDUtjhnaHLdjbsNKjHnSP6IyD1zUd30D8PCEa5CkZk069L8IbE6yKclZwHbgwIRrkKRmTXR5p6pOJnkH8Pf0PrL54ao6vEKHG2l5aEa1OGZoc9yOuQ1jH/NE38iVJE2XP4wuSQ0x9CWpITMf+km2Jrk/ydEkuxa5PUne393+1SQvn0ad4zTAmH+jG+tXk3wuyUunUec4nW7Mff1+Nsnj3XdCZtogY07y2iR3Jzmc5J8mXeO4DfDYPifJ3yT5Sjfmt0+jznFK8uEkJ5Lcu8Tt482wqprZf/TeDP534AXAWcBXgItP6fMG4JP0viNwGXDHtOuewJhfBazrtl/fwpj7+n0a+DvgTdOuewJ/5+fR+zb787vr50+77gmM+T3An3Tbc8C3gLOmXfuI434N8HLg3iVuH2uGzfpM/4nTOlTV94AfnNah3zbgo9XzBeB5SS6cdKFjdNoxV9Xnquq/u6tfoPd9iFk2yN8Z4J3AJ4ATkyxuhQwy5jcDt1TVMYCqmvVxDzLmAn40SYDn0gv9k5Mtc7yq6rP0xrGUsWbYrIf+euChvuvzXdty+8yS5Y7nanqzhFl22jEnWQ/8GvDBCda1kgb5O/80sC7JZ5LcleRtE6tuZQwy5j8HXkzvS533AO+qqu9PprypGWuGzfr59Ac5rcNAp36YIQOPJ8kv0gv9n1/RilbeIGN+L/Duqnq8NwmceYOMeS3wCuBy4FnA55N8oar+baWLWyGDjPkK4G7gdcBPAbcn+eeq+vZKFzdFY82wWQ/9QU7rcKad+mGg8ST5GeAm4PVV9c0J1bZSBhnzFmBfF/jnAW9IcrKq/noyJY7doI/tb1TVd4HvJvks8FJgVkN/kDG/HbiheovdR5M8CLwIuHMyJU7FWDNs1pd3BjmtwwHgbd074JcBj1bV8UkXOkanHXOS5wO3AG+d4Vlfv9OOuao2VdXGqtoI3Az8zgwHPgz22L4V+IUka5M8G3glcGTCdY7TIGM+Ru+VDUkuoHcW3q9NtMrJG2uGzfRMv5Y4rUOS3+pu/yC9T3K8ATgK/C+9mcLMGnDMfwT8OHBjN/M9WTN8dsIBx3xGGWTMVXUkyaeArwLfp/dLdIt+7G8WDPh3/mPgI0nuobfs8e6qmunTLSf5OPBa4Lwk88B1wDNgZTLM0zBIUkNmfXlHkrQMhr4kNcTQl6SGGPqS1BBDX5IaYuhLUkMMfUlqyP8D8s/Repza2ZAAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig,ax = plt.subplots()\n",
    "bins = np.linspace(0,1)\n",
    "plt.hist([normed_pbind_df['Norm_PBind (REF)'],normed_pbind_df['Norm_PBind (ALT)']],bins=bins)\n",
    "# plt.hist(normed_pbind_df['Norm_PBind (ALT)'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp</th>\n",
       "      <th>TF</th>\n",
       "      <th>H (REF)</th>\n",
       "      <th>H (ALT)</th>\n",
       "      <th>pbind (REF)</th>\n",
       "      <th>pbind (ALT)</th>\n",
       "      <th>Norm_PBind (REF)</th>\n",
       "      <th>Norm_PBind (ALT)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>TFAP2A(var.2)</td>\n",
       "      <td>0.633006</td>\n",
       "      <td>-0.435252</td>\n",
       "      <td>8.768534e-02</td>\n",
       "      <td>2.551912e-01</td>\n",
       "      <td>0.131395</td>\n",
       "      <td>0.113429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>BARX2</td>\n",
       "      <td>15.359863</td>\n",
       "      <td>9.436166</td>\n",
       "      <td>3.115243e-09</td>\n",
       "      <td>1.164449e-06</td>\n",
       "      <td>0.014385</td>\n",
       "      <td>0.014385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>Esrrg</td>\n",
       "      <td>10.279685</td>\n",
       "      <td>9.026033</td>\n",
       "      <td>2.620004e-03</td>\n",
       "      <td>9.178180e-03</td>\n",
       "      <td>0.987035</td>\n",
       "      <td>0.986951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>MEIS2</td>\n",
       "      <td>1.848022</td>\n",
       "      <td>1.381386</td>\n",
       "      <td>1.075637e+00</td>\n",
       "      <td>1.715232e+00</td>\n",
       "      <td>0.699580</td>\n",
       "      <td>0.551245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>MEIS3</td>\n",
       "      <td>-2.361144</td>\n",
       "      <td>0.190768</td>\n",
       "      <td>1.509059e-01</td>\n",
       "      <td>1.176048e-02</td>\n",
       "      <td>0.012091</td>\n",
       "      <td>0.013871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4213</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>PBX1</td>\n",
       "      <td>7.282949</td>\n",
       "      <td>7.959959</td>\n",
       "      <td>1.750528e-05</td>\n",
       "      <td>8.895038e-06</td>\n",
       "      <td>0.024842</td>\n",
       "      <td>0.024842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4214</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>LHX1</td>\n",
       "      <td>11.281669</td>\n",
       "      <td>10.860066</td>\n",
       "      <td>2.079952e-06</td>\n",
       "      <td>3.170685e-06</td>\n",
       "      <td>0.141669</td>\n",
       "      <td>0.141669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4215</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>ISL2</td>\n",
       "      <td>8.907163</td>\n",
       "      <td>8.738833</td>\n",
       "      <td>3.008563e-06</td>\n",
       "      <td>3.560113e-06</td>\n",
       "      <td>0.021734</td>\n",
       "      <td>0.021734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4216</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>Dux</td>\n",
       "      <td>26.649268</td>\n",
       "      <td>26.227665</td>\n",
       "      <td>2.794816e-13</td>\n",
       "      <td>4.260427e-13</td>\n",
       "      <td>0.094784</td>\n",
       "      <td>0.094784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4217</td>\n",
       "      <td>6:4907848</td>\n",
       "      <td>DUX4</td>\n",
       "      <td>50.458770</td>\n",
       "      <td>50.037166</td>\n",
       "      <td>9.772932e-22</td>\n",
       "      <td>1.489789e-21</td>\n",
       "      <td>0.889093</td>\n",
       "      <td>0.889093</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4218 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             snp             TF    H (REF)    H (ALT)   pbind (REF)  \\\n",
       "0     3:13308152  TFAP2A(var.2)   0.633006  -0.435252  8.768534e-02   \n",
       "1     3:13308152          BARX2  15.359863   9.436166  3.115243e-09   \n",
       "2     3:13308152          Esrrg  10.279685   9.026033  2.620004e-03   \n",
       "3     3:13308152          MEIS2   1.848022   1.381386  1.075637e+00   \n",
       "4     3:13308152          MEIS3  -2.361144   0.190768  1.509059e-01   \n",
       "...          ...            ...        ...        ...           ...   \n",
       "4213   6:4907848           PBX1   7.282949   7.959959  1.750528e-05   \n",
       "4214   6:4907848           LHX1  11.281669  10.860066  2.079952e-06   \n",
       "4215   6:4907848           ISL2   8.907163   8.738833  3.008563e-06   \n",
       "4216   6:4907848            Dux  26.649268  26.227665  2.794816e-13   \n",
       "4217   6:4907848           DUX4  50.458770  50.037166  9.772932e-22   \n",
       "\n",
       "       pbind (ALT)  Norm_PBind (REF)  Norm_PBind (ALT)  \n",
       "0     2.551912e-01          0.131395          0.113429  \n",
       "1     1.164449e-06          0.014385          0.014385  \n",
       "2     9.178180e-03          0.987035          0.986951  \n",
       "3     1.715232e+00          0.699580          0.551245  \n",
       "4     1.176048e-02          0.012091          0.013871  \n",
       "...            ...               ...               ...  \n",
       "4213  8.895038e-06          0.024842          0.024842  \n",
       "4214  3.170685e-06          0.141669          0.141669  \n",
       "4215  3.560113e-06          0.021734          0.021734  \n",
       "4216  4.260427e-13          0.094784          0.094784  \n",
       "4217  1.489789e-21          0.889093          0.889093  \n",
       "\n",
       "[4218 rows x 8 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normed_pbind_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_seq_combos(seq,al,tf_len):\n",
    "    seqs_list = []\n",
    "    complseqs_list = []\n",
    "    for num in range(tf_len):\n",
    "        newseq = ''\n",
    "        curr_ref_start_pos = tf_len-num-1\n",
    "        curr_ref_end_pos = curr_ref_start_pos+tf_len\n",
    "        curr_seq = seq[curr_ref_start_pos:curr_ref_end_pos]\n",
    "        tempseq = ''\n",
    "        for n,b in enumerate(curr_seq):\n",
    "            if((n+curr_ref_start_pos) == (tf_len-1)):\n",
    "                tempseq = ''.join([tempseq,al])\n",
    "            else:\n",
    "                tempseq = ''.join([tempseq,b])\n",
    "        seqs_list.append(tempseq)\n",
    "        complseqs_list.append(get_complseq(tempseq))\n",
    "    return seqs_list,complseqs_list\n",
    "\n",
    "\n",
    "def get_scoredict_entry(seq,pwm,bg_z_df,bgfreqs,tfname,tfpbind_cutoff=None):\n",
    "    fracPWM = get_fracPWM_from_matrix(pwm)\n",
    "    lnfracPWM = get_lnPWM_from_fracPWM(fracPWM,bgfreqs)\n",
    "    rawscore_list = get_matrix_scores(pwm,seq)\n",
    "    pos_counts = get_matrix_counts(pwm)\n",
    "    tot_count = sum(pos_counts)\n",
    "    score_ln = get_matrix_scores(lnfracPWM,seq)\n",
    "    curr_fracscore = sum([rawscore_list[x]/pos_counts[x] for x in range(len(pos_counts))])\n",
    "    curr_H = np.sum(score_ln)\n",
    "    curr_bindingp = calculate_bindingP(curr_H,bg_z_df,tfname)\n",
    "    if(tfpbind_cutoff is not None):\n",
    "        if(curr_bindingp >= float(tfpbind_cutoff)):\n",
    "            curr_scoredict = {'tf_name':tfname,'raw_score':sum(rawscore_list),'tf_len':len(pwm),'counts_perpos':min(pos_counts),\n",
    "                    'fraction_score':curr_fracscore,'H':curr_H,'bindingP':curr_bindingp}#,'orientation':'+','direction':'forward','motif_pos':p}\n",
    "    else:\n",
    "        curr_scoredict = {'tf_name':tfname,'raw_score':sum(rawscore_list),'tf_len':len(pwm),'counts_perpos':min(pos_counts),\n",
    "                    'fraction_score':curr_fracscore,'H':curr_H,'bindingP':curr_bindingp}\n",
    "    return curr_scoredict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_seqlist = get_seq_combos(ref_full_seq,'G',len(curr_matrix))\n",
    "bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'][['frac_A','frac_C','frac_G','frac_T']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GCATTCATTAGG'"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_random_bgseqs(slen,reps,fastaloc,chrmsizes,seqinfo_file=None):\n",
    "    # x = pybedtools.BedTool()\n",
    "    chrms_touse = list(chrmsizes['chrom'])\n",
    "    bgseq_list = []\n",
    "    if(seqinfo_file is not None):\n",
    "        genseqs_df = read_csv(seqinfo_file,delimiter='\\t')\n",
    "        for r,seq in genseqs_df.iterrows():\n",
    "            chrmfile = '{0}/chr{1}.fa'.format(fastaloc,seq['Chrm'])\n",
    "            curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(seq['Chrm'],seq['Start'],seq['End']),chrmfile)\n",
    "            bgseq_list.append([seq['Chrm'],seq['Start'],seq['End'],curr_seq.upper()])\n",
    "    while(len(bgseq_list) < int(reps)):\n",
    "        is_valid_seq = True\n",
    "        try:\n",
    "            curr_chrm = 8\n",
    "            curr_start = random.randint(1,chrmsizes.loc[chrmsizes['chrom'] == 'chr{0}'.format(curr_chrm)]['size'].values[0])\n",
    "            curr_end = curr_start + slen\n",
    "            chrmfile = '{0}/chr{1}.fa'.format(fastaloc,curr_chrm)\n",
    "            curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(curr_chrm,curr_start,curr_end),chrmfile).upper()\n",
    "            for b in curr_seq:\n",
    "                if(b not in 'ACTG'):\n",
    "                    is_valid_seq = False\n",
    "                    continue\n",
    "            if(is_valid_seq):\n",
    "                bgseq_list.append([curr_chrm,curr_start,curr_end,curr_seq])\n",
    "        except:\n",
    "            continue\n",
    "            # bgseq_list.append({'chrm':curr_chrm,'start':curr_start,'end':curr_end,'seq':curr_seq.upper()})\n",
    "    return bgseq_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[8, 99052390, 99052400, 'TGAAAAAAACT'],\n",
       " [8, 35673058, 35673068, 'TGGTGAGTGAG'],\n",
       " [8, 114551456, 114551466, 'AAACATTCTAA'],\n",
       " [8, 4262645, 4262655, 'ATTGGTTATTT'],\n",
       " [8, 40835500, 40835510, 'GTTGGCCAGGC']]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_random_bgseqs(10,5,'/local3/jake/TF_binding/testing',chrmsizes_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# test_finaloutput_df = read_csv('/local3/jake/TF_binding/testing/final_PWM.alltraits.allsnps.withPBinding100reps.output',delimiter='\\t')\n",
    "test_finaloutput_df = read_csv('/local3/jake/TF_binding/testing/final_PWM.alltraits.allsnps.withPBinding100reps.output',delimiter='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.07563699327 MEIS2 0.5182201881921262\n",
      "14.7703128976 FOXC2 0.9365897172428472\n",
      "6.6773563475 HOXD9 0.8697468301927145\n",
      "1.55498734326 Znf281 0.6086086286742479\n",
      "387.908761496 Foxo1 0.997428702824405\n",
      "1.38167730713 FOXC2 0.5801278380550885\n",
      "6.03604829404 NKX2-2 0.857874767452769\n",
      "7.434444318200001 NKX2-3 0.8814385438719917\n",
      "7.72985977408 GCM2 0.8854506228185611\n",
      "134.608632857 IRF6 0.9926258382012114\n",
      "1.25819402688 MZF1(var.2) 0.5571682556505344\n",
      "2008.59665436 TBX1 0.9995023877065943\n",
      "20.7294432022 TBX15 0.9539794926774795\n",
      "151.173459035 NKX2-2 0.9934285518227927\n",
      "435.24958535 ZNF384 0.9977077342109218\n",
      "55.4901279226 NFAT5 0.9822977919013617\n",
      "4.79641804599 ZBTB7C 0.8274796620831028\n",
      "348350.756348 KLF14 0.9999971293384294\n",
      "284.893105939 ZNF740 0.9965021891775955\n",
      "11612.9830514 ZNF148 0.9999138968951843\n",
      "3.693068696 TFAP2A(var.3) 0.7869198034854442\n",
      "10.2753420248 TFAP2C 0.9113108943573858\n",
      "84.78731061069999 GCM2 0.9883432643723044\n",
      "1.38167730713 FOXC2 0.5801278380550885\n",
      "6.03604829404 NKX2-2 0.857874767452769\n",
      "7.434444318200001 NKX2-3 0.8814385438719917\n",
      "134.608632857 IRF6 0.9926258382012114\n",
      "7.72985977408 GCM2 0.8854506228185611\n",
      "151.173459035 NKX2-2 0.9934285518227927\n",
      "1.25819402688 MZF1(var.2) 0.5571682556505344\n",
      "43.0510549313 Isl1 0.9772990680572725\n",
      "3.64387523604 CEBPG 0.7846626041455211\n",
      "22816.4730168 NKX6-3 0.9999561739374356\n",
      "6.466962926819999 GMEB2 0.866076742338465\n",
      "8.91965098265 HOXA10 0.8991900015683797\n",
      "435.24958535 ZNF384 0.9977077342109218\n",
      "635.390948389 HOXD9 0.9984286388696577\n",
      "1.1877366331 STAT3 0.5429065889967941\n",
      "9124.66818663 Wt1 0.9998904189830767\n",
      "9376.74021471 ZNF148 0.9998933645017768\n",
      "2.02107589017 KLF4 0.6689920954134196\n",
      "6.26576084695 ZBTB7C 0.86236816472938\n",
      "3.96294102994 SP8 0.798506572218684\n",
      "57822.75852999999 KLF5 0.9999827060705596\n",
      "63.7388777961 RUNX3 0.9845533312586525\n",
      "5.720935496009999 NKX2-2 0.8512111891869447\n",
      "1.9118772678 HOXA6 0.6565789324092586\n",
      "1.90165982925 HOXB8 0.6553696646585946\n",
      "2.4095149828 NFAT5 0.7067031513146812\n",
      "1.9076348564 E2F6 0.6560778607401975\n",
      "1.38167730713 FOXC2 0.5801278380550885\n",
      "6.03604829404 NKX2-2 0.857874767452769\n",
      "7.434444318200001 NKX2-3 0.8814385438719917\n",
      "134.608632857 IRF6 0.9926258382012114\n",
      "635.390948389 HOXD9 0.9984286388696577\n",
      "1.1877366331 STAT3 0.5429065889967941\n",
      "7.72985977408 GCM2 0.8854506228185611\n",
      "19.2388483028 HOXA6 0.9505900738502128\n",
      "1.88697369529 HOXB8 0.6536165183511264\n",
      "1.25819402688 MZF1(var.2) 0.5571682556505344\n",
      "9124.66818663 Wt1 0.9998904189830767\n",
      "9376.74021471 ZNF148 0.9998933645017768\n",
      "2.02107589017 KLF4 0.6689920954134196\n",
      "6.26576084695 ZBTB7C 0.86236816472938\n",
      "3.96294102994 SP8 0.798506572218684\n",
      "57822.75852999999 KLF5 0.9999827060705596\n",
      "63.7388777961 RUNX3 0.9845533312586525\n",
      "43.0510549313 Isl1 0.9772990680572725\n",
      "3.64387523604 CEBPG 0.7846626041455211\n",
      "22816.4730168 NKX6-3 0.9999561739374356\n",
      "6.466962926819999 GMEB2 0.866076742338465\n",
      "8.91965098265 HOXA10 0.8991900015683797\n",
      "1.82782750582 SOX15 0.6463716411466944\n",
      "151.173459035 NKX2-2 0.9934285518227927\n",
      "4.02649041612 ZNF148 0.8010540322939619\n",
      "150.481750785 Wt1 0.993398544743392\n",
      "49.422872193100005 KLF5 0.9801677303075245\n",
      "1264.01280482 Foxo1 0.9992094941678112\n",
      "29035.1497023 Wt1 0.9999655601720528\n",
      "3550.88478929 ZNF148 0.99971845933657\n",
      "93533.6537695 KLF5 0.9999893087753075\n",
      "1.35238239551 SP8 0.5748990462134594\n",
      "4.82721897645 Sox3 0.8283915528074578\n",
      "1.56593771153 Sox6 0.6102789262935098\n",
      "1.96773770938 TBX1 0.6630429984296087\n",
      "1.37612845296 TBX3 0.5791473315544025\n",
      "16.859210301300006 Ahr::Arnt 0.9440064827544442\n",
      "3.10427960046 ZNF384 0.7563518820965045\n",
      "1020.43572196 TEF 0.9990209858745915\n",
      "2.35656902964 ONECUT1 0.7020767363430136\n",
      "499.765799435 FOXC2 0.9980030585133174\n",
      "1.5126913704 CDX2 0.6020203627946007\n",
      "3.0685082947900004 RAX 0.7542096691113159\n",
      "7.329741468360001 NOTO 0.8799482548404529\n",
      "3.50295287518 CDX4 0.7779235031436573\n",
      "4120.36052429 HOXD9 0.9997573616784785\n",
      "1.6060906913 Sox3 0.6162834995201519\n",
      "616.150799032 ZNF384 0.9983796504815902\n",
      "37.2564488704 Foxo1 0.9738606161959973\n",
      "1.36684074128 Sox3 0.5774958650318488\n",
      "94.1967236686 STAT3 0.9894954368021956\n",
      "4.92939986699 Sox6 0.8313488679407801\n",
      "1.75635621325 SOX15 0.6372021891814247\n",
      "2.1977342665 TBX18 0.6872785801885982\n",
      "40.3761765663 TBX6 0.9758315029810893\n",
      "8.735508952610001 TBX3 0.8972832334843601\n",
      "18.4898160997 Ahr::Arnt 0.948691152605734\n",
      "3.03057503725 MEIS3 0.7518964438669867\n",
      "1197.50745058 MEIS2 0.999165628883225\n",
      "16.4027841182 Stat4 0.9425379299537927\n",
      "8.95737649211 NKX2-2 0.899571940380626\n",
      "6.04097208216 NKX2-3 0.8579741563619879\n",
      "22.5448882046 GATA6 0.9575279359446154\n",
      "1.35634072752 KLF5 0.575613158010382\n",
      "2.57682522646 HES1 0.7204224593922622\n",
      "3.26906306607 ZBTB14 0.7657565642570934\n",
      "2.5613696674 KLF4 0.7192091545134248\n",
      "2.4095149828 NFAT5 0.7067031513146812\n"
     ]
    }
   ],
   "source": [
    "count = 0\n",
    "for i,p in test_finaloutput_df.iterrows():\n",
    "    if(p['P binding (REF)'] > 1):\n",
    "        curr_z = bg_z_df.loc[bg_z_df['TF_name'] == p['TF_name']]['BG Z score'].values[0]\n",
    "        print(p['P binding (REF)'],p['TF_name'],math.exp(-p['H (REF)'])/(curr_z+math.exp(-p['H (REF)'])))\n",
    "        count += 1\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "118"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>beta</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>trait_num</th>\n",
       "      <th>TF_name</th>\n",
       "      <th>PWM_fracscore_ref</th>\n",
       "      <th>PWM_fracscore_alt</th>\n",
       "      <th>TF_len</th>\n",
       "      <th>TF_count</th>\n",
       "      <th>H (REF)</th>\n",
       "      <th>Hprime (ALT)</th>\n",
       "      <th>P binding (REF)</th>\n",
       "      <th>P binding (ALT)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>13308152</td>\n",
       "      <td>3:13308152</td>\n",
       "      <td>1.252538e-11</td>\n",
       "      <td>6.312838</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>MEIS2</td>\n",
       "      <td>4.472486</td>\n",
       "      <td>4.498092</td>\n",
       "      <td>8</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>1.848022</td>\n",
       "      <td>1.381386</td>\n",
       "      <td>1.075637</td>\n",
       "      <td>1.715232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>9</td>\n",
       "      <td>117741852</td>\n",
       "      <td>9:117741852</td>\n",
       "      <td>3.720925e-10</td>\n",
       "      <td>8.344869</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>6</td>\n",
       "      <td>FOXC2</td>\n",
       "      <td>6.312845</td>\n",
       "      <td>5.851529</td>\n",
       "      <td>12</td>\n",
       "      <td>20559.0</td>\n",
       "      <td>-3.212667</td>\n",
       "      <td>-1.366008</td>\n",
       "      <td>14.770313</td>\n",
       "      <td>2.330214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>4</td>\n",
       "      <td>36936148</td>\n",
       "      <td>4:36936148</td>\n",
       "      <td>1.048648e-08</td>\n",
       "      <td>7.487932</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6</td>\n",
       "      <td>HOXD9</td>\n",
       "      <td>5.371421</td>\n",
       "      <td>5.368861</td>\n",
       "      <td>10</td>\n",
       "      <td>16367.0</td>\n",
       "      <td>3.687649</td>\n",
       "      <td>7.927654</td>\n",
       "      <td>6.677356</td>\n",
       "      <td>0.096204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>135</td>\n",
       "      <td>2</td>\n",
       "      <td>75543351</td>\n",
       "      <td>2:75543351</td>\n",
       "      <td>2.109130e-08</td>\n",
       "      <td>6.243562</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>Znf281</td>\n",
       "      <td>5.801887</td>\n",
       "      <td>5.795648</td>\n",
       "      <td>11</td>\n",
       "      <td>6572.0</td>\n",
       "      <td>2.039124</td>\n",
       "      <td>2.557725</td>\n",
       "      <td>1.554987</td>\n",
       "      <td>0.925766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>149</td>\n",
       "      <td>3</td>\n",
       "      <td>78699639</td>\n",
       "      <td>3:78699639</td>\n",
       "      <td>1.260930e-09</td>\n",
       "      <td>6.238320</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>7</td>\n",
       "      <td>Foxo1</td>\n",
       "      <td>5.876905</td>\n",
       "      <td>5.872895</td>\n",
       "      <td>11</td>\n",
       "      <td>2494.0</td>\n",
       "      <td>0.032564</td>\n",
       "      <td>0.471262</td>\n",
       "      <td>387.908761</td>\n",
       "      <td>250.152886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4069</td>\n",
       "      <td>10</td>\n",
       "      <td>101227322</td>\n",
       "      <td>10:101227322</td>\n",
       "      <td>7.738416e-09</td>\n",
       "      <td>4.733058</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>KLF5</td>\n",
       "      <td>4.566361</td>\n",
       "      <td>3.802130</td>\n",
       "      <td>10</td>\n",
       "      <td>13615.0</td>\n",
       "      <td>5.514446</td>\n",
       "      <td>7.867854</td>\n",
       "      <td>1.356341</td>\n",
       "      <td>0.128913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4089</td>\n",
       "      <td>21</td>\n",
       "      <td>37815373</td>\n",
       "      <td>21:37815373</td>\n",
       "      <td>9.301903e-09</td>\n",
       "      <td>6.797307</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>24</td>\n",
       "      <td>HES1</td>\n",
       "      <td>4.075568</td>\n",
       "      <td>4.075568</td>\n",
       "      <td>10</td>\n",
       "      <td>18530.0</td>\n",
       "      <td>2.791947</td>\n",
       "      <td>2.791947</td>\n",
       "      <td>2.576825</td>\n",
       "      <td>2.576825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4104</td>\n",
       "      <td>6</td>\n",
       "      <td>161296413</td>\n",
       "      <td>6:161296413</td>\n",
       "      <td>1.050335e-08</td>\n",
       "      <td>2.978600</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>24</td>\n",
       "      <td>ZBTB14</td>\n",
       "      <td>5.336410</td>\n",
       "      <td>6.323006</td>\n",
       "      <td>12</td>\n",
       "      <td>4551.0</td>\n",
       "      <td>0.897615</td>\n",
       "      <td>-4.694942</td>\n",
       "      <td>3.269063</td>\n",
       "      <td>877.485065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4140</td>\n",
       "      <td>12</td>\n",
       "      <td>125145063</td>\n",
       "      <td>12:125145063</td>\n",
       "      <td>4.790847e-01</td>\n",
       "      <td>0.535178</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>24</td>\n",
       "      <td>KLF4</td>\n",
       "      <td>7.116014</td>\n",
       "      <td>6.262822</td>\n",
       "      <td>12</td>\n",
       "      <td>55821.0</td>\n",
       "      <td>-7.345788</td>\n",
       "      <td>-3.910319</td>\n",
       "      <td>2.561370</td>\n",
       "      <td>0.082502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4186</td>\n",
       "      <td>20</td>\n",
       "      <td>48717609</td>\n",
       "      <td>20:48717609</td>\n",
       "      <td>1.321490e-08</td>\n",
       "      <td>1.918270</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>NFAT5</td>\n",
       "      <td>5.019356</td>\n",
       "      <td>5.019356</td>\n",
       "      <td>10</td>\n",
       "      <td>1004.0</td>\n",
       "      <td>-2.362002</td>\n",
       "      <td>-2.365170</td>\n",
       "      <td>2.409515</td>\n",
       "      <td>2.417159</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>118 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos        snp_id        pvalue      beta ref_allele  \\\n",
       "3       3   13308152    3:13308152  1.252538e-11  6.312838          C   \n",
       "37      9  117741852   9:117741852  3.720925e-10  8.344869          A   \n",
       "72      4   36936148    4:36936148  1.048648e-08  7.487932          C   \n",
       "135     2   75543351    2:75543351  2.109130e-08  6.243562          G   \n",
       "149     3   78699639    3:78699639  1.260930e-09  6.238320          C   \n",
       "...   ...        ...           ...           ...       ...        ...   \n",
       "4069   10  101227322  10:101227322  7.738416e-09  4.733058          G   \n",
       "4089   21   37815373   21:37815373  9.301903e-09  6.797307          C   \n",
       "4104    6  161296413   6:161296413  1.050335e-08  2.978600          C   \n",
       "4140   12  125145063  12:125145063  4.790847e-01  0.535178          G   \n",
       "4186   20   48717609   20:48717609  1.321490e-08  1.918270          C   \n",
       "\n",
       "     alt_allele  trait_num TF_name  PWM_fracscore_ref  PWM_fracscore_alt  \\\n",
       "3             A          6   MEIS2           4.472486           4.498092   \n",
       "37            G          6   FOXC2           6.312845           5.851529   \n",
       "72            T          6   HOXD9           5.371421           5.368861   \n",
       "135           A          6  Znf281           5.801887           5.795648   \n",
       "149           T          7   Foxo1           5.876905           5.872895   \n",
       "...         ...        ...     ...                ...                ...   \n",
       "4069          A         24    KLF5           4.566361           3.802130   \n",
       "4089          T         24    HES1           4.075568           4.075568   \n",
       "4104          T         24  ZBTB14           5.336410           6.323006   \n",
       "4140          C         24    KLF4           7.116014           6.262822   \n",
       "4186          G          8   NFAT5           5.019356           5.019356   \n",
       "\n",
       "      TF_len  TF_count   H (REF)  Hprime (ALT)  P binding (REF)  \\\n",
       "3          8    1355.0  1.848022      1.381386         1.075637   \n",
       "37        12   20559.0 -3.212667     -1.366008        14.770313   \n",
       "72        10   16367.0  3.687649      7.927654         6.677356   \n",
       "135       11    6572.0  2.039124      2.557725         1.554987   \n",
       "149       11    2494.0  0.032564      0.471262       387.908761   \n",
       "...      ...       ...       ...           ...              ...   \n",
       "4069      10   13615.0  5.514446      7.867854         1.356341   \n",
       "4089      10   18530.0  2.791947      2.791947         2.576825   \n",
       "4104      12    4551.0  0.897615     -4.694942         3.269063   \n",
       "4140      12   55821.0 -7.345788     -3.910319         2.561370   \n",
       "4186      10    1004.0 -2.362002     -2.365170         2.409515   \n",
       "\n",
       "      P binding (ALT)  \n",
       "3            1.715232  \n",
       "37           2.330214  \n",
       "72           0.096204  \n",
       "135          0.925766  \n",
       "149        250.152886  \n",
       "...               ...  \n",
       "4069         0.128913  \n",
       "4089         2.576825  \n",
       "4104       877.485065  \n",
       "4140         0.082502  \n",
       "4186         2.417159  \n",
       "\n",
       "[118 rows x 17 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_finaloutput_df.loc[test_finaloutput_df['P binding (REF)'] > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF_name</th>\n",
       "      <th>BG Z score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>318</td>\n",
       "      <td>TFAP2A(var.2)</td>\n",
       "      <td>6.055668</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           TF_name  BG Z score\n",
       "318  TFAP2A(var.2)    6.055668"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bg_z_df.loc[bg_z_df['TF_name'] == 'TFAP2A(var.2)']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GGATTACTTACAGTTAAACGTAA'"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "''.join(list(reversed(ref_full_seq))) == get_rev_seq(ref_full_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AATGCAAATTGACATTCATTAGG\n",
      "11 11 23\n",
      "ACATTCATTAGG 12\n",
      "10 11 22\n",
      "GACATTCATTAG 12\n",
      "9 11 21\n",
      "TGACATTCATTA 12\n",
      "8 11 20\n",
      "TTGACATTCATT 12\n",
      "7 11 19\n",
      "ATTGACATTCAT 12\n",
      "6 11 18\n",
      "AATTGACATTCA 12\n",
      "5 11 17\n",
      "AAATTGACATTC 12\n",
      "4 11 16\n",
      "CAAATTGACATT 12\n",
      "3 11 15\n",
      "GCAAATTGACAT 12\n",
      "2 11 14\n",
      "TGCAAATTGACA 12\n",
      "1 11 13\n",
      "ATGCAAATTGAC 12\n",
      "0 11 12\n",
      "AATGCAAATTGA 12\n"
     ]
    }
   ],
   "source": [
    "print(ref_full_seq)\n",
    "seqlist = []\n",
    "for n in range(tf_len):\n",
    "    newseq = ''\n",
    "    curr_ref_start_pos = tf_len-n-1\n",
    "    curr_ref_end_pos = curr_ref_start_pos+tf_len\n",
    "    print((curr_ref_start_pos),(tf_len-1),curr_ref_end_pos)\n",
    "    curr_seq = ref_full_seq[curr_ref_start_pos:curr_ref_end_pos]\n",
    "    print(curr_seq,len(curr_seq))\n",
    "    seqlist.append([curr_seq,curr_ref_start_pos])\n",
    "#     for s in ref_full_seq\n",
    "\n",
    "new_seqlist = []\n",
    "al = 'A'\n",
    "for s in seqlist:\n",
    "    tempseq = ''\n",
    "    for n,b in enumerate(s[0]):\n",
    "        if((n+s[1]) == (tf_len-1)):\n",
    "            tempseq = ''.join([tempseq,al])\n",
    "        else:\n",
    "            tempseq = ''.join([tempseq,b])\n",
    "    new_seqlist.append(tempseq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for num in range(tf_len):\n",
    "    newseq = ''\n",
    "    curr_ref_start_pos = tf_len-num-1\n",
    "    curr_ref_end_pos = curr_ref_start_pos+tf_len\n",
    "    curr_seq = seq[curr_ref_start_pos:curr_ref_end_pos]\n",
    "    print((curr_ref_start_pos),(tf_len-1),curr_ref_end_pos)\n",
    "    tempseq = ''\n",
    "    for n,b in enumerate(curr_seq):\n",
    "        if((n+curr_ref_start_pos) == (tf_len-1)):\n",
    "            tempseq = ''.join([tempseq,al])\n",
    "        else:\n",
    "            tempseq = ''.join([tempseq,b])\n",
    "    seqs_list.append(tempseq)\n",
    "    complseqs_list.append(get_complseq(tempseq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'gene_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-db4651ed197e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mposition\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m61768012\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mchromosome\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m8\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mref_pos_end\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgene_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos_end'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0mref_pos_start\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgene_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos_start'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0mgene_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'relative_start'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m  \u001b[0mgene_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pos_start'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mref_pos_start\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'gene_df' is not defined"
     ]
    }
   ],
   "source": [
    "# gene_df = read_csv('/local3/jake/TF_binding/testing/14.5M.TopVars.LE.chrm8.pos61768012.TF_genes',header=None,delimiter='\\t')\n",
    "# gene_df.columns = ['pos_start','pos_end','tf_name']\n",
    "position = 61768012\n",
    "chromosome = 8\n",
    "ref_pos_end = max(gene_df['pos_end'])\n",
    "ref_pos_start = min(gene_df['pos_start'])\n",
    "gene_df['relative_start'] =  gene_df['pos_start']-ref_pos_start\n",
    "gene_df['relative_end'] = gene_df['pos_end']-ref_pos_start\n",
    "\n",
    "matrix_loc = '/local3/jake/TF_binding/JASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac/'\n",
    "# matrix_loc = 'gene_dfJASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac'\n",
    "transfac_matrix_list = os.listdir(matrix_loc)\n",
    "infodicts_list = []\n",
    "for f in transfac_matrix_list:\n",
    "    curr_infodict = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    infodicts_list.append(curr_infodict)\n",
    "    \n",
    "# for i,g in gene_df.iterrows():\n",
    "#     curr_score = get_matrix_byTFName(g[2],infodicts_list,'ACT')\n",
    "#     print('score for {0} is {1}'.format(g[2],curr_score))\n",
    "    \n",
    "\n",
    "# ref_full_seq = get_hg19reference_sequence(ref_pos_start,ref_pos_end,'/local3/jake/TF_binding/chr4.fa')\n",
    "ref_full_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(8,ref_pos_start,(ref_pos_end)),'/local3/jake/TF_binding/testing/chr8.fa')\n",
    "\n",
    "updated_pos = (position-ref_pos_start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'gene_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-b5c251069ba9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mbgfrac_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/local3/jake/TF_binding/ACTG_count.all_chrms.fractions.v2.txt'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdelimiter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mscore_dict_bytf\u001b[0m \u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mg\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgene_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miterrows\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0mcurr_relative_pos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mabs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mupdated_pos\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'relative_start'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mcurr_refseq\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_seq\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mref_full_seq\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'relative_start'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'relative_end'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcurr_relative_pos\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'G'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'gene_df' is not defined"
     ]
    }
   ],
   "source": [
    "bgfrac_df = read_csv('/local3/jake/TF_binding/ACTG_count.all_chrms.fractions.txt',delimiter='\\t')\n",
    "score_dict_bytf ={}\n",
    "for i,g in gene_df.iterrows():\n",
    "    curr_relative_pos = abs(updated_pos-g['relative_start'])\n",
    "    curr_refseq = make_seq(ref_full_seq[g['relative_start']:(g['relative_end']+1)],curr_relative_pos,'G')\n",
    "    curr_altseq = make_seq(ref_full_seq[g['relative_start']:(g['relative_end']+1)],curr_relative_pos,'A')\n",
    "    curr_matrix = get_matrix_byTF(g['tf_name'],infodicts_list)\n",
    "    curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "    curr_lnfracPWM = get_fraclnPWM_from_matrixdict(curr_matrix)\n",
    "    # curr_refscore,curr_altscore,curr_counts = get_matrix_scores(tfname=g['tf_name'],matrix_dict=curr_matrix,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    # curr_totcount = sum(curr_counts)\n",
    "    refscore_list = get_matrix_scores(curr_matrix,curr_refseq)\n",
    "    altscore_list = get_matrix_scores(curr_matrix,curr_altseq)\n",
    "    pos_counts = get_matrix_counts(curr_matrix,curr_matrix['TF_len'])\n",
    "    tot_count = sum(pos_counts)\n",
    "    curr_scoredict = {'ref_score':sum(refscore_list),'alt_score':sum(altscore_list),'tf_len':curr_matrix['TF_len'],'counts_perpos':min(pos_counts)}\n",
    "    curr_scoredict['ref_fraction_score'] = sum([refscore_list[x]/pos_counts[x] for x in range(len(pos_counts))])\n",
    "    curr_scoredict['alt_fraction_score'] = sum([altscore_list[x]/pos_counts[x] for x in range(len(pos_counts))])\n",
    "\n",
    "    refscore_ln = get_matrix_scores(curr_lnfracPWM,curr_refseq)\n",
    "    altscore_ln = get_matrix_scores(curr_lnfracPWM,curr_altseq)\n",
    "    curr_scoredict['ref_fraction_ln_score'] = sum(refscore_ln)\n",
    "    curr_scoredict['alt_fraction_ln_score'] = sum(altscore_ln)\n",
    "\n",
    "    # frac_refvals,frac_altvals = get_frac_matrix_scores(tfname=g['tf_name'],pwm=curr_fracPWM,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    # fracln_refvals,fracln_altvals = get_frac_matrix_scores(tfname=g['tf_name'],pwm=curr_lnfracPWM,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    bgfreqs = np.array(bgfrac_df.loc[bgfrac_df['Chrm'] == str(chromosome)][['frac_A','frac_C','frac_G','frac_T']])\n",
    "    ref_Y = compute_Y(curr_refseq,bgfreqs)\n",
    "    alt_Y = compute_Y(curr_altseq,bgfreqs)\n",
    "    ref_H = refscore_ln - ref_Y\n",
    "    alt_H = altscore_ln - alt_Y\n",
    "    curr_scoredict['H'] = np.sum(ref_H)\n",
    "    curr_scoredict['Hprime'] = np.sum(alt_H)\n",
    "\n",
    "    score_dict_bytf[g['tf_name']] = curr_scoredict\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "seqA = 'AAAAAAAAAAAAAA'\n",
    "seqB = 'GAAAAAAAAAAAAA'\n",
    "curr_matrix = read_JASPAR_transfac_pfms('/local3/jake/TF_binding/TF_1.transfac')\n",
    "bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'][['frac_A','frac_C','frac_G','frac_T']]\n",
    "# curr_lnfracPWM = get_fraclnPWM_from_matrixdict(curr_matrix)\n",
    "curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "curr_lnfracPWM = get_lnPWM_from_fracPWM(curr_fracPWM,bgfreqs)\n",
    "Ascore_ln = get_matrix_scores(curr_lnfracPWM,seqA)\n",
    "Bscore_ln = get_matrix_scores(curr_lnfracPWM,seqB)\n",
    "\n",
    "A_Y = compute_Y(seqA,np.array(bgfreqs))\n",
    "B_Y = compute_Y(seqB,np.array(bgfreqs))\n",
    "A_H = Ascore_ln - A_Y\n",
    "B_H = Bscore_ln - B_Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([], shape=(0, 4), dtype=float64)"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sum_A = np.sum(bgfrac_df['count_A'])\n",
    "# sum_C = np.sum(bgfrac_df['count_C'])\n",
    "# sum_G = np.sum(bgfrac_df['count_G'])\n",
    "# sum_T = np.sum(bgfrac_df['count_T'])\n",
    "# sum_total = np.sum(bgfrac_df['total'])\n",
    "# bgfrac_df = bgfrac_df.append({'Chrm':'Total','count_A':sum_A,'count_C':sum_C,'count_G':sum_G,'count_T':sum_T,'total':sum_total,'frac_A':(sum_A/sum_total),'frac_C':(sum_C/sum_total),'frac_G':(sum_G/sum_total),'frac_T':(sum_T/sum_total)},ignore_index=True)\n",
    "np.array(bgfrac_df.loc[bgfrac_df['Chrm'] == 8][['frac_A','frac_C','frac_G','frac_T']])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrm</th>\n",
       "      <th>count_A</th>\n",
       "      <th>count_C</th>\n",
       "      <th>count_G</th>\n",
       "      <th>count_T</th>\n",
       "      <th>total</th>\n",
       "      <th>frac_A</th>\n",
       "      <th>frac_C</th>\n",
       "      <th>frac_G</th>\n",
       "      <th>frac_T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>65570891</td>\n",
       "      <td>47024412</td>\n",
       "      <td>47016562</td>\n",
       "      <td>65668756</td>\n",
       "      <td>2.252806e+08</td>\n",
       "      <td>0.291063</td>\n",
       "      <td>0.208737</td>\n",
       "      <td>0.208702</td>\n",
       "      <td>0.291498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>71102632</td>\n",
       "      <td>47915465</td>\n",
       "      <td>47947042</td>\n",
       "      <td>71239379</td>\n",
       "      <td>2.382045e+08</td>\n",
       "      <td>0.298494</td>\n",
       "      <td>0.201153</td>\n",
       "      <td>0.201285</td>\n",
       "      <td>0.299068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>58713343</td>\n",
       "      <td>38653197</td>\n",
       "      <td>38670110</td>\n",
       "      <td>58760485</td>\n",
       "      <td>1.947971e+08</td>\n",
       "      <td>0.301408</td>\n",
       "      <td>0.198428</td>\n",
       "      <td>0.198515</td>\n",
       "      <td>0.301650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>57932980</td>\n",
       "      <td>35885806</td>\n",
       "      <td>35890822</td>\n",
       "      <td>57952068</td>\n",
       "      <td>1.876617e+08</td>\n",
       "      <td>0.308710</td>\n",
       "      <td>0.191226</td>\n",
       "      <td>0.191253</td>\n",
       "      <td>0.308811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>53672554</td>\n",
       "      <td>35089383</td>\n",
       "      <td>35129186</td>\n",
       "      <td>53804137</td>\n",
       "      <td>1.776953e+08</td>\n",
       "      <td>0.302048</td>\n",
       "      <td>0.197469</td>\n",
       "      <td>0.197693</td>\n",
       "      <td>0.302789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>50554433</td>\n",
       "      <td>33143287</td>\n",
       "      <td>33163423</td>\n",
       "      <td>50533923</td>\n",
       "      <td>1.673951e+08</td>\n",
       "      <td>0.302007</td>\n",
       "      <td>0.197994</td>\n",
       "      <td>0.198115</td>\n",
       "      <td>0.301884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>45997757</td>\n",
       "      <td>31671670</td>\n",
       "      <td>31636979</td>\n",
       "      <td>46047257</td>\n",
       "      <td>1.553537e+08</td>\n",
       "      <td>0.296084</td>\n",
       "      <td>0.203868</td>\n",
       "      <td>0.203645</td>\n",
       "      <td>0.296403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>42767293</td>\n",
       "      <td>28703983</td>\n",
       "      <td>28702621</td>\n",
       "      <td>42715025</td>\n",
       "      <td>1.428889e+08</td>\n",
       "      <td>0.299304</td>\n",
       "      <td>0.200883</td>\n",
       "      <td>0.200874</td>\n",
       "      <td>0.298939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>35260078</td>\n",
       "      <td>24826212</td>\n",
       "      <td>24813259</td>\n",
       "      <td>35243882</td>\n",
       "      <td>1.201434e+08</td>\n",
       "      <td>0.293483</td>\n",
       "      <td>0.206638</td>\n",
       "      <td>0.206530</td>\n",
       "      <td>0.293348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>38330752</td>\n",
       "      <td>27308648</td>\n",
       "      <td>27298423</td>\n",
       "      <td>38376915</td>\n",
       "      <td>1.313147e+08</td>\n",
       "      <td>0.291900</td>\n",
       "      <td>0.207963</td>\n",
       "      <td>0.207885</td>\n",
       "      <td>0.292251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>38307244</td>\n",
       "      <td>27236798</td>\n",
       "      <td>27268038</td>\n",
       "      <td>38317436</td>\n",
       "      <td>1.311295e+08</td>\n",
       "      <td>0.292133</td>\n",
       "      <td>0.207709</td>\n",
       "      <td>0.207947</td>\n",
       "      <td>0.292211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>38604831</td>\n",
       "      <td>26634995</td>\n",
       "      <td>26617050</td>\n",
       "      <td>38624517</td>\n",
       "      <td>1.304814e+08</td>\n",
       "      <td>0.295865</td>\n",
       "      <td>0.204129</td>\n",
       "      <td>0.203991</td>\n",
       "      <td>0.296016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>29336945</td>\n",
       "      <td>18412698</td>\n",
       "      <td>18414776</td>\n",
       "      <td>29425459</td>\n",
       "      <td>9.558988e+07</td>\n",
       "      <td>0.306904</td>\n",
       "      <td>0.192622</td>\n",
       "      <td>0.192644</td>\n",
       "      <td>0.307830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>25992966</td>\n",
       "      <td>18027132</td>\n",
       "      <td>18071947</td>\n",
       "      <td>26197495</td>\n",
       "      <td>8.828954e+07</td>\n",
       "      <td>0.294406</td>\n",
       "      <td>0.204182</td>\n",
       "      <td>0.204690</td>\n",
       "      <td>0.296723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>23620876</td>\n",
       "      <td>17247582</td>\n",
       "      <td>17228387</td>\n",
       "      <td>23597921</td>\n",
       "      <td>8.169477e+07</td>\n",
       "      <td>0.289136</td>\n",
       "      <td>0.211122</td>\n",
       "      <td>0.210887</td>\n",
       "      <td>0.288855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>21724083</td>\n",
       "      <td>17630040</td>\n",
       "      <td>17701988</td>\n",
       "      <td>21828642</td>\n",
       "      <td>7.888475e+07</td>\n",
       "      <td>0.275390</td>\n",
       "      <td>0.223491</td>\n",
       "      <td>0.224403</td>\n",
       "      <td>0.276716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>21159933</td>\n",
       "      <td>17727956</td>\n",
       "      <td>17700340</td>\n",
       "      <td>21206981</td>\n",
       "      <td>7.779521e+07</td>\n",
       "      <td>0.271995</td>\n",
       "      <td>0.227880</td>\n",
       "      <td>0.227525</td>\n",
       "      <td>0.272600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>22465380</td>\n",
       "      <td>14838685</td>\n",
       "      <td>14863671</td>\n",
       "      <td>22489493</td>\n",
       "      <td>7.465723e+07</td>\n",
       "      <td>0.300914</td>\n",
       "      <td>0.198758</td>\n",
       "      <td>0.199092</td>\n",
       "      <td>0.301237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>14390632</td>\n",
       "      <td>13478255</td>\n",
       "      <td>13511145</td>\n",
       "      <td>14428951</td>\n",
       "      <td>5.580898e+07</td>\n",
       "      <td>0.257855</td>\n",
       "      <td>0.241507</td>\n",
       "      <td>0.242096</td>\n",
       "      <td>0.258542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>20</td>\n",
       "      <td>16523053</td>\n",
       "      <td>13107828</td>\n",
       "      <td>13149412</td>\n",
       "      <td>16725227</td>\n",
       "      <td>5.950552e+07</td>\n",
       "      <td>0.277673</td>\n",
       "      <td>0.220279</td>\n",
       "      <td>0.220978</td>\n",
       "      <td>0.281070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>10422924</td>\n",
       "      <td>7160212</td>\n",
       "      <td>7174721</td>\n",
       "      <td>10348785</td>\n",
       "      <td>3.510664e+07</td>\n",
       "      <td>0.296893</td>\n",
       "      <td>0.203956</td>\n",
       "      <td>0.204369</td>\n",
       "      <td>0.294781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "      <td>9094775</td>\n",
       "      <td>8375984</td>\n",
       "      <td>8369235</td>\n",
       "      <td>9054551</td>\n",
       "      <td>3.489454e+07</td>\n",
       "      <td>0.260636</td>\n",
       "      <td>0.240037</td>\n",
       "      <td>0.239844</td>\n",
       "      <td>0.259483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Total</td>\n",
       "      <td>791546355</td>\n",
       "      <td>550100228</td>\n",
       "      <td>550339137</td>\n",
       "      <td>792587285</td>\n",
       "      <td>2.684573e+09</td>\n",
       "      <td>0.294850</td>\n",
       "      <td>0.204912</td>\n",
       "      <td>0.205001</td>\n",
       "      <td>0.295238</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chrm    count_A    count_C    count_G    count_T         total    frac_A  \\\n",
       "0       1   65570891   47024412   47016562   65668756  2.252806e+08  0.291063   \n",
       "1       2   71102632   47915465   47947042   71239379  2.382045e+08  0.298494   \n",
       "2       3   58713343   38653197   38670110   58760485  1.947971e+08  0.301408   \n",
       "3       4   57932980   35885806   35890822   57952068  1.876617e+08  0.308710   \n",
       "4       5   53672554   35089383   35129186   53804137  1.776953e+08  0.302048   \n",
       "5       6   50554433   33143287   33163423   50533923  1.673951e+08  0.302007   \n",
       "6       7   45997757   31671670   31636979   46047257  1.553537e+08  0.296084   \n",
       "7       8   42767293   28703983   28702621   42715025  1.428889e+08  0.299304   \n",
       "8       9   35260078   24826212   24813259   35243882  1.201434e+08  0.293483   \n",
       "9      10   38330752   27308648   27298423   38376915  1.313147e+08  0.291900   \n",
       "10     11   38307244   27236798   27268038   38317436  1.311295e+08  0.292133   \n",
       "11     12   38604831   26634995   26617050   38624517  1.304814e+08  0.295865   \n",
       "12     13   29336945   18412698   18414776   29425459  9.558988e+07  0.306904   \n",
       "13     14   25992966   18027132   18071947   26197495  8.828954e+07  0.294406   \n",
       "14     15   23620876   17247582   17228387   23597921  8.169477e+07  0.289136   \n",
       "15     16   21724083   17630040   17701988   21828642  7.888475e+07  0.275390   \n",
       "16     17   21159933   17727956   17700340   21206981  7.779521e+07  0.271995   \n",
       "17     18   22465380   14838685   14863671   22489493  7.465723e+07  0.300914   \n",
       "18     19   14390632   13478255   13511145   14428951  5.580898e+07  0.257855   \n",
       "19     20   16523053   13107828   13149412   16725227  5.950552e+07  0.277673   \n",
       "20     21   10422924    7160212    7174721   10348785  3.510664e+07  0.296893   \n",
       "21     22    9094775    8375984    8369235    9054551  3.489454e+07  0.260636   \n",
       "22  Total  791546355  550100228  550339137  792587285  2.684573e+09  0.294850   \n",
       "\n",
       "      frac_C    frac_G    frac_T  \n",
       "0   0.208737  0.208702  0.291498  \n",
       "1   0.201153  0.201285  0.299068  \n",
       "2   0.198428  0.198515  0.301650  \n",
       "3   0.191226  0.191253  0.308811  \n",
       "4   0.197469  0.197693  0.302789  \n",
       "5   0.197994  0.198115  0.301884  \n",
       "6   0.203868  0.203645  0.296403  \n",
       "7   0.200883  0.200874  0.298939  \n",
       "8   0.206638  0.206530  0.293348  \n",
       "9   0.207963  0.207885  0.292251  \n",
       "10  0.207709  0.207947  0.292211  \n",
       "11  0.204129  0.203991  0.296016  \n",
       "12  0.192622  0.192644  0.307830  \n",
       "13  0.204182  0.204690  0.296723  \n",
       "14  0.211122  0.210887  0.288855  \n",
       "15  0.223491  0.224403  0.276716  \n",
       "16  0.227880  0.227525  0.272600  \n",
       "17  0.198758  0.199092  0.301237  \n",
       "18  0.241507  0.242096  0.258542  \n",
       "19  0.220279  0.220978  0.281070  \n",
       "20  0.203956  0.204369  0.294781  \n",
       "21  0.240037  0.239844  0.259483  \n",
       "22  0.204912  0.205001  0.295238  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bgfrac_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_mat = get_matrix_byTF('TEF',infodicts_list)\n",
    "test_fracmat = get_fracPWM_from_matrix(test_mat)\n",
    "test_lnfracmat = get_lnPWM_from_fracPWM(test_fracmat,bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == 'Total']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'A': 0.6735225273982176,\n",
       "  'C': -0.3554471495320632,\n",
       "  'T': -0.08175289925020757,\n",
       "  'G': -0.14459640764447515},\n",
       " {'A': -0.5177994648194508,\n",
       "  'C': 0.7835377014306947,\n",
       "  'T': 4.873093687874733,\n",
       "  'G': -0.6913982749716533},\n",
       " {'A': 2.4887931408833537,\n",
       "  'C': 3.691325718965144,\n",
       "  'T': -1.188277683467875,\n",
       "  'G': 4.844439435478379},\n",
       " {'A': 2.3242229598990236,\n",
       "  'C': 6.642466237754624,\n",
       "  'T': -1.1815761438448353,\n",
       "  'G': 3.1771645415297476},\n",
       " {'A': -1.190937592440007,\n",
       "  'C': 6.634418984027275,\n",
       "  'T': 6.999621128163717,\n",
       "  'G': 1.9435053083729816},\n",
       " {'A': 4.1992464783911005,\n",
       "  'C': -1.463514238717256,\n",
       "  'T': 0.9883738365415188,\n",
       "  'G': 6.726164493593496},\n",
       " {'A': -0.22037920958245874,\n",
       "  'C': 6.604900579569244,\n",
       "  'T': 6.970102723705686,\n",
       "  'G': -1.1257183578630328},\n",
       " {'A': 7.016455283009747,\n",
       "  'C': 1.7547275337902164,\n",
       "  'T': -1.1714750478583316,\n",
       "  'G': 2.868811906397717},\n",
       " {'A': -1.1539257019909293,\n",
       "  'C': 2.3276254526226685,\n",
       "  'T': 1.9490366833804107,\n",
       "  'G': 4.186958431263203},\n",
       " {'A': -1.2124437976609705,\n",
       "  'C': 6.612912778806312,\n",
       "  'T': 4.580219650144383,\n",
       "  'G': 3.6176147118271706},\n",
       " {'A': 7.040754323085755,\n",
       "  'C': -0.9827765807475464,\n",
       "  'T': -0.25907928790245394,\n",
       "  'G': 1.0788786213936117},\n",
       " {'A': -0.26422203833057184,\n",
       "  'C': -0.43591324163753215,\n",
       "  'T': 0.7102438286499426,\n",
       "  'G': 0.28608924140683917}]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_lnfracmat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/local3/jake/TF_binding/tophit_SNPs/output_v2/'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-14-fbc78aa974ec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mresults_loc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'/local3/jake/TF_binding/tophit_SNPs/output_v2/'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mpwmresults_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresults_loc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0massocfile_loc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'/local3/mhansen/Projects/CTA/Analysis/GWAS/All_Inds/Results/5M_Filtered'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0minfofile_loc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'/local3/jake/TF_binding/tophit_SNPs/'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/local3/jake/TF_binding/tophit_SNPs/output_v2/'"
     ]
    }
   ],
   "source": [
    "results_loc = '/local3/jake/TF_binding/tophit_SNPs/output_v2/'\n",
    "pwmresults_list = os.listdir(results_loc)\n",
    "\n",
    "assocfile_loc = '/local3/mhansen/Projects/CTA/Analysis/GWAS/All_Inds/Results/5M_Filtered'\n",
    "infofile_loc = '/local3/jake/TF_binding/tophit_SNPs/'\n",
    "infofile_list = os.listdir(infofile_loc)\n",
    "\n",
    "final_df_dictlist = []\n",
    "for i in infofile_list:\n",
    "    if('.full_assoc' not in i):\n",
    "        continue\n",
    "    curr_df = read_csv('{0}{1}'.format(infofile_loc,i),delimiter='\\t')\n",
    "    curr_prefix = '.'.join(i.split('.')[:-1])\n",
    "    temp_dict = {}\n",
    "    for num,snp in curr_df.iterrows():\n",
    "        temp_dict['chr'] = snp['#Chr']\n",
    "        temp_dict['pos'] = snp['Pos']\n",
    "        temp_dict['snp_id'] = snp['VID']\n",
    "        temp_dict['beta'] = snp['Beta']\n",
    "        temp_dict['pvalue'] = snp['pval']\n",
    "        temp_dict['ref_allele'] = snp['Ref']\n",
    "        temp_dict['alt_allele'] = snp['Alt']\n",
    "        temp_dict['trait_num'] = curr_prefix.split('.')[0]\n",
    "        curr_pwmfile = '{0}{1}.chrm{2}.pos{3}.PWM_scores'.format(results_loc,curr_prefix,snp['#Chr'],snp['Pos'])\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'args' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-ae9873ab5b8e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0mgene_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/local3/jake/TF_binding/tophit_SNPs/output/24.5M.TopVars.LE.chrm4.pos165034923.TF_genes'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdelimiter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0mgene_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'pos_start'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'pos_end'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m \u001b[0mtransfac_matrix_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmatrix_loc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m \u001b[0minfodicts_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtransfac_matrix_list\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'args' is not defined"
     ]
    }
   ],
   "source": [
    "test_pwmfile = '24.5M.TopVars.LE.chrm4.pos165034923.PWM_scores'\n",
    "# position = int(args.position.split(':')[1])\n",
    "position = 165034923\n",
    "# chromosome = int(args.position.split(':')[0])\n",
    "chromosome = 4\n",
    "\n",
    "# gene_df = read_csv('{0}'.format(args.input_gene_file),header=None,delimiter='\\t')\n",
    "gene_df = read_csv('/local3/jake/TF_binding/tophit_SNPs/output/24.5M.TopVars.LE.chrm4.pos165034923.TF_genes',header=None,delimiter='\\t')\n",
    "gene_df.columns = ['pos_start','pos_end','tf_name']\n",
    "# transfac_matrix_list = os.listdir(args.matrix_loc)\n",
    "matrix_loc = '/local3/jake/TF_binding/JASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac/'\n",
    "# matrix_loc = 'gene_dfJASPAR2020_CORE_vertebrates_non-redundant_pfms_transfac'\n",
    "transfac_matrix_list = os.listdir(matrix_loc)\n",
    "infodicts_list = []\n",
    "for f in transfac_matrix_list:\n",
    "    curr_infodict = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    infodicts_list.append(curr_infodict)\n",
    "\n",
    "ref_pos_end = max(gene_df['pos_end'])\n",
    "ref_pos_start = min(gene_df['pos_start'])\n",
    "# ref_full_seq = get_hg19reference_sequence(ref_pos_start,ref_pos_end,args.ref_fasta_file)\n",
    "ref_full_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(chromosome,ref_pos_start,ref_pos_end),args.ref_fasta_file)\n",
    "updated_pos = (position-ref_pos_start)\n",
    "gene_df['relative_start'] =  gene_df['pos_start']-ref_pos_start\n",
    "gene_df['relative_end'] = gene_df['pos_end']-ref_pos_start\n",
    "\n",
    "score_dict_bytf ={}\n",
    "for i,g in gene_df.iterrows():\n",
    "    curr_relative_pos = abs(updated_pos-g['relative_start'])\n",
    "    curr_seq = ref_full_seq[g['relative_start']:(g['relative_end']+1)]\n",
    "    curr_matrix = get_matrix_byTF(g['tf_name'],infodicts_list)\n",
    "    curr_refscore,curr_altscore,curr_counts = get_matrix_scores(tfname=g['tf_name'],matrix_dict=curr_matrix,ref_al=args.ref_al,alt_al=args.alt_al,position=curr_relative_pos,reference_seq=curr_seq)\n",
    "    curr_totcount = sum(curr_counts)\n",
    "    score_dict_bytf[g['tf_name']] = {'ref_score':curr_refscore,'alt_score':curr_altscore,'ref_fraction_score':(curr_refscore/curr_totcount),'alt_fraction_score':(curr_altscore/curr_totcount),'tf_len':len(curr_seq),'counts_perpos':curr_counts[0]}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#Make the final results df; combining the different .PWM output files\n",
    "results_loc = '/local3/jake/TF_binding/testing/tophit_SNPs/output_v5/'\n",
    "pwmresults_list = os.listdir(results_loc)\n",
    "\n",
    "assocfile_loc = '/local3/mhansen/Projects/CTA/Analysis/GWAS/All_Inds/Results/5M_Filtered'\n",
    "infofile_loc = '/local3/jake/TF_binding/testing/tophit_SNPs/'\n",
    "infofile_list = os.listdir(infofile_loc)\n",
    "\n",
    "final_df_dictlist = []\n",
    "for i in infofile_list:\n",
    "    if('.full_assoc' not in i):\n",
    "        continue\n",
    "    elif(i == '8.1M_5M.TopVars.LE.5M.full_assoc' or i == '7.1M_5M.TopVars.LE.5M.full_assoc'):\n",
    "        continue\n",
    "    curr_df = read_csv('{0}{1}'.format(infofile_loc,i),delimiter='\\t')\n",
    "    curr_prefix = '.'.join(i.split('.')[:-1])\n",
    "    temp_dict = {}\n",
    "    for num,snp in curr_df.iterrows():\n",
    "        temp_dict['chr'] = snp['#Chr']\n",
    "        temp_dict['pos'] = snp['Pos']\n",
    "        temp_dict['snp_id'] = snp['VID']\n",
    "        if(i == '8.1M_5M.TopVars.LE.full_assoc' or i == '7.1M_5M.TopVars.LE.full_assoc' ):\n",
    "            temp_dict['beta'] = snp['Beta_FE']\n",
    "            temp_dict['pvalue'] = snp['pval_FE']\n",
    "            if(i == '8.1M_5M.TopVars.LE.full_assoc'):\n",
    "                d = read_csv('{0}{1}'.format(infofile_loc,'8.1M_5M.TopVars.LE.5M.full_assoc'),delimiter='\\t')\n",
    "                temp_dict['ref_allele'] = d.loc[num]['Ref']\n",
    "                temp_dict['alt_allele'] = d.loc[num]['Alt']\n",
    "            if(i == '7.1M_5M.TopVars.LE.full_assoc'):\n",
    "                d = read_csv('{0}{1}'.format(infofile_loc,'7.1M_5M.TopVars.LE.5M.full_assoc'),delimiter='\\t')\n",
    "                temp_dict['ref_allele'] = d.loc[num]['Ref']\n",
    "                temp_dict['alt_allele'] = d.loc[num]['Alt']\n",
    "        else:\n",
    "            temp_dict['pvalue'] = snp['pval']\n",
    "            temp_dict['beta'] = snp['Beta']\n",
    "            temp_dict['ref_allele'] = snp['Ref']\n",
    "            temp_dict['alt_allele'] = snp['Alt']\n",
    "        temp_dict['trait_num'] = curr_prefix.split('.')[0]\n",
    "        curr_pwmfile = '{0}{1}.chrm{2}.pos{3}.PWM_scores'.format(results_loc,curr_prefix,snp['#Chr'],snp['Pos'])\n",
    "        curr_pwm_df = read_csv(curr_pwmfile,skiprows=1,delimiter='\\t')\n",
    "        for index,tf in curr_pwm_df.iterrows():\n",
    "            curr_dict = temp_dict.copy()\n",
    "            curr_dict['TF_name'] = tf['TF_Name']\n",
    "            curr_dict['PWM_fracscore_ref'] = tf['PWM Fraction Score (REF allele)'] #f(float(tf['PWM Fraction Score (REF allele)']),5)#\"{0:.5f}\".format(tf['PWM Fraction Score (REF allele)'])\n",
    "            curr_dict['PWM_fracscore_alt'] = tf['PWM Fraction Score (ALT allele)'] #f(float(tf['PWM Fraction Score (Alt allele)']),5)#\"{0:.5f}\".format(tf['PWM Fraction Score (ALT allele)'])\n",
    "            curr_dict['TF_len'] = tf['TF Length']\n",
    "            curr_dict['TF_count'] = tf['TF Counts per position']\n",
    "#             curr_dict['PWM_ln_fracscore_ref'] = \"{0:.5f}\".format(tf['REF Log(PWM Fraction Score)'])\n",
    "#             curr_dict['PWM_ln_fracscore_alt'] = \"{0:.5f}\".format(tf['ALT Log(PWM Fraction Score)'])\n",
    "            curr_dict['H (REF)'] = tf['H (REF)'] #f(float(tf['H (REF)']),5)#\"{0:.5f}\".format(tf['H (REF)'])\n",
    "            curr_dict['Hprime (ALT)'] = tf['Hprime (ALT)'] #f(float(tf['Hprime (ALT)']),5)#\"{0:.5f}\".format(tf['Hprime (ALT)'])\n",
    "            curr_dict['P binding (REF)'] = tf['P binding (REF)'] #f(float(tf['P binding (REF)']),5)#\"{0:.5f}\".format(tf['P binding (REF)'])\n",
    "            curr_dict['P binding (ALT)'] = tf['P binding (ALT)'] #f(float(tf['P binding (ALT)']),5)#\"{0:.5f}\".format(tf['P binding (ALT)'])\n",
    "            final_df_dictlist.append(curr_dict)\n",
    "\n",
    "final_df = DataFrame(final_df_dictlist)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>beta</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>trait_num</th>\n",
       "      <th>TF_name</th>\n",
       "      <th>PWM_fracscore_ref</th>\n",
       "      <th>PWM_fracscore_alt</th>\n",
       "      <th>TF_len</th>\n",
       "      <th>TF_count</th>\n",
       "      <th>H (REF)</th>\n",
       "      <th>Hprime (ALT)</th>\n",
       "      <th>P binding (REF)</th>\n",
       "      <th>P binding (ALT)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>149</td>\n",
       "      <td>3</td>\n",
       "      <td>78699639</td>\n",
       "      <td>3:78699639</td>\n",
       "      <td>1.260930e-09</td>\n",
       "      <td>6.238320</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>7</td>\n",
       "      <td>Foxo1</td>\n",
       "      <td>5.876905</td>\n",
       "      <td>5.872895</td>\n",
       "      <td>11</td>\n",
       "      <td>2494.0</td>\n",
       "      <td>0.032564</td>\n",
       "      <td>0.471262</td>\n",
       "      <td>387.908761</td>\n",
       "      <td>250.152886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>328</td>\n",
       "      <td>15</td>\n",
       "      <td>48683454</td>\n",
       "      <td>15:48683454</td>\n",
       "      <td>8.597562e-12</td>\n",
       "      <td>2.115559</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>25</td>\n",
       "      <td>IRF6</td>\n",
       "      <td>5.044620</td>\n",
       "      <td>5.028799</td>\n",
       "      <td>9</td>\n",
       "      <td>2665.0</td>\n",
       "      <td>0.344800</td>\n",
       "      <td>2.130402</td>\n",
       "      <td>134.608633</td>\n",
       "      <td>22.573342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>433</td>\n",
       "      <td>11</td>\n",
       "      <td>61137147</td>\n",
       "      <td>11:61137147</td>\n",
       "      <td>7.483225e-10</td>\n",
       "      <td>2.144577</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>25</td>\n",
       "      <td>TBX1</td>\n",
       "      <td>5.015816</td>\n",
       "      <td>4.020404</td>\n",
       "      <td>8</td>\n",
       "      <td>11935.0</td>\n",
       "      <td>-1.666418</td>\n",
       "      <td>7.720827</td>\n",
       "      <td>2008.596654</td>\n",
       "      <td>0.168292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>443</td>\n",
       "      <td>15</td>\n",
       "      <td>28238895</td>\n",
       "      <td>15:28238895</td>\n",
       "      <td>8.381399e-10</td>\n",
       "      <td>2.188545</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>25</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>7.162255</td>\n",
       "      <td>7.162255</td>\n",
       "      <td>14</td>\n",
       "      <td>26027.0</td>\n",
       "      <td>-5.118098</td>\n",
       "      <td>-5.118098</td>\n",
       "      <td>151.173459</td>\n",
       "      <td>151.173459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>523</td>\n",
       "      <td>15</td>\n",
       "      <td>48296524</td>\n",
       "      <td>15:48296524</td>\n",
       "      <td>1.368194e-08</td>\n",
       "      <td>1.720186</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>25</td>\n",
       "      <td>ZNF384</td>\n",
       "      <td>7.695108</td>\n",
       "      <td>7.765220</td>\n",
       "      <td>12</td>\n",
       "      <td>30437.0</td>\n",
       "      <td>-7.035665</td>\n",
       "      <td>-6.987407</td>\n",
       "      <td>435.249585</td>\n",
       "      <td>414.744055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>775</td>\n",
       "      <td>14</td>\n",
       "      <td>59697005</td>\n",
       "      <td>14:59697005</td>\n",
       "      <td>5.008992e-09</td>\n",
       "      <td>0.975950</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>20</td>\n",
       "      <td>KLF14</td>\n",
       "      <td>7.611608</td>\n",
       "      <td>7.609984</td>\n",
       "      <td>14</td>\n",
       "      <td>1604.0</td>\n",
       "      <td>-6.209140</td>\n",
       "      <td>-5.690172</td>\n",
       "      <td>348350.756348</td>\n",
       "      <td>207315.553111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>776</td>\n",
       "      <td>14</td>\n",
       "      <td>59697005</td>\n",
       "      <td>14:59697005</td>\n",
       "      <td>5.008992e-09</td>\n",
       "      <td>0.975950</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>20</td>\n",
       "      <td>ZNF740</td>\n",
       "      <td>7.570647</td>\n",
       "      <td>6.623519</td>\n",
       "      <td>13</td>\n",
       "      <td>1059.0</td>\n",
       "      <td>-3.911958</td>\n",
       "      <td>0.590858</td>\n",
       "      <td>284.893106</td>\n",
       "      <td>3.155977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>777</td>\n",
       "      <td>14</td>\n",
       "      <td>59697005</td>\n",
       "      <td>14:59697005</td>\n",
       "      <td>5.008992e-09</td>\n",
       "      <td>0.975950</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>20</td>\n",
       "      <td>ZNF148</td>\n",
       "      <td>7.317058</td>\n",
       "      <td>6.585022</td>\n",
       "      <td>12</td>\n",
       "      <td>11203.0</td>\n",
       "      <td>-7.368900</td>\n",
       "      <td>-4.846557</td>\n",
       "      <td>11612.983051</td>\n",
       "      <td>932.189034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>984</td>\n",
       "      <td>15</td>\n",
       "      <td>48683454</td>\n",
       "      <td>15:48683454</td>\n",
       "      <td>8.913702e-11</td>\n",
       "      <td>1.348175</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>26</td>\n",
       "      <td>IRF6</td>\n",
       "      <td>5.044620</td>\n",
       "      <td>5.028799</td>\n",
       "      <td>9</td>\n",
       "      <td>2665.0</td>\n",
       "      <td>0.344800</td>\n",
       "      <td>2.130402</td>\n",
       "      <td>134.608633</td>\n",
       "      <td>22.573342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1041</td>\n",
       "      <td>15</td>\n",
       "      <td>28238895</td>\n",
       "      <td>15:28238895</td>\n",
       "      <td>5.209608e-10</td>\n",
       "      <td>1.485306</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>26</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>7.162255</td>\n",
       "      <td>7.162255</td>\n",
       "      <td>14</td>\n",
       "      <td>26027.0</td>\n",
       "      <td>-5.118098</td>\n",
       "      <td>-5.118098</td>\n",
       "      <td>151.173459</td>\n",
       "      <td>151.173459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1241</td>\n",
       "      <td>8</td>\n",
       "      <td>6410987</td>\n",
       "      <td>8:6410987</td>\n",
       "      <td>1.721366e-08</td>\n",
       "      <td>2.507842</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>26</td>\n",
       "      <td>NKX6-3</td>\n",
       "      <td>6.062696</td>\n",
       "      <td>5.247429</td>\n",
       "      <td>9</td>\n",
       "      <td>6909.0</td>\n",
       "      <td>-7.040466</td>\n",
       "      <td>-5.152894</td>\n",
       "      <td>22816.473017</td>\n",
       "      <td>3455.304880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1410</td>\n",
       "      <td>15</td>\n",
       "      <td>48296524</td>\n",
       "      <td>15:48296524</td>\n",
       "      <td>2.308487e-08</td>\n",
       "      <td>1.135363</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>26</td>\n",
       "      <td>ZNF384</td>\n",
       "      <td>7.695108</td>\n",
       "      <td>7.765220</td>\n",
       "      <td>12</td>\n",
       "      <td>30437.0</td>\n",
       "      <td>-7.035665</td>\n",
       "      <td>-6.987407</td>\n",
       "      <td>435.249585</td>\n",
       "      <td>414.744055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1533</td>\n",
       "      <td>5</td>\n",
       "      <td>32944420</td>\n",
       "      <td>5:32944420</td>\n",
       "      <td>2.959992e-08</td>\n",
       "      <td>2.554335</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>26</td>\n",
       "      <td>HOXD9</td>\n",
       "      <td>5.770177</td>\n",
       "      <td>5.368568</td>\n",
       "      <td>10</td>\n",
       "      <td>16367.0</td>\n",
       "      <td>-0.867869</td>\n",
       "      <td>7.923143</td>\n",
       "      <td>635.390948</td>\n",
       "      <td>0.096639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1558</td>\n",
       "      <td>18</td>\n",
       "      <td>34121148</td>\n",
       "      <td>18:34121148</td>\n",
       "      <td>3.193288e-08</td>\n",
       "      <td>4.692228</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>26</td>\n",
       "      <td>Wt1</td>\n",
       "      <td>7.943770</td>\n",
       "      <td>8.079233</td>\n",
       "      <td>14</td>\n",
       "      <td>4695.0</td>\n",
       "      <td>-6.213578</td>\n",
       "      <td>-7.654303</td>\n",
       "      <td>9124.668187</td>\n",
       "      <td>38540.377390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1559</td>\n",
       "      <td>18</td>\n",
       "      <td>34121148</td>\n",
       "      <td>18:34121148</td>\n",
       "      <td>3.193288e-08</td>\n",
       "      <td>4.692228</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>26</td>\n",
       "      <td>ZNF148</td>\n",
       "      <td>7.421137</td>\n",
       "      <td>7.421137</td>\n",
       "      <td>12</td>\n",
       "      <td>11203.0</td>\n",
       "      <td>-7.155008</td>\n",
       "      <td>-7.155008</td>\n",
       "      <td>9376.740215</td>\n",
       "      <td>9376.740215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1564</td>\n",
       "      <td>18</td>\n",
       "      <td>34121148</td>\n",
       "      <td>18:34121148</td>\n",
       "      <td>3.193288e-08</td>\n",
       "      <td>4.692228</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>26</td>\n",
       "      <td>KLF5</td>\n",
       "      <td>5.659934</td>\n",
       "      <td>5.624972</td>\n",
       "      <td>10</td>\n",
       "      <td>13615.0</td>\n",
       "      <td>-5.145902</td>\n",
       "      <td>0.607487</td>\n",
       "      <td>57822.758530</td>\n",
       "      <td>183.414645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2087</td>\n",
       "      <td>15</td>\n",
       "      <td>48683454</td>\n",
       "      <td>15:48683454</td>\n",
       "      <td>7.755283e-10</td>\n",
       "      <td>0.997581</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>27</td>\n",
       "      <td>IRF6</td>\n",
       "      <td>5.044620</td>\n",
       "      <td>5.028799</td>\n",
       "      <td>9</td>\n",
       "      <td>2665.0</td>\n",
       "      <td>0.344800</td>\n",
       "      <td>2.130402</td>\n",
       "      <td>134.608633</td>\n",
       "      <td>22.573342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2220</td>\n",
       "      <td>5</td>\n",
       "      <td>32944420</td>\n",
       "      <td>5:32944420</td>\n",
       "      <td>2.213560e-09</td>\n",
       "      <td>2.146916</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>27</td>\n",
       "      <td>HOXD9</td>\n",
       "      <td>5.770177</td>\n",
       "      <td>5.368568</td>\n",
       "      <td>10</td>\n",
       "      <td>16367.0</td>\n",
       "      <td>-0.867869</td>\n",
       "      <td>7.923143</td>\n",
       "      <td>635.390948</td>\n",
       "      <td>0.096639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2437</td>\n",
       "      <td>18</td>\n",
       "      <td>34121148</td>\n",
       "      <td>18:34121148</td>\n",
       "      <td>8.382969e-09</td>\n",
       "      <td>3.808910</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>27</td>\n",
       "      <td>Wt1</td>\n",
       "      <td>7.943770</td>\n",
       "      <td>8.079233</td>\n",
       "      <td>14</td>\n",
       "      <td>4695.0</td>\n",
       "      <td>-6.213578</td>\n",
       "      <td>-7.654303</td>\n",
       "      <td>9124.668187</td>\n",
       "      <td>38540.377390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2438</td>\n",
       "      <td>18</td>\n",
       "      <td>34121148</td>\n",
       "      <td>18:34121148</td>\n",
       "      <td>8.382969e-09</td>\n",
       "      <td>3.808910</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>27</td>\n",
       "      <td>ZNF148</td>\n",
       "      <td>7.421137</td>\n",
       "      <td>7.421137</td>\n",
       "      <td>12</td>\n",
       "      <td>11203.0</td>\n",
       "      <td>-7.155008</td>\n",
       "      <td>-7.155008</td>\n",
       "      <td>9376.740215</td>\n",
       "      <td>9376.740215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2443</td>\n",
       "      <td>18</td>\n",
       "      <td>34121148</td>\n",
       "      <td>18:34121148</td>\n",
       "      <td>8.382969e-09</td>\n",
       "      <td>3.808910</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>27</td>\n",
       "      <td>KLF5</td>\n",
       "      <td>5.659934</td>\n",
       "      <td>5.624972</td>\n",
       "      <td>10</td>\n",
       "      <td>13615.0</td>\n",
       "      <td>-5.145902</td>\n",
       "      <td>0.607487</td>\n",
       "      <td>57822.758530</td>\n",
       "      <td>183.414645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2477</td>\n",
       "      <td>8</td>\n",
       "      <td>6410987</td>\n",
       "      <td>8:6410987</td>\n",
       "      <td>1.175133e-08</td>\n",
       "      <td>1.977696</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>27</td>\n",
       "      <td>NKX6-3</td>\n",
       "      <td>6.062696</td>\n",
       "      <td>5.247429</td>\n",
       "      <td>9</td>\n",
       "      <td>6909.0</td>\n",
       "      <td>-7.040466</td>\n",
       "      <td>-5.152894</td>\n",
       "      <td>22816.473017</td>\n",
       "      <td>3455.304880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2669</td>\n",
       "      <td>15</td>\n",
       "      <td>28238895</td>\n",
       "      <td>15:28238895</td>\n",
       "      <td>1.490933e-08</td>\n",
       "      <td>1.056619</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>27</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>7.162255</td>\n",
       "      <td>7.162255</td>\n",
       "      <td>14</td>\n",
       "      <td>26027.0</td>\n",
       "      <td>-5.118098</td>\n",
       "      <td>-5.118098</td>\n",
       "      <td>151.173459</td>\n",
       "      <td>151.173459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2907</td>\n",
       "      <td>4</td>\n",
       "      <td>137664056</td>\n",
       "      <td>4:137664056</td>\n",
       "      <td>7.071539e-08</td>\n",
       "      <td>-1.514088</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23</td>\n",
       "      <td>Wt1</td>\n",
       "      <td>6.424068</td>\n",
       "      <td>5.671778</td>\n",
       "      <td>14</td>\n",
       "      <td>4695.0</td>\n",
       "      <td>-2.108683</td>\n",
       "      <td>1.915546</td>\n",
       "      <td>150.481751</td>\n",
       "      <td>2.690194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2922</td>\n",
       "      <td>21</td>\n",
       "      <td>41166390</td>\n",
       "      <td>21:41166390</td>\n",
       "      <td>1.009829e-07</td>\n",
       "      <td>-1.690931</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23</td>\n",
       "      <td>Foxo1</td>\n",
       "      <td>6.167602</td>\n",
       "      <td>5.787891</td>\n",
       "      <td>11</td>\n",
       "      <td>2494.0</td>\n",
       "      <td>-1.148713</td>\n",
       "      <td>0.566090</td>\n",
       "      <td>1264.012805</td>\n",
       "      <td>227.521401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2994</td>\n",
       "      <td>5</td>\n",
       "      <td>79639772</td>\n",
       "      <td>5:79639772</td>\n",
       "      <td>1.388130e-07</td>\n",
       "      <td>-1.274068</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23</td>\n",
       "      <td>Wt1</td>\n",
       "      <td>7.977636</td>\n",
       "      <td>7.225346</td>\n",
       "      <td>14</td>\n",
       "      <td>4695.0</td>\n",
       "      <td>-7.371103</td>\n",
       "      <td>-3.398698</td>\n",
       "      <td>29035.149702</td>\n",
       "      <td>546.676153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2995</td>\n",
       "      <td>5</td>\n",
       "      <td>79639772</td>\n",
       "      <td>5:79639772</td>\n",
       "      <td>1.388130e-07</td>\n",
       "      <td>-1.274068</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23</td>\n",
       "      <td>ZNF148</td>\n",
       "      <td>7.405695</td>\n",
       "      <td>6.865750</td>\n",
       "      <td>12</td>\n",
       "      <td>11203.0</td>\n",
       "      <td>-6.183973</td>\n",
       "      <td>-3.951693</td>\n",
       "      <td>3550.884789</td>\n",
       "      <td>380.951632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3000</td>\n",
       "      <td>5</td>\n",
       "      <td>79639772</td>\n",
       "      <td>5:79639772</td>\n",
       "      <td>1.388130e-07</td>\n",
       "      <td>-1.274068</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23</td>\n",
       "      <td>KLF5</td>\n",
       "      <td>5.669556</td>\n",
       "      <td>4.669849</td>\n",
       "      <td>10</td>\n",
       "      <td>13615.0</td>\n",
       "      <td>-5.626841</td>\n",
       "      <td>4.319318</td>\n",
       "      <td>93533.653770</td>\n",
       "      <td>4.481321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3421</td>\n",
       "      <td>2</td>\n",
       "      <td>222726924</td>\n",
       "      <td>2:222726924</td>\n",
       "      <td>2.066333e-09</td>\n",
       "      <td>7.635113</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>24</td>\n",
       "      <td>TEF</td>\n",
       "      <td>5.502854</td>\n",
       "      <td>5.502854</td>\n",
       "      <td>12</td>\n",
       "      <td>3604.0</td>\n",
       "      <td>0.167770</td>\n",
       "      <td>0.167770</td>\n",
       "      <td>1020.435722</td>\n",
       "      <td>1020.435722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3496</td>\n",
       "      <td>10</td>\n",
       "      <td>63473567</td>\n",
       "      <td>10:63473567</td>\n",
       "      <td>2.201977e-09</td>\n",
       "      <td>4.665304</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>24</td>\n",
       "      <td>FOXC2</td>\n",
       "      <td>6.867753</td>\n",
       "      <td>6.505517</td>\n",
       "      <td>12</td>\n",
       "      <td>20559.0</td>\n",
       "      <td>-6.734187</td>\n",
       "      <td>-5.612187</td>\n",
       "      <td>499.765799</td>\n",
       "      <td>162.737618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3516</td>\n",
       "      <td>10</td>\n",
       "      <td>63473567</td>\n",
       "      <td>10:63473567</td>\n",
       "      <td>2.201977e-09</td>\n",
       "      <td>4.665304</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>24</td>\n",
       "      <td>HOXD9</td>\n",
       "      <td>5.637873</td>\n",
       "      <td>4.853097</td>\n",
       "      <td>10</td>\n",
       "      <td>16367.0</td>\n",
       "      <td>-2.737325</td>\n",
       "      <td>0.134049</td>\n",
       "      <td>4120.360524</td>\n",
       "      <td>233.299207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3574</td>\n",
       "      <td>15</td>\n",
       "      <td>54456230</td>\n",
       "      <td>15:54456230</td>\n",
       "      <td>2.438254e-09</td>\n",
       "      <td>4.719345</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>24</td>\n",
       "      <td>ZNF384</td>\n",
       "      <td>7.735224</td>\n",
       "      <td>6.743930</td>\n",
       "      <td>12</td>\n",
       "      <td>30437.0</td>\n",
       "      <td>-7.383237</td>\n",
       "      <td>-0.379210</td>\n",
       "      <td>616.150799</td>\n",
       "      <td>0.559598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3646</td>\n",
       "      <td>11</td>\n",
       "      <td>118643571</td>\n",
       "      <td>11:118643571</td>\n",
       "      <td>3.082888e-09</td>\n",
       "      <td>8.068627</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>24</td>\n",
       "      <td>MEIS2</td>\n",
       "      <td>5.324302</td>\n",
       "      <td>4.533285</td>\n",
       "      <td>8</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>-5.167063</td>\n",
       "      <td>-2.048615</td>\n",
       "      <td>1197.507451</td>\n",
       "      <td>52.960710</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr        pos        snp_id        pvalue      beta ref_allele  \\\n",
       "149     3   78699639    3:78699639  1.260930e-09  6.238320          C   \n",
       "328    15   48683454   15:48683454  8.597562e-12  2.115559          T   \n",
       "433    11   61137147   11:61137147  7.483225e-10  2.144577          T   \n",
       "443    15   28238895   15:28238895  8.381399e-10  2.188545          A   \n",
       "523    15   48296524   15:48296524  1.368194e-08  1.720186          C   \n",
       "775    14   59697005   14:59697005  5.008992e-09  0.975950          G   \n",
       "776    14   59697005   14:59697005  5.008992e-09  0.975950          G   \n",
       "777    14   59697005   14:59697005  5.008992e-09  0.975950          G   \n",
       "984    15   48683454   15:48683454  8.913702e-11  1.348175          T   \n",
       "1041   15   28238895   15:28238895  5.209608e-10  1.485306          A   \n",
       "1241    8    6410987     8:6410987  1.721366e-08  2.507842          T   \n",
       "1410   15   48296524   15:48296524  2.308487e-08  1.135363          C   \n",
       "1533    5   32944420    5:32944420  2.959992e-08  2.554335          A   \n",
       "1558   18   34121148   18:34121148  3.193288e-08  4.692228          A   \n",
       "1559   18   34121148   18:34121148  3.193288e-08  4.692228          A   \n",
       "1564   18   34121148   18:34121148  3.193288e-08  4.692228          A   \n",
       "2087   15   48683454   15:48683454  7.755283e-10  0.997581          T   \n",
       "2220    5   32944420    5:32944420  2.213560e-09  2.146916          A   \n",
       "2437   18   34121148   18:34121148  8.382969e-09  3.808910          A   \n",
       "2438   18   34121148   18:34121148  8.382969e-09  3.808910          A   \n",
       "2443   18   34121148   18:34121148  8.382969e-09  3.808910          A   \n",
       "2477    8    6410987     8:6410987  1.175133e-08  1.977696          T   \n",
       "2669   15   28238895   15:28238895  1.490933e-08  1.056619          A   \n",
       "2907    4  137664056   4:137664056  7.071539e-08 -1.514088          G   \n",
       "2922   21   41166390   21:41166390  1.009829e-07 -1.690931          G   \n",
       "2994    5   79639772    5:79639772  1.388130e-07 -1.274068          G   \n",
       "2995    5   79639772    5:79639772  1.388130e-07 -1.274068          G   \n",
       "3000    5   79639772    5:79639772  1.388130e-07 -1.274068          G   \n",
       "3421    2  222726924   2:222726924  2.066333e-09  7.635113          T   \n",
       "3496   10   63473567   10:63473567  2.201977e-09  4.665304          C   \n",
       "3516   10   63473567   10:63473567  2.201977e-09  4.665304          C   \n",
       "3574   15   54456230   15:54456230  2.438254e-09  4.719345          A   \n",
       "3646   11  118643571  11:118643571  3.082888e-09  8.068627          C   \n",
       "\n",
       "     alt_allele trait_num TF_name  PWM_fracscore_ref  PWM_fracscore_alt  \\\n",
       "149           T         7   Foxo1           5.876905           5.872895   \n",
       "328           G        25    IRF6           5.044620           5.028799   \n",
       "433           A        25    TBX1           5.015816           4.020404   \n",
       "443           G        25  NKX2-2           7.162255           7.162255   \n",
       "523           T        25  ZNF384           7.695108           7.765220   \n",
       "775           A        20   KLF14           7.611608           7.609984   \n",
       "776           A        20  ZNF740           7.570647           6.623519   \n",
       "777           A        20  ZNF148           7.317058           6.585022   \n",
       "984           G        26    IRF6           5.044620           5.028799   \n",
       "1041          G        26  NKX2-2           7.162255           7.162255   \n",
       "1241          C        26  NKX6-3           6.062696           5.247429   \n",
       "1410          T        26  ZNF384           7.695108           7.765220   \n",
       "1533          T        26   HOXD9           5.770177           5.368568   \n",
       "1558          C        26     Wt1           7.943770           8.079233   \n",
       "1559          C        26  ZNF148           7.421137           7.421137   \n",
       "1564          C        26    KLF5           5.659934           5.624972   \n",
       "2087          G        27    IRF6           5.044620           5.028799   \n",
       "2220          T        27   HOXD9           5.770177           5.368568   \n",
       "2437          C        27     Wt1           7.943770           8.079233   \n",
       "2438          C        27  ZNF148           7.421137           7.421137   \n",
       "2443          C        27    KLF5           5.659934           5.624972   \n",
       "2477          C        27  NKX6-3           6.062696           5.247429   \n",
       "2669          G        27  NKX2-2           7.162255           7.162255   \n",
       "2907          A        23     Wt1           6.424068           5.671778   \n",
       "2922          A        23   Foxo1           6.167602           5.787891   \n",
       "2994          A        23     Wt1           7.977636           7.225346   \n",
       "2995          A        23  ZNF148           7.405695           6.865750   \n",
       "3000          A        23    KLF5           5.669556           4.669849   \n",
       "3421          G        24     TEF           5.502854           5.502854   \n",
       "3496          T        24   FOXC2           6.867753           6.505517   \n",
       "3516          T        24   HOXD9           5.637873           4.853097   \n",
       "3574          G        24  ZNF384           7.735224           6.743930   \n",
       "3646          T        24   MEIS2           5.324302           4.533285   \n",
       "\n",
       "      TF_len  TF_count   H (REF)  Hprime (ALT)  P binding (REF)  \\\n",
       "149       11    2494.0  0.032564      0.471262       387.908761   \n",
       "328        9    2665.0  0.344800      2.130402       134.608633   \n",
       "433        8   11935.0 -1.666418      7.720827      2008.596654   \n",
       "443       14   26027.0 -5.118098     -5.118098       151.173459   \n",
       "523       12   30437.0 -7.035665     -6.987407       435.249585   \n",
       "775       14    1604.0 -6.209140     -5.690172    348350.756348   \n",
       "776       13    1059.0 -3.911958      0.590858       284.893106   \n",
       "777       12   11203.0 -7.368900     -4.846557     11612.983051   \n",
       "984        9    2665.0  0.344800      2.130402       134.608633   \n",
       "1041      14   26027.0 -5.118098     -5.118098       151.173459   \n",
       "1241       9    6909.0 -7.040466     -5.152894     22816.473017   \n",
       "1410      12   30437.0 -7.035665     -6.987407       435.249585   \n",
       "1533      10   16367.0 -0.867869      7.923143       635.390948   \n",
       "1558      14    4695.0 -6.213578     -7.654303      9124.668187   \n",
       "1559      12   11203.0 -7.155008     -7.155008      9376.740215   \n",
       "1564      10   13615.0 -5.145902      0.607487     57822.758530   \n",
       "2087       9    2665.0  0.344800      2.130402       134.608633   \n",
       "2220      10   16367.0 -0.867869      7.923143       635.390948   \n",
       "2437      14    4695.0 -6.213578     -7.654303      9124.668187   \n",
       "2438      12   11203.0 -7.155008     -7.155008      9376.740215   \n",
       "2443      10   13615.0 -5.145902      0.607487     57822.758530   \n",
       "2477       9    6909.0 -7.040466     -5.152894     22816.473017   \n",
       "2669      14   26027.0 -5.118098     -5.118098       151.173459   \n",
       "2907      14    4695.0 -2.108683      1.915546       150.481751   \n",
       "2922      11    2494.0 -1.148713      0.566090      1264.012805   \n",
       "2994      14    4695.0 -7.371103     -3.398698     29035.149702   \n",
       "2995      12   11203.0 -6.183973     -3.951693      3550.884789   \n",
       "3000      10   13615.0 -5.626841      4.319318     93533.653770   \n",
       "3421      12    3604.0  0.167770      0.167770      1020.435722   \n",
       "3496      12   20559.0 -6.734187     -5.612187       499.765799   \n",
       "3516      10   16367.0 -2.737325      0.134049      4120.360524   \n",
       "3574      12   30437.0 -7.383237     -0.379210       616.150799   \n",
       "3646       8    1355.0 -5.167063     -2.048615      1197.507451   \n",
       "\n",
       "      P binding (ALT)  \n",
       "149        250.152886  \n",
       "328         22.573342  \n",
       "433          0.168292  \n",
       "443        151.173459  \n",
       "523        414.744055  \n",
       "775     207315.553111  \n",
       "776          3.155977  \n",
       "777        932.189034  \n",
       "984         22.573342  \n",
       "1041       151.173459  \n",
       "1241      3455.304880  \n",
       "1410       414.744055  \n",
       "1533         0.096639  \n",
       "1558     38540.377390  \n",
       "1559      9376.740215  \n",
       "1564       183.414645  \n",
       "2087        22.573342  \n",
       "2220         0.096639  \n",
       "2437     38540.377390  \n",
       "2438      9376.740215  \n",
       "2443       183.414645  \n",
       "2477      3455.304880  \n",
       "2669       151.173459  \n",
       "2907         2.690194  \n",
       "2922       227.521401  \n",
       "2994       546.676153  \n",
       "2995       380.951632  \n",
       "3000         4.481321  \n",
       "3421      1020.435722  \n",
       "3496       162.737618  \n",
       "3516       233.299207  \n",
       "3574         0.559598  \n",
       "3646        52.960710  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.loc[final_df['P binding (REF)'] > 100]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bg_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('/local3/jake/TF_binding/testing/tophit_SNPs/final_PWM.alltraits.allsnps.withPBinding.output',sep='\\t',index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import decimal\n",
    "\n",
    "def float_to_decimal(f):\n",
    "    # http://docs.python.org/library/decimal.html#decimal-faq\n",
    "    \"Convert a floating point number to a Decimal with no loss of information\"\n",
    "    n, d = f.as_integer_ratio()\n",
    "    numerator, denominator = decimal.Decimal(n), decimal.Decimal(d)\n",
    "    ctx = decimal.Context(prec=60)\n",
    "    result = ctx.divide(numerator, denominator)\n",
    "    while ctx.flags[decimal.Inexact]:\n",
    "        ctx.flags[decimal.Inexact] = False\n",
    "        ctx.prec *= 2\n",
    "        result = ctx.divide(numerator, denominator)\n",
    "    return result \n",
    "\n",
    "def f(number, sigfig):\n",
    "    # http://stackoverflow.com/questions/2663612/nicely-representing-a-floating-point-number-in-python/2663623#2663623\n",
    "    assert(sigfig>0)\n",
    "    try:\n",
    "        d=decimal.Decimal(number)\n",
    "    except TypeError:\n",
    "        d=float_to_decimal(float(number))\n",
    "    sign,digits,exponent=d.as_tuple()\n",
    "    if len(digits) < sigfig:\n",
    "        digits = list(digits)\n",
    "        digits.extend([0] * (sigfig - len(digits)))    \n",
    "    shift=d.adjusted()\n",
    "    result=int(''.join(map(str,digits[:sigfig])))\n",
    "    # Round the result\n",
    "    if len(digits)>sigfig and digits[sigfig]>=5: result+=1\n",
    "    result=list(str(result))\n",
    "    # Rounding can change the length of result\n",
    "    # If so, adjust shift\n",
    "    shift+=len(result)-sigfig\n",
    "    # reset len of result to sigfig\n",
    "    result=result[:sigfig]\n",
    "    if shift >= sigfig-1:\n",
    "        # Tack more zeros on the end\n",
    "        result+=['0']*(shift-sigfig+1)\n",
    "    elif 0<=shift:\n",
    "        # Place the decimal point in between digits\n",
    "        result.insert(shift+1,'.')\n",
    "    else:\n",
    "        # Tack zeros on the front\n",
    "        assert(shift<0)\n",
    "        result=['0.']+['0']*(-shift-1)+result\n",
    "    if sign:\n",
    "        result.insert(0,'-')\n",
    "    return ''.join(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomesize_file = open('/local3/jake/TF_binding/genomesize.all_chrms.hg19.txt','w')\n",
    "for row,c in bgfrac_df.iterrows():\n",
    "    genomesize_file.write('chr{0}\\t{1}\\n'.format(int(c['Chrm']),int(c['total'])))\n",
    "\n",
    "genomesize_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_full_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(chromosome,ref_pos_start,ref_pos_end),'/local3/jake/TF_binding/chr8.fa')\n",
    "tfname_list = list(set(final_df['TF_name']))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "decorated() missing 1 required positional argument: 'self'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-51-c28ff6094562>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# pybedtools.BedTool.random(genome='/local3/jake/TF_binding/genomesize.all_chrms.hg19.txt',l=10, n=10)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;31m# x = pybedtools.BedTool()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0my\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpybedtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mBedTool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrandom\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ml\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenome\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'hg19'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: decorated() missing 1 required positional argument: 'self'"
     ]
    }
   ],
   "source": [
    "test_vals = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/local3/jake/TF_binding/testing/chr8.fa'"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'{0}/{1}.fa'.format(ref_fasta_loc,'chr8')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "bg_H_by_TF = {}\n",
    "for f in transfac_matrix_list:\n",
    "    curr_matrix = read_JASPAR_transfac_pfms('{0}/{1}'.format(matrix_loc,f))\n",
    "    bgseqs = get_random_bgseqs(curr_matrix['TF_len'],2,ref_fasta_loc,chrmsizes_df)\n",
    "    \n",
    "    bg_H_list = []\n",
    "    for s in bgseqs:\n",
    "        curr_seq = s['seq']\n",
    "        bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == str(s['chrm'])][['frac_A','frac_C','frac_G','frac_T']]\n",
    "        curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "        curr_lnfracPWM = get_lnPWM_from_fracPWM(curr_fracPWM,bgfreqs)\n",
    "        curr_H = np.sum(get_matrix_scores(curr_lnfracPWM,curr_seq))\n",
    "        bg_H_list.append(curr_H)\n",
    "    curr_z = [pow(math.e,-x) for x in bg_H_list]\n",
    "    bg_H_by_TF[curr_matrix['Matrix_Name']] = sum(curr_z)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "bg_Z_byTF = {}\n",
    "for tf,h in bg_H_by_TF.items():\n",
    "    curr_z = [pow(math.e,-x) for x in h]\n",
    "    bg_Z_byTF[tf] = sum(curr_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "bgseqs = get_random_bgseqs(10,5,'/local3/jake/TF_binding')\n",
    "bg_H_list = []\n",
    "bg_H_by_TF = {}\n",
    "for s in bgseqs:\n",
    "    curr_seq = s['seq']\n",
    "    curr_matrix = get_matrix_byTF('TEF',infodicts_list)\n",
    "    bgfreqs = bgfrac_df.loc[bgfrac_df['Chrm'] == '8'][['frac_A','frac_C','frac_G','frac_T']]\n",
    "    curr_fracPWM = get_fracPWM_from_matrixdict(curr_matrix)\n",
    "    curr_lnfracPWM = get_lnPWM_from_fracPWM(curr_fracPWM,bgfreqs)\n",
    "    curr_H = np.sum(get_matrix_scores(curr_lnfracPWM,curr_seq))\n",
    "    bg_H_list.append(curr_H)\n",
    "#     bg_H_list.append()\n",
    "bg_H_by_TF['TEST_TF'] = bg_H_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'Matrix_ID': 'MA0843.1',\n",
       "  'Matrix_Name': 'TEF',\n",
       "  1: {'A': 542.0, 'C': 1054.0, 'G': 854.0, 'T': 1155.0},\n",
       "  2: {'A': 1972.0, 'C': 373.0, 'G': 1631.0, 'T': 9.0},\n",
       "  3: {'A': 91.0, 'C': 19.0, 'G': 6.0, 'T': 3602.0},\n",
       "  4: {'A': 108.0, 'C': 1.0, 'G': 32.0, 'T': 3602.0},\n",
       "  5: {'A': 3602.0, 'C': 1.0, 'G': 109.0, 'T': 1.0},\n",
       "  6: {'A': 18.0, 'C': 3602.0, 'G': 1.0, 'T': 447.0},\n",
       "  7: {'A': 1325.0, 'C': 1.0, 'G': 2278.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 134.0, 'G': 44.0, 'T': 3602.0},\n",
       "  9: {'A': 3602.0, 'C': 77.0, 'G': 12.0, 'T': 162.0},\n",
       "  10: {'A': 3602.0, 'C': 1.0, 'G': 20.0, 'T': 11.0},\n",
       "  11: {'A': 1.0, 'C': 2121.0, 'G': 270.0, 'T': 1482.0},\n",
       "  12: {'A': 1384.0, 'C': 1142.0, 'G': 555.0, 'T': 523.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TEF-1-related',\n",
       "  'tf_class': 'TEA',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q10587',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0715.1',\n",
       "  'Matrix_Name': 'PROP1',\n",
       "  1: {'A': 64.0, 'C': 13.0, 'G': 11.0, 'T': 576.0},\n",
       "  2: {'A': 576.0, 'C': 1.0, 'G': 1.0, 'T': 12.0},\n",
       "  3: {'A': 576.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  4: {'A': 42.0, 'C': 40.0, 'G': 19.0, 'T': 576.0},\n",
       "  5: {'A': 29.0, 'C': 210.0, 'G': 21.0, 'T': 368.0},\n",
       "  6: {'A': 184.0, 'C': 58.0, 'G': 79.0, 'T': 259.0},\n",
       "  7: {'A': 576.0, 'C': 39.0, 'G': 1.0, 'T': 34.0},\n",
       "  8: {'A': 576.0, 'C': 4.0, 'G': 47.0, 'T': 7.0},\n",
       "  9: {'A': 3.0, 'C': 1.0, 'G': 1.0, 'T': 576.0},\n",
       "  10: {'A': 14.0, 'C': 19.0, 'G': 7.0, 'T': 576.0},\n",
       "  11: {'A': 576.0, 'C': 1.0, 'G': 16.0, 'T': 48.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'O75360',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1504.1',\n",
       "  'Matrix_Name': 'HOXC4',\n",
       "  1: {'A': 2732.0, 'C': 1868.0, 'G': 2810.0, 'T': 1354.0},\n",
       "  2: {'A': 189.0, 'C': 3112.0, 'G': 1.0, 'T': 8760.0},\n",
       "  3: {'A': 3608.0, 'C': 5153.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 8760.0, 'C': 1.0, 'G': 584.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 8760.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 8760.0},\n",
       "  7: {'A': 8760.0, 'C': 1.0, 'G': 826.0, 'T': 1.0},\n",
       "  8: {'A': 2777.0, 'C': 1680.0, 'G': 2714.0, 'T': 1593.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P09017',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0739.1',\n",
       "  'Matrix_Name': 'Hic1',\n",
       "  1: {'A': 6338.0, 'C': 720.0, 'G': 4671.0, 'T': 667.0},\n",
       "  2: {'A': 63.0, 'C': 423.0, 'G': 74.0, 'T': 11008.0},\n",
       "  3: {'A': 951.0, 'C': 589.0, 'G': 11008.0, 'T': 184.0},\n",
       "  4: {'A': 51.0, 'C': 11008.0, 'G': 122.0, 'T': 1.0},\n",
       "  5: {'A': 89.0, 'C': 11008.0, 'G': 32.0, 'T': 69.0},\n",
       "  6: {'A': 9009.0, 'C': 2000.0, 'G': 4.0, 'T': 318.0},\n",
       "  7: {'A': 6141.0, 'C': 1987.0, 'G': 2268.0, 'T': 615.0},\n",
       "  8: {'A': 1221.0, 'C': 11008.0, 'G': 1630.0, 'T': 1489.0},\n",
       "  9: {'A': 1611.0, 'C': 4823.0, 'G': 1885.0, 'T': 2693.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '15231840',\n",
       "  'uniprot_ids': 'Q9R1Y5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0857.1',\n",
       "  'Matrix_Name': 'Rarb',\n",
       "  1: {'A': 1099.0, 'C': 258.0, 'G': 182.0, 'T': 414.0},\n",
       "  2: {'A': 1349.0, 'C': 34.0, 'G': 240.0, 'T': 92.0},\n",
       "  3: {'A': 1310.0, 'C': 3.0, 'G': 244.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1965.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 2.0, 'G': 1942.0, 'T': 32.0},\n",
       "  6: {'A': 7.0, 'C': 5.0, 'G': 16.0, 'T': 1778.0},\n",
       "  7: {'A': 2.0, 'C': 1317.0, 'G': 10.0, 'T': 9.0},\n",
       "  8: {'A': 1619.0, 'C': 1.0, 'G': 15.0, 'T': 1.0},\n",
       "  9: {'A': 1560.0, 'C': 12.0, 'G': 4.0, 'T': 85.0},\n",
       "  10: {'A': 1444.0, 'C': 5.0, 'G': 174.0, 'T': 25.0},\n",
       "  11: {'A': 1542.0, 'C': 1.0, 'G': 31.0, 'T': 1.0},\n",
       "  12: {'A': 2.0, 'C': 1.0, 'G': 1973.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 2.0, 'G': 1775.0, 'T': 273.0},\n",
       "  14: {'A': 2.0, 'C': 9.0, 'G': 5.0, 'T': 1718.0},\n",
       "  15: {'A': 2.0, 'C': 1410.0, 'G': 3.0, 'T': 9.0},\n",
       "  16: {'A': 1564.0, 'C': 1.0, 'G': 4.0, 'T': 3.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P22605',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1553.1',\n",
       "  'Matrix_Name': 'RARG(var.3)',\n",
       "  1: {'A': 1690.0, 'C': 297.0, 'G': 979.0, 'T': 480.0},\n",
       "  2: {'A': 2762.0, 'C': 333.0, 'G': 682.0, 'T': 109.0},\n",
       "  3: {'A': 112.0, 'C': 38.0, 'G': 3443.0, 'T': 157.0},\n",
       "  4: {'A': 40.0, 'C': 29.0, 'G': 3443.0, 'T': 1110.0},\n",
       "  5: {'A': 345.0, 'C': 294.0, 'G': 324.0, 'T': 3443.0},\n",
       "  6: {'A': 29.0, 'C': 3443.0, 'G': 59.0, 'T': 151.0},\n",
       "  7: {'A': 2608.0, 'C': 10.0, 'G': 836.0, 'T': 23.0},\n",
       "  8: {'A': 34.0, 'C': 51.0, 'G': 14.0, 'T': 3443.0},\n",
       "  9: {'A': 92.0, 'C': 47.0, 'G': 3443.0, 'T': 13.0},\n",
       "  10: {'A': 3443.0, 'C': 197.0, 'G': 129.0, 'T': 201.0},\n",
       "  11: {'A': 998.0, 'C': 3443.0, 'G': 6.0, 'T': 13.0},\n",
       "  12: {'A': 107.0, 'C': 3443.0, 'G': 19.0, 'T': 49.0},\n",
       "  13: {'A': 55.0, 'C': 688.0, 'G': 276.0, 'T': 2757.0},\n",
       "  14: {'A': 428.0, 'C': 999.0, 'G': 283.0, 'T': 1737.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '1738600',\n",
       "  'uniprot_ids': 'P13631',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0066.1',\n",
       "  'Matrix_Name': 'PPARG',\n",
       "  1: {'A': 4.0, 'C': 9.0, 'G': 15.0, 'T': 4.0},\n",
       "  2: {'A': 4.0, 'C': 1.0, 'G': 1.0, 'T': 26.0},\n",
       "  3: {'A': 20.0, 'C': 1.0, 'G': 10.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 2.0, 'G': 28.0, 'T': 1.0},\n",
       "  5: {'A': 2.0, 'C': 1.0, 'G': 27.0, 'T': 2.0},\n",
       "  6: {'A': 1.0, 'C': 2.0, 'G': 5.0, 'T': 24.0},\n",
       "  7: {'A': 3.0, 'C': 24.0, 'G': 4.0, 'T': 1.0},\n",
       "  8: {'A': 27.0, 'C': 2.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 6.0, 'C': 16.0, 'G': 5.0, 'T': 5.0},\n",
       "  10: {'A': 6.0, 'C': 8.0, 'G': 11.0, 'T': 7.0},\n",
       "  11: {'A': 5.0, 'C': 3.0, 'G': 19.0, 'T': 5.0},\n",
       "  12: {'A': 2.0, 'C': 1.0, 'G': 3.0, 'T': 26.0},\n",
       "  13: {'A': 3.0, 'C': 6.0, 'G': 21.0, 'T': 2.0},\n",
       "  14: {'A': 23.0, 'C': 6.0, 'G': 1.0, 'T': 2.0},\n",
       "  15: {'A': 2.0, 'C': 28.0, 'G': 1.0, 'T': 1.0},\n",
       "  16: {'A': 1.0, 'C': 26.0, 'G': 1.0, 'T': 4.0},\n",
       "  17: {'A': 4.0, 'C': 13.0, 'G': 1.0, 'T': 14.0},\n",
       "  18: {'A': 23.0, 'C': 6.0, 'G': 1.0, 'T': 2.0},\n",
       "  19: {'A': 6.0, 'C': 13.0, 'G': 7.0, 'T': 6.0},\n",
       "  20: {'A': 8.0, 'C': 1.0, 'G': 2.0, 'T': 21.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 20,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '11139380',\n",
       "  'uniprot_ids': 'P37231',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0812.1',\n",
       "  'Matrix_Name': 'TFAP2B(var.2)',\n",
       "  1: {'A': 1785.0, 'C': 1413.0, 'G': 233.0, 'T': 1122.0},\n",
       "  2: {'A': 37.0, 'C': 610.0, 'G': 3941.0, 'T': 31.0},\n",
       "  3: {'A': 1.0, 'C': 4550.0, 'G': 5.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 4550.0, 'G': 2.0, 'T': 159.0},\n",
       "  5: {'A': 17.0, 'C': 1205.0, 'G': 613.0, 'T': 2717.0},\n",
       "  6: {'A': 577.0, 'C': 2127.0, 'G': 1499.0, 'T': 349.0},\n",
       "  7: {'A': 3481.0, 'C': 609.0, 'G': 447.0, 'T': 15.0},\n",
       "  8: {'A': 236.0, 'C': 5.0, 'G': 4550.0, 'T': 3.0},\n",
       "  9: {'A': 7.0, 'C': 11.0, 'G': 4550.0, 'T': 2.0},\n",
       "  10: {'A': 48.0, 'C': 4251.0, 'G': 300.0, 'T': 26.0},\n",
       "  11: {'A': 1823.0, 'C': 372.0, 'G': 1150.0, 'T': 1208.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'Q92481',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1493.1',\n",
       "  'Matrix_Name': 'HES6',\n",
       "  1: {'A': 13.0, 'C': 40.0, 'G': 440.0, 'T': 44.0},\n",
       "  2: {'A': 17.0, 'C': 36.0, 'G': 440.0, 'T': 14.0},\n",
       "  3: {'A': 18.0, 'C': 440.0, 'G': 20.0, 'T': 16.0},\n",
       "  4: {'A': 440.0, 'C': 1.0, 'G': 10.0, 'T': 12.0},\n",
       "  5: {'A': 3.0, 'C': 440.0, 'G': 10.0, 'T': 17.0},\n",
       "  6: {'A': 11.0, 'C': 13.0, 'G': 440.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 8.0, 'G': 30.0, 'T': 440.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 440.0, 'T': 1.0},\n",
       "  9: {'A': 34.0, 'C': 22.0, 'G': 1.0, 'T': 440.0},\n",
       "  10: {'A': 80.0, 'C': 153.0, 'G': 1.0, 'T': 210.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11959828',\n",
       "  'uniprot_ids': 'Q96HZ4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0483.1',\n",
       "  'Matrix_Name': 'Gfi1b',\n",
       "  1: {'A': 1125.0, 'C': 439.0, 'G': 105.0, 'T': 96.0},\n",
       "  2: {'A': 1762.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 1760.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 339.0, 'G': 29.0, 'T': 1396.0},\n",
       "  5: {'A': 1.0, 'C': 1762.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1034.0, 'C': 22.0, 'G': 1.0, 'T': 708.0},\n",
       "  7: {'A': 62.0, 'C': 1313.0, 'G': 389.0, 'T': 1.0},\n",
       "  8: {'A': 1117.0, 'C': 1.0, 'G': 2.0, 'T': 645.0},\n",
       "  9: {'A': 58.0, 'C': 2.0, 'G': 1677.0, 'T': 28.0},\n",
       "  10: {'A': 143.0, 'C': 1329.0, 'G': 77.0, 'T': 216.0},\n",
       "  11: {'A': 741.0, 'C': 434.0, 'G': 131.0, 'T': 459.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19773260',\n",
       "  'uniprot_ids': 'O70237',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1140.2',\n",
       "  'Matrix_Name': 'JUNB(var.2)',\n",
       "  1: {'A': 51.0, 'C': 47.0, 'G': 163.0, 'T': 48.0},\n",
       "  2: {'A': 306.0, 'C': 8.0, 'G': 16.0, 'T': 7.0},\n",
       "  3: {'A': 2.0, 'C': 2.0, 'G': 5.0, 'T': 306.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 306.0, 'T': 14.0},\n",
       "  5: {'A': 306.0, 'C': 2.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 306.0, 'G': 1.0, 'T': 3.0},\n",
       "  7: {'A': 1.0, 'C': 2.0, 'G': 306.0, 'T': 1.0},\n",
       "  8: {'A': 2.0, 'C': 4.0, 'G': 2.0, 'T': 306.0},\n",
       "  9: {'A': 18.0, 'C': 306.0, 'G': 4.0, 'T': 2.0},\n",
       "  10: {'A': 306.0, 'C': 3.0, 'G': 2.0, 'T': 2.0},\n",
       "  11: {'A': 1.0, 'C': 115.0, 'G': 11.0, 'T': 192.0},\n",
       "  12: {'A': 41.0, 'C': 188.0, 'G': 44.0, 'T': 36.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '14585541',\n",
       "  'uniprot_ids': 'P17275',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0735.1',\n",
       "  'Matrix_Name': 'GLIS1',\n",
       "  1: {'A': 3237.0, 'C': 1843.0, 'G': 101.0, 'T': 892.0},\n",
       "  2: {'A': 4.0, 'C': 45.0, 'G': 8618.0, 'T': 1017.0},\n",
       "  3: {'A': 6315.0, 'C': 1.0, 'G': 7.0, 'T': 9.0},\n",
       "  4: {'A': 1.0, 'C': 4808.0, 'G': 3.0, 'T': 1.0},\n",
       "  5: {'A': 5.0, 'C': 4770.0, 'G': 3.0, 'T': 4.0},\n",
       "  6: {'A': 1.0, 'C': 4797.0, 'G': 12.0, 'T': 1.0},\n",
       "  7: {'A': 4.0, 'C': 4818.0, 'G': 6.0, 'T': 3.0},\n",
       "  8: {'A': 1.0, 'C': 4703.0, 'G': 1.0, 'T': 8.0},\n",
       "  9: {'A': 7.0, 'C': 4506.0, 'G': 6.0, 'T': 3.0},\n",
       "  10: {'A': 6782.0, 'C': 4.0, 'G': 161.0, 'T': 7.0},\n",
       "  11: {'A': 1.0, 'C': 4550.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 8.0, 'C': 1.0, 'G': 10082.0, 'T': 52.0},\n",
       "  13: {'A': 5760.0, 'C': 5.0, 'G': 33.0, 'T': 2025.0},\n",
       "  14: {'A': 2847.0, 'C': 1341.0, 'G': 1.0, 'T': 2787.0},\n",
       "  15: {'A': 8.0, 'C': 6.0, 'G': 9573.0, 'T': 25.0},\n",
       "  16: {'A': 858.0, 'C': 3055.0, 'G': 1345.0, 'T': 1030.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '22391303',\n",
       "  'uniprot_ids': 'Q8NBF1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0865.1',\n",
       "  'Matrix_Name': 'E2F8',\n",
       "  1: {'A': 4.0, 'C': 4.0, 'G': 2.0, 'T': 88.0},\n",
       "  2: {'A': 1.0, 'C': 4.0, 'G': 1.0, 'T': 91.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 93.0},\n",
       "  4: {'A': 1.0, 'C': 99.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 97.0, 'G': 1.0, 'T': 4.0},\n",
       "  6: {'A': 1.0, 'C': 98.0, 'G': 2.0, 'T': 1.0},\n",
       "  7: {'A': 5.0, 'C': 8.0, 'G': 120.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 97.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 3.0, 'C': 94.0, 'G': 5.0, 'T': 1.0},\n",
       "  10: {'A': 81.0, 'C': 2.0, 'G': 4.0, 'T': 2.0},\n",
       "  11: {'A': 84.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 80.0, 'C': 3.0, 'G': 2.0, 'T': 5.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '18836037',\n",
       "  'uniprot_ids': 'A0AVK6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0762.1',\n",
       "  'Matrix_Name': 'ETV2',\n",
       "  1: {'A': 1185.0, 'C': 252.0, 'G': 399.0, 'T': 370.0},\n",
       "  2: {'A': 1185.0, 'C': 11.0, 'G': 24.0, 'T': 6.0},\n",
       "  3: {'A': 58.0, 'C': 1185.0, 'G': 110.0, 'T': 1.0},\n",
       "  4: {'A': 117.0, 'C': 1185.0, 'G': 4.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1185.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1185.0, 'T': 1.0},\n",
       "  7: {'A': 1185.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1185.0, 'C': 2.0, 'G': 1.0, 'T': 144.0},\n",
       "  9: {'A': 847.0, 'C': 3.0, 'G': 338.0, 'T': 1.0},\n",
       "  10: {'A': 19.0, 'C': 441.0, 'G': 67.0, 'T': 1185.0},\n",
       "  11: {'A': 743.0, 'C': 162.0, 'G': 442.0, 'T': 232.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'Q3KNT2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0106.3',\n",
       "  'Matrix_Name': 'TP53',\n",
       "  1: {'A': 7545.0, 'C': 1038.0, 'G': 6564.0, 'T': 2269.0},\n",
       "  2: {'A': 10515.0, 'C': 117.0, 'G': 2434.0, 'T': 736.0},\n",
       "  3: {'A': 46.0, 'C': 19690.0, 'G': 14.0, 'T': 838.0},\n",
       "  4: {'A': 11932.0, 'C': 205.0, 'G': 654.0, 'T': 740.0},\n",
       "  5: {'A': 1711.0, 'C': 375.0, 'G': 138.0, 'T': 18894.0},\n",
       "  6: {'A': 9.0, 'C': 3.0, 'G': 20394.0, 'T': 1.0},\n",
       "  7: {'A': 245.0, 'C': 12015.0, 'G': 110.0, 'T': 7107.0},\n",
       "  8: {'A': 1229.0, 'C': 17287.0, 'G': 418.0, 'T': 2643.0},\n",
       "  9: {'A': 1146.0, 'C': 11876.0, 'G': 2152.0, 'T': 3775.0},\n",
       "  10: {'A': 3852.0, 'C': 1475.0, 'G': 9955.0, 'T': 2119.0},\n",
       "  11: {'A': 3585.0, 'C': 757.0, 'G': 17847.0, 'T': 1547.0},\n",
       "  12: {'A': 7105.0, 'C': 27.0, 'G': 8822.0, 'T': 191.0},\n",
       "  13: {'A': 1.0, 'C': 19351.0, 'G': 5.0, 'T': 2.0},\n",
       "  14: {'A': 12926.0, 'C': 251.0, 'G': 158.0, 'T': 1156.0},\n",
       "  15: {'A': 1826.0, 'C': 119.0, 'G': 90.0, 'T': 17865.0},\n",
       "  16: {'A': 328.0, 'C': 34.0, 'G': 20550.0, 'T': 69.0},\n",
       "  17: {'A': 880.0, 'C': 3339.0, 'G': 117.0, 'T': 17860.0},\n",
       "  18: {'A': 1473.0, 'C': 9184.0, 'G': 826.0, 'T': 9121.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'p53-related',\n",
       "  'tf_class': 'p53',\n",
       "  'pubmed_ids': '1588974',\n",
       "  'uniprot_ids': 'P04637',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0760.1',\n",
       "  'Matrix_Name': 'ERF',\n",
       "  1: {'A': 206.0, 'C': 5.0, 'G': 34.0, 'T': 4.0},\n",
       "  2: {'A': 8.0, 'C': 206.0, 'G': 9.0, 'T': 35.0},\n",
       "  3: {'A': 26.0, 'C': 206.0, 'G': 7.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 206.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 4.0, 'G': 206.0, 'T': 1.0},\n",
       "  6: {'A': 206.0, 'C': 4.0, 'G': 4.0, 'T': 3.0},\n",
       "  7: {'A': 206.0, 'C': 1.0, 'G': 1.0, 'T': 31.0},\n",
       "  8: {'A': 83.0, 'C': 18.0, 'G': 206.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 42.0, 'G': 12.0, 'T': 206.0},\n",
       "  10: {'A': 63.0, 'C': 29.0, 'G': 93.0, 'T': 23.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'P50548',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0782.2',\n",
       "  'Matrix_Name': 'PKNOX1',\n",
       "  1: {'A': 9812.0, 'C': 9504.0, 'G': 9600.0, 'T': 11051.0},\n",
       "  2: {'A': 10246.0, 'C': 9829.0, 'G': 9536.0, 'T': 10356.0},\n",
       "  3: {'A': 5223.0, 'C': 1626.0, 'G': 3323.0, 'T': 29795.0},\n",
       "  4: {'A': 1655.0, 'C': 1161.0, 'G': 36107.0, 'T': 1044.0},\n",
       "  5: {'A': 36760.0, 'C': 679.0, 'G': 1331.0, 'T': 1197.0},\n",
       "  6: {'A': 2326.0, 'C': 4124.0, 'G': 26183.0, 'T': 7334.0},\n",
       "  7: {'A': 1141.0, 'C': 879.0, 'G': 1028.0, 'T': 36919.0},\n",
       "  8: {'A': 556.0, 'C': 1005.0, 'G': 37714.0, 'T': 692.0},\n",
       "  9: {'A': 36375.0, 'C': 411.0, 'G': 2460.0, 'T': 721.0},\n",
       "  10: {'A': 652.0, 'C': 37847.0, 'G': 581.0, 'T': 887.0},\n",
       "  11: {'A': 33983.0, 'C': 945.0, 'G': 2409.0, 'T': 2630.0},\n",
       "  12: {'A': 2746.0, 'C': 4760.0, 'G': 27368.0, 'T': 5093.0},\n",
       "  13: {'A': 8453.0, 'C': 16342.0, 'G': 9129.0, 'T': 6043.0},\n",
       "  14: {'A': 8618.0, 'C': 9387.0, 'G': 9061.0, 'T': 12901.0},\n",
       "  15: {'A': 10523.0, 'C': 9269.0, 'G': 10297.0, 'T': 9878.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23602564',\n",
       "  'uniprot_ids': 'P55347',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0115.1',\n",
       "  'Matrix_Name': 'NR1H2::RXRA',\n",
       "  1: {'A': 18.0, 'C': 6.0, 'G': 1.0, 'T': 4.0},\n",
       "  2: {'A': 18.0, 'C': 2.0, 'G': 6.0, 'T': 3.0},\n",
       "  3: {'A': 21.0, 'C': 1.0, 'G': 6.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 26.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 25.0, 'T': 2.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 26.0},\n",
       "  7: {'A': 1.0, 'C': 26.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 26.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 25.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  10: {'A': 26.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 26.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 26.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 26.0},\n",
       "  14: {'A': 1.0, 'C': 26.0, 'G': 1.0, 'T': 1.0},\n",
       "  15: {'A': 26.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  16: {'A': 21.0, 'C': 2.0, 'G': 3.0, 'T': 3.0},\n",
       "  17: {'A': 1.0, 'C': 16.0, 'G': 7.0, 'T': 5.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '10187832',\n",
       "  'uniprot_ids': 'P55055;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1464.1',\n",
       "  'Matrix_Name': 'ARNT2',\n",
       "  1: {'A': 11782.0, 'C': 4266.0, 'G': 28842.0, 'T': 1952.0},\n",
       "  2: {'A': 9573.0, 'C': 4902.0, 'G': 9526.0, 'T': 40623.0},\n",
       "  3: {'A': 506.0, 'C': 40623.0, 'G': 116.0, 'T': 1.0},\n",
       "  4: {'A': 40623.0, 'C': 123.0, 'G': 533.0, 'T': 461.0},\n",
       "  5: {'A': 1.0, 'C': 40623.0, 'G': 1.0, 'T': 149.0},\n",
       "  6: {'A': 85.0, 'C': 1.0, 'G': 40623.0, 'T': 1.0},\n",
       "  7: {'A': 652.0, 'C': 933.0, 'G': 112.0, 'T': 40623.0},\n",
       "  8: {'A': 15.0, 'C': 1.0, 'G': 40623.0, 'T': 972.0},\n",
       "  9: {'A': 18560.0, 'C': 22064.0, 'G': 3913.0, 'T': 7160.0},\n",
       "  10: {'A': 7893.0, 'C': 12449.0, 'G': 12032.0, 'T': 8252.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q9HBZ2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0131.2',\n",
       "  'Matrix_Name': 'HINFP',\n",
       "  1: {'A': 24.0, 'C': 341.0, 'G': 107.0, 'T': 101.0},\n",
       "  2: {'A': 286.0, 'C': 132.0, 'G': 114.0, 'T': 54.0},\n",
       "  3: {'A': 277.0, 'C': 46.0, 'G': 218.0, 'T': 18.0},\n",
       "  4: {'A': 2.0, 'C': 389.0, 'G': 98.0, 'T': 8.0},\n",
       "  5: {'A': 1.0, 'C': 15.0, 'G': 485.0, 'T': 11.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 471.0},\n",
       "  7: {'A': 1.0, 'C': 405.0, 'G': 14.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 414.0, 'G': 1.0, 'T': 4.0},\n",
       "  9: {'A': 1.0, 'C': 32.0, 'G': 473.0, 'T': 22.0},\n",
       "  10: {'A': 2.0, 'C': 396.0, 'G': 16.0, 'T': 13.0},\n",
       "  11: {'A': 86.0, 'C': 56.0, 'G': 189.0, 'T': 98.0},\n",
       "  12: {'A': 78.0, 'C': 85.0, 'G': 155.0, 'T': 108.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '14752047',\n",
       "  'uniprot_ids': 'Q9BQA5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0658.1',\n",
       "  'Matrix_Name': 'LHX6',\n",
       "  1: {'A': 1858.0, 'C': 770.0, 'G': 1549.0, 'T': 265.0},\n",
       "  2: {'A': 530.0, 'C': 3016.0, 'G': 792.0, 'T': 1423.0},\n",
       "  3: {'A': 1.0, 'C': 254.0, 'G': 29.0, 'T': 4439.0},\n",
       "  4: {'A': 4439.0, 'C': 91.0, 'G': 488.0, 'T': 1.0},\n",
       "  5: {'A': 4439.0, 'C': 15.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 99.0, 'G': 16.0, 'T': 4439.0},\n",
       "  7: {'A': 1.0, 'C': 767.0, 'G': 109.0, 'T': 4439.0},\n",
       "  8: {'A': 4439.0, 'C': 1.0, 'G': 244.0, 'T': 3.0},\n",
       "  9: {'A': 1669.0, 'C': 638.0, 'G': 2770.0, 'T': 195.0},\n",
       "  10: {'A': 549.0, 'C': 1655.0, 'G': 729.0, 'T': 1508.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9UPM6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1635.1',\n",
       "  'Matrix_Name': 'BHLHE22(var.2)',\n",
       "  1: {'A': 3887.0, 'C': 5622.0, 'G': 4448.0, 'T': 4403.0},\n",
       "  2: {'A': 4454.0, 'C': 5111.0, 'G': 7268.0, 'T': 1527.0},\n",
       "  3: {'A': 1.0, 'C': 18332.0, 'G': 12.0, 'T': 15.0},\n",
       "  4: {'A': 18268.0, 'C': 38.0, 'G': 53.0, 'T': 1.0},\n",
       "  5: {'A': 2.0, 'C': 54.0, 'G': 18289.0, 'T': 15.0},\n",
       "  6: {'A': 15.0, 'C': 18289.0, 'G': 54.0, 'T': 2.0},\n",
       "  7: {'A': 1.0, 'C': 53.0, 'G': 38.0, 'T': 18268.0},\n",
       "  8: {'A': 15.0, 'C': 16.0, 'G': 18328.0, 'T': 1.0},\n",
       "  9: {'A': 1527.0, 'C': 7270.0, 'G': 5113.0, 'T': 4450.0},\n",
       "  10: {'A': 4399.0, 'C': 4448.0, 'G': 5626.0, 'T': 3887.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22284184',\n",
       "  'uniprot_ids': 'Q8NFJ8',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0807.1',\n",
       "  'Matrix_Name': 'TBX5',\n",
       "  1: {'A': 1805.0, 'C': 90.0, 'G': 410.0, 'T': 208.0},\n",
       "  2: {'A': 236.0, 'C': 191.0, 'G': 1805.0, 'T': 134.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1805.0, 'T': 29.0},\n",
       "  4: {'A': 46.0, 'C': 132.0, 'G': 167.0, 'T': 1805.0},\n",
       "  5: {'A': 28.0, 'C': 1.0, 'G': 1805.0, 'T': 1.0},\n",
       "  6: {'A': 163.0, 'C': 226.0, 'G': 202.0, 'T': 1805.0},\n",
       "  7: {'A': 127.0, 'C': 617.0, 'G': 1805.0, 'T': 916.0},\n",
       "  8: {'A': 1805.0, 'C': 77.0, 'G': 344.0, 'T': 284.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TBX2-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'Q99593',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0138.2',\n",
       "  'Matrix_Name': 'REST',\n",
       "  1: {'A': 212.0, 'C': 175.0, 'G': 367.0, 'T': 841.0},\n",
       "  2: {'A': 59.0, 'C': 270.0, 'G': 147.0, 'T': 1125.0},\n",
       "  3: {'A': 77.0, 'C': 1367.0, 'G': 51.0, 'T': 106.0},\n",
       "  4: {'A': 1453.0, 'C': 31.0, 'G': 95.0, 'T': 27.0},\n",
       "  5: {'A': 35.0, 'C': 45.0, 'G': 1517.0, 'T': 11.0},\n",
       "  6: {'A': 123.0, 'C': 979.0, 'G': 324.0, 'T': 183.0},\n",
       "  7: {'A': 1576.0, 'C': 8.0, 'G': 13.0, 'T': 13.0},\n",
       "  8: {'A': 4.0, 'C': 1587.0, 'G': 13.0, 'T': 6.0},\n",
       "  9: {'A': 36.0, 'C': 1482.0, 'G': 21.0, 'T': 71.0},\n",
       "  10: {'A': 914.0, 'C': 202.0, 'G': 163.0, 'T': 330.0},\n",
       "  11: {'A': 220.0, 'C': 376.0, 'G': 125.0, 'T': 887.0},\n",
       "  12: {'A': 40.0, 'C': 8.0, 'G': 1552.0, 'T': 8.0},\n",
       "  13: {'A': 21.0, 'C': 6.0, 'G': 1578.0, 'T': 3.0},\n",
       "  14: {'A': 1407.0, 'C': 113.0, 'G': 35.0, 'T': 52.0},\n",
       "  15: {'A': 14.0, 'C': 1281.0, 'G': 234.0, 'T': 75.0},\n",
       "  16: {'A': 1575.0, 'C': 10.0, 'G': 8.0, 'T': 11.0},\n",
       "  17: {'A': 43.0, 'C': 14.0, 'G': 1531.0, 'T': 10.0},\n",
       "  18: {'A': 206.0, 'C': 1008.0, 'G': 184.0, 'T': 199.0},\n",
       "  19: {'A': 367.0, 'C': 32.0, 'G': 689.0, 'T': 508.0},\n",
       "  20: {'A': 214.0, 'C': 934.0, 'G': 320.0, 'T': 126.0},\n",
       "  21: {'A': 180.0, 'C': 1115.0, 'G': 38.0, 'T': 261.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 21,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '17540862',\n",
       "  'uniprot_ids': 'Q13127',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0647.1',\n",
       "  'Matrix_Name': 'GRHL1',\n",
       "  1: {'A': 1346.0, 'C': 473.0, 'G': 248.0, 'T': 428.0},\n",
       "  2: {'A': 2492.0, 'C': 110.0, 'G': 410.0, 'T': 419.0},\n",
       "  3: {'A': 2492.0, 'C': 4.0, 'G': 113.0, 'T': 63.0},\n",
       "  4: {'A': 2492.0, 'C': 2.0, 'G': 8.0, 'T': 29.0},\n",
       "  5: {'A': 1.0, 'C': 2492.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 372.0, 'C': 2492.0, 'G': 52.0, 'T': 222.0},\n",
       "  7: {'A': 333.0, 'C': 53.0, 'G': 2492.0, 'T': 201.0},\n",
       "  8: {'A': 1.0, 'C': 2.0, 'G': 2492.0, 'T': 1.0},\n",
       "  9: {'A': 77.0, 'C': 5.0, 'G': 3.0, 'T': 2492.0},\n",
       "  10: {'A': 128.0, 'C': 187.0, 'G': 13.0, 'T': 2492.0},\n",
       "  11: {'A': 762.0, 'C': 538.0, 'G': 245.0, 'T': 2492.0},\n",
       "  12: {'A': 686.0, 'C': 290.0, 'G': 656.0, 'T': 863.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Grainyhead-related',\n",
       "  'tf_class': 'Grainyhead',\n",
       "  'pubmed_ids': '18288204',\n",
       "  'uniprot_ids': 'Q9NZI5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1128.1',\n",
       "  'Matrix_Name': 'FOSL1::JUN',\n",
       "  1: {'A': 209.0, 'C': 247.0, 'G': 254.0, 'T': 294.0},\n",
       "  2: {'A': 173.0, 'C': 160.0, 'G': 381.0, 'T': 290.0},\n",
       "  3: {'A': 603.0, 'C': 90.0, 'G': 284.0, 'T': 27.0},\n",
       "  4: {'A': 2.0, 'C': 3.0, 'G': 2.0, 'T': 997.0},\n",
       "  5: {'A': 4.0, 'C': 3.0, 'G': 958.0, 'T': 39.0},\n",
       "  6: {'A': 995.0, 'C': 3.0, 'G': 3.0, 'T': 4.0},\n",
       "  7: {'A': 27.0, 'C': 827.0, 'G': 52.0, 'T': 98.0},\n",
       "  8: {'A': 34.0, 'C': 3.0, 'G': 141.0, 'T': 827.0},\n",
       "  9: {'A': 160.0, 'C': 836.0, 'G': 3.0, 'T': 5.0},\n",
       "  10: {'A': 985.0, 'C': 5.0, 'G': 10.0, 'T': 5.0},\n",
       "  11: {'A': 58.0, 'C': 246.0, 'G': 120.0, 'T': 580.0},\n",
       "  12: {'A': 297.0, 'C': 309.0, 'G': 174.0, 'T': 224.0},\n",
       "  13: {'A': 278.0, 'C': 290.0, 'G': 238.0, 'T': 197.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15407;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1642.1',\n",
       "  'Matrix_Name': 'NEUROG2(var.2)',\n",
       "  1: {'A': 8691.0, 'C': 7354.0, 'G': 10814.0, 'T': 7536.0},\n",
       "  2: {'A': 9048.0, 'C': 6844.0, 'G': 10124.0, 'T': 8379.0},\n",
       "  3: {'A': 12090.0, 'C': 6468.0, 'G': 11115.0, 'T': 4722.0},\n",
       "  4: {'A': 16449.0, 'C': 8155.0, 'G': 8040.0, 'T': 1751.0},\n",
       "  5: {'A': 279.0, 'C': 33560.0, 'G': 318.0, 'T': 238.0},\n",
       "  6: {'A': 33721.0, 'C': 212.0, 'G': 302.0, 'T': 160.0},\n",
       "  7: {'A': 401.0, 'C': 794.0, 'G': 30393.0, 'T': 2807.0},\n",
       "  8: {'A': 32747.0, 'C': 953.0, 'G': 419.0, 'T': 276.0},\n",
       "  9: {'A': 650.0, 'C': 905.0, 'G': 324.0, 'T': 32516.0},\n",
       "  10: {'A': 281.0, 'C': 206.0, 'G': 33423.0, 'T': 485.0},\n",
       "  11: {'A': 1524.0, 'C': 2967.0, 'G': 24999.0, 'T': 4905.0},\n",
       "  12: {'A': 6666.0, 'C': 10442.0, 'G': 7582.0, 'T': 9705.0},\n",
       "  13: {'A': 9465.0, 'C': 9035.0, 'G': 8710.0, 'T': 7185.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '19389376',\n",
       "  'uniprot_ids': 'Q9H2A3',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0140.2',\n",
       "  'Matrix_Name': 'GATA1::TAL1',\n",
       "  1: {'A': 1115.0, 'C': 2039.0, 'G': 968.0, 'T': 837.0},\n",
       "  2: {'A': 207.0, 'C': 808.0, 'G': 160.0, 'T': 3784.0},\n",
       "  3: {'A': 319.0, 'C': 1.0, 'G': 1.0, 'T': 4638.0},\n",
       "  4: {'A': 4956.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 4956.0},\n",
       "  6: {'A': 1.0, 'C': 4956.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 2050.0, 'C': 64.0, 'G': 56.0, 'T': 2789.0},\n",
       "  8: {'A': 659.0, 'C': 1332.0, 'G': 2215.0, 'T': 753.0},\n",
       "  9: {'A': 964.0, 'C': 1215.0, 'G': 1245.0, 'T': 1535.0},\n",
       "  10: {'A': 1192.0, 'C': 1128.0, 'G': 1521.0, 'T': 1118.0},\n",
       "  11: {'A': 1779.0, 'C': 1131.0, 'G': 991.0, 'T': 1058.0},\n",
       "  12: {'A': 1039.0, 'C': 1238.0, 'G': 1713.0, 'T': 969.0},\n",
       "  13: {'A': 1353.0, 'C': 1143.0, 'G': 1467.0, 'T': 996.0},\n",
       "  14: {'A': 1886.0, 'C': 1034.0, 'G': 821.0, 'T': 1218.0},\n",
       "  15: {'A': 1382.0, 'C': 1238.0, 'G': 1885.0, 'T': 454.0},\n",
       "  16: {'A': 111.0, 'C': 4823.0, 'G': 24.0, 'T': 1.0},\n",
       "  17: {'A': 4660.0, 'C': 1.0, 'G': 1.0, 'T': 297.0},\n",
       "  18: {'A': 161.0, 'C': 891.0, 'G': 3545.0, 'T': 362.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '20566737',\n",
       "  'uniprot_ids': 'P15976;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1494.1',\n",
       "  'Matrix_Name': 'HNF4A(var.2)',\n",
       "  1: {'A': 213.0, 'C': 71.0, 'G': 326.0, 'T': 44.0},\n",
       "  2: {'A': 302.0, 'C': 8.0, 'G': 351.0, 'T': 8.0},\n",
       "  3: {'A': 4.0, 'C': 3.0, 'G': 652.0, 'T': 8.0},\n",
       "  4: {'A': 3.0, 'C': 2.0, 'G': 18.0, 'T': 652.0},\n",
       "  5: {'A': 101.0, 'C': 652.0, 'G': 42.0, 'T': 105.0},\n",
       "  6: {'A': 8.0, 'C': 652.0, 'G': 5.0, 'T': 16.0},\n",
       "  7: {'A': 652.0, 'C': 1.0, 'G': 8.0, 'T': 1.0},\n",
       "  8: {'A': 652.0, 'C': 2.0, 'G': 3.0, 'T': 4.0},\n",
       "  9: {'A': 652.0, 'C': 1.0, 'G': 6.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 652.0, 'T': 2.0},\n",
       "  11: {'A': 4.0, 'C': 1.0, 'G': 652.0, 'T': 65.0},\n",
       "  12: {'A': 1.0, 'C': 5.0, 'G': 25.0, 'T': 652.0},\n",
       "  13: {'A': 3.0, 'C': 652.0, 'G': 8.0, 'T': 36.0},\n",
       "  14: {'A': 466.0, 'C': 6.0, 'G': 187.0, 'T': 10.0},\n",
       "  15: {'A': 175.0, 'C': 218.0, 'G': 102.0, 'T': 160.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '25164757',\n",
       "  'uniprot_ids': 'P41235',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0889.1',\n",
       "  'Matrix_Name': 'GBX1',\n",
       "  1: {'A': 10002.0, 'C': 6728.0, 'G': 7839.0, 'T': 2775.0},\n",
       "  2: {'A': 1675.0, 'C': 14979.0, 'G': 6965.0, 'T': 3724.0},\n",
       "  3: {'A': 244.0, 'C': 13544.0, 'G': 16.0, 'T': 13797.0},\n",
       "  4: {'A': 27340.0, 'C': 590.0, 'G': 759.0, 'T': 158.0},\n",
       "  5: {'A': 27340.0, 'C': 173.0, 'G': 30.0, 'T': 24.0},\n",
       "  6: {'A': 46.0, 'C': 129.0, 'G': 2.0, 'T': 27340.0},\n",
       "  7: {'A': 313.0, 'C': 439.0, 'G': 247.0, 'T': 27340.0},\n",
       "  8: {'A': 27340.0, 'C': 51.0, 'G': 9.0, 'T': 322.0},\n",
       "  9: {'A': 5721.0, 'C': 6378.0, 'G': 12204.0, 'T': 3040.0},\n",
       "  10: {'A': 3527.0, 'C': 11718.0, 'G': 5442.0, 'T': 6656.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q14549',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1569.1',\n",
       "  'Matrix_Name': 'TFAP2E',\n",
       "  1: {'A': 2555.0, 'C': 5505.0, 'G': 1435.0, 'T': 4958.0},\n",
       "  2: {'A': 1.0, 'C': 6925.0, 'G': 14450.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 14450.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 14450.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 4769.0, 'G': 2054.0, 'T': 7859.0},\n",
       "  6: {'A': 1441.0, 'C': 5929.0, 'G': 5767.0, 'T': 1317.0},\n",
       "  7: {'A': 7717.0, 'C': 2420.0, 'G': 4464.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 14450.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 14450.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 14450.0, 'G': 6819.0, 'T': 1.0},\n",
       "  11: {'A': 4905.0, 'C': 1507.0, 'G': 5311.0, 'T': 2730.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'Q6VUC0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0063.2',\n",
       "  'Matrix_Name': 'NKX2-5',\n",
       "  1: {'A': 1724.0, 'C': 1025.0, 'G': 1688.0, 'T': 1011.0},\n",
       "  2: {'A': 1527.0, 'C': 1647.0, 'G': 1319.0, 'T': 955.0},\n",
       "  3: {'A': 97.0, 'C': 5003.0, 'G': 25.0, 'T': 323.0},\n",
       "  4: {'A': 5247.0, 'C': 49.0, 'G': 77.0, 'T': 75.0},\n",
       "  5: {'A': 145.0, 'C': 5099.0, 'G': 50.0, 'T': 154.0},\n",
       "  6: {'A': 136.0, 'C': 83.0, 'G': 62.0, 'T': 5167.0},\n",
       "  7: {'A': 223.0, 'C': 4320.0, 'G': 63.0, 'T': 842.0},\n",
       "  8: {'A': 4820.0, 'C': 121.0, 'G': 218.0, 'T': 289.0},\n",
       "  9: {'A': 4932.0, 'C': 166.0, 'G': 99.0, 'T': 251.0},\n",
       "  10: {'A': 1486.0, 'C': 1407.0, 'G': 1479.0, 'T': 1076.0},\n",
       "  11: {'A': 1828.0, 'C': 1252.0, 'G': 1085.0, 'T': 1283.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '27013732',\n",
       "  'uniprot_ids': 'P52952',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0095.2',\n",
       "  'Matrix_Name': 'YY1',\n",
       "  1: {'A': 1127.0, 'C': 4584.0, 'G': 802.0, 'T': 662.0},\n",
       "  2: {'A': 6976.0, 'C': 1.0, 'G': 182.0, 'T': 16.0},\n",
       "  3: {'A': 6742.0, 'C': 100.0, 'G': 269.0, 'T': 64.0},\n",
       "  4: {'A': 2507.0, 'C': 1118.0, 'G': 3283.0, 'T': 267.0},\n",
       "  5: {'A': 7172.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 13.0, 'G': 1.0, 'T': 7160.0},\n",
       "  7: {'A': 12.0, 'C': 1.0, 'G': 7161.0, 'T': 1.0},\n",
       "  8: {'A': 14.0, 'C': 1.0, 'G': 7159.0, 'T': 1.0},\n",
       "  9: {'A': 813.0, 'C': 5638.0, 'G': 39.0, 'T': 685.0},\n",
       "  10: {'A': 868.0, 'C': 1682.0, 'G': 2766.0, 'T': 1859.0},\n",
       "  11: {'A': 900.0, 'C': 876.0, 'G': 4656.0, 'T': 743.0},\n",
       "  12: {'A': 1333.0, 'C': 4569.0, 'G': 392.0, 'T': 881.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '18950698',\n",
       "  'uniprot_ids': 'P25490',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0507.1',\n",
       "  'Matrix_Name': 'POU2F2',\n",
       "  1: {'A': 416.0, 'C': 636.0, 'G': 403.0, 'T': 836.0},\n",
       "  2: {'A': 531.0, 'C': 267.0, 'G': 594.0, 'T': 899.0},\n",
       "  3: {'A': 606.0, 'C': 826.0, 'G': 416.0, 'T': 443.0},\n",
       "  4: {'A': 2017.0, 'C': 20.0, 'G': 2.0, 'T': 252.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2288.0},\n",
       "  6: {'A': 257.0, 'C': 3.0, 'G': 1.0, 'T': 2030.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2288.0},\n",
       "  8: {'A': 159.0, 'C': 1.0, 'G': 1998.0, 'T': 133.0},\n",
       "  9: {'A': 1.0, 'C': 2272.0, 'G': 1.0, 'T': 17.0},\n",
       "  10: {'A': 2258.0, 'C': 31.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 2286.0},\n",
       "  12: {'A': 1160.0, 'C': 40.0, 'G': 614.0, 'T': 477.0},\n",
       "  13: {'A': 595.0, 'C': 362.0, 'G': 108.0, 'T': 1226.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '11937630',\n",
       "  'uniprot_ids': 'P09086',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0709.1',\n",
       "  'Matrix_Name': 'Msx3',\n",
       "  1: {'A': 1382.0, 'C': 3248.0, 'G': 2497.0, 'T': 1257.0},\n",
       "  2: {'A': 313.0, 'C': 5799.0, 'G': 30.0, 'T': 2583.0},\n",
       "  3: {'A': 8381.0, 'C': 161.0, 'G': 238.0, 'T': 95.0},\n",
       "  4: {'A': 8381.0, 'C': 90.0, 'G': 39.0, 'T': 6.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 8381.0},\n",
       "  6: {'A': 122.0, 'C': 216.0, 'G': 106.0, 'T': 8381.0},\n",
       "  7: {'A': 8381.0, 'C': 60.0, 'G': 89.0, 'T': 214.0},\n",
       "  8: {'A': 2690.0, 'C': 1738.0, 'G': 2596.0, 'T': 1361.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P70354',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0663.1',\n",
       "  'Matrix_Name': 'MLX',\n",
       "  1: {'A': 178.0, 'C': 5.0, 'G': 111.0, 'T': 42.0},\n",
       "  2: {'A': 1.0, 'C': 18.0, 'G': 62.0, 'T': 323.0},\n",
       "  3: {'A': 4.0, 'C': 252.0, 'G': 4.0, 'T': 5.0},\n",
       "  4: {'A': 235.0, 'C': 1.0, 'G': 7.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 258.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 2.0, 'C': 1.0, 'G': 559.0, 'T': 2.0},\n",
       "  7: {'A': 3.0, 'C': 2.0, 'G': 2.0, 'T': 316.0},\n",
       "  8: {'A': 3.0, 'C': 1.0, 'G': 556.0, 'T': 6.0},\n",
       "  9: {'A': 241.0, 'C': 10.0, 'G': 36.0, 'T': 17.0},\n",
       "  10: {'A': 24.0, 'C': 85.0, 'G': 7.0, 'T': 236.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '14742444',\n",
       "  'uniprot_ids': 'Q9UH92',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0720.1',\n",
       "  'Matrix_Name': 'Shox2',\n",
       "  1: {'A': 2391.0, 'C': 7696.0, 'G': 3293.0, 'T': 4803.0},\n",
       "  2: {'A': 980.0, 'C': 9028.0, 'G': 229.0, 'T': 9153.0},\n",
       "  3: {'A': 18180.0, 'C': 627.0, 'G': 427.0, 'T': 767.0},\n",
       "  4: {'A': 18180.0, 'C': 139.0, 'G': 4.0, 'T': 80.0},\n",
       "  5: {'A': 30.0, 'C': 13.0, 'G': 1.0, 'T': 18180.0},\n",
       "  6: {'A': 598.0, 'C': 546.0, 'G': 871.0, 'T': 18180.0},\n",
       "  7: {'A': 18180.0, 'C': 194.0, 'G': 37.0, 'T': 896.0},\n",
       "  8: {'A': 6543.0, 'C': 3022.0, 'G': 6338.0, 'T': 2279.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P70390',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1596.1',\n",
       "  'Matrix_Name': 'ZNF460',\n",
       "  1: {'A': 413.0, 'C': 132.0, 'G': 1024.0, 'T': 47.0},\n",
       "  2: {'A': 50.0, 'C': 1265.0, 'G': 186.0, 'T': 115.0},\n",
       "  3: {'A': 20.0, 'C': 1461.0, 'G': 68.0, 'T': 67.0},\n",
       "  4: {'A': 28.0, 'C': 219.0, 'G': 84.0, 'T': 1285.0},\n",
       "  5: {'A': 18.0, 'C': 1520.0, 'G': 48.0, 'T': 30.0},\n",
       "  6: {'A': 852.0, 'C': 496.0, 'G': 182.0, 'T': 86.0},\n",
       "  7: {'A': 62.0, 'C': 24.0, 'G': 1494.0, 'T': 36.0},\n",
       "  8: {'A': 7.0, 'C': 1589.0, 'G': 8.0, 'T': 12.0},\n",
       "  9: {'A': 2.0, 'C': 1597.0, 'G': 10.0, 'T': 7.0},\n",
       "  10: {'A': 14.0, 'C': 13.0, 'G': 18.0, 'T': 1571.0},\n",
       "  11: {'A': 4.0, 'C': 1585.0, 'G': 16.0, 'T': 11.0},\n",
       "  12: {'A': 6.0, 'C': 1584.0, 'G': 16.0, 'T': 10.0},\n",
       "  13: {'A': 13.0, 'C': 1534.0, 'G': 28.0, 'T': 41.0},\n",
       "  14: {'A': 517.0, 'C': 111.0, 'G': 931.0, 'T': 57.0},\n",
       "  15: {'A': 988.0, 'C': 176.0, 'G': 422.0, 'T': 30.0},\n",
       "  16: {'A': 244.0, 'C': 159.0, 'G': 1148.0, 'T': 65.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q14592',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1647.1',\n",
       "  'Matrix_Name': 'PRDM4',\n",
       "  1: {'A': 1981.0, 'C': 1164.0, 'G': 2554.0, 'T': 1560.0},\n",
       "  2: {'A': 2042.0, 'C': 1013.0, 'G': 2080.0, 'T': 2124.0},\n",
       "  3: {'A': 282.0, 'C': 6505.0, 'G': 317.0, 'T': 155.0},\n",
       "  4: {'A': 861.0, 'C': 287.0, 'G': 567.0, 'T': 5544.0},\n",
       "  5: {'A': 294.0, 'C': 111.0, 'G': 6814.0, 'T': 40.0},\n",
       "  6: {'A': 148.0, 'C': 138.0, 'G': 83.0, 'T': 6890.0},\n",
       "  7: {'A': 170.0, 'C': 406.0, 'G': 394.0, 'T': 6289.0},\n",
       "  8: {'A': 196.0, 'C': 139.0, 'G': 112.0, 'T': 6812.0},\n",
       "  9: {'A': 482.0, 'C': 6354.0, 'G': 130.0, 'T': 293.0},\n",
       "  10: {'A': 2112.0, 'C': 1457.0, 'G': 727.0, 'T': 2963.0},\n",
       "  11: {'A': 2581.0, 'C': 1533.0, 'G': 1345.0, 'T': 1800.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23918801',\n",
       "  'uniprot_ids': 'Q9UKN5',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1657.1',\n",
       "  'Matrix_Name': 'ZNF652',\n",
       "  1: {'A': 2046.0, 'C': 2745.0, 'G': 4107.0, 'T': 3479.0},\n",
       "  2: {'A': 6159.0, 'C': 2127.0, 'G': 2231.0, 'T': 1860.0},\n",
       "  3: {'A': 11419.0, 'C': 373.0, 'G': 318.0, 'T': 267.0},\n",
       "  4: {'A': 11221.0, 'C': 90.0, 'G': 661.0, 'T': 405.0},\n",
       "  5: {'A': 268.0, 'C': 574.0, 'G': 10999.0, 'T': 536.0},\n",
       "  6: {'A': 7607.0, 'C': 107.0, 'G': 4280.0, 'T': 383.0},\n",
       "  7: {'A': 179.0, 'C': 110.0, 'G': 11941.0, 'T': 147.0},\n",
       "  8: {'A': 221.0, 'C': 156.0, 'G': 183.0, 'T': 11817.0},\n",
       "  9: {'A': 241.0, 'C': 275.0, 'G': 250.0, 'T': 11611.0},\n",
       "  10: {'A': 11709.0, 'C': 216.0, 'G': 206.0, 'T': 246.0},\n",
       "  11: {'A': 9190.0, 'C': 1217.0, 'G': 1033.0, 'T': 937.0},\n",
       "  12: {'A': 4036.0, 'C': 2716.0, 'G': 2186.0, 'T': 3439.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q9Y2D9',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0804.1',\n",
       "  'Matrix_Name': 'TBX19',\n",
       "  1: {'A': 10248.0, 'C': 5395.0, 'G': 7956.0, 'T': 15260.0},\n",
       "  2: {'A': 7364.0, 'C': 4214.0, 'G': 7814.0, 'T': 21131.0},\n",
       "  3: {'A': 344.0, 'C': 1044.0, 'G': 222.0, 'T': 44290.0},\n",
       "  4: {'A': 13893.0, 'C': 24810.0, 'G': 9199.0, 'T': 2403.0},\n",
       "  5: {'A': 23301.0, 'C': 457.0, 'G': 12032.0, 'T': 1890.0},\n",
       "  6: {'A': 77.0, 'C': 31490.0, 'G': 97.0, 'T': 229.0},\n",
       "  7: {'A': 31155.0, 'C': 51.0, 'G': 1900.0, 'T': 28.0},\n",
       "  8: {'A': 2676.0, 'C': 29631.0, 'G': 1508.0, 'T': 5304.0},\n",
       "  9: {'A': 7974.0, 'C': 15622.0, 'G': 7433.0, 'T': 4384.0},\n",
       "  10: {'A': 5456.0, 'C': 4320.0, 'G': 2832.0, 'T': 40121.0},\n",
       "  11: {'A': 29963.0, 'C': 1693.0, 'G': 4296.0, 'T': 4117.0},\n",
       "  12: {'A': 5374.0, 'C': 7006.0, 'G': 19802.0, 'T': 8984.0},\n",
       "  13: {'A': 8128.0, 'C': 1731.0, 'G': 43954.0, 'T': 3922.0},\n",
       "  14: {'A': 83.0, 'C': 2984.0, 'G': 266.0, 'T': 43500.0},\n",
       "  15: {'A': 279.0, 'C': 21.0, 'G': 48758.0, 'T': 129.0},\n",
       "  16: {'A': 2766.0, 'C': 12023.0, 'G': 695.0, 'T': 26201.0},\n",
       "  17: {'A': 2536.0, 'C': 6136.0, 'G': 39867.0, 'T': 13126.0},\n",
       "  18: {'A': 31607.0, 'C': 112.0, 'G': 859.0, 'T': 236.0},\n",
       "  19: {'A': 19938.0, 'C': 4954.0, 'G': 4530.0, 'T': 5443.0},\n",
       "  20: {'A': 16039.0, 'C': 5962.0, 'G': 5925.0, 'T': 8229.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 20,\n",
       "  'tf_family': 'Brachyury-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'O60806',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0594.2',\n",
       "  'Matrix_Name': 'HOXA9',\n",
       "  1: {'A': 652.0, 'C': 382.0, 'G': 994.0, 'T': 34.0},\n",
       "  2: {'A': 278.0, 'C': 535.0, 'G': 146.0, 'T': 1645.0},\n",
       "  3: {'A': 78.0, 'C': 1645.0, 'G': 17.0, 'T': 72.0},\n",
       "  4: {'A': 628.0, 'C': 18.0, 'G': 1645.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 48.0, 'G': 1.0, 'T': 1645.0},\n",
       "  6: {'A': 1645.0, 'C': 124.0, 'G': 75.0, 'T': 1263.0},\n",
       "  7: {'A': 1645.0, 'C': 87.0, 'G': 24.0, 'T': 178.0},\n",
       "  8: {'A': 1645.0, 'C': 875.0, 'G': 216.0, 'T': 876.0},\n",
       "  9: {'A': 371.0, 'C': 486.0, 'G': 337.0, 'T': 454.0},\n",
       "  10: {'A': 351.0, 'C': 450.0, 'G': 499.0, 'T': 348.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '12923056',\n",
       "  'uniprot_ids': 'P31269',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1496.1',\n",
       "  'Matrix_Name': 'HOXA4',\n",
       "  1: {'A': 1090.0, 'C': 398.0, 'G': 1366.0, 'T': 95.0},\n",
       "  2: {'A': 1.0, 'C': 968.0, 'G': 1.0, 'T': 2455.0},\n",
       "  3: {'A': 894.0, 'C': 1562.0, 'G': 177.0, 'T': 1.0},\n",
       "  4: {'A': 2455.0, 'C': 1.0, 'G': 685.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 40.0, 'T': 2455.0},\n",
       "  6: {'A': 1.0, 'C': 174.0, 'G': 172.0, 'T': 2455.0},\n",
       "  7: {'A': 2455.0, 'C': 147.0, 'G': 261.0, 'T': 1.0},\n",
       "  8: {'A': 637.0, 'C': 594.0, 'G': 841.0, 'T': 386.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7901228',\n",
       "  'uniprot_ids': 'Q00056',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0691.1',\n",
       "  'Matrix_Name': 'TFAP4',\n",
       "  1: {'A': 3424.0, 'C': 501.0, 'G': 338.0, 'T': 229.0},\n",
       "  2: {'A': 2160.0, 'C': 547.0, 'G': 63.0, 'T': 1265.0},\n",
       "  3: {'A': 1.0, 'C': 3424.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 3424.0, 'C': 3.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 3.0, 'C': 86.0, 'G': 3424.0, 'T': 15.0},\n",
       "  6: {'A': 22.0, 'C': 3424.0, 'G': 50.0, 'T': 7.0},\n",
       "  7: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 3424.0},\n",
       "  8: {'A': 1.0, 'C': 2.0, 'G': 3424.0, 'T': 1.0},\n",
       "  9: {'A': 1811.0, 'C': 115.0, 'G': 334.0, 'T': 1614.0},\n",
       "  10: {'A': 300.0, 'C': 683.0, 'G': 434.0, 'T': 3424.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '18818310',\n",
       "  'uniprot_ids': 'Q01664',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0614.1',\n",
       "  'Matrix_Name': 'Foxj2',\n",
       "  1: {'A': 462.0, 'C': 1.0, 'G': 539.0, 'T': 1.0},\n",
       "  2: {'A': 51.0, 'C': 26.0, 'G': 1.0, 'T': 925.0},\n",
       "  3: {'A': 796.0, 'C': 206.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 978.0, 'C': 1.0, 'G': 1.0, 'T': 23.0},\n",
       "  5: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 955.0, 'G': 1.0, 'T': 47.0},\n",
       "  7: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 594.0, 'C': 126.0, 'G': 126.0, 'T': 157.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '24025729',\n",
       "  'uniprot_ids': 'Q9ES18',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0683.1',\n",
       "  'Matrix_Name': 'POU4F2',\n",
       "  1: {'A': 1491.0, 'C': 480.0, 'G': 772.0, 'T': 510.0},\n",
       "  2: {'A': 111.0, 'C': 122.0, 'G': 54.0, 'T': 2758.0},\n",
       "  3: {'A': 194.0, 'C': 35.0, 'G': 3441.0, 'T': 82.0},\n",
       "  4: {'A': 1161.0, 'C': 2472.0, 'G': 3.0, 'T': 6.0},\n",
       "  5: {'A': 3080.0, 'C': 7.0, 'G': 5.0, 'T': 12.0},\n",
       "  6: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 2830.0},\n",
       "  7: {'A': 2991.0, 'C': 10.0, 'G': 15.0, 'T': 259.0},\n",
       "  8: {'A': 2622.0, 'C': 83.0, 'G': 54.0, 'T': 738.0},\n",
       "  9: {'A': 77.0, 'C': 55.0, 'G': 29.0, 'T': 2736.0},\n",
       "  10: {'A': 305.0, 'C': 10.0, 'G': 9.0, 'T': 2670.0},\n",
       "  11: {'A': 3099.0, 'C': 27.0, 'G': 235.0, 'T': 58.0},\n",
       "  12: {'A': 3188.0, 'C': 5.0, 'G': 10.0, 'T': 1.0},\n",
       "  13: {'A': 20.0, 'C': 31.0, 'G': 70.0, 'T': 2762.0},\n",
       "  14: {'A': 131.0, 'C': 169.0, 'G': 2194.0, 'T': 1073.0},\n",
       "  15: {'A': 3079.0, 'C': 233.0, 'G': 270.0, 'T': 237.0},\n",
       "  16: {'A': 740.0, 'C': 677.0, 'G': 2853.0, 'T': 286.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18077589',\n",
       "  'uniprot_ids': 'Q12837',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1525.1',\n",
       "  'Matrix_Name': 'NFATC4',\n",
       "  1: {'A': 1653.0, 'C': 917.0, 'G': 964.0, 'T': 820.0},\n",
       "  2: {'A': 4350.0, 'C': 330.0, 'G': 2370.0, 'T': 844.0},\n",
       "  3: {'A': 96.0, 'C': 1535.0, 'G': 1.0, 'T': 2817.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 4350.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 4350.0, 'T': 1.0},\n",
       "  6: {'A': 4350.0, 'C': 40.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 4350.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 4350.0, 'C': 1085.0, 'G': 128.0, 'T': 260.0},\n",
       "  9: {'A': 4350.0, 'C': 3277.0, 'G': 1790.0, 'T': 1293.0},\n",
       "  10: {'A': 1133.0, 'C': 633.0, 'G': 349.0, 'T': 2240.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NFAT-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '12374807',\n",
       "  'uniprot_ids': 'Q14934',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0608.1',\n",
       "  'Matrix_Name': 'Creb3l2',\n",
       "  1: {'A': 3.0, 'C': 3.0, 'G': 746.0, 'T': 251.0},\n",
       "  2: {'A': 251.0, 'C': 624.0, 'G': 2.0, 'T': 127.0},\n",
       "  3: {'A': 2.0, 'C': 873.0, 'G': 127.0, 'T': 2.0},\n",
       "  4: {'A': 997.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  5: {'A': 2.0, 'C': 997.0, 'G': 2.0, 'T': 2.0},\n",
       "  6: {'A': 2.0, 'C': 2.0, 'G': 997.0, 'T': 2.0},\n",
       "  7: {'A': 2.0, 'C': 2.0, 'G': 2.0, 'T': 997.0},\n",
       "  8: {'A': 2.0, 'C': 144.0, 'G': 855.0, 'T': 2.0},\n",
       "  9: {'A': 251.0, 'C': 3.0, 'G': 251.0, 'T': 499.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q8BH52',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0592.3',\n",
       "  'Matrix_Name': 'ESRRA',\n",
       "  1: {'A': 5232.0, 'C': 5067.0, 'G': 9008.0, 'T': 6692.0},\n",
       "  2: {'A': 3547.0, 'C': 8790.0, 'G': 6313.0, 'T': 7349.0},\n",
       "  3: {'A': 2306.0, 'C': 6447.0, 'G': 1851.0, 'T': 15395.0},\n",
       "  4: {'A': 973.0, 'C': 21855.0, 'G': 2324.0, 'T': 847.0},\n",
       "  5: {'A': 24682.0, 'C': 356.0, 'G': 557.0, 'T': 404.0},\n",
       "  6: {'A': 24745.0, 'C': 416.0, 'G': 570.0, 'T': 268.0},\n",
       "  7: {'A': 481.0, 'C': 341.0, 'G': 24677.0, 'T': 500.0},\n",
       "  8: {'A': 224.0, 'C': 147.0, 'G': 25333.0, 'T': 295.0},\n",
       "  9: {'A': 1004.0, 'C': 380.0, 'G': 950.0, 'T': 23665.0},\n",
       "  10: {'A': 294.0, 'C': 23990.0, 'G': 787.0, 'T': 928.0},\n",
       "  11: {'A': 24649.0, 'C': 227.0, 'G': 782.0, 'T': 341.0},\n",
       "  12: {'A': 5536.0, 'C': 9141.0, 'G': 4456.0, 'T': 6866.0},\n",
       "  13: {'A': 9326.0, 'C': 5587.0, 'G': 5893.0, 'T': 5193.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '17488637',\n",
       "  'uniprot_ids': 'P11474',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0643.1',\n",
       "  'Matrix_Name': 'Esrrg',\n",
       "  1: {'A': 3490.0, 'C': 5975.0, 'G': 2801.0, 'T': 14910.0},\n",
       "  2: {'A': 706.0, 'C': 10010.0, 'G': 4902.0, 'T': 491.0},\n",
       "  3: {'A': 14910.0, 'C': 1.0, 'G': 604.0, 'T': 196.0},\n",
       "  4: {'A': 14910.0, 'C': 13.0, 'G': 35.0, 'T': 6.0},\n",
       "  5: {'A': 13.0, 'C': 8.0, 'G': 14910.0, 'T': 299.0},\n",
       "  6: {'A': 1.0, 'C': 18.0, 'G': 14910.0, 'T': 15.0},\n",
       "  7: {'A': 473.0, 'C': 42.0, 'G': 845.0, 'T': 14910.0},\n",
       "  8: {'A': 8.0, 'C': 14910.0, 'G': 272.0, 'T': 1281.0},\n",
       "  9: {'A': 14910.0, 'C': 35.0, 'G': 1562.0, 'T': 61.0},\n",
       "  10: {'A': 4306.0, 'C': 3208.0, 'G': 1182.0, 'T': 6217.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15475169',\n",
       "  'uniprot_ids': 'P62509',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1633.1',\n",
       "  'Matrix_Name': 'BACH1',\n",
       "  1: {'A': 3258.0, 'C': 3947.0, 'G': 4836.0, 'T': 3520.0},\n",
       "  2: {'A': 6930.0, 'C': 3834.0, 'G': 3849.0, 'T': 948.0},\n",
       "  3: {'A': 106.0, 'C': 268.0, 'G': 150.0, 'T': 15037.0},\n",
       "  4: {'A': 98.0, 'C': 150.0, 'G': 14832.0, 'T': 481.0},\n",
       "  5: {'A': 15047.0, 'C': 231.0, 'G': 168.0, 'T': 115.0},\n",
       "  6: {'A': 444.0, 'C': 12598.0, 'G': 2084.0, 'T': 435.0},\n",
       "  7: {'A': 651.0, 'C': 221.0, 'G': 267.0, 'T': 14422.0},\n",
       "  8: {'A': 577.0, 'C': 14564.0, 'G': 179.0, 'T': 241.0},\n",
       "  9: {'A': 14779.0, 'C': 195.0, 'G': 311.0, 'T': 276.0},\n",
       "  10: {'A': 340.0, 'C': 624.0, 'G': 13051.0, 'T': 1546.0},\n",
       "  11: {'A': 770.0, 'C': 13383.0, 'G': 859.0, 'T': 549.0},\n",
       "  12: {'A': 8800.0, 'C': 2168.0, 'G': 1972.0, 'T': 2621.0},\n",
       "  13: {'A': 4493.0, 'C': 3575.0, 'G': 3554.0, 'T': 3939.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8887638',\n",
       "  'uniprot_ids': 'O14867',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0789.1',\n",
       "  'Matrix_Name': 'POU3F4',\n",
       "  1: {'A': 942.0, 'C': 350.0, 'G': 137.0, 'T': 11118.0},\n",
       "  2: {'A': 11118.0, 'C': 24.0, 'G': 69.0, 'T': 27.0},\n",
       "  3: {'A': 60.0, 'C': 60.0, 'G': 25.0, 'T': 11118.0},\n",
       "  4: {'A': 54.0, 'C': 20.0, 'G': 11118.0, 'T': 246.0},\n",
       "  5: {'A': 137.0, 'C': 11118.0, 'G': 603.0, 'T': 2566.0},\n",
       "  6: {'A': 6448.0, 'C': 11.0, 'G': 57.0, 'T': 4671.0},\n",
       "  7: {'A': 11118.0, 'C': 333.0, 'G': 316.0, 'T': 886.0},\n",
       "  8: {'A': 11118.0, 'C': 48.0, 'G': 89.0, 'T': 169.0},\n",
       "  9: {'A': 706.0, 'C': 248.0, 'G': 293.0, 'T': 11118.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8876240;',\n",
       "  'uniprot_ids': 'P49335',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0787.1',\n",
       "  'Matrix_Name': 'POU3F2',\n",
       "  1: {'A': 592.0, 'C': 62.0, 'G': 172.0, 'T': 701.0},\n",
       "  2: {'A': 164.0, 'C': 66.0, 'G': 41.0, 'T': 1293.0},\n",
       "  3: {'A': 1293.0, 'C': 17.0, 'G': 26.0, 'T': 1.0},\n",
       "  4: {'A': 29.0, 'C': 31.0, 'G': 13.0, 'T': 1293.0},\n",
       "  5: {'A': 20.0, 'C': 10.0, 'G': 1293.0, 'T': 61.0},\n",
       "  6: {'A': 69.0, 'C': 1293.0, 'G': 154.0, 'T': 373.0},\n",
       "  7: {'A': 689.0, 'C': 5.0, 'G': 2.0, 'T': 605.0},\n",
       "  8: {'A': 1293.0, 'C': 17.0, 'G': 33.0, 'T': 46.0},\n",
       "  9: {'A': 1293.0, 'C': 7.0, 'G': 1.0, 'T': 14.0},\n",
       "  10: {'A': 64.0, 'C': 20.0, 'G': 29.0, 'T': 1293.0},\n",
       "  11: {'A': 138.0, 'C': 166.0, 'G': 453.0, 'T': 841.0},\n",
       "  12: {'A': 1293.0, 'C': 263.0, 'G': 262.0, 'T': 563.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9852081',\n",
       "  'uniprot_ids': 'P20265',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0822.1',\n",
       "  'Matrix_Name': 'HES7',\n",
       "  1: {'A': 169.0, 'C': 791.0, 'G': 44.0, 'T': 864.0},\n",
       "  2: {'A': 25.0, 'C': 20.0, 'G': 1865.0, 'T': 4.0},\n",
       "  3: {'A': 161.0, 'C': 30.0, 'G': 1865.0, 'T': 3.0},\n",
       "  4: {'A': 2.0, 'C': 1865.0, 'G': 1.0, 'T': 4.0},\n",
       "  5: {'A': 1865.0, 'C': 4.0, 'G': 8.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1865.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 5.0, 'C': 2.0, 'G': 1865.0, 'T': 2.0},\n",
       "  8: {'A': 4.0, 'C': 4.0, 'G': 2.0, 'T': 1865.0},\n",
       "  9: {'A': 3.0, 'C': 2.0, 'G': 1865.0, 'T': 1.0},\n",
       "  10: {'A': 9.0, 'C': 1865.0, 'G': 6.0, 'T': 94.0},\n",
       "  11: {'A': 1.0, 'C': 1865.0, 'G': 10.0, 'T': 10.0},\n",
       "  12: {'A': 947.0, 'C': 48.0, 'G': 735.0, 'T': 138.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16342160',\n",
       "  'uniprot_ids': 'Q9BYE0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1528.1',\n",
       "  'Matrix_Name': 'NFIX(var.2)',\n",
       "  1: {'A': 3183.0, 'C': 4288.0, 'G': 6641.0, 'T': 3239.0},\n",
       "  2: {'A': 4531.0, 'C': 3913.0, 'G': 3398.0, 'T': 17347.0},\n",
       "  3: {'A': 128.0, 'C': 232.0, 'G': 2387.0, 'T': 17347.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 17347.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 17347.0, 'T': 1.0},\n",
       "  6: {'A': 1820.0, 'C': 17347.0, 'G': 1.0, 'T': 176.0},\n",
       "  7: {'A': 7105.0, 'C': 1891.0, 'G': 5091.0, 'T': 3263.0},\n",
       "  8: {'A': 3126.0, 'C': 5921.0, 'G': 3847.0, 'T': 4456.0},\n",
       "  9: {'A': 3062.0, 'C': 4770.0, 'G': 5992.0, 'T': 3526.0},\n",
       "  10: {'A': 3424.0, 'C': 2530.0, 'G': 7943.0, 'T': 3454.0},\n",
       "  11: {'A': 1774.0, 'C': 2301.0, 'G': 1424.0, 'T': 17347.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 17347.0, 'T': 460.0},\n",
       "  13: {'A': 1.0, 'C': 17347.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 1.0, 'C': 17347.0, 'G': 1.0, 'T': 1.0},\n",
       "  15: {'A': 17347.0, 'C': 1857.0, 'G': 64.0, 'T': 315.0},\n",
       "  16: {'A': 8331.0, 'C': 2166.0, 'G': 9017.0, 'T': 3323.0},\n",
       "  17: {'A': 2859.0, 'C': 5305.0, 'G': 5741.0, 'T': 3444.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '22456058',\n",
       "  'uniprot_ids': 'Q14938',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1487.1',\n",
       "  'Matrix_Name': 'FOXE1',\n",
       "  1: {'A': 55.0, 'C': 290.0, 'G': 166.0, 'T': 156.0},\n",
       "  2: {'A': 289.0, 'C': 382.0, 'G': 284.0, 'T': 1.0},\n",
       "  3: {'A': 38.0, 'C': 299.0, 'G': 39.0, 'T': 367.0},\n",
       "  4: {'A': 192.0, 'C': 1.0, 'G': 1.0, 'T': 665.0},\n",
       "  5: {'A': 665.0, 'C': 16.0, 'G': 31.0, 'T': 1.0},\n",
       "  6: {'A': 665.0, 'C': 58.0, 'G': 59.0, 'T': 195.0},\n",
       "  7: {'A': 665.0, 'C': 1.0, 'G': 229.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 325.0, 'G': 1.0, 'T': 341.0},\n",
       "  9: {'A': 665.0, 'C': 11.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 665.0, 'C': 9.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 665.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 665.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 665.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  14: {'A': 522.0, 'C': 85.0, 'G': 124.0, 'T': 143.0},\n",
       "  15: {'A': 149.0, 'C': 181.0, 'G': 101.0, 'T': 236.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23675434',\n",
       "  'uniprot_ids': 'O00358',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0030.1',\n",
       "  'Matrix_Name': 'FOXF2',\n",
       "  1: {'A': 2.0, 'C': 11.0, 'G': 8.0, 'T': 10.0},\n",
       "  2: {'A': 11.0, 'C': 8.0, 'G': 6.0, 'T': 6.0},\n",
       "  3: {'A': 18.0, 'C': 5.0, 'G': 3.0, 'T': 5.0},\n",
       "  4: {'A': 14.0, 'C': 6.0, 'G': 6.0, 'T': 6.0},\n",
       "  5: {'A': 4.0, 'C': 12.0, 'G': 9.0, 'T': 1.0},\n",
       "  6: {'A': 8.0, 'C': 1.0, 'G': 21.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 28.0},\n",
       "  8: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 26.0, 'G': 1.0, 'T': 3.0},\n",
       "  12: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 17.0, 'C': 5.0, 'G': 3.0, 'T': 6.0},\n",
       "  14: {'A': 8.0, 'C': 5.0, 'G': 7.0, 'T': 11.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '7957066',\n",
       "  'uniprot_ids': 'Q12947',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0730.1',\n",
       "  'Matrix_Name': 'RARA(var.2)',\n",
       "  1: {'A': 1090.0, 'C': 25.0, 'G': 314.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1489.0, 'T': 3.0},\n",
       "  3: {'A': 1.0, 'C': 3.0, 'G': 1452.0, 'T': 72.0},\n",
       "  4: {'A': 7.0, 'C': 12.0, 'G': 18.0, 'T': 1144.0},\n",
       "  5: {'A': 11.0, 'C': 1981.0, 'G': 10.0, 'T': 22.0},\n",
       "  6: {'A': 1392.0, 'C': 17.0, 'G': 12.0, 'T': 3.0},\n",
       "  7: {'A': 302.0, 'C': 398.0, 'G': 85.0, 'T': 582.0},\n",
       "  8: {'A': 214.0, 'C': 433.0, 'G': 585.0, 'T': 224.0},\n",
       "  9: {'A': 148.0, 'C': 678.0, 'G': 158.0, 'T': 503.0},\n",
       "  10: {'A': 973.0, 'C': 119.0, 'G': 199.0, 'T': 229.0},\n",
       "  11: {'A': 1016.0, 'C': 18.0, 'G': 349.0, 'T': 8.0},\n",
       "  12: {'A': 1127.0, 'C': 17.0, 'G': 270.0, 'T': 2.0},\n",
       "  13: {'A': 3.0, 'C': 1.0, 'G': 1456.0, 'T': 5.0},\n",
       "  14: {'A': 2.0, 'C': 8.0, 'G': 1441.0, 'T': 36.0},\n",
       "  15: {'A': 9.0, 'C': 12.0, 'G': 14.0, 'T': 1132.0},\n",
       "  16: {'A': 5.0, 'C': 1924.0, 'G': 27.0, 'T': 23.0},\n",
       "  17: {'A': 1487.0, 'C': 4.0, 'G': 26.0, 'T': 11.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P10276',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0767.1',\n",
       "  'Matrix_Name': 'GCM2',\n",
       "  1: {'A': 180.0, 'C': 341.0, 'G': 376.0, 'T': 428.0},\n",
       "  2: {'A': 1321.0, 'C': 78.0, 'G': 300.0, 'T': 46.0},\n",
       "  3: {'A': 24.0, 'C': 42.0, 'G': 25.0, 'T': 1321.0},\n",
       "  4: {'A': 264.0, 'C': 33.0, 'G': 1321.0, 'T': 1.0},\n",
       "  5: {'A': 85.0, 'C': 1321.0, 'G': 139.0, 'T': 128.0},\n",
       "  6: {'A': 52.0, 'C': 2.0, 'G': 1321.0, 'T': 67.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1321.0, 'T': 14.0},\n",
       "  8: {'A': 2.0, 'C': 1.0, 'G': 1321.0, 'T': 1.0},\n",
       "  9: {'A': 93.0, 'C': 534.0, 'G': 92.0, 'T': 1321.0},\n",
       "  10: {'A': 600.0, 'C': 141.0, 'G': 342.0, 'T': 241.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'GCM',\n",
       "  'tf_class': 'GCM',\n",
       "  'pubmed_ids': '8962155',\n",
       "  'uniprot_ids': 'O75603',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0117.2',\n",
       "  'Matrix_Name': 'Mafb',\n",
       "  1: {'A': 1238.0, 'C': 175.0, 'G': 327.0, 'T': 447.0},\n",
       "  2: {'A': 1264.0, 'C': 122.0, 'G': 160.0, 'T': 640.0},\n",
       "  3: {'A': 1081.0, 'C': 217.0, 'G': 183.0, 'T': 705.0},\n",
       "  4: {'A': 745.0, 'C': 399.0, 'G': 430.0, 'T': 612.0},\n",
       "  5: {'A': 265.0, 'C': 619.0, 'G': 32.0, 'T': 2183.0},\n",
       "  6: {'A': 3.0, 'C': 15.0, 'G': 2183.0, 'T': 6.0},\n",
       "  7: {'A': 296.0, 'C': 2183.0, 'G': 6.0, 'T': 8.0},\n",
       "  8: {'A': 32.0, 'C': 37.0, 'G': 3.0, 'T': 2183.0},\n",
       "  9: {'A': 46.0, 'C': 14.0, 'G': 2183.0, 'T': 171.0},\n",
       "  10: {'A': 2183.0, 'C': 14.0, 'G': 18.0, 'T': 14.0},\n",
       "  11: {'A': 183.0, 'C': 2183.0, 'G': 491.0, 'T': 274.0},\n",
       "  12: {'A': 583.0, 'C': 139.0, 'G': 647.0, 'T': 817.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9571165',\n",
       "  'uniprot_ids': 'P54841',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0040.1',\n",
       "  'Matrix_Name': 'Foxq1',\n",
       "  1: {'A': 5.0, 'C': 5.0, 'G': 4.0, 'T': 8.0},\n",
       "  2: {'A': 14.0, 'C': 2.0, 'G': 4.0, 'T': 2.0},\n",
       "  3: {'A': 6.0, 'C': 3.0, 'G': 1.0, 'T': 12.0},\n",
       "  4: {'A': 4.0, 'C': 1.0, 'G': 1.0, 'T': 16.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 19.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 19.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 19.0},\n",
       "  9: {'A': 18.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 2.0, 'G': 5.0, 'T': 14.0},\n",
       "  11: {'A': 7.0, 'C': 1.0, 'G': 4.0, 'T': 10.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '8139574',\n",
       "  'uniprot_ids': 'Q63244',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0898.1',\n",
       "  'Matrix_Name': 'Hmx3',\n",
       "  1: {'A': 50.0, 'C': 22.0, 'G': 22.0, 'T': 10.0},\n",
       "  2: {'A': 13.0, 'C': 50.0, 'G': 26.0, 'T': 14.0},\n",
       "  3: {'A': 49.0, 'C': 24.0, 'G': 20.0, 'T': 11.0},\n",
       "  4: {'A': 77.0, 'C': 8.0, 'G': 12.0, 'T': 7.0},\n",
       "  5: {'A': 16.0, 'C': 11.0, 'G': 60.0, 'T': 16.0},\n",
       "  6: {'A': 3.0, 'C': 95.0, 'G': 1.0, 'T': 5.0},\n",
       "  7: {'A': 85.0, 'C': 1.0, 'G': 17.0, 'T': 2.0},\n",
       "  8: {'A': 88.0, 'C': 13.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 100.0},\n",
       "  10: {'A': 2.0, 'C': 5.0, 'G': 1.0, 'T': 95.0},\n",
       "  11: {'A': 93.0, 'C': 1.0, 'G': 3.0, 'T': 6.0},\n",
       "  12: {'A': 88.0, 'C': 3.0, 'G': 4.0, 'T': 8.0},\n",
       "  13: {'A': 40.0, 'C': 16.0, 'G': 13.0, 'T': 36.0},\n",
       "  14: {'A': 27.0, 'C': 19.0, 'G': 39.0, 'T': 19.0},\n",
       "  15: {'A': 48.0, 'C': 22.0, 'G': 27.0, 'T': 7.0},\n",
       "  16: {'A': 53.0, 'C': 12.0, 'G': 19.0, 'T': 20.0},\n",
       "  17: {'A': 14.0, 'C': 11.0, 'G': 14.0, 'T': 65.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P42581',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0834.1',\n",
       "  'Matrix_Name': 'ATF7',\n",
       "  1: {'A': 3923.0, 'C': 6607.0, 'G': 4162.0, 'T': 4642.0},\n",
       "  2: {'A': 5131.0, 'C': 2441.0, 'G': 7300.0, 'T': 4462.0},\n",
       "  3: {'A': 19330.0, 'C': 369.0, 'G': 3773.0, 'T': 42.0},\n",
       "  4: {'A': 12.0, 'C': 5.0, 'G': 44.0, 'T': 19330.0},\n",
       "  5: {'A': 27.0, 'C': 8.0, 'G': 19330.0, 'T': 1760.0},\n",
       "  6: {'A': 19330.0, 'C': 9.0, 'G': 59.0, 'T': 13.0},\n",
       "  7: {'A': 21.0, 'C': 19330.0, 'G': 34.0, 'T': 69.0},\n",
       "  8: {'A': 80.0, 'C': 10.0, 'G': 19330.0, 'T': 5.0},\n",
       "  9: {'A': 26.0, 'C': 17.0, 'G': 78.0, 'T': 19330.0},\n",
       "  10: {'A': 2699.0, 'C': 19330.0, 'G': 52.0, 'T': 23.0},\n",
       "  11: {'A': 19330.0, 'C': 3.0, 'G': 50.0, 'T': 12.0},\n",
       "  12: {'A': 83.0, 'C': 7306.0, 'G': 441.0, 'T': 19330.0},\n",
       "  13: {'A': 5629.0, 'C': 8117.0, 'G': 2130.0, 'T': 3457.0},\n",
       "  14: {'A': 5681.0, 'C': 4921.0, 'G': 5893.0, 'T': 2840.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '2516827',\n",
       "  'uniprot_ids': 'P17544',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0726.1',\n",
       "  'Matrix_Name': 'VSX2',\n",
       "  1: {'A': 428.0, 'C': 9555.0, 'G': 446.0, 'T': 4256.0},\n",
       "  2: {'A': 1081.0, 'C': 2726.0, 'G': 364.0, 'T': 13810.0},\n",
       "  3: {'A': 13810.0, 'C': 398.0, 'G': 344.0, 'T': 403.0},\n",
       "  4: {'A': 13810.0, 'C': 125.0, 'G': 124.0, 'T': 64.0},\n",
       "  5: {'A': 119.0, 'C': 187.0, 'G': 93.0, 'T': 13810.0},\n",
       "  6: {'A': 627.0, 'C': 568.0, 'G': 448.0, 'T': 13810.0},\n",
       "  7: {'A': 13810.0, 'C': 322.0, 'G': 2110.0, 'T': 1015.0},\n",
       "  8: {'A': 2892.0, 'C': 3244.0, 'G': 4289.0, 'T': 3388.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P58304',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0634.1',\n",
       "  'Matrix_Name': 'ALX3',\n",
       "  1: {'A': 1252.0, 'C': 2175.0, 'G': 1063.0, 'T': 3391.0},\n",
       "  2: {'A': 988.0, 'C': 3403.0, 'G': 1151.0, 'T': 2338.0},\n",
       "  3: {'A': 795.0, 'C': 3368.0, 'G': 346.0, 'T': 4511.0},\n",
       "  4: {'A': 7878.0, 'C': 507.0, 'G': 886.0, 'T': 630.0},\n",
       "  5: {'A': 7878.0, 'C': 318.0, 'G': 142.0, 'T': 232.0},\n",
       "  6: {'A': 77.0, 'C': 73.0, 'G': 9.0, 'T': 7878.0},\n",
       "  7: {'A': 698.0, 'C': 730.0, 'G': 240.0, 'T': 7878.0},\n",
       "  8: {'A': 7878.0, 'C': 450.0, 'G': 88.0, 'T': 978.0},\n",
       "  9: {'A': 2598.0, 'C': 1672.0, 'G': 2206.0, 'T': 1404.0},\n",
       "  10: {'A': 2760.0, 'C': 2447.0, 'G': 1504.0, 'T': 1169.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'O95076',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0892.1',\n",
       "  'Matrix_Name': 'GSX1',\n",
       "  1: {'A': 332.0, 'C': 392.0, 'G': 210.0, 'T': 253.0},\n",
       "  2: {'A': 165.0, 'C': 536.0, 'G': 259.0, 'T': 227.0},\n",
       "  3: {'A': 125.0, 'C': 451.0, 'G': 49.0, 'T': 734.0},\n",
       "  4: {'A': 782.0, 'C': 403.0, 'G': 61.0, 'T': 178.0},\n",
       "  5: {'A': 1184.0, 'C': 107.0, 'G': 124.0, 'T': 51.0},\n",
       "  6: {'A': 91.0, 'C': 80.0, 'G': 4.0, 'T': 1184.0},\n",
       "  7: {'A': 107.0, 'C': 107.0, 'G': 1.0, 'T': 1184.0},\n",
       "  8: {'A': 1184.0, 'C': 1.0, 'G': 10.0, 'T': 104.0},\n",
       "  9: {'A': 449.0, 'C': 250.0, 'G': 314.0, 'T': 174.0},\n",
       "  10: {'A': 351.0, 'C': 306.0, 'G': 215.0, 'T': 316.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'A4IFQ3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0258.2',\n",
       "  'Matrix_Name': 'ESR2',\n",
       "  1: {'A': 5411.0, 'C': 97.0, 'G': 2564.0, 'T': 175.0},\n",
       "  2: {'A': 430.0, 'C': 1.0, 'G': 7171.0, 'T': 645.0},\n",
       "  3: {'A': 60.0, 'C': 1.0, 'G': 8144.0, 'T': 42.0},\n",
       "  4: {'A': 75.0, 'C': 148.0, 'G': 990.0, 'T': 7034.0},\n",
       "  5: {'A': 1.0, 'C': 7622.0, 'G': 534.0, 'T': 90.0},\n",
       "  6: {'A': 8099.0, 'C': 2.0, 'G': 58.0, 'T': 88.0},\n",
       "  7: {'A': 503.0, 'C': 4203.0, 'G': 2739.0, 'T': 802.0},\n",
       "  8: {'A': 2093.0, 'C': 3236.0, 'G': 1689.0, 'T': 1229.0},\n",
       "  9: {'A': 1809.0, 'C': 2511.0, 'G': 2174.0, 'T': 1753.0},\n",
       "  10: {'A': 1603.0, 'C': 976.0, 'G': 534.0, 'T': 5134.0},\n",
       "  11: {'A': 1122.0, 'C': 399.0, 'G': 6659.0, 'T': 67.0},\n",
       "  12: {'A': 3279.0, 'C': 2057.0, 'G': 1392.0, 'T': 1519.0},\n",
       "  13: {'A': 435.0, 'C': 6470.0, 'G': 364.0, 'T': 978.0},\n",
       "  14: {'A': 1112.0, 'C': 5855.0, 'G': 1.0, 'T': 1279.0},\n",
       "  15: {'A': 835.0, 'C': 2709.0, 'G': 607.0, 'T': 4096.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '18272478',\n",
       "  'uniprot_ids': 'Q92731',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0524.2',\n",
       "  'Matrix_Name': 'TFAP2C',\n",
       "  1: {'A': 1290.0, 'C': 1428.0, 'G': 309.0, 'T': 2811.0},\n",
       "  2: {'A': 879.0, 'C': 1694.0, 'G': 5835.0, 'T': 140.0},\n",
       "  3: {'A': 71.0, 'C': 5835.0, 'G': 622.0, 'T': 52.0},\n",
       "  4: {'A': 25.0, 'C': 5835.0, 'G': 10.0, 'T': 122.0},\n",
       "  5: {'A': 38.0, 'C': 3595.0, 'G': 385.0, 'T': 1821.0},\n",
       "  6: {'A': 517.0, 'C': 2125.0, 'G': 1328.0, 'T': 1868.0},\n",
       "  7: {'A': 2406.0, 'C': 1451.0, 'G': 1703.0, 'T': 278.0},\n",
       "  8: {'A': 2217.0, 'C': 342.0, 'G': 3262.0, 'T': 18.0},\n",
       "  9: {'A': 164.0, 'C': 13.0, 'G': 5835.0, 'T': 17.0},\n",
       "  10: {'A': 112.0, 'C': 743.0, 'G': 5835.0, 'T': 38.0},\n",
       "  11: {'A': 255.0, 'C': 5835.0, 'G': 1173.0, 'T': 600.0},\n",
       "  12: {'A': 3486.0, 'C': 390.0, 'G': 1036.0, 'T': 926.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'Q92754',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0068.2',\n",
       "  'Matrix_Name': 'PAX4',\n",
       "  1: {'A': 38.0, 'C': 334.0, 'G': 48.0, 'T': 106.0},\n",
       "  2: {'A': 59.0, 'C': 31.0, 'G': 121.0, 'T': 334.0},\n",
       "  3: {'A': 334.0, 'C': 3.0, 'G': 1.0, 'T': 4.0},\n",
       "  4: {'A': 334.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 6.0, 'T': 334.0},\n",
       "  6: {'A': 19.0, 'C': 12.0, 'G': 11.0, 'T': 334.0},\n",
       "  7: {'A': 334.0, 'C': 90.0, 'G': 12.0, 'T': 13.0},\n",
       "  8: {'A': 151.0, 'C': 94.0, 'G': 334.0, 'T': 42.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '10567552',\n",
       "  'uniprot_ids': 'O43316',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1527.1',\n",
       "  'Matrix_Name': 'NFIC(var.2)',\n",
       "  1: {'A': 399.0, 'C': 561.0, 'G': 762.0, 'T': 400.0},\n",
       "  2: {'A': 168.0, 'C': 617.0, 'G': 66.0, 'T': 2119.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 6.0, 'T': 2119.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 2119.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 4.0, 'G': 2119.0, 'T': 1.0},\n",
       "  6: {'A': 101.0, 'C': 2119.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 697.0, 'C': 269.0, 'G': 695.0, 'T': 462.0},\n",
       "  8: {'A': 390.0, 'C': 706.0, 'G': 493.0, 'T': 532.0},\n",
       "  9: {'A': 444.0, 'C': 581.0, 'G': 679.0, 'T': 418.0},\n",
       "  10: {'A': 477.0, 'C': 383.0, 'G': 882.0, 'T': 380.0},\n",
       "  11: {'A': 204.0, 'C': 341.0, 'G': 98.0, 'T': 2119.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 2119.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 2119.0, 'G': 14.0, 'T': 1.0},\n",
       "  14: {'A': 4.0, 'C': 2119.0, 'G': 1.0, 'T': 1.0},\n",
       "  15: {'A': 2119.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  16: {'A': 1087.0, 'C': 20.0, 'G': 1033.0, 'T': 155.0},\n",
       "  17: {'A': 365.0, 'C': 696.0, 'G': 648.0, 'T': 413.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '22456058',\n",
       "  'uniprot_ids': 'P08651',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0671.1',\n",
       "  'Matrix_Name': 'NFIX',\n",
       "  1: {'A': 7952.0, 'C': 8811.0, 'G': 8726.0, 'T': 7379.0},\n",
       "  2: {'A': 7833.0, 'C': 6135.0, 'G': 10856.0, 'T': 8044.0},\n",
       "  3: {'A': 11280.0, 'C': 9696.0, 'G': 5912.0, 'T': 32865.0},\n",
       "  4: {'A': 133.0, 'C': 187.0, 'G': 32865.0, 'T': 2540.0},\n",
       "  5: {'A': 4.0, 'C': 32865.0, 'G': 3258.0, 'T': 30.0},\n",
       "  6: {'A': 1.0, 'C': 32865.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 32865.0, 'C': 2231.0, 'G': 181.0, 'T': 111.0},\n",
       "  8: {'A': 32865.0, 'C': 1357.0, 'G': 12689.0, 'T': 2400.0},\n",
       "  9: {'A': 8252.0, 'C': 9040.0, 'G': 9136.0, 'T': 6440.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '20178747',\n",
       "  'uniprot_ids': 'Q14938',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1584.1',\n",
       "  'Matrix_Name': 'ZIC5',\n",
       "  1: {'A': 6069.0, 'C': 6878.0, 'G': 5213.0, 'T': 3107.0},\n",
       "  2: {'A': 556.0, 'C': 3319.0, 'G': 17330.0, 'T': 5688.0},\n",
       "  3: {'A': 19267.0, 'C': 2321.0, 'G': 2766.0, 'T': 1.0},\n",
       "  4: {'A': 251.0, 'C': 21202.0, 'G': 1.0, 'T': 18.0},\n",
       "  5: {'A': 148.0, 'C': 21164.0, 'G': 59.0, 'T': 58.0},\n",
       "  6: {'A': 1312.0, 'C': 20222.0, 'G': 471.0, 'T': 329.0},\n",
       "  7: {'A': 6.0, 'C': 21264.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 68.0, 'C': 21236.0, 'G': 3.0, 'T': 3.0},\n",
       "  9: {'A': 11.0, 'C': 17686.0, 'G': 1.0, 'T': 3578.0},\n",
       "  10: {'A': 2111.0, 'C': 22.0, 'G': 20413.0, 'T': 24.0},\n",
       "  11: {'A': 3.0, 'C': 19608.0, 'G': 236.0, 'T': 5844.0},\n",
       "  12: {'A': 2537.0, 'C': 3845.0, 'G': 3253.0, 'T': 18008.0},\n",
       "  13: {'A': 252.0, 'C': 9.0, 'G': 21154.0, 'T': 3.0},\n",
       "  14: {'A': 5256.0, 'C': 9446.0, 'G': 1726.0, 'T': 10268.0},\n",
       "  15: {'A': 53.0, 'C': 1734.0, 'G': 19962.0, 'T': 2757.0},\n",
       "  16: {'A': 8074.0, 'C': 11607.0, 'G': 793.0, 'T': 5357.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'Q96T25',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0164.1',\n",
       "  'Matrix_Name': 'Nr2e3',\n",
       "  1: {'A': 5.0, 'C': 13.0, 'G': 3.0, 'T': 6.0},\n",
       "  2: {'A': 24.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 24.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 2.0, 'G': 23.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 24.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 24.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 24.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15634773',\n",
       "  'uniprot_ids': 'Q9QXZ7',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1126.1',\n",
       "  'Matrix_Name': 'FOS::JUN(var.2)',\n",
       "  1: {'A': 267.0, 'C': 64.0, 'G': 420.0, 'T': 253.0},\n",
       "  2: {'A': 557.0, 'C': 46.0, 'G': 370.0, 'T': 31.0},\n",
       "  3: {'A': 48.0, 'C': 75.0, 'G': 63.0, 'T': 817.0},\n",
       "  4: {'A': 43.0, 'C': 113.0, 'G': 656.0, 'T': 192.0},\n",
       "  5: {'A': 749.0, 'C': 32.0, 'G': 136.0, 'T': 87.0},\n",
       "  6: {'A': 81.0, 'C': 839.0, 'G': 42.0, 'T': 42.0},\n",
       "  7: {'A': 132.0, 'C': 115.0, 'G': 723.0, 'T': 35.0},\n",
       "  8: {'A': 41.0, 'C': 24.0, 'G': 33.0, 'T': 906.0},\n",
       "  9: {'A': 54.0, 'C': 847.0, 'G': 45.0, 'T': 58.0},\n",
       "  10: {'A': 908.0, 'C': 25.0, 'G': 37.0, 'T': 34.0},\n",
       "  11: {'A': 31.0, 'C': 235.0, 'G': 61.0, 'T': 677.0},\n",
       "  12: {'A': 300.0, 'C': 323.0, 'G': 161.0, 'T': 221.0},\n",
       "  13: {'A': 323.0, 'C': 314.0, 'G': 162.0, 'T': 205.0},\n",
       "  14: {'A': 191.0, 'C': 179.0, 'G': 262.0, 'T': 372.0},\n",
       "  15: {'A': 162.0, 'C': 76.0, 'G': 239.0, 'T': 527.0},\n",
       "  16: {'A': 204.0, 'C': 292.0, 'G': 208.0, 'T': 299.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P01100;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1523.1',\n",
       "  'Matrix_Name': 'MSANTD3',\n",
       "  1: {'A': 99.0, 'C': 270.0, 'G': 516.0, 'T': 118.0},\n",
       "  2: {'A': 112.0, 'C': 212.0, 'G': 221.0, 'T': 457.0},\n",
       "  3: {'A': 333.0, 'C': 273.0, 'G': 250.0, 'T': 146.0},\n",
       "  4: {'A': 65.0, 'C': 724.0, 'G': 32.0, 'T': 181.0},\n",
       "  5: {'A': 981.0, 'C': 1.0, 'G': 3.0, 'T': 17.0},\n",
       "  6: {'A': 32.0, 'C': 897.0, 'G': 28.0, 'T': 46.0},\n",
       "  7: {'A': 38.0, 'C': 8.0, 'G': 1.0, 'T': 955.0},\n",
       "  8: {'A': 48.0, 'C': 863.0, 'G': 30.0, 'T': 61.0},\n",
       "  9: {'A': 776.0, 'C': 141.0, 'G': 25.0, 'T': 60.0},\n",
       "  10: {'A': 272.0, 'C': 569.0, 'G': 64.0, 'T': 97.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Myb/SANT',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '12354778',\n",
       "  'uniprot_ids': 'Q96H12',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0112.3',\n",
       "  'Matrix_Name': 'ESR1',\n",
       "  1: {'A': 233.0, 'C': 66.0, 'G': 102.0, 'T': 89.0},\n",
       "  2: {'A': 403.0, 'C': 7.0, 'G': 207.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 624.0, 'T': 35.0},\n",
       "  4: {'A': 3.0, 'C': 1.0, 'G': 639.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 4.0, 'T': 550.0},\n",
       "  6: {'A': 2.0, 'C': 453.0, 'G': 1.0, 'T': 3.0},\n",
       "  7: {'A': 472.0, 'C': 1.0, 'G': 31.0, 'T': 1.0},\n",
       "  8: {'A': 109.0, 'C': 339.0, 'G': 54.0, 'T': 24.0},\n",
       "  9: {'A': 128.0, 'C': 224.0, 'G': 341.0, 'T': 66.0},\n",
       "  10: {'A': 236.0, 'C': 89.0, 'G': 285.0, 'T': 14.0},\n",
       "  11: {'A': 1.0, 'C': 24.0, 'G': 2.0, 'T': 511.0},\n",
       "  12: {'A': 3.0, 'C': 1.0, 'G': 633.0, 'T': 1.0},\n",
       "  13: {'A': 492.0, 'C': 2.0, 'G': 2.0, 'T': 1.0},\n",
       "  14: {'A': 1.0, 'C': 461.0, 'G': 3.0, 'T': 1.0},\n",
       "  15: {'A': 16.0, 'C': 456.0, 'G': 1.0, 'T': 1.0},\n",
       "  16: {'A': 1.0, 'C': 144.0, 'G': 8.0, 'T': 503.0},\n",
       "  17: {'A': 36.0, 'C': 69.0, 'G': 210.0, 'T': 205.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '19339991',\n",
       "  'uniprot_ids': 'P03372',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1120.1',\n",
       "  'Matrix_Name': 'SOX13',\n",
       "  1: {'A': 1540.0, 'C': 664.0, 'G': 1396.0, 'T': 956.0},\n",
       "  2: {'A': 1796.0, 'C': 959.0, 'G': 1050.0, 'T': 751.0},\n",
       "  3: {'A': 4452.0, 'C': 23.0, 'G': 41.0, 'T': 40.0},\n",
       "  4: {'A': 56.0, 'C': 4277.0, 'G': 61.0, 'T': 162.0},\n",
       "  5: {'A': 4330.0, 'C': 56.0, 'G': 51.0, 'T': 119.0},\n",
       "  6: {'A': 4391.0, 'C': 49.0, 'G': 64.0, 'T': 52.0},\n",
       "  7: {'A': 157.0, 'C': 78.0, 'G': 29.0, 'T': 4292.0},\n",
       "  8: {'A': 586.0, 'C': 58.0, 'G': 3838.0, 'T': 74.0},\n",
       "  9: {'A': 779.0, 'C': 648.0, 'G': 2667.0, 'T': 462.0},\n",
       "  10: {'A': 1229.0, 'C': 1258.0, 'G': 1075.0, 'T': 994.0},\n",
       "  11: {'A': 1229.0, 'C': 1127.0, 'G': 1047.0, 'T': 1153.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '27521520',\n",
       "  'uniprot_ids': 'Q9UN79',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1112.2',\n",
       "  'Matrix_Name': 'NR4A1',\n",
       "  1: {'A': 3485.0, 'C': 3284.0, 'G': 4160.0, 'T': 5002.0},\n",
       "  2: {'A': 4464.0, 'C': 3564.0, 'G': 3381.0, 'T': 4522.0},\n",
       "  3: {'A': 14314.0, 'C': 237.0, 'G': 824.0, 'T': 556.0},\n",
       "  4: {'A': 15269.0, 'C': 118.0, 'G': 334.0, 'T': 210.0},\n",
       "  5: {'A': 15685.0, 'C': 63.0, 'G': 129.0, 'T': 54.0},\n",
       "  6: {'A': 428.0, 'C': 93.0, 'G': 12435.0, 'T': 2975.0},\n",
       "  7: {'A': 301.0, 'C': 167.0, 'G': 14965.0, 'T': 498.0},\n",
       "  8: {'A': 366.0, 'C': 825.0, 'G': 1359.0, 'T': 13381.0},\n",
       "  9: {'A': 71.0, 'C': 14950.0, 'G': 152.0, 'T': 758.0},\n",
       "  10: {'A': 15127.0, 'C': 152.0, 'G': 330.0, 'T': 322.0},\n",
       "  11: {'A': 4457.0, 'C': 3988.0, 'G': 4076.0, 'T': 3410.0},\n",
       "  12: {'A': 5240.0, 'C': 3371.0, 'G': 3888.0, 'T': 3432.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'NGFI-B-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '12082103',\n",
       "  'uniprot_ids': 'P22736',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0781.1',\n",
       "  'Matrix_Name': 'PAX9',\n",
       "  1: {'A': 155.0, 'C': 2411.0, 'G': 1177.0, 'T': 245.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 3812.0, 'T': 11.0},\n",
       "  3: {'A': 68.0, 'C': 14.0, 'G': 7.0, 'T': 3108.0},\n",
       "  4: {'A': 1.0, 'C': 3776.0, 'G': 1.0, 'T': 172.0},\n",
       "  5: {'A': 3242.0, 'C': 72.0, 'G': 3.0, 'T': 2.0},\n",
       "  6: {'A': 1.0, 'C': 3861.0, 'G': 1.0, 'T': 268.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 3946.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 3696.0, 'G': 728.0, 'T': 1.0},\n",
       "  9: {'A': 1841.0, 'C': 84.0, 'G': 14.0, 'T': 1142.0},\n",
       "  10: {'A': 2.0, 'C': 69.0, 'G': 1.0, 'T': 3189.0},\n",
       "  11: {'A': 4.0, 'C': 1413.0, 'G': 3553.0, 'T': 4.0},\n",
       "  12: {'A': 3084.0, 'C': 1.0, 'G': 469.0, 'T': 1.0},\n",
       "  13: {'A': 623.0, 'C': 1424.0, 'G': 963.0, 'T': 445.0},\n",
       "  14: {'A': 7.0, 'C': 208.0, 'G': 1.0, 'T': 3133.0},\n",
       "  15: {'A': 143.0, 'C': 9.0, 'G': 3609.0, 'T': 252.0},\n",
       "  16: {'A': 1250.0, 'C': 2391.0, 'G': 472.0, 'T': 1.0},\n",
       "  17: {'A': 2185.0, 'C': 1335.0, 'G': 645.0, 'T': 853.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '19132093',\n",
       "  'uniprot_ids': 'P55771',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1130.1',\n",
       "  'Matrix_Name': 'FOSL2::JUN',\n",
       "  1: {'A': 190.0, 'C': 249.0, 'G': 296.0, 'T': 269.0},\n",
       "  2: {'A': 215.0, 'C': 180.0, 'G': 339.0, 'T': 270.0},\n",
       "  3: {'A': 543.0, 'C': 122.0, 'G': 283.0, 'T': 56.0},\n",
       "  4: {'A': 2.0, 'C': 2.0, 'G': 2.0, 'T': 999.0},\n",
       "  5: {'A': 2.0, 'C': 2.0, 'G': 841.0, 'T': 159.0},\n",
       "  6: {'A': 848.0, 'C': 124.0, 'G': 2.0, 'T': 30.0},\n",
       "  7: {'A': 104.0, 'C': 142.0, 'G': 699.0, 'T': 59.0},\n",
       "  8: {'A': 2.0, 'C': 2.0, 'G': 2.0, 'T': 999.0},\n",
       "  9: {'A': 21.0, 'C': 980.0, 'G': 2.0, 'T': 2.0},\n",
       "  10: {'A': 999.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  11: {'A': 20.0, 'C': 300.0, 'G': 94.0, 'T': 590.0},\n",
       "  12: {'A': 270.0, 'C': 342.0, 'G': 181.0, 'T': 210.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15408;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1565.1',\n",
       "  'Matrix_Name': 'TBX18',\n",
       "  1: {'A': 217.0, 'C': 87.0, 'G': 317.0, 'T': 161.0},\n",
       "  2: {'A': 344.0, 'C': 104.0, 'G': 436.0, 'T': 149.0},\n",
       "  3: {'A': 779.0, 'C': 63.0, 'G': 160.0, 'T': 60.0},\n",
       "  4: {'A': 74.0, 'C': 58.0, 'G': 779.0, 'T': 48.0},\n",
       "  5: {'A': 15.0, 'C': 1.0, 'G': 779.0, 'T': 1.0},\n",
       "  6: {'A': 8.0, 'C': 81.0, 'G': 1.0, 'T': 779.0},\n",
       "  7: {'A': 22.0, 'C': 7.0, 'G': 779.0, 'T': 3.0},\n",
       "  8: {'A': 101.0, 'C': 99.0, 'G': 110.0, 'T': 779.0},\n",
       "  9: {'A': 53.0, 'C': 162.0, 'G': 779.0, 'T': 33.0},\n",
       "  10: {'A': 779.0, 'C': 2.0, 'G': 6.0, 'T': 35.0},\n",
       "  11: {'A': 779.0, 'C': 63.0, 'G': 266.0, 'T': 96.0},\n",
       "  12: {'A': 306.0, 'C': 138.0, 'G': 247.0, 'T': 91.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TBX1-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '17584735',\n",
       "  'uniprot_ids': 'O95935',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0075.3',\n",
       "  'Matrix_Name': 'PRRX2',\n",
       "  1: {'A': 779.0, 'C': 6721.0, 'G': 1614.0, 'T': 1934.0},\n",
       "  2: {'A': 123.0, 'C': 2009.0, 'G': 132.0, 'T': 4712.0},\n",
       "  3: {'A': 6721.0, 'C': 30.0, 'G': 4.0, 'T': 1.0},\n",
       "  4: {'A': 6721.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6721.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6721.0},\n",
       "  7: {'A': 6721.0, 'C': 2.0, 'G': 400.0, 'T': 1.0},\n",
       "  8: {'A': 2292.0, 'C': 1272.0, 'G': 1863.0, 'T': 1296.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7901837',\n",
       "  'uniprot_ids': 'Q99811',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0704.1',\n",
       "  'Matrix_Name': 'Lhx4',\n",
       "  1: {'A': 423.0, 'C': 818.0, 'G': 496.0, 'T': 1162.0},\n",
       "  2: {'A': 218.0, 'C': 913.0, 'G': 1.0, 'T': 2897.0},\n",
       "  3: {'A': 2897.0, 'C': 429.0, 'G': 310.0, 'T': 256.0},\n",
       "  4: {'A': 2897.0, 'C': 1.0, 'G': 1.0, 'T': 156.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2897.0},\n",
       "  6: {'A': 1.0, 'C': 447.0, 'G': 58.0, 'T': 2897.0},\n",
       "  7: {'A': 2897.0, 'C': 40.0, 'G': 317.0, 'T': 136.0},\n",
       "  8: {'A': 1315.0, 'C': 450.0, 'G': 858.0, 'T': 276.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P53776',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0743.2',\n",
       "  'Matrix_Name': 'SCRT1',\n",
       "  1: {'A': 16321.0, 'C': 14119.0, 'G': 14308.0, 'T': 15465.0},\n",
       "  2: {'A': 22933.0, 'C': 10315.0, 'G': 14515.0, 'T': 12450.0},\n",
       "  3: {'A': 17037.0, 'C': 8694.0, 'G': 10155.0, 'T': 24327.0},\n",
       "  4: {'A': 4414.0, 'C': 7426.0, 'G': 16397.0, 'T': 31976.0},\n",
       "  5: {'A': 1457.0, 'C': 54836.0, 'G': 1298.0, 'T': 2622.0},\n",
       "  6: {'A': 45607.0, 'C': 8752.0, 'G': 1836.0, 'T': 4018.0},\n",
       "  7: {'A': 57272.0, 'C': 371.0, 'G': 1567.0, 'T': 1003.0},\n",
       "  8: {'A': 214.0, 'C': 59558.0, 'G': 225.0, 'T': 216.0},\n",
       "  9: {'A': 58759.0, 'C': 235.0, 'G': 160.0, 'T': 1059.0},\n",
       "  10: {'A': 7018.0, 'C': 558.0, 'G': 51331.0, 'T': 1306.0},\n",
       "  11: {'A': 1665.0, 'C': 467.0, 'G': 56501.0, 'T': 1580.0},\n",
       "  12: {'A': 884.0, 'C': 1099.0, 'G': 260.0, 'T': 57970.0},\n",
       "  13: {'A': 7814.0, 'C': 3297.0, 'G': 41191.0, 'T': 7911.0},\n",
       "  14: {'A': 10723.0, 'C': 8789.0, 'G': 26974.0, 'T': 13727.0},\n",
       "  15: {'A': 13383.0, 'C': 14904.0, 'G': 13471.0, 'T': 18455.0},\n",
       "  16: {'A': 16485.0, 'C': 12838.0, 'G': 11050.0, 'T': 19840.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q9BWW7',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0136.2',\n",
       "  'Matrix_Name': 'ELF5',\n",
       "  1: {'A': 2417.0, 'C': 253.0, 'G': 302.0, 'T': 794.0},\n",
       "  2: {'A': 1222.0, 'C': 2299.0, 'G': 956.0, 'T': 867.0},\n",
       "  3: {'A': 988.0, 'C': 2454.0, 'G': 1267.0, 'T': 98.0},\n",
       "  4: {'A': 1459.0, 'C': 2529.0, 'G': 310.0, 'T': 57.0},\n",
       "  5: {'A': 4.0, 'C': 1.0, 'G': 3308.0, 'T': 21.0},\n",
       "  6: {'A': 6.0, 'C': 1.0, 'G': 3316.0, 'T': 26.0},\n",
       "  7: {'A': 2700.0, 'C': 12.0, 'G': 2.0, 'T': 5.0},\n",
       "  8: {'A': 2647.0, 'C': 62.0, 'G': 1.0, 'T': 246.0},\n",
       "  9: {'A': 891.0, 'C': 156.0, 'G': 2858.0, 'T': 85.0},\n",
       "  10: {'A': 478.0, 'C': 717.0, 'G': 313.0, 'T': 2471.0},\n",
       "  11: {'A': 960.0, 'C': 372.0, 'G': 763.0, 'T': 664.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'Q9UKW6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1649.1',\n",
       "  'Matrix_Name': 'ZBTB12',\n",
       "  1: {'A': 1464.0, 'C': 1019.0, 'G': 1399.0, 'T': 889.0},\n",
       "  2: {'A': 753.0, 'C': 1342.0, 'G': 785.0, 'T': 1891.0},\n",
       "  3: {'A': 145.0, 'C': 4378.0, 'G': 176.0, 'T': 72.0},\n",
       "  4: {'A': 131.0, 'C': 217.0, 'G': 191.0, 'T': 4232.0},\n",
       "  5: {'A': 1842.0, 'C': 74.0, 'G': 2798.0, 'T': 57.0},\n",
       "  6: {'A': 87.0, 'C': 81.0, 'G': 4479.0, 'T': 124.0},\n",
       "  7: {'A': 4476.0, 'C': 115.0, 'G': 107.0, 'T': 73.0},\n",
       "  8: {'A': 4509.0, 'C': 66.0, 'G': 104.0, 'T': 92.0},\n",
       "  9: {'A': 52.0, 'C': 4512.0, 'G': 51.0, 'T': 156.0},\n",
       "  10: {'A': 1327.0, 'C': 1791.0, 'G': 648.0, 'T': 1005.0},\n",
       "  11: {'A': 964.0, 'C': 1482.0, 'G': 993.0, 'T': 1332.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q9Y330',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0706.1',\n",
       "  'Matrix_Name': 'MEOX2',\n",
       "  1: {'A': 8100.0, 'C': 3822.0, 'G': 6834.0, 'T': 5177.0},\n",
       "  2: {'A': 3263.0, 'C': 8943.0, 'G': 14988.0, 'T': 2755.0},\n",
       "  3: {'A': 1470.0, 'C': 5056.0, 'G': 643.0, 'T': 23931.0},\n",
       "  4: {'A': 23931.0, 'C': 8989.0, 'G': 3017.0, 'T': 901.0},\n",
       "  5: {'A': 23931.0, 'C': 789.0, 'G': 273.0, 'T': 493.0},\n",
       "  6: {'A': 77.0, 'C': 39.0, 'G': 170.0, 'T': 23931.0},\n",
       "  7: {'A': 1153.0, 'C': 6040.0, 'G': 4975.0, 'T': 23931.0},\n",
       "  8: {'A': 23931.0, 'C': 360.0, 'G': 4100.0, 'T': 1033.0},\n",
       "  9: {'A': 9635.0, 'C': 3381.0, 'G': 4087.0, 'T': 6831.0},\n",
       "  10: {'A': 5919.0, 'C': 7196.0, 'G': 5709.0, 'T': 5109.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q6FHY5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0070.1',\n",
       "  'Matrix_Name': 'PBX1',\n",
       "  1: {'A': 6.0, 'C': 7.0, 'G': 3.0, 'T': 6.0},\n",
       "  2: {'A': 4.0, 'C': 10.0, 'G': 4.0, 'T': 4.0},\n",
       "  3: {'A': 17.0, 'C': 2.0, 'G': 2.0, 'T': 1.0},\n",
       "  4: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 17.0},\n",
       "  5: {'A': 1.0, 'C': 19.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 18.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 18.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 18.0},\n",
       "  9: {'A': 1.0, 'C': 19.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 17.0, 'C': 2.0, 'G': 1.0, 'T': 2.0},\n",
       "  11: {'A': 13.0, 'C': 1.0, 'G': 2.0, 'T': 6.0},\n",
       "  12: {'A': 9.0, 'C': 3.0, 'G': 3.0, 'T': 7.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7910944',\n",
       "  'uniprot_ids': 'P40424',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1111.1',\n",
       "  'Matrix_Name': 'NR2F2',\n",
       "  1: {'A': 489.0, 'C': 693.0, 'G': 419.0, 'T': 607.0},\n",
       "  2: {'A': 1028.0, 'C': 457.0, 'G': 321.0, 'T': 402.0},\n",
       "  3: {'A': 1789.0, 'C': 38.0, 'G': 174.0, 'T': 207.0},\n",
       "  4: {'A': 2085.0, 'C': 16.0, 'G': 83.0, 'T': 24.0},\n",
       "  5: {'A': 38.0, 'C': 17.0, 'G': 2109.0, 'T': 44.0},\n",
       "  6: {'A': 34.0, 'C': 13.0, 'G': 2133.0, 'T': 28.0},\n",
       "  7: {'A': 18.0, 'C': 23.0, 'G': 31.0, 'T': 2136.0},\n",
       "  8: {'A': 31.0, 'C': 1979.0, 'G': 27.0, 'T': 171.0},\n",
       "  9: {'A': 2112.0, 'C': 17.0, 'G': 44.0, 'T': 35.0},\n",
       "  10: {'A': 634.0, 'C': 504.0, 'G': 545.0, 'T': 525.0},\n",
       "  11: {'A': 848.0, 'C': 385.0, 'G': 573.0, 'T': 402.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '1324415',\n",
       "  'uniprot_ids': 'P24468',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0017.2',\n",
       "  'Matrix_Name': 'NR2F1',\n",
       "  1: {'A': 4803.0, 'C': 11447.0, 'G': 3277.0, 'T': 3981.0},\n",
       "  2: {'A': 7267.0, 'C': 1693.0, 'G': 1154.0, 'T': 1949.0},\n",
       "  3: {'A': 6334.0, 'C': 388.0, 'G': 4870.0, 'T': 470.0},\n",
       "  4: {'A': 8497.0, 'C': 1.0, 'G': 3792.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 31.0, 'G': 12060.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 12060.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 12060.0},\n",
       "  8: {'A': 1.0, 'C': 12060.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 12060.0, 'C': 1.0, 'G': 92.0, 'T': 1.0},\n",
       "  10: {'A': 5117.0, 'C': 3056.0, 'G': 1619.0, 'T': 2271.0},\n",
       "  11: {'A': 3230.0, 'C': 2967.0, 'G': 4396.0, 'T': 1470.0},\n",
       "  12: {'A': 3277.0, 'C': 2471.0, 'G': 3887.0, 'T': 2428.0},\n",
       "  13: {'A': 5682.0, 'C': 2913.0, 'G': 36864.0, 'T': 3467.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '8496174',\n",
       "  'uniprot_ids': 'P10589',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1539.1',\n",
       "  'Matrix_Name': 'NR2F6(var.3)',\n",
       "  1: {'A': 6350.0, 'C': 728.0, 'G': 3737.0, 'T': 938.0},\n",
       "  2: {'A': 8960.0, 'C': 27.0, 'G': 2790.0, 'T': 21.0},\n",
       "  3: {'A': 32.0, 'C': 2.0, 'G': 11750.0, 'T': 125.0},\n",
       "  4: {'A': 39.0, 'C': 13.0, 'G': 11750.0, 'T': 146.0},\n",
       "  5: {'A': 27.0, 'C': 79.0, 'G': 254.0, 'T': 11750.0},\n",
       "  6: {'A': 5.0, 'C': 11750.0, 'G': 217.0, 'T': 444.0},\n",
       "  7: {'A': 9115.0, 'C': 40.0, 'G': 2635.0, 'T': 47.0},\n",
       "  8: {'A': 2493.0, 'C': 3334.0, 'G': 2973.0, 'T': 2953.0},\n",
       "  9: {'A': 67.0, 'C': 334.0, 'G': 68.0, 'T': 11750.0},\n",
       "  10: {'A': 388.0, 'C': 207.0, 'G': 11750.0, 'T': 10.0},\n",
       "  11: {'A': 11750.0, 'C': 256.0, 'G': 108.0, 'T': 4.0},\n",
       "  12: {'A': 160.0, 'C': 11750.0, 'G': 1.0, 'T': 37.0},\n",
       "  13: {'A': 83.0, 'C': 11750.0, 'G': 10.0, 'T': 25.0},\n",
       "  14: {'A': 24.0, 'C': 2845.0, 'G': 27.0, 'T': 8906.0},\n",
       "  15: {'A': 911.0, 'C': 3746.0, 'G': 719.0, 'T': 6377.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'P10588',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1550.1',\n",
       "  'Matrix_Name': 'PPARD',\n",
       "  1: {'A': 4798.0, 'C': 2890.0, 'G': 2379.0, 'T': 3117.0},\n",
       "  2: {'A': 4471.0, 'C': 1312.0, 'G': 7621.0, 'T': 978.0},\n",
       "  3: {'A': 6329.0, 'C': 76.0, 'G': 6809.0, 'T': 14.0},\n",
       "  4: {'A': 102.0, 'C': 21.0, 'G': 13068.0, 'T': 88.0},\n",
       "  5: {'A': 127.0, 'C': 11.0, 'G': 12075.0, 'T': 2113.0},\n",
       "  6: {'A': 22.0, 'C': 29.0, 'G': 416.0, 'T': 12958.0},\n",
       "  7: {'A': 47.0, 'C': 11670.0, 'G': 932.0, 'T': 1895.0},\n",
       "  8: {'A': 13106.0, 'C': 1.0, 'G': 152.0, 'T': 1.0},\n",
       "  9: {'A': 12528.0, 'C': 35.0, 'G': 867.0, 'T': 401.0},\n",
       "  10: {'A': 12694.0, 'C': 8.0, 'G': 1082.0, 'T': 1.0},\n",
       "  11: {'A': 36.0, 'C': 13.0, 'G': 13136.0, 'T': 46.0},\n",
       "  12: {'A': 30.0, 'C': 2.0, 'G': 12250.0, 'T': 1833.0},\n",
       "  13: {'A': 1.0, 'C': 40.0, 'G': 407.0, 'T': 12922.0},\n",
       "  14: {'A': 116.0, 'C': 11921.0, 'G': 656.0, 'T': 1601.0},\n",
       "  15: {'A': 12303.0, 'C': 40.0, 'G': 1642.0, 'T': 52.0},\n",
       "  16: {'A': 3807.0, 'C': 3780.0, 'G': 2028.0, 'T': 3569.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '8725145',\n",
       "  'uniprot_ids': 'Q03181',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1629.1',\n",
       "  'Matrix_Name': 'Zic2',\n",
       "  1: {'A': 3146.0, 'C': 3508.0, 'G': 2762.0, 'T': 2507.0},\n",
       "  2: {'A': 2451.0, 'C': 1756.0, 'G': 3822.0, 'T': 3894.0},\n",
       "  3: {'A': 1162.0, 'C': 9934.0, 'G': 586.0, 'T': 241.0},\n",
       "  4: {'A': 6889.0, 'C': 1766.0, 'G': 830.0, 'T': 2438.0},\n",
       "  5: {'A': 136.0, 'C': 11477.0, 'G': 189.0, 'T': 121.0},\n",
       "  6: {'A': 11075.0, 'C': 392.0, 'G': 187.0, 'T': 269.0},\n",
       "  7: {'A': 141.0, 'C': 229.0, 'G': 11485.0, 'T': 68.0},\n",
       "  8: {'A': 81.0, 'C': 11273.0, 'G': 270.0, 'T': 299.0},\n",
       "  9: {'A': 10626.0, 'C': 119.0, 'G': 736.0, 'T': 442.0},\n",
       "  10: {'A': 193.0, 'C': 134.0, 'G': 11185.0, 'T': 411.0},\n",
       "  11: {'A': 97.0, 'C': 205.0, 'G': 11466.0, 'T': 155.0},\n",
       "  12: {'A': 3061.0, 'C': 1477.0, 'G': 4912.0, 'T': 2473.0},\n",
       "  13: {'A': 3639.0, 'C': 1662.0, 'G': 4980.0, 'T': 1642.0},\n",
       "  14: {'A': 2334.0, 'C': 2169.0, 'G': 5315.0, 'T': 2105.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11038359',\n",
       "  'uniprot_ids': 'Q62520',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0462.2',\n",
       "  'Matrix_Name': 'BATF::JUN',\n",
       "  1: {'A': 10766.0, 'C': 4763.0, 'G': 9318.0, 'T': 13311.0},\n",
       "  2: {'A': 18790.0, 'C': 7261.0, 'G': 6730.0, 'T': 5377.0},\n",
       "  3: {'A': 368.0, 'C': 276.0, 'G': 183.0, 'T': 37331.0},\n",
       "  4: {'A': 835.0, 'C': 401.0, 'G': 33136.0, 'T': 3786.0},\n",
       "  5: {'A': 37281.0, 'C': 205.0, 'G': 351.0, 'T': 321.0},\n",
       "  6: {'A': 1400.0, 'C': 31169.0, 'G': 4189.0, 'T': 1400.0},\n",
       "  7: {'A': 756.0, 'C': 242.0, 'G': 540.0, 'T': 36620.0},\n",
       "  8: {'A': 3956.0, 'C': 32771.0, 'G': 523.0, 'T': 908.0},\n",
       "  9: {'A': 36855.0, 'C': 319.0, 'G': 425.0, 'T': 559.0},\n",
       "  10: {'A': 6300.0, 'C': 6106.0, 'G': 7746.0, 'T': 18006.0},\n",
       "  11: {'A': 13408.0, 'C': 9669.0, 'G': 4924.0, 'T': 10157.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'B-ATF-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22992523',\n",
       "  'uniprot_ids': 'Q16520;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1148.1',\n",
       "  'Matrix_Name': 'PPARA::RXRA',\n",
       "  1: {'A': 549.0, 'C': 141.0, 'G': 109.0, 'T': 205.0},\n",
       "  2: {'A': 432.0, 'C': 74.0, 'G': 61.0, 'T': 438.0},\n",
       "  3: {'A': 248.0, 'C': 212.0, 'G': 333.0, 'T': 211.0},\n",
       "  4: {'A': 62.0, 'C': 95.0, 'G': 239.0, 'T': 609.0},\n",
       "  5: {'A': 493.0, 'C': 96.0, 'G': 371.0, 'T': 43.0},\n",
       "  6: {'A': 69.0, 'C': 20.0, 'G': 844.0, 'T': 72.0},\n",
       "  7: {'A': 81.0, 'C': 16.0, 'G': 857.0, 'T': 50.0},\n",
       "  8: {'A': 43.0, 'C': 158.0, 'G': 246.0, 'T': 557.0},\n",
       "  9: {'A': 132.0, 'C': 446.0, 'G': 202.0, 'T': 224.0},\n",
       "  10: {'A': 888.0, 'C': 17.0, 'G': 68.0, 'T': 31.0},\n",
       "  11: {'A': 670.0, 'C': 25.0, 'G': 209.0, 'T': 100.0},\n",
       "  12: {'A': 808.0, 'C': 26.0, 'G': 144.0, 'T': 26.0},\n",
       "  13: {'A': 112.0, 'C': 17.0, 'G': 863.0, 'T': 12.0},\n",
       "  14: {'A': 19.0, 'C': 15.0, 'G': 785.0, 'T': 185.0},\n",
       "  15: {'A': 18.0, 'C': 50.0, 'G': 116.0, 'T': 819.0},\n",
       "  16: {'A': 91.0, 'C': 599.0, 'G': 147.0, 'T': 167.0},\n",
       "  17: {'A': 690.0, 'C': 38.0, 'G': 231.0, 'T': 44.0},\n",
       "  18: {'A': 223.0, 'C': 238.0, 'G': 244.0, 'T': 299.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '8386511',\n",
       "  'uniprot_ids': 'Q07869;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1560.1',\n",
       "  'Matrix_Name': 'SOHLH2',\n",
       "  1: {'A': 231.0, 'C': 486.0, 'G': 185.0, 'T': 314.0},\n",
       "  2: {'A': 173.0, 'C': 265.0, 'G': 1213.0, 'T': 277.0},\n",
       "  3: {'A': 46.0, 'C': 1213.0, 'G': 30.0, 'T': 71.0},\n",
       "  4: {'A': 1213.0, 'C': 1.0, 'G': 42.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1213.0, 'G': 12.0, 'T': 8.0},\n",
       "  6: {'A': 15.0, 'C': 1.0, 'G': 1213.0, 'T': 16.0},\n",
       "  7: {'A': 37.0, 'C': 66.0, 'G': 1.0, 'T': 1213.0},\n",
       "  8: {'A': 42.0, 'C': 5.0, 'G': 1213.0, 'T': 5.0},\n",
       "  9: {'A': 273.0, 'C': 1213.0, 'G': 173.0, 'T': 154.0},\n",
       "  10: {'A': 382.0, 'C': 240.0, 'G': 370.0, 'T': 225.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '26869299',\n",
       "  'uniprot_ids': 'Q9NX45',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1114.1',\n",
       "  'Matrix_Name': 'PBX3',\n",
       "  1: {'A': 1596.0, 'C': 1712.0, 'G': 2193.0, 'T': 1524.0},\n",
       "  2: {'A': 1511.0, 'C': 1725.0, 'G': 1935.0, 'T': 1854.0},\n",
       "  3: {'A': 1321.0, 'C': 1850.0, 'G': 1929.0, 'T': 1925.0},\n",
       "  4: {'A': 597.0, 'C': 226.0, 'G': 504.0, 'T': 5698.0},\n",
       "  5: {'A': 229.0, 'C': 228.0, 'G': 6407.0, 'T': 161.0},\n",
       "  6: {'A': 6427.0, 'C': 126.0, 'G': 215.0, 'T': 257.0},\n",
       "  7: {'A': 336.0, 'C': 789.0, 'G': 4930.0, 'T': 970.0},\n",
       "  8: {'A': 248.0, 'C': 200.0, 'G': 286.0, 'T': 6291.0},\n",
       "  9: {'A': 147.0, 'C': 243.0, 'G': 6551.0, 'T': 84.0},\n",
       "  10: {'A': 6165.0, 'C': 109.0, 'G': 605.0, 'T': 146.0},\n",
       "  11: {'A': 151.0, 'C': 6525.0, 'G': 210.0, 'T': 139.0},\n",
       "  12: {'A': 6092.0, 'C': 130.0, 'G': 426.0, 'T': 377.0},\n",
       "  13: {'A': 370.0, 'C': 413.0, 'G': 5692.0, 'T': 550.0},\n",
       "  14: {'A': 885.0, 'C': 1378.0, 'G': 4078.0, 'T': 684.0},\n",
       "  15: {'A': 1252.0, 'C': 2655.0, 'G': 1592.0, 'T': 1526.0},\n",
       "  16: {'A': 1870.0, 'C': 1620.0, 'G': 2069.0, 'T': 1466.0},\n",
       "  17: {'A': 1496.0, 'C': 1802.0, 'G': 2218.0, 'T': 1509.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '10082572',\n",
       "  'uniprot_ids': 'P40426',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1475.1',\n",
       "  'Matrix_Name': 'CREB3L4(var.2)',\n",
       "  1: {'A': 230.0, 'C': 225.0, 'G': 571.0, 'T': 271.0},\n",
       "  2: {'A': 505.0, 'C': 222.0, 'G': 791.0, 'T': 76.0},\n",
       "  3: {'A': 1.0, 'C': 139.0, 'G': 49.0, 'T': 1294.0},\n",
       "  4: {'A': 36.0, 'C': 266.0, 'G': 1294.0, 'T': 259.0},\n",
       "  5: {'A': 1294.0, 'C': 28.0, 'G': 84.0, 'T': 40.0},\n",
       "  6: {'A': 1.0, 'C': 1294.0, 'G': 7.0, 'T': 44.0},\n",
       "  7: {'A': 82.0, 'C': 4.0, 'G': 1294.0, 'T': 1.0},\n",
       "  8: {'A': 17.0, 'C': 101.0, 'G': 37.0, 'T': 1294.0},\n",
       "  9: {'A': 177.0, 'C': 1294.0, 'G': 228.0, 'T': 60.0},\n",
       "  10: {'A': 1294.0, 'C': 36.0, 'G': 154.0, 'T': 41.0},\n",
       "  11: {'A': 117.0, 'C': 749.0, 'G': 283.0, 'T': 547.0},\n",
       "  12: {'A': 235.0, 'C': 583.0, 'G': 266.0, 'T': 213.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21711675',\n",
       "  'uniprot_ids': 'Q8TEY5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0711.1',\n",
       "  'Matrix_Name': 'OTX1',\n",
       "  1: {'A': 5119.0, 'C': 6502.0, 'G': 3187.0, 'T': 8637.0},\n",
       "  2: {'A': 746.0, 'C': 91.0, 'G': 1.0, 'T': 23442.0},\n",
       "  3: {'A': 23442.0, 'C': 1.0, 'G': 255.0, 'T': 545.0},\n",
       "  4: {'A': 23442.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 244.0, 'C': 1.0, 'G': 3161.0, 'T': 23442.0},\n",
       "  6: {'A': 648.0, 'C': 23442.0, 'G': 660.0, 'T': 379.0},\n",
       "  7: {'A': 790.0, 'C': 23442.0, 'G': 8663.0, 'T': 3364.0},\n",
       "  8: {'A': 1874.0, 'C': 6183.0, 'G': 10600.0, 'T': 4788.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P32242',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1578.1',\n",
       "  'Matrix_Name': 'VEZF1',\n",
       "  1: {'A': 72.0, 'C': 791.0, 'G': 138.0, 'T': 1.0},\n",
       "  2: {'A': 2.0, 'C': 998.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 5.0, 'C': 996.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 3.0, 'C': 998.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 7.0, 'C': 990.0, 'G': 1.0, 'T': 5.0},\n",
       "  6: {'A': 11.0, 'C': 978.0, 'G': 4.0, 'T': 9.0},\n",
       "  7: {'A': 407.0, 'C': 358.0, 'G': 111.0, 'T': 126.0},\n",
       "  8: {'A': 45.0, 'C': 551.0, 'G': 71.0, 'T': 335.0},\n",
       "  9: {'A': 241.0, 'C': 172.0, 'G': 212.0, 'T': 378.0},\n",
       "  10: {'A': 334.0, 'C': 174.0, 'G': 153.0, 'T': 340.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '20062523',\n",
       "  'uniprot_ids': 'Q14119',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1484.1',\n",
       "  'Matrix_Name': 'ETS2',\n",
       "  1: {'A': 5262.0, 'C': 2831.0, 'G': 5703.0, 'T': 2946.0},\n",
       "  2: {'A': 16739.0, 'C': 472.0, 'G': 3137.0, 'T': 728.0},\n",
       "  3: {'A': 644.0, 'C': 16739.0, 'G': 1307.0, 'T': 1.0},\n",
       "  4: {'A': 1280.0, 'C': 16739.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 8.0, 'C': 1.0, 'G': 16739.0, 'T': 6.0},\n",
       "  6: {'A': 5.0, 'C': 8.0, 'G': 16739.0, 'T': 1.0},\n",
       "  7: {'A': 16739.0, 'C': 14.0, 'G': 8.0, 'T': 1.0},\n",
       "  8: {'A': 11583.0, 'C': 205.0, 'G': 1.0, 'T': 4954.0},\n",
       "  9: {'A': 3051.0, 'C': 1.0, 'G': 13731.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 8137.0, 'G': 416.0, 'T': 8191.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'P15036',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0591.1',\n",
       "  'Matrix_Name': 'Bach1::Mafk',\n",
       "  1: {'A': 46.0, 'C': 15.0, 'G': 42.0, 'T': 15.0},\n",
       "  2: {'A': 15.0, 'C': 37.0, 'G': 47.0, 'T': 19.0},\n",
       "  3: {'A': 16.0, 'C': 42.0, 'G': 55.0, 'T': 5.0},\n",
       "  4: {'A': 67.0, 'C': 17.0, 'G': 33.0, 'T': 1.0},\n",
       "  5: {'A': 3.0, 'C': 2.0, 'G': 1.0, 'T': 112.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 108.0, 'T': 8.0},\n",
       "  7: {'A': 114.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 1.0, 'C': 95.0, 'G': 21.0, 'T': 1.0},\n",
       "  9: {'A': 2.0, 'C': 1.0, 'G': 5.0, 'T': 110.0},\n",
       "  10: {'A': 6.0, 'C': 108.0, 'G': 3.0, 'T': 1.0},\n",
       "  11: {'A': 109.0, 'C': 2.0, 'G': 4.0, 'T': 3.0},\n",
       "  12: {'A': 1.0, 'C': 2.0, 'G': 114.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 115.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 99.0, 'C': 5.0, 'G': 7.0, 'T': 7.0},\n",
       "  15: {'A': 13.0, 'C': 43.0, 'G': 39.0, 'T': 23.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11069897',\n",
       "  'uniprot_ids': 'P97302;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0067.1',\n",
       "  'Matrix_Name': 'Pax2',\n",
       "  1: {'A': 11.0, 'C': 8.0, 'G': 6.0, 'T': 10.0},\n",
       "  2: {'A': 8.0, 'C': 2.0, 'G': 22.0, 'T': 3.0},\n",
       "  3: {'A': 4.0, 'C': 3.0, 'G': 1.0, 'T': 27.0},\n",
       "  4: {'A': 1.0, 'C': 29.0, 'G': 2.0, 'T': 3.0},\n",
       "  5: {'A': 27.0, 'C': 2.0, 'G': 4.0, 'T': 2.0},\n",
       "  6: {'A': 3.0, 'C': 18.0, 'G': 2.0, 'T': 12.0},\n",
       "  7: {'A': 3.0, 'C': 1.0, 'G': 20.0, 'T': 11.0},\n",
       "  8: {'A': 2.0, 'C': 12.0, 'G': 12.0, 'T': 9.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '8132558',\n",
       "  'uniprot_ids': 'P32114',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0705.1',\n",
       "  'Matrix_Name': 'Lhx8',\n",
       "  1: {'A': 473.0, 'C': 2589.0, 'G': 926.0, 'T': 1109.0},\n",
       "  2: {'A': 1.0, 'C': 1997.0, 'G': 1.0, 'T': 5093.0},\n",
       "  3: {'A': 5093.0, 'C': 161.0, 'G': 1671.0, 'T': 1.0},\n",
       "  4: {'A': 5093.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 5093.0},\n",
       "  6: {'A': 1.0, 'C': 2084.0, 'G': 89.0, 'T': 5093.0},\n",
       "  7: {'A': 5093.0, 'C': 1.0, 'G': 1930.0, 'T': 1.0},\n",
       "  8: {'A': 1367.0, 'C': 1157.0, 'G': 2157.0, 'T': 415.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'O35652',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0099.3',\n",
       "  'Matrix_Name': 'FOS::JUN',\n",
       "  1: {'A': 593.0, 'C': 112.0, 'G': 258.0, 'T': 41.0},\n",
       "  2: {'A': 4.0, 'C': 6.0, 'G': 3.0, 'T': 991.0},\n",
       "  3: {'A': 8.0, 'C': 2.0, 'G': 825.0, 'T': 169.0},\n",
       "  4: {'A': 821.0, 'C': 132.0, 'G': 5.0, 'T': 46.0},\n",
       "  5: {'A': 79.0, 'C': 181.0, 'G': 689.0, 'T': 55.0},\n",
       "  6: {'A': 6.0, 'C': 3.0, 'G': 3.0, 'T': 992.0},\n",
       "  7: {'A': 31.0, 'C': 964.0, 'G': 4.0, 'T': 5.0},\n",
       "  8: {'A': 993.0, 'C': 3.0, 'G': 3.0, 'T': 5.0},\n",
       "  9: {'A': 25.0, 'C': 314.0, 'G': 96.0, 'T': 569.0},\n",
       "  10: {'A': 278.0, 'C': 398.0, 'G': 165.0, 'T': 163.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P01100;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0712.2',\n",
       "  'Matrix_Name': 'OTX2',\n",
       "  1: {'A': 42532.0, 'C': 24446.0, 'G': 22125.0, 'T': 31580.0},\n",
       "  2: {'A': 39167.0, 'C': 17329.0, 'G': 31826.0, 'T': 32361.0},\n",
       "  3: {'A': 37156.0, 'C': 5323.0, 'G': 55815.0, 'T': 22389.0},\n",
       "  4: {'A': 5436.0, 'C': 828.0, 'G': 111742.0, 'T': 2677.0},\n",
       "  5: {'A': 2919.0, 'C': 1621.0, 'G': 113843.0, 'T': 2300.0},\n",
       "  6: {'A': 116412.0, 'C': 2221.0, 'G': 707.0, 'T': 1343.0},\n",
       "  7: {'A': 766.0, 'C': 599.0, 'G': 788.0, 'T': 118530.0},\n",
       "  8: {'A': 1914.0, 'C': 1772.0, 'G': 1408.0, 'T': 115589.0},\n",
       "  9: {'A': 114799.0, 'C': 637.0, 'G': 1303.0, 'T': 3944.0},\n",
       "  10: {'A': 30849.0, 'C': 12348.0, 'G': 52480.0, 'T': 25006.0},\n",
       "  11: {'A': 39803.0, 'C': 25480.0, 'G': 27154.0, 'T': 28246.0},\n",
       "  12: {'A': 36007.0, 'C': 20624.0, 'G': 29815.0, 'T': 34237.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P32243',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0797.1',\n",
       "  'Matrix_Name': 'TGIF2',\n",
       "  1: {'A': 478.0, 'C': 422.0, 'G': 154.0, 'T': 11802.0},\n",
       "  2: {'A': 98.0, 'C': 49.0, 'G': 11802.0, 'T': 53.0},\n",
       "  3: {'A': 11802.0, 'C': 35.0, 'G': 397.0, 'T': 156.0},\n",
       "  4: {'A': 128.0, 'C': 11802.0, 'G': 52.0, 'T': 157.0},\n",
       "  5: {'A': 11802.0, 'C': 26.0, 'G': 271.0, 'T': 42.0},\n",
       "  6: {'A': 325.0, 'C': 263.0, 'G': 11802.0, 'T': 2691.0},\n",
       "  7: {'A': 1502.0, 'C': 7460.0, 'G': 4342.0, 'T': 92.0},\n",
       "  8: {'A': 72.0, 'C': 411.0, 'G': 58.0, 'T': 11802.0},\n",
       "  9: {'A': 151.0, 'C': 78.0, 'G': 11802.0, 'T': 84.0},\n",
       "  10: {'A': 602.0, 'C': 622.0, 'G': 124.0, 'T': 11802.0},\n",
       "  11: {'A': 71.0, 'C': 11802.0, 'G': 137.0, 'T': 114.0},\n",
       "  12: {'A': 11802.0, 'C': 173.0, 'G': 396.0, 'T': 421.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8537382',\n",
       "  'uniprot_ids': 'Q9GZN2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0884.1',\n",
       "  'Matrix_Name': 'DUXA',\n",
       "  1: {'A': 225.0, 'C': 314.0, 'G': 195.0, 'T': 284.0},\n",
       "  2: {'A': 242.0, 'C': 286.0, 'G': 112.0, 'T': 1015.0},\n",
       "  3: {'A': 656.0, 'C': 6.0, 'G': 360.0, 'T': 13.0},\n",
       "  4: {'A': 1015.0, 'C': 7.0, 'G': 12.0, 'T': 6.0},\n",
       "  5: {'A': 137.0, 'C': 395.0, 'G': 126.0, 'T': 621.0},\n",
       "  6: {'A': 13.0, 'C': 393.0, 'G': 194.0, 'T': 418.0},\n",
       "  7: {'A': 24.0, 'C': 345.0, 'G': 117.0, 'T': 1015.0},\n",
       "  8: {'A': 1015.0, 'C': 57.0, 'G': 48.0, 'T': 23.0},\n",
       "  9: {'A': 1015.0, 'C': 8.0, 'G': 11.0, 'T': 8.0},\n",
       "  10: {'A': 1.0, 'C': 2.0, 'G': 6.0, 'T': 1015.0},\n",
       "  11: {'A': 1.0, 'C': 1015.0, 'G': 4.0, 'T': 109.0},\n",
       "  12: {'A': 1015.0, 'C': 30.0, 'G': 53.0, 'T': 93.0},\n",
       "  13: {'A': 365.0, 'C': 179.0, 'G': 284.0, 'T': 190.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '24278031',\n",
       "  'uniprot_ids': 'A6NLW8',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0490.2',\n",
       "  'Matrix_Name': 'JUNB',\n",
       "  1: {'A': 22773.0, 'C': 16584.0, 'G': 18151.0, 'T': 22352.0},\n",
       "  2: {'A': 23562.0, 'C': 15808.0, 'G': 20404.0, 'T': 20086.0},\n",
       "  3: {'A': 56608.0, 'C': 7818.0, 'G': 11567.0, 'T': 3867.0},\n",
       "  4: {'A': 539.0, 'C': 655.0, 'G': 323.0, 'T': 78343.0},\n",
       "  5: {'A': 771.0, 'C': 551.0, 'G': 74758.0, 'T': 3780.0},\n",
       "  6: {'A': 78102.0, 'C': 524.0, 'G': 551.0, 'T': 683.0},\n",
       "  7: {'A': 4756.0, 'C': 59785.0, 'G': 10520.0, 'T': 4799.0},\n",
       "  8: {'A': 2000.0, 'C': 290.0, 'G': 914.0, 'T': 76656.0},\n",
       "  9: {'A': 5244.0, 'C': 73200.0, 'G': 618.0, 'T': 798.0},\n",
       "  10: {'A': 77676.0, 'C': 589.0, 'G': 901.0, 'T': 694.0},\n",
       "  11: {'A': 3723.0, 'C': 8877.0, 'G': 7509.0, 'T': 59751.0},\n",
       "  12: {'A': 21545.0, 'C': 19231.0, 'G': 16492.0, 'T': 22592.0},\n",
       "  13: {'A': 21138.0, 'C': 20654.0, 'G': 15069.0, 'T': 22999.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21526160',\n",
       "  'uniprot_ids': 'P17275',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0832.1',\n",
       "  'Matrix_Name': 'Tcf21',\n",
       "  1: {'A': 56.0, 'C': 30.0, 'G': 71.0, 'T': 18.0},\n",
       "  2: {'A': 29.0, 'C': 77.0, 'G': 23.0, 'T': 47.0},\n",
       "  3: {'A': 172.0, 'C': 2.0, 'G': 24.0, 'T': 4.0},\n",
       "  4: {'A': 172.0, 'C': 11.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 172.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 172.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 172.0, 'T': 3.0},\n",
       "  8: {'A': 1.0, 'C': 172.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 172.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 172.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 5.0, 'G': 21.0, 'T': 172.0},\n",
       "  12: {'A': 6.0, 'C': 24.0, 'G': 1.0, 'T': 172.0},\n",
       "  13: {'A': 51.0, 'C': 24.0, 'G': 81.0, 'T': 20.0},\n",
       "  14: {'A': 31.0, 'C': 50.0, 'G': 33.0, 'T': 61.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '26020271',\n",
       "  'uniprot_ids': 'O35437',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0624.1',\n",
       "  'Matrix_Name': 'NFATC1',\n",
       "  1: {'A': 384.0, 'C': 135.0, 'G': 168.0, 'T': 317.0},\n",
       "  2: {'A': 227.0, 'C': 194.0, 'G': 190.0, 'T': 393.0},\n",
       "  3: {'A': 159.0, 'C': 61.0, 'G': 123.0, 'T': 661.0},\n",
       "  4: {'A': 20.0, 'C': 36.0, 'G': 8.0, 'T': 940.0},\n",
       "  5: {'A': 22.0, 'C': 30.0, 'G': 16.0, 'T': 936.0},\n",
       "  6: {'A': 15.0, 'C': 972.0, 'G': 9.0, 'T': 8.0},\n",
       "  7: {'A': 10.0, 'C': 968.0, 'G': 6.0, 'T': 19.0},\n",
       "  8: {'A': 517.0, 'C': 46.0, 'G': 358.0, 'T': 84.0},\n",
       "  9: {'A': 231.0, 'C': 263.0, 'G': 127.0, 'T': 384.0},\n",
       "  10: {'A': 231.0, 'C': 228.0, 'G': 195.0, 'T': 350.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NFAT-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'G1MTN6',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1463.1',\n",
       "  'Matrix_Name': 'ARGFX',\n",
       "  1: {'A': 3894.0, 'C': 1751.0, 'G': 4468.0, 'T': 6480.0},\n",
       "  2: {'A': 2559.0, 'C': 16590.0, 'G': 1749.0, 'T': 4424.0},\n",
       "  3: {'A': 271.0, 'C': 1131.0, 'G': 97.0, 'T': 16590.0},\n",
       "  4: {'A': 16590.0, 'C': 47.0, 'G': 1.0, 'T': 5.0},\n",
       "  5: {'A': 16590.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 16590.0},\n",
       "  7: {'A': 1.0, 'C': 45.0, 'G': 1.0, 'T': 16590.0},\n",
       "  8: {'A': 16590.0, 'C': 500.0, 'G': 1493.0, 'T': 1049.0},\n",
       "  9: {'A': 10341.0, 'C': 564.0, 'G': 6250.0, 'T': 901.0},\n",
       "  10: {'A': 6984.0, 'C': 4099.0, 'G': 2352.0, 'T': 3159.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'A6NJG6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0152.1',\n",
       "  'Matrix_Name': 'NFATC2',\n",
       "  1: {'A': 4.0, 'C': 2.0, 'G': 3.0, 'T': 21.0},\n",
       "  2: {'A': 2.0, 'C': 3.0, 'G': 3.0, 'T': 22.0},\n",
       "  3: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 25.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 27.0},\n",
       "  5: {'A': 2.0, 'C': 26.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 27.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 19.0, 'C': 4.0, 'G': 2.0, 'T': 5.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'NFAT-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'Q13469',\n",
       "  'data_type': 'COMPILED'},\n",
       " {'Matrix_ID': 'MA1651.1',\n",
       "  'Matrix_Name': 'ZFP42',\n",
       "  1: {'A': 369.0, 'C': 470.0, 'G': 503.0, 'T': 414.0},\n",
       "  2: {'A': 455.0, 'C': 421.0, 'G': 413.0, 'T': 467.0},\n",
       "  3: {'A': 406.0, 'C': 403.0, 'G': 391.0, 'T': 556.0},\n",
       "  4: {'A': 392.0, 'C': 661.0, 'G': 328.0, 'T': 375.0},\n",
       "  5: {'A': 330.0, 'C': 992.0, 'G': 199.0, 'T': 235.0},\n",
       "  6: {'A': 1609.0, 'C': 33.0, 'G': 80.0, 'T': 34.0},\n",
       "  7: {'A': 1554.0, 'C': 55.0, 'G': 70.0, 'T': 77.0},\n",
       "  8: {'A': 818.0, 'C': 162.0, 'G': 669.0, 'T': 107.0},\n",
       "  9: {'A': 1725.0, 'C': 8.0, 'G': 14.0, 'T': 9.0},\n",
       "  10: {'A': 7.0, 'C': 14.0, 'G': 8.0, 'T': 1727.0},\n",
       "  11: {'A': 6.0, 'C': 6.0, 'G': 1742.0, 'T': 2.0},\n",
       "  12: {'A': 26.0, 'C': 30.0, 'G': 1575.0, 'T': 125.0},\n",
       "  13: {'A': 32.0, 'C': 1704.0, 'G': 8.0, 'T': 12.0},\n",
       "  14: {'A': 43.0, 'C': 117.0, 'G': 326.0, 'T': 1270.0},\n",
       "  15: {'A': 83.0, 'C': 282.0, 'G': 1242.0, 'T': 149.0},\n",
       "  16: {'A': 126.0, 'C': 1378.0, 'G': 79.0, 'T': 173.0},\n",
       "  17: {'A': 167.0, 'C': 1080.0, 'G': 188.0, 'T': 321.0},\n",
       "  18: {'A': 418.0, 'C': 458.0, 'G': 360.0, 'T': 520.0},\n",
       "  19: {'A': 268.0, 'C': 652.0, 'G': 425.0, 'T': 411.0},\n",
       "  20: {'A': 367.0, 'C': 564.0, 'G': 448.0, 'T': 377.0},\n",
       "  21: {'A': 442.0, 'C': 406.0, 'G': 505.0, 'T': 403.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 21,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q96MM3',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1117.1',\n",
       "  'Matrix_Name': 'RELB',\n",
       "  1: {'A': 797.0, 'C': 415.0, 'G': 1576.0, 'T': 506.0},\n",
       "  2: {'A': 1232.0, 'C': 562.0, 'G': 795.0, 'T': 705.0},\n",
       "  3: {'A': 2834.0, 'C': 128.0, 'G': 158.0, 'T': 174.0},\n",
       "  4: {'A': 211.0, 'C': 108.0, 'G': 160.0, 'T': 2815.0},\n",
       "  5: {'A': 91.0, 'C': 104.0, 'G': 54.0, 'T': 3045.0},\n",
       "  6: {'A': 145.0, 'C': 2929.0, 'G': 41.0, 'T': 179.0},\n",
       "  7: {'A': 93.0, 'C': 3099.0, 'G': 29.0, 'T': 73.0},\n",
       "  8: {'A': 147.0, 'C': 3014.0, 'G': 59.0, 'T': 74.0},\n",
       "  9: {'A': 80.0, 'C': 3036.0, 'G': 108.0, 'T': 70.0},\n",
       "  10: {'A': 797.0, 'C': 629.0, 'G': 985.0, 'T': 883.0},\n",
       "  11: {'A': 712.0, 'C': 922.0, 'G': 997.0, 'T': 663.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'NF-kappaB-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '17996728',\n",
       "  'uniprot_ids': 'Q01201',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0681.2',\n",
       "  'Matrix_Name': 'PHOX2B',\n",
       "  1: {'A': 18531.0, 'C': 7302.0, 'G': 8173.0, 'T': 19234.0},\n",
       "  2: {'A': 19444.0, 'C': 7037.0, 'G': 6745.0, 'T': 20014.0},\n",
       "  3: {'A': 10345.0, 'C': 2749.0, 'G': 2543.0, 'T': 37603.0},\n",
       "  4: {'A': 42697.0, 'C': 1936.0, 'G': 2290.0, 'T': 6317.0},\n",
       "  5: {'A': 51240.0, 'C': 497.0, 'G': 509.0, 'T': 994.0},\n",
       "  6: {'A': 1743.0, 'C': 616.0, 'G': 468.0, 'T': 50413.0},\n",
       "  7: {'A': 2452.0, 'C': 36620.0, 'G': 1874.0, 'T': 12294.0},\n",
       "  8: {'A': 43075.0, 'C': 1896.0, 'G': 589.0, 'T': 7680.0},\n",
       "  9: {'A': 48551.0, 'C': 1454.0, 'G': 1419.0, 'T': 1816.0},\n",
       "  10: {'A': 50676.0, 'C': 452.0, 'G': 762.0, 'T': 1350.0},\n",
       "  11: {'A': 922.0, 'C': 330.0, 'G': 373.0, 'T': 51615.0},\n",
       "  12: {'A': 4104.0, 'C': 1924.0, 'G': 1233.0, 'T': 45979.0},\n",
       "  13: {'A': 42870.0, 'C': 1937.0, 'G': 2156.0, 'T': 6277.0},\n",
       "  14: {'A': 16089.0, 'C': 4910.0, 'G': 5570.0, 'T': 26671.0},\n",
       "  15: {'A': 20367.0, 'C': 8281.0, 'G': 7631.0, 'T': 16961.0},\n",
       "  16: {'A': 18566.0, 'C': 7410.0, 'G': 7000.0, 'T': 20264.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '22323723',\n",
       "  'uniprot_ids': 'Q99453',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0509.2',\n",
       "  'Matrix_Name': 'RFX1',\n",
       "  1: {'A': 3239.0, 'C': 5592.0, 'G': 3750.0, 'T': 5415.0},\n",
       "  2: {'A': 3486.0, 'C': 2609.0, 'G': 5167.0, 'T': 6734.0},\n",
       "  3: {'A': 430.0, 'C': 108.0, 'G': 17250.0, 'T': 208.0},\n",
       "  4: {'A': 133.0, 'C': 301.0, 'G': 171.0, 'T': 17391.0},\n",
       "  5: {'A': 487.0, 'C': 519.0, 'G': 169.0, 'T': 16821.0},\n",
       "  6: {'A': 1090.0, 'C': 303.0, 'G': 15837.0, 'T': 766.0},\n",
       "  7: {'A': 123.0, 'C': 17172.0, 'G': 100.0, 'T': 601.0},\n",
       "  8: {'A': 65.0, 'C': 6615.0, 'G': 104.0, 'T': 11212.0},\n",
       "  9: {'A': 16568.0, 'C': 314.0, 'G': 774.0, 'T': 340.0},\n",
       "  10: {'A': 2379.0, 'C': 844.0, 'G': 2718.0, 'T': 12055.0},\n",
       "  11: {'A': 2059.0, 'C': 484.0, 'G': 14465.0, 'T': 988.0},\n",
       "  12: {'A': 2608.0, 'C': 1565.0, 'G': 12434.0, 'T': 1389.0},\n",
       "  13: {'A': 4512.0, 'C': 5985.0, 'G': 3570.0, 'T': 3929.0},\n",
       "  14: {'A': 9395.0, 'C': 1918.0, 'G': 3718.0, 'T': 2965.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RFX-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'P48743',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0692.1',\n",
       "  'Matrix_Name': 'TFEB',\n",
       "  1: {'A': 4968.0, 'C': 1554.0, 'G': 3749.0, 'T': 344.0},\n",
       "  2: {'A': 2546.0, 'C': 4937.0, 'G': 2732.0, 'T': 8716.0},\n",
       "  3: {'A': 18.0, 'C': 8716.0, 'G': 1.0, 'T': 4.0},\n",
       "  4: {'A': 8716.0, 'C': 287.0, 'G': 63.0, 'T': 275.0},\n",
       "  5: {'A': 1.0, 'C': 8716.0, 'G': 84.0, 'T': 1156.0},\n",
       "  6: {'A': 2763.0, 'C': 289.0, 'G': 8716.0, 'T': 6.0},\n",
       "  7: {'A': 455.0, 'C': 170.0, 'G': 151.0, 'T': 8716.0},\n",
       "  8: {'A': 10.0, 'C': 11.0, 'G': 8716.0, 'T': 12.0},\n",
       "  9: {'A': 8716.0, 'C': 2343.0, 'G': 873.0, 'T': 808.0},\n",
       "  10: {'A': 284.0, 'C': 5890.0, 'G': 559.0, 'T': 2827.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '1748288',\n",
       "  'uniprot_ids': 'P19484',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0627.2',\n",
       "  'Matrix_Name': 'POU2F3',\n",
       "  1: {'A': 17400.0, 'C': 11384.0, 'G': 10723.0, 'T': 16526.0},\n",
       "  2: {'A': 26661.0, 'C': 6825.0, 'G': 8771.0, 'T': 13776.0},\n",
       "  3: {'A': 8523.0, 'C': 6886.0, 'G': 3030.0, 'T': 37594.0},\n",
       "  4: {'A': 54551.0, 'C': 195.0, 'G': 355.0, 'T': 932.0},\n",
       "  5: {'A': 1091.0, 'C': 323.0, 'G': 1252.0, 'T': 53367.0},\n",
       "  6: {'A': 981.0, 'C': 252.0, 'G': 51972.0, 'T': 2828.0},\n",
       "  7: {'A': 947.0, 'C': 50289.0, 'G': 1080.0, 'T': 3717.0},\n",
       "  8: {'A': 54916.0, 'C': 81.0, 'G': 139.0, 'T': 897.0},\n",
       "  9: {'A': 51383.0, 'C': 568.0, 'G': 805.0, 'T': 3277.0},\n",
       "  10: {'A': 55303.0, 'C': 165.0, 'G': 234.0, 'T': 331.0},\n",
       "  11: {'A': 5116.0, 'C': 3479.0, 'G': 4099.0, 'T': 43339.0},\n",
       "  12: {'A': 16289.0, 'C': 10596.0, 'G': 13238.0, 'T': 15910.0},\n",
       "  13: {'A': 18908.0, 'C': 11937.0, 'G': 7942.0, 'T': 17246.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q9UKI9',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0035.4',\n",
       "  'Matrix_Name': 'GATA1',\n",
       "  1: {'A': 22210.0, 'C': 12210.0, 'G': 13956.0, 'T': 23456.0},\n",
       "  2: {'A': 17329.0, 'C': 14490.0, 'G': 11089.0, 'T': 28924.0},\n",
       "  3: {'A': 3954.0, 'C': 62420.0, 'G': 2713.0, 'T': 2745.0},\n",
       "  4: {'A': 1315.0, 'C': 596.0, 'G': 653.0, 'T': 69268.0},\n",
       "  5: {'A': 49693.0, 'C': 711.0, 'G': 857.0, 'T': 20571.0},\n",
       "  6: {'A': 67551.0, 'C': 1293.0, 'G': 989.0, 'T': 1999.0},\n",
       "  7: {'A': 2207.0, 'C': 1239.0, 'G': 619.0, 'T': 67767.0},\n",
       "  8: {'A': 2568.0, 'C': 65938.0, 'G': 1472.0, 'T': 1854.0},\n",
       "  9: {'A': 7398.0, 'C': 3026.0, 'G': 766.0, 'T': 60642.0},\n",
       "  10: {'A': 26546.0, 'C': 11187.0, 'G': 14359.0, 'T': 19740.0},\n",
       "  11: {'A': 22657.0, 'C': 15262.0, 'G': 9326.0, 'T': 24587.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '23142663',\n",
       "  'uniprot_ids': 'P15976',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0116.1',\n",
       "  'Matrix_Name': 'Znf423',\n",
       "  1: {'A': 8.0, 'C': 5.0, 'G': 23.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 17.0, 'G': 18.0, 'T': 1.0},\n",
       "  3: {'A': 17.0, 'C': 18.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 18.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 34.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 34.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 2.0, 'C': 18.0, 'G': 1.0, 'T': 16.0},\n",
       "  8: {'A': 25.0, 'C': 1.0, 'G': 1.0, 'T': 10.0},\n",
       "  9: {'A': 14.0, 'C': 1.0, 'G': 17.0, 'T': 5.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 34.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 34.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 18.0, 'T': 17.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 17.0, 'T': 18.0},\n",
       "  14: {'A': 1.0, 'C': 9.0, 'G': 17.0, 'T': 10.0},\n",
       "  15: {'A': 12.0, 'C': 23.0, 'G': 1.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '9774661',\n",
       "  'uniprot_ids': 'O08961',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0502.2',\n",
       "  'Matrix_Name': 'NFYB',\n",
       "  1: {'A': 814.0, 'C': 3804.0, 'G': 1422.0, 'T': 1418.0},\n",
       "  2: {'A': 439.0, 'C': 2706.0, 'G': 902.0, 'T': 3411.0},\n",
       "  3: {'A': 194.0, 'C': 5448.0, 'G': 1482.0, 'T': 334.0},\n",
       "  4: {'A': 6984.0, 'C': 182.0, 'G': 58.0, 'T': 234.0},\n",
       "  5: {'A': 135.0, 'C': 242.0, 'G': 228.0, 'T': 6853.0},\n",
       "  6: {'A': 85.0, 'C': 38.0, 'G': 103.0, 'T': 7232.0},\n",
       "  7: {'A': 54.0, 'C': 42.0, 'G': 7251.0, 'T': 111.0},\n",
       "  8: {'A': 64.0, 'C': 59.0, 'G': 7273.0, 'T': 62.0},\n",
       "  9: {'A': 155.0, 'C': 6489.0, 'G': 174.0, 'T': 640.0},\n",
       "  10: {'A': 155.0, 'C': 5824.0, 'G': 140.0, 'T': 1339.0},\n",
       "  11: {'A': 3122.0, 'C': 1671.0, 'G': 1636.0, 'T': 1029.0},\n",
       "  12: {'A': 2318.0, 'C': 1490.0, 'G': 2664.0, 'T': 986.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': '',\n",
       "  'tf_class': 'Heteromeric',\n",
       "  'pubmed_ids': '9334236',\n",
       "  'uniprot_ids': 'P25208',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1508.1',\n",
       "  'Matrix_Name': 'IKZF1',\n",
       "  1: {'A': 3296.0, 'C': 2056.0, 'G': 3874.0, 'T': 1696.0},\n",
       "  2: {'A': 4188.0, 'C': 2293.0, 'G': 2869.0, 'T': 1572.0},\n",
       "  3: {'A': 8123.0, 'C': 1159.0, 'G': 1008.0, 'T': 632.0},\n",
       "  4: {'A': 9138.0, 'C': 474.0, 'G': 957.0, 'T': 353.0},\n",
       "  5: {'A': 674.0, 'C': 8679.0, 'G': 1273.0, 'T': 296.0},\n",
       "  6: {'A': 10195.0, 'C': 213.0, 'G': 253.0, 'T': 261.0},\n",
       "  7: {'A': 126.0, 'C': 107.0, 'G': 10570.0, 'T': 119.0},\n",
       "  8: {'A': 419.0, 'C': 179.0, 'G': 10186.0, 'T': 138.0},\n",
       "  9: {'A': 10324.0, 'C': 248.0, 'G': 298.0, 'T': 52.0},\n",
       "  10: {'A': 10255.0, 'C': 115.0, 'G': 262.0, 'T': 290.0},\n",
       "  11: {'A': 3301.0, 'C': 1676.0, 'G': 4920.0, 'T': 1025.0},\n",
       "  12: {'A': 2222.0, 'C': 2992.0, 'G': 2318.0, 'T': 3390.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '27664237',\n",
       "  'uniprot_ids': 'Q13422',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0690.1',\n",
       "  'Matrix_Name': 'TBX21',\n",
       "  1: {'A': 2323.0, 'C': 264.0, 'G': 938.0, 'T': 873.0},\n",
       "  2: {'A': 4396.0, 'C': 31.0, 'G': 360.0, 'T': 147.0},\n",
       "  3: {'A': 613.0, 'C': 240.0, 'G': 4396.0, 'T': 399.0},\n",
       "  4: {'A': 3.0, 'C': 18.0, 'G': 4396.0, 'T': 34.0},\n",
       "  5: {'A': 7.0, 'C': 181.0, 'G': 17.0, 'T': 4396.0},\n",
       "  6: {'A': 2.0, 'C': 1.0, 'G': 4396.0, 'T': 3.0},\n",
       "  7: {'A': 186.0, 'C': 332.0, 'G': 35.0, 'T': 4396.0},\n",
       "  8: {'A': 81.0, 'C': 127.0, 'G': 4396.0, 'T': 365.0},\n",
       "  9: {'A': 4396.0, 'C': 1.0, 'G': 39.0, 'T': 1.0},\n",
       "  10: {'A': 4396.0, 'C': 442.0, 'G': 318.0, 'T': 1288.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'TBrain-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '16285859',\n",
       "  'uniprot_ids': 'Q9UL17',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1603.1',\n",
       "  'Matrix_Name': 'Dmrt1',\n",
       "  1: {'A': 2594.0, 'C': 1114.0, 'G': 12330.0, 'T': 2261.0},\n",
       "  2: {'A': 7474.0, 'C': 4223.0, 'G': 1247.0, 'T': 5355.0},\n",
       "  3: {'A': 3916.0, 'C': 1229.0, 'G': 919.0, 'T': 12235.0},\n",
       "  4: {'A': 17934.0, 'C': 70.0, 'G': 83.0, 'T': 212.0},\n",
       "  5: {'A': 212.0, 'C': 17634.0, 'G': 140.0, 'T': 313.0},\n",
       "  6: {'A': 17490.0, 'C': 111.0, 'G': 105.0, 'T': 593.0},\n",
       "  7: {'A': 15569.0, 'C': 483.0, 'G': 457.0, 'T': 1790.0},\n",
       "  8: {'A': 12434.0, 'C': 206.0, 'G': 323.0, 'T': 5336.0},\n",
       "  9: {'A': 728.0, 'C': 186.0, 'G': 17033.0, 'T': 352.0},\n",
       "  10: {'A': 548.0, 'C': 162.0, 'G': 201.0, 'T': 17388.0},\n",
       "  11: {'A': 10368.0, 'C': 1234.0, 'G': 2522.0, 'T': 4175.0},\n",
       "  12: {'A': 5910.0, 'C': 1334.0, 'G': 4133.0, 'T': 6922.0},\n",
       "  13: {'A': 2689.0, 'C': 11810.0, 'G': 1390.0, 'T': 2410.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'DMRT',\n",
       "  'tf_class': 'DM-type',\n",
       "  'pubmed_ids': '26005864',\n",
       "  'uniprot_ids': 'Q9QZ59',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1542.1',\n",
       "  'Matrix_Name': 'OSR1',\n",
       "  1: {'A': 1029.0, 'C': 1303.0, 'G': 718.0, 'T': 1844.0},\n",
       "  2: {'A': 449.0, 'C': 236.0, 'G': 4891.0, 'T': 18.0},\n",
       "  3: {'A': 1.0, 'C': 4891.0, 'G': 1.0, 'T': 19.0},\n",
       "  4: {'A': 5.0, 'C': 100.0, 'G': 1.0, 'T': 4891.0},\n",
       "  5: {'A': 4891.0, 'C': 84.0, 'G': 39.0, 'T': 445.0},\n",
       "  6: {'A': 11.0, 'C': 4891.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 2104.0, 'G': 6.0, 'T': 2788.0},\n",
       "  8: {'A': 63.0, 'C': 5.0, 'G': 4891.0, 'T': 25.0},\n",
       "  9: {'A': 408.0, 'C': 613.0, 'G': 505.0, 'T': 4891.0},\n",
       "  10: {'A': 1103.0, 'C': 546.0, 'G': 1513.0, 'T': 1732.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Other',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '29084951',\n",
       "  'uniprot_ids': 'Q8TAX0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0693.2',\n",
       "  'Matrix_Name': 'VDR',\n",
       "  1: {'A': 223.0, 'C': 281.0, 'G': 44.0, 'T': 456.0},\n",
       "  2: {'A': 357.0, 'C': 46.0, 'G': 568.0, 'T': 33.0},\n",
       "  3: {'A': 556.0, 'C': 25.0, 'G': 387.0, 'T': 35.0},\n",
       "  4: {'A': 25.0, 'C': 26.0, 'G': 920.0, 'T': 33.0},\n",
       "  5: {'A': 18.0, 'C': 24.0, 'G': 71.0, 'T': 891.0},\n",
       "  6: {'A': 62.0, 'C': 32.0, 'G': 83.0, 'T': 828.0},\n",
       "  7: {'A': 56.0, 'C': 744.0, 'G': 29.0, 'T': 175.0},\n",
       "  8: {'A': 887.0, 'C': 20.0, 'G': 77.0, 'T': 20.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '8065900',\n",
       "  'uniprot_ids': 'P11473',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0027.2',\n",
       "  'Matrix_Name': 'EN1',\n",
       "  1: {'A': 428.0, 'C': 1175.0, 'G': 766.0, 'T': 526.0},\n",
       "  2: {'A': 105.0, 'C': 1329.0, 'G': 6.0, 'T': 1563.0},\n",
       "  3: {'A': 2892.0, 'C': 100.0, 'G': 72.0, 'T': 1.0},\n",
       "  4: {'A': 2892.0, 'C': 29.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2892.0},\n",
       "  6: {'A': 58.0, 'C': 484.0, 'G': 163.0, 'T': 2892.0},\n",
       "  7: {'A': 2892.0, 'C': 1.0, 'G': 31.0, 'T': 133.0},\n",
       "  8: {'A': 690.0, 'C': 684.0, 'G': 1217.0, 'T': 303.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8096059',\n",
       "  'uniprot_ids': 'Q05925',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0043.3',\n",
       "  'Matrix_Name': 'HLF',\n",
       "  1: {'A': 6811.0, 'C': 5064.0, 'G': 5627.0, 'T': 6926.0},\n",
       "  2: {'A': 6192.0, 'C': 4010.0, 'G': 6298.0, 'T': 7928.0},\n",
       "  3: {'A': 8790.0, 'C': 4433.0, 'G': 9794.0, 'T': 1411.0},\n",
       "  4: {'A': 296.0, 'C': 243.0, 'G': 149.0, 'T': 23740.0},\n",
       "  5: {'A': 163.0, 'C': 97.0, 'G': 896.0, 'T': 23272.0},\n",
       "  6: {'A': 22747.0, 'C': 126.0, 'G': 1345.0, 'T': 210.0},\n",
       "  7: {'A': 198.0, 'C': 1549.0, 'G': 254.0, 'T': 22427.0},\n",
       "  8: {'A': 2419.0, 'C': 113.0, 'G': 21706.0, 'T': 190.0},\n",
       "  9: {'A': 370.0, 'C': 17204.0, 'G': 200.0, 'T': 6654.0},\n",
       "  10: {'A': 24157.0, 'C': 123.0, 'G': 49.0, 'T': 99.0},\n",
       "  11: {'A': 24056.0, 'C': 68.0, 'G': 190.0, 'T': 114.0},\n",
       "  12: {'A': 1400.0, 'C': 13238.0, 'G': 3301.0, 'T': 6489.0},\n",
       "  13: {'A': 8898.0, 'C': 6413.0, 'G': 3219.0, 'T': 5898.0},\n",
       "  14: {'A': 6903.0, 'C': 5268.0, 'G': 5524.0, 'T': 6733.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8065331',\n",
       "  'uniprot_ids': 'Q16534',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1473.1',\n",
       "  'Matrix_Name': 'CDX4',\n",
       "  1: {'A': 3714.0, 'C': 2230.0, 'G': 12157.0, 'T': 5270.0},\n",
       "  2: {'A': 2534.0, 'C': 1065.0, 'G': 23368.0, 'T': 99.0},\n",
       "  3: {'A': 1.0, 'C': 14683.0, 'G': 1.0, 'T': 8687.0},\n",
       "  4: {'A': 12393.0, 'C': 10977.0, 'G': 26.0, 'T': 1.0},\n",
       "  5: {'A': 23368.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 23368.0},\n",
       "  7: {'A': 23368.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 23368.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 23368.0, 'C': 58.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 23368.0, 'C': 6494.0, 'G': 2169.0, 'T': 4828.0},\n",
       "  11: {'A': 2716.0, 'C': 11455.0, 'G': 3759.0, 'T': 5442.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'O14627',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1474.1',\n",
       "  'Matrix_Name': 'CREB3L4',\n",
       "  1: {'A': 877.0, 'C': 1803.0, 'G': 960.0, 'T': 2091.0},\n",
       "  2: {'A': 829.0, 'C': 248.0, 'G': 3892.0, 'T': 1404.0},\n",
       "  3: {'A': 1119.0, 'C': 3892.0, 'G': 166.0, 'T': 247.0},\n",
       "  4: {'A': 261.0, 'C': 3892.0, 'G': 113.0, 'T': 222.0},\n",
       "  5: {'A': 3892.0, 'C': 51.0, 'G': 209.0, 'T': 9.0},\n",
       "  6: {'A': 14.0, 'C': 3892.0, 'G': 21.0, 'T': 131.0},\n",
       "  7: {'A': 307.0, 'C': 66.0, 'G': 3892.0, 'T': 1.0},\n",
       "  8: {'A': 54.0, 'C': 302.0, 'G': 64.0, 'T': 3892.0},\n",
       "  9: {'A': 501.0, 'C': 3892.0, 'G': 691.0, 'T': 280.0},\n",
       "  10: {'A': 3892.0, 'C': 516.0, 'G': 770.0, 'T': 276.0},\n",
       "  11: {'A': 285.0, 'C': 2585.0, 'G': 730.0, 'T': 1309.0},\n",
       "  12: {'A': 665.0, 'C': 1741.0, 'G': 789.0, 'T': 699.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21711675',\n",
       "  'uniprot_ids': 'Q8TEY5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0696.1',\n",
       "  'Matrix_Name': 'ZIC1',\n",
       "  1: {'A': 1114.0, 'C': 3935.0, 'G': 25703.0, 'T': 8961.0},\n",
       "  2: {'A': 22723.0, 'C': 2661.0, 'G': 7409.0, 'T': 24.0},\n",
       "  3: {'A': 837.0, 'C': 28950.0, 'G': 59.0, 'T': 126.0},\n",
       "  4: {'A': 1265.0, 'C': 29218.0, 'G': 205.0, 'T': 282.0},\n",
       "  5: {'A': 2607.0, 'C': 25528.0, 'G': 328.0, 'T': 928.0},\n",
       "  6: {'A': 1.0, 'C': 29400.0, 'G': 5.0, 'T': 8.0},\n",
       "  7: {'A': 186.0, 'C': 29533.0, 'G': 11.0, 'T': 15.0},\n",
       "  8: {'A': 23.0, 'C': 18413.0, 'G': 1.0, 'T': 10314.0},\n",
       "  9: {'A': 2157.0, 'C': 46.0, 'G': 31996.0, 'T': 15.0},\n",
       "  10: {'A': 24.0, 'C': 27427.0, 'G': 440.0, 'T': 6736.0},\n",
       "  11: {'A': 1216.0, 'C': 1758.0, 'G': 5532.0, 'T': 27085.0},\n",
       "  12: {'A': 74.0, 'C': 25.0, 'G': 34650.0, 'T': 2.0},\n",
       "  13: {'A': 3122.0, 'C': 7694.0, 'G': 1691.0, 'T': 16191.0},\n",
       "  14: {'A': 36.0, 'C': 521.0, 'G': 34432.0, 'T': 2726.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q15915',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1568.1',\n",
       "  'Matrix_Name': 'TCF21(var.2)',\n",
       "  1: {'A': 114.0, 'C': 399.0, 'G': 176.0, 'T': 187.0},\n",
       "  2: {'A': 873.0, 'C': 21.0, 'G': 353.0, 'T': 195.0},\n",
       "  3: {'A': 90.0, 'C': 873.0, 'G': 206.0, 'T': 29.0},\n",
       "  4: {'A': 5.0, 'C': 873.0, 'G': 13.0, 'T': 3.0},\n",
       "  5: {'A': 873.0, 'C': 1.0, 'G': 165.0, 'T': 15.0},\n",
       "  6: {'A': 1.0, 'C': 55.0, 'G': 68.0, 'T': 873.0},\n",
       "  7: {'A': 873.0, 'C': 44.0, 'G': 72.0, 'T': 1.0},\n",
       "  8: {'A': 22.0, 'C': 108.0, 'G': 1.0, 'T': 873.0},\n",
       "  9: {'A': 5.0, 'C': 1.0, 'G': 873.0, 'T': 19.0},\n",
       "  10: {'A': 20.0, 'C': 87.0, 'G': 534.0, 'T': 340.0},\n",
       "  11: {'A': 88.0, 'C': 442.0, 'G': 7.0, 'T': 432.0},\n",
       "  12: {'A': 302.0, 'C': 151.0, 'G': 316.0, 'T': 106.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'O43680',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0897.1',\n",
       "  'Matrix_Name': 'Hmx2',\n",
       "  1: {'A': 51.0, 'C': 21.0, 'G': 24.0, 'T': 8.0},\n",
       "  2: {'A': 20.0, 'C': 45.0, 'G': 24.0, 'T': 15.0},\n",
       "  3: {'A': 56.0, 'C': 20.0, 'G': 18.0, 'T': 10.0},\n",
       "  4: {'A': 78.0, 'C': 7.0, 'G': 13.0, 'T': 7.0},\n",
       "  5: {'A': 18.0, 'C': 16.0, 'G': 58.0, 'T': 12.0},\n",
       "  6: {'A': 2.0, 'C': 98.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 86.0, 'C': 1.0, 'G': 16.0, 'T': 1.0},\n",
       "  8: {'A': 86.0, 'C': 14.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 100.0},\n",
       "  10: {'A': 3.0, 'C': 7.0, 'G': 1.0, 'T': 94.0},\n",
       "  11: {'A': 94.0, 'C': 1.0, 'G': 2.0, 'T': 6.0},\n",
       "  12: {'A': 90.0, 'C': 2.0, 'G': 3.0, 'T': 9.0},\n",
       "  13: {'A': 37.0, 'C': 20.0, 'G': 12.0, 'T': 35.0},\n",
       "  14: {'A': 26.0, 'C': 22.0, 'G': 42.0, 'T': 15.0},\n",
       "  15: {'A': 42.0, 'C': 28.0, 'G': 28.0, 'T': 6.0},\n",
       "  16: {'A': 48.0, 'C': 11.0, 'G': 26.0, 'T': 19.0},\n",
       "  17: {'A': 10.0, 'C': 8.0, 'G': 10.0, 'T': 75.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P43687',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA1656.1',\n",
       "  'Matrix_Name': 'ZNF449',\n",
       "  1: {'A': 1810.0, 'C': 2717.0, 'G': 1567.0, 'T': 1570.0},\n",
       "  2: {'A': 2105.0, 'C': 2321.0, 'G': 1922.0, 'T': 1316.0},\n",
       "  3: {'A': 4136.0, 'C': 1760.0, 'G': 1464.0, 'T': 304.0},\n",
       "  4: {'A': 6240.0, 'C': 583.0, 'G': 225.0, 'T': 616.0},\n",
       "  5: {'A': 109.0, 'C': 93.0, 'G': 7369.0, 'T': 93.0},\n",
       "  6: {'A': 140.0, 'C': 7089.0, 'G': 323.0, 'T': 112.0},\n",
       "  7: {'A': 65.0, 'C': 7541.0, 'G': 29.0, 'T': 29.0},\n",
       "  8: {'A': 69.0, 'C': 7480.0, 'G': 55.0, 'T': 60.0},\n",
       "  9: {'A': 6852.0, 'C': 301.0, 'G': 183.0, 'T': 328.0},\n",
       "  10: {'A': 6621.0, 'C': 97.0, 'G': 830.0, 'T': 116.0},\n",
       "  11: {'A': 185.0, 'C': 7245.0, 'G': 168.0, 'T': 66.0},\n",
       "  12: {'A': 1144.0, 'C': 5030.0, 'G': 474.0, 'T': 1016.0},\n",
       "  13: {'A': 2596.0, 'C': 1876.0, 'G': 1586.0, 'T': 1606.0},\n",
       "  14: {'A': 1671.0, 'C': 2196.0, 'G': 2238.0, 'T': 1559.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q6P9G9',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0648.1',\n",
       "  'Matrix_Name': 'GSC',\n",
       "  1: {'A': 1250.0, 'C': 1913.0, 'G': 2040.0, 'T': 883.0},\n",
       "  2: {'A': 956.0, 'C': 2886.0, 'G': 554.0, 'T': 1689.0},\n",
       "  3: {'A': 1.0, 'C': 55.0, 'G': 1.0, 'T': 6082.0},\n",
       "  4: {'A': 6082.0, 'C': 72.0, 'G': 177.0, 'T': 90.0},\n",
       "  5: {'A': 6082.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 26.0, 'G': 145.0, 'T': 6082.0},\n",
       "  7: {'A': 417.0, 'C': 6082.0, 'G': 453.0, 'T': 788.0},\n",
       "  8: {'A': 316.0, 'C': 6082.0, 'G': 829.0, 'T': 1100.0},\n",
       "  9: {'A': 837.0, 'C': 2399.0, 'G': 1733.0, 'T': 1116.0},\n",
       "  10: {'A': 1098.0, 'C': 2198.0, 'G': 620.0, 'T': 2169.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P56915',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0754.1',\n",
       "  'Matrix_Name': 'CUX1',\n",
       "  1: {'A': 4100.0, 'C': 4650.0, 'G': 2402.0, 'T': 25660.0},\n",
       "  2: {'A': 25660.0, 'C': 3297.0, 'G': 15551.0, 'T': 7606.0},\n",
       "  3: {'A': 25660.0, 'C': 110.0, 'G': 193.0, 'T': 168.0},\n",
       "  4: {'A': 72.0, 'C': 295.0, 'G': 78.0, 'T': 25660.0},\n",
       "  5: {'A': 52.0, 'C': 25660.0, 'G': 54.0, 'T': 596.0},\n",
       "  6: {'A': 9581.0, 'C': 265.0, 'G': 16080.0, 'T': 131.0},\n",
       "  7: {'A': 25660.0, 'C': 46.0, 'G': 86.0, 'T': 232.0},\n",
       "  8: {'A': 151.0, 'C': 96.0, 'G': 30.0, 'T': 25660.0},\n",
       "  9: {'A': 25660.0, 'C': 9421.0, 'G': 7561.0, 'T': 7080.0},\n",
       "  10: {'A': 8169.0, 'C': 7634.0, 'G': 3959.0, 'T': 5901.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HD-CUT',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7799919',\n",
       "  'uniprot_ids': 'P39880',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0827.1',\n",
       "  'Matrix_Name': 'OLIG3',\n",
       "  1: {'A': 1916.0, 'C': 205.0, 'G': 428.0, 'T': 133.0},\n",
       "  2: {'A': 801.0, 'C': 1116.0, 'G': 209.0, 'T': 7.0},\n",
       "  3: {'A': 223.0, 'C': 1916.0, 'G': 1.0, 'T': 4.0},\n",
       "  4: {'A': 1916.0, 'C': 11.0, 'G': 100.0, 'T': 23.0},\n",
       "  5: {'A': 1.0, 'C': 27.0, 'G': 73.0, 'T': 1916.0},\n",
       "  6: {'A': 1916.0, 'C': 104.0, 'G': 12.0, 'T': 1.0},\n",
       "  7: {'A': 57.0, 'C': 225.0, 'G': 3.0, 'T': 1916.0},\n",
       "  8: {'A': 5.0, 'C': 5.0, 'G': 1916.0, 'T': 276.0},\n",
       "  9: {'A': 29.0, 'C': 663.0, 'G': 905.0, 'T': 1011.0},\n",
       "  10: {'A': 266.0, 'C': 965.0, 'G': 177.0, 'T': 1916.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20959288',\n",
       "  'uniprot_ids': 'Q7RTU3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1632.1',\n",
       "  'Matrix_Name': 'ATF2',\n",
       "  1: {'A': 20123.0, 'C': 10796.0, 'G': 11911.0, 'T': 16568.0},\n",
       "  2: {'A': 19158.0, 'C': 10293.0, 'G': 18925.0, 'T': 11022.0},\n",
       "  3: {'A': 48583.0, 'C': 3146.0, 'G': 6230.0, 'T': 1439.0},\n",
       "  4: {'A': 517.0, 'C': 900.0, 'G': 329.0, 'T': 57652.0},\n",
       "  5: {'A': 1171.0, 'C': 409.0, 'G': 55319.0, 'T': 2499.0},\n",
       "  6: {'A': 57762.0, 'C': 397.0, 'G': 263.0, 'T': 976.0},\n",
       "  7: {'A': 1052.0, 'C': 16971.0, 'G': 23208.0, 'T': 18167.0},\n",
       "  8: {'A': 1366.0, 'C': 887.0, 'G': 56883.0, 'T': 262.0},\n",
       "  9: {'A': 964.0, 'C': 317.0, 'G': 284.0, 'T': 57833.0},\n",
       "  10: {'A': 4373.0, 'C': 53465.0, 'G': 650.0, 'T': 910.0},\n",
       "  11: {'A': 57900.0, 'C': 219.0, 'G': 769.0, 'T': 510.0},\n",
       "  12: {'A': 2950.0, 'C': 14154.0, 'G': 5972.0, 'T': 36322.0},\n",
       "  13: {'A': 13721.0, 'C': 15786.0, 'G': 11554.0, 'T': 18337.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20581861',\n",
       "  'uniprot_ids': 'P15336',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0727.1',\n",
       "  'Matrix_Name': 'NR3C2',\n",
       "  1: {'A': 8407.0, 'C': 6689.0, 'G': 20194.0, 'T': 12759.0},\n",
       "  2: {'A': 15034.0, 'C': 227.0, 'G': 26932.0, 'T': 536.0},\n",
       "  3: {'A': 58.0, 'C': 9.0, 'G': 55732.0, 'T': 61.0},\n",
       "  4: {'A': 17524.0, 'C': 1378.0, 'G': 6651.0, 'T': 13655.0},\n",
       "  5: {'A': 30043.0, 'C': 16.0, 'G': 14.0, 'T': 10.0},\n",
       "  6: {'A': 1.0, 'C': 49954.0, 'G': 6.0, 'T': 1.0},\n",
       "  7: {'A': 33516.0, 'C': 29.0, 'G': 1054.0, 'T': 222.0},\n",
       "  8: {'A': 7348.0, 'C': 18274.0, 'G': 3145.0, 'T': 16490.0},\n",
       "  9: {'A': 17997.0, 'C': 4834.0, 'G': 8539.0, 'T': 11063.0},\n",
       "  10: {'A': 22921.0, 'C': 1934.0, 'G': 14023.0, 'T': 4024.0},\n",
       "  11: {'A': 402.0, 'C': 997.0, 'G': 23.0, 'T': 43466.0},\n",
       "  12: {'A': 1.0, 'C': 10.0, 'G': 58889.0, 'T': 1.0},\n",
       "  13: {'A': 5.0, 'C': 24.0, 'G': 4.0, 'T': 44514.0},\n",
       "  14: {'A': 13790.0, 'C': 7367.0, 'G': 1347.0, 'T': 18548.0},\n",
       "  15: {'A': 17.0, 'C': 42688.0, 'G': 10.0, 'T': 84.0},\n",
       "  16: {'A': 561.0, 'C': 25681.0, 'G': 583.0, 'T': 16505.0},\n",
       "  17: {'A': 9075.0, 'C': 17398.0, 'G': 7775.0, 'T': 8793.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15563547',\n",
       "  'uniprot_ids': 'P08235',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1121.1',\n",
       "  'Matrix_Name': 'TEAD2',\n",
       "  1: {'A': 138.0, 'C': 146.0, 'G': 113.0, 'T': 149.0},\n",
       "  2: {'A': 89.0, 'C': 239.0, 'G': 82.0, 'T': 136.0},\n",
       "  3: {'A': 442.0, 'C': 18.0, 'G': 71.0, 'T': 15.0},\n",
       "  4: {'A': 40.0, 'C': 479.0, 'G': 15.0, 'T': 12.0},\n",
       "  5: {'A': 518.0, 'C': 6.0, 'G': 12.0, 'T': 10.0},\n",
       "  6: {'A': 3.0, 'C': 11.0, 'G': 7.0, 'T': 525.0},\n",
       "  7: {'A': 16.0, 'C': 9.0, 'G': 10.0, 'T': 511.0},\n",
       "  8: {'A': 9.0, 'C': 501.0, 'G': 11.0, 'T': 25.0},\n",
       "  9: {'A': 9.0, 'C': 514.0, 'G': 4.0, 'T': 19.0},\n",
       "  10: {'A': 264.0, 'C': 61.0, 'G': 57.0, 'T': 164.0},\n",
       "  11: {'A': 113.0, 'C': 112.0, 'G': 208.0, 'T': 113.0},\n",
       "  12: {'A': 112.0, 'C': 173.0, 'G': 148.0, 'T': 113.0},\n",
       "  13: {'A': 110.0, 'C': 222.0, 'G': 118.0, 'T': 96.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'TEF-1-related',\n",
       "  'tf_class': 'TEA',\n",
       "  'pubmed_ids': '9571041',\n",
       "  'uniprot_ids': 'Q15562',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0862.1',\n",
       "  'Matrix_Name': 'GMEB2',\n",
       "  1: {'A': 35.0, 'C': 76.0, 'G': 110.0, 'T': 240.0},\n",
       "  2: {'A': 26.0, 'C': 11.0, 'G': 87.0, 'T': 240.0},\n",
       "  3: {'A': 240.0, 'C': 1.0, 'G': 6.0, 'T': 1.0},\n",
       "  4: {'A': 5.0, 'C': 240.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 5.0, 'G': 240.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 22.0, 'G': 1.0, 'T': 240.0},\n",
       "  7: {'A': 240.0, 'C': 67.0, 'G': 13.0, 'T': 10.0},\n",
       "  8: {'A': 240.0, 'C': 85.0, 'G': 25.0, 'T': 6.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'GMEB',\n",
       "  'tf_class': 'SAND',\n",
       "  'pubmed_ids': '11743720',\n",
       "  'uniprot_ids': 'Q9UKD1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0900.2',\n",
       "  'Matrix_Name': 'HOXA2',\n",
       "  1: {'A': 8881.0, 'C': 19453.0, 'G': 22687.0, 'T': 5932.0},\n",
       "  2: {'A': 1.0, 'C': 3637.0, 'G': 1.0, 'T': 56950.0},\n",
       "  3: {'A': 39675.0, 'C': 17276.0, 'G': 291.0, 'T': 1.0},\n",
       "  4: {'A': 56950.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 56950.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 56950.0},\n",
       "  7: {'A': 56950.0, 'C': 1.0, 'G': 2053.0, 'T': 1.0},\n",
       "  8: {'A': 10642.0, 'C': 16331.0, 'G': 22247.0, 'T': 7732.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'O43364',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0076.2',\n",
       "  'Matrix_Name': 'ELK4',\n",
       "  1: {'A': 299.0, 'C': 1459.0, 'G': 887.0, 'T': 786.0},\n",
       "  2: {'A': 392.0, 'C': 1906.0, 'G': 571.0, 'T': 562.0},\n",
       "  3: {'A': 2035.0, 'C': 146.0, 'G': 1039.0, 'T': 211.0},\n",
       "  4: {'A': 1.0, 'C': 2689.0, 'G': 17.0, 'T': 724.0},\n",
       "  5: {'A': 9.0, 'C': 1.0, 'G': 1.0, 'T': 3420.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3428.0},\n",
       "  7: {'A': 1.0, 'C': 3428.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 3428.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 2916.0, 'T': 513.0},\n",
       "  10: {'A': 1.0, 'C': 502.0, 'G': 2844.0, 'T': 84.0},\n",
       "  11: {'A': 291.0, 'C': 1254.0, 'G': 669.0, 'T': 1217.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '8524663',\n",
       "  'uniprot_ids': 'P28324',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0031.1',\n",
       "  'Matrix_Name': 'FOXD1',\n",
       "  1: {'A': 2.0, 'C': 2.0, 'G': 18.0, 'T': 2.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 21.0},\n",
       "  3: {'A': 20.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 21.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 19.0, 'C': 2.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 2.0, 'C': 19.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 21.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 8.0, 'C': 3.0, 'G': 4.0, 'T': 9.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '7957066',\n",
       "  'uniprot_ids': 'Q16676',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0163.1',\n",
       "  'Matrix_Name': 'PLAG1',\n",
       "  1: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  2: {'A': 4.0, 'C': 1.0, 'G': 15.0, 'T': 2.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 18.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 15.0, 'G': 5.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 16.0, 'G': 2.0, 'T': 3.0},\n",
       "  7: {'A': 5.0, 'C': 11.0, 'G': 2.0, 'T': 4.0},\n",
       "  8: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 7.0},\n",
       "  9: {'A': 12.0, 'C': 6.0, 'G': 3.0, 'T': 1.0},\n",
       "  10: {'A': 3.0, 'C': 1.0, 'G': 15.0, 'T': 3.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 17.0, 'T': 3.0},\n",
       "  14: {'A': 3.0, 'C': 1.0, 'G': 17.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16041365',\n",
       "  'uniprot_ids': 'Q6DJT9',\n",
       "  'data_type': 'bacterial'},\n",
       " {'Matrix_ID': 'MA1109.1',\n",
       "  'Matrix_Name': 'NEUROD1',\n",
       "  1: {'A': 562.0, 'C': 430.0, 'G': 727.0, 'T': 567.0},\n",
       "  2: {'A': 756.0, 'C': 324.0, 'G': 907.0, 'T': 299.0},\n",
       "  3: {'A': 1290.0, 'C': 453.0, 'G': 495.0, 'T': 48.0},\n",
       "  4: {'A': 18.0, 'C': 2241.0, 'G': 14.0, 'T': 13.0},\n",
       "  5: {'A': 2242.0, 'C': 8.0, 'G': 17.0, 'T': 19.0},\n",
       "  6: {'A': 17.0, 'C': 25.0, 'G': 2153.0, 'T': 91.0},\n",
       "  7: {'A': 2035.0, 'C': 151.0, 'G': 78.0, 'T': 22.0},\n",
       "  8: {'A': 56.0, 'C': 52.0, 'G': 20.0, 'T': 2158.0},\n",
       "  9: {'A': 14.0, 'C': 8.0, 'G': 2232.0, 'T': 32.0},\n",
       "  10: {'A': 60.0, 'C': 94.0, 'G': 1963.0, 'T': 169.0},\n",
       "  11: {'A': 374.0, 'C': 912.0, 'G': 419.0, 'T': 581.0},\n",
       "  12: {'A': 693.0, 'C': 541.0, 'G': 600.0, 'T': 452.0},\n",
       "  13: {'A': 539.0, 'C': 592.0, 'G': 629.0, 'T': 526.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9858593',\n",
       "  'uniprot_ids': 'Q13562',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0721.1',\n",
       "  'Matrix_Name': 'UNCX',\n",
       "  1: {'A': 4981.0, 'C': 13364.0, 'G': 5712.0, 'T': 12872.0},\n",
       "  2: {'A': 3976.0, 'C': 16801.0, 'G': 2087.0, 'T': 20127.0},\n",
       "  3: {'A': 36927.0, 'C': 2541.0, 'G': 4583.0, 'T': 2630.0},\n",
       "  4: {'A': 36927.0, 'C': 1303.0, 'G': 421.0, 'T': 891.0},\n",
       "  5: {'A': 482.0, 'C': 807.0, 'G': 118.0, 'T': 36927.0},\n",
       "  6: {'A': 3112.0, 'C': 3959.0, 'G': 1783.0, 'T': 36927.0},\n",
       "  7: {'A': 36927.0, 'C': 2212.0, 'G': 732.0, 'T': 3886.0},\n",
       "  8: {'A': 13812.0, 'C': 7826.0, 'G': 10481.0, 'T': 4811.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'A6NJT0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0656.1',\n",
       "  'Matrix_Name': 'JDP2(var.2)',\n",
       "  1: {'A': 4261.0, 'C': 5484.0, 'G': 11037.0, 'T': 3624.0},\n",
       "  2: {'A': 24404.0, 'C': 1320.0, 'G': 1404.0, 'T': 18.0},\n",
       "  3: {'A': 4.0, 'C': 10.0, 'G': 4.0, 'T': 24404.0},\n",
       "  4: {'A': 12.0, 'C': 1.0, 'G': 24404.0, 'T': 1940.0},\n",
       "  5: {'A': 24404.0, 'C': 8.0, 'G': 7.0, 'T': 7.0},\n",
       "  6: {'A': 6.0, 'C': 24404.0, 'G': 7.0, 'T': 339.0},\n",
       "  7: {'A': 588.0, 'C': 6.0, 'G': 24404.0, 'T': 3.0},\n",
       "  8: {'A': 16.0, 'C': 25.0, 'G': 8.0, 'T': 24404.0},\n",
       "  9: {'A': 1778.0, 'C': 24404.0, 'G': 5.0, 'T': 2.0},\n",
       "  10: {'A': 24404.0, 'C': 11.0, 'G': 10.0, 'T': 1.0},\n",
       "  11: {'A': 48.0, 'C': 12120.0, 'G': 836.0, 'T': 12284.0},\n",
       "  12: {'A': 4384.0, 'C': 12118.0, 'G': 4125.0, 'T': 3779.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9154808',\n",
       "  'uniprot_ids': 'Q8WYK2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1604.1',\n",
       "  'Matrix_Name': 'Ebf2',\n",
       "  1: {'A': 4672.0, 'C': 5871.0, 'G': 7679.0, 'T': 8043.0},\n",
       "  2: {'A': 3212.0, 'C': 7197.0, 'G': 3220.0, 'T': 12636.0},\n",
       "  3: {'A': 197.0, 'C': 25498.0, 'G': 377.0, 'T': 193.0},\n",
       "  4: {'A': 514.0, 'C': 24253.0, 'G': 260.0, 'T': 1238.0},\n",
       "  5: {'A': 436.0, 'C': 25138.0, 'G': 243.0, 'T': 448.0},\n",
       "  6: {'A': 19236.0, 'C': 3745.0, 'G': 1428.0, 'T': 1856.0},\n",
       "  7: {'A': 19156.0, 'C': 1348.0, 'G': 3636.0, 'T': 2125.0},\n",
       "  8: {'A': 738.0, 'C': 240.0, 'G': 25104.0, 'T': 183.0},\n",
       "  9: {'A': 1277.0, 'C': 221.0, 'G': 24387.0, 'T': 380.0},\n",
       "  10: {'A': 972.0, 'C': 581.0, 'G': 24125.0, 'T': 587.0},\n",
       "  11: {'A': 21803.0, 'C': 1438.0, 'G': 1702.0, 'T': 1322.0},\n",
       "  12: {'A': 7077.0, 'C': 7669.0, 'G': 6898.0, 'T': 4621.0},\n",
       "  13: {'A': 6668.0, 'C': 7159.0, 'G': 4834.0, 'T': 7604.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Early',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '23499423',\n",
       "  'uniprot_ids': 'O08792',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0811.1',\n",
       "  'Matrix_Name': 'TFAP2B',\n",
       "  1: {'A': 717.0, 'C': 764.0, 'G': 181.0, 'T': 1521.0},\n",
       "  2: {'A': 673.0, 'C': 1007.0, 'G': 3180.0, 'T': 103.0},\n",
       "  3: {'A': 24.0, 'C': 3180.0, 'G': 262.0, 'T': 24.0},\n",
       "  4: {'A': 2.0, 'C': 3180.0, 'G': 1.0, 'T': 35.0},\n",
       "  5: {'A': 12.0, 'C': 2055.0, 'G': 153.0, 'T': 964.0},\n",
       "  6: {'A': 210.0, 'C': 1200.0, 'G': 680.0, 'T': 1093.0},\n",
       "  7: {'A': 1298.0, 'C': 816.0, 'G': 889.0, 'T': 180.0},\n",
       "  8: {'A': 1275.0, 'C': 214.0, 'G': 1694.0, 'T': 1.0},\n",
       "  9: {'A': 84.0, 'C': 18.0, 'G': 3180.0, 'T': 1.0},\n",
       "  10: {'A': 38.0, 'C': 418.0, 'G': 3180.0, 'T': 13.0},\n",
       "  11: {'A': 93.0, 'C': 3180.0, 'G': 485.0, 'T': 240.0},\n",
       "  12: {'A': 1843.0, 'C': 190.0, 'G': 671.0, 'T': 480.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'Q92481',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0733.1',\n",
       "  'Matrix_Name': 'EGR4',\n",
       "  1: {'A': 1168.0, 'C': 1460.0, 'G': 1005.0, 'T': 1954.0},\n",
       "  2: {'A': 1144.0, 'C': 1665.0, 'G': 528.0, 'T': 2132.0},\n",
       "  3: {'A': 2746.0, 'C': 1925.0, 'G': 152.0, 'T': 159.0},\n",
       "  4: {'A': 102.0, 'C': 4467.0, 'G': 151.0, 'T': 320.0},\n",
       "  5: {'A': 557.0, 'C': 476.0, 'G': 7054.0, 'T': 708.0},\n",
       "  6: {'A': 12.0, 'C': 4448.0, 'G': 64.0, 'T': 89.0},\n",
       "  7: {'A': 121.0, 'C': 4449.0, 'G': 108.0, 'T': 77.0},\n",
       "  8: {'A': 1.0, 'C': 4404.0, 'G': 111.0, 'T': 132.0},\n",
       "  9: {'A': 4501.0, 'C': 1056.0, 'G': 349.0, 'T': 364.0},\n",
       "  10: {'A': 45.0, 'C': 4403.0, 'G': 115.0, 'T': 246.0},\n",
       "  11: {'A': 234.0, 'C': 1794.0, 'G': 6863.0, 'T': 254.0},\n",
       "  12: {'A': 11.0, 'C': 4355.0, 'G': 184.0, 'T': 220.0},\n",
       "  13: {'A': 4609.0, 'C': 840.0, 'G': 830.0, 'T': 484.0},\n",
       "  14: {'A': 1722.0, 'C': 1400.0, 'G': 410.0, 'T': 1779.0},\n",
       "  15: {'A': 1484.0, 'C': 820.0, 'G': 656.0, 'T': 2763.0},\n",
       "  16: {'A': 1192.0, 'C': 1119.0, 'G': 936.0, 'T': 2423.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7891721',\n",
       "  'uniprot_ids': 'Q05215',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1524.1',\n",
       "  'Matrix_Name': 'MSGN1',\n",
       "  1: {'A': 1191.0, 'C': 2081.0, 'G': 1616.0, 'T': 1489.0},\n",
       "  2: {'A': 3097.0, 'C': 577.0, 'G': 3278.0, 'T': 409.0},\n",
       "  3: {'A': 664.0, 'C': 6374.0, 'G': 1614.0, 'T': 727.0},\n",
       "  4: {'A': 1.0, 'C': 6374.0, 'G': 18.0, 'T': 1.0},\n",
       "  5: {'A': 6374.0, 'C': 1.0, 'G': 375.0, 'T': 1.0},\n",
       "  6: {'A': 2425.0, 'C': 628.0, 'G': 113.0, 'T': 3951.0},\n",
       "  7: {'A': 3767.0, 'C': 191.0, 'G': 815.0, 'T': 2608.0},\n",
       "  8: {'A': 11.0, 'C': 288.0, 'G': 18.0, 'T': 6374.0},\n",
       "  9: {'A': 1.0, 'C': 11.0, 'G': 6374.0, 'T': 1.0},\n",
       "  10: {'A': 413.0, 'C': 1133.0, 'G': 3105.0, 'T': 3270.0},\n",
       "  11: {'A': 515.0, 'C': 3270.0, 'G': 1504.0, 'T': 3105.0},\n",
       "  12: {'A': 1389.0, 'C': 1977.0, 'G': 1782.0, 'T': 1229.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21750544',\n",
       "  'uniprot_ids': 'A6NI15',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0006.1',\n",
       "  'Matrix_Name': 'Ahr::Arnt',\n",
       "  1: {'A': 4.0, 'C': 9.0, 'G': 3.0, 'T': 12.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 24.0, 'T': 2.0},\n",
       "  3: {'A': 1.0, 'C': 24.0, 'G': 1.0, 'T': 2.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 24.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 25.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 25.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 6,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '7592839',\n",
       "  'uniprot_ids': 'P30561;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0029.1',\n",
       "  'Matrix_Name': 'Mecom',\n",
       "  1: {'A': 15.0, 'C': 3.0, 'G': 7.0, 'T': 6.0},\n",
       "  2: {'A': 21.0, 'C': 2.0, 'G': 3.0, 'T': 5.0},\n",
       "  3: {'A': 1.0, 'C': 2.0, 'G': 26.0, 'T': 2.0},\n",
       "  4: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 2.0, 'C': 11.0, 'G': 1.0, 'T': 17.0},\n",
       "  6: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 27.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 28.0, 'T': 1.0},\n",
       "  9: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 4.0, 'G': 1.0, 'T': 25.0},\n",
       "  11: {'A': 25.0, 'C': 2.0, 'G': 1.0, 'T': 3.0},\n",
       "  12: {'A': 24.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  13: {'A': 7.0, 'C': 8.0, 'G': 8.0, 'T': 8.0},\n",
       "  14: {'A': 16.0, 'C': 7.0, 'G': 4.0, 'T': 4.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '8321231',\n",
       "  'uniprot_ids': 'A4QPC8',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0158.2',\n",
       "  'Matrix_Name': 'HOXA5',\n",
       "  1: {'A': 143.0, 'C': 833.0, 'G': 1760.0, 'T': 398.0},\n",
       "  2: {'A': 1.0, 'C': 1053.0, 'G': 1.0, 'T': 2079.0},\n",
       "  3: {'A': 3131.0, 'C': 1633.0, 'G': 82.0, 'T': 1.0},\n",
       "  4: {'A': 3131.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3131.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 886.0, 'T': 3131.0},\n",
       "  7: {'A': 3131.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 481.0, 'C': 1464.0, 'G': 1134.0, 'T': 55.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P20719',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0122.3',\n",
       "  'Matrix_Name': 'Nkx3-2',\n",
       "  1: {'A': 3888.0, 'C': 1475.0, 'G': 1737.0, 'T': 3124.0},\n",
       "  2: {'A': 3422.0, 'C': 1615.0, 'G': 1728.0, 'T': 3459.0},\n",
       "  3: {'A': 5715.0, 'C': 953.0, 'G': 1467.0, 'T': 2089.0},\n",
       "  4: {'A': 5599.0, 'C': 1110.0, 'G': 1155.0, 'T': 2360.0},\n",
       "  5: {'A': 2605.0, 'C': 5080.0, 'G': 1505.0, 'T': 1034.0},\n",
       "  6: {'A': 80.0, 'C': 9901.0, 'G': 24.0, 'T': 219.0},\n",
       "  7: {'A': 9908.0, 'C': 83.0, 'G': 20.0, 'T': 213.0},\n",
       "  8: {'A': 152.0, 'C': 9966.0, 'G': 11.0, 'T': 95.0},\n",
       "  9: {'A': 51.0, 'C': 33.0, 'G': 30.0, 'T': 10110.0},\n",
       "  10: {'A': 96.0, 'C': 139.0, 'G': 69.0, 'T': 9920.0},\n",
       "  11: {'A': 9681.0, 'C': 148.0, 'G': 69.0, 'T': 326.0},\n",
       "  12: {'A': 5621.0, 'C': 992.0, 'G': 1369.0, 'T': 2242.0},\n",
       "  13: {'A': 3398.0, 'C': 2084.0, 'G': 1851.0, 'T': 2891.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P97503',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1588.1',\n",
       "  'Matrix_Name': 'ZNF136',\n",
       "  1: {'A': 1130.0, 'C': 143.0, 'G': 1821.0, 'T': 111.0},\n",
       "  2: {'A': 1629.0, 'C': 152.0, 'G': 287.0, 'T': 1137.0},\n",
       "  3: {'A': 3079.0, 'C': 30.0, 'G': 70.0, 'T': 26.0},\n",
       "  4: {'A': 39.0, 'C': 35.0, 'G': 35.0, 'T': 3096.0},\n",
       "  5: {'A': 28.0, 'C': 70.0, 'G': 28.0, 'T': 3079.0},\n",
       "  6: {'A': 58.0, 'C': 2911.0, 'G': 32.0, 'T': 204.0},\n",
       "  7: {'A': 40.0, 'C': 84.0, 'G': 47.0, 'T': 3034.0},\n",
       "  8: {'A': 434.0, 'C': 94.0, 'G': 512.0, 'T': 2165.0},\n",
       "  9: {'A': 288.0, 'C': 18.0, 'G': 2800.0, 'T': 99.0},\n",
       "  10: {'A': 111.0, 'C': 49.0, 'G': 2996.0, 'T': 49.0},\n",
       "  11: {'A': 158.0, 'C': 673.0, 'G': 34.0, 'T': 2340.0},\n",
       "  12: {'A': 84.0, 'C': 61.0, 'G': 103.0, 'T': 2957.0},\n",
       "  13: {'A': 166.0, 'C': 41.0, 'G': 2905.0, 'T': 93.0},\n",
       "  14: {'A': 1890.0, 'C': 121.0, 'G': 1049.0, 'T': 145.0},\n",
       "  15: {'A': 576.0, 'C': 1974.0, 'G': 83.0, 'T': 572.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '27852650',\n",
       "  'uniprot_ids': 'P52737',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0084.1',\n",
       "  'Matrix_Name': 'SRY',\n",
       "  1: {'A': 6.0, 'C': 6.0, 'G': 11.0, 'T': 9.0},\n",
       "  2: {'A': 9.0, 'C': 4.0, 'G': 5.0, 'T': 14.0},\n",
       "  3: {'A': 16.0, 'C': 1.0, 'G': 4.0, 'T': 11.0},\n",
       "  4: {'A': 19.0, 'C': 4.0, 'G': 4.0, 'T': 5.0},\n",
       "  5: {'A': 26.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  6: {'A': 1.0, 'C': 27.0, 'G': 1.0, 'T': 3.0},\n",
       "  7: {'A': 29.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 28.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  9: {'A': 8.0, 'C': 1.0, 'G': 3.0, 'T': 20.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '8190643',\n",
       "  'uniprot_ids': 'Q05066',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1132.1',\n",
       "  'Matrix_Name': 'JUN::JUNB',\n",
       "  1: {'A': 204.0, 'C': 129.0, 'G': 409.0, 'T': 262.0},\n",
       "  2: {'A': 628.0, 'C': 135.0, 'G': 190.0, 'T': 51.0},\n",
       "  3: {'A': 12.0, 'C': 11.0, 'G': 35.0, 'T': 947.0},\n",
       "  4: {'A': 11.0, 'C': 22.0, 'G': 922.0, 'T': 48.0},\n",
       "  5: {'A': 945.0, 'C': 11.0, 'G': 29.0, 'T': 19.0},\n",
       "  6: {'A': 28.0, 'C': 827.0, 'G': 116.0, 'T': 34.0},\n",
       "  7: {'A': 59.0, 'C': 25.0, 'G': 354.0, 'T': 566.0},\n",
       "  8: {'A': 162.0, 'C': 708.0, 'G': 11.0, 'T': 123.0},\n",
       "  9: {'A': 932.0, 'C': 25.0, 'G': 19.0, 'T': 28.0},\n",
       "  10: {'A': 189.0, 'C': 249.0, 'G': 65.0, 'T': 501.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15240010',\n",
       "  'uniprot_ids': 'P05412;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0800.1',\n",
       "  'Matrix_Name': 'EOMES',\n",
       "  1: {'A': 4527.0, 'C': 912.0, 'G': 2275.0, 'T': 2123.0},\n",
       "  2: {'A': 9833.0, 'C': 159.0, 'G': 1260.0, 'T': 654.0},\n",
       "  3: {'A': 1776.0, 'C': 1005.0, 'G': 9833.0, 'T': 1306.0},\n",
       "  4: {'A': 24.0, 'C': 93.0, 'G': 9833.0, 'T': 95.0},\n",
       "  5: {'A': 20.0, 'C': 601.0, 'G': 51.0, 'T': 9833.0},\n",
       "  6: {'A': 10.0, 'C': 9.0, 'G': 9833.0, 'T': 4.0},\n",
       "  7: {'A': 496.0, 'C': 1541.0, 'G': 99.0, 'T': 9833.0},\n",
       "  8: {'A': 211.0, 'C': 551.0, 'G': 9833.0, 'T': 981.0},\n",
       "  9: {'A': 9833.0, 'C': 29.0, 'G': 100.0, 'T': 25.0},\n",
       "  10: {'A': 9833.0, 'C': 1957.0, 'G': 1023.0, 'T': 3352.0},\n",
       "  11: {'A': 9833.0, 'C': 4338.0, 'G': 2942.0, 'T': 3782.0},\n",
       "  12: {'A': 5650.0, 'C': 1961.0, 'G': 1967.0, 'T': 4184.0},\n",
       "  13: {'A': 2502.0, 'C': 2578.0, 'G': 1955.0, 'T': 2801.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'TBrain-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'O95936',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0081.2',\n",
       "  'Matrix_Name': 'SPIB',\n",
       "  1: {'A': 5348.0, 'C': 5798.0, 'G': 4027.0, 'T': 11761.0},\n",
       "  2: {'A': 4683.0, 'C': 8294.0, 'G': 4545.0, 'T': 9412.0},\n",
       "  3: {'A': 2180.0, 'C': 6430.0, 'G': 3194.0, 'T': 15130.0},\n",
       "  4: {'A': 380.0, 'C': 23972.0, 'G': 1722.0, 'T': 860.0},\n",
       "  5: {'A': 24320.0, 'C': 620.0, 'G': 670.0, 'T': 1324.0},\n",
       "  6: {'A': 240.0, 'C': 22731.0, 'G': 2517.0, 'T': 1446.0},\n",
       "  7: {'A': 451.0, 'C': 237.0, 'G': 158.0, 'T': 26088.0},\n",
       "  8: {'A': 191.0, 'C': 201.0, 'G': 244.0, 'T': 26298.0},\n",
       "  9: {'A': 207.0, 'C': 26146.0, 'G': 154.0, 'T': 427.0},\n",
       "  10: {'A': 276.0, 'C': 25493.0, 'G': 130.0, 'T': 1035.0},\n",
       "  11: {'A': 866.0, 'C': 4518.0, 'G': 1830.0, 'T': 19720.0},\n",
       "  12: {'A': 1021.0, 'C': 20114.0, 'G': 2317.0, 'T': 3482.0},\n",
       "  13: {'A': 4964.0, 'C': 3923.0, 'G': 782.0, 'T': 17265.0},\n",
       "  14: {'A': 2839.0, 'C': 5426.0, 'G': 2088.0, 'T': 16581.0},\n",
       "  15: {'A': 4160.0, 'C': 6030.0, 'G': 2410.0, 'T': 14334.0},\n",
       "  16: {'A': 4027.0, 'C': 8590.0, 'G': 4166.0, 'T': 10151.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '7624145',\n",
       "  'uniprot_ids': 'Q01892',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0153.2',\n",
       "  'Matrix_Name': 'HNF1B',\n",
       "  1: {'A': 1529.0, 'C': 1031.0, 'G': 30840.0, 'T': 1629.0},\n",
       "  2: {'A': 542.0, 'C': 2553.0, 'G': 1435.0, 'T': 30840.0},\n",
       "  3: {'A': 5731.0, 'C': 3467.0, 'G': 779.0, 'T': 30840.0},\n",
       "  4: {'A': 30840.0, 'C': 14.0, 'G': 399.0, 'T': 50.0},\n",
       "  5: {'A': 30840.0, 'C': 1164.0, 'G': 141.0, 'T': 613.0},\n",
       "  6: {'A': 1945.0, 'C': 810.0, 'G': 67.0, 'T': 30840.0},\n",
       "  7: {'A': 6503.0, 'C': 9237.0, 'G': 9627.0, 'T': 5475.0},\n",
       "  8: {'A': 30840.0, 'C': 74.0, 'G': 601.0, 'T': 1048.0},\n",
       "  9: {'A': 1324.0, 'C': 417.0, 'G': 1590.0, 'T': 30840.0},\n",
       "  10: {'A': 81.0, 'C': 666.0, 'G': 2.0, 'T': 30840.0},\n",
       "  11: {'A': 30840.0, 'C': 795.0, 'G': 5272.0, 'T': 3120.0},\n",
       "  12: {'A': 30840.0, 'C': 1473.0, 'G': 1981.0, 'T': 407.0},\n",
       "  13: {'A': 1791.0, 'C': 17115.0, 'G': 726.0, 'T': 13725.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P35680',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0790.1',\n",
       "  'Matrix_Name': 'POU4F1',\n",
       "  1: {'A': 2419.0, 'C': 1036.0, 'G': 1086.0, 'T': 950.0},\n",
       "  2: {'A': 154.0, 'C': 237.0, 'G': 126.0, 'T': 3554.0},\n",
       "  3: {'A': 1033.0, 'C': 377.0, 'G': 3858.0, 'T': 751.0},\n",
       "  4: {'A': 1721.0, 'C': 1662.0, 'G': 30.0, 'T': 77.0},\n",
       "  5: {'A': 2951.0, 'C': 48.0, 'G': 26.0, 'T': 140.0},\n",
       "  6: {'A': 92.0, 'C': 16.0, 'G': 16.0, 'T': 3512.0},\n",
       "  7: {'A': 2812.0, 'C': 103.0, 'G': 146.0, 'T': 1317.0},\n",
       "  8: {'A': 2916.0, 'C': 125.0, 'G': 145.0, 'T': 639.0},\n",
       "  9: {'A': 466.0, 'C': 107.0, 'G': 130.0, 'T': 3202.0},\n",
       "  10: {'A': 391.0, 'C': 73.0, 'G': 263.0, 'T': 3219.0},\n",
       "  11: {'A': 2826.0, 'C': 669.0, 'G': 697.0, 'T': 1117.0},\n",
       "  12: {'A': 3182.0, 'C': 37.0, 'G': 55.0, 'T': 16.0},\n",
       "  13: {'A': 277.0, 'C': 360.0, 'G': 221.0, 'T': 3103.0},\n",
       "  14: {'A': 1055.0, 'C': 653.0, 'G': 3455.0, 'T': 1577.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18077589',\n",
       "  'uniprot_ids': 'Q01851',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1468.1',\n",
       "  'Matrix_Name': 'ATOH7',\n",
       "  1: {'A': 713.0, 'C': 98.0, 'G': 143.0, 'T': 63.0},\n",
       "  2: {'A': 303.0, 'C': 212.0, 'G': 155.0, 'T': 46.0},\n",
       "  3: {'A': 1.0, 'C': 713.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 713.0, 'C': 1.0, 'G': 103.0, 'T': 59.0},\n",
       "  5: {'A': 16.0, 'C': 107.0, 'G': 60.0, 'T': 713.0},\n",
       "  6: {'A': 713.0, 'C': 181.0, 'G': 79.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 26.0, 'G': 26.0, 'T': 713.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 713.0, 'T': 1.0},\n",
       "  9: {'A': 6.0, 'C': 156.0, 'G': 262.0, 'T': 292.0},\n",
       "  10: {'A': 18.0, 'C': 401.0, 'G': 3.0, 'T': 313.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23293286',\n",
       "  'uniprot_ids': 'Q8N100',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1101.2',\n",
       "  'Matrix_Name': 'BACH2',\n",
       "  1: {'A': 26747.0, 'C': 12204.0, 'G': 13721.0, 'T': 15602.0},\n",
       "  2: {'A': 38328.0, 'C': 8824.0, 'G': 9606.0, 'T': 24470.0},\n",
       "  3: {'A': 59308.0, 'C': 5535.0, 'G': 8491.0, 'T': 15406.0},\n",
       "  4: {'A': 12405.0, 'C': 21219.0, 'G': 21258.0, 'T': 13392.0},\n",
       "  5: {'A': 9370.0, 'C': 61581.0, 'G': 6728.0, 'T': 8117.0},\n",
       "  6: {'A': 62994.0, 'C': 261.0, 'G': 15305.0, 'T': 1.0},\n",
       "  7: {'A': 2.0, 'C': 8.0, 'G': 2.0, 'T': 68267.0},\n",
       "  8: {'A': 7.0, 'C': 1.0, 'G': 68241.0, 'T': 55.0},\n",
       "  9: {'A': 67847.0, 'C': 1089.0, 'G': 14.0, 'T': 7.0},\n",
       "  10: {'A': 51.0, 'C': 37439.0, 'G': 30786.0, 'T': 19.0},\n",
       "  11: {'A': 4.0, 'C': 21.0, 'G': 857.0, 'T': 67976.0},\n",
       "  12: {'A': 116.0, 'C': 68216.0, 'G': 1.0, 'T': 4.0},\n",
       "  13: {'A': 68265.0, 'C': 3.0, 'G': 12.0, 'T': 1.0},\n",
       "  14: {'A': 48.0, 'C': 11635.0, 'G': 2029.0, 'T': 63747.0},\n",
       "  15: {'A': 4523.0, 'C': 35221.0, 'G': 30499.0, 'T': 5275.0},\n",
       "  16: {'A': 19651.0, 'C': 17683.0, 'G': 19156.0, 'T': 11784.0},\n",
       "  17: {'A': 18458.0, 'C': 13439.0, 'G': 7651.0, 'T': 54750.0},\n",
       "  18: {'A': 23587.0, 'C': 10704.0, 'G': 8222.0, 'T': 39013.0},\n",
       "  19: {'A': 15262.0, 'C': 15624.0, 'G': 11269.0, 'T': 26119.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 19,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '28461702',\n",
       "  'uniprot_ids': 'Q9BYV9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0904.2',\n",
       "  'Matrix_Name': 'HOXB5',\n",
       "  1: {'A': 382.0, 'C': 1121.0, 'G': 1274.0, 'T': 295.0},\n",
       "  2: {'A': 1.0, 'C': 538.0, 'G': 1.0, 'T': 3068.0},\n",
       "  3: {'A': 3068.0, 'C': 1032.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 3068.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3068.0},\n",
       "  6: {'A': 7.0, 'C': 27.0, 'G': 1062.0, 'T': 3068.0},\n",
       "  7: {'A': 3068.0, 'C': 1.0, 'G': 565.0, 'T': 1.0},\n",
       "  8: {'A': 254.0, 'C': 1220.0, 'G': 1147.0, 'T': 451.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P09067',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1558.1',\n",
       "  'Matrix_Name': 'SNAI1',\n",
       "  1: {'A': 5275.0, 'C': 3188.0, 'G': 6411.0, 'T': 5113.0},\n",
       "  2: {'A': 6994.0, 'C': 502.0, 'G': 12990.0, 'T': 3519.0},\n",
       "  3: {'A': 447.0, 'C': 19983.0, 'G': 276.0, 'T': 341.0},\n",
       "  4: {'A': 19983.0, 'C': 298.0, 'G': 92.0, 'T': 569.0},\n",
       "  5: {'A': 1767.0, 'C': 518.0, 'G': 19983.0, 'T': 307.0},\n",
       "  6: {'A': 73.0, 'C': 6.0, 'G': 19983.0, 'T': 29.0},\n",
       "  7: {'A': 39.0, 'C': 1.0, 'G': 43.0, 'T': 19983.0},\n",
       "  8: {'A': 707.0, 'C': 1.0, 'G': 19983.0, 'T': 312.0},\n",
       "  9: {'A': 1640.0, 'C': 10939.0, 'G': 5307.0, 'T': 9045.0},\n",
       "  10: {'A': 10089.0, 'C': 4593.0, 'G': 9895.0, 'T': 6357.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '26170144',\n",
       "  'uniprot_ids': 'O95863',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0854.1',\n",
       "  'Matrix_Name': 'Alx1',\n",
       "  1: {'A': 20.0, 'C': 45.0, 'G': 18.0, 'T': 21.0},\n",
       "  2: {'A': 8.0, 'C': 22.0, 'G': 66.0, 'T': 8.0},\n",
       "  3: {'A': 39.0, 'C': 34.0, 'G': 15.0, 'T': 16.0},\n",
       "  4: {'A': 44.0, 'C': 10.0, 'G': 32.0, 'T': 18.0},\n",
       "  5: {'A': 6.0, 'C': 41.0, 'G': 3.0, 'T': 54.0},\n",
       "  6: {'A': 2.0, 'C': 8.0, 'G': 1.0, 'T': 93.0},\n",
       "  7: {'A': 99.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  8: {'A': 99.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 99.0},\n",
       "  10: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 99.0},\n",
       "  11: {'A': 93.0, 'C': 1.0, 'G': 8.0, 'T': 2.0},\n",
       "  12: {'A': 54.0, 'C': 3.0, 'G': 41.0, 'T': 6.0},\n",
       "  13: {'A': 16.0, 'C': 22.0, 'G': 12.0, 'T': 54.0},\n",
       "  14: {'A': 24.0, 'C': 33.0, 'G': 17.0, 'T': 30.0},\n",
       "  15: {'A': 40.0, 'C': 32.0, 'G': 13.0, 'T': 18.0},\n",
       "  16: {'A': 25.0, 'C': 30.0, 'G': 24.0, 'T': 25.0},\n",
       "  17: {'A': 16.0, 'C': 41.0, 'G': 15.0, 'T': 32.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q8C8B0',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0503.1',\n",
       "  'Matrix_Name': 'Nkx2-5(var.2)',\n",
       "  1: {'A': 1791.0, 'C': 328.0, 'G': 822.0, 'T': 492.0},\n",
       "  2: {'A': 1283.0, 'C': 193.0, 'G': 1468.0, 'T': 489.0},\n",
       "  3: {'A': 345.0, 'C': 1778.0, 'G': 1293.0, 'T': 17.0},\n",
       "  4: {'A': 1.0, 'C': 3430.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 2468.0, 'C': 7.0, 'G': 1.0, 'T': 957.0},\n",
       "  6: {'A': 1.0, 'C': 3430.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3430.0},\n",
       "  8: {'A': 1.0, 'C': 3392.0, 'G': 1.0, 'T': 39.0},\n",
       "  9: {'A': 2859.0, 'C': 432.0, 'G': 1.0, 'T': 141.0},\n",
       "  10: {'A': 2266.0, 'C': 4.0, 'G': 826.0, 'T': 337.0},\n",
       "  11: {'A': 503.0, 'C': 957.0, 'G': 1796.0, 'T': 177.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7797561',\n",
       "  'uniprot_ids': 'P42582',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0024.3',\n",
       "  'Matrix_Name': 'E2F1',\n",
       "  1: {'A': 255.0, 'C': 116.0, 'G': 90.0, 'T': 566.0},\n",
       "  2: {'A': 242.0, 'C': 47.0, 'G': 101.0, 'T': 639.0},\n",
       "  3: {'A': 209.0, 'C': 37.0, 'G': 146.0, 'T': 628.0},\n",
       "  4: {'A': 55.0, 'C': 72.0, 'G': 889.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 33.0, 'G': 951.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 600.0, 'G': 1.0, 'T': 3.0},\n",
       "  7: {'A': 1.0, 'C': 2.0, 'G': 1010.0, 'T': 1.0},\n",
       "  8: {'A': 3.0, 'C': 589.0, 'G': 18.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 578.0, 'G': 47.0, 'T': 51.0},\n",
       "  10: {'A': 525.0, 'C': 172.0, 'G': 21.0, 'T': 120.0},\n",
       "  11: {'A': 511.0, 'C': 81.0, 'G': 59.0, 'T': 183.0},\n",
       "  12: {'A': 494.0, 'C': 60.0, 'G': 113.0, 'T': 220.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '9372931',\n",
       "  'uniprot_ids': 'Q01094',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1149.1',\n",
       "  'Matrix_Name': 'RARA::RXRG',\n",
       "  1: {'A': 346.0, 'C': 114.0, 'G': 414.0, 'T': 130.0},\n",
       "  2: {'A': 497.0, 'C': 39.0, 'G': 461.0, 'T': 7.0},\n",
       "  3: {'A': 14.0, 'C': 4.0, 'G': 955.0, 'T': 31.0},\n",
       "  4: {'A': 12.0, 'C': 4.0, 'G': 829.0, 'T': 159.0},\n",
       "  5: {'A': 6.0, 'C': 16.0, 'G': 72.0, 'T': 910.0},\n",
       "  6: {'A': 19.0, 'C': 832.0, 'G': 59.0, 'T': 94.0},\n",
       "  7: {'A': 875.0, 'C': 6.0, 'G': 112.0, 'T': 11.0},\n",
       "  8: {'A': 273.0, 'C': 240.0, 'G': 155.0, 'T': 335.0},\n",
       "  9: {'A': 216.0, 'C': 250.0, 'G': 321.0, 'T': 216.0},\n",
       "  10: {'A': 212.0, 'C': 213.0, 'G': 273.0, 'T': 306.0},\n",
       "  11: {'A': 345.0, 'C': 200.0, 'G': 232.0, 'T': 228.0},\n",
       "  12: {'A': 349.0, 'C': 134.0, 'G': 398.0, 'T': 122.0},\n",
       "  13: {'A': 496.0, 'C': 91.0, 'G': 391.0, 'T': 26.0},\n",
       "  14: {'A': 27.0, 'C': 5.0, 'G': 928.0, 'T': 44.0},\n",
       "  15: {'A': 13.0, 'C': 6.0, 'G': 816.0, 'T': 170.0},\n",
       "  16: {'A': 61.0, 'C': 113.0, 'G': 227.0, 'T': 602.0},\n",
       "  17: {'A': 74.0, 'C': 697.0, 'G': 98.0, 'T': 136.0},\n",
       "  18: {'A': 703.0, 'C': 54.0, 'G': 207.0, 'T': 41.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '10698945',\n",
       "  'uniprot_ids': 'P10276;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0855.1',\n",
       "  'Matrix_Name': 'RXRB',\n",
       "  1: {'A': 1431.0, 'C': 343.0, 'G': 3157.0, 'T': 225.0},\n",
       "  2: {'A': 1660.0, 'C': 9.0, 'G': 3738.0, 'T': 4.0},\n",
       "  3: {'A': 15.0, 'C': 2.0, 'G': 3956.0, 'T': 11.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 3643.0, 'T': 493.0},\n",
       "  5: {'A': 1.0, 'C': 3.0, 'G': 12.0, 'T': 4482.0},\n",
       "  6: {'A': 38.0, 'C': 7054.0, 'G': 113.0, 'T': 351.0},\n",
       "  7: {'A': 5824.0, 'C': 1.0, 'G': 13.0, 'T': 2.0},\n",
       "  8: {'A': 5821.0, 'C': 6.0, 'G': 120.0, 'T': 32.0},\n",
       "  9: {'A': 5730.0, 'C': 2.0, 'G': 222.0, 'T': 1.0},\n",
       "  10: {'A': 2.0, 'C': 1.0, 'G': 3868.0, 'T': 8.0},\n",
       "  11: {'A': 3.0, 'C': 1.0, 'G': 3746.0, 'T': 313.0},\n",
       "  12: {'A': 2.0, 'C': 3.0, 'G': 20.0, 'T': 4432.0},\n",
       "  13: {'A': 17.0, 'C': 7054.0, 'G': 103.0, 'T': 185.0},\n",
       "  14: {'A': 5739.0, 'C': 14.0, 'G': 300.0, 'T': 4.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P28702',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0486.2',\n",
       "  'Matrix_Name': 'HSF1',\n",
       "  1: {'A': 65.0, 'C': 19.0, 'G': 26.0, 'T': 1456.0},\n",
       "  2: {'A': 12.0, 'C': 5.0, 'G': 18.0, 'T': 1456.0},\n",
       "  3: {'A': 10.0, 'C': 1456.0, 'G': 10.0, 'T': 6.0},\n",
       "  4: {'A': 5.0, 'C': 362.0, 'G': 239.0, 'T': 1456.0},\n",
       "  5: {'A': 1456.0, 'C': 425.0, 'G': 385.0, 'T': 4.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1456.0, 'T': 1.0},\n",
       "  7: {'A': 1456.0, 'C': 9.0, 'G': 3.0, 'T': 8.0},\n",
       "  8: {'A': 1456.0, 'C': 5.0, 'G': 5.0, 'T': 28.0},\n",
       "  9: {'A': 71.0, 'C': 714.0, 'G': 160.0, 'T': 516.0},\n",
       "  10: {'A': 491.0, 'C': 191.0, 'G': 593.0, 'T': 184.0},\n",
       "  11: {'A': 26.0, 'C': 6.0, 'G': 1.0, 'T': 1456.0},\n",
       "  12: {'A': 7.0, 'C': 3.0, 'G': 5.0, 'T': 1456.0},\n",
       "  13: {'A': 4.0, 'C': 1456.0, 'G': 4.0, 'T': 5.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'HSF',\n",
       "  'tf_class': 'Heat',\n",
       "  'pubmed_ids': '17500587',\n",
       "  'uniprot_ids': 'Q00613',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0511.2',\n",
       "  'Matrix_Name': 'RUNX2',\n",
       "  1: {'A': 3873.0, 'C': 1093.0, 'G': 327.0, 'T': 2749.0},\n",
       "  2: {'A': 6622.0, 'C': 557.0, 'G': 1216.0, 'T': 583.0},\n",
       "  3: {'A': 6622.0, 'C': 326.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 6622.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 6622.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1915.0, 'C': 37.0, 'G': 4708.0, 'T': 121.0},\n",
       "  7: {'A': 1.0, 'C': 6622.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 6622.0, 'C': 477.0, 'G': 459.0, 'T': 192.0},\n",
       "  9: {'A': 4203.0, 'C': 678.0, 'G': 1135.0, 'T': 608.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Runt-related',\n",
       "  'tf_class': 'Runt',\n",
       "  'pubmed_ids': '20483917',\n",
       "  'uniprot_ids': 'Q13950',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0878.2',\n",
       "  'Matrix_Name': 'CDX1',\n",
       "  1: {'A': 3247.0, 'C': 2472.0, 'G': 8785.0, 'T': 4499.0},\n",
       "  2: {'A': 1843.0, 'C': 1.0, 'G': 18999.0, 'T': 4.0},\n",
       "  3: {'A': 1.0, 'C': 13343.0, 'G': 2.0, 'T': 5656.0},\n",
       "  4: {'A': 7214.0, 'C': 11786.0, 'G': 29.0, 'T': 1.0},\n",
       "  5: {'A': 18999.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 18999.0},\n",
       "  7: {'A': 18999.0, 'C': 1.0, 'G': 11.0, 'T': 1.0},\n",
       "  8: {'A': 18999.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 18999.0, 'C': 1.0, 'G': 20.0, 'T': 1.0},\n",
       "  10: {'A': 18999.0, 'C': 7225.0, 'G': 1688.0, 'T': 3372.0},\n",
       "  11: {'A': 1783.0, 'C': 10694.0, 'G': 2694.0, 'T': 3830.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P47902',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0665.1',\n",
       "  'Matrix_Name': 'MSC',\n",
       "  1: {'A': 70.0, 'C': 1.0, 'G': 19.0, 'T': 8.0},\n",
       "  2: {'A': 70.0, 'C': 13.0, 'G': 4.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 70.0, 'G': 4.0, 'T': 2.0},\n",
       "  4: {'A': 70.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 13.0, 'G': 70.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 70.0, 'G': 4.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 70.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 70.0, 'T': 5.0},\n",
       "  9: {'A': 4.0, 'C': 11.0, 'G': 5.0, 'T': 70.0},\n",
       "  10: {'A': 1.0, 'C': 10.0, 'G': 5.0, 'T': 70.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '24175993',\n",
       "  'uniprot_ids': 'O60682',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0632.2',\n",
       "  'Matrix_Name': 'TCFL5',\n",
       "  1: {'A': 2811.0, 'C': 10583.0, 'G': 22298.0, 'T': 36855.0},\n",
       "  2: {'A': 1.0, 'C': 72544.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 72544.0, 'C': 1.0, 'G': 29711.0, 'T': 9360.0},\n",
       "  4: {'A': 1.0, 'C': 72544.0, 'G': 1.0, 'T': 1644.0},\n",
       "  5: {'A': 1567.0, 'C': 1.0, 'G': 72544.0, 'T': 1.0},\n",
       "  6: {'A': 1766.0, 'C': 57093.0, 'G': 1.0, 'T': 15452.0},\n",
       "  7: {'A': 3521.0, 'C': 658.0, 'G': 72544.0, 'T': 1.0},\n",
       "  8: {'A': 1594.0, 'C': 72544.0, 'G': 4595.0, 'T': 636.0},\n",
       "  9: {'A': 72544.0, 'C': 44395.0, 'G': 19097.0, 'T': 25922.0},\n",
       "  10: {'A': 10589.0, 'C': 47539.0, 'G': 5748.0, 'T': 8671.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15585666',\n",
       "  'uniprot_ids': 'Q9UL49',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0759.1',\n",
       "  'Matrix_Name': 'ELK3',\n",
       "  1: {'A': 2238.0, 'C': 295.0, 'G': 714.0, 'T': 457.0},\n",
       "  2: {'A': 270.0, 'C': 2238.0, 'G': 224.0, 'T': 78.0},\n",
       "  3: {'A': 121.0, 'C': 2238.0, 'G': 15.0, 'T': 9.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 2238.0, 'T': 1.0},\n",
       "  5: {'A': 3.0, 'C': 5.0, 'G': 2238.0, 'T': 79.0},\n",
       "  6: {'A': 2238.0, 'C': 8.0, 'G': 4.0, 'T': 2.0},\n",
       "  7: {'A': 2238.0, 'C': 44.0, 'G': 1.0, 'T': 354.0},\n",
       "  8: {'A': 499.0, 'C': 246.0, 'G': 2238.0, 'T': 59.0},\n",
       "  9: {'A': 152.0, 'C': 488.0, 'G': 169.0, 'T': 2238.0},\n",
       "  10: {'A': 779.0, 'C': 319.0, 'G': 650.0, 'T': 493.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'P41970',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0895.1',\n",
       "  'Matrix_Name': 'HMBOX1',\n",
       "  1: {'A': 488.0, 'C': 450.0, 'G': 195.0, 'T': 102.0},\n",
       "  2: {'A': 123.0, 'C': 684.0, 'G': 187.0, 'T': 548.0},\n",
       "  3: {'A': 76.0, 'C': 26.0, 'G': 64.0, 'T': 1232.0},\n",
       "  4: {'A': 1232.0, 'C': 1.0, 'G': 368.0, 'T': 45.0},\n",
       "  5: {'A': 3.0, 'C': 52.0, 'G': 1232.0, 'T': 6.0},\n",
       "  6: {'A': 27.0, 'C': 2.0, 'G': 6.0, 'T': 1232.0},\n",
       "  7: {'A': 238.0, 'C': 51.0, 'G': 9.0, 'T': 1232.0},\n",
       "  8: {'A': 1232.0, 'C': 5.0, 'G': 18.0, 'T': 103.0},\n",
       "  9: {'A': 754.0, 'C': 479.0, 'G': 131.0, 'T': 141.0},\n",
       "  10: {'A': 203.0, 'C': 543.0, 'G': 371.0, 'T': 118.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q6NT76',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0489.1',\n",
       "  'Matrix_Name': 'JUN(var.2)',\n",
       "  1: {'A': 4853.0, 'C': 1251.0, 'G': 2819.0, 'T': 2037.0},\n",
       "  2: {'A': 3928.0, 'C': 1325.0, 'G': 4111.0, 'T': 1596.0},\n",
       "  3: {'A': 3727.0, 'C': 1128.0, 'G': 4331.0, 'T': 1774.0},\n",
       "  4: {'A': 4397.0, 'C': 1123.0, 'G': 3462.0, 'T': 1978.0},\n",
       "  5: {'A': 4147.0, 'C': 594.0, 'G': 4214.0, 'T': 2005.0},\n",
       "  6: {'A': 5931.0, 'C': 994.0, 'G': 4034.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 10957.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 10957.0, 'T': 1.0},\n",
       "  9: {'A': 10954.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  10: {'A': 383.0, 'C': 5705.0, 'G': 4091.0, 'T': 781.0},\n",
       "  11: {'A': 200.0, 'C': 1.0, 'G': 1.0, 'T': 10758.0},\n",
       "  12: {'A': 957.0, 'C': 10001.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 10957.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 377.0, 'C': 2510.0, 'G': 2646.0, 'T': 5427.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21703547',\n",
       "  'uniprot_ids': 'P05412',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0661.1',\n",
       "  'Matrix_Name': 'MEOX1',\n",
       "  1: {'A': 907.0, 'C': 705.0, 'G': 987.0, 'T': 406.0},\n",
       "  2: {'A': 151.0, 'C': 1638.0, 'G': 1364.0, 'T': 116.0},\n",
       "  3: {'A': 65.0, 'C': 407.0, 'G': 54.0, 'T': 3001.0},\n",
       "  4: {'A': 3001.0, 'C': 849.0, 'G': 152.0, 'T': 40.0},\n",
       "  5: {'A': 3001.0, 'C': 1.0, 'G': 3.0, 'T': 4.0},\n",
       "  6: {'A': 9.0, 'C': 30.0, 'G': 1.0, 'T': 3001.0},\n",
       "  7: {'A': 15.0, 'C': 243.0, 'G': 329.0, 'T': 3001.0},\n",
       "  8: {'A': 3001.0, 'C': 1.0, 'G': 339.0, 'T': 7.0},\n",
       "  9: {'A': 957.0, 'C': 912.0, 'G': 450.0, 'T': 684.0},\n",
       "  10: {'A': 573.0, 'C': 1310.0, 'G': 615.0, 'T': 507.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P50221',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0515.1',\n",
       "  'Matrix_Name': 'Sox6',\n",
       "  1: {'A': 5.0, 'C': 140.0, 'G': 51.0, 'T': 57.0},\n",
       "  2: {'A': 1.0, 'C': 222.0, 'G': 1.0, 'T': 29.0},\n",
       "  3: {'A': 162.0, 'C': 1.0, 'G': 1.0, 'T': 89.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 250.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 250.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 250.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 250.0},\n",
       "  8: {'A': 1.0, 'C': 116.0, 'G': 6.0, 'T': 130.0},\n",
       "  9: {'A': 5.0, 'C': 113.0, 'G': 1.0, 'T': 134.0},\n",
       "  10: {'A': 15.0, 'C': 77.0, 'G': 32.0, 'T': 129.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '21985497',\n",
       "  'uniprot_ids': 'P40645',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0078.1',\n",
       "  'Matrix_Name': 'Sox17',\n",
       "  1: {'A': 8.0, 'C': 10.0, 'G': 7.0, 'T': 10.0},\n",
       "  2: {'A': 9.0, 'C': 9.0, 'G': 5.0, 'T': 12.0},\n",
       "  3: {'A': 4.0, 'C': 19.0, 'G': 2.0, 'T': 10.0},\n",
       "  4: {'A': 31.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 31.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 32.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 32.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 3.0, 'T': 30.0},\n",
       "  9: {'A': 1.0, 'C': 18.0, 'G': 11.0, 'T': 5.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '8636240',\n",
       "  'uniprot_ids': 'Q61473',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1498.1',\n",
       "  'Matrix_Name': 'HOXA7',\n",
       "  1: {'A': 482.0, 'C': 840.0, 'G': 2303.0, 'T': 606.0},\n",
       "  2: {'A': 1.0, 'C': 2320.0, 'G': 10.0, 'T': 1909.0},\n",
       "  3: {'A': 4228.0, 'C': 3336.0, 'G': 418.0, 'T': 227.0},\n",
       "  4: {'A': 4228.0, 'C': 84.0, 'G': 26.0, 'T': 149.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 4228.0},\n",
       "  6: {'A': 160.0, 'C': 1.0, 'G': 816.0, 'T': 4228.0},\n",
       "  7: {'A': 4228.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 829.0, 'C': 1721.0, 'G': 978.0, 'T': 703.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31268',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1562.1',\n",
       "  'Matrix_Name': 'SOX14',\n",
       "  1: {'A': 266.0, 'C': 595.0, 'G': 215.0, 'T': 183.0},\n",
       "  2: {'A': 69.0, 'C': 595.0, 'G': 45.0, 'T': 100.0},\n",
       "  3: {'A': 45.0, 'C': 19.0, 'G': 595.0, 'T': 71.0},\n",
       "  4: {'A': 595.0, 'C': 9.0, 'G': 28.0, 'T': 38.0},\n",
       "  5: {'A': 595.0, 'C': 14.0, 'G': 1.0, 'T': 20.0},\n",
       "  6: {'A': 5.0, 'C': 595.0, 'G': 11.0, 'T': 18.0},\n",
       "  7: {'A': 595.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 595.0, 'C': 18.0, 'G': 1.0, 'T': 6.0},\n",
       "  9: {'A': 1.0, 'C': 3.0, 'G': 17.0, 'T': 595.0},\n",
       "  10: {'A': 76.0, 'C': 68.0, 'G': 373.0, 'T': 81.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '17852354',\n",
       "  'uniprot_ids': 'O95416',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0872.1',\n",
       "  'Matrix_Name': 'TFAP2A(var.3)',\n",
       "  1: {'A': 263.0, 'C': 155.0, 'G': 135.0, 'T': 838.0},\n",
       "  2: {'A': 1.0, 'C': 358.0, 'G': 1360.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1369.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1374.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 49.0, 'C': 1116.0, 'G': 96.0, 'T': 132.0},\n",
       "  6: {'A': 56.0, 'C': 525.0, 'G': 213.0, 'T': 651.0},\n",
       "  7: {'A': 189.0, 'C': 507.0, 'G': 547.0, 'T': 252.0},\n",
       "  8: {'A': 557.0, 'C': 225.0, 'G': 539.0, 'T': 76.0},\n",
       "  9: {'A': 193.0, 'C': 67.0, 'G': 1421.0, 'T': 54.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 1912.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 1904.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1100.0, 'G': 367.0, 'T': 1.0},\n",
       "  13: {'A': 722.0, 'C': 103.0, 'G': 199.0, 'T': 261.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'P05549',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0156.2',\n",
       "  'Matrix_Name': 'FEV',\n",
       "  1: {'A': 7250.0, 'C': 575.0, 'G': 1556.0, 'T': 1630.0},\n",
       "  2: {'A': 1166.0, 'C': 7250.0, 'G': 719.0, 'T': 244.0},\n",
       "  3: {'A': 615.0, 'C': 7250.0, 'G': 52.0, 'T': 4.0},\n",
       "  4: {'A': 14.0, 'C': 1.0, 'G': 7250.0, 'T': 3.0},\n",
       "  5: {'A': 3.0, 'C': 2.0, 'G': 7250.0, 'T': 10.0},\n",
       "  6: {'A': 7250.0, 'C': 12.0, 'G': 4.0, 'T': 19.0},\n",
       "  7: {'A': 7250.0, 'C': 155.0, 'G': 7.0, 'T': 2197.0},\n",
       "  8: {'A': 3319.0, 'C': 367.0, 'G': 7250.0, 'T': 93.0},\n",
       "  9: {'A': 445.0, 'C': 1822.0, 'G': 625.0, 'T': 7250.0},\n",
       "  10: {'A': 2135.0, 'C': 1359.0, 'G': 2258.0, 'T': 1500.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'Q99581',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1576.1',\n",
       "  'Matrix_Name': 'THRB(var.3)',\n",
       "  1: {'A': 120.0, 'C': 32.0, 'G': 158.0, 'T': 83.0},\n",
       "  2: {'A': 1.0, 'C': 9.0, 'G': 1.0, 'T': 382.0},\n",
       "  3: {'A': 29.0, 'C': 90.0, 'G': 390.0, 'T': 46.0},\n",
       "  4: {'A': 390.0, 'C': 20.0, 'G': 18.0, 'T': 72.0},\n",
       "  5: {'A': 1.0, 'C': 390.0, 'G': 5.0, 'T': 18.0},\n",
       "  6: {'A': 1.0, 'C': 390.0, 'G': 60.0, 'T': 21.0},\n",
       "  7: {'A': 14.0, 'C': 76.0, 'G': 39.0, 'T': 390.0},\n",
       "  8: {'A': 12.0, 'C': 135.0, 'G': 94.0, 'T': 152.0},\n",
       "  9: {'A': 264.0, 'C': 29.0, 'G': 90.0, 'T': 10.0},\n",
       "  10: {'A': 43.0, 'C': 222.0, 'G': 72.0, 'T': 55.0},\n",
       "  11: {'A': 103.0, 'C': 87.0, 'G': 206.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 7.0, 'T': 394.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 390.0, 'T': 4.0},\n",
       "  14: {'A': 390.0, 'C': 12.0, 'G': 1.0, 'T': 87.0},\n",
       "  15: {'A': 1.0, 'C': 390.0, 'G': 1.0, 'T': 3.0},\n",
       "  16: {'A': 8.0, 'C': 390.0, 'G': 4.0, 'T': 5.0},\n",
       "  17: {'A': 1.0, 'C': 10.0, 'G': 1.0, 'T': 390.0},\n",
       "  18: {'A': 1.0, 'C': 176.0, 'G': 74.0, 'T': 215.0},\n",
       "  19: {'A': 324.0, 'C': 9.0, 'G': 75.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 19,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '19741045',\n",
       "  'uniprot_ids': 'P10828',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0796.1',\n",
       "  'Matrix_Name': 'TGIF1',\n",
       "  1: {'A': 40.0, 'C': 20.0, 'G': 2.0, 'T': 13222.0},\n",
       "  2: {'A': 10.0, 'C': 9.0, 'G': 13500.0, 'T': 5.0},\n",
       "  3: {'A': 8829.0, 'C': 4.0, 'G': 21.0, 'T': 1.0},\n",
       "  4: {'A': 2.0, 'C': 13305.0, 'G': 6.0, 'T': 8.0},\n",
       "  5: {'A': 8862.0, 'C': 6.0, 'G': 23.0, 'T': 5.0},\n",
       "  6: {'A': 15.0, 'C': 21.0, 'G': 12674.0, 'T': 964.0},\n",
       "  7: {'A': 453.0, 'C': 10563.0, 'G': 2521.0, 'T': 13.0},\n",
       "  8: {'A': 2.0, 'C': 63.0, 'G': 9.0, 'T': 12776.0},\n",
       "  9: {'A': 12.0, 'C': 14.0, 'G': 13262.0, 'T': 10.0},\n",
       "  10: {'A': 46.0, 'C': 25.0, 'G': 4.0, 'T': 12657.0},\n",
       "  11: {'A': 37.0, 'C': 13486.0, 'G': 14.0, 'T': 18.0},\n",
       "  12: {'A': 9035.0, 'C': 18.0, 'G': 33.0, 'T': 34.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8537382',\n",
       "  'uniprot_ids': 'Q15583',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0673.1',\n",
       "  'Matrix_Name': 'NKX2-8',\n",
       "  1: {'A': 1943.0, 'C': 3530.0, 'G': 2844.0, 'T': 250.0},\n",
       "  2: {'A': 71.0, 'C': 8563.0, 'G': 1.0, 'T': 2156.0},\n",
       "  3: {'A': 8563.0, 'C': 596.0, 'G': 1.0, 'T': 517.0},\n",
       "  4: {'A': 15.0, 'C': 8563.0, 'G': 1.0, 'T': 111.0},\n",
       "  5: {'A': 1.0, 'C': 92.0, 'G': 1.0, 'T': 8563.0},\n",
       "  6: {'A': 24.0, 'C': 3198.0, 'G': 1.0, 'T': 8563.0},\n",
       "  7: {'A': 1739.0, 'C': 2268.0, 'G': 4287.0, 'T': 272.0},\n",
       "  8: {'A': 8563.0, 'C': 909.0, 'G': 697.0, 'T': 2159.0},\n",
       "  9: {'A': 3195.0, 'C': 1965.0, 'G': 2594.0, 'T': 812.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '11042197',\n",
       "  'uniprot_ids': 'O15522',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0484.2',\n",
       "  'Matrix_Name': 'HNF4G',\n",
       "  1: {'A': 3844.0, 'C': 2787.0, 'G': 5561.0, 'T': 6034.0},\n",
       "  2: {'A': 3656.0, 'C': 5661.0, 'G': 3821.0, 'T': 5088.0},\n",
       "  3: {'A': 435.0, 'C': 16489.0, 'G': 420.0, 'T': 882.0},\n",
       "  4: {'A': 17012.0, 'C': 450.0, 'G': 457.0, 'T': 307.0},\n",
       "  5: {'A': 15705.0, 'C': 508.0, 'G': 1577.0, 'T': 436.0},\n",
       "  6: {'A': 16964.0, 'C': 173.0, 'G': 770.0, 'T': 319.0},\n",
       "  7: {'A': 209.0, 'C': 330.0, 'G': 17216.0, 'T': 471.0},\n",
       "  8: {'A': 278.0, 'C': 228.0, 'G': 842.0, 'T': 16878.0},\n",
       "  9: {'A': 1119.0, 'C': 14918.0, 'G': 698.0, 'T': 1491.0},\n",
       "  10: {'A': 565.0, 'C': 16496.0, 'G': 227.0, 'T': 938.0},\n",
       "  11: {'A': 14054.0, 'C': 1191.0, 'G': 1300.0, 'T': 1681.0},\n",
       "  12: {'A': 5009.0, 'C': 4074.0, 'G': 5327.0, 'T': 3816.0},\n",
       "  13: {'A': 5813.0, 'C': 3719.0, 'G': 4663.0, 'T': 4031.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '22383578',\n",
       "  'uniprot_ids': 'Q14541',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1559.1',\n",
       "  'Matrix_Name': 'SNAI3',\n",
       "  1: {'A': 2953.0, 'C': 662.0, 'G': 1526.0, 'T': 1111.0},\n",
       "  2: {'A': 3918.0, 'C': 1.0, 'G': 2332.0, 'T': 184.0},\n",
       "  3: {'A': 24.0, 'C': 6249.0, 'G': 1.0, 'T': 36.0},\n",
       "  4: {'A': 6249.0, 'C': 1.0, 'G': 35.0, 'T': 4.0},\n",
       "  5: {'A': 119.0, 'C': 30.0, 'G': 6249.0, 'T': 44.0},\n",
       "  6: {'A': 1.0, 'C': 19.0, 'G': 6249.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 13.0, 'G': 29.0, 'T': 6249.0},\n",
       "  8: {'A': 95.0, 'C': 1.0, 'G': 6249.0, 'T': 31.0},\n",
       "  9: {'A': 433.0, 'C': 6249.0, 'G': 1702.0, 'T': 3162.0},\n",
       "  10: {'A': 4681.0, 'C': 689.0, 'G': 1570.0, 'T': 554.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23762348',\n",
       "  'uniprot_ids': 'Q3KNW1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0056.2',\n",
       "  'Matrix_Name': 'MZF1',\n",
       "  1: {'A': 2142.0, 'C': 2261.0, 'G': 1809.0, 'T': 2034.0},\n",
       "  2: {'A': 2478.0, 'C': 1945.0, 'G': 1544.0, 'T': 2279.0},\n",
       "  3: {'A': 5856.0, 'C': 1017.0, 'G': 1000.0, 'T': 373.0},\n",
       "  4: {'A': 7150.0, 'C': 427.0, 'G': 386.0, 'T': 283.0},\n",
       "  5: {'A': 107.0, 'C': 57.0, 'G': 99.0, 'T': 7983.0},\n",
       "  6: {'A': 101.0, 'C': 7950.0, 'G': 101.0, 'T': 94.0},\n",
       "  7: {'A': 136.0, 'C': 7940.0, 'G': 55.0, 'T': 115.0},\n",
       "  8: {'A': 112.0, 'C': 7920.0, 'G': 59.0, 'T': 155.0},\n",
       "  9: {'A': 118.0, 'C': 7674.0, 'G': 84.0, 'T': 370.0},\n",
       "  10: {'A': 6037.0, 'C': 612.0, 'G': 485.0, 'T': 1112.0},\n",
       "  11: {'A': 1694.0, 'C': 3008.0, 'G': 1524.0, 'T': 2020.0},\n",
       "  12: {'A': 2097.0, 'C': 2079.0, 'G': 1483.0, 'T': 2587.0},\n",
       "  13: {'A': 1763.0, 'C': 2207.0, 'G': 1774.0, 'T': 2502.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '8114711',\n",
       "  'uniprot_ids': 'P28698',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0496.3',\n",
       "  'Matrix_Name': 'MAFK',\n",
       "  1: {'A': 11947.0, 'C': 10790.0, 'G': 9970.0, 'T': 19031.0},\n",
       "  2: {'A': 9603.0, 'C': 10656.0, 'G': 21905.0, 'T': 9574.0},\n",
       "  3: {'A': 14090.0, 'C': 27001.0, 'G': 6707.0, 'T': 3940.0},\n",
       "  4: {'A': 1617.0, 'C': 1569.0, 'G': 1059.0, 'T': 47493.0},\n",
       "  5: {'A': 1109.0, 'C': 1367.0, 'G': 46696.0, 'T': 2566.0},\n",
       "  6: {'A': 47485.0, 'C': 1671.0, 'G': 1307.0, 'T': 1275.0},\n",
       "  7: {'A': 2493.0, 'C': 39126.0, 'G': 7574.0, 'T': 2545.0},\n",
       "  8: {'A': 2723.0, 'C': 1134.0, 'G': 1082.0, 'T': 46799.0},\n",
       "  9: {'A': 1819.0, 'C': 47873.0, 'G': 958.0, 'T': 1088.0},\n",
       "  10: {'A': 47676.0, 'C': 883.0, 'G': 1268.0, 'T': 1911.0},\n",
       "  11: {'A': 1440.0, 'C': 1922.0, 'G': 44014.0, 'T': 4362.0},\n",
       "  12: {'A': 2665.0, 'C': 43459.0, 'G': 3328.0, 'T': 2286.0},\n",
       "  13: {'A': 24803.0, 'C': 14139.0, 'G': 4591.0, 'T': 8205.0},\n",
       "  14: {'A': 15934.0, 'C': 10738.0, 'G': 10513.0, 'T': 14553.0},\n",
       "  15: {'A': 14819.0, 'C': 9757.0, 'G': 8542.0, 'T': 18620.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8264639',\n",
       "  'uniprot_ids': 'O60675',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0751.1',\n",
       "  'Matrix_Name': 'ZIC4',\n",
       "  1: {'A': 579.0, 'C': 935.0, 'G': 4127.0, 'T': 1888.0},\n",
       "  2: {'A': 3014.0, 'C': 813.0, 'G': 1554.0, 'T': 22.0},\n",
       "  3: {'A': 316.0, 'C': 2871.0, 'G': 53.0, 'T': 99.0},\n",
       "  4: {'A': 210.0, 'C': 2891.0, 'G': 122.0, 'T': 148.0},\n",
       "  5: {'A': 615.0, 'C': 2758.0, 'G': 226.0, 'T': 276.0},\n",
       "  6: {'A': 9.0, 'C': 3000.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 115.0, 'C': 2986.0, 'G': 1.0, 'T': 12.0},\n",
       "  8: {'A': 25.0, 'C': 2223.0, 'G': 7.0, 'T': 1079.0},\n",
       "  9: {'A': 1096.0, 'C': 69.0, 'G': 4796.0, 'T': 70.0},\n",
       "  10: {'A': 25.0, 'C': 2280.0, 'G': 253.0, 'T': 1018.0},\n",
       "  11: {'A': 1053.0, 'C': 866.0, 'G': 1904.0, 'T': 3489.0},\n",
       "  12: {'A': 259.0, 'C': 19.0, 'G': 5087.0, 'T': 68.0},\n",
       "  13: {'A': 705.0, 'C': 1045.0, 'G': 600.0, 'T': 1455.0},\n",
       "  14: {'A': 75.0, 'C': 453.0, 'G': 4523.0, 'T': 1157.0},\n",
       "  15: {'A': 1084.0, 'C': 1225.0, 'G': 364.0, 'T': 917.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q8N9L1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0686.1',\n",
       "  'Matrix_Name': 'SPDEF',\n",
       "  1: {'A': 4629.0, 'C': 102.0, 'G': 177.0, 'T': 913.0},\n",
       "  2: {'A': 1891.0, 'C': 2738.0, 'G': 537.0, 'T': 78.0},\n",
       "  3: {'A': 53.0, 'C': 4629.0, 'G': 93.0, 'T': 1.0},\n",
       "  4: {'A': 238.0, 'C': 4629.0, 'G': 2.0, 'T': 2.0},\n",
       "  5: {'A': 3.0, 'C': 3.0, 'G': 4629.0, 'T': 1.0},\n",
       "  6: {'A': 7.0, 'C': 1.0, 'G': 4629.0, 'T': 9.0},\n",
       "  7: {'A': 4629.0, 'C': 1.0, 'G': 7.0, 'T': 3.0},\n",
       "  8: {'A': 533.0, 'C': 70.0, 'G': 12.0, 'T': 4629.0},\n",
       "  9: {'A': 1567.0, 'C': 602.0, 'G': 4629.0, 'T': 67.0},\n",
       "  10: {'A': 59.0, 'C': 1346.0, 'G': 29.0, 'T': 4629.0},\n",
       "  11: {'A': 1912.0, 'C': 569.0, 'G': 1340.0, 'T': 811.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '23449978',\n",
       "  'uniprot_ids': 'O95238',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0833.2',\n",
       "  'Matrix_Name': 'ATF4',\n",
       "  1: {'A': 12095.0, 'C': 5055.0, 'G': 9023.0, 'T': 7107.0},\n",
       "  2: {'A': 11783.0, 'C': 4366.0, 'G': 10153.0, 'T': 6978.0},\n",
       "  3: {'A': 20882.0, 'C': 5782.0, 'G': 5971.0, 'T': 645.0},\n",
       "  4: {'A': 499.0, 'C': 387.0, 'G': 298.0, 'T': 32096.0},\n",
       "  5: {'A': 229.0, 'C': 199.0, 'G': 32316.0, 'T': 536.0},\n",
       "  6: {'A': 32088.0, 'C': 218.0, 'G': 544.0, 'T': 430.0},\n",
       "  7: {'A': 244.0, 'C': 1627.0, 'G': 329.0, 'T': 31080.0},\n",
       "  8: {'A': 157.0, 'C': 52.0, 'G': 32959.0, 'T': 112.0},\n",
       "  9: {'A': 4907.0, 'C': 23001.0, 'G': 106.0, 'T': 5266.0},\n",
       "  10: {'A': 32905.0, 'C': 170.0, 'G': 75.0, 'T': 130.0},\n",
       "  11: {'A': 32964.0, 'C': 71.0, 'G': 113.0, 'T': 132.0},\n",
       "  12: {'A': 804.0, 'C': 6135.0, 'G': 2809.0, 'T': 23532.0},\n",
       "  13: {'A': 12608.0, 'C': 9847.0, 'G': 4719.0, 'T': 6106.0},\n",
       "  14: {'A': 11050.0, 'C': 6065.0, 'G': 5732.0, 'T': 10433.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'ATF-4-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22819556',\n",
       "  'uniprot_ids': 'P18848',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1501.1',\n",
       "  'Matrix_Name': 'HOXB7',\n",
       "  1: {'A': 189.0, 'C': 145.0, 'G': 381.0, 'T': 117.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 829.0, 'T': 88.0},\n",
       "  3: {'A': 1.0, 'C': 360.0, 'G': 52.0, 'T': 829.0},\n",
       "  4: {'A': 829.0, 'C': 223.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 829.0, 'C': 30.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 32.0, 'T': 829.0},\n",
       "  7: {'A': 80.0, 'C': 4.0, 'G': 1.0, 'T': 829.0},\n",
       "  8: {'A': 829.0, 'C': 1.0, 'G': 152.0, 'T': 1.0},\n",
       "  9: {'A': 325.0, 'C': 166.0, 'G': 269.0, 'T': 72.0},\n",
       "  10: {'A': 133.0, 'C': 355.0, 'G': 251.0, 'T': 93.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P09629',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0719.1',\n",
       "  'Matrix_Name': 'RHOXF1',\n",
       "  1: {'A': 3225.0, 'C': 2645.0, 'G': 1856.0, 'T': 3194.0},\n",
       "  2: {'A': 2838.0, 'C': 1.0, 'G': 479.0, 'T': 10917.0},\n",
       "  3: {'A': 5544.0, 'C': 1.0, 'G': 5374.0, 'T': 1368.0},\n",
       "  4: {'A': 10917.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 3541.0, 'T': 7377.0},\n",
       "  6: {'A': 1.0, 'C': 10917.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 10917.0, 'G': 1.0, 'T': 2276.0},\n",
       "  8: {'A': 3224.0, 'C': 3454.0, 'G': 2118.0, 'T': 2124.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q8NHV9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1577.1',\n",
       "  'Matrix_Name': 'TLX2',\n",
       "  1: {'A': 998.0, 'C': 2573.0, 'G': 1852.0, 'T': 2380.0},\n",
       "  2: {'A': 1436.0, 'C': 7800.0, 'G': 1113.0, 'T': 8157.0},\n",
       "  3: {'A': 7800.0, 'C': 1.0, 'G': 81.0, 'T': 1.0},\n",
       "  4: {'A': 7800.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7800.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7800.0},\n",
       "  7: {'A': 7800.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 2185.0, 'C': 1324.0, 'G': 3440.0, 'T': 853.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '10869550',\n",
       "  'uniprot_ids': 'O43763',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1520.1',\n",
       "  'Matrix_Name': 'MAF',\n",
       "  1: {'A': 969.0, 'C': 517.0, 'G': 588.0, 'T': 923.0},\n",
       "  2: {'A': 214.0, 'C': 517.0, 'G': 52.0, 'T': 2398.0},\n",
       "  3: {'A': 81.0, 'C': 46.0, 'G': 2939.0, 'T': 9.0},\n",
       "  4: {'A': 365.0, 'C': 2838.0, 'G': 54.0, 'T': 220.0},\n",
       "  5: {'A': 440.0, 'C': 244.0, 'G': 117.0, 'T': 2860.0},\n",
       "  6: {'A': 317.0, 'C': 186.0, 'G': 2740.0, 'T': 683.0},\n",
       "  7: {'A': 2713.0, 'C': 344.0, 'G': 117.0, 'T': 433.0},\n",
       "  8: {'A': 388.0, 'C': 1278.0, 'G': 1432.0, 'T': 345.0},\n",
       "  9: {'A': 530.0, 'C': 90.0, 'G': 405.0, 'T': 2716.0},\n",
       "  10: {'A': 610.0, 'C': 2777.0, 'G': 157.0, 'T': 302.0},\n",
       "  11: {'A': 2905.0, 'C': 104.0, 'G': 315.0, 'T': 360.0},\n",
       "  12: {'A': 217.0, 'C': 52.0, 'G': 2822.0, 'T': 330.0},\n",
       "  13: {'A': 49.0, 'C': 2965.0, 'G': 57.0, 'T': 67.0},\n",
       "  14: {'A': 2354.0, 'C': 73.0, 'G': 563.0, 'T': 211.0},\n",
       "  15: {'A': 978.0, 'C': 576.0, 'G': 487.0, 'T': 956.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9571165',\n",
       "  'uniprot_ids': 'O75444',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0725.1',\n",
       "  'Matrix_Name': 'VSX1',\n",
       "  1: {'A': 757.0, 'C': 8454.0, 'G': 562.0, 'T': 5120.0},\n",
       "  2: {'A': 1837.0, 'C': 3132.0, 'G': 573.0, 'T': 13573.0},\n",
       "  3: {'A': 13573.0, 'C': 627.0, 'G': 554.0, 'T': 555.0},\n",
       "  4: {'A': 13573.0, 'C': 155.0, 'G': 173.0, 'T': 91.0},\n",
       "  5: {'A': 173.0, 'C': 200.0, 'G': 104.0, 'T': 13573.0},\n",
       "  6: {'A': 899.0, 'C': 762.0, 'G': 985.0, 'T': 13573.0},\n",
       "  7: {'A': 13573.0, 'C': 517.0, 'G': 3338.0, 'T': 1373.0},\n",
       "  8: {'A': 2627.0, 'C': 3556.0, 'G': 3659.0, 'T': 3733.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q9NZR4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0147.3',\n",
       "  'Matrix_Name': 'MYC',\n",
       "  1: {'A': 1039.0, 'C': 1346.0, 'G': 1650.0, 'T': 911.0},\n",
       "  2: {'A': 1020.0, 'C': 1288.0, 'G': 1758.0, 'T': 880.0},\n",
       "  3: {'A': 439.0, 'C': 3747.0, 'G': 473.0, 'T': 287.0},\n",
       "  4: {'A': 45.0, 'C': 4841.0, 'G': 32.0, 'T': 28.0},\n",
       "  5: {'A': 4601.0, 'C': 60.0, 'G': 171.0, 'T': 114.0},\n",
       "  6: {'A': 55.0, 'C': 4526.0, 'G': 71.0, 'T': 294.0},\n",
       "  7: {'A': 170.0, 'C': 158.0, 'G': 4579.0, 'T': 39.0},\n",
       "  8: {'A': 60.0, 'C': 163.0, 'G': 64.0, 'T': 4659.0},\n",
       "  9: {'A': 24.0, 'C': 57.0, 'G': 4818.0, 'T': 47.0},\n",
       "  10: {'A': 420.0, 'C': 2802.0, 'G': 1199.0, 'T': 525.0},\n",
       "  11: {'A': 962.0, 'C': 1525.0, 'G': 1037.0, 'T': 1422.0},\n",
       "  12: {'A': 756.0, 'C': 1786.0, 'G': 1349.0, 'T': 1055.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '18555785',\n",
       "  'uniprot_ids': 'P01106',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0836.2',\n",
       "  'Matrix_Name': 'CEBPD',\n",
       "  1: {'A': 5118.0, 'C': 3599.0, 'G': 5550.0, 'T': 7189.0},\n",
       "  2: {'A': 9694.0, 'C': 3598.0, 'G': 5736.0, 'T': 2428.0},\n",
       "  3: {'A': 211.0, 'C': 298.0, 'G': 201.0, 'T': 20746.0},\n",
       "  4: {'A': 200.0, 'C': 82.0, 'G': 234.0, 'T': 20940.0},\n",
       "  5: {'A': 1868.0, 'C': 300.0, 'G': 17310.0, 'T': 1978.0},\n",
       "  6: {'A': 264.0, 'C': 20684.0, 'G': 215.0, 'T': 293.0},\n",
       "  7: {'A': 15975.0, 'C': 1023.0, 'G': 2626.0, 'T': 1832.0},\n",
       "  8: {'A': 1342.0, 'C': 17459.0, 'G': 378.0, 'T': 2277.0},\n",
       "  9: {'A': 20541.0, 'C': 653.0, 'G': 76.0, 'T': 186.0},\n",
       "  10: {'A': 20375.0, 'C': 316.0, 'G': 405.0, 'T': 360.0},\n",
       "  11: {'A': 2103.0, 'C': 6343.0, 'G': 3135.0, 'T': 9875.0},\n",
       "  12: {'A': 7278.0, 'C': 5122.0, 'G': 3584.0, 'T': 5472.0},\n",
       "  13: {'A': 5597.0, 'C': 5176.0, 'G': 4380.0, 'T': 6303.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8632009',\n",
       "  'uniprot_ids': 'P49716',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1489.1',\n",
       "  'Matrix_Name': 'FOXN3',\n",
       "  1: {'A': 10.0, 'C': 10.0, 'G': 972.0, 'T': 10.0},\n",
       "  2: {'A': 10.0, 'C': 10.0, 'G': 10.0, 'T': 972.0},\n",
       "  3: {'A': 972.0, 'C': 10.0, 'G': 10.0, 'T': 10.0},\n",
       "  4: {'A': 972.0, 'C': 10.0, 'G': 10.0, 'T': 10.0},\n",
       "  5: {'A': 972.0, 'C': 10.0, 'G': 10.0, 'T': 10.0},\n",
       "  6: {'A': 10.0, 'C': 972.0, 'G': 10.0, 'T': 10.0},\n",
       "  7: {'A': 972.0, 'C': 10.0, 'G': 10.0, 'T': 10.0},\n",
       "  8: {'A': 972.0, 'C': 10.0, 'G': 10.0, 'T': 10.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '30826165',\n",
       "  'uniprot_ids': 'O00409',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0839.1',\n",
       "  'Matrix_Name': 'CREB3L1',\n",
       "  1: {'A': 248.0, 'C': 144.0, 'G': 182.0, 'T': 101.0},\n",
       "  2: {'A': 39.0, 'C': 160.0, 'G': 30.0, 'T': 672.0},\n",
       "  3: {'A': 21.0, 'C': 4.0, 'G': 672.0, 'T': 19.0},\n",
       "  4: {'A': 62.0, 'C': 672.0, 'G': 1.0, 'T': 3.0},\n",
       "  5: {'A': 4.0, 'C': 672.0, 'G': 1.0, 'T': 6.0},\n",
       "  6: {'A': 672.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 672.0, 'G': 4.0, 'T': 1.0},\n",
       "  8: {'A': 22.0, 'C': 4.0, 'G': 672.0, 'T': 1.0},\n",
       "  9: {'A': 8.0, 'C': 1.0, 'G': 4.0, 'T': 672.0},\n",
       "  10: {'A': 122.0, 'C': 672.0, 'G': 41.0, 'T': 11.0},\n",
       "  11: {'A': 672.0, 'C': 34.0, 'G': 73.0, 'T': 40.0},\n",
       "  12: {'A': 61.0, 'C': 187.0, 'G': 126.0, 'T': 301.0},\n",
       "  13: {'A': 77.0, 'C': 672.0, 'G': 65.0, 'T': 80.0},\n",
       "  14: {'A': 672.0, 'C': 66.0, 'G': 168.0, 'T': 128.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21041443',\n",
       "  'uniprot_ids': 'Q96BA8',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0485.2',\n",
       "  'Matrix_Name': 'HOXC9',\n",
       "  1: {'A': 4348.0, 'C': 1761.0, 'G': 15542.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 148.0, 'G': 1.0, 'T': 15542.0},\n",
       "  3: {'A': 1.0, 'C': 15542.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 4251.0, 'C': 1.0, 'G': 15542.0, 'T': 1.0},\n",
       "  5: {'A': 6.0, 'C': 6.0, 'G': 12.0, 'T': 15542.0},\n",
       "  6: {'A': 15542.0, 'C': 1.0, 'G': 1.0, 'T': 2095.0},\n",
       "  7: {'A': 15542.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 15542.0, 'C': 13.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 15542.0, 'C': 2034.0, 'G': 1367.0, 'T': 3542.0},\n",
       "  10: {'A': 3257.0, 'C': 5038.0, 'G': 2192.0, 'T': 5058.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9079637',\n",
       "  'uniprot_ids': 'P31274',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0633.1',\n",
       "  'Matrix_Name': 'Twist2',\n",
       "  1: {'A': 303.0, 'C': 201.0, 'G': 299.0, 'T': 201.0},\n",
       "  2: {'A': 283.0, 'C': 581.0, 'G': 70.0, 'T': 70.0},\n",
       "  3: {'A': 1.0, 'C': 1001.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1001.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 218.0, 'G': 1.0, 'T': 784.0},\n",
       "  6: {'A': 790.0, 'C': 1.0, 'G': 212.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1001.0, 'T': 1.0},\n",
       "  9: {'A': 51.0, 'C': 51.0, 'G': 156.0, 'T': 746.0},\n",
       "  10: {'A': 182.0, 'C': 182.0, 'G': 273.0, 'T': 366.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '10594029',\n",
       "  'uniprot_ids': 'Q9D030',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1648.1',\n",
       "  'Matrix_Name': 'TCF12(var.2)',\n",
       "  1: {'A': 13015.0, 'C': 19394.0, 'G': 16997.0, 'T': 12474.0},\n",
       "  2: {'A': 14836.0, 'C': 17595.0, 'G': 18561.0, 'T': 10888.0},\n",
       "  3: {'A': 956.0, 'C': 57849.0, 'G': 1187.0, 'T': 1888.0},\n",
       "  4: {'A': 52530.0, 'C': 2008.0, 'G': 5608.0, 'T': 1734.0},\n",
       "  5: {'A': 745.0, 'C': 55599.0, 'G': 3781.0, 'T': 1755.0},\n",
       "  6: {'A': 1914.0, 'C': 55766.0, 'G': 3537.0, 'T': 663.0},\n",
       "  7: {'A': 1381.0, 'C': 3887.0, 'G': 2626.0, 'T': 53986.0},\n",
       "  8: {'A': 2820.0, 'C': 2089.0, 'G': 55923.0, 'T': 1048.0},\n",
       "  9: {'A': 1090.0, 'C': 55062.0, 'G': 2315.0, 'T': 3413.0},\n",
       "  10: {'A': 13868.0, 'C': 17604.0, 'G': 14294.0, 'T': 16114.0},\n",
       "  11: {'A': 11415.0, 'C': 18155.0, 'G': 20061.0, 'T': 12249.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'E2A-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8163514',\n",
       "  'uniprot_ids': 'Q99081',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1645.1',\n",
       "  'Matrix_Name': 'NKX2-2',\n",
       "  1: {'A': 6954.0, 'C': 6960.0, 'G': 4905.0, 'T': 7208.0},\n",
       "  2: {'A': 9465.0, 'C': 5639.0, 'G': 4855.0, 'T': 6068.0},\n",
       "  3: {'A': 7296.0, 'C': 5559.0, 'G': 6330.0, 'T': 6842.0},\n",
       "  4: {'A': 2439.0, 'C': 20233.0, 'G': 2294.0, 'T': 1061.0},\n",
       "  5: {'A': 619.0, 'C': 24342.0, 'G': 177.0, 'T': 889.0},\n",
       "  6: {'A': 23749.0, 'C': 464.0, 'G': 258.0, 'T': 1556.0},\n",
       "  7: {'A': 479.0, 'C': 24694.0, 'G': 421.0, 'T': 433.0},\n",
       "  8: {'A': 491.0, 'C': 453.0, 'G': 296.0, 'T': 24787.0},\n",
       "  9: {'A': 398.0, 'C': 24059.0, 'G': 839.0, 'T': 731.0},\n",
       "  10: {'A': 21327.0, 'C': 2773.0, 'G': 581.0, 'T': 1346.0},\n",
       "  11: {'A': 20004.0, 'C': 2105.0, 'G': 2267.0, 'T': 1651.0},\n",
       "  12: {'A': 6910.0, 'C': 6628.0, 'G': 7590.0, 'T': 4899.0},\n",
       "  13: {'A': 7981.0, 'C': 5943.0, 'G': 5555.0, 'T': 6548.0},\n",
       "  14: {'A': 7414.0, 'C': 6272.0, 'G': 6051.0, 'T': 6290.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'O95096',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0829.2',\n",
       "  'Matrix_Name': 'SREBF1(var.2)',\n",
       "  1: {'A': 2536.0, 'C': 1209.0, 'G': 1992.0, 'T': 1408.0},\n",
       "  2: {'A': 2041.0, 'C': 1660.0, 'G': 2255.0, 'T': 1189.0},\n",
       "  3: {'A': 4922.0, 'C': 485.0, 'G': 1577.0, 'T': 161.0},\n",
       "  4: {'A': 179.0, 'C': 166.0, 'G': 129.0, 'T': 6671.0},\n",
       "  5: {'A': 104.0, 'C': 6771.0, 'G': 159.0, 'T': 111.0},\n",
       "  6: {'A': 6893.0, 'C': 62.0, 'G': 58.0, 'T': 132.0},\n",
       "  7: {'A': 49.0, 'C': 5503.0, 'G': 1468.0, 'T': 125.0},\n",
       "  8: {'A': 192.0, 'C': 4188.0, 'G': 2703.0, 'T': 62.0},\n",
       "  9: {'A': 131.0, 'C': 101.0, 'G': 46.0, 'T': 6867.0},\n",
       "  10: {'A': 64.0, 'C': 222.0, 'G': 6787.0, 'T': 72.0},\n",
       "  11: {'A': 6711.0, 'C': 119.0, 'G': 117.0, 'T': 198.0},\n",
       "  12: {'A': 142.0, 'C': 1457.0, 'G': 511.0, 'T': 5035.0},\n",
       "  13: {'A': 1174.0, 'C': 2262.0, 'G': 1591.0, 'T': 2118.0},\n",
       "  14: {'A': 1371.0, 'C': 2001.0, 'G': 1135.0, 'T': 2638.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11591434',\n",
       "  'uniprot_ids': 'P36956',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0060.3',\n",
       "  'Matrix_Name': 'NFYA',\n",
       "  1: {'A': 1898.0, 'C': 758.0, 'G': 1816.0, 'T': 357.0},\n",
       "  2: {'A': 1925.0, 'C': 405.0, 'G': 1859.0, 'T': 640.0},\n",
       "  3: {'A': 24.0, 'C': 4658.0, 'G': 35.0, 'T': 112.0},\n",
       "  4: {'A': 32.0, 'C': 4714.0, 'G': 29.0, 'T': 54.0},\n",
       "  5: {'A': 4684.0, 'C': 30.0, 'G': 18.0, 'T': 97.0},\n",
       "  6: {'A': 4585.0, 'C': 95.0, 'G': 125.0, 'T': 24.0},\n",
       "  7: {'A': 38.0, 'C': 133.0, 'G': 12.0, 'T': 4646.0},\n",
       "  8: {'A': 167.0, 'C': 4216.0, 'G': 320.0, 'T': 126.0},\n",
       "  9: {'A': 4089.0, 'C': 214.0, 'G': 490.0, 'T': 36.0},\n",
       "  10: {'A': 612.0, 'C': 1132.0, 'G': 2714.0, 'T': 371.0},\n",
       "  11: {'A': 1856.0, 'C': 1370.0, 'G': 888.0, 'T': 715.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'NFY',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '9469818',\n",
       "  'uniprot_ids': 'P23511',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1141.1',\n",
       "  'Matrix_Name': 'FOS::JUND',\n",
       "  1: {'A': 184.0, 'C': 245.0, 'G': 316.0, 'T': 259.0},\n",
       "  2: {'A': 183.0, 'C': 168.0, 'G': 390.0, 'T': 264.0},\n",
       "  3: {'A': 594.0, 'C': 105.0, 'G': 267.0, 'T': 39.0},\n",
       "  4: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 1000.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 872.0, 'T': 130.0},\n",
       "  6: {'A': 895.0, 'C': 92.0, 'G': 1.0, 'T': 16.0},\n",
       "  7: {'A': 76.0, 'C': 41.0, 'G': 868.0, 'T': 20.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  9: {'A': 30.0, 'C': 971.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 22.0, 'C': 291.0, 'G': 78.0, 'T': 613.0},\n",
       "  12: {'A': 251.0, 'C': 371.0, 'G': 196.0, 'T': 186.0},\n",
       "  13: {'A': 265.0, 'C': 265.0, 'G': 293.0, 'T': 181.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P01100;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0087.1',\n",
       "  'Matrix_Name': 'Sox5',\n",
       "  1: {'A': 24.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 23.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 24.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 23.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 23.0},\n",
       "  6: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 22.0},\n",
       "  7: {'A': 9.0, 'C': 4.0, 'G': 5.0, 'T': 9.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '1396566',\n",
       "  'uniprot_ids': 'P35710',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1593.1',\n",
       "  'Matrix_Name': 'ZNF317',\n",
       "  1: {'A': 2960.0, 'C': 1202.0, 'G': 1986.0, 'T': 5319.0},\n",
       "  2: {'A': 4275.0, 'C': 2142.0, 'G': 3671.0, 'T': 1379.0},\n",
       "  3: {'A': 9382.0, 'C': 358.0, 'G': 481.0, 'T': 1246.0},\n",
       "  4: {'A': 282.0, 'C': 10822.0, 'G': 184.0, 'T': 179.0},\n",
       "  5: {'A': 10345.0, 'C': 86.0, 'G': 318.0, 'T': 718.0},\n",
       "  6: {'A': 692.0, 'C': 193.0, 'G': 10375.0, 'T': 207.0},\n",
       "  7: {'A': 396.0, 'C': 10609.0, 'G': 224.0, 'T': 238.0},\n",
       "  8: {'A': 9324.0, 'C': 251.0, 'G': 290.0, 'T': 1602.0},\n",
       "  9: {'A': 285.0, 'C': 187.0, 'G': 10731.0, 'T': 264.0},\n",
       "  10: {'A': 10826.0, 'C': 209.0, 'G': 207.0, 'T': 225.0},\n",
       "  11: {'A': 2073.0, 'C': 4400.0, 'G': 1335.0, 'T': 3659.0},\n",
       "  12: {'A': 3853.0, 'C': 1710.0, 'G': 1482.0, 'T': 4422.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28273063',\n",
       "  'uniprot_ids': 'Q96PQ6',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1597.1',\n",
       "  'Matrix_Name': 'ZNF528',\n",
       "  1: {'A': 85.0, 'C': 381.0, 'G': 67.0, 'T': 87.0},\n",
       "  2: {'A': 70.0, 'C': 368.0, 'G': 88.0, 'T': 94.0},\n",
       "  3: {'A': 101.0, 'C': 405.0, 'G': 65.0, 'T': 49.0},\n",
       "  4: {'A': 525.0, 'C': 13.0, 'G': 62.0, 'T': 20.0},\n",
       "  5: {'A': 33.0, 'C': 4.0, 'G': 574.0, 'T': 9.0},\n",
       "  6: {'A': 8.0, 'C': 6.0, 'G': 599.0, 'T': 7.0},\n",
       "  7: {'A': 23.0, 'C': 3.0, 'G': 591.0, 'T': 3.0},\n",
       "  8: {'A': 609.0, 'C': 4.0, 'G': 4.0, 'T': 3.0},\n",
       "  9: {'A': 580.0, 'C': 5.0, 'G': 27.0, 'T': 8.0},\n",
       "  10: {'A': 6.0, 'C': 2.0, 'G': 609.0, 'T': 3.0},\n",
       "  11: {'A': 36.0, 'C': 415.0, 'G': 55.0, 'T': 114.0},\n",
       "  12: {'A': 16.0, 'C': 591.0, 'G': 4.0, 'T': 9.0},\n",
       "  13: {'A': 607.0, 'C': 3.0, 'G': 8.0, 'T': 2.0},\n",
       "  14: {'A': 12.0, 'C': 48.0, 'G': 12.0, 'T': 548.0},\n",
       "  15: {'A': 2.0, 'C': 98.0, 'G': 15.0, 'T': 505.0},\n",
       "  16: {'A': 12.0, 'C': 130.0, 'G': 63.0, 'T': 415.0},\n",
       "  17: {'A': 17.0, 'C': 559.0, 'G': 9.0, 'T': 35.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '25690854',\n",
       "  'uniprot_ids': 'Q3MIS6',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0501.1',\n",
       "  'Matrix_Name': 'MAF::NFE2',\n",
       "  1: {'A': 727.0, 'C': 27.0, 'G': 335.0, 'T': 5.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1091.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1083.0, 'T': 9.0},\n",
       "  4: {'A': 1091.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 17.0, 'C': 872.0, 'G': 167.0, 'T': 38.0},\n",
       "  6: {'A': 111.0, 'C': 2.0, 'G': 13.0, 'T': 968.0},\n",
       "  7: {'A': 81.0, 'C': 938.0, 'G': 23.0, 'T': 52.0},\n",
       "  8: {'A': 979.0, 'C': 1.0, 'G': 52.0, 'T': 62.0},\n",
       "  9: {'A': 9.0, 'C': 5.0, 'G': 1042.0, 'T': 38.0},\n",
       "  10: {'A': 1.0, 'C': 1064.0, 'G': 23.0, 'T': 6.0},\n",
       "  11: {'A': 932.0, 'C': 18.0, 'G': 74.0, 'T': 70.0},\n",
       "  12: {'A': 384.0, 'C': 204.0, 'G': 284.0, 'T': 222.0},\n",
       "  13: {'A': 362.0, 'C': 101.0, 'G': 131.0, 'T': 500.0},\n",
       "  14: {'A': 293.0, 'C': 103.0, 'G': 92.0, 'T': 606.0},\n",
       "  15: {'A': 212.0, 'C': 284.0, 'G': 107.0, 'T': 491.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9166829',\n",
       "  'uniprot_ids': 'O75444;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0613.1',\n",
       "  'Matrix_Name': 'FOXG1',\n",
       "  1: {'A': 375.0, 'C': 2.0, 'G': 624.0, 'T': 2.0},\n",
       "  2: {'A': 2.0, 'C': 2.0, 'G': 2.0, 'T': 998.0},\n",
       "  3: {'A': 998.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  4: {'A': 998.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  5: {'A': 998.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  6: {'A': 2.0, 'C': 998.0, 'G': 2.0, 'T': 2.0},\n",
       "  7: {'A': 998.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  8: {'A': 499.0, 'C': 168.0, 'G': 3.0, 'T': 334.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '15342912',\n",
       "  'uniprot_ids': 'P55316',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0879.1',\n",
       "  'Matrix_Name': 'Dlx1',\n",
       "  1: {'A': 2124.0, 'C': 2826.0, 'G': 1815.0, 'T': 1520.0},\n",
       "  2: {'A': 2585.0, 'C': 2796.0, 'G': 1593.0, 'T': 1312.0},\n",
       "  3: {'A': 2296.0, 'C': 2754.0, 'G': 150.0, 'T': 5529.0},\n",
       "  4: {'A': 8282.0, 'C': 862.0, 'G': 417.0, 'T': 348.0},\n",
       "  5: {'A': 8282.0, 'C': 106.0, 'G': 43.0, 'T': 536.0},\n",
       "  6: {'A': 1003.0, 'C': 42.0, 'G': 50.0, 'T': 8282.0},\n",
       "  7: {'A': 158.0, 'C': 360.0, 'G': 512.0, 'T': 8282.0},\n",
       "  8: {'A': 8282.0, 'C': 104.0, 'G': 37.0, 'T': 1911.0},\n",
       "  9: {'A': 1775.0, 'C': 1895.0, 'G': 2078.0, 'T': 2537.0},\n",
       "  10: {'A': 1614.0, 'C': 3127.0, 'G': 1894.0, 'T': 1649.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q64317',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0619.1',\n",
       "  'Matrix_Name': 'LIN54',\n",
       "  1: {'A': 412.0, 'C': 92.0, 'G': 213.0, 'T': 286.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  4: {'A': 1.0, 'C': 213.0, 'G': 1.0, 'T': 789.0},\n",
       "  5: {'A': 439.0, 'C': 1.0, 'G': 563.0, 'T': 1.0},\n",
       "  6: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 105.0, 'C': 212.0, 'G': 1.0, 'T': 686.0},\n",
       "  9: {'A': 244.0, 'C': 250.0, 'G': 160.0, 'T': 351.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': '',\n",
       "  'tf_class': 'CRC',\n",
       "  'pubmed_ids': '19725879',\n",
       "  'uniprot_ids': 'F1NCE0',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0794.1',\n",
       "  'Matrix_Name': 'PROX1',\n",
       "  1: {'A': 175.0, 'C': 1504.0, 'G': 544.0, 'T': 1148.0},\n",
       "  2: {'A': 3368.0, 'C': 18.0, 'G': 95.0, 'T': 47.0},\n",
       "  3: {'A': 3368.0, 'C': 230.0, 'G': 49.0, 'T': 421.0},\n",
       "  4: {'A': 64.0, 'C': 1.0, 'G': 3368.0, 'T': 2.0},\n",
       "  5: {'A': 3368.0, 'C': 22.0, 'G': 85.0, 'T': 3.0},\n",
       "  6: {'A': 32.0, 'C': 3368.0, 'G': 1.0, 'T': 37.0},\n",
       "  7: {'A': 16.0, 'C': 3.0, 'G': 3368.0, 'T': 300.0},\n",
       "  8: {'A': 12.0, 'C': 1693.0, 'G': 9.0, 'T': 1676.0},\n",
       "  9: {'A': 4.0, 'C': 3368.0, 'G': 13.0, 'T': 19.0},\n",
       "  10: {'A': 5.0, 'C': 4.0, 'G': 129.0, 'T': 3368.0},\n",
       "  11: {'A': 18.0, 'C': 172.0, 'G': 40.0, 'T': 3368.0},\n",
       "  12: {'A': 2134.0, 'C': 554.0, 'G': 608.0, 'T': 75.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'HD-PROS',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '21040746',\n",
       "  'uniprot_ids': 'Q92786',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1480.1',\n",
       "  'Matrix_Name': 'DPRX',\n",
       "  1: {'A': 4697.0, 'C': 1412.0, 'G': 1510.0, 'T': 946.0},\n",
       "  2: {'A': 937.0, 'C': 1012.0, 'G': 8563.0, 'T': 438.0},\n",
       "  3: {'A': 5802.0, 'C': 2762.0, 'G': 111.0, 'T': 371.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 8563.0},\n",
       "  5: {'A': 8563.0, 'C': 1731.0, 'G': 1122.0, 'T': 1072.0},\n",
       "  6: {'A': 8563.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 199.0, 'T': 8563.0},\n",
       "  8: {'A': 1015.0, 'C': 8563.0, 'G': 223.0, 'T': 351.0},\n",
       "  9: {'A': 304.0, 'C': 8563.0, 'G': 755.0, 'T': 3232.0},\n",
       "  10: {'A': 1522.0, 'C': 2971.0, 'G': 2270.0, 'T': 1803.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'A6NFQ7',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0644.1',\n",
       "  'Matrix_Name': 'ESX1',\n",
       "  1: {'A': 10666.0, 'C': 8154.0, 'G': 9486.0, 'T': 7322.0},\n",
       "  2: {'A': 6474.0, 'C': 15224.0, 'G': 5363.0, 'T': 8565.0},\n",
       "  3: {'A': 2885.0, 'C': 17982.0, 'G': 473.0, 'T': 17643.0},\n",
       "  4: {'A': 35624.0, 'C': 3019.0, 'G': 3417.0, 'T': 2401.0},\n",
       "  5: {'A': 35624.0, 'C': 2441.0, 'G': 868.0, 'T': 1090.0},\n",
       "  6: {'A': 757.0, 'C': 996.0, 'G': 5.0, 'T': 35624.0},\n",
       "  7: {'A': 3722.0, 'C': 4546.0, 'G': 1275.0, 'T': 35624.0},\n",
       "  8: {'A': 35624.0, 'C': 478.0, 'G': 450.0, 'T': 2502.0},\n",
       "  9: {'A': 12066.0, 'C': 7660.0, 'G': 10211.0, 'T': 5690.0},\n",
       "  10: {'A': 7587.0, 'C': 14584.0, 'G': 6232.0, 'T': 7224.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q8N693',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0492.1',\n",
       "  'Matrix_Name': 'JUND(var.2)',\n",
       "  1: {'A': 11449.0, 'C': 4327.0, 'G': 7997.0, 'T': 9862.0},\n",
       "  2: {'A': 13437.0, 'C': 3229.0, 'G': 8722.0, 'T': 8247.0},\n",
       "  3: {'A': 11262.0, 'C': 4837.0, 'G': 8111.0, 'T': 9425.0},\n",
       "  4: {'A': 13617.0, 'C': 2453.0, 'G': 15762.0, 'T': 1803.0},\n",
       "  5: {'A': 25348.0, 'C': 170.0, 'G': 8116.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 33632.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 33512.0, 'T': 121.0},\n",
       "  8: {'A': 33632.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 10407.0, 'G': 8288.0, 'T': 14939.0},\n",
       "  10: {'A': 5125.0, 'C': 1825.0, 'G': 26684.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 3936.0, 'G': 1.0, 'T': 29697.0},\n",
       "  12: {'A': 8794.0, 'C': 24698.0, 'G': 142.0, 'T': 1.0},\n",
       "  13: {'A': 33632.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 268.0, 'C': 5471.0, 'G': 2408.0, 'T': 25488.0},\n",
       "  15: {'A': 10256.0, 'C': 10842.0, 'G': 6352.0, 'T': 6185.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21526160',\n",
       "  'uniprot_ids': 'P17535',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0090.3',\n",
       "  'Matrix_Name': 'TEAD1',\n",
       "  1: {'A': 17690.0, 'C': 19273.0, 'G': 14430.0, 'T': 19009.0},\n",
       "  2: {'A': 16652.0, 'C': 22060.0, 'G': 12054.0, 'T': 19636.0},\n",
       "  3: {'A': 58031.0, 'C': 1249.0, 'G': 9180.0, 'T': 1942.0},\n",
       "  4: {'A': 6097.0, 'C': 61277.0, 'G': 1924.0, 'T': 1104.0},\n",
       "  5: {'A': 68107.0, 'C': 647.0, 'G': 947.0, 'T': 701.0},\n",
       "  6: {'A': 536.0, 'C': 701.0, 'G': 647.0, 'T': 68518.0},\n",
       "  7: {'A': 4977.0, 'C': 623.0, 'G': 1211.0, 'T': 63591.0},\n",
       "  8: {'A': 1349.0, 'C': 66954.0, 'G': 766.0, 'T': 1333.0},\n",
       "  9: {'A': 703.0, 'C': 67636.0, 'G': 375.0, 'T': 1688.0},\n",
       "  10: {'A': 57191.0, 'C': 1730.0, 'G': 2052.0, 'T': 9429.0},\n",
       "  11: {'A': 9125.0, 'C': 6216.0, 'G': 47857.0, 'T': 7204.0},\n",
       "  12: {'A': 13253.0, 'C': 20003.0, 'G': 26526.0, 'T': 10620.0},\n",
       "  13: {'A': 19195.0, 'C': 21359.0, 'G': 16345.0, 'T': 13503.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'TEF-1-related',\n",
       "  'tf_class': 'TEA',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P28347',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0640.2',\n",
       "  'Matrix_Name': 'ELF3',\n",
       "  1: {'A': 11698.0, 'C': 12662.0, 'G': 10256.0, 'T': 13890.0},\n",
       "  2: {'A': 9835.0, 'C': 13182.0, 'G': 11895.0, 'T': 13594.0},\n",
       "  3: {'A': 5685.0, 'C': 24732.0, 'G': 6931.0, 'T': 11158.0},\n",
       "  4: {'A': 6457.0, 'C': 27974.0, 'G': 4123.0, 'T': 9952.0},\n",
       "  5: {'A': 40644.0, 'C': 2041.0, 'G': 2616.0, 'T': 3205.0},\n",
       "  6: {'A': 555.0, 'C': 41951.0, 'G': 1366.0, 'T': 4634.0},\n",
       "  7: {'A': 1172.0, 'C': 1369.0, 'G': 769.0, 'T': 45196.0},\n",
       "  8: {'A': 433.0, 'C': 944.0, 'G': 955.0, 'T': 46174.0},\n",
       "  9: {'A': 663.0, 'C': 46482.0, 'G': 567.0, 'T': 794.0},\n",
       "  10: {'A': 448.0, 'C': 47108.0, 'G': 474.0, 'T': 476.0},\n",
       "  11: {'A': 777.0, 'C': 1316.0, 'G': 1990.0, 'T': 44423.0},\n",
       "  12: {'A': 2378.0, 'C': 9279.0, 'G': 28320.0, 'T': 8529.0},\n",
       "  13: {'A': 9011.0, 'C': 13292.0, 'G': 13946.0, 'T': 12257.0},\n",
       "  14: {'A': 10660.0, 'C': 9086.0, 'G': 6002.0, 'T': 22758.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '9234700',\n",
       "  'uniprot_ids': 'P78545',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0910.2',\n",
       "  'Matrix_Name': 'HOXD8',\n",
       "  1: {'A': 538.0, 'C': 406.0, 'G': 2288.0, 'T': 400.0},\n",
       "  2: {'A': 1.0, 'C': 2151.0, 'G': 1.0, 'T': 1481.0},\n",
       "  3: {'A': 3630.0, 'C': 3007.0, 'G': 48.0, 'T': 1.0},\n",
       "  4: {'A': 3630.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3630.0},\n",
       "  6: {'A': 179.0, 'C': 1.0, 'G': 508.0, 'T': 3630.0},\n",
       "  7: {'A': 3630.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 944.0, 'C': 1684.0, 'G': 552.0, 'T': 453.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P13378',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1118.1',\n",
       "  'Matrix_Name': 'SIX1',\n",
       "  1: {'A': 97.0, 'C': 44.0, 'G': 860.0, 'T': 111.0},\n",
       "  2: {'A': 401.0, 'C': 78.0, 'G': 91.0, 'T': 542.0},\n",
       "  3: {'A': 1050.0, 'C': 19.0, 'G': 7.0, 'T': 36.0},\n",
       "  4: {'A': 1077.0, 'C': 9.0, 'G': 11.0, 'T': 15.0},\n",
       "  5: {'A': 44.0, 'C': 960.0, 'G': 13.0, 'T': 95.0},\n",
       "  6: {'A': 113.0, 'C': 872.0, 'G': 32.0, 'T': 95.0},\n",
       "  7: {'A': 52.0, 'C': 47.0, 'G': 31.0, 'T': 982.0},\n",
       "  8: {'A': 49.0, 'C': 10.0, 'G': 1026.0, 'T': 27.0},\n",
       "  9: {'A': 1071.0, 'C': 8.0, 'G': 12.0, 'T': 21.0},\n",
       "  10: {'A': 147.0, 'C': 180.0, 'G': 290.0, 'T': 495.0},\n",
       "  11: {'A': 547.0, 'C': 361.0, 'G': 74.0, 'T': 130.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HD-SINE',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9826681',\n",
       "  'uniprot_ids': 'Q15475',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1154.1',\n",
       "  'Matrix_Name': 'ZNF282',\n",
       "  1: {'A': 22.0, 'C': 197.0, 'G': 51.0, 'T': 46.0},\n",
       "  2: {'A': 40.0, 'C': 22.0, 'G': 8.0, 'T': 379.0},\n",
       "  3: {'A': 17.0, 'C': 1.0, 'G': 9.0, 'T': 388.0},\n",
       "  4: {'A': 4.0, 'C': 3.0, 'G': 4.0, 'T': 381.0},\n",
       "  5: {'A': 1.0, 'C': 210.0, 'G': 3.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 209.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 1.0, 'C': 206.0, 'G': 4.0, 'T': 3.0},\n",
       "  8: {'A': 141.0, 'C': 100.0, 'G': 1.0, 'T': 22.0},\n",
       "  9: {'A': 13.0, 'C': 133.0, 'G': 3.0, 'T': 127.0},\n",
       "  10: {'A': 257.0, 'C': 4.0, 'G': 38.0, 'T': 62.0},\n",
       "  11: {'A': 278.0, 'C': 1.0, 'G': 1.0, 'T': 3.0},\n",
       "  12: {'A': 3.0, 'C': 211.0, 'G': 5.0, 'T': 4.0},\n",
       "  13: {'A': 243.0, 'C': 45.0, 'G': 3.0, 'T': 8.0},\n",
       "  14: {'A': 2.0, 'C': 211.0, 'G': 3.0, 'T': 3.0},\n",
       "  15: {'A': 15.0, 'C': 1.0, 'G': 298.0, 'T': 130.0},\n",
       "  16: {'A': 158.0, 'C': 41.0, 'G': 61.0, 'T': 44.0},\n",
       "  17: {'A': 45.0, 'C': 99.0, 'G': 98.0, 'T': 52.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28092692',\n",
       "  'uniprot_ids': 'Q9UDV7',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0678.1',\n",
       "  'Matrix_Name': 'OLIG2',\n",
       "  1: {'A': 1610.0, 'C': 91.0, 'G': 284.0, 'T': 109.0},\n",
       "  2: {'A': 681.0, 'C': 930.0, 'G': 143.0, 'T': 17.0},\n",
       "  3: {'A': 75.0, 'C': 1610.0, 'G': 5.0, 'T': 1.0},\n",
       "  4: {'A': 1610.0, 'C': 3.0, 'G': 54.0, 'T': 23.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1610.0},\n",
       "  6: {'A': 1610.0, 'C': 23.0, 'G': 19.0, 'T': 5.0},\n",
       "  7: {'A': 22.0, 'C': 160.0, 'G': 1.0, 'T': 1610.0},\n",
       "  8: {'A': 11.0, 'C': 4.0, 'G': 1610.0, 'T': 118.0},\n",
       "  9: {'A': 5.0, 'C': 397.0, 'G': 867.0, 'T': 745.0},\n",
       "  10: {'A': 147.0, 'C': 690.0, 'G': 58.0, 'T': 1610.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20959288',\n",
       "  'uniprot_ids': 'Q13516',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1541.1',\n",
       "  'Matrix_Name': 'NR6A1',\n",
       "  1: {'A': 172.0, 'C': 198.0, 'G': 314.0, 'T': 217.0},\n",
       "  2: {'A': 256.0, 'C': 474.0, 'G': 349.0, 'T': 897.0},\n",
       "  3: {'A': 125.0, 'C': 897.0, 'G': 343.0, 'T': 133.0},\n",
       "  4: {'A': 897.0, 'C': 5.0, 'G': 45.0, 'T': 12.0},\n",
       "  5: {'A': 897.0, 'C': 92.0, 'G': 73.0, 'T': 52.0},\n",
       "  6: {'A': 14.0, 'C': 10.0, 'G': 897.0, 'T': 63.0},\n",
       "  7: {'A': 20.0, 'C': 13.0, 'G': 444.0, 'T': 454.0},\n",
       "  8: {'A': 2.0, 'C': 75.0, 'G': 27.0, 'T': 897.0},\n",
       "  9: {'A': 8.0, 'C': 897.0, 'G': 30.0, 'T': 41.0},\n",
       "  10: {'A': 897.0, 'C': 1.0, 'G': 3.0, 'T': 15.0},\n",
       "  11: {'A': 897.0, 'C': 97.0, 'G': 36.0, 'T': 66.0},\n",
       "  12: {'A': 22.0, 'C': 16.0, 'G': 897.0, 'T': 63.0},\n",
       "  13: {'A': 30.0, 'C': 12.0, 'G': 525.0, 'T': 374.0},\n",
       "  14: {'A': 92.0, 'C': 186.0, 'G': 356.0, 'T': 897.0},\n",
       "  15: {'A': 40.0, 'C': 897.0, 'G': 110.0, 'T': 255.0},\n",
       "  16: {'A': 897.0, 'C': 21.0, 'G': 360.0, 'T': 42.0},\n",
       "  17: {'A': 283.0, 'C': 178.0, 'G': 208.0, 'T': 231.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'GCNF-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '11578963',\n",
       "  'uniprot_ids': 'Q15406',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0463.2',\n",
       "  'Matrix_Name': 'BCL6',\n",
       "  1: {'A': 1715.0, 'C': 1346.0, 'G': 1290.0, 'T': 1499.0},\n",
       "  2: {'A': 1157.0, 'C': 2488.0, 'G': 900.0, 'T': 4601.0},\n",
       "  3: {'A': 976.0, 'C': 145.0, 'G': 5570.0, 'T': 377.0},\n",
       "  4: {'A': 196.0, 'C': 5788.0, 'G': 107.0, 'T': 150.0},\n",
       "  5: {'A': 355.0, 'C': 258.0, 'G': 72.0, 'T': 5757.0},\n",
       "  6: {'A': 648.0, 'C': 790.0, 'G': 823.0, 'T': 5332.0},\n",
       "  7: {'A': 48.0, 'C': 8.0, 'G': 6.0, 'T': 5832.0},\n",
       "  8: {'A': 2.0, 'C': 5825.0, 'G': 5.0, 'T': 187.0},\n",
       "  9: {'A': 160.0, 'C': 979.0, 'G': 4641.0, 'T': 3169.0},\n",
       "  10: {'A': 5715.0, 'C': 113.0, 'G': 76.0, 'T': 220.0},\n",
       "  11: {'A': 641.0, 'C': 45.0, 'G': 5693.0, 'T': 124.0},\n",
       "  12: {'A': 35.0, 'C': 88.0, 'G': 5792.0, 'T': 205.0},\n",
       "  13: {'A': 5617.0, 'C': 154.0, 'G': 461.0, 'T': 339.0},\n",
       "  14: {'A': 5657.0, 'C': 89.0, 'G': 137.0, 'T': 592.0},\n",
       "  15: {'A': 1569.0, 'C': 1911.0, 'G': 312.0, 'T': 3516.0},\n",
       "  16: {'A': 1300.0, 'C': 1067.0, 'G': 1578.0, 'T': 1904.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7945383',\n",
       "  'uniprot_ids': 'P41182',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1500.1',\n",
       "  'Matrix_Name': 'HOXB6',\n",
       "  1: {'A': 1358.0, 'C': 2469.0, 'G': 2396.0, 'T': 3725.0},\n",
       "  2: {'A': 1.0, 'C': 711.0, 'G': 1.0, 'T': 6193.0},\n",
       "  3: {'A': 6193.0, 'C': 648.0, 'G': 1.0, 'T': 565.0},\n",
       "  4: {'A': 6193.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 437.0, 'G': 728.0, 'T': 6193.0},\n",
       "  6: {'A': 404.0, 'C': 927.0, 'G': 4140.0, 'T': 6193.0},\n",
       "  7: {'A': 3129.0, 'C': 1.0, 'G': 3065.0, 'T': 1.0},\n",
       "  8: {'A': 692.0, 'C': 4869.0, 'G': 997.0, 'T': 1325.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P17509',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1548.1',\n",
       "  'Matrix_Name': 'PLAGL2',\n",
       "  1: {'A': 225.0, 'C': 225.0, 'G': 225.0, 'T': 328.0},\n",
       "  2: {'A': 40.0, 'C': 40.0, 'G': 814.0, 'T': 108.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1000.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 152.0, 'G': 849.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 800.0, 'G': 201.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 926.0, 'G': 1.0, 'T': 75.0},\n",
       "  9: {'A': 91.0, 'C': 685.0, 'G': 26.0, 'T': 199.0},\n",
       "  10: {'A': 211.0, 'C': 211.0, 'G': 211.0, 'T': 370.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11832486',\n",
       "  'uniprot_ids': 'Q9UPG8',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1589.1',\n",
       "  'Matrix_Name': 'ZNF140',\n",
       "  1: {'A': 326.0, 'C': 273.0, 'G': 259.0, 'T': 986.0},\n",
       "  2: {'A': 1141.0, 'C': 174.0, 'G': 342.0, 'T': 187.0},\n",
       "  3: {'A': 280.0, 'C': 39.0, 'G': 1224.0, 'T': 301.0},\n",
       "  4: {'A': 30.0, 'C': 11.0, 'G': 1789.0, 'T': 14.0},\n",
       "  5: {'A': 1709.0, 'C': 15.0, 'G': 109.0, 'T': 11.0},\n",
       "  6: {'A': 27.0, 'C': 12.0, 'G': 1788.0, 'T': 17.0},\n",
       "  7: {'A': 11.0, 'C': 876.0, 'G': 37.0, 'T': 920.0},\n",
       "  8: {'A': 523.0, 'C': 7.0, 'G': 1195.0, 'T': 119.0},\n",
       "  9: {'A': 12.0, 'C': 2.0, 'G': 1813.0, 'T': 17.0},\n",
       "  10: {'A': 1824.0, 'C': 3.0, 'G': 8.0, 'T': 9.0},\n",
       "  11: {'A': 1817.0, 'C': 10.0, 'G': 13.0, 'T': 4.0},\n",
       "  12: {'A': 9.0, 'C': 88.0, 'G': 28.0, 'T': 1719.0},\n",
       "  13: {'A': 72.0, 'C': 34.0, 'G': 49.0, 'T': 1689.0},\n",
       "  14: {'A': 20.0, 'C': 11.0, 'G': 1807.0, 'T': 6.0},\n",
       "  15: {'A': 30.0, 'C': 1607.0, 'G': 23.0, 'T': 184.0},\n",
       "  16: {'A': 222.0, 'C': 129.0, 'G': 70.0, 'T': 1423.0},\n",
       "  17: {'A': 311.0, 'C': 132.0, 'G': 1242.0, 'T': 159.0},\n",
       "  18: {'A': 339.0, 'C': 160.0, 'G': 1169.0, 'T': 176.0},\n",
       "  19: {'A': 456.0, 'C': 191.0, 'G': 1006.0, 'T': 191.0},\n",
       "  20: {'A': 235.0, 'C': 291.0, 'G': 249.0, 'T': 1069.0},\n",
       "  21: {'A': 281.0, 'C': 967.0, 'G': 254.0, 'T': 342.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 21,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'P52738',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1531.1',\n",
       "  'Matrix_Name': 'NR1D1',\n",
       "  1: {'A': 3234.0, 'C': 359.0, 'G': 3363.0, 'T': 44.0},\n",
       "  2: {'A': 17.0, 'C': 3.0, 'G': 6596.0, 'T': 724.0},\n",
       "  3: {'A': 52.0, 'C': 1.0, 'G': 6596.0, 'T': 13.0},\n",
       "  4: {'A': 1.0, 'C': 207.0, 'G': 1648.0, 'T': 6596.0},\n",
       "  5: {'A': 1.0, 'C': 6596.0, 'G': 3.0, 'T': 4244.0},\n",
       "  6: {'A': 6596.0, 'C': 3.0, 'G': 3.0, 'T': 8.0},\n",
       "  7: {'A': 626.0, 'C': 1242.0, 'G': 4429.0, 'T': 302.0},\n",
       "  8: {'A': 21.0, 'C': 118.0, 'G': 214.0, 'T': 6596.0},\n",
       "  9: {'A': 3577.0, 'C': 93.0, 'G': 3019.0, 'T': 12.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 6596.0, 'T': 1114.0},\n",
       "  11: {'A': 9.0, 'C': 1.0, 'G': 6596.0, 'T': 7.0},\n",
       "  12: {'A': 214.0, 'C': 430.0, 'G': 701.0, 'T': 6596.0},\n",
       "  13: {'A': 2.0, 'C': 6596.0, 'G': 90.0, 'T': 345.0},\n",
       "  14: {'A': 6596.0, 'C': 5.0, 'G': 918.0, 'T': 21.0},\n",
       "  15: {'A': 1675.0, 'C': 1533.0, 'G': 1810.0, 'T': 1581.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'P20393',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0475.2',\n",
       "  'Matrix_Name': 'FLI1',\n",
       "  1: {'A': 43587.0, 'C': 1461.0, 'G': 5133.0, 'T': 4484.0},\n",
       "  2: {'A': 2927.0, 'C': 43587.0, 'G': 1859.0, 'T': 293.0},\n",
       "  3: {'A': 2807.0, 'C': 43587.0, 'G': 91.0, 'T': 1.0},\n",
       "  4: {'A': 6.0, 'C': 1.0, 'G': 43587.0, 'T': 3.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 43587.0, 'T': 61.0},\n",
       "  6: {'A': 43587.0, 'C': 22.0, 'G': 20.0, 'T': 12.0},\n",
       "  7: {'A': 43587.0, 'C': 231.0, 'G': 25.0, 'T': 6066.0},\n",
       "  8: {'A': 19103.0, 'C': 564.0, 'G': 24485.0, 'T': 84.0},\n",
       "  9: {'A': 898.0, 'C': 11174.0, 'G': 1987.0, 'T': 32414.0},\n",
       "  10: {'A': 10643.0, 'C': 7717.0, 'G': 18561.0, 'T': 6669.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '7517940',\n",
       "  'uniprot_ids': 'Q01543',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0847.2',\n",
       "  'Matrix_Name': 'FOXD2',\n",
       "  1: {'A': 1405.0, 'C': 1835.0, 'G': 2315.0, 'T': 443.0},\n",
       "  2: {'A': 866.0, 'C': 2407.0, 'G': 310.0, 'T': 3590.0},\n",
       "  3: {'A': 2184.0, 'C': 37.0, 'G': 4.0, 'T': 3813.0},\n",
       "  4: {'A': 5996.0, 'C': 93.0, 'G': 1295.0, 'T': 24.0},\n",
       "  5: {'A': 2483.0, 'C': 1209.0, 'G': 753.0, 'T': 1554.0},\n",
       "  6: {'A': 2609.0, 'C': 102.0, 'G': 5996.0, 'T': 7.0},\n",
       "  7: {'A': 149.0, 'C': 1006.0, 'G': 1.0, 'T': 5996.0},\n",
       "  8: {'A': 5996.0, 'C': 659.0, 'G': 12.0, 'T': 12.0},\n",
       "  9: {'A': 5996.0, 'C': 210.0, 'G': 18.0, 'T': 12.0},\n",
       "  10: {'A': 5996.0, 'C': 1.0, 'G': 26.0, 'T': 34.0},\n",
       "  11: {'A': 1.0, 'C': 5996.0, 'G': 1.0, 'T': 2298.0},\n",
       "  12: {'A': 5996.0, 'C': 54.0, 'G': 126.0, 'T': 210.0},\n",
       "  13: {'A': 2401.0, 'C': 1004.0, 'G': 516.0, 'T': 2077.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '28520732',\n",
       "  'uniprot_ids': 'O60548',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0664.1',\n",
       "  'Matrix_Name': 'MLXIPL',\n",
       "  1: {'A': 346.0, 'C': 32.0, 'G': 221.0, 'T': 114.0},\n",
       "  2: {'A': 127.0, 'C': 29.0, 'G': 134.0, 'T': 469.0},\n",
       "  3: {'A': 1.0, 'C': 449.0, 'G': 12.0, 'T': 1.0},\n",
       "  4: {'A': 412.0, 'C': 1.0, 'G': 3.0, 'T': 6.0},\n",
       "  5: {'A': 3.0, 'C': 453.0, 'G': 13.0, 'T': 1.0},\n",
       "  6: {'A': 21.0, 'C': 3.0, 'G': 736.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 4.0, 'T': 563.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 728.0, 'T': 1.0},\n",
       "  9: {'A': 357.0, 'C': 111.0, 'G': 88.0, 'T': 114.0},\n",
       "  10: {'A': 139.0, 'C': 182.0, 'G': 54.0, 'T': 443.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '14742444',\n",
       "  'uniprot_ids': 'Q9NP71',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0645.1',\n",
       "  'Matrix_Name': 'ETV6',\n",
       "  1: {'A': 9415.0, 'C': 6563.0, 'G': 4685.0, 'T': 2003.0},\n",
       "  2: {'A': 343.0, 'C': 2395.0, 'G': 4293.0, 'T': 442.0},\n",
       "  3: {'A': 1477.0, 'C': 6380.0, 'G': 398.0, 'T': 309.0},\n",
       "  4: {'A': 8.0, 'C': 1.0, 'G': 6688.0, 'T': 14.0},\n",
       "  5: {'A': 22.0, 'C': 1.0, 'G': 6688.0, 'T': 23.0},\n",
       "  6: {'A': 6688.0, 'C': 1.0, 'G': 8.0, 'T': 1.0},\n",
       "  7: {'A': 6688.0, 'C': 14.0, 'G': 33.0, 'T': 42.0},\n",
       "  8: {'A': 769.0, 'C': 53.0, 'G': 6688.0, 'T': 1.0},\n",
       "  9: {'A': 185.0, 'C': 835.0, 'G': 234.0, 'T': 6688.0},\n",
       "  10: {'A': 2050.0, 'C': 343.0, 'G': 3206.0, 'T': 1092.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20400516',\n",
       "  'uniprot_ids': 'P41212',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0642.1',\n",
       "  'Matrix_Name': 'EN2',\n",
       "  1: {'A': 403.0, 'C': 565.0, 'G': 457.0, 'T': 226.0},\n",
       "  2: {'A': 213.0, 'C': 682.0, 'G': 413.0, 'T': 344.0},\n",
       "  3: {'A': 11.0, 'C': 1025.0, 'G': 36.0, 'T': 625.0},\n",
       "  4: {'A': 1649.0, 'C': 298.0, 'G': 165.0, 'T': 54.0},\n",
       "  5: {'A': 1649.0, 'C': 106.0, 'G': 8.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1649.0},\n",
       "  7: {'A': 1.0, 'C': 421.0, 'G': 204.0, 'T': 1649.0},\n",
       "  8: {'A': 1649.0, 'C': 1.0, 'G': 46.0, 'T': 1.0},\n",
       "  9: {'A': 351.0, 'C': 492.0, 'G': 631.0, 'T': 178.0},\n",
       "  10: {'A': 182.0, 'C': 709.0, 'G': 477.0, 'T': 284.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P19622',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0761.2',\n",
       "  'Matrix_Name': 'ETV1',\n",
       "  1: {'A': 12623.0, 'C': 7469.0, 'G': 9644.0, 'T': 7607.0},\n",
       "  2: {'A': 11360.0, 'C': 8639.0, 'G': 10408.0, 'T': 6936.0},\n",
       "  3: {'A': 22802.0, 'C': 4340.0, 'G': 6016.0, 'T': 4185.0},\n",
       "  4: {'A': 1849.0, 'C': 32140.0, 'G': 2390.0, 'T': 964.0},\n",
       "  5: {'A': 33306.0, 'C': 3065.0, 'G': 574.0, 'T': 398.0},\n",
       "  6: {'A': 232.0, 'C': 276.0, 'G': 36484.0, 'T': 351.0},\n",
       "  7: {'A': 379.0, 'C': 272.0, 'G': 36122.0, 'T': 570.0},\n",
       "  8: {'A': 36131.0, 'C': 472.0, 'G': 376.0, 'T': 364.0},\n",
       "  9: {'A': 33369.0, 'C': 487.0, 'G': 413.0, 'T': 3074.0},\n",
       "  10: {'A': 4236.0, 'C': 1308.0, 'G': 31257.0, 'T': 542.0},\n",
       "  11: {'A': 3344.0, 'C': 3890.0, 'G': 3147.0, 'T': 26962.0},\n",
       "  12: {'A': 7923.0, 'C': 5061.0, 'G': 18968.0, 'T': 5391.0},\n",
       "  13: {'A': 11679.0, 'C': 7476.0, 'G': 10821.0, 'T': 7367.0},\n",
       "  14: {'A': 9862.0, 'C': 9115.0, 'G': 9699.0, 'T': 8667.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '25866208',\n",
       "  'uniprot_ids': 'P50549',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0079.4',\n",
       "  'Matrix_Name': 'SP1',\n",
       "  1: {'A': 7867.0, 'C': 13274.0, 'G': 9692.0, 'T': 18834.0},\n",
       "  2: {'A': 28234.0, 'C': 1873.0, 'G': 1000.0, 'T': 19878.0},\n",
       "  3: {'A': 28896.0, 'C': 1531.0, 'G': 19478.0, 'T': 1185.0},\n",
       "  4: {'A': 14585.0, 'C': 1.0, 'G': 37544.0, 'T': 154.0},\n",
       "  5: {'A': 963.0, 'C': 48855.0, 'G': 83.0, 'T': 125.0},\n",
       "  6: {'A': 56.0, 'C': 49650.0, 'G': 1.0, 'T': 1671.0},\n",
       "  7: {'A': 42155.0, 'C': 12673.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 49661.0, 'G': 1.0, 'T': 32.0},\n",
       "  9: {'A': 960.0, 'C': 1.0, 'G': 48958.0, 'T': 868.0},\n",
       "  10: {'A': 1.0, 'C': 49667.0, 'G': 1.0, 'T': 18.0},\n",
       "  11: {'A': 9.0, 'C': 49663.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 48773.0, 'G': 1.0, 'T': 1441.0},\n",
       "  13: {'A': 24667.0, 'C': 24197.0, 'G': 16.0, 'T': 1374.0},\n",
       "  14: {'A': 1196.0, 'C': 37613.0, 'G': 920.0, 'T': 10718.0},\n",
       "  15: {'A': 12599.0, 'C': 15607.0, 'G': 3045.0, 'T': 18416.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '22021377',\n",
       "  'uniprot_ids': 'P08047',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1547.1',\n",
       "  'Matrix_Name': 'PITX2',\n",
       "  1: {'A': 6051.0, 'C': 5520.0, 'G': 4141.0, 'T': 5437.0},\n",
       "  2: {'A': 260.0, 'C': 1.0, 'G': 1.0, 'T': 10957.0},\n",
       "  3: {'A': 10957.0, 'C': 1.0, 'G': 18.0, 'T': 1.0},\n",
       "  4: {'A': 10957.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 148.0, 'T': 10957.0},\n",
       "  6: {'A': 1.0, 'C': 10957.0, 'G': 1.0, 'T': 1456.0},\n",
       "  7: {'A': 198.0, 'C': 10957.0, 'G': 249.0, 'T': 365.0},\n",
       "  8: {'A': 2120.0, 'C': 5774.0, 'G': 1693.0, 'T': 1372.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9685346',\n",
       "  'uniprot_ids': 'Q99697',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1653.1',\n",
       "  'Matrix_Name': 'ZNF148',\n",
       "  1: {'A': 1400.0, 'C': 4594.0, 'G': 2242.0, 'T': 2967.0},\n",
       "  2: {'A': 1264.0, 'C': 4058.0, 'G': 3216.0, 'T': 2665.0},\n",
       "  3: {'A': 83.0, 'C': 10800.0, 'G': 206.0, 'T': 114.0},\n",
       "  4: {'A': 133.0, 'C': 10397.0, 'G': 472.0, 'T': 201.0},\n",
       "  5: {'A': 137.0, 'C': 10527.0, 'G': 402.0, 'T': 137.0},\n",
       "  6: {'A': 90.0, 'C': 10834.0, 'G': 212.0, 'T': 67.0},\n",
       "  7: {'A': 134.0, 'C': 15.0, 'G': 1013.0, 'T': 10041.0},\n",
       "  8: {'A': 67.0, 'C': 10955.0, 'G': 113.0, 'T': 68.0},\n",
       "  9: {'A': 353.0, 'C': 9923.0, 'G': 461.0, 'T': 466.0},\n",
       "  10: {'A': 443.0, 'C': 9284.0, 'G': 393.0, 'T': 1083.0},\n",
       "  11: {'A': 847.0, 'C': 7240.0, 'G': 1925.0, 'T': 1191.0},\n",
       "  12: {'A': 842.0, 'C': 7115.0, 'G': 2065.0, 'T': 1181.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '15635072',\n",
       "  'uniprot_ids': 'Q9UQR1',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0689.1',\n",
       "  'Matrix_Name': 'TBX20',\n",
       "  1: {'A': 198.0, 'C': 5.0, 'G': 122.0, 'T': 251.0},\n",
       "  2: {'A': 573.0, 'C': 24.0, 'G': 55.0, 'T': 25.0},\n",
       "  3: {'A': 51.0, 'C': 35.0, 'G': 573.0, 'T': 27.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 573.0, 'T': 41.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 11.0, 'T': 573.0},\n",
       "  6: {'A': 1.0, 'C': 2.0, 'G': 573.0, 'T': 5.0},\n",
       "  7: {'A': 9.0, 'C': 2.0, 'G': 12.0, 'T': 573.0},\n",
       "  8: {'A': 49.0, 'C': 11.0, 'G': 573.0, 'T': 83.0},\n",
       "  9: {'A': 573.0, 'C': 2.0, 'G': 7.0, 'T': 11.0},\n",
       "  10: {'A': 573.0, 'C': 17.0, 'G': 54.0, 'T': 171.0},\n",
       "  11: {'A': 155.0, 'C': 61.0, 'G': 301.0, 'T': 60.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'TBX1-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '23326246',\n",
       "  'uniprot_ids': 'Q9UMR3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0866.1',\n",
       "  'Matrix_Name': 'SOX21',\n",
       "  1: {'A': 15807.0, 'C': 84.0, 'G': 200.0, 'T': 107.0},\n",
       "  2: {'A': 15807.0, 'C': 251.0, 'G': 2009.0, 'T': 233.0},\n",
       "  3: {'A': 16.0, 'C': 15807.0, 'G': 29.0, 'T': 25.0},\n",
       "  4: {'A': 15807.0, 'C': 8.0, 'G': 23.0, 'T': 2.0},\n",
       "  5: {'A': 15807.0, 'C': 14.0, 'G': 57.0, 'T': 1.0},\n",
       "  6: {'A': 7.0, 'C': 4.0, 'G': 31.0, 'T': 15807.0},\n",
       "  7: {'A': 349.0, 'C': 16.0, 'G': 9625.0, 'T': 5820.0},\n",
       "  8: {'A': 1910.0, 'C': 1272.0, 'G': 7902.0, 'T': 4726.0},\n",
       "  9: {'A': 129.0, 'C': 7222.0, 'G': 74.0, 'T': 8385.0},\n",
       "  10: {'A': 15807.0, 'C': 21.0, 'G': 41.0, 'T': 2.0},\n",
       "  11: {'A': 65.0, 'C': 10.0, 'G': 10734.0, 'T': 5074.0},\n",
       "  12: {'A': 85.0, 'C': 16.0, 'G': 78.0, 'T': 15807.0},\n",
       "  13: {'A': 11.0, 'C': 5.0, 'G': 15807.0, 'T': 9.0},\n",
       "  14: {'A': 34.0, 'C': 18.0, 'G': 38.0, 'T': 15807.0},\n",
       "  15: {'A': 280.0, 'C': 791.0, 'G': 485.0, 'T': 15807.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '12920151',\n",
       "  'uniprot_ids': 'Q9Y651',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1532.1',\n",
       "  'Matrix_Name': 'NR1D2',\n",
       "  1: {'A': 739.0, 'C': 122.0, 'G': 918.0, 'T': 63.0},\n",
       "  2: {'A': 37.0, 'C': 2.0, 'G': 1656.0, 'T': 558.0},\n",
       "  3: {'A': 44.0, 'C': 5.0, 'G': 1656.0, 'T': 57.0},\n",
       "  4: {'A': 6.0, 'C': 93.0, 'G': 471.0, 'T': 1656.0},\n",
       "  5: {'A': 62.0, 'C': 1656.0, 'G': 24.0, 'T': 2519.0},\n",
       "  6: {'A': 1656.0, 'C': 1.0, 'G': 30.0, 'T': 92.0},\n",
       "  7: {'A': 237.0, 'C': 300.0, 'G': 866.0, 'T': 256.0},\n",
       "  8: {'A': 70.0, 'C': 129.0, 'G': 192.0, 'T': 1656.0},\n",
       "  9: {'A': 766.0, 'C': 35.0, 'G': 891.0, 'T': 41.0},\n",
       "  10: {'A': 15.0, 'C': 1.0, 'G': 1656.0, 'T': 798.0},\n",
       "  11: {'A': 41.0, 'C': 5.0, 'G': 1656.0, 'T': 18.0},\n",
       "  12: {'A': 137.0, 'C': 126.0, 'G': 269.0, 'T': 1656.0},\n",
       "  13: {'A': 33.0, 'C': 1656.0, 'G': 135.0, 'T': 663.0},\n",
       "  14: {'A': 1656.0, 'C': 19.0, 'G': 555.0, 'T': 95.0},\n",
       "  15: {'A': 391.0, 'C': 279.0, 'G': 458.0, 'T': 531.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'Q14995',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0713.1',\n",
       "  'Matrix_Name': 'PHOX2A',\n",
       "  1: {'A': 2165.0, 'C': 1241.0, 'G': 498.0, 'T': 11435.0},\n",
       "  2: {'A': 11435.0, 'C': 226.0, 'G': 1540.0, 'T': 1002.0},\n",
       "  3: {'A': 11435.0, 'C': 1.0, 'G': 35.0, 'T': 6.0},\n",
       "  4: {'A': 1363.0, 'C': 4425.0, 'G': 630.0, 'T': 11435.0},\n",
       "  5: {'A': 640.0, 'C': 5305.0, 'G': 2301.0, 'T': 6131.0},\n",
       "  6: {'A': 1263.0, 'C': 4581.0, 'G': 807.0, 'T': 6856.0},\n",
       "  7: {'A': 11435.0, 'C': 718.0, 'G': 1897.0, 'T': 438.0},\n",
       "  8: {'A': 11435.0, 'C': 200.0, 'G': 310.0, 'T': 133.0},\n",
       "  9: {'A': 1.0, 'C': 11.0, 'G': 3.0, 'T': 11435.0},\n",
       "  10: {'A': 1136.0, 'C': 779.0, 'G': 82.0, 'T': 11435.0},\n",
       "  11: {'A': 11435.0, 'C': 230.0, 'G': 817.0, 'T': 1658.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'O14813',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0841.1',\n",
       "  'Matrix_Name': 'NFE2',\n",
       "  1: {'A': 5590.0, 'C': 7209.0, 'G': 5304.0, 'T': 732.0},\n",
       "  2: {'A': 18832.0, 'C': 10.0, 'G': 1307.0, 'T': 6.0},\n",
       "  3: {'A': 3.0, 'C': 9.0, 'G': 1.0, 'T': 18832.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 18832.0, 'T': 3.0},\n",
       "  5: {'A': 18832.0, 'C': 12.0, 'G': 1.0, 'T': 5.0},\n",
       "  6: {'A': 38.0, 'C': 14361.0, 'G': 4473.0, 'T': 18.0},\n",
       "  7: {'A': 1.0, 'C': 21.0, 'G': 4.0, 'T': 18832.0},\n",
       "  8: {'A': 2.0, 'C': 18832.0, 'G': 1.0, 'T': 4.0},\n",
       "  9: {'A': 18832.0, 'C': 14.0, 'G': 9.0, 'T': 6.0},\n",
       "  10: {'A': 1.0, 'C': 2690.0, 'G': 11.0, 'T': 18832.0},\n",
       "  11: {'A': 488.0, 'C': 9861.0, 'G': 5068.0, 'T': 3418.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '10498903',\n",
       "  'uniprot_ids': 'Q16621',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0477.2',\n",
       "  'Matrix_Name': 'FOSL1',\n",
       "  1: {'A': 14980.0, 'C': 12492.0, 'G': 14849.0, 'T': 15364.0},\n",
       "  2: {'A': 15922.0, 'C': 11019.0, 'G': 16856.0, 'T': 13888.0},\n",
       "  3: {'A': 35057.0, 'C': 6704.0, 'G': 13128.0, 'T': 2796.0},\n",
       "  4: {'A': 261.0, 'C': 461.0, 'G': 184.0, 'T': 56779.0},\n",
       "  5: {'A': 516.0, 'C': 302.0, 'G': 54620.0, 'T': 2247.0},\n",
       "  6: {'A': 56807.0, 'C': 288.0, 'G': 287.0, 'T': 303.0},\n",
       "  7: {'A': 2558.0, 'C': 43919.0, 'G': 8616.0, 'T': 2592.0},\n",
       "  8: {'A': 1061.0, 'C': 164.0, 'G': 599.0, 'T': 55861.0},\n",
       "  9: {'A': 3134.0, 'C': 53712.0, 'G': 279.0, 'T': 560.0},\n",
       "  10: {'A': 56409.0, 'C': 345.0, 'G': 567.0, 'T': 364.0},\n",
       "  11: {'A': 2717.0, 'C': 10400.0, 'G': 6534.0, 'T': 38034.0},\n",
       "  12: {'A': 14672.0, 'C': 15641.0, 'G': 12315.0, 'T': 15057.0},\n",
       "  13: {'A': 13957.0, 'C': 17638.0, 'G': 11066.0, 'T': 15024.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P15407',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0828.1',\n",
       "  'Matrix_Name': 'SREBF2(var.2)',\n",
       "  1: {'A': 2899.0, 'C': 285.0, 'G': 1060.0, 'T': 14.0},\n",
       "  2: {'A': 386.0, 'C': 131.0, 'G': 17.0, 'T': 3958.0},\n",
       "  3: {'A': 8.0, 'C': 3958.0, 'G': 17.0, 'T': 2.0},\n",
       "  4: {'A': 3958.0, 'C': 3.0, 'G': 76.0, 'T': 13.0},\n",
       "  5: {'A': 4.0, 'C': 3958.0, 'G': 83.0, 'T': 12.0},\n",
       "  6: {'A': 22.0, 'C': 342.0, 'G': 3958.0, 'T': 1.0},\n",
       "  7: {'A': 45.0, 'C': 241.0, 'G': 1.0, 'T': 3958.0},\n",
       "  8: {'A': 4.0, 'C': 14.0, 'G': 3958.0, 'T': 9.0},\n",
       "  9: {'A': 3958.0, 'C': 9.0, 'G': 40.0, 'T': 123.0},\n",
       "  10: {'A': 21.0, 'C': 2115.0, 'G': 105.0, 'T': 1844.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11591434',\n",
       "  'uniprot_ids': 'Q12772',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0771.1',\n",
       "  'Matrix_Name': 'HSF4',\n",
       "  1: {'A': 1919.0, 'C': 1027.0, 'G': 1117.0, 'T': 10068.0},\n",
       "  2: {'A': 729.0, 'C': 92.0, 'G': 986.0, 'T': 10068.0},\n",
       "  3: {'A': 95.0, 'C': 10068.0, 'G': 112.0, 'T': 71.0},\n",
       "  4: {'A': 63.0, 'C': 4330.0, 'G': 3253.0, 'T': 10068.0},\n",
       "  5: {'A': 10068.0, 'C': 4013.0, 'G': 4791.0, 'T': 34.0},\n",
       "  6: {'A': 4.0, 'C': 13.0, 'G': 10068.0, 'T': 9.0},\n",
       "  7: {'A': 10068.0, 'C': 833.0, 'G': 50.0, 'T': 838.0},\n",
       "  8: {'A': 10068.0, 'C': 415.0, 'G': 370.0, 'T': 1655.0},\n",
       "  9: {'A': 1357.0, 'C': 3687.0, 'G': 2185.0, 'T': 2842.0},\n",
       "  10: {'A': 2603.0, 'C': 2221.0, 'G': 3326.0, 'T': 1921.0},\n",
       "  11: {'A': 1318.0, 'C': 285.0, 'G': 385.0, 'T': 10068.0},\n",
       "  12: {'A': 321.0, 'C': 81.0, 'G': 137.0, 'T': 10068.0},\n",
       "  13: {'A': 99.0, 'C': 10068.0, 'G': 101.0, 'T': 214.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'HSF',\n",
       "  'tf_class': 'Heat',\n",
       "  'pubmed_ids': '25850300',\n",
       "  'uniprot_ids': 'Q9ULV5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0741.1',\n",
       "  'Matrix_Name': 'KLF16',\n",
       "  1: {'A': 667.0, 'C': 536.0, 'G': 1613.0, 'T': 241.0},\n",
       "  2: {'A': 902.0, 'C': 1613.0, 'G': 78.0, 'T': 117.0},\n",
       "  3: {'A': 63.0, 'C': 1613.0, 'G': 7.0, 'T': 46.0},\n",
       "  4: {'A': 1151.0, 'C': 462.0, 'G': 33.0, 'T': 1.0},\n",
       "  5: {'A': 67.0, 'C': 1613.0, 'G': 18.0, 'T': 20.0},\n",
       "  6: {'A': 498.0, 'C': 174.0, 'G': 1613.0, 'T': 404.0},\n",
       "  7: {'A': 9.0, 'C': 1613.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 3.0, 'C': 1613.0, 'G': 9.0, 'T': 1.0},\n",
       "  9: {'A': 17.0, 'C': 1613.0, 'G': 4.0, 'T': 98.0},\n",
       "  10: {'A': 438.0, 'C': 1176.0, 'G': 20.0, 'T': 41.0},\n",
       "  11: {'A': 33.0, 'C': 1613.0, 'G': 24.0, 'T': 371.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16266294',\n",
       "  'uniprot_ids': 'Q9BXK1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0655.1',\n",
       "  'Matrix_Name': 'JDP2',\n",
       "  1: {'A': 1635.0, 'C': 20.0, 'G': 141.0, 'T': 9.0},\n",
       "  2: {'A': 6.0, 'C': 7.0, 'G': 2.0, 'T': 1635.0},\n",
       "  3: {'A': 1.0, 'C': 4.0, 'G': 1635.0, 'T': 23.0},\n",
       "  4: {'A': 1635.0, 'C': 2.0, 'G': 3.0, 'T': 26.0},\n",
       "  5: {'A': 15.0, 'C': 1201.0, 'G': 435.0, 'T': 14.0},\n",
       "  6: {'A': 7.0, 'C': 1.0, 'G': 7.0, 'T': 1635.0},\n",
       "  7: {'A': 23.0, 'C': 1635.0, 'G': 4.0, 'T': 10.0},\n",
       "  8: {'A': 1635.0, 'C': 1.0, 'G': 1.0, 'T': 8.0},\n",
       "  9: {'A': 23.0, 'C': 405.0, 'G': 6.0, 'T': 1635.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9154808',\n",
       "  'uniprot_ids': 'Q8WYK2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0605.2',\n",
       "  'Matrix_Name': 'ATF3',\n",
       "  1: {'A': 8506.0, 'C': 8221.0, 'G': 16895.0, 'T': 6927.0},\n",
       "  2: {'A': 24742.0, 'C': 1355.0, 'G': 15806.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 40547.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 40547.0, 'T': 1367.0},\n",
       "  5: {'A': 40547.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 40547.0, 'G': 1.0, 'T': 557.0},\n",
       "  7: {'A': 892.0, 'C': 1.0, 'G': 40547.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 40547.0},\n",
       "  9: {'A': 1521.0, 'C': 40547.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 40547.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 15738.0, 'G': 1095.0, 'T': 24809.0},\n",
       "  12: {'A': 6920.0, 'C': 16821.0, 'G': 8243.0, 'T': 8565.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '12815047',\n",
       "  'uniprot_ids': 'P18847',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0004.1',\n",
       "  'Matrix_Name': 'Arnt',\n",
       "  1: {'A': 5.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  2: {'A': 20.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 21.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 21.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 21.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 21.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 6,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '7592839',\n",
       "  'uniprot_ids': 'P53762',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0899.1',\n",
       "  'Matrix_Name': 'HOXA10',\n",
       "  1: {'A': 2316.0, 'C': 1424.0, 'G': 2994.0, 'T': 1652.0},\n",
       "  2: {'A': 2426.0, 'C': 1758.0, 'G': 8382.0, 'T': 1342.0},\n",
       "  3: {'A': 154.0, 'C': 3868.0, 'G': 60.0, 'T': 4516.0},\n",
       "  4: {'A': 8382.0, 'C': 4517.0, 'G': 1023.0, 'T': 958.0},\n",
       "  5: {'A': 8382.0, 'C': 456.0, 'G': 719.0, 'T': 268.0},\n",
       "  6: {'A': 368.0, 'C': 12.0, 'G': 1.0, 'T': 8382.0},\n",
       "  7: {'A': 5961.0, 'C': 85.0, 'G': 122.0, 'T': 2423.0},\n",
       "  8: {'A': 8382.0, 'C': 37.0, 'G': 77.0, 'T': 385.0},\n",
       "  9: {'A': 8382.0, 'C': 569.0, 'G': 126.0, 'T': 332.0},\n",
       "  10: {'A': 8382.0, 'C': 2083.0, 'G': 1561.0, 'T': 2059.0},\n",
       "  11: {'A': 3018.0, 'C': 2090.0, 'G': 1231.0, 'T': 2046.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31260',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0504.1',\n",
       "  'Matrix_Name': 'NR2C2',\n",
       "  1: {'A': 130.0, 'C': 127.0, 'G': 91.0, 'T': 51.0},\n",
       "  2: {'A': 83.0, 'C': 71.0, 'G': 222.0, 'T': 23.0},\n",
       "  3: {'A': 194.0, 'C': 7.0, 'G': 196.0, 'T': 2.0},\n",
       "  4: {'A': 1.0, 'C': 9.0, 'G': 367.0, 'T': 22.0},\n",
       "  5: {'A': 11.0, 'C': 4.0, 'G': 356.0, 'T': 28.0},\n",
       "  6: {'A': 4.0, 'C': 1.0, 'G': 80.0, 'T': 314.0},\n",
       "  7: {'A': 7.0, 'C': 310.0, 'G': 55.0, 'T': 27.0},\n",
       "  8: {'A': 344.0, 'C': 1.0, 'G': 49.0, 'T': 5.0},\n",
       "  9: {'A': 188.0, 'C': 1.0, 'G': 209.0, 'T': 1.0},\n",
       "  10: {'A': 325.0, 'C': 1.0, 'G': 72.0, 'T': 1.0},\n",
       "  11: {'A': 11.0, 'C': 1.0, 'G': 386.0, 'T': 1.0},\n",
       "  12: {'A': 12.0, 'C': 1.0, 'G': 337.0, 'T': 49.0},\n",
       "  13: {'A': 1.0, 'C': 28.0, 'G': 90.0, 'T': 280.0},\n",
       "  14: {'A': 10.0, 'C': 316.0, 'G': 36.0, 'T': 37.0},\n",
       "  15: {'A': 291.0, 'C': 1.0, 'G': 93.0, 'T': 14.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '11478808',\n",
       "  'uniprot_ids': 'P49116',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0785.1',\n",
       "  'Matrix_Name': 'POU2F1',\n",
       "  1: {'A': 5077.0, 'C': 1939.0, 'G': 1495.0, 'T': 2076.0},\n",
       "  2: {'A': 5352.0, 'C': 1047.0, 'G': 2143.0, 'T': 5232.0},\n",
       "  3: {'A': 757.0, 'C': 1348.0, 'G': 475.0, 'T': 10583.0},\n",
       "  4: {'A': 10583.0, 'C': 13.0, 'G': 30.0, 'T': 44.0},\n",
       "  5: {'A': 154.0, 'C': 502.0, 'G': 214.0, 'T': 10583.0},\n",
       "  6: {'A': 187.0, 'C': 83.0, 'G': 10583.0, 'T': 1327.0},\n",
       "  7: {'A': 301.0, 'C': 10583.0, 'G': 1352.0, 'T': 3376.0},\n",
       "  8: {'A': 6933.0, 'C': 34.0, 'G': 47.0, 'T': 3652.0},\n",
       "  9: {'A': 10583.0, 'C': 109.0, 'G': 880.0, 'T': 788.0},\n",
       "  10: {'A': 10583.0, 'C': 38.0, 'G': 52.0, 'T': 159.0},\n",
       "  11: {'A': 1423.0, 'C': 458.0, 'G': 756.0, 'T': 10583.0},\n",
       "  12: {'A': 1977.0, 'C': 1841.0, 'G': 2505.0, 'T': 4263.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '1361172',\n",
       "  'uniprot_ids': 'P14859',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0714.1',\n",
       "  'Matrix_Name': 'PITX3',\n",
       "  1: {'A': 1627.0, 'C': 1925.0, 'G': 1635.0, 'T': 1070.0},\n",
       "  2: {'A': 1603.0, 'C': 2900.0, 'G': 807.0, 'T': 3355.0},\n",
       "  3: {'A': 494.0, 'C': 52.0, 'G': 10.0, 'T': 6254.0},\n",
       "  4: {'A': 6254.0, 'C': 33.0, 'G': 261.0, 'T': 159.0},\n",
       "  5: {'A': 6254.0, 'C': 1.0, 'G': 16.0, 'T': 1.0},\n",
       "  6: {'A': 263.0, 'C': 139.0, 'G': 1102.0, 'T': 6254.0},\n",
       "  7: {'A': 319.0, 'C': 6254.0, 'G': 167.0, 'T': 237.0},\n",
       "  8: {'A': 228.0, 'C': 6254.0, 'G': 240.0, 'T': 969.0},\n",
       "  9: {'A': 1139.0, 'C': 2951.0, 'G': 971.0, 'T': 1196.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'O75364',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0146.2',\n",
       "  'Matrix_Name': 'Zfx',\n",
       "  1: {'A': 51.0, 'C': 179.0, 'G': 180.0, 'T': 71.0},\n",
       "  2: {'A': 60.0, 'C': 171.0, 'G': 174.0, 'T': 76.0},\n",
       "  3: {'A': 92.0, 'C': 152.0, 'G': 200.0, 'T': 38.0},\n",
       "  4: {'A': 73.0, 'C': 50.0, 'G': 298.0, 'T': 62.0},\n",
       "  5: {'A': 11.0, 'C': 298.0, 'G': 145.0, 'T': 31.0},\n",
       "  6: {'A': 7.0, 'C': 362.0, 'G': 3.0, 'T': 113.0},\n",
       "  7: {'A': 31.0, 'C': 125.0, 'G': 184.0, 'T': 145.0},\n",
       "  8: {'A': 192.0, 'C': 154.0, 'G': 123.0, 'T': 16.0},\n",
       "  9: {'A': 9.0, 'C': 3.0, 'G': 471.0, 'T': 2.0},\n",
       "  10: {'A': 1.0, 'C': 4.0, 'G': 478.0, 'T': 2.0},\n",
       "  11: {'A': 1.0, 'C': 482.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 481.0, 'G': 1.0, 'T': 2.0},\n",
       "  13: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 481.0},\n",
       "  14: {'A': 85.0, 'C': 123.0, 'G': 219.0, 'T': 57.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '18555785',\n",
       "  'uniprot_ids': 'P17012',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1122.1',\n",
       "  'Matrix_Name': 'TFDP1',\n",
       "  1: {'A': 412.0, 'C': 492.0, 'G': 778.0, 'T': 329.0},\n",
       "  2: {'A': 264.0, 'C': 685.0, 'G': 941.0, 'T': 121.0},\n",
       "  3: {'A': 49.0, 'C': 72.0, 'G': 1863.0, 'T': 27.0},\n",
       "  4: {'A': 29.0, 'C': 1887.0, 'G': 23.0, 'T': 72.0},\n",
       "  5: {'A': 15.0, 'C': 55.0, 'G': 1933.0, 'T': 8.0},\n",
       "  6: {'A': 37.0, 'C': 81.0, 'G': 1866.0, 'T': 27.0},\n",
       "  7: {'A': 28.0, 'C': 33.0, 'G': 1920.0, 'T': 30.0},\n",
       "  8: {'A': 1856.0, 'C': 25.0, 'G': 99.0, 'T': 31.0},\n",
       "  9: {'A': 1646.0, 'C': 171.0, 'G': 138.0, 'T': 56.0},\n",
       "  10: {'A': 466.0, 'C': 648.0, 'G': 649.0, 'T': 248.0},\n",
       "  11: {'A': 382.0, 'C': 605.0, 'G': 649.0, 'T': 375.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '9027491',\n",
       "  'uniprot_ids': 'Q14186',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0677.1',\n",
       "  'Matrix_Name': 'Nr2f6',\n",
       "  1: {'A': 153.0, 'C': 69.0, 'G': 164.0, 'T': 56.0},\n",
       "  2: {'A': 217.0, 'C': 12.0, 'G': 159.0, 'T': 7.0},\n",
       "  3: {'A': 11.0, 'C': 8.0, 'G': 313.0, 'T': 22.0},\n",
       "  4: {'A': 12.0, 'C': 9.0, 'G': 306.0, 'T': 25.0},\n",
       "  5: {'A': 7.0, 'C': 19.0, 'G': 26.0, 'T': 386.0},\n",
       "  6: {'A': 12.0, 'C': 524.0, 'G': 19.0, 'T': 57.0},\n",
       "  7: {'A': 455.0, 'C': 7.0, 'G': 9.0, 'T': 9.0},\n",
       "  8: {'A': 426.0, 'C': 12.0, 'G': 43.0, 'T': 36.0},\n",
       "  9: {'A': 451.0, 'C': 3.0, 'G': 36.0, 'T': 1.0},\n",
       "  10: {'A': 3.0, 'C': 3.0, 'G': 321.0, 'T': 6.0},\n",
       "  11: {'A': 5.0, 'C': 2.0, 'G': 315.0, 'T': 17.0},\n",
       "  12: {'A': 3.0, 'C': 11.0, 'G': 16.0, 'T': 420.0},\n",
       "  13: {'A': 2.0, 'C': 615.0, 'G': 22.0, 'T': 60.0},\n",
       "  14: {'A': 450.0, 'C': 8.0, 'G': 45.0, 'T': 8.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '12690040',\n",
       "  'uniprot_ids': 'P43136',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1513.1',\n",
       "  'Matrix_Name': 'KLF15',\n",
       "  1: {'A': 1185.0, 'C': 2904.0, 'G': 5539.0, 'T': 1745.0},\n",
       "  2: {'A': 938.0, 'C': 6507.0, 'G': 2118.0, 'T': 1810.0},\n",
       "  3: {'A': 41.0, 'C': 11235.0, 'G': 87.0, 'T': 10.0},\n",
       "  4: {'A': 94.0, 'C': 10941.0, 'G': 334.0, 'T': 4.0},\n",
       "  5: {'A': 1.0, 'C': 11369.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 11370.0, 'T': 1.0},\n",
       "  7: {'A': 12.0, 'C': 11346.0, 'G': 3.0, 'T': 12.0},\n",
       "  8: {'A': 56.0, 'C': 10955.0, 'G': 360.0, 'T': 2.0},\n",
       "  9: {'A': 13.0, 'C': 11255.0, 'G': 101.0, 'T': 4.0},\n",
       "  10: {'A': 1567.0, 'C': 6268.0, 'G': 1935.0, 'T': 1603.0},\n",
       "  11: {'A': 1068.0, 'C': 5014.0, 'G': 3125.0, 'T': 2166.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q9UIH9',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0102.4',\n",
       "  'Matrix_Name': 'CEBPA',\n",
       "  1: {'A': 24011.0, 'C': 14824.0, 'G': 15848.0, 'T': 24324.0},\n",
       "  2: {'A': 17689.0, 'C': 12802.0, 'G': 20063.0, 'T': 28453.0},\n",
       "  3: {'A': 54375.0, 'C': 7285.0, 'G': 12773.0, 'T': 4574.0},\n",
       "  4: {'A': 870.0, 'C': 827.0, 'G': 723.0, 'T': 76587.0},\n",
       "  5: {'A': 1003.0, 'C': 283.0, 'G': 2366.0, 'T': 75355.0},\n",
       "  6: {'A': 9253.0, 'C': 937.0, 'G': 57888.0, 'T': 10929.0},\n",
       "  7: {'A': 1149.0, 'C': 75125.0, 'G': 679.0, 'T': 2054.0},\n",
       "  8: {'A': 62410.0, 'C': 3536.0, 'G': 6752.0, 'T': 6309.0},\n",
       "  9: {'A': 6096.0, 'C': 61539.0, 'G': 1584.0, 'T': 9788.0},\n",
       "  10: {'A': 71917.0, 'C': 5656.0, 'G': 341.0, 'T': 1093.0},\n",
       "  11: {'A': 75178.0, 'C': 1035.0, 'G': 1382.0, 'T': 1412.0},\n",
       "  12: {'A': 4507.0, 'C': 16194.0, 'G': 7313.0, 'T': 50993.0},\n",
       "  13: {'A': 28053.0, 'C': 17397.0, 'G': 12933.0, 'T': 20624.0},\n",
       "  14: {'A': 21659.0, 'C': 17148.0, 'G': 16046.0, 'T': 24154.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8632009',\n",
       "  'uniprot_ids': 'P49715',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1136.1',\n",
       "  'Matrix_Name': 'FOSB::JUNB(var.2)',\n",
       "  1: {'A': 562.0, 'C': 60.0, 'G': 369.0, 'T': 13.0},\n",
       "  2: {'A': 31.0, 'C': 37.0, 'G': 17.0, 'T': 918.0},\n",
       "  3: {'A': 14.0, 'C': 47.0, 'G': 701.0, 'T': 243.0},\n",
       "  4: {'A': 870.0, 'C': 45.0, 'G': 61.0, 'T': 29.0},\n",
       "  5: {'A': 31.0, 'C': 843.0, 'G': 28.0, 'T': 102.0},\n",
       "  6: {'A': 53.0, 'C': 36.0, 'G': 894.0, 'T': 21.0},\n",
       "  7: {'A': 57.0, 'C': 81.0, 'G': 29.0, 'T': 837.0},\n",
       "  8: {'A': 174.0, 'C': 777.0, 'G': 37.0, 'T': 16.0},\n",
       "  9: {'A': 924.0, 'C': 17.0, 'G': 31.0, 'T': 32.0},\n",
       "  10: {'A': 12.0, 'C': 261.0, 'G': 66.0, 'T': 665.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P53539;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0707.1',\n",
       "  'Matrix_Name': 'MNX1',\n",
       "  1: {'A': 214.0, 'C': 26.0, 'G': 215.0, 'T': 159.0},\n",
       "  2: {'A': 128.0, 'C': 168.0, 'G': 192.0, 'T': 126.0},\n",
       "  3: {'A': 2.0, 'C': 218.0, 'G': 8.0, 'T': 394.0},\n",
       "  4: {'A': 611.0, 'C': 122.0, 'G': 1.0, 'T': 57.0},\n",
       "  5: {'A': 611.0, 'C': 37.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 9.0, 'C': 44.0, 'G': 1.0, 'T': 611.0},\n",
       "  7: {'A': 141.0, 'C': 22.0, 'G': 66.0, 'T': 611.0},\n",
       "  8: {'A': 611.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 320.0, 'C': 55.0, 'G': 113.0, 'T': 125.0},\n",
       "  10: {'A': 265.0, 'C': 124.0, 'G': 110.0, 'T': 116.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P50219',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0792.1',\n",
       "  'Matrix_Name': 'POU5F1B',\n",
       "  1: {'A': 1327.0, 'C': 1779.0, 'G': 809.0, 'T': 9520.0},\n",
       "  2: {'A': 9520.0, 'C': 22.0, 'G': 68.0, 'T': 85.0},\n",
       "  3: {'A': 172.0, 'C': 365.0, 'G': 207.0, 'T': 9520.0},\n",
       "  4: {'A': 323.0, 'C': 95.0, 'G': 9520.0, 'T': 1791.0},\n",
       "  5: {'A': 399.0, 'C': 9520.0, 'G': 1294.0, 'T': 3829.0},\n",
       "  6: {'A': 6300.0, 'C': 52.0, 'G': 66.0, 'T': 3221.0},\n",
       "  7: {'A': 9520.0, 'C': 145.0, 'G': 857.0, 'T': 897.0},\n",
       "  8: {'A': 9520.0, 'C': 66.0, 'G': 98.0, 'T': 204.0},\n",
       "  9: {'A': 1560.0, 'C': 538.0, 'G': 751.0, 'T': 9520.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '2350782',\n",
       "  'uniprot_ids': 'Q06416',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0469.3',\n",
       "  'Matrix_Name': 'E2F3',\n",
       "  1: {'A': 11775.0, 'C': 2814.0, 'G': 8685.0, 'T': 6154.0},\n",
       "  2: {'A': 9759.0, 'C': 154.0, 'G': 337.0, 'T': 25119.0},\n",
       "  3: {'A': 4707.0, 'C': 82.0, 'G': 84.0, 'T': 28523.0},\n",
       "  4: {'A': 2829.0, 'C': 321.0, 'G': 110.0, 'T': 27784.0},\n",
       "  5: {'A': 2538.0, 'C': 99.0, 'G': 3969.0, 'T': 27497.0},\n",
       "  6: {'A': 73.0, 'C': 3847.0, 'G': 29312.0, 'T': 1.0},\n",
       "  7: {'A': 12.0, 'C': 1.0, 'G': 29422.0, 'T': 13.0},\n",
       "  8: {'A': 18.0, 'C': 29394.0, 'G': 10.0, 'T': 25.0},\n",
       "  9: {'A': 27.0, 'C': 15.0, 'G': 29387.0, 'T': 26.0},\n",
       "  10: {'A': 15.0, 'C': 29428.0, 'G': 1.0, 'T': 9.0},\n",
       "  11: {'A': 1.0, 'C': 29369.0, 'G': 3756.0, 'T': 72.0},\n",
       "  12: {'A': 27526.0, 'C': 3863.0, 'G': 79.0, 'T': 2531.0},\n",
       "  13: {'A': 27861.0, 'C': 114.0, 'G': 273.0, 'T': 2678.0},\n",
       "  14: {'A': 28548.0, 'C': 81.0, 'G': 93.0, 'T': 4553.0},\n",
       "  15: {'A': 25060.0, 'C': 398.0, 'G': 115.0, 'T': 9733.0},\n",
       "  16: {'A': 6057.0, 'C': 8776.0, 'G': 2757.0, 'T': 11839.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '18836037',\n",
       "  'uniprot_ids': 'O00716',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0527.1',\n",
       "  'Matrix_Name': 'ZBTB33',\n",
       "  1: {'A': 83.0, 'C': 257.0, 'G': 147.0, 'T': 222.0},\n",
       "  2: {'A': 7.0, 'C': 67.0, 'G': 33.0, 'T': 602.0},\n",
       "  3: {'A': 29.0, 'C': 660.0, 'G': 12.0, 'T': 8.0},\n",
       "  4: {'A': 11.0, 'C': 90.0, 'G': 6.0, 'T': 602.0},\n",
       "  5: {'A': 1.0, 'C': 706.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 36.0, 'C': 1.0, 'G': 668.0, 'T': 4.0},\n",
       "  7: {'A': 3.0, 'C': 669.0, 'G': 1.0, 'T': 36.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 706.0, 'T': 1.0},\n",
       "  9: {'A': 462.0, 'C': 81.0, 'G': 119.0, 'T': 47.0},\n",
       "  10: {'A': 9.0, 'C': 58.0, 'G': 451.0, 'T': 191.0},\n",
       "  11: {'A': 417.0, 'C': 95.0, 'G': 142.0, 'T': 55.0},\n",
       "  12: {'A': 86.0, 'C': 233.0, 'G': 137.0, 'T': 253.0},\n",
       "  13: {'A': 20.0, 'C': 329.0, 'G': 44.0, 'T': 316.0},\n",
       "  14: {'A': 80.0, 'C': 146.0, 'G': 110.0, 'T': 373.0},\n",
       "  15: {'A': 91.0, 'C': 234.0, 'G': 267.0, 'T': 117.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Other',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23693142',\n",
       "  'uniprot_ids': 'Q86T24',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1544.1',\n",
       "  'Matrix_Name': 'OVOL1',\n",
       "  1: {'A': 6193.0, 'C': 2531.0, 'G': 3016.0, 'T': 3270.0},\n",
       "  2: {'A': 9557.0, 'C': 506.0, 'G': 5451.0, 'T': 4725.0},\n",
       "  3: {'A': 8032.0, 'C': 794.0, 'G': 3335.0, 'T': 6975.0},\n",
       "  4: {'A': 15007.0, 'C': 390.0, 'G': 1.0, 'T': 2101.0},\n",
       "  5: {'A': 1.0, 'C': 15007.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 15007.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 15007.0, 'T': 5.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 5.0, 'T': 15007.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 15007.0},\n",
       "  10: {'A': 15007.0, 'C': 10.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 952.0, 'C': 5768.0, 'G': 2466.0, 'T': 15007.0},\n",
       "  12: {'A': 3356.0, 'C': 3696.0, 'G': 3416.0, 'T': 4542.0},\n",
       "  13: {'A': 1614.0, 'C': 6592.0, 'G': 523.0, 'T': 8416.0},\n",
       "  14: {'A': 2924.0, 'C': 3937.0, 'G': 4873.0, 'T': 3276.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '26504231',\n",
       "  'uniprot_ids': 'O14753',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0821.1',\n",
       "  'Matrix_Name': 'HES5',\n",
       "  1: {'A': 217.0, 'C': 780.0, 'G': 31.0, 'T': 536.0},\n",
       "  2: {'A': 48.0, 'C': 7.0, 'G': 1560.0, 'T': 2.0},\n",
       "  3: {'A': 765.0, 'C': 64.0, 'G': 1560.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1560.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1560.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 2.0, 'C': 1560.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 1560.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 8.0, 'G': 1.0, 'T': 1560.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1560.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1560.0, 'G': 15.0, 'T': 268.0},\n",
       "  11: {'A': 1.0, 'C': 1560.0, 'G': 2.0, 'T': 15.0},\n",
       "  12: {'A': 652.0, 'C': 154.0, 'G': 476.0, 'T': 280.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15186484',\n",
       "  'uniprot_ids': 'Q5TA89',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0745.2',\n",
       "  'Matrix_Name': 'SNAI2',\n",
       "  1: {'A': 14659.0, 'C': 10032.0, 'G': 11170.0, 'T': 9959.0},\n",
       "  2: {'A': 7669.0, 'C': 12315.0, 'G': 9491.0, 'T': 16345.0},\n",
       "  3: {'A': 10323.0, 'C': 6394.0, 'G': 26743.0, 'T': 2360.0},\n",
       "  4: {'A': 409.0, 'C': 44464.0, 'G': 324.0, 'T': 623.0},\n",
       "  5: {'A': 44964.0, 'C': 203.0, 'G': 425.0, 'T': 228.0},\n",
       "  6: {'A': 113.0, 'C': 45145.0, 'G': 327.0, 'T': 235.0},\n",
       "  7: {'A': 247.0, 'C': 44740.0, 'G': 488.0, 'T': 345.0},\n",
       "  8: {'A': 79.0, 'C': 314.0, 'G': 197.0, 'T': 45230.0},\n",
       "  9: {'A': 180.0, 'C': 255.0, 'G': 45258.0, 'T': 127.0},\n",
       "  10: {'A': 815.0, 'C': 6468.0, 'G': 1264.0, 'T': 37273.0},\n",
       "  11: {'A': 12018.0, 'C': 18772.0, 'G': 6282.0, 'T': 8748.0},\n",
       "  12: {'A': 13587.0, 'C': 11017.0, 'G': 12298.0, 'T': 8918.0},\n",
       "  13: {'A': 6216.0, 'C': 13477.0, 'G': 8220.0, 'T': 17907.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'O43623',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0786.1',\n",
       "  'Matrix_Name': 'POU3F1',\n",
       "  1: {'A': 1035.0, 'C': 127.0, 'G': 323.0, 'T': 1438.0},\n",
       "  2: {'A': 506.0, 'C': 187.0, 'G': 129.0, 'T': 2472.0},\n",
       "  3: {'A': 2472.0, 'C': 8.0, 'G': 27.0, 'T': 22.0},\n",
       "  4: {'A': 45.0, 'C': 54.0, 'G': 22.0, 'T': 2472.0},\n",
       "  5: {'A': 38.0, 'C': 14.0, 'G': 2472.0, 'T': 233.0},\n",
       "  6: {'A': 171.0, 'C': 2472.0, 'G': 414.0, 'T': 1120.0},\n",
       "  7: {'A': 1424.0, 'C': 2.0, 'G': 22.0, 'T': 1049.0},\n",
       "  8: {'A': 2472.0, 'C': 104.0, 'G': 131.0, 'T': 285.0},\n",
       "  9: {'A': 2472.0, 'C': 21.0, 'G': 26.0, 'T': 64.0},\n",
       "  10: {'A': 260.0, 'C': 96.0, 'G': 129.0, 'T': 2472.0},\n",
       "  11: {'A': 383.0, 'C': 303.0, 'G': 621.0, 'T': 1168.0},\n",
       "  12: {'A': 1511.0, 'C': 479.0, 'G': 453.0, 'T': 962.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '1361172',\n",
       "  'uniprot_ids': 'Q03052',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0849.1',\n",
       "  'Matrix_Name': 'FOXO6',\n",
       "  1: {'A': 1446.0, 'C': 1.0, 'G': 7026.0, 'T': 1.0},\n",
       "  2: {'A': 155.0, 'C': 1.0, 'G': 33.0, 'T': 7026.0},\n",
       "  3: {'A': 7026.0, 'C': 1148.0, 'G': 8.0, 'T': 1.0},\n",
       "  4: {'A': 7026.0, 'C': 196.0, 'G': 1.0, 'T': 27.0},\n",
       "  5: {'A': 7026.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 7026.0, 'G': 32.0, 'T': 386.0},\n",
       "  7: {'A': 7026.0, 'C': 1.0, 'G': 14.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '10880363',\n",
       "  'uniprot_ids': 'A8MYZ6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0724.1',\n",
       "  'Matrix_Name': 'VENTX',\n",
       "  1: {'A': 17345.0, 'C': 984.0, 'G': 1935.0, 'T': 3663.0},\n",
       "  2: {'A': 9982.0, 'C': 13943.0, 'G': 6942.0, 'T': 5913.0},\n",
       "  3: {'A': 563.0, 'C': 23924.0, 'G': 1.0, 'T': 3965.0},\n",
       "  4: {'A': 7796.0, 'C': 5651.0, 'G': 16129.0, 'T': 627.0},\n",
       "  5: {'A': 23924.0, 'C': 204.0, 'G': 1.0, 'T': 75.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 23924.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 23924.0},\n",
       "  8: {'A': 23924.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 8821.0, 'C': 707.0, 'G': 15104.0, 'T': 2190.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'O95231',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0776.1',\n",
       "  'Matrix_Name': 'MYBL1',\n",
       "  1: {'A': 1592.0, 'C': 26.0, 'G': 392.0, 'T': 202.0},\n",
       "  2: {'A': 348.0, 'C': 2210.0, 'G': 123.0, 'T': 113.0},\n",
       "  3: {'A': 259.0, 'C': 2210.0, 'G': 267.0, 'T': 1.0},\n",
       "  4: {'A': 50.0, 'C': 8.0, 'G': 2210.0, 'T': 5.0},\n",
       "  5: {'A': 10.0, 'C': 27.0, 'G': 1.0, 'T': 2210.0},\n",
       "  6: {'A': 11.0, 'C': 9.0, 'G': 1.0, 'T': 2210.0},\n",
       "  7: {'A': 2210.0, 'C': 1.0, 'G': 13.0, 'T': 15.0},\n",
       "  8: {'A': 2210.0, 'C': 19.0, 'G': 25.0, 'T': 1.0},\n",
       "  9: {'A': 15.0, 'C': 2210.0, 'G': 2.0, 'T': 13.0},\n",
       "  10: {'A': 165.0, 'C': 1734.0, 'G': 2210.0, 'T': 946.0},\n",
       "  11: {'A': 494.0, 'C': 363.0, 'G': 2210.0, 'T': 889.0},\n",
       "  12: {'A': 333.0, 'C': 766.0, 'G': 41.0, 'T': 1073.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Myb/SANT',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '21998171',\n",
       "  'uniprot_ids': 'P10243',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0770.1',\n",
       "  'Matrix_Name': 'HSF2',\n",
       "  1: {'A': 174.0, 'C': 38.0, 'G': 92.0, 'T': 3188.0},\n",
       "  2: {'A': 24.0, 'C': 3.0, 'G': 48.0, 'T': 3188.0},\n",
       "  3: {'A': 17.0, 'C': 3188.0, 'G': 3.0, 'T': 1.0},\n",
       "  4: {'A': 2.0, 'C': 711.0, 'G': 294.0, 'T': 3188.0},\n",
       "  5: {'A': 3188.0, 'C': 566.0, 'G': 793.0, 'T': 1.0},\n",
       "  6: {'A': 3.0, 'C': 1.0, 'G': 3188.0, 'T': 2.0},\n",
       "  7: {'A': 3188.0, 'C': 50.0, 'G': 5.0, 'T': 29.0},\n",
       "  8: {'A': 3188.0, 'C': 2.0, 'G': 24.0, 'T': 109.0},\n",
       "  9: {'A': 171.0, 'C': 1407.0, 'G': 631.0, 'T': 982.0},\n",
       "  10: {'A': 1000.0, 'C': 469.0, 'G': 1210.0, 'T': 512.0},\n",
       "  11: {'A': 143.0, 'C': 69.0, 'G': 89.0, 'T': 3188.0},\n",
       "  12: {'A': 46.0, 'C': 90.0, 'G': 44.0, 'T': 3188.0},\n",
       "  13: {'A': 90.0, 'C': 3188.0, 'G': 118.0, 'T': 365.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'HSF',\n",
       "  'tf_class': 'Heat',\n",
       "  'pubmed_ids': '12027891',\n",
       "  'uniprot_ids': 'Q03933',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0742.1',\n",
       "  'Matrix_Name': 'Klf12',\n",
       "  1: {'A': 168.0, 'C': 36.0, 'G': 1159.0, 'T': 226.0},\n",
       "  2: {'A': 554.0, 'C': 12.0, 'G': 532.0, 'T': 1.0},\n",
       "  3: {'A': 20.0, 'C': 669.0, 'G': 9.0, 'T': 10.0},\n",
       "  4: {'A': 6.0, 'C': 671.0, 'G': 13.0, 'T': 4.0},\n",
       "  5: {'A': 854.0, 'C': 28.0, 'G': 3.0, 'T': 1.0},\n",
       "  6: {'A': 10.0, 'C': 676.0, 'G': 1.0, 'T': 5.0},\n",
       "  7: {'A': 216.0, 'C': 2.0, 'G': 1349.0, 'T': 17.0},\n",
       "  8: {'A': 1.0, 'C': 688.0, 'G': 1.0, 'T': 5.0},\n",
       "  9: {'A': 1.0, 'C': 707.0, 'G': 10.0, 'T': 14.0},\n",
       "  10: {'A': 1.0, 'C': 716.0, 'G': 1.0, 'T': 14.0},\n",
       "  11: {'A': 411.0, 'C': 121.0, 'G': 1.0, 'T': 484.0},\n",
       "  12: {'A': 239.0, 'C': 220.0, 'G': 100.0, 'T': 404.0},\n",
       "  13: {'A': 309.0, 'C': 240.0, 'G': 85.0, 'T': 291.0},\n",
       "  14: {'A': 239.0, 'C': 187.0, 'G': 118.0, 'T': 430.0},\n",
       "  15: {'A': 272.0, 'C': 211.0, 'G': 142.0, 'T': 331.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16266294',\n",
       "  'uniprot_ids': 'O35738',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0729.1',\n",
       "  'Matrix_Name': 'RARA',\n",
       "  1: {'A': 251.0, 'C': 2.0, 'G': 514.0, 'T': 5.0},\n",
       "  2: {'A': 508.0, 'C': 2.0, 'G': 36.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 746.0, 'T': 1.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 747.0, 'T': 20.0},\n",
       "  5: {'A': 1.0, 'C': 6.0, 'G': 2.0, 'T': 560.0},\n",
       "  6: {'A': 3.0, 'C': 708.0, 'G': 27.0, 'T': 1.0},\n",
       "  7: {'A': 529.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  8: {'A': 527.0, 'C': 1.0, 'G': 29.0, 'T': 163.0},\n",
       "  9: {'A': 527.0, 'C': 1.0, 'G': 4.0, 'T': 38.0},\n",
       "  10: {'A': 523.0, 'C': 6.0, 'G': 5.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 760.0, 'T': 2.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 755.0, 'T': 102.0},\n",
       "  13: {'A': 4.0, 'C': 1.0, 'G': 1.0, 'T': 547.0},\n",
       "  14: {'A': 1.0, 'C': 706.0, 'G': 1.0, 'T': 3.0},\n",
       "  15: {'A': 541.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  16: {'A': 540.0, 'C': 37.0, 'G': 31.0, 'T': 182.0},\n",
       "  17: {'A': 3.0, 'C': 1.0, 'G': 229.0, 'T': 329.0},\n",
       "  18: {'A': 1.0, 'C': 116.0, 'G': 227.0, 'T': 213.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P10276',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1144.1',\n",
       "  'Matrix_Name': 'FOSL2::JUND',\n",
       "  1: {'A': 138.0, 'C': 144.0, 'G': 447.0, 'T': 276.0},\n",
       "  2: {'A': 635.0, 'C': 61.0, 'G': 292.0, 'T': 15.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 991.0, 'T': 11.0},\n",
       "  5: {'A': 999.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 47.0, 'C': 605.0, 'G': 293.0, 'T': 59.0},\n",
       "  7: {'A': 7.0, 'C': 1.0, 'G': 61.0, 'T': 935.0},\n",
       "  8: {'A': 141.0, 'C': 861.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 39.0, 'C': 267.0, 'G': 114.0, 'T': 584.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15408;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1652.1',\n",
       "  'Matrix_Name': 'ZKSCAN5',\n",
       "  1: {'A': 871.0, 'C': 637.0, 'G': 1162.0, 'T': 661.0},\n",
       "  2: {'A': 823.0, 'C': 705.0, 'G': 1198.0, 'T': 605.0},\n",
       "  3: {'A': 1511.0, 'C': 316.0, 'G': 802.0, 'T': 702.0},\n",
       "  4: {'A': 137.0, 'C': 31.0, 'G': 3112.0, 'T': 51.0},\n",
       "  5: {'A': 39.0, 'C': 28.0, 'G': 3237.0, 'T': 27.0},\n",
       "  6: {'A': 3200.0, 'C': 63.0, 'G': 48.0, 'T': 20.0},\n",
       "  7: {'A': 1447.0, 'C': 28.0, 'G': 1828.0, 'T': 28.0},\n",
       "  8: {'A': 93.0, 'C': 38.0, 'G': 3150.0, 'T': 50.0},\n",
       "  9: {'A': 74.0, 'C': 87.0, 'G': 77.0, 'T': 3093.0},\n",
       "  10: {'A': 44.0, 'C': 25.0, 'G': 3252.0, 'T': 10.0},\n",
       "  11: {'A': 3200.0, 'C': 42.0, 'G': 63.0, 'T': 26.0},\n",
       "  12: {'A': 376.0, 'C': 371.0, 'G': 2035.0, 'T': 549.0},\n",
       "  13: {'A': 1269.0, 'C': 505.0, 'G': 1079.0, 'T': 478.0},\n",
       "  14: {'A': 1170.0, 'C': 570.0, 'G': 1053.0, 'T': 538.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'Q9Y2L8',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0109.1',\n",
       "  'Matrix_Name': 'HLTF',\n",
       "  1: {'A': 17.0, 'C': 11.0, 'G': 13.0, 'T': 14.0},\n",
       "  2: {'A': 21.0, 'C': 13.0, 'G': 6.0, 'T': 17.0},\n",
       "  3: {'A': 27.0, 'C': 34.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 60.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 26.0, 'C': 1.0, 'G': 1.0, 'T': 35.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 60.0},\n",
       "  7: {'A': 25.0, 'C': 6.0, 'G': 13.0, 'T': 19.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 23.0, 'T': 38.0},\n",
       "  9: {'A': 16.0, 'C': 15.0, 'G': 15.0, 'T': 9.0},\n",
       "  10: {'A': 13.0, 'C': 13.0, 'G': 11.0, 'T': 16.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Myb/SANT',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '12198246',\n",
       "  'uniprot_ids': 'Q95216',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0638.1',\n",
       "  'Matrix_Name': 'CREB3',\n",
       "  1: {'A': 80.0, 'C': 79.0, 'G': 108.0, 'T': 51.0},\n",
       "  2: {'A': 33.0, 'C': 79.0, 'G': 76.0, 'T': 285.0},\n",
       "  3: {'A': 21.0, 'C': 1.0, 'G': 391.0, 'T': 130.0},\n",
       "  4: {'A': 93.0, 'C': 208.0, 'G': 3.0, 'T': 21.0},\n",
       "  5: {'A': 8.0, 'C': 242.0, 'G': 17.0, 'T': 8.0},\n",
       "  6: {'A': 306.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 1.0, 'C': 246.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 4.0, 'C': 1.0, 'G': 448.0, 'T': 3.0},\n",
       "  9: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 377.0},\n",
       "  10: {'A': 11.0, 'C': 246.0, 'G': 9.0, 'T': 5.0},\n",
       "  11: {'A': 301.0, 'C': 3.0, 'G': 13.0, 'T': 7.0},\n",
       "  12: {'A': 8.0, 'C': 116.0, 'G': 29.0, 'T': 162.0},\n",
       "  13: {'A': 66.0, 'C': 211.0, 'G': 36.0, 'T': 60.0},\n",
       "  14: {'A': 120.0, 'C': 51.0, 'G': 99.0, 'T': 44.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15845366',\n",
       "  'uniprot_ids': 'O43889',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0682.2',\n",
       "  'Matrix_Name': 'PITX1',\n",
       "  1: {'A': 2899.0, 'C': 6100.0, 'G': 2383.0, 'T': 4433.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 10532.0},\n",
       "  3: {'A': 10532.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 10532.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1040.0, 'T': 10532.0},\n",
       "  6: {'A': 144.0, 'C': 10532.0, 'G': 1.0, 'T': 142.0},\n",
       "  7: {'A': 69.0, 'C': 10532.0, 'G': 135.0, 'T': 548.0},\n",
       "  8: {'A': 2349.0, 'C': 4195.0, 'G': 1877.0, 'T': 2114.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8675014',\n",
       "  'uniprot_ids': 'P78337',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0873.1',\n",
       "  'Matrix_Name': 'HOXD12',\n",
       "  1: {'A': 630.0, 'C': 156.0, 'G': 526.0, 'T': 59.0},\n",
       "  2: {'A': 486.0, 'C': 388.0, 'G': 1368.0, 'T': 13.0},\n",
       "  3: {'A': 2.0, 'C': 50.0, 'G': 9.0, 'T': 1368.0},\n",
       "  4: {'A': 4.0, 'C': 1368.0, 'G': 6.0, 'T': 1.0},\n",
       "  5: {'A': 208.0, 'C': 3.0, 'G': 1368.0, 'T': 1.0},\n",
       "  6: {'A': 2.0, 'C': 11.0, 'G': 1.0, 'T': 1368.0},\n",
       "  7: {'A': 1368.0, 'C': 1.0, 'G': 3.0, 'T': 223.0},\n",
       "  8: {'A': 1368.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  9: {'A': 1368.0, 'C': 1.0, 'G': 7.0, 'T': 45.0},\n",
       "  10: {'A': 1368.0, 'C': 186.0, 'G': 49.0, 'T': 318.0},\n",
       "  11: {'A': 539.0, 'C': 483.0, 'G': 78.0, 'T': 272.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P35452',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0815.1',\n",
       "  'Matrix_Name': 'TFAP2C(var.3)',\n",
       "  1: {'A': 533.0, 'C': 351.0, 'G': 216.0, 'T': 1467.0},\n",
       "  2: {'A': 1.0, 'C': 660.0, 'G': 2380.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 2497.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 2409.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 18.0, 'C': 2074.0, 'G': 96.0, 'T': 331.0},\n",
       "  6: {'A': 90.0, 'C': 1002.0, 'G': 405.0, 'T': 1218.0},\n",
       "  7: {'A': 366.0, 'C': 914.0, 'G': 925.0, 'T': 426.0},\n",
       "  8: {'A': 1014.0, 'C': 363.0, 'G': 985.0, 'T': 74.0},\n",
       "  9: {'A': 381.0, 'C': 111.0, 'G': 2360.0, 'T': 21.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 3083.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 3138.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 2129.0, 'G': 520.0, 'T': 1.0},\n",
       "  13: {'A': 1283.0, 'C': 175.0, 'G': 344.0, 'T': 436.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'Q92754',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1420.1',\n",
       "  'Matrix_Name': 'IRF5',\n",
       "  1: {'A': 36.0, 'C': 170.0, 'G': 23.0, 'T': 43.0},\n",
       "  2: {'A': 11.0, 'C': 178.0, 'G': 18.0, 'T': 39.0},\n",
       "  3: {'A': 16.0, 'C': 4.0, 'G': 358.0, 'T': 22.0},\n",
       "  4: {'A': 224.0, 'C': 3.0, 'G': 7.0, 'T': 8.0},\n",
       "  5: {'A': 225.0, 'C': 6.0, 'G': 5.0, 'T': 45.0},\n",
       "  6: {'A': 226.0, 'C': 1.0, 'G': 6.0, 'T': 2.0},\n",
       "  7: {'A': 2.0, 'C': 181.0, 'G': 1.0, 'T': 5.0},\n",
       "  8: {'A': 1.0, 'C': 183.0, 'G': 1.0, 'T': 38.0},\n",
       "  9: {'A': 106.0, 'C': 31.0, 'G': 366.0, 'T': 9.0},\n",
       "  10: {'A': 225.0, 'C': 12.0, 'G': 8.0, 'T': 2.0},\n",
       "  11: {'A': 207.0, 'C': 7.0, 'G': 26.0, 'T': 41.0},\n",
       "  12: {'A': 224.0, 'C': 85.0, 'G': 13.0, 'T': 16.0},\n",
       "  13: {'A': 30.0, 'C': 172.0, 'G': 26.0, 'T': 23.0},\n",
       "  14: {'A': 23.0, 'C': 81.0, 'G': 24.0, 'T': 158.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '25896227',\n",
       "  'uniprot_ids': 'Q13568',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1644.1',\n",
       "  'Matrix_Name': 'NFYC',\n",
       "  1: {'A': 5185.0, 'C': 2116.0, 'G': 4644.0, 'T': 1675.0},\n",
       "  2: {'A': 4352.0, 'C': 1747.0, 'G': 4982.0, 'T': 2539.0},\n",
       "  3: {'A': 288.0, 'C': 12508.0, 'G': 395.0, 'T': 429.0},\n",
       "  4: {'A': 381.0, 'C': 12881.0, 'G': 112.0, 'T': 246.0},\n",
       "  5: {'A': 12777.0, 'C': 326.0, 'G': 104.0, 'T': 413.0},\n",
       "  6: {'A': 12318.0, 'C': 377.0, 'G': 550.0, 'T': 375.0},\n",
       "  7: {'A': 260.0, 'C': 375.0, 'G': 72.0, 'T': 12913.0},\n",
       "  8: {'A': 533.0, 'C': 11225.0, 'G': 1465.0, 'T': 397.0},\n",
       "  9: {'A': 11827.0, 'C': 712.0, 'G': 892.0, 'T': 189.0},\n",
       "  10: {'A': 2168.0, 'C': 3709.0, 'G': 5968.0, 'T': 1775.0},\n",
       "  11: {'A': 5182.0, 'C': 3232.0, 'G': 2609.0, 'T': 2597.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': '',\n",
       "  'tf_class': 'Heteromeric',\n",
       "  'pubmed_ids': '11895773',\n",
       "  'uniprot_ids': 'Q13952',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0774.1',\n",
       "  'Matrix_Name': 'MEIS2',\n",
       "  1: {'A': 409.0, 'C': 265.0, 'G': 315.0, 'T': 1352.0},\n",
       "  2: {'A': 32.0, 'C': 4.0, 'G': 41.0, 'T': 1352.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1352.0, 'T': 1.0},\n",
       "  4: {'A': 1352.0, 'C': 4.0, 'G': 26.0, 'T': 63.0},\n",
       "  5: {'A': 10.0, 'C': 1352.0, 'G': 1.0, 'T': 6.0},\n",
       "  6: {'A': 1352.0, 'C': 8.0, 'G': 5.0, 'T': 1.0},\n",
       "  7: {'A': 84.0, 'C': 99.0, 'G': 1352.0, 'T': 68.0},\n",
       "  8: {'A': 158.0, 'C': 702.0, 'G': 651.0, 'T': 89.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9405651',\n",
       "  'uniprot_ids': 'O14770',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0838.1',\n",
       "  'Matrix_Name': 'CEBPG',\n",
       "  1: {'A': 12769.0, 'C': 2442.0, 'G': 5250.0, 'T': 285.0},\n",
       "  2: {'A': 184.0, 'C': 100.0, 'G': 214.0, 'T': 20743.0},\n",
       "  3: {'A': 47.0, 'C': 19.0, 'G': 905.0, 'T': 20743.0},\n",
       "  4: {'A': 8600.0, 'C': 3.0, 'G': 12144.0, 'T': 293.0},\n",
       "  5: {'A': 115.0, 'C': 20743.0, 'G': 31.0, 'T': 450.0},\n",
       "  6: {'A': 1391.0, 'C': 185.0, 'G': 20743.0, 'T': 382.0},\n",
       "  7: {'A': 535.0, 'C': 20743.0, 'G': 45.0, 'T': 1130.0},\n",
       "  8: {'A': 20743.0, 'C': 1.0, 'G': 46.0, 'T': 17.0},\n",
       "  9: {'A': 20743.0, 'C': 1.0, 'G': 24.0, 'T': 47.0},\n",
       "  10: {'A': 160.0, 'C': 8887.0, 'G': 348.0, 'T': 11856.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8632009',\n",
       "  'uniprot_ids': 'P53567',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0149.1',\n",
       "  'Matrix_Name': 'EWSR1-FLI1',\n",
       "  1: {'A': 1.0, 'C': 1.0, 'G': 106.0, 'T': 1.0},\n",
       "  2: {'A': 3.0, 'C': 1.0, 'G': 104.0, 'T': 1.0},\n",
       "  3: {'A': 105.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  4: {'A': 105.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  5: {'A': 2.0, 'C': 1.0, 'G': 105.0, 'T': 1.0},\n",
       "  6: {'A': 3.0, 'C': 1.0, 'G': 103.0, 'T': 2.0},\n",
       "  7: {'A': 104.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  8: {'A': 103.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 105.0, 'T': 2.0},\n",
       "  10: {'A': 1.0, 'C': 3.0, 'G': 104.0, 'T': 1.0},\n",
       "  11: {'A': 100.0, 'C': 5.0, 'G': 3.0, 'T': 1.0},\n",
       "  12: {'A': 106.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 106.0, 'T': 1.0},\n",
       "  14: {'A': 1.0, 'C': 3.0, 'G': 104.0, 'T': 1.0},\n",
       "  15: {'A': 101.0, 'C': 4.0, 'G': 1.0, 'T': 3.0},\n",
       "  16: {'A': 103.0, 'C': 1.0, 'G': 3.0, 'T': 2.0},\n",
       "  17: {'A': 6.0, 'C': 1.0, 'G': 98.0, 'T': 4.0},\n",
       "  18: {'A': 4.0, 'C': 4.0, 'G': 98.0, 'T': 3.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '19305498',\n",
       "  'uniprot_ids': 'F1JVV7;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1139.1',\n",
       "  'Matrix_Name': 'FOSL2::JUNB(var.2)',\n",
       "  1: {'A': 228.0, 'C': 151.0, 'G': 360.0, 'T': 264.0},\n",
       "  2: {'A': 639.0, 'C': 66.0, 'G': 269.0, 'T': 30.0},\n",
       "  3: {'A': 9.0, 'C': 18.0, 'G': 7.0, 'T': 970.0},\n",
       "  4: {'A': 13.0, 'C': 22.0, 'G': 781.0, 'T': 188.0},\n",
       "  5: {'A': 973.0, 'C': 6.0, 'G': 14.0, 'T': 11.0},\n",
       "  6: {'A': 11.0, 'C': 900.0, 'G': 18.0, 'T': 75.0},\n",
       "  7: {'A': 75.0, 'C': 17.0, 'G': 901.0, 'T': 11.0},\n",
       "  8: {'A': 11.0, 'C': 13.0, 'G': 7.0, 'T': 973.0},\n",
       "  9: {'A': 187.0, 'C': 779.0, 'G': 24.0, 'T': 14.0},\n",
       "  10: {'A': 969.0, 'C': 6.0, 'G': 20.0, 'T': 9.0},\n",
       "  11: {'A': 29.0, 'C': 267.0, 'G': 68.0, 'T': 640.0},\n",
       "  12: {'A': 264.0, 'C': 359.0, 'G': 152.0, 'T': 229.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15408;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1135.1',\n",
       "  'Matrix_Name': 'FOSB::JUNB',\n",
       "  1: {'A': 120.0, 'C': 183.0, 'G': 430.0, 'T': 271.0},\n",
       "  2: {'A': 599.0, 'C': 62.0, 'G': 322.0, 'T': 21.0},\n",
       "  3: {'A': 2.0, 'C': 1.0, 'G': 2.0, 'T': 999.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 981.0, 'T': 20.0},\n",
       "  5: {'A': 998.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  6: {'A': 35.0, 'C': 597.0, 'G': 310.0, 'T': 62.0},\n",
       "  7: {'A': 3.0, 'C': 2.0, 'G': 40.0, 'T': 959.0},\n",
       "  8: {'A': 192.0, 'C': 809.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 999.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  10: {'A': 38.0, 'C': 260.0, 'G': 127.0, 'T': 579.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P53539;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1108.2',\n",
       "  'Matrix_Name': 'MXI1',\n",
       "  1: {'A': 5787.0, 'C': 4725.0, 'G': 6327.0, 'T': 5213.0},\n",
       "  2: {'A': 6185.0, 'C': 5842.0, 'G': 6079.0, 'T': 3946.0},\n",
       "  3: {'A': 110.0, 'C': 21383.0, 'G': 98.0, 'T': 461.0},\n",
       "  4: {'A': 21871.0, 'C': 2.0, 'G': 171.0, 'T': 8.0},\n",
       "  5: {'A': 11.0, 'C': 21553.0, 'G': 19.0, 'T': 469.0},\n",
       "  6: {'A': 20507.0, 'C': 1.0, 'G': 1543.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 508.0, 'G': 1.0, 'T': 21542.0},\n",
       "  8: {'A': 528.0, 'C': 115.0, 'G': 21233.0, 'T': 176.0},\n",
       "  9: {'A': 2834.0, 'C': 5330.0, 'G': 7229.0, 'T': 6659.0},\n",
       "  10: {'A': 4946.0, 'C': 6344.0, 'G': 5395.0, 'T': 5367.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8425219',\n",
       "  'uniprot_ids': 'P50539',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1125.1',\n",
       "  'Matrix_Name': 'ZNF384',\n",
       "  1: {'A': 9461.0, 'C': 4946.0, 'G': 5651.0, 'T': 10379.0},\n",
       "  2: {'A': 9117.0, 'C': 7026.0, 'G': 5134.0, 'T': 9160.0},\n",
       "  3: {'A': 9523.0, 'C': 3457.0, 'G': 4679.0, 'T': 12778.0},\n",
       "  4: {'A': 13126.0, 'C': 7692.0, 'G': 4722.0, 'T': 4897.0},\n",
       "  5: {'A': 30355.0, 'C': 8.0, 'G': 14.0, 'T': 60.0},\n",
       "  6: {'A': 30192.0, 'C': 36.0, 'G': 20.0, 'T': 189.0},\n",
       "  7: {'A': 30073.0, 'C': 22.0, 'G': 27.0, 'T': 315.0},\n",
       "  8: {'A': 30253.0, 'C': 14.0, 'G': 16.0, 'T': 154.0},\n",
       "  9: {'A': 30203.0, 'C': 27.0, 'G': 15.0, 'T': 192.0},\n",
       "  10: {'A': 30407.0, 'C': 7.0, 'G': 1.0, 'T': 22.0},\n",
       "  11: {'A': 20323.0, 'C': 2909.0, 'G': 2768.0, 'T': 4437.0},\n",
       "  12: {'A': 18917.0, 'C': 3964.0, 'G': 2899.0, 'T': 4657.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11149472',\n",
       "  'uniprot_ids': 'Q8TF68',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0799.1',\n",
       "  'Matrix_Name': 'RFX4',\n",
       "  1: {'A': 2197.0, 'C': 4158.0, 'G': 2888.0, 'T': 2409.0},\n",
       "  2: {'A': 2260.0, 'C': 18.0, 'G': 15890.0, 'T': 247.0},\n",
       "  3: {'A': 418.0, 'C': 381.0, 'G': 204.0, 'T': 13638.0},\n",
       "  4: {'A': 6539.0, 'C': 2706.0, 'G': 122.0, 'T': 10464.0},\n",
       "  5: {'A': 4223.0, 'C': 465.0, 'G': 6676.0, 'T': 2316.0},\n",
       "  6: {'A': 4.0, 'C': 9324.0, 'G': 19.0, 'T': 1627.0},\n",
       "  7: {'A': 2.0, 'C': 8837.0, 'G': 46.0, 'T': 3473.0},\n",
       "  8: {'A': 10326.0, 'C': 857.0, 'G': 937.0, 'T': 968.0},\n",
       "  9: {'A': 966.0, 'C': 665.0, 'G': 524.0, 'T': 15459.0},\n",
       "  10: {'A': 2945.0, 'C': 69.0, 'G': 14991.0, 'T': 2.0},\n",
       "  11: {'A': 4465.0, 'C': 560.0, 'G': 13779.0, 'T': 108.0},\n",
       "  12: {'A': 1341.0, 'C': 5758.0, 'G': 99.0, 'T': 3911.0},\n",
       "  13: {'A': 8912.0, 'C': 128.0, 'G': 1786.0, 'T': 4987.0},\n",
       "  14: {'A': 9969.0, 'C': 146.0, 'G': 190.0, 'T': 26.0},\n",
       "  15: {'A': 34.0, 'C': 8785.0, 'G': 12.0, 'T': 305.0},\n",
       "  16: {'A': 2001.0, 'C': 3461.0, 'G': 5110.0, 'T': 1854.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'RFX-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '8754849',\n",
       "  'uniprot_ids': 'Q33E94',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0612.2',\n",
       "  'Matrix_Name': 'EMX1',\n",
       "  1: {'A': 1277.0, 'C': 3426.0, 'G': 3067.0, 'T': 1778.0},\n",
       "  2: {'A': 20.0, 'C': 2264.0, 'G': 1.0, 'T': 7282.0},\n",
       "  3: {'A': 9545.0, 'C': 1236.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 9545.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 9545.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 934.0, 'T': 9545.0},\n",
       "  7: {'A': 9545.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1334.0, 'C': 2634.0, 'G': 4517.0, 'T': 1063.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q04741',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1570.1',\n",
       "  'Matrix_Name': 'TFAP4(var.2)',\n",
       "  1: {'A': 480.0, 'C': 131.0, 'G': 198.0, 'T': 40.0},\n",
       "  2: {'A': 218.0, 'C': 145.0, 'G': 1.0, 'T': 130.0},\n",
       "  3: {'A': 1.0, 'C': 480.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 480.0, 'C': 1.0, 'G': 15.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 221.0, 'G': 162.0, 'T': 480.0},\n",
       "  6: {'A': 480.0, 'C': 150.0, 'G': 288.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 480.0},\n",
       "  8: {'A': 1.0, 'C': 4.0, 'G': 480.0, 'T': 1.0},\n",
       "  9: {'A': 104.0, 'C': 15.0, 'G': 157.0, 'T': 207.0},\n",
       "  10: {'A': 61.0, 'C': 211.0, 'G': 111.0, 'T': 480.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9741833',\n",
       "  'uniprot_ids': 'Q01664',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0157.2',\n",
       "  'Matrix_Name': 'FOXO3',\n",
       "  1: {'A': 22.0, 'C': 1.0, 'G': 533.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 14.0, 'T': 533.0},\n",
       "  3: {'A': 533.0, 'C': 17.0, 'G': 6.0, 'T': 2.0},\n",
       "  4: {'A': 533.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 533.0, 'C': 11.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 16.0, 'C': 533.0, 'G': 1.0, 'T': 4.0},\n",
       "  7: {'A': 533.0, 'C': 12.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 318.0, 'C': 13.0, 'G': 37.0, 'T': 216.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'O43524',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0135.1',\n",
       "  'Matrix_Name': 'Lhx3',\n",
       "  1: {'A': 10.0, 'C': 1.0, 'G': 4.0, 'T': 9.0},\n",
       "  2: {'A': 17.0, 'C': 3.0, 'G': 3.0, 'T': 1.0},\n",
       "  3: {'A': 20.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 21.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 21.0},\n",
       "  6: {'A': 20.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 21.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 3.0, 'C': 3.0, 'G': 1.0, 'T': 17.0},\n",
       "  9: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 20.0},\n",
       "  10: {'A': 17.0, 'C': 2.0, 'G': 1.0, 'T': 4.0},\n",
       "  11: {'A': 10.0, 'C': 1.0, 'G': 4.0, 'T': 9.0},\n",
       "  12: {'A': 3.0, 'C': 9.0, 'G': 1.0, 'T': 11.0},\n",
       "  13: {'A': 3.0, 'C': 10.0, 'G': 4.0, 'T': 7.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '11602361',\n",
       "  'uniprot_ids': 'P50481',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1124.1',\n",
       "  'Matrix_Name': 'ZNF24',\n",
       "  1: {'A': 2056.0, 'C': 17057.0, 'G': 1346.0, 'T': 2849.0},\n",
       "  2: {'A': 18029.0, 'C': 1817.0, 'G': 1968.0, 'T': 1494.0},\n",
       "  3: {'A': 959.0, 'C': 2103.0, 'G': 740.0, 'T': 19506.0},\n",
       "  4: {'A': 433.0, 'C': 545.0, 'G': 462.0, 'T': 21868.0},\n",
       "  5: {'A': 327.0, 'C': 22138.0, 'G': 119.0, 'T': 724.0},\n",
       "  6: {'A': 22438.0, 'C': 220.0, 'G': 502.0, 'T': 148.0},\n",
       "  7: {'A': 98.0, 'C': 888.0, 'G': 129.0, 'T': 22193.0},\n",
       "  8: {'A': 183.0, 'C': 331.0, 'G': 183.0, 'T': 22611.0},\n",
       "  9: {'A': 355.0, 'C': 22069.0, 'G': 94.0, 'T': 790.0},\n",
       "  10: {'A': 22035.0, 'C': 403.0, 'G': 590.0, 'T': 280.0},\n",
       "  11: {'A': 1011.0, 'C': 2063.0, 'G': 812.0, 'T': 19422.0},\n",
       "  12: {'A': 986.0, 'C': 2129.0, 'G': 1046.0, 'T': 19147.0},\n",
       "  13: {'A': 2063.0, 'C': 17702.0, 'G': 857.0, 'T': 2686.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11532988',\n",
       "  'uniprot_ids': 'P17028',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0801.1',\n",
       "  'Matrix_Name': 'MGA',\n",
       "  1: {'A': 30490.0, 'C': 1152.0, 'G': 8874.0, 'T': 4161.0},\n",
       "  2: {'A': 4311.0, 'C': 3738.0, 'G': 30490.0, 'T': 1930.0},\n",
       "  3: {'A': 17.0, 'C': 510.0, 'G': 30490.0, 'T': 105.0},\n",
       "  4: {'A': 13.0, 'C': 3535.0, 'G': 97.0, 'T': 30490.0},\n",
       "  5: {'A': 15.0, 'C': 2.0, 'G': 30490.0, 'T': 12.0},\n",
       "  6: {'A': 1043.0, 'C': 3090.0, 'G': 382.0, 'T': 30490.0},\n",
       "  7: {'A': 690.0, 'C': 1754.0, 'G': 30490.0, 'T': 6466.0},\n",
       "  8: {'A': 30490.0, 'C': 111.0, 'G': 921.0, 'T': 275.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TBX6-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'Q8IWI9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0091.1',\n",
       "  'Matrix_Name': 'TAL1::TCF3',\n",
       "  1: {'A': 14.0, 'C': 15.0, 'G': 9.0, 'T': 10.0},\n",
       "  2: {'A': 10.0, 'C': 11.0, 'G': 21.0, 'T': 6.0},\n",
       "  3: {'A': 40.0, 'C': 1.0, 'G': 4.0, 'T': 3.0},\n",
       "  4: {'A': 21.0, 'C': 25.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 44.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 45.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 2.0, 'G': 12.0, 'T': 33.0},\n",
       "  8: {'A': 13.0, 'C': 33.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 45.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 44.0, 'T': 2.0},\n",
       "  11: {'A': 1.0, 'C': 4.0, 'G': 21.0, 'T': 22.0},\n",
       "  12: {'A': 5.0, 'C': 5.0, 'G': 3.0, 'T': 35.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8289805',\n",
       "  'uniprot_ids': 'P17542;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0780.1',\n",
       "  'Matrix_Name': 'PAX3',\n",
       "  1: {'A': 1995.0, 'C': 244.0, 'G': 395.0, 'T': 7946.0},\n",
       "  2: {'A': 7946.0, 'C': 33.0, 'G': 97.0, 'T': 32.0},\n",
       "  3: {'A': 7946.0, 'C': 25.0, 'G': 13.0, 'T': 14.0},\n",
       "  4: {'A': 10.0, 'C': 102.0, 'G': 18.0, 'T': 7946.0},\n",
       "  5: {'A': 13.0, 'C': 5268.0, 'G': 63.0, 'T': 2679.0},\n",
       "  6: {'A': 2685.0, 'C': 49.0, 'G': 5262.0, 'T': 8.0},\n",
       "  7: {'A': 7946.0, 'C': 23.0, 'G': 58.0, 'T': 32.0},\n",
       "  8: {'A': 1.0, 'C': 15.0, 'G': 15.0, 'T': 7946.0},\n",
       "  9: {'A': 59.0, 'C': 98.0, 'G': 43.0, 'T': 7946.0},\n",
       "  10: {'A': 7946.0, 'C': 234.0, 'G': 198.0, 'T': 1062.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '19458195',\n",
       "  'uniprot_ids': 'P23760',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1607.1',\n",
       "  'Matrix_Name': 'Foxl2',\n",
       "  1: {'A': 6236.0, 'C': 3032.0, 'G': 4622.0, 'T': 4215.0},\n",
       "  2: {'A': 5914.0, 'C': 2913.0, 'G': 3522.0, 'T': 5756.0},\n",
       "  3: {'A': 6434.0, 'C': 1577.0, 'G': 1600.0, 'T': 8494.0},\n",
       "  4: {'A': 11159.0, 'C': 2052.0, 'G': 2108.0, 'T': 2786.0},\n",
       "  5: {'A': 2986.0, 'C': 897.0, 'G': 1372.0, 'T': 12850.0},\n",
       "  6: {'A': 2608.0, 'C': 368.0, 'G': 14649.0, 'T': 480.0},\n",
       "  7: {'A': 1098.0, 'C': 302.0, 'G': 420.0, 'T': 16285.0},\n",
       "  8: {'A': 17228.0, 'C': 559.0, 'G': 113.0, 'T': 205.0},\n",
       "  9: {'A': 17213.0, 'C': 337.0, 'G': 188.0, 'T': 367.0},\n",
       "  10: {'A': 17147.0, 'C': 217.0, 'G': 412.0, 'T': 329.0},\n",
       "  11: {'A': 283.0, 'C': 15699.0, 'G': 152.0, 'T': 1971.0},\n",
       "  12: {'A': 15624.0, 'C': 808.0, 'G': 402.0, 'T': 1271.0},\n",
       "  13: {'A': 5230.0, 'C': 4247.0, 'G': 3845.0, 'T': 4783.0},\n",
       "  14: {'A': 5144.0, 'C': 4330.0, 'G': 4530.0, 'T': 4101.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '17192407',\n",
       "  'uniprot_ids': 'O88470',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0616.2',\n",
       "  'Matrix_Name': 'HES2',\n",
       "  1: {'A': 776.0, 'C': 1194.0, 'G': 8484.0, 'T': 799.0},\n",
       "  2: {'A': 1502.0, 'C': 1108.0, 'G': 8484.0, 'T': 1582.0},\n",
       "  3: {'A': 119.0, 'C': 8484.0, 'G': 86.0, 'T': 74.0},\n",
       "  4: {'A': 6456.0, 'C': 1.0, 'G': 2029.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 8484.0, 'G': 1.0, 'T': 130.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 8484.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1099.0, 'G': 1.0, 'T': 8484.0},\n",
       "  8: {'A': 54.0, 'C': 16.0, 'G': 8484.0, 'T': 1.0},\n",
       "  9: {'A': 513.0, 'C': 4635.0, 'G': 1290.0, 'T': 3850.0},\n",
       "  10: {'A': 967.0, 'C': 4465.0, 'G': 1736.0, 'T': 1320.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8354270',\n",
       "  'uniprot_ids': 'Q9Y543',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0679.2',\n",
       "  'Matrix_Name': 'ONECUT1',\n",
       "  1: {'A': 26519.0, 'C': 10014.0, 'G': 11815.0, 'T': 13875.0},\n",
       "  2: {'A': 28609.0, 'C': 8238.0, 'G': 12369.0, 'T': 13007.0},\n",
       "  3: {'A': 30061.0, 'C': 8479.0, 'G': 9657.0, 'T': 14026.0},\n",
       "  4: {'A': 37927.0, 'C': 7556.0, 'G': 6150.0, 'T': 10590.0},\n",
       "  5: {'A': 52147.0, 'C': 1012.0, 'G': 7795.0, 'T': 1269.0},\n",
       "  6: {'A': 60016.0, 'C': 359.0, 'G': 1167.0, 'T': 681.0},\n",
       "  7: {'A': 483.0, 'C': 295.0, 'G': 242.0, 'T': 61203.0},\n",
       "  8: {'A': 1346.0, 'C': 60182.0, 'G': 279.0, 'T': 416.0},\n",
       "  9: {'A': 59596.0, 'C': 439.0, 'G': 1227.0, 'T': 961.0},\n",
       "  10: {'A': 60871.0, 'C': 248.0, 'G': 367.0, 'T': 737.0},\n",
       "  11: {'A': 226.0, 'C': 357.0, 'G': 163.0, 'T': 61477.0},\n",
       "  12: {'A': 48525.0, 'C': 3452.0, 'G': 6073.0, 'T': 4173.0},\n",
       "  13: {'A': 31040.0, 'C': 10187.0, 'G': 7661.0, 'T': 13335.0},\n",
       "  14: {'A': 25105.0, 'C': 10402.0, 'G': 8009.0, 'T': 18707.0},\n",
       "  15: {'A': 18745.0, 'C': 10390.0, 'G': 10130.0, 'T': 22958.0},\n",
       "  16: {'A': 19933.0, 'C': 11848.0, 'G': 11207.0, 'T': 19235.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'HD-CUT',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8887657',\n",
       "  'uniprot_ids': 'Q9UBC0',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0036.3',\n",
       "  'Matrix_Name': 'GATA2',\n",
       "  1: {'A': 3304.0, 'C': 2979.0, 'G': 3202.0, 'T': 4732.0},\n",
       "  2: {'A': 1987.0, 'C': 4100.0, 'G': 2967.0, 'T': 5163.0},\n",
       "  3: {'A': 941.0, 'C': 10608.0, 'G': 1224.0, 'T': 1444.0},\n",
       "  4: {'A': 445.0, 'C': 260.0, 'G': 151.0, 'T': 13361.0},\n",
       "  5: {'A': 418.0, 'C': 35.0, 'G': 22.0, 'T': 13742.0},\n",
       "  6: {'A': 14099.0, 'C': 15.0, 'G': 44.0, 'T': 59.0},\n",
       "  7: {'A': 146.0, 'C': 78.0, 'G': 49.0, 'T': 13944.0},\n",
       "  8: {'A': 97.0, 'C': 13787.0, 'G': 127.0, 'T': 206.0},\n",
       "  9: {'A': 2374.0, 'C': 493.0, 'G': 368.0, 'T': 10982.0},\n",
       "  10: {'A': 2913.0, 'C': 3715.0, 'G': 3657.0, 'T': 3932.0},\n",
       "  11: {'A': 3279.0, 'C': 3875.0, 'G': 2217.0, 'T': 4846.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '8321207',\n",
       "  'uniprot_ids': 'P23769',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0806.1',\n",
       "  'Matrix_Name': 'TBX4',\n",
       "  1: {'A': 18647.0, 'C': 438.0, 'G': 2354.0, 'T': 1284.0},\n",
       "  2: {'A': 3083.0, 'C': 2329.0, 'G': 18647.0, 'T': 1163.0},\n",
       "  3: {'A': 39.0, 'C': 190.0, 'G': 18647.0, 'T': 34.0},\n",
       "  4: {'A': 1.0, 'C': 1645.0, 'G': 192.0, 'T': 18647.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 18647.0, 'T': 1.0},\n",
       "  6: {'A': 1552.0, 'C': 2860.0, 'G': 305.0, 'T': 18647.0},\n",
       "  7: {'A': 1356.0, 'C': 5397.0, 'G': 18647.0, 'T': 8385.0},\n",
       "  8: {'A': 18647.0, 'C': 589.0, 'G': 1734.0, 'T': 966.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TBX2-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'P57082',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0881.1',\n",
       "  'Matrix_Name': 'Dlx4',\n",
       "  1: {'A': 2638.0, 'C': 3929.0, 'G': 3680.0, 'T': 1812.0},\n",
       "  2: {'A': 725.0, 'C': 6842.0, 'G': 146.0, 'T': 5214.0},\n",
       "  3: {'A': 12055.0, 'C': 1501.0, 'G': 740.0, 'T': 884.0},\n",
       "  4: {'A': 12055.0, 'C': 745.0, 'G': 56.0, 'T': 500.0},\n",
       "  5: {'A': 89.0, 'C': 86.0, 'G': 39.0, 'T': 12055.0},\n",
       "  6: {'A': 441.0, 'C': 869.0, 'G': 543.0, 'T': 12055.0},\n",
       "  7: {'A': 12055.0, 'C': 317.0, 'G': 135.0, 'T': 1128.0},\n",
       "  8: {'A': 3066.0, 'C': 3704.0, 'G': 2287.0, 'T': 3000.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P70436',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0606.1',\n",
       "  'Matrix_Name': 'NFAT5',\n",
       "  1: {'A': 372.0, 'C': 211.0, 'G': 211.0, 'T': 211.0},\n",
       "  2: {'A': 177.0, 'C': 89.0, 'G': 89.0, 'T': 650.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  5: {'A': 90.0, 'C': 85.0, 'G': 1.0, 'T': 828.0},\n",
       "  6: {'A': 1.0, 'C': 916.0, 'G': 1.0, 'T': 86.0},\n",
       "  7: {'A': 1.0, 'C': 827.0, 'G': 86.0, 'T': 90.0},\n",
       "  8: {'A': 832.0, 'C': 85.0, 'G': 1.0, 'T': 86.0},\n",
       "  9: {'A': 125.0, 'C': 288.0, 'G': 42.0, 'T': 549.0},\n",
       "  10: {'A': 269.0, 'C': 164.0, 'G': 249.0, 'T': 323.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NFAT-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '12374807',\n",
       "  'uniprot_ids': 'A0A1D5NT38',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0144.2',\n",
       "  'Matrix_Name': 'STAT3',\n",
       "  1: {'A': 1937.0, 'C': 8572.0, 'G': 3802.0, 'T': 7313.0},\n",
       "  2: {'A': 709.0, 'C': 1.0, 'G': 1.0, 'T': 20913.0},\n",
       "  3: {'A': 251.0, 'C': 347.0, 'G': 5851.0, 'T': 15175.0},\n",
       "  4: {'A': 1007.0, 'C': 19586.0, 'G': 1.0, 'T': 1030.0},\n",
       "  5: {'A': 841.0, 'C': 7781.0, 'G': 1.0, 'T': 13001.0},\n",
       "  6: {'A': 3693.0, 'C': 1.0, 'G': 11457.0, 'T': 6473.0},\n",
       "  7: {'A': 1575.0, 'C': 1.0, 'G': 20047.0, 'T': 1.0},\n",
       "  8: {'A': 2533.0, 'C': 1.0, 'G': 18687.0, 'T': 403.0},\n",
       "  9: {'A': 21621.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 21621.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 9312.0, 'C': 963.0, 'G': 6580.0, 'T': 4769.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '18555785',\n",
       "  'uniprot_ids': 'P40763',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1502.1',\n",
       "  'Matrix_Name': 'HOXB8',\n",
       "  1: {'A': 523.0, 'C': 537.0, 'G': 2780.0, 'T': 638.0},\n",
       "  2: {'A': 1.0, 'C': 2561.0, 'G': 1.0, 'T': 1915.0},\n",
       "  3: {'A': 4475.0, 'C': 3375.0, 'G': 90.0, 'T': 1.0},\n",
       "  4: {'A': 4475.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 4475.0},\n",
       "  6: {'A': 285.0, 'C': 1.0, 'G': 543.0, 'T': 4475.0},\n",
       "  7: {'A': 4475.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1175.0, 'C': 1705.0, 'G': 831.0, 'T': 767.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P17481',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0637.1',\n",
       "  'Matrix_Name': 'CENPB',\n",
       "  1: {'A': 686.0, 'C': 5581.0, 'G': 4.0, 'T': 735.0},\n",
       "  2: {'A': 101.0, 'C': 5581.0, 'G': 5.0, 'T': 145.0},\n",
       "  3: {'A': 3.0, 'C': 5581.0, 'G': 2.0, 'T': 15.0},\n",
       "  4: {'A': 174.0, 'C': 187.0, 'G': 5581.0, 'T': 211.0},\n",
       "  5: {'A': 13.0, 'C': 5581.0, 'G': 1424.0, 'T': 2558.0},\n",
       "  6: {'A': 2210.0, 'C': 630.0, 'G': 1148.0, 'T': 1596.0},\n",
       "  7: {'A': 298.0, 'C': 434.0, 'G': 1.0, 'T': 5581.0},\n",
       "  8: {'A': 1707.0, 'C': 1639.0, 'G': 1158.0, 'T': 1080.0},\n",
       "  9: {'A': 1294.0, 'C': 1768.0, 'G': 1480.0, 'T': 1042.0},\n",
       "  10: {'A': 2169.0, 'C': 823.0, 'G': 761.0, 'T': 1831.0},\n",
       "  11: {'A': 3687.0, 'C': 695.0, 'G': 1894.0, 'T': 598.0},\n",
       "  12: {'A': 285.0, 'C': 5581.0, 'G': 12.0, 'T': 101.0},\n",
       "  13: {'A': 873.0, 'C': 86.0, 'G': 5581.0, 'T': 1.0},\n",
       "  14: {'A': 5581.0, 'C': 583.0, 'G': 62.0, 'T': 1904.0},\n",
       "  15: {'A': 5581.0, 'C': 287.0, 'G': 60.0, 'T': 202.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': '',\n",
       "  'tf_class': 'CENP-B',\n",
       "  'pubmed_ids': '12460987',\n",
       "  'uniprot_ids': 'P07199',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0750.2',\n",
       "  'Matrix_Name': 'ZBTB7A',\n",
       "  1: {'A': 4137.0, 'C': 3513.0, 'G': 4168.0, 'T': 2827.0},\n",
       "  2: {'A': 2784.0, 'C': 5357.0, 'G': 4398.0, 'T': 2106.0},\n",
       "  3: {'A': 478.0, 'C': 12470.0, 'G': 1497.0, 'T': 200.0},\n",
       "  4: {'A': 2491.0, 'C': 11294.0, 'G': 625.0, 'T': 235.0},\n",
       "  5: {'A': 174.0, 'C': 211.0, 'G': 14131.0, 'T': 129.0},\n",
       "  6: {'A': 115.0, 'C': 164.0, 'G': 14240.0, 'T': 126.0},\n",
       "  7: {'A': 13855.0, 'C': 275.0, 'G': 336.0, 'T': 179.0},\n",
       "  8: {'A': 13720.0, 'C': 344.0, 'G': 262.0, 'T': 319.0},\n",
       "  9: {'A': 630.0, 'C': 712.0, 'G': 13113.0, 'T': 190.0},\n",
       "  10: {'A': 1142.0, 'C': 3434.0, 'G': 1624.0, 'T': 8445.0},\n",
       "  11: {'A': 1745.0, 'C': 2486.0, 'G': 9143.0, 'T': 1271.0},\n",
       "  12: {'A': 2710.0, 'C': 4387.0, 'G': 5863.0, 'T': 1685.0},\n",
       "  13: {'A': 2851.0, 'C': 4857.0, 'G': 4670.0, 'T': 2267.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '12750370',\n",
       "  'uniprot_ids': 'O95365',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0666.1',\n",
       "  'Matrix_Name': 'MSX1',\n",
       "  1: {'A': 462.0, 'C': 1325.0, 'G': 788.0, 'T': 502.0},\n",
       "  2: {'A': 83.0, 'C': 2118.0, 'G': 34.0, 'T': 957.0},\n",
       "  3: {'A': 3074.0, 'C': 333.0, 'G': 140.0, 'T': 17.0},\n",
       "  4: {'A': 3074.0, 'C': 108.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 32.0, 'T': 3074.0},\n",
       "  6: {'A': 64.0, 'C': 206.0, 'G': 252.0, 'T': 3074.0},\n",
       "  7: {'A': 3074.0, 'C': 119.0, 'G': 106.0, 'T': 92.0},\n",
       "  8: {'A': 856.0, 'C': 815.0, 'G': 946.0, 'T': 461.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P28360',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0818.1',\n",
       "  'Matrix_Name': 'BHLHE22',\n",
       "  1: {'A': 521.0, 'C': 27.0, 'G': 173.0, 'T': 22.0},\n",
       "  2: {'A': 190.0, 'C': 230.0, 'G': 100.0, 'T': 5.0},\n",
       "  3: {'A': 7.0, 'C': 521.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 521.0, 'C': 7.0, 'G': 25.0, 'T': 9.0},\n",
       "  5: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 521.0},\n",
       "  6: {'A': 521.0, 'C': 5.0, 'G': 3.0, 'T': 3.0},\n",
       "  7: {'A': 12.0, 'C': 49.0, 'G': 1.0, 'T': 521.0},\n",
       "  8: {'A': 1.0, 'C': 3.0, 'G': 521.0, 'T': 5.0},\n",
       "  9: {'A': 5.0, 'C': 194.0, 'G': 124.0, 'T': 202.0},\n",
       "  10: {'A': 37.0, 'C': 255.0, 'G': 22.0, 'T': 521.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22284184',\n",
       "  'uniprot_ids': 'Q8NFJ8',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0695.1',\n",
       "  'Matrix_Name': 'ZBTB7C',\n",
       "  1: {'A': 273.0, 'C': 119.0, 'G': 464.0, 'T': 165.0},\n",
       "  2: {'A': 184.0, 'C': 1019.0, 'G': 216.0, 'T': 111.0},\n",
       "  3: {'A': 352.0, 'C': 129.0, 'G': 1019.0, 'T': 138.0},\n",
       "  4: {'A': 1019.0, 'C': 226.0, 'G': 9.0, 'T': 25.0},\n",
       "  5: {'A': 40.0, 'C': 1019.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1019.0, 'G': 6.0, 'T': 15.0},\n",
       "  7: {'A': 1019.0, 'C': 270.0, 'G': 9.0, 'T': 11.0},\n",
       "  8: {'A': 19.0, 'C': 1019.0, 'G': 8.0, 'T': 18.0},\n",
       "  9: {'A': 91.0, 'C': 1019.0, 'G': 55.0, 'T': 91.0},\n",
       "  10: {'A': 254.0, 'C': 139.0, 'G': 528.0, 'T': 101.0},\n",
       "  11: {'A': 1019.0, 'C': 445.0, 'G': 252.0, 'T': 420.0},\n",
       "  12: {'A': 334.0, 'C': 316.0, 'G': 197.0, 'T': 175.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'A1YPR0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0810.1',\n",
       "  'Matrix_Name': 'TFAP2A(var.2)',\n",
       "  1: {'A': 679.0, 'C': 1184.0, 'G': 174.0, 'T': 1972.0},\n",
       "  2: {'A': 839.0, 'C': 1907.0, 'G': 4007.0, 'T': 148.0},\n",
       "  3: {'A': 16.0, 'C': 4007.0, 'G': 230.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 4007.0, 'G': 11.0, 'T': 34.0},\n",
       "  5: {'A': 11.0, 'C': 2813.0, 'G': 222.0, 'T': 964.0},\n",
       "  6: {'A': 198.0, 'C': 1654.0, 'G': 832.0, 'T': 1325.0},\n",
       "  7: {'A': 1325.0, 'C': 1440.0, 'G': 1055.0, 'T': 190.0},\n",
       "  8: {'A': 1698.0, 'C': 408.0, 'G': 1874.0, 'T': 29.0},\n",
       "  9: {'A': 192.0, 'C': 35.0, 'G': 4007.0, 'T': 1.0},\n",
       "  10: {'A': 6.0, 'C': 789.0, 'G': 4007.0, 'T': 11.0},\n",
       "  11: {'A': 102.0, 'C': 4007.0, 'G': 469.0, 'T': 227.0},\n",
       "  12: {'A': 1923.0, 'C': 435.0, 'G': 980.0, 'T': 672.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'P05549',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1106.1',\n",
       "  'Matrix_Name': 'HIF1A',\n",
       "  1: {'A': 198.0, 'C': 230.0, 'G': 415.0, 'T': 141.0},\n",
       "  2: {'A': 106.0, 'C': 271.0, 'G': 264.0, 'T': 343.0},\n",
       "  3: {'A': 977.0, 'C': 1.0, 'G': 3.0, 'T': 3.0},\n",
       "  4: {'A': 1.0, 'C': 980.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 2.0, 'C': 1.0, 'G': 980.0, 'T': 1.0},\n",
       "  6: {'A': 3.0, 'C': 34.0, 'G': 12.0, 'T': 935.0},\n",
       "  7: {'A': 85.0, 'C': 42.0, 'G': 845.0, 'T': 12.0},\n",
       "  8: {'A': 89.0, 'C': 838.0, 'G': 21.0, 'T': 36.0},\n",
       "  9: {'A': 207.0, 'C': 291.0, 'G': 258.0, 'T': 228.0},\n",
       "  10: {'A': 165.0, 'C': 316.0, 'G': 272.0, 'T': 231.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20061373',\n",
       "  'uniprot_ids': 'Q16665',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0526.3',\n",
       "  'Matrix_Name': 'USF2',\n",
       "  1: {'A': 9448.0, 'C': 7704.0, 'G': 9958.0, 'T': 11112.0},\n",
       "  2: {'A': 10548.0, 'C': 5510.0, 'G': 14330.0, 'T': 7834.0},\n",
       "  3: {'A': 6563.0, 'C': 3200.0, 'G': 26612.0, 'T': 1847.0},\n",
       "  4: {'A': 2755.0, 'C': 6525.0, 'G': 3098.0, 'T': 25844.0},\n",
       "  5: {'A': 366.0, 'C': 36958.0, 'G': 526.0, 'T': 372.0},\n",
       "  6: {'A': 36934.0, 'C': 547.0, 'G': 330.0, 'T': 411.0},\n",
       "  7: {'A': 313.0, 'C': 10980.0, 'G': 1911.0, 'T': 25018.0},\n",
       "  8: {'A': 1745.0, 'C': 789.0, 'G': 35414.0, 'T': 274.0},\n",
       "  9: {'A': 358.0, 'C': 199.0, 'G': 499.0, 'T': 37166.0},\n",
       "  10: {'A': 337.0, 'C': 561.0, 'G': 36979.0, 'T': 345.0},\n",
       "  11: {'A': 31294.0, 'C': 2133.0, 'G': 3249.0, 'T': 1546.0},\n",
       "  12: {'A': 1449.0, 'C': 28476.0, 'G': 2402.0, 'T': 5895.0},\n",
       "  13: {'A': 7699.0, 'C': 13807.0, 'G': 4692.0, 'T': 12024.0},\n",
       "  14: {'A': 10036.0, 'C': 9794.0, 'G': 7668.0, 'T': 10724.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q15853',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1646.1',\n",
       "  'Matrix_Name': 'OSR2',\n",
       "  1: {'A': 4438.0, 'C': 3642.0, 'G': 3247.0, 'T': 3485.0},\n",
       "  2: {'A': 4245.0, 'C': 4207.0, 'G': 3007.0, 'T': 3353.0},\n",
       "  3: {'A': 8521.0, 'C': 2222.0, 'G': 2118.0, 'T': 1951.0},\n",
       "  4: {'A': 255.0, 'C': 13938.0, 'G': 407.0, 'T': 212.0},\n",
       "  5: {'A': 13600.0, 'C': 198.0, 'G': 738.0, 'T': 276.0},\n",
       "  6: {'A': 217.0, 'C': 230.0, 'G': 14207.0, 'T': 158.0},\n",
       "  7: {'A': 11679.0, 'C': 704.0, 'G': 902.0, 'T': 1527.0},\n",
       "  8: {'A': 13726.0, 'C': 289.0, 'G': 517.0, 'T': 280.0},\n",
       "  9: {'A': 417.0, 'C': 171.0, 'G': 13983.0, 'T': 241.0},\n",
       "  10: {'A': 667.0, 'C': 12110.0, 'G': 842.0, 'T': 1193.0},\n",
       "  11: {'A': 4387.0, 'C': 4275.0, 'G': 2267.0, 'T': 3883.0},\n",
       "  12: {'A': 4049.0, 'C': 3305.0, 'G': 4727.0, 'T': 2731.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q8N2R0',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0083.3',\n",
       "  'Matrix_Name': 'SRF',\n",
       "  1: {'A': 215.0, 'C': 126.0, 'G': 81.0, 'T': 1130.0},\n",
       "  2: {'A': 76.0, 'C': 153.0, 'G': 1145.0, 'T': 373.0},\n",
       "  3: {'A': 628.0, 'C': 612.0, 'G': 145.0, 'T': 168.0},\n",
       "  4: {'A': 1.0, 'C': 1772.0, 'G': 16.0, 'T': 1.0},\n",
       "  5: {'A': 4.0, 'C': 1788.0, 'G': 12.0, 'T': 20.0},\n",
       "  6: {'A': 1138.0, 'C': 14.0, 'G': 19.0, 'T': 62.0},\n",
       "  7: {'A': 55.0, 'C': 1.0, 'G': 7.0, 'T': 1572.0},\n",
       "  8: {'A': 1164.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  9: {'A': 2.0, 'C': 6.0, 'G': 9.0, 'T': 1611.0},\n",
       "  10: {'A': 1163.0, 'C': 1.0, 'G': 1.0, 'T': 17.0},\n",
       "  11: {'A': 440.0, 'C': 13.0, 'G': 1.0, 'T': 1075.0},\n",
       "  12: {'A': 9.0, 'C': 10.0, 'G': 1897.0, 'T': 1.0},\n",
       "  13: {'A': 9.0, 'C': 3.0, 'G': 1873.0, 'T': 8.0},\n",
       "  14: {'A': 263.0, 'C': 106.0, 'G': 671.0, 'T': 687.0},\n",
       "  15: {'A': 338.0, 'C': 1024.0, 'G': 146.0, 'T': 73.0},\n",
       "  16: {'A': 914.0, 'C': 64.0, 'G': 99.0, 'T': 249.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Responders',\n",
       "  'tf_class': 'MADS',\n",
       "  'pubmed_ids': '2243767',\n",
       "  'uniprot_ids': 'P11831',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0907.1',\n",
       "  'Matrix_Name': 'HOXC13',\n",
       "  1: {'A': 166.0, 'C': 289.0, 'G': 496.0, 'T': 310.0},\n",
       "  2: {'A': 45.0, 'C': 1258.0, 'G': 59.0, 'T': 14.0},\n",
       "  3: {'A': 5.0, 'C': 22.0, 'G': 6.0, 'T': 1258.0},\n",
       "  4: {'A': 27.0, 'C': 1258.0, 'G': 11.0, 'T': 133.0},\n",
       "  5: {'A': 252.0, 'C': 5.0, 'G': 1258.0, 'T': 5.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 6.0, 'T': 1258.0},\n",
       "  7: {'A': 1258.0, 'C': 2.0, 'G': 3.0, 'T': 166.0},\n",
       "  8: {'A': 1258.0, 'C': 5.0, 'G': 3.0, 'T': 9.0},\n",
       "  9: {'A': 1258.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  10: {'A': 1258.0, 'C': 174.0, 'G': 90.0, 'T': 206.0},\n",
       "  11: {'A': 390.0, 'C': 359.0, 'G': 148.0, 'T': 363.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31276',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1107.2',\n",
       "  'Matrix_Name': 'KLF9',\n",
       "  1: {'A': 5360.0, 'C': 10413.0, 'G': 5201.0, 'T': 4609.0},\n",
       "  2: {'A': 8016.0, 'C': 5109.0, 'G': 7780.0, 'T': 4678.0},\n",
       "  3: {'A': 2863.0, 'C': 4343.0, 'G': 16853.0, 'T': 1524.0},\n",
       "  4: {'A': 4665.0, 'C': 17840.0, 'G': 1661.0, 'T': 1417.0},\n",
       "  5: {'A': 288.0, 'C': 24710.0, 'G': 292.0, 'T': 293.0},\n",
       "  6: {'A': 21900.0, 'C': 2560.0, 'G': 709.0, 'T': 414.0},\n",
       "  7: {'A': 222.0, 'C': 24864.0, 'G': 354.0, 'T': 143.0},\n",
       "  8: {'A': 19188.0, 'C': 1071.0, 'G': 4341.0, 'T': 983.0},\n",
       "  9: {'A': 225.0, 'C': 24366.0, 'G': 736.0, 'T': 256.0},\n",
       "  10: {'A': 2429.0, 'C': 22118.0, 'G': 606.0, 'T': 430.0},\n",
       "  11: {'A': 147.0, 'C': 24683.0, 'G': 206.0, 'T': 547.0},\n",
       "  12: {'A': 18838.0, 'C': 4340.0, 'G': 853.0, 'T': 1552.0},\n",
       "  13: {'A': 793.0, 'C': 20645.0, 'G': 1779.0, 'T': 2366.0},\n",
       "  14: {'A': 6949.0, 'C': 8798.0, 'G': 3290.0, 'T': 6546.0},\n",
       "  15: {'A': 4111.0, 'C': 10521.0, 'G': 4424.0, 'T': 6527.0},\n",
       "  16: {'A': 5876.0, 'C': 9520.0, 'G': 5682.0, 'T': 4505.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '12620113',\n",
       "  'uniprot_ids': 'Q13886',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0703.2',\n",
       "  'Matrix_Name': 'LMX1B',\n",
       "  1: {'A': 1895.0, 'C': 618.0, 'G': 891.0, 'T': 664.0},\n",
       "  2: {'A': 2006.0, 'C': 681.0, 'G': 757.0, 'T': 2059.0},\n",
       "  3: {'A': 1110.0, 'C': 549.0, 'G': 937.0, 'T': 2956.0},\n",
       "  4: {'A': 1.0, 'C': 600.0, 'G': 191.0, 'T': 3466.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 4065.0},\n",
       "  6: {'A': 4065.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 4065.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 4065.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 4065.0},\n",
       "  10: {'A': 4065.0, 'C': 13.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 2320.0, 'C': 786.0, 'G': 351.0, 'T': 610.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'O60663',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1102.2',\n",
       "  'Matrix_Name': 'CTCFL',\n",
       "  1: {'A': 5826.0, 'C': 4181.0, 'G': 4845.0, 'T': 3189.0},\n",
       "  2: {'A': 2660.0, 'C': 7034.0, 'G': 7095.0, 'T': 1252.0},\n",
       "  3: {'A': 1302.0, 'C': 12868.0, 'G': 2034.0, 'T': 1837.0},\n",
       "  4: {'A': 17271.0, 'C': 414.0, 'G': 330.0, 'T': 26.0},\n",
       "  5: {'A': 87.0, 'C': 482.0, 'G': 17286.0, 'T': 186.0},\n",
       "  6: {'A': 685.0, 'C': 601.0, 'G': 16512.0, 'T': 243.0},\n",
       "  7: {'A': 350.0, 'C': 1142.0, 'G': 15426.0, 'T': 1123.0},\n",
       "  8: {'A': 194.0, 'C': 768.0, 'G': 16820.0, 'T': 259.0},\n",
       "  9: {'A': 352.0, 'C': 271.0, 'G': 17002.0, 'T': 416.0},\n",
       "  10: {'A': 182.0, 'C': 17594.0, 'G': 69.0, 'T': 196.0},\n",
       "  11: {'A': 4167.0, 'C': 3016.0, 'G': 9300.0, 'T': 1558.0},\n",
       "  12: {'A': 1772.0, 'C': 7728.0, 'G': 6342.0, 'T': 2199.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '26268681',\n",
       "  'uniprot_ids': 'Q8NI51',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0505.1',\n",
       "  'Matrix_Name': 'Nr5a2',\n",
       "  1: {'A': 646.0, 'C': 225.0, 'G': 635.0, 'T': 200.0},\n",
       "  2: {'A': 986.0, 'C': 171.0, 'G': 337.0, 'T': 212.0},\n",
       "  3: {'A': 277.0, 'C': 199.0, 'G': 991.0, 'T': 239.0},\n",
       "  4: {'A': 123.0, 'C': 379.0, 'G': 367.0, 'T': 837.0},\n",
       "  5: {'A': 72.0, 'C': 352.0, 'G': 24.0, 'T': 1258.0},\n",
       "  6: {'A': 9.0, 'C': 1617.0, 'G': 75.0, 'T': 5.0},\n",
       "  7: {'A': 1686.0, 'C': 15.0, 'G': 1.0, 'T': 4.0},\n",
       "  8: {'A': 1680.0, 'C': 1.0, 'G': 24.0, 'T': 1.0},\n",
       "  9: {'A': 40.0, 'C': 1.0, 'G': 1664.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 1703.0, 'T': 1.0},\n",
       "  11: {'A': 170.0, 'C': 585.0, 'G': 62.0, 'T': 889.0},\n",
       "  12: {'A': 1.0, 'C': 1569.0, 'G': 13.0, 'T': 123.0},\n",
       "  13: {'A': 1438.0, 'C': 21.0, 'G': 91.0, 'T': 156.0},\n",
       "  14: {'A': 272.0, 'C': 346.0, 'G': 741.0, 'T': 347.0},\n",
       "  15: {'A': 250.0, 'C': 694.0, 'G': 455.0, 'T': 307.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'FTZ-F1-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '12853459',\n",
       "  'uniprot_ids': 'P45448',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1418.1',\n",
       "  'Matrix_Name': 'IRF3',\n",
       "  1: {'A': 4597.0, 'C': 6119.0, 'G': 7840.0, 'T': 4121.0},\n",
       "  2: {'A': 3598.0, 'C': 7116.0, 'G': 9274.0, 'T': 2953.0},\n",
       "  3: {'A': 8452.0, 'C': 701.0, 'G': 14774.0, 'T': 2100.0},\n",
       "  4: {'A': 12888.0, 'C': 18.0, 'G': 9048.0, 'T': 630.0},\n",
       "  5: {'A': 18312.0, 'C': 84.0, 'G': 3485.0, 'T': 2154.0},\n",
       "  6: {'A': 19158.0, 'C': 215.0, 'G': 194.0, 'T': 378.0},\n",
       "  7: {'A': 14211.0, 'C': 14601.0, 'G': 5576.0, 'T': 6866.0},\n",
       "  8: {'A': 289.0, 'C': 7339.0, 'G': 26853.0, 'T': 3059.0},\n",
       "  9: {'A': 155.0, 'C': 15.0, 'G': 28000.0, 'T': 20.0},\n",
       "  10: {'A': 19386.0, 'C': 16.0, 'G': 56.0, 'T': 11.0},\n",
       "  11: {'A': 19351.0, 'C': 16.0, 'G': 60.0, 'T': 55.0},\n",
       "  12: {'A': 19429.0, 'C': 23.0, 'G': 44.0, 'T': 325.0},\n",
       "  13: {'A': 424.0, 'C': 18364.0, 'G': 1151.0, 'T': 865.0},\n",
       "  14: {'A': 712.0, 'C': 18317.0, 'G': 3397.0, 'T': 2349.0},\n",
       "  15: {'A': 395.0, 'C': 9.0, 'G': 27996.0, 'T': 9.0},\n",
       "  16: {'A': 19323.0, 'C': 138.0, 'G': 84.0, 'T': 27.0},\n",
       "  17: {'A': 19387.0, 'C': 52.0, 'G': 37.0, 'T': 487.0},\n",
       "  18: {'A': 19411.0, 'C': 85.0, 'G': 57.0, 'T': 788.0},\n",
       "  19: {'A': 570.0, 'C': 17759.0, 'G': 3275.0, 'T': 517.0},\n",
       "  20: {'A': 4483.0, 'C': 12662.0, 'G': 4798.0, 'T': 20226.0},\n",
       "  21: {'A': 8275.0, 'C': 2073.0, 'G': 10607.0, 'T': 2390.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 21,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '10938111',\n",
       "  'uniprot_ids': 'Q14653',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0700.2',\n",
       "  'Matrix_Name': 'LHX2',\n",
       "  1: {'A': 2375.0, 'C': 989.0, 'G': 1262.0, 'T': 1711.0},\n",
       "  2: {'A': 2265.0, 'C': 950.0, 'G': 1253.0, 'T': 1869.0},\n",
       "  3: {'A': 899.0, 'C': 3413.0, 'G': 1046.0, 'T': 979.0},\n",
       "  4: {'A': 45.0, 'C': 5613.0, 'G': 38.0, 'T': 641.0},\n",
       "  5: {'A': 5791.0, 'C': 72.0, 'G': 53.0, 'T': 421.0},\n",
       "  6: {'A': 6023.0, 'C': 102.0, 'G': 78.0, 'T': 134.0},\n",
       "  7: {'A': 130.0, 'C': 59.0, 'G': 65.0, 'T': 6083.0},\n",
       "  8: {'A': 68.0, 'C': 6.0, 'G': 6.0, 'T': 6257.0},\n",
       "  9: {'A': 6276.0, 'C': 14.0, 'G': 7.0, 'T': 40.0},\n",
       "  10: {'A': 2168.0, 'C': 977.0, 'G': 1464.0, 'T': 1728.0},\n",
       "  11: {'A': 1935.0, 'C': 1344.0, 'G': 1036.0, 'T': 2022.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P50458',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1599.1',\n",
       "  'Matrix_Name': 'ZNF682',\n",
       "  1: {'A': 206.0, 'C': 424.0, 'G': 215.0, 'T': 217.0},\n",
       "  2: {'A': 305.0, 'C': 235.0, 'G': 350.0, 'T': 172.0},\n",
       "  3: {'A': 111.0, 'C': 86.0, 'G': 775.0, 'T': 90.0},\n",
       "  4: {'A': 104.0, 'C': 43.0, 'G': 878.0, 'T': 37.0},\n",
       "  5: {'A': 101.0, 'C': 796.0, 'G': 106.0, 'T': 59.0},\n",
       "  6: {'A': 49.0, 'C': 598.0, 'G': 56.0, 'T': 359.0},\n",
       "  7: {'A': 954.0, 'C': 73.0, 'G': 21.0, 'T': 14.0},\n",
       "  8: {'A': 1000.0, 'C': 32.0, 'G': 23.0, 'T': 7.0},\n",
       "  9: {'A': 7.0, 'C': 8.0, 'G': 1039.0, 'T': 8.0},\n",
       "  10: {'A': 14.0, 'C': 1000.0, 'G': 14.0, 'T': 34.0},\n",
       "  11: {'A': 6.0, 'C': 1034.0, 'G': 10.0, 'T': 12.0},\n",
       "  12: {'A': 8.0, 'C': 1035.0, 'G': 11.0, 'T': 8.0},\n",
       "  13: {'A': 67.0, 'C': 897.0, 'G': 33.0, 'T': 65.0},\n",
       "  14: {'A': 368.0, 'C': 118.0, 'G': 119.0, 'T': 457.0},\n",
       "  15: {'A': 325.0, 'C': 255.0, 'G': 301.0, 'T': 181.0},\n",
       "  16: {'A': 193.0, 'C': 261.0, 'G': 195.0, 'T': 413.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28092692',\n",
       "  'uniprot_ids': 'O95780',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0033.2',\n",
       "  'Matrix_Name': 'FOXL1',\n",
       "  1: {'A': 2103.0, 'C': 14.0, 'G': 2679.0, 'T': 141.0},\n",
       "  2: {'A': 11.0, 'C': 417.0, 'G': 1.0, 'T': 4781.0},\n",
       "  3: {'A': 4781.0, 'C': 629.0, 'G': 58.0, 'T': 1.0},\n",
       "  4: {'A': 4781.0, 'C': 39.0, 'G': 1.0, 'T': 16.0},\n",
       "  5: {'A': 4781.0, 'C': 83.0, 'G': 51.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 3724.0, 'G': 1.0, 'T': 1058.0},\n",
       "  7: {'A': 4781.0, 'C': 55.0, 'G': 1.0, 'T': 18.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23836653',\n",
       "  'uniprot_ids': 'Q12952',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0674.1',\n",
       "  'Matrix_Name': 'NKX6-1',\n",
       "  1: {'A': 404.0, 'C': 696.0, 'G': 735.0, 'T': 382.0},\n",
       "  2: {'A': 1.0, 'C': 574.0, 'G': 1.0, 'T': 1641.0},\n",
       "  3: {'A': 1408.0, 'C': 806.0, 'G': 1.0, 'T': 196.0},\n",
       "  4: {'A': 2213.0, 'C': 201.0, 'G': 13.0, 'T': 41.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2213.0},\n",
       "  6: {'A': 59.0, 'C': 46.0, 'G': 1.0, 'T': 2213.0},\n",
       "  7: {'A': 2213.0, 'C': 28.0, 'G': 133.0, 'T': 18.0},\n",
       "  8: {'A': 1451.0, 'C': 295.0, 'G': 143.0, 'T': 328.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P78426',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0795.1',\n",
       "  'Matrix_Name': 'SMAD3',\n",
       "  1: {'A': 555.0, 'C': 1830.0, 'G': 88.0, 'T': 1663.0},\n",
       "  2: {'A': 21.0, 'C': 17.0, 'G': 3492.0, 'T': 19.0},\n",
       "  3: {'A': 104.0, 'C': 30.0, 'G': 84.0, 'T': 3492.0},\n",
       "  4: {'A': 10.0, 'C': 3492.0, 'G': 12.0, 'T': 7.0},\n",
       "  5: {'A': 1.0, 'C': 47.0, 'G': 32.0, 'T': 3492.0},\n",
       "  6: {'A': 3492.0, 'C': 31.0, 'G': 23.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 11.0, 'G': 3492.0, 'T': 4.0},\n",
       "  8: {'A': 3492.0, 'C': 200.0, 'G': 6.0, 'T': 57.0},\n",
       "  9: {'A': 21.0, 'C': 3492.0, 'G': 11.0, 'T': 21.0},\n",
       "  10: {'A': 3492.0, 'C': 435.0, 'G': 317.0, 'T': 759.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SMAD',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '12582250',\n",
       "  'uniprot_ids': 'P84022',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0675.1',\n",
       "  'Matrix_Name': 'NKX6-2',\n",
       "  1: {'A': 3762.0, 'C': 6073.0, 'G': 5229.0, 'T': 3875.0},\n",
       "  2: {'A': 1173.0, 'C': 7832.0, 'G': 815.0, 'T': 11105.0},\n",
       "  3: {'A': 11355.0, 'C': 7582.0, 'G': 1054.0, 'T': 2583.0},\n",
       "  4: {'A': 18936.0, 'C': 311.0, 'G': 192.0, 'T': 547.0},\n",
       "  5: {'A': 585.0, 'C': 1.0, 'G': 1.0, 'T': 18936.0},\n",
       "  6: {'A': 2469.0, 'C': 2234.0, 'G': 552.0, 'T': 18936.0},\n",
       "  7: {'A': 18936.0, 'C': 371.0, 'G': 776.0, 'T': 679.0},\n",
       "  8: {'A': 9972.0, 'C': 3458.0, 'G': 1661.0, 'T': 3847.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9C056',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1638.1',\n",
       "  'Matrix_Name': 'HAND2',\n",
       "  1: {'A': 10137.0, 'C': 5565.0, 'G': 7464.0, 'T': 7121.0},\n",
       "  2: {'A': 8837.0, 'C': 11299.0, 'G': 6995.0, 'T': 3156.0},\n",
       "  3: {'A': 20.0, 'C': 30063.0, 'G': 133.0, 'T': 71.0},\n",
       "  4: {'A': 29223.0, 'C': 411.0, 'G': 648.0, 'T': 5.0},\n",
       "  5: {'A': 13.0, 'C': 43.0, 'G': 30145.0, 'T': 86.0},\n",
       "  6: {'A': 30193.0, 'C': 31.0, 'G': 39.0, 'T': 24.0},\n",
       "  7: {'A': 15.0, 'C': 2277.0, 'G': 32.0, 'T': 27963.0},\n",
       "  8: {'A': 235.0, 'C': 150.0, 'G': 29852.0, 'T': 50.0},\n",
       "  9: {'A': 5218.0, 'C': 7691.0, 'G': 8879.0, 'T': 8499.0},\n",
       "  10: {'A': 6166.0, 'C': 10228.0, 'G': 6109.0, 'T': 7784.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11812799',\n",
       "  'uniprot_ids': 'P61296',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0059.1',\n",
       "  'Matrix_Name': 'MAX::MYC',\n",
       "  1: {'A': 8.0, 'C': 2.0, 'G': 10.0, 'T': 5.0},\n",
       "  2: {'A': 16.0, 'C': 2.0, 'G': 5.0, 'T': 2.0},\n",
       "  3: {'A': 3.0, 'C': 10.0, 'G': 10.0, 'T': 2.0},\n",
       "  4: {'A': 2.0, 'C': 21.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 22.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 21.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 21.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 21.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 22.0, 'T': 1.0},\n",
       "  10: {'A': 2.0, 'C': 2.0, 'G': 19.0, 'T': 2.0},\n",
       "  11: {'A': 4.0, 'C': 6.0, 'G': 1.0, 'T': 14.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8265351',\n",
       "  'uniprot_ids': 'P61244;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0493.1',\n",
       "  'Matrix_Name': 'Klf1',\n",
       "  1: {'A': 151.0, 'C': 77.0, 'G': 203.0, 'T': 99.0},\n",
       "  2: {'A': 168.0, 'C': 41.0, 'G': 306.0, 'T': 15.0},\n",
       "  3: {'A': 68.0, 'C': 397.0, 'G': 33.0, 'T': 32.0},\n",
       "  4: {'A': 14.0, 'C': 462.0, 'G': 1.0, 'T': 53.0},\n",
       "  5: {'A': 405.0, 'C': 119.0, 'G': 1.0, 'T': 5.0},\n",
       "  6: {'A': 1.0, 'C': 527.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 401.0, 'C': 1.0, 'G': 98.0, 'T': 30.0},\n",
       "  8: {'A': 1.0, 'C': 527.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 527.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 509.0, 'G': 1.0, 'T': 19.0},\n",
       "  11: {'A': 281.0, 'C': 36.0, 'G': 1.0, 'T': 212.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7682653',\n",
       "  'uniprot_ids': 'P46099',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0151.1',\n",
       "  'Matrix_Name': 'Arid3a',\n",
       "  1: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 28.0},\n",
       "  3: {'A': 2.0, 'C': 10.0, 'G': 1.0, 'T': 18.0},\n",
       "  4: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 21.0, 'C': 1.0, 'G': 2.0, 'T': 7.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 6,\n",
       "  'tf_family': 'ARID-related',\n",
       "  'tf_class': 'ARID',\n",
       "  'pubmed_ids': '8543152',\n",
       "  'uniprot_ids': 'Q62431',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0050.2',\n",
       "  'Matrix_Name': 'IRF1',\n",
       "  1: {'A': 317.0, 'C': 321.0, 'G': 267.0, 'T': 461.0},\n",
       "  2: {'A': 252.0, 'C': 285.0, 'G': 252.0, 'T': 577.0},\n",
       "  3: {'A': 174.0, 'C': 325.0, 'G': 213.0, 'T': 654.0},\n",
       "  4: {'A': 139.0, 'C': 431.0, 'G': 149.0, 'T': 647.0},\n",
       "  5: {'A': 736.0, 'C': 102.0, 'G': 362.0, 'T': 166.0},\n",
       "  6: {'A': 25.0, 'C': 793.0, 'G': 507.0, 'T': 41.0},\n",
       "  7: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 1351.0},\n",
       "  8: {'A': 1.0, 'C': 32.0, 'G': 1.0, 'T': 1332.0},\n",
       "  9: {'A': 1.0, 'C': 57.0, 'G': 5.0, 'T': 1303.0},\n",
       "  10: {'A': 1.0, 'C': 1250.0, 'G': 1.0, 'T': 114.0},\n",
       "  11: {'A': 914.0, 'C': 40.0, 'G': 239.0, 'T': 173.0},\n",
       "  12: {'A': 21.0, 'C': 788.0, 'G': 346.0, 'T': 211.0},\n",
       "  13: {'A': 17.0, 'C': 2.0, 'G': 1.0, 'T': 1346.0},\n",
       "  14: {'A': 8.0, 'C': 15.0, 'G': 6.0, 'T': 1337.0},\n",
       "  15: {'A': 12.0, 'C': 59.0, 'G': 1.0, 'T': 1294.0},\n",
       "  16: {'A': 18.0, 'C': 1024.0, 'G': 29.0, 'T': 295.0},\n",
       "  17: {'A': 514.0, 'C': 246.0, 'G': 264.0, 'T': 342.0},\n",
       "  18: {'A': 219.0, 'C': 465.0, 'G': 292.0, 'T': 390.0},\n",
       "  19: {'A': 146.0, 'C': 187.0, 'G': 115.0, 'T': 918.0},\n",
       "  20: {'A': 199.0, 'C': 194.0, 'G': 100.0, 'T': 873.0},\n",
       "  21: {'A': 167.0, 'C': 332.0, 'G': 152.0, 'T': 715.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 21,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '21803131',\n",
       "  'uniprot_ids': 'P10914',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0825.1',\n",
       "  'Matrix_Name': 'MNT',\n",
       "  1: {'A': 280.0, 'C': 129.0, 'G': 146.0, 'T': 130.0},\n",
       "  2: {'A': 188.0, 'C': 229.0, 'G': 204.0, 'T': 64.0},\n",
       "  3: {'A': 28.0, 'C': 682.0, 'G': 1.0, 'T': 6.0},\n",
       "  4: {'A': 682.0, 'C': 1.0, 'G': 1.0, 'T': 22.0},\n",
       "  5: {'A': 1.0, 'C': 682.0, 'G': 8.0, 'T': 17.0},\n",
       "  6: {'A': 16.0, 'C': 1.0, 'G': 682.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 16.0, 'T': 682.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 682.0, 'T': 26.0},\n",
       "  9: {'A': 151.0, 'C': 282.0, 'G': 132.0, 'T': 121.0},\n",
       "  10: {'A': 160.0, 'C': 230.0, 'G': 76.0, 'T': 220.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '12695332',\n",
       "  'uniprot_ids': 'Q99583',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0107.1',\n",
       "  'Matrix_Name': 'RELA',\n",
       "  1: {'A': 1.0, 'C': 5.0, 'G': 12.0, 'T': 4.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 18.0, 'T': 2.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  4: {'A': 12.0, 'C': 1.0, 'G': 8.0, 'T': 1.0},\n",
       "  5: {'A': 11.0, 'C': 4.0, 'G': 5.0, 'T': 2.0},\n",
       "  6: {'A': 3.0, 'C': 1.0, 'G': 1.0, 'T': 17.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 19.0},\n",
       "  8: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 17.0},\n",
       "  9: {'A': 1.0, 'C': 19.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 19.0, 'G': 1.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NF-kappaB-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '1406630',\n",
       "  'uniprot_ids': 'Q04206',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0100.3',\n",
       "  'Matrix_Name': 'MYB',\n",
       "  1: {'A': 426.0, 'C': 423.0, 'G': 301.0, 'T': 381.0},\n",
       "  2: {'A': 293.0, 'C': 429.0, 'G': 398.0, 'T': 411.0},\n",
       "  3: {'A': 1.0, 'C': 1514.0, 'G': 6.0, 'T': 10.0},\n",
       "  4: {'A': 1479.0, 'C': 4.0, 'G': 40.0, 'T': 8.0},\n",
       "  5: {'A': 1520.0, 'C': 3.0, 'G': 1.0, 'T': 7.0},\n",
       "  6: {'A': 1.0, 'C': 1528.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 2.0, 'C': 61.0, 'G': 69.0, 'T': 1399.0},\n",
       "  8: {'A': 9.0, 'C': 4.0, 'G': 1517.0, 'T': 1.0},\n",
       "  9: {'A': 402.0, 'C': 483.0, 'G': 148.0, 'T': 498.0},\n",
       "  10: {'A': 423.0, 'C': 529.0, 'G': 226.0, 'T': 353.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Myb/SANT',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '8467793',\n",
       "  'uniprot_ids': 'P10242',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1155.1',\n",
       "  'Matrix_Name': 'ZSCAN4',\n",
       "  1: {'A': 258.0, 'C': 384.0, 'G': 113.0, 'T': 1374.0},\n",
       "  2: {'A': 291.0, 'C': 3.0, 'G': 1738.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1677.0, 'G': 2.0, 'T': 1.0},\n",
       "  4: {'A': 1293.0, 'C': 1.0, 'G': 1.0, 'T': 3.0},\n",
       "  5: {'A': 1.0, 'C': 1666.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1282.0, 'C': 1.0, 'G': 3.0, 'T': 3.0},\n",
       "  7: {'A': 4.0, 'C': 1703.0, 'G': 1.0, 'T': 74.0},\n",
       "  8: {'A': 773.0, 'C': 375.0, 'G': 321.0, 'T': 9.0},\n",
       "  9: {'A': 8.0, 'C': 1685.0, 'G': 1.0, 'T': 3.0},\n",
       "  10: {'A': 9.0, 'C': 12.0, 'G': 76.0, 'T': 1700.0},\n",
       "  11: {'A': 4.0, 'C': 51.0, 'G': 1874.0, 'T': 219.0},\n",
       "  12: {'A': 1119.0, 'C': 108.0, 'G': 231.0, 'T': 363.0},\n",
       "  13: {'A': 1226.0, 'C': 280.0, 'G': 1.0, 'T': 7.0},\n",
       "  14: {'A': 1270.0, 'C': 1.0, 'G': 3.0, 'T': 2.0},\n",
       "  15: {'A': 1001.0, 'C': 201.0, 'G': 366.0, 'T': 230.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28092692',\n",
       "  'uniprot_ids': 'Q8NAM6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1554.1',\n",
       "  'Matrix_Name': 'RFX7',\n",
       "  1: {'A': 1.0, 'C': 165.0, 'G': 64.0, 'T': 59.0},\n",
       "  2: {'A': 1.0, 'C': 3.0, 'G': 280.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 25.0, 'G': 1.0, 'T': 280.0},\n",
       "  4: {'A': 1.0, 'C': 28.0, 'G': 1.0, 'T': 280.0},\n",
       "  5: {'A': 96.0, 'C': 1.0, 'G': 280.0, 'T': 39.0},\n",
       "  6: {'A': 3.0, 'C': 280.0, 'G': 36.0, 'T': 20.0},\n",
       "  7: {'A': 1.0, 'C': 116.0, 'G': 14.0, 'T': 165.0},\n",
       "  8: {'A': 280.0, 'C': 23.0, 'G': 105.0, 'T': 28.0},\n",
       "  9: {'A': 38.0, 'C': 107.0, 'G': 28.0, 'T': 110.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'RFX-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'O14593',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0595.1',\n",
       "  'Matrix_Name': 'SREBF1',\n",
       "  1: {'A': 29.0, 'C': 15.0, 'G': 17.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 59.0},\n",
       "  3: {'A': 1.0, 'C': 59.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 59.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 52.0, 'G': 8.0, 'T': 1.0},\n",
       "  6: {'A': 7.0, 'C': 39.0, 'G': 14.0, 'T': 2.0},\n",
       "  7: {'A': 1.0, 'C': 45.0, 'G': 1.0, 'T': 15.0},\n",
       "  8: {'A': 3.0, 'C': 46.0, 'G': 12.0, 'T': 1.0},\n",
       "  9: {'A': 59.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 27.0, 'G': 8.0, 'T': 26.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8156598',\n",
       "  'uniprot_ids': 'P36956',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1503.1',\n",
       "  'Matrix_Name': 'HOXB9',\n",
       "  1: {'A': 623.0, 'C': 281.0, 'G': 1843.0, 'T': 3.0},\n",
       "  2: {'A': 1.0, 'C': 44.0, 'G': 1.0, 'T': 1843.0},\n",
       "  3: {'A': 1.0, 'C': 1843.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 645.0, 'C': 7.0, 'G': 1843.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 1843.0},\n",
       "  6: {'A': 1843.0, 'C': 13.0, 'G': 4.0, 'T': 463.0},\n",
       "  7: {'A': 1843.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1843.0, 'C': 43.0, 'G': 1.0, 'T': 18.0},\n",
       "  9: {'A': 1843.0, 'C': 296.0, 'G': 202.0, 'T': 487.0},\n",
       "  10: {'A': 418.0, 'C': 459.0, 'G': 262.0, 'T': 707.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P17482',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0610.1',\n",
       "  'Matrix_Name': 'DMRT3',\n",
       "  1: {'A': 454.0, 'C': 241.0, 'G': 155.0, 'T': 155.0},\n",
       "  2: {'A': 540.0, 'C': 124.0, 'G': 124.0, 'T': 217.0},\n",
       "  3: {'A': 107.0, 'C': 23.0, 'G': 23.0, 'T': 852.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1000.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  6: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 203.0, 'C': 1.0, 'G': 105.0, 'T': 695.0},\n",
       "  8: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 619.0, 'C': 189.0, 'G': 98.0, 'T': 98.0},\n",
       "  10: {'A': 510.0, 'C': 128.0, 'G': 128.0, 'T': 237.0},\n",
       "  11: {'A': 229.0, 'C': 229.0, 'G': 229.0, 'T': 316.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'DMRT',\n",
       "  'tf_class': 'DM-type',\n",
       "  'pubmed_ids': '17605809',\n",
       "  'uniprot_ids': 'Q9NQL9',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1622.1',\n",
       "  'Matrix_Name': 'Smad2::Smad3',\n",
       "  1: {'A': 2996.0, 'C': 2255.0, 'G': 2616.0, 'T': 2482.0},\n",
       "  2: {'A': 2509.0, 'C': 2516.0, 'G': 2325.0, 'T': 2999.0},\n",
       "  3: {'A': 2500.0, 'C': 1643.0, 'G': 3480.0, 'T': 2726.0},\n",
       "  4: {'A': 4452.0, 'C': 2213.0, 'G': 2889.0, 'T': 795.0},\n",
       "  5: {'A': 80.0, 'C': 60.0, 'G': 57.0, 'T': 10152.0},\n",
       "  6: {'A': 91.0, 'C': 83.0, 'G': 9769.0, 'T': 406.0},\n",
       "  7: {'A': 10142.0, 'C': 45.0, 'G': 59.0, 'T': 103.0},\n",
       "  8: {'A': 628.0, 'C': 7481.0, 'G': 1543.0, 'T': 697.0},\n",
       "  9: {'A': 248.0, 'C': 54.0, 'G': 69.0, 'T': 9978.0},\n",
       "  10: {'A': 712.0, 'C': 9444.0, 'G': 91.0, 'T': 102.0},\n",
       "  11: {'A': 10057.0, 'C': 86.0, 'G': 80.0, 'T': 126.0},\n",
       "  12: {'A': 684.0, 'C': 2048.0, 'G': 1980.0, 'T': 5637.0},\n",
       "  13: {'A': 2949.0, 'C': 2480.0, 'G': 2155.0, 'T': 2765.0},\n",
       "  14: {'A': 2747.0, 'C': 2596.0, 'G': 2525.0, 'T': 2481.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'SMAD',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '9732876',\n",
       "  'uniprot_ids': 'Q15796;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1556.1',\n",
       "  'Matrix_Name': 'RXRG(var.2)',\n",
       "  1: {'A': 804.0, 'C': 165.0, 'G': 1072.0, 'T': 80.0},\n",
       "  2: {'A': 1265.0, 'C': 8.0, 'G': 854.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 108.0, 'G': 2118.0, 'T': 1.0},\n",
       "  4: {'A': 8.0, 'C': 32.0, 'G': 2118.0, 'T': 100.0},\n",
       "  5: {'A': 13.0, 'C': 6.0, 'G': 34.0, 'T': 2118.0},\n",
       "  6: {'A': 1.0, 'C': 2118.0, 'G': 2.0, 'T': 17.0},\n",
       "  7: {'A': 2118.0, 'C': 16.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 839.0, 'G': 1.0, 'T': 1280.0},\n",
       "  9: {'A': 65.0, 'C': 56.0, 'G': 2118.0, 'T': 27.0},\n",
       "  10: {'A': 2118.0, 'C': 215.0, 'G': 79.0, 'T': 120.0},\n",
       "  11: {'A': 145.0, 'C': 2118.0, 'G': 64.0, 'T': 65.0},\n",
       "  12: {'A': 145.0, 'C': 2118.0, 'G': 177.0, 'T': 38.0},\n",
       "  13: {'A': 24.0, 'C': 883.0, 'G': 58.0, 'T': 1236.0},\n",
       "  14: {'A': 136.0, 'C': 997.0, 'G': 227.0, 'T': 762.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'P48443',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0002.2',\n",
       "  'Matrix_Name': 'RUNX1',\n",
       "  1: {'A': 288.0, 'C': 497.0, 'G': 697.0, 'T': 522.0},\n",
       "  2: {'A': 235.0, 'C': 486.0, 'G': 468.0, 'T': 815.0},\n",
       "  3: {'A': 124.0, 'C': 1073.0, 'G': 150.0, 'T': 657.0},\n",
       "  4: {'A': 58.0, 'C': 1.0, 'G': 8.0, 'T': 1937.0},\n",
       "  5: {'A': 1.0, 'C': 76.0, 'G': 1873.0, 'T': 54.0},\n",
       "  6: {'A': 88.0, 'C': 128.0, 'G': 71.0, 'T': 1717.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1988.0, 'T': 14.0},\n",
       "  8: {'A': 18.0, 'C': 43.0, 'G': 1849.0, 'T': 94.0},\n",
       "  9: {'A': 11.0, 'C': 401.0, 'G': 252.0, 'T': 1340.0},\n",
       "  10: {'A': 132.0, 'C': 464.0, 'G': 82.0, 'T': 1326.0},\n",
       "  11: {'A': 501.0, 'C': 159.0, 'G': 290.0, 'T': 1054.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Runt-related',\n",
       "  'tf_class': 'Runt',\n",
       "  'pubmed_ids': '-',\n",
       "  'uniprot_ids': 'Q01196',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1476.1',\n",
       "  'Matrix_Name': 'DLX5',\n",
       "  1: {'A': 2116.0, 'C': 3458.0, 'G': 5503.0, 'T': 1846.0},\n",
       "  2: {'A': 126.0, 'C': 7615.0, 'G': 1.0, 'T': 5306.0},\n",
       "  3: {'A': 12920.0, 'C': 69.0, 'G': 48.0, 'T': 1.0},\n",
       "  4: {'A': 12920.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 12920.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 12920.0},\n",
       "  7: {'A': 12920.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 2286.0, 'C': 3812.0, 'G': 3097.0, 'T': 3728.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P56178',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0716.1',\n",
       "  'Matrix_Name': 'PRRX1',\n",
       "  1: {'A': 4133.0, 'C': 9064.0, 'G': 4340.0, 'T': 6365.0},\n",
       "  2: {'A': 2374.0, 'C': 11123.0, 'G': 1122.0, 'T': 12776.0},\n",
       "  3: {'A': 23898.0, 'C': 1089.0, 'G': 1777.0, 'T': 811.0},\n",
       "  4: {'A': 23898.0, 'C': 504.0, 'G': 203.0, 'T': 210.0},\n",
       "  5: {'A': 1.0, 'C': 4.0, 'G': 1.0, 'T': 23898.0},\n",
       "  6: {'A': 1265.0, 'C': 2303.0, 'G': 1005.0, 'T': 23898.0},\n",
       "  7: {'A': 23898.0, 'C': 645.0, 'G': 137.0, 'T': 1230.0},\n",
       "  8: {'A': 9499.0, 'C': 4270.0, 'G': 6367.0, 'T': 3766.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P54821',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1471.1',\n",
       "  'Matrix_Name': 'BARX2',\n",
       "  1: {'A': 8676.0, 'C': 6474.0, 'G': 6746.0, 'T': 6495.0},\n",
       "  2: {'A': 16963.0, 'C': 3614.0, 'G': 3633.0, 'T': 11427.0},\n",
       "  3: {'A': 18518.0, 'C': 70.0, 'G': 48.0, 'T': 9872.0},\n",
       "  4: {'A': 28389.0, 'C': 31.0, 'G': 100.0, 'T': 7508.0},\n",
       "  5: {'A': 28389.0, 'C': 3349.0, 'G': 2921.0, 'T': 2848.0},\n",
       "  6: {'A': 2815.0, 'C': 16935.0, 'G': 1343.0, 'T': 11454.0},\n",
       "  7: {'A': 15760.0, 'C': 28389.0, 'G': 5136.0, 'T': 3453.0},\n",
       "  8: {'A': 28389.0, 'C': 50.0, 'G': 10010.0, 'T': 1.0},\n",
       "  9: {'A': 8.0, 'C': 1.0, 'G': 42.0, 'T': 28389.0},\n",
       "  10: {'A': 20.0, 'C': 2052.0, 'G': 1.0, 'T': 28389.0},\n",
       "  11: {'A': 28389.0, 'C': 326.0, 'G': 416.0, 'T': 4260.0},\n",
       "  12: {'A': 6426.0, 'C': 9282.0, 'G': 7423.0, 'T': 5261.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '10051657',\n",
       "  'uniprot_ids': 'Q9UMQ3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1530.1',\n",
       "  'Matrix_Name': 'NKX6-3',\n",
       "  1: {'A': 1401.0, 'C': 2732.0, 'G': 2724.0, 'T': 52.0},\n",
       "  2: {'A': 1.0, 'C': 4563.0, 'G': 1.0, 'T': 6907.0},\n",
       "  3: {'A': 154.0, 'C': 6754.0, 'G': 534.0, 'T': 627.0},\n",
       "  4: {'A': 6907.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6907.0},\n",
       "  6: {'A': 1.0, 'C': 468.0, 'G': 1.0, 'T': 6907.0},\n",
       "  7: {'A': 6907.0, 'C': 1.0, 'G': 702.0, 'T': 1.0},\n",
       "  8: {'A': 6907.0, 'C': 790.0, 'G': 380.0, 'T': 3754.0},\n",
       "  9: {'A': 3051.0, 'C': 1371.0, 'G': 886.0, 'T': 1602.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'A6NJ46',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1507.1',\n",
       "  'Matrix_Name': 'HOXD4',\n",
       "  1: {'A': 1187.0, 'C': 706.0, 'G': 1156.0, 'T': 360.0},\n",
       "  2: {'A': 1.0, 'C': 1185.0, 'G': 1.0, 'T': 3406.0},\n",
       "  3: {'A': 1543.0, 'C': 1863.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 3406.0, 'C': 1.0, 'G': 177.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3406.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3406.0},\n",
       "  7: {'A': 3406.0, 'C': 1.0, 'G': 195.0, 'T': 1.0},\n",
       "  8: {'A': 1103.0, 'C': 526.0, 'G': 1148.0, 'T': 632.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9079695',\n",
       "  'uniprot_ids': 'P09016',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0597.1',\n",
       "  'Matrix_Name': 'THAP1',\n",
       "  1: {'A': 19.0, 'C': 94.0, 'G': 28.0, 'T': 62.0},\n",
       "  2: {'A': 16.0, 'C': 48.0, 'G': 13.0, 'T': 126.0},\n",
       "  3: {'A': 21.0, 'C': 1.0, 'G': 138.0, 'T': 43.0},\n",
       "  4: {'A': 8.0, 'C': 183.0, 'G': 1.0, 'T': 11.0},\n",
       "  5: {'A': 4.0, 'C': 196.0, 'G': 1.0, 'T': 2.0},\n",
       "  6: {'A': 59.0, 'C': 137.0, 'G': 2.0, 'T': 5.0},\n",
       "  7: {'A': 48.0, 'C': 18.0, 'G': 78.0, 'T': 59.0},\n",
       "  8: {'A': 31.0, 'C': 72.0, 'G': 42.0, 'T': 58.0},\n",
       "  9: {'A': 117.0, 'C': 35.0, 'G': 18.0, 'T': 33.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'THAP-related',\n",
       "  'tf_class': 'C2CH',\n",
       "  'pubmed_ids': '15863623',\n",
       "  'uniprot_ids': 'Q9NVV9',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0600.2',\n",
       "  'Matrix_Name': 'RFX2',\n",
       "  1: {'A': 1382.0, 'C': 5654.0, 'G': 4043.0, 'T': 2337.0},\n",
       "  2: {'A': 271.0, 'C': 13.0, 'G': 13943.0, 'T': 9.0},\n",
       "  3: {'A': 29.0, 'C': 39.0, 'G': 15.0, 'T': 13837.0},\n",
       "  4: {'A': 1991.0, 'C': 889.0, 'G': 15.0, 'T': 12504.0},\n",
       "  5: {'A': 3949.0, 'C': 238.0, 'G': 8027.0, 'T': 1176.0},\n",
       "  6: {'A': 13.0, 'C': 13198.0, 'G': 94.0, 'T': 1460.0},\n",
       "  7: {'A': 3.0, 'C': 12327.0, 'G': 6.0, 'T': 3387.0},\n",
       "  8: {'A': 9774.0, 'C': 39.0, 'G': 66.0, 'T': 185.0},\n",
       "  9: {'A': 216.0, 'C': 89.0, 'G': 30.0, 'T': 13480.0},\n",
       "  10: {'A': 3133.0, 'C': 8.0, 'G': 12114.0, 'T': 3.0},\n",
       "  11: {'A': 2008.0, 'C': 129.0, 'G': 13076.0, 'T': 4.0},\n",
       "  12: {'A': 1360.0, 'C': 8299.0, 'G': 355.0, 'T': 4298.0},\n",
       "  13: {'A': 9267.0, 'C': 27.0, 'G': 959.0, 'T': 1195.0},\n",
       "  14: {'A': 9877.0, 'C': 15.0, 'G': 26.0, 'T': 17.0},\n",
       "  15: {'A': 8.0, 'C': 14068.0, 'G': 2.0, 'T': 310.0},\n",
       "  16: {'A': 2416.0, 'C': 3655.0, 'G': 5452.0, 'T': 1442.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'RFX-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '8754849',\n",
       "  'uniprot_ids': 'P48378',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1506.1',\n",
       "  'Matrix_Name': 'HOXD10',\n",
       "  1: {'A': 1259.0, 'C': 529.0, 'G': 2401.0, 'T': 460.0},\n",
       "  2: {'A': 1103.0, 'C': 413.0, 'G': 4646.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 41.0, 'G': 3.0, 'T': 4646.0},\n",
       "  4: {'A': 33.0, 'C': 4646.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1274.0, 'C': 2.0, 'G': 4646.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 152.0, 'T': 4646.0},\n",
       "  7: {'A': 4646.0, 'C': 1.0, 'G': 19.0, 'T': 371.0},\n",
       "  8: {'A': 4646.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 4646.0, 'C': 37.0, 'G': 1.0, 'T': 8.0},\n",
       "  10: {'A': 4646.0, 'C': 464.0, 'G': 263.0, 'T': 729.0},\n",
       "  11: {'A': 1317.0, 'C': 1451.0, 'G': 484.0, 'T': 1397.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P28358',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0041.1',\n",
       "  'Matrix_Name': 'Foxd3',\n",
       "  1: {'A': 12.0, 'C': 4.0, 'G': 27.0, 'T': 8.0},\n",
       "  2: {'A': 31.0, 'C': 3.0, 'G': 1.0, 'T': 16.0},\n",
       "  3: {'A': 25.0, 'C': 2.0, 'G': 1.0, 'T': 23.0},\n",
       "  4: {'A': 2.0, 'C': 5.0, 'G': 1.0, 'T': 43.0},\n",
       "  5: {'A': 13.0, 'C': 1.0, 'G': 35.0, 'T': 2.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 48.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 47.0},\n",
       "  8: {'A': 6.0, 'C': 1.0, 'G': 11.0, 'T': 33.0},\n",
       "  9: {'A': 15.0, 'C': 1.0, 'G': 22.0, 'T': 13.0},\n",
       "  10: {'A': 7.0, 'C': 8.0, 'G': 1.0, 'T': 35.0},\n",
       "  11: {'A': 2.0, 'C': 3.0, 'G': 5.0, 'T': 41.0},\n",
       "  12: {'A': 1.0, 'C': 13.0, 'G': 1.0, 'T': 36.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '8139574',\n",
       "  'uniprot_ids': 'Q63245',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0856.1',\n",
       "  'Matrix_Name': 'RXRG',\n",
       "  1: {'A': 7825.0, 'C': 3801.0, 'G': 21427.0, 'T': 1887.0},\n",
       "  2: {'A': 14208.0, 'C': 64.0, 'G': 20866.0, 'T': 21.0},\n",
       "  3: {'A': 50.0, 'C': 5.0, 'G': 23555.0, 'T': 78.0},\n",
       "  4: {'A': 103.0, 'C': 10.0, 'G': 22313.0, 'T': 3947.0},\n",
       "  5: {'A': 6.0, 'C': 36.0, 'G': 127.0, 'T': 39124.0},\n",
       "  6: {'A': 29.0, 'C': 56981.0, 'G': 735.0, 'T': 3943.0},\n",
       "  7: {'A': 37923.0, 'C': 1.0, 'G': 77.0, 'T': 5.0},\n",
       "  8: {'A': 34469.0, 'C': 15.0, 'G': 971.0, 'T': 575.0},\n",
       "  9: {'A': 33004.0, 'C': 1.0, 'G': 1635.0, 'T': 6.0},\n",
       "  10: {'A': 10.0, 'C': 2.0, 'G': 29533.0, 'T': 21.0},\n",
       "  11: {'A': 13.0, 'C': 2.0, 'G': 27904.0, 'T': 3914.0},\n",
       "  12: {'A': 3.0, 'C': 18.0, 'G': 135.0, 'T': 37559.0},\n",
       "  13: {'A': 79.0, 'C': 49964.0, 'G': 936.0, 'T': 3000.0},\n",
       "  14: {'A': 36678.0, 'C': 44.0, 'G': 2324.0, 'T': 28.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P48443',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0508.3',\n",
       "  'Matrix_Name': 'PRDM1',\n",
       "  1: {'A': 8763.0, 'C': 17789.0, 'G': 6639.0, 'T': 22294.0},\n",
       "  2: {'A': 17154.0, 'C': 12289.0, 'G': 8769.0, 'T': 17273.0},\n",
       "  3: {'A': 1200.0, 'C': 52753.0, 'G': 702.0, 'T': 830.0},\n",
       "  4: {'A': 1284.0, 'C': 468.0, 'G': 827.0, 'T': 52906.0},\n",
       "  5: {'A': 2182.0, 'C': 1347.0, 'G': 1012.0, 'T': 50944.0},\n",
       "  6: {'A': 1814.0, 'C': 489.0, 'G': 562.0, 'T': 52620.0},\n",
       "  7: {'A': 1378.0, 'C': 53466.0, 'G': 208.0, 'T': 433.0},\n",
       "  8: {'A': 3550.0, 'C': 1077.0, 'G': 1391.0, 'T': 49467.0},\n",
       "  9: {'A': 1711.0, 'C': 52822.0, 'G': 291.0, 'T': 661.0},\n",
       "  10: {'A': 10523.0, 'C': 13568.0, 'G': 4099.0, 'T': 27295.0},\n",
       "  11: {'A': 10125.0, 'C': 17519.0, 'G': 7862.0, 'T': 19979.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'O75626',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0874.1',\n",
       "  'Matrix_Name': 'Arx',\n",
       "  1: {'A': 25.0, 'C': 23.0, 'G': 41.0, 'T': 15.0},\n",
       "  2: {'A': 14.0, 'C': 17.0, 'G': 20.0, 'T': 53.0},\n",
       "  3: {'A': 17.0, 'C': 38.0, 'G': 30.0, 'T': 20.0},\n",
       "  4: {'A': 23.0, 'C': 57.0, 'G': 14.0, 'T': 11.0},\n",
       "  5: {'A': 51.0, 'C': 7.0, 'G': 39.0, 'T': 8.0},\n",
       "  6: {'A': 2.0, 'C': 39.0, 'G': 2.0, 'T': 61.0},\n",
       "  7: {'A': 2.0, 'C': 14.0, 'G': 1.0, 'T': 87.0},\n",
       "  8: {'A': 99.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 100.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  10: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 100.0},\n",
       "  11: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 99.0},\n",
       "  12: {'A': 87.0, 'C': 1.0, 'G': 14.0, 'T': 2.0},\n",
       "  13: {'A': 61.0, 'C': 2.0, 'G': 39.0, 'T': 2.0},\n",
       "  14: {'A': 17.0, 'C': 17.0, 'G': 15.0, 'T': 55.0},\n",
       "  15: {'A': 19.0, 'C': 12.0, 'G': 45.0, 'T': 28.0},\n",
       "  16: {'A': 10.0, 'C': 37.0, 'G': 38.0, 'T': 19.0},\n",
       "  17: {'A': 54.0, 'C': 11.0, 'G': 8.0, 'T': 31.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'O35085',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0744.2',\n",
       "  'Matrix_Name': 'SCRT2',\n",
       "  1: {'A': 18856.0, 'C': 13346.0, 'G': 17681.0, 'T': 15731.0},\n",
       "  2: {'A': 23311.0, 'C': 12733.0, 'G': 17498.0, 'T': 12072.0},\n",
       "  3: {'A': 29946.0, 'C': 6320.0, 'G': 12474.0, 'T': 16874.0},\n",
       "  4: {'A': 5354.0, 'C': 7237.0, 'G': 35770.0, 'T': 17253.0},\n",
       "  5: {'A': 1379.0, 'C': 59162.0, 'G': 1340.0, 'T': 3733.0},\n",
       "  6: {'A': 53426.0, 'C': 5537.0, 'G': 2266.0, 'T': 4385.0},\n",
       "  7: {'A': 62367.0, 'C': 414.0, 'G': 1879.0, 'T': 954.0},\n",
       "  8: {'A': 232.0, 'C': 64857.0, 'G': 320.0, 'T': 205.0},\n",
       "  9: {'A': 63881.0, 'C': 295.0, 'G': 253.0, 'T': 1185.0},\n",
       "  10: {'A': 7307.0, 'C': 729.0, 'G': 56106.0, 'T': 1472.0},\n",
       "  11: {'A': 1863.0, 'C': 973.0, 'G': 60637.0, 'T': 2141.0},\n",
       "  12: {'A': 2079.0, 'C': 3476.0, 'G': 1169.0, 'T': 58890.0},\n",
       "  13: {'A': 10482.0, 'C': 5662.0, 'G': 38134.0, 'T': 11336.0},\n",
       "  14: {'A': 13384.0, 'C': 11352.0, 'G': 25289.0, 'T': 15589.0},\n",
       "  15: {'A': 16059.0, 'C': 16086.0, 'G': 14717.0, 'T': 18752.0},\n",
       "  16: {'A': 17882.0, 'C': 13847.0, 'G': 12431.0, 'T': 21454.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q9NQ03',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0048.2',\n",
       "  'Matrix_Name': 'NHLH1',\n",
       "  1: {'A': 789.0, 'C': 2167.0, 'G': 182.0, 'T': 112.0},\n",
       "  2: {'A': 462.0, 'C': 180.0, 'G': 2167.0, 'T': 143.0},\n",
       "  3: {'A': 4.0, 'C': 2167.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 2167.0, 'C': 4.0, 'G': 2.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 302.0, 'G': 2167.0, 'T': 115.0},\n",
       "  6: {'A': 97.0, 'C': 2167.0, 'G': 125.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 5.0, 'G': 5.0, 'T': 2167.0},\n",
       "  8: {'A': 3.0, 'C': 5.0, 'G': 2167.0, 'T': 2.0},\n",
       "  9: {'A': 297.0, 'C': 2167.0, 'G': 103.0, 'T': 351.0},\n",
       "  10: {'A': 309.0, 'C': 737.0, 'G': 2167.0, 'T': 1490.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8289804',\n",
       "  'uniprot_ids': 'Q02575',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0057.1',\n",
       "  'Matrix_Name': 'MZF1(var.2)',\n",
       "  1: {'A': 2.0, 'C': 5.0, 'G': 8.0, 'T': 5.0},\n",
       "  2: {'A': 3.0, 'C': 1.0, 'G': 8.0, 'T': 8.0},\n",
       "  3: {'A': 16.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 12.0, 'T': 6.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 16.0, 'T': 2.0},\n",
       "  6: {'A': 1.0, 'C': 3.0, 'G': 15.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 15.0, 'T': 3.0},\n",
       "  8: {'A': 4.0, 'C': 2.0, 'G': 9.0, 'T': 5.0},\n",
       "  9: {'A': 11.0, 'C': 1.0, 'G': 5.0, 'T': 3.0},\n",
       "  10: {'A': 9.0, 'C': 3.0, 'G': 5.0, 'T': 3.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '8114711',\n",
       "  'uniprot_ids': 'P28698',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1116.1',\n",
       "  'Matrix_Name': 'RBPJ',\n",
       "  1: {'A': 346.0, 'C': 843.0, 'G': 605.0, 'T': 612.0},\n",
       "  2: {'A': 490.0, 'C': 873.0, 'G': 788.0, 'T': 255.0},\n",
       "  3: {'A': 3.0, 'C': 13.0, 'G': 3.0, 'T': 2387.0},\n",
       "  4: {'A': 36.0, 'C': 26.0, 'G': 2342.0, 'T': 2.0},\n",
       "  5: {'A': 10.0, 'C': 21.0, 'G': 2314.0, 'T': 61.0},\n",
       "  6: {'A': 8.0, 'C': 9.0, 'G': 2310.0, 'T': 79.0},\n",
       "  7: {'A': 2370.0, 'C': 1.0, 'G': 5.0, 'T': 30.0},\n",
       "  8: {'A': 2383.0, 'C': 16.0, 'G': 2.0, 'T': 5.0},\n",
       "  9: {'A': 743.0, 'C': 656.0, 'G': 593.0, 'T': 414.0},\n",
       "  10: {'A': 566.0, 'C': 580.0, 'G': 712.0, 'T': 548.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'CSL-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '23651858',\n",
       "  'uniprot_ids': 'Q06330',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1537.1',\n",
       "  'Matrix_Name': 'NR2F1(var.2)',\n",
       "  1: {'A': 12165.0, 'C': 4592.0, 'G': 13409.0, 'T': 3793.0},\n",
       "  2: {'A': 20463.0, 'C': 437.0, 'G': 13495.0, 'T': 281.0},\n",
       "  3: {'A': 318.0, 'C': 123.0, 'G': 33957.0, 'T': 768.0},\n",
       "  4: {'A': 469.0, 'C': 43.0, 'G': 33957.0, 'T': 745.0},\n",
       "  5: {'A': 9.0, 'C': 466.0, 'G': 1341.0, 'T': 33957.0},\n",
       "  6: {'A': 11.0, 'C': 33957.0, 'G': 642.0, 'T': 2240.0},\n",
       "  7: {'A': 33957.0, 'C': 7.0, 'G': 607.0, 'T': 20.0},\n",
       "  8: {'A': 33957.0, 'C': 229.0, 'G': 3362.0, 'T': 2712.0},\n",
       "  9: {'A': 33957.0, 'C': 49.0, 'G': 2543.0, 'T': 42.0},\n",
       "  10: {'A': 14.0, 'C': 1.0, 'G': 33957.0, 'T': 60.0},\n",
       "  11: {'A': 243.0, 'C': 1.0, 'G': 33957.0, 'T': 495.0},\n",
       "  12: {'A': 22.0, 'C': 752.0, 'G': 1855.0, 'T': 33957.0},\n",
       "  13: {'A': 38.0, 'C': 33957.0, 'G': 1544.0, 'T': 4149.0},\n",
       "  14: {'A': 33957.0, 'C': 280.0, 'G': 6238.0, 'T': 721.0},\n",
       "  15: {'A': 9118.0, 'C': 9478.0, 'G': 7333.0, 'T': 8030.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'P10589',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0474.2',\n",
       "  'Matrix_Name': 'ERG',\n",
       "  1: {'A': 4567.0, 'C': 269.0, 'G': 849.0, 'T': 929.0},\n",
       "  2: {'A': 647.0, 'C': 4567.0, 'G': 399.0, 'T': 98.0},\n",
       "  3: {'A': 636.0, 'C': 4567.0, 'G': 23.0, 'T': 7.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 4567.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 2.0, 'G': 4567.0, 'T': 30.0},\n",
       "  6: {'A': 4567.0, 'C': 21.0, 'G': 1.0, 'T': 8.0},\n",
       "  7: {'A': 4567.0, 'C': 120.0, 'G': 8.0, 'T': 1246.0},\n",
       "  8: {'A': 1991.0, 'C': 101.0, 'G': 2577.0, 'T': 29.0},\n",
       "  9: {'A': 138.0, 'C': 1587.0, 'G': 356.0, 'T': 4567.0},\n",
       "  10: {'A': 1007.0, 'C': 1101.0, 'G': 1669.0, 'T': 793.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'P11308',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0753.2',\n",
       "  'Matrix_Name': 'ZNF740',\n",
       "  1: {'A': 187.0, 'C': 503.0, 'G': 183.0, 'T': 186.0},\n",
       "  2: {'A': 163.0, 'C': 678.0, 'G': 73.0, 'T': 379.0},\n",
       "  3: {'A': 728.0, 'C': 344.0, 'G': 1056.0, 'T': 313.0},\n",
       "  4: {'A': 61.0, 'C': 1056.0, 'G': 38.0, 'T': 50.0},\n",
       "  5: {'A': 111.0, 'C': 1056.0, 'G': 15.0, 'T': 2.0},\n",
       "  6: {'A': 19.0, 'C': 1056.0, 'G': 5.0, 'T': 17.0},\n",
       "  7: {'A': 23.0, 'C': 1056.0, 'G': 8.0, 'T': 16.0},\n",
       "  8: {'A': 24.0, 'C': 1056.0, 'G': 4.0, 'T': 12.0},\n",
       "  9: {'A': 11.0, 'C': 1056.0, 'G': 8.0, 'T': 7.0},\n",
       "  10: {'A': 3.0, 'C': 1056.0, 'G': 4.0, 'T': 4.0},\n",
       "  11: {'A': 116.0, 'C': 1056.0, 'G': 7.0, 'T': 5.0},\n",
       "  12: {'A': 1056.0, 'C': 240.0, 'G': 69.0, 'T': 135.0},\n",
       "  13: {'A': 208.0, 'C': 1056.0, 'G': 55.0, 'T': 247.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Other',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'Q8NDX6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0830.2',\n",
       "  'Matrix_Name': 'TCF4',\n",
       "  1: {'A': 6049.0, 'C': 10053.0, 'G': 7141.0, 'T': 6226.0},\n",
       "  2: {'A': 7035.0, 'C': 6755.0, 'G': 8619.0, 'T': 7060.0},\n",
       "  3: {'A': 4169.0, 'C': 5063.0, 'G': 18499.0, 'T': 1738.0},\n",
       "  4: {'A': 244.0, 'C': 28687.0, 'G': 322.0, 'T': 216.0},\n",
       "  5: {'A': 27024.0, 'C': 636.0, 'G': 1296.0, 'T': 513.0},\n",
       "  6: {'A': 217.0, 'C': 26766.0, 'G': 1734.0, 'T': 752.0},\n",
       "  7: {'A': 357.0, 'C': 27725.0, 'G': 1198.0, 'T': 189.0},\n",
       "  8: {'A': 373.0, 'C': 1017.0, 'G': 680.0, 'T': 27399.0},\n",
       "  9: {'A': 252.0, 'C': 541.0, 'G': 28433.0, 'T': 243.0},\n",
       "  10: {'A': 1695.0, 'C': 18936.0, 'G': 5282.0, 'T': 3556.0},\n",
       "  11: {'A': 5535.0, 'C': 12873.0, 'G': 4934.0, 'T': 6127.0},\n",
       "  12: {'A': 6414.0, 'C': 8659.0, 'G': 7965.0, 'T': 6431.0},\n",
       "  13: {'A': 5134.0, 'C': 9844.0, 'G': 7663.0, 'T': 6828.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'E2A-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '31081034',\n",
       "  'uniprot_ids': 'P15884',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0466.2',\n",
       "  'Matrix_Name': 'CEBPB',\n",
       "  1: {'A': 10586.0, 'C': 5008.0, 'G': 6283.0, 'T': 860.0},\n",
       "  2: {'A': 34.0, 'C': 35.0, 'G': 9.0, 'T': 22733.0},\n",
       "  3: {'A': 18.0, 'C': 15.0, 'G': 2277.0, 'T': 22733.0},\n",
       "  4: {'A': 10931.0, 'C': 35.0, 'G': 11804.0, 'T': 2238.0},\n",
       "  5: {'A': 394.0, 'C': 22733.0, 'G': 222.0, 'T': 2218.0},\n",
       "  6: {'A': 3522.0, 'C': 556.0, 'G': 22733.0, 'T': 487.0},\n",
       "  7: {'A': 2528.0, 'C': 22733.0, 'G': 4.0, 'T': 2136.0},\n",
       "  8: {'A': 22733.0, 'C': 637.0, 'G': 10.0, 'T': 1.0},\n",
       "  9: {'A': 22733.0, 'C': 10.0, 'G': 1.0, 'T': 10.0},\n",
       "  10: {'A': 156.0, 'C': 10618.0, 'G': 941.0, 'T': 12117.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8632009',\n",
       "  'uniprot_ids': 'P17676',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0625.1',\n",
       "  'Matrix_Name': 'NFATC3',\n",
       "  1: {'A': 422.0, 'C': 94.0, 'G': 140.0, 'T': 349.0},\n",
       "  2: {'A': 180.0, 'C': 182.0, 'G': 169.0, 'T': 473.0},\n",
       "  3: {'A': 118.0, 'C': 41.0, 'G': 74.0, 'T': 771.0},\n",
       "  4: {'A': 15.0, 'C': 11.0, 'G': 6.0, 'T': 972.0},\n",
       "  5: {'A': 4.0, 'C': 24.0, 'G': 10.0, 'T': 967.0},\n",
       "  6: {'A': 37.0, 'C': 934.0, 'G': 2.0, 'T': 31.0},\n",
       "  7: {'A': 9.0, 'C': 960.0, 'G': 24.0, 'T': 11.0},\n",
       "  8: {'A': 553.0, 'C': 40.0, 'G': 343.0, 'T': 68.0},\n",
       "  9: {'A': 174.0, 'C': 252.0, 'G': 134.0, 'T': 444.0},\n",
       "  10: {'A': 216.0, 'C': 212.0, 'G': 174.0, 'T': 401.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NFAT-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '9143705',\n",
       "  'uniprot_ids': 'Q12968',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0042.2',\n",
       "  'Matrix_Name': 'FOXI1',\n",
       "  1: {'A': 619.0, 'C': 40.0, 'G': 5176.0, 'T': 1.0},\n",
       "  2: {'A': 75.0, 'C': 61.0, 'G': 36.0, 'T': 5176.0},\n",
       "  3: {'A': 5176.0, 'C': 541.0, 'G': 38.0, 'T': 1.0},\n",
       "  4: {'A': 5176.0, 'C': 24.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 5176.0, 'C': 24.0, 'G': 30.0, 'T': 67.0},\n",
       "  6: {'A': 158.0, 'C': 5176.0, 'G': 49.0, 'T': 811.0},\n",
       "  7: {'A': 5176.0, 'C': 13.0, 'G': 26.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23836653',\n",
       "  'uniprot_ids': 'Q12951',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0680.1',\n",
       "  'Matrix_Name': 'PAX7',\n",
       "  1: {'A': 680.0, 'C': 69.0, 'G': 99.0, 'T': 3709.0},\n",
       "  2: {'A': 3709.0, 'C': 19.0, 'G': 17.0, 'T': 7.0},\n",
       "  3: {'A': 3709.0, 'C': 4.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 7.0, 'C': 22.0, 'G': 4.0, 'T': 3709.0},\n",
       "  5: {'A': 11.0, 'C': 2935.0, 'G': 25.0, 'T': 775.0},\n",
       "  6: {'A': 1136.0, 'C': 28.0, 'G': 2574.0, 'T': 1.0},\n",
       "  7: {'A': 3709.0, 'C': 1.0, 'G': 11.0, 'T': 11.0},\n",
       "  8: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 3709.0},\n",
       "  9: {'A': 22.0, 'C': 26.0, 'G': 10.0, 'T': 3709.0},\n",
       "  10: {'A': 3709.0, 'C': 73.0, 'G': 48.0, 'T': 250.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '10567552',\n",
       "  'uniprot_ids': 'P23759',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1497.1',\n",
       "  'Matrix_Name': 'HOXA6',\n",
       "  1: {'A': 1276.0, 'C': 1596.0, 'G': 2729.0, 'T': 1604.0},\n",
       "  2: {'A': 1338.0, 'C': 1726.0, 'G': 7202.0, 'T': 1606.0},\n",
       "  3: {'A': 88.0, 'C': 2968.0, 'G': 101.0, 'T': 4235.0},\n",
       "  4: {'A': 7202.0, 'C': 3769.0, 'G': 834.0, 'T': 1483.0},\n",
       "  5: {'A': 7202.0, 'C': 746.0, 'G': 909.0, 'T': 12.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7202.0},\n",
       "  7: {'A': 1912.0, 'C': 234.0, 'G': 554.0, 'T': 7202.0},\n",
       "  8: {'A': 7202.0, 'C': 95.0, 'G': 503.0, 'T': 664.0},\n",
       "  9: {'A': 2701.0, 'C': 1378.0, 'G': 1633.0, 'T': 1493.0},\n",
       "  10: {'A': 1235.0, 'C': 2458.0, 'G': 2021.0, 'T': 1490.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31267',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0479.1',\n",
       "  'Matrix_Name': 'FOXH1',\n",
       "  1: {'A': 940.0, 'C': 2841.0, 'G': 1491.0, 'T': 2943.0},\n",
       "  2: {'A': 1579.0, 'C': 2334.0, 'G': 2138.0, 'T': 2164.0},\n",
       "  3: {'A': 974.0, 'C': 3262.0, 'G': 3157.0, 'T': 822.0},\n",
       "  4: {'A': 8212.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 7548.0, 'C': 1.0, 'G': 230.0, 'T': 436.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 8212.0},\n",
       "  7: {'A': 2375.0, 'C': 5774.0, 'G': 1.0, 'T': 65.0},\n",
       "  8: {'A': 1415.0, 'C': 6798.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 8212.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 8212.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 7139.0, 'C': 1.0, 'G': 1.0, 'T': 1074.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '9702198',\n",
       "  'uniprot_ids': 'O75593',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1518.1',\n",
       "  'Matrix_Name': 'LHX1',\n",
       "  1: {'A': 2685.0, 'C': 4119.0, 'G': 5235.0, 'T': 2919.0},\n",
       "  2: {'A': 950.0, 'C': 9231.0, 'G': 573.0, 'T': 5725.0},\n",
       "  3: {'A': 14955.0, 'C': 4879.0, 'G': 4036.0, 'T': 2085.0},\n",
       "  4: {'A': 14955.0, 'C': 271.0, 'G': 40.0, 'T': 570.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 14955.0},\n",
       "  6: {'A': 1102.0, 'C': 5414.0, 'G': 3101.0, 'T': 14955.0},\n",
       "  7: {'A': 14955.0, 'C': 496.0, 'G': 618.0, 'T': 2106.0},\n",
       "  8: {'A': 3024.0, 'C': 5038.0, 'G': 3909.0, 'T': 2987.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P48742',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0660.1',\n",
       "  'Matrix_Name': 'MEF2B',\n",
       "  1: {'A': 1386.0, 'C': 73.0, 'G': 1767.0, 'T': 506.0},\n",
       "  2: {'A': 19.0, 'C': 3151.0, 'G': 1.0, 'T': 13.0},\n",
       "  3: {'A': 1.0, 'C': 26.0, 'G': 1.0, 'T': 3151.0},\n",
       "  4: {'A': 3151.0, 'C': 1.0, 'G': 1.0, 'T': 16.0},\n",
       "  5: {'A': 1496.0, 'C': 3.0, 'G': 1.0, 'T': 1656.0},\n",
       "  6: {'A': 3151.0, 'C': 1.0, 'G': 13.0, 'T': 711.0},\n",
       "  7: {'A': 3151.0, 'C': 4.0, 'G': 3.0, 'T': 144.0},\n",
       "  8: {'A': 3151.0, 'C': 1.0, 'G': 1.0, 'T': 46.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3151.0},\n",
       "  10: {'A': 3151.0, 'C': 4.0, 'G': 3.0, 'T': 4.0},\n",
       "  11: {'A': 99.0, 'C': 20.0, 'G': 3151.0, 'T': 4.0},\n",
       "  12: {'A': 1014.0, 'C': 2139.0, 'G': 100.0, 'T': 507.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Regulators',\n",
       "  'tf_class': 'MADS',\n",
       "  'pubmed_ids': '9430690',\n",
       "  'uniprot_ids': 'Q02080',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0491.2',\n",
       "  'Matrix_Name': 'JUND',\n",
       "  1: {'A': 28468.0, 'C': 21966.0, 'G': 25007.0, 'T': 30508.0},\n",
       "  2: {'A': 29235.0, 'C': 19713.0, 'G': 31348.0, 'T': 25653.0},\n",
       "  3: {'A': 70167.0, 'C': 10174.0, 'G': 20587.0, 'T': 5021.0},\n",
       "  4: {'A': 645.0, 'C': 758.0, 'G': 444.0, 'T': 104102.0},\n",
       "  5: {'A': 1366.0, 'C': 874.0, 'G': 99788.0, 'T': 3921.0},\n",
       "  6: {'A': 103798.0, 'C': 651.0, 'G': 574.0, 'T': 926.0},\n",
       "  7: {'A': 4635.0, 'C': 76029.0, 'G': 20634.0, 'T': 4651.0},\n",
       "  8: {'A': 2054.0, 'C': 497.0, 'G': 1148.0, 'T': 102250.0},\n",
       "  9: {'A': 4916.0, 'C': 98629.0, 'G': 954.0, 'T': 1450.0},\n",
       "  10: {'A': 103421.0, 'C': 752.0, 'G': 966.0, 'T': 810.0},\n",
       "  11: {'A': 2797.0, 'C': 12026.0, 'G': 7909.0, 'T': 83217.0},\n",
       "  12: {'A': 27970.0, 'C': 28251.0, 'G': 22250.0, 'T': 27478.0},\n",
       "  13: {'A': 28213.0, 'C': 29794.0, 'G': 19237.0, 'T': 28705.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21526160',\n",
       "  'uniprot_ids': 'P17535',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0150.2',\n",
       "  'Matrix_Name': 'Nfe2l2',\n",
       "  1: {'A': 167.0, 'C': 339.0, 'G': 110.0, 'T': 114.0},\n",
       "  2: {'A': 402.0, 'C': 79.0, 'G': 160.0, 'T': 89.0},\n",
       "  3: {'A': 119.0, 'C': 240.0, 'G': 276.0, 'T': 95.0},\n",
       "  4: {'A': 204.0, 'C': 248.0, 'G': 153.0, 'T': 125.0},\n",
       "  5: {'A': 489.0, 'C': 29.0, 'G': 206.0, 'T': 6.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 727.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 712.0, 'T': 16.0},\n",
       "  8: {'A': 727.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 21.0, 'C': 534.0, 'G': 119.0, 'T': 56.0},\n",
       "  10: {'A': 85.0, 'C': 21.0, 'G': 17.0, 'T': 607.0},\n",
       "  11: {'A': 66.0, 'C': 550.0, 'G': 47.0, 'T': 67.0},\n",
       "  12: {'A': 578.0, 'C': 4.0, 'G': 91.0, 'T': 57.0},\n",
       "  13: {'A': 11.0, 'C': 12.0, 'G': 699.0, 'T': 8.0},\n",
       "  14: {'A': 5.0, 'C': 700.0, 'G': 20.0, 'T': 5.0},\n",
       "  15: {'A': 624.0, 'C': 27.0, 'G': 32.0, 'T': 47.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'Q60795',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1618.1',\n",
       "  'Matrix_Name': 'Ptf1a',\n",
       "  1: {'A': 4093.0, 'C': 2990.0, 'G': 3662.0, 'T': 3700.0},\n",
       "  2: {'A': 4477.0, 'C': 3663.0, 'G': 3569.0, 'T': 2736.0},\n",
       "  3: {'A': 8555.0, 'C': 2748.0, 'G': 2415.0, 'T': 727.0},\n",
       "  4: {'A': 134.0, 'C': 14007.0, 'G': 131.0, 'T': 173.0},\n",
       "  5: {'A': 14065.0, 'C': 119.0, 'G': 154.0, 'T': 107.0},\n",
       "  6: {'A': 251.0, 'C': 467.0, 'G': 13073.0, 'T': 654.0},\n",
       "  7: {'A': 12133.0, 'C': 1725.0, 'G': 438.0, 'T': 149.0},\n",
       "  8: {'A': 127.0, 'C': 225.0, 'G': 119.0, 'T': 13974.0},\n",
       "  9: {'A': 242.0, 'C': 123.0, 'G': 13869.0, 'T': 211.0},\n",
       "  10: {'A': 966.0, 'C': 2059.0, 'G': 2855.0, 'T': 8565.0},\n",
       "  11: {'A': 1512.0, 'C': 2342.0, 'G': 2296.0, 'T': 8295.0},\n",
       "  12: {'A': 3492.0, 'C': 3602.0, 'G': 3665.0, 'T': 3686.0},\n",
       "  13: {'A': 3755.0, 'C': 3690.0, 'G': 3377.0, 'T': 3623.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23754747',\n",
       "  'uniprot_ids': 'Q9QX98',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0525.2',\n",
       "  'Matrix_Name': 'TP63',\n",
       "  1: {'A': 1587.0, 'C': 27.0, 'G': 998.0, 'T': 116.0},\n",
       "  2: {'A': 2262.0, 'C': 2.0, 'G': 186.0, 'T': 22.0},\n",
       "  3: {'A': 1.0, 'C': 2534.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 2247.0, 'C': 1.0, 'G': 77.0, 'T': 5.0},\n",
       "  5: {'A': 693.0, 'C': 34.0, 'G': 1.0, 'T': 2354.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 4022.0, 'T': 1.0},\n",
       "  7: {'A': 15.0, 'C': 86.0, 'G': 1.0, 'T': 3026.0},\n",
       "  8: {'A': 98.0, 'C': 1026.0, 'G': 27.0, 'T': 1962.0},\n",
       "  9: {'A': 650.0, 'C': 479.0, 'G': 2071.0, 'T': 14.0},\n",
       "  10: {'A': 113.0, 'C': 682.0, 'G': 1657.0, 'T': 830.0},\n",
       "  11: {'A': 994.0, 'C': 2.0, 'G': 2133.0, 'T': 48.0},\n",
       "  12: {'A': 2193.0, 'C': 1.0, 'G': 49.0, 'T': 46.0},\n",
       "  13: {'A': 1.0, 'C': 2789.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 2152.0, 'C': 1.0, 'G': 2.0, 'T': 193.0},\n",
       "  15: {'A': 76.0, 'C': 618.0, 'G': 9.0, 'T': 2824.0},\n",
       "  16: {'A': 1.0, 'C': 1.0, 'G': 3533.0, 'T': 1.0},\n",
       "  17: {'A': 50.0, 'C': 203.0, 'G': 2.0, 'T': 3127.0},\n",
       "  18: {'A': 702.0, 'C': 2100.0, 'G': 199.0, 'T': 607.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'p53-related',\n",
       "  'tf_class': 'p53',\n",
       "  'pubmed_ids': '17188034',\n",
       "  'uniprot_ids': 'Q9H3D4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0025.2',\n",
       "  'Matrix_Name': 'NFIL3',\n",
       "  1: {'A': 6369.0, 'C': 3905.0, 'G': 5616.0, 'T': 7110.0},\n",
       "  2: {'A': 9042.0, 'C': 3897.0, 'G': 7262.0, 'T': 2799.0},\n",
       "  3: {'A': 871.0, 'C': 682.0, 'G': 519.0, 'T': 20928.0},\n",
       "  4: {'A': 469.0, 'C': 209.0, 'G': 1715.0, 'T': 20607.0},\n",
       "  5: {'A': 21354.0, 'C': 221.0, 'G': 991.0, 'T': 434.0},\n",
       "  6: {'A': 660.0, 'C': 1512.0, 'G': 493.0, 'T': 20335.0},\n",
       "  7: {'A': 1461.0, 'C': 468.0, 'G': 20460.0, 'T': 611.0},\n",
       "  8: {'A': 1359.0, 'C': 12005.0, 'G': 455.0, 'T': 9181.0},\n",
       "  9: {'A': 21849.0, 'C': 544.0, 'G': 172.0, 'T': 435.0},\n",
       "  10: {'A': 21912.0, 'C': 252.0, 'G': 361.0, 'T': 475.0},\n",
       "  11: {'A': 2194.0, 'C': 4960.0, 'G': 2702.0, 'T': 13144.0},\n",
       "  12: {'A': 7672.0, 'C': 5631.0, 'G': 4153.0, 'T': 5544.0},\n",
       "  13: {'A': 6693.0, 'C': 4930.0, 'G': 4483.0, 'T': 6894.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9122243',\n",
       "  'uniprot_ids': 'Q16649',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0880.1',\n",
       "  'Matrix_Name': 'Dlx3',\n",
       "  1: {'A': 1698.0, 'C': 2339.0, 'G': 2080.0, 'T': 1209.0},\n",
       "  2: {'A': 573.0, 'C': 4134.0, 'G': 165.0, 'T': 3191.0},\n",
       "  3: {'A': 7324.0, 'C': 996.0, 'G': 588.0, 'T': 644.0},\n",
       "  4: {'A': 7324.0, 'C': 537.0, 'G': 63.0, 'T': 300.0},\n",
       "  5: {'A': 174.0, 'C': 67.0, 'G': 61.0, 'T': 7324.0},\n",
       "  6: {'A': 400.0, 'C': 536.0, 'G': 453.0, 'T': 7324.0},\n",
       "  7: {'A': 7324.0, 'C': 203.0, 'G': 102.0, 'T': 692.0},\n",
       "  8: {'A': 2028.0, 'C': 2132.0, 'G': 1357.0, 'T': 1810.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q64205',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0902.2',\n",
       "  'Matrix_Name': 'HOXB2',\n",
       "  1: {'A': 808.0, 'C': 2039.0, 'G': 2926.0, 'T': 1057.0},\n",
       "  2: {'A': 52.0, 'C': 2835.0, 'G': 1.0, 'T': 3994.0},\n",
       "  3: {'A': 6828.0, 'C': 3277.0, 'G': 200.0, 'T': 1.0},\n",
       "  4: {'A': 6828.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6828.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1877.0, 'T': 6828.0},\n",
       "  7: {'A': 6828.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1274.0, 'C': 2032.0, 'G': 2934.0, 'T': 590.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P14652',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0009.2',\n",
       "  'Matrix_Name': 'TBXT',\n",
       "  1: {'A': 53.0, 'C': 157.0, 'G': 23.0, 'T': 6887.0},\n",
       "  2: {'A': 2242.0, 'C': 4779.0, 'G': 1322.0, 'T': 255.0},\n",
       "  3: {'A': 3817.0, 'C': 56.0, 'G': 1416.0, 'T': 218.0},\n",
       "  4: {'A': 1.0, 'C': 5263.0, 'G': 1.0, 'T': 3.0},\n",
       "  5: {'A': 4108.0, 'C': 1.0, 'G': 142.0, 'T': 1.0},\n",
       "  6: {'A': 219.0, 'C': 5105.0, 'G': 112.0, 'T': 401.0},\n",
       "  7: {'A': 2278.0, 'C': 4419.0, 'G': 1668.0, 'T': 810.0},\n",
       "  8: {'A': 630.0, 'C': 612.0, 'G': 215.0, 'T': 6552.0},\n",
       "  9: {'A': 4162.0, 'C': 122.0, 'G': 492.0, 'T': 418.0},\n",
       "  10: {'A': 1067.0, 'C': 2005.0, 'G': 5820.0, 'T': 2922.0},\n",
       "  11: {'A': 666.0, 'C': 136.0, 'G': 7305.0, 'T': 367.0},\n",
       "  12: {'A': 1.0, 'C': 191.0, 'G': 14.0, 'T': 6778.0},\n",
       "  13: {'A': 13.0, 'C': 1.0, 'G': 7629.0, 'T': 1.0},\n",
       "  14: {'A': 505.0, 'C': 2353.0, 'G': 70.0, 'T': 5976.0},\n",
       "  15: {'A': 374.0, 'C': 1238.0, 'G': 7059.0, 'T': 1658.0},\n",
       "  16: {'A': 4172.0, 'C': 13.0, 'G': 112.0, 'T': 18.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Brachyury-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '7588606',\n",
       "  'uniprot_ids': 'O15178',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1499.1',\n",
       "  'Matrix_Name': 'HOXB4',\n",
       "  1: {'A': 4684.0, 'C': 1378.0, 'G': 4398.0, 'T': 1022.0},\n",
       "  2: {'A': 1.0, 'C': 2104.0, 'G': 1.0, 'T': 9081.0},\n",
       "  3: {'A': 3622.0, 'C': 5460.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 9081.0, 'C': 1.0, 'G': 266.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 9081.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 272.0, 'T': 9081.0},\n",
       "  7: {'A': 9081.0, 'C': 1.0, 'G': 56.0, 'T': 1.0},\n",
       "  8: {'A': 4398.0, 'C': 1410.0, 'G': 4684.0, 'T': 1389.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P17483',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1580.1',\n",
       "  'Matrix_Name': 'ZBTB32',\n",
       "  1: {'A': 883.0, 'C': 765.0, 'G': 66.0, 'T': 1850.0},\n",
       "  2: {'A': 357.0, 'C': 4.0, 'G': 3058.0, 'T': 157.0},\n",
       "  3: {'A': 6.0, 'C': 2.0, 'G': 4.0, 'T': 3564.0},\n",
       "  4: {'A': 3561.0, 'C': 6.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 3561.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 3557.0, 'C': 4.0, 'G': 5.0, 'T': 9.0},\n",
       "  7: {'A': 1.0, 'C': 4.0, 'G': 3563.0, 'T': 7.0},\n",
       "  8: {'A': 1.0, 'C': 5.0, 'G': 1.0, 'T': 3559.0},\n",
       "  9: {'A': 2391.0, 'C': 110.0, 'G': 724.0, 'T': 340.0},\n",
       "  10: {'A': 1560.0, 'C': 214.0, 'G': 216.0, 'T': 1575.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Other',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23794217',\n",
       "  'uniprot_ids': 'Q9Y2Y4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0765.2',\n",
       "  'Matrix_Name': 'ETV5',\n",
       "  1: {'A': 4532.0, 'C': 8004.0, 'G': 7118.0, 'T': 6668.0},\n",
       "  2: {'A': 4783.0, 'C': 10308.0, 'G': 6130.0, 'T': 5101.0},\n",
       "  3: {'A': 19411.0, 'C': 2444.0, 'G': 3018.0, 'T': 1449.0},\n",
       "  4: {'A': 313.0, 'C': 24042.0, 'G': 995.0, 'T': 972.0},\n",
       "  5: {'A': 1044.0, 'C': 1146.0, 'G': 409.0, 'T': 23723.0},\n",
       "  6: {'A': 223.0, 'C': 957.0, 'G': 817.0, 'T': 24325.0},\n",
       "  7: {'A': 244.0, 'C': 25485.0, 'G': 337.0, 'T': 256.0},\n",
       "  8: {'A': 368.0, 'C': 25062.0, 'G': 499.0, 'T': 393.0},\n",
       "  9: {'A': 1296.0, 'C': 1744.0, 'G': 18758.0, 'T': 4524.0},\n",
       "  10: {'A': 3018.0, 'C': 6922.0, 'G': 11945.0, 'T': 4437.0},\n",
       "  11: {'A': 4680.0, 'C': 8379.0, 'G': 6040.0, 'T': 7223.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '25866208',\n",
       "  'uniprot_ids': 'P41161',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0816.1',\n",
       "  'Matrix_Name': 'Ascl2',\n",
       "  1: {'A': 238.0, 'C': 37.0, 'G': 64.0, 'T': 17.0},\n",
       "  2: {'A': 129.0, 'C': 42.0, 'G': 173.0, 'T': 2.0},\n",
       "  3: {'A': 4.0, 'C': 301.0, 'G': 1.0, 'T': 2.0},\n",
       "  4: {'A': 301.0, 'C': 1.0, 'G': 1.0, 'T': 16.0},\n",
       "  5: {'A': 8.0, 'C': 24.0, 'G': 301.0, 'T': 19.0},\n",
       "  6: {'A': 4.0, 'C': 301.0, 'G': 3.0, 'T': 2.0},\n",
       "  7: {'A': 5.0, 'C': 1.0, 'G': 1.0, 'T': 301.0},\n",
       "  8: {'A': 1.0, 'C': 2.0, 'G': 301.0, 'T': 3.0},\n",
       "  9: {'A': 7.0, 'C': 207.0, 'G': 16.0, 'T': 95.0},\n",
       "  10: {'A': 35.0, 'C': 102.0, 'G': 29.0, 'T': 200.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'O35885',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0783.1',\n",
       "  'Matrix_Name': 'PKNOX2',\n",
       "  1: {'A': 38.0, 'C': 1.0, 'G': 10.0, 'T': 1193.0},\n",
       "  2: {'A': 2.0, 'C': 7.0, 'G': 1390.0, 'T': 1.0},\n",
       "  3: {'A': 1125.0, 'C': 4.0, 'G': 14.0, 'T': 7.0},\n",
       "  4: {'A': 1.0, 'C': 1251.0, 'G': 1.0, 'T': 10.0},\n",
       "  5: {'A': 1170.0, 'C': 1.0, 'G': 2.0, 'T': 5.0},\n",
       "  6: {'A': 8.0, 'C': 1.0, 'G': 1299.0, 'T': 119.0},\n",
       "  7: {'A': 40.0, 'C': 305.0, 'G': 703.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 5.0, 'G': 1.0, 'T': 1205.0},\n",
       "  9: {'A': 4.0, 'C': 2.0, 'G': 1371.0, 'T': 1.0},\n",
       "  10: {'A': 11.0, 'C': 30.0, 'G': 6.0, 'T': 1168.0},\n",
       "  11: {'A': 12.0, 'C': 1266.0, 'G': 7.0, 'T': 4.0},\n",
       "  12: {'A': 1191.0, 'C': 2.0, 'G': 22.0, 'T': 10.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '25233378',\n",
       "  'uniprot_ids': 'Q96KN3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0817.1',\n",
       "  'Matrix_Name': 'BHLHE23',\n",
       "  1: {'A': 2551.0, 'C': 687.0, 'G': 859.0, 'T': 913.0},\n",
       "  2: {'A': 5006.0, 'C': 56.0, 'G': 891.0, 'T': 45.0},\n",
       "  3: {'A': 2780.0, 'C': 1287.0, 'G': 933.0, 'T': 9.0},\n",
       "  4: {'A': 22.0, 'C': 5006.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 5006.0, 'C': 1.0, 'G': 68.0, 'T': 18.0},\n",
       "  6: {'A': 1.0, 'C': 5.0, 'G': 2.0, 'T': 5006.0},\n",
       "  7: {'A': 5006.0, 'C': 6.0, 'G': 4.0, 'T': 1.0},\n",
       "  8: {'A': 48.0, 'C': 116.0, 'G': 6.0, 'T': 5006.0},\n",
       "  9: {'A': 1.0, 'C': 5.0, 'G': 5006.0, 'T': 8.0},\n",
       "  10: {'A': 14.0, 'C': 1204.0, 'G': 1772.0, 'T': 2019.0},\n",
       "  11: {'A': 244.0, 'C': 1469.0, 'G': 93.0, 'T': 5006.0},\n",
       "  12: {'A': 1524.0, 'C': 912.0, 'G': 965.0, 'T': 1607.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9144210',\n",
       "  'uniprot_ids': 'Q8NDY6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0851.1',\n",
       "  'Matrix_Name': 'Foxj3',\n",
       "  1: {'A': 28.0, 'C': 27.0, 'G': 22.0, 'T': 27.0},\n",
       "  2: {'A': 35.0, 'C': 14.0, 'G': 32.0, 'T': 23.0},\n",
       "  3: {'A': 49.0, 'C': 20.0, 'G': 17.0, 'T': 18.0},\n",
       "  4: {'A': 52.0, 'C': 14.0, 'G': 18.0, 'T': 20.0},\n",
       "  5: {'A': 36.0, 'C': 14.0, 'G': 34.0, 'T': 21.0},\n",
       "  6: {'A': 31.0, 'C': 2.0, 'G': 69.0, 'T': 1.0},\n",
       "  7: {'A': 2.0, 'C': 3.0, 'G': 1.0, 'T': 98.0},\n",
       "  8: {'A': 92.0, 'C': 9.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 97.0, 'C': 3.0, 'G': 1.0, 'T': 4.0},\n",
       "  10: {'A': 100.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  11: {'A': 1.0, 'C': 82.0, 'G': 1.0, 'T': 19.0},\n",
       "  12: {'A': 100.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  13: {'A': 80.0, 'C': 8.0, 'G': 7.0, 'T': 10.0},\n",
       "  14: {'A': 58.0, 'C': 10.0, 'G': 8.0, 'T': 28.0},\n",
       "  15: {'A': 23.0, 'C': 35.0, 'G': 27.0, 'T': 19.0},\n",
       "  16: {'A': 28.0, 'C': 28.0, 'G': 25.0, 'T': 23.0},\n",
       "  17: {'A': 25.0, 'C': 40.0, 'G': 18.0, 'T': 20.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q8BUR3',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA1654.1',\n",
       "  'Matrix_Name': 'ZNF16',\n",
       "  1: {'A': 665.0, 'C': 92.0, 'G': 59.0, 'T': 21.0},\n",
       "  2: {'A': 363.0, 'C': 223.0, 'G': 228.0, 'T': 23.0},\n",
       "  3: {'A': 49.0, 'C': 80.0, 'G': 129.0, 'T': 579.0},\n",
       "  4: {'A': 354.0, 'C': 15.0, 'G': 313.0, 'T': 155.0},\n",
       "  5: {'A': 18.0, 'C': 3.0, 'G': 810.0, 'T': 6.0},\n",
       "  6: {'A': 13.0, 'C': 5.0, 'G': 805.0, 'T': 14.0},\n",
       "  7: {'A': 2.0, 'C': 11.0, 'G': 822.0, 'T': 2.0},\n",
       "  8: {'A': 833.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  9: {'A': 14.0, 'C': 2.0, 'G': 817.0, 'T': 4.0},\n",
       "  10: {'A': 8.0, 'C': 795.0, 'G': 5.0, 'T': 29.0},\n",
       "  11: {'A': 11.0, 'C': 749.0, 'G': 1.0, 'T': 76.0},\n",
       "  12: {'A': 801.0, 'C': 18.0, 'G': 13.0, 'T': 5.0},\n",
       "  13: {'A': 32.0, 'C': 175.0, 'G': 32.0, 'T': 598.0},\n",
       "  14: {'A': 166.0, 'C': 27.0, 'G': 437.0, 'T': 207.0},\n",
       "  15: {'A': 25.0, 'C': 17.0, 'G': 783.0, 'T': 12.0},\n",
       "  16: {'A': 612.0, 'C': 93.0, 'G': 96.0, 'T': 36.0},\n",
       "  17: {'A': 525.0, 'C': 18.0, 'G': 216.0, 'T': 78.0},\n",
       "  18: {'A': 67.0, 'C': 5.0, 'G': 712.0, 'T': 53.0},\n",
       "  19: {'A': 90.0, 'C': 8.0, 'G': 726.0, 'T': 13.0},\n",
       "  20: {'A': 60.0, 'C': 161.0, 'G': 76.0, 'T': 540.0},\n",
       "  21: {'A': 33.0, 'C': 57.0, 'G': 139.0, 'T': 608.0},\n",
       "  22: {'A': 63.0, 'C': 271.0, 'G': 137.0, 'T': 366.0},\n",
       "  23: {'A': 68.0, 'C': 110.0, 'G': 96.0, 'T': 563.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 23,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '27852650',\n",
       "  'uniprot_ids': 'P17020',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0046.2',\n",
       "  'Matrix_Name': 'HNF1A',\n",
       "  1: {'A': 5210.0, 'C': 2442.0, 'G': 3491.0, 'T': 3010.0},\n",
       "  2: {'A': 6982.0, 'C': 416.0, 'G': 7169.0, 'T': 637.0},\n",
       "  3: {'A': 377.0, 'C': 1061.0, 'G': 904.0, 'T': 14150.0},\n",
       "  4: {'A': 2915.0, 'C': 2171.0, 'G': 545.0, 'T': 14150.0},\n",
       "  5: {'A': 14150.0, 'C': 4.0, 'G': 247.0, 'T': 15.0},\n",
       "  6: {'A': 14150.0, 'C': 634.0, 'G': 9.0, 'T': 365.0},\n",
       "  7: {'A': 1275.0, 'C': 424.0, 'G': 8.0, 'T': 14150.0},\n",
       "  8: {'A': 3341.0, 'C': 3751.0, 'G': 4541.0, 'T': 2519.0},\n",
       "  9: {'A': 14150.0, 'C': 10.0, 'G': 340.0, 'T': 652.0},\n",
       "  10: {'A': 712.0, 'C': 14.0, 'G': 846.0, 'T': 14150.0},\n",
       "  11: {'A': 43.0, 'C': 229.0, 'G': 5.0, 'T': 14150.0},\n",
       "  12: {'A': 14150.0, 'C': 238.0, 'G': 1338.0, 'T': 1639.0},\n",
       "  13: {'A': 14150.0, 'C': 696.0, 'G': 1096.0, 'T': 175.0},\n",
       "  14: {'A': 1489.0, 'C': 14150.0, 'G': 656.0, 'T': 915.0},\n",
       "  15: {'A': 3973.0, 'C': 3428.0, 'G': 2276.0, 'T': 4476.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9047360',\n",
       "  'uniprot_ids': 'P20823',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0813.1',\n",
       "  'Matrix_Name': 'TFAP2B(var.3)',\n",
       "  1: {'A': 307.0, 'C': 154.0, 'G': 96.0, 'T': 752.0},\n",
       "  2: {'A': 1.0, 'C': 330.0, 'G': 1064.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 925.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 872.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 6.0, 'C': 775.0, 'G': 23.0, 'T': 137.0},\n",
       "  6: {'A': 33.0, 'C': 470.0, 'G': 133.0, 'T': 572.0},\n",
       "  7: {'A': 148.0, 'C': 346.0, 'G': 393.0, 'T': 385.0},\n",
       "  8: {'A': 584.0, 'C': 201.0, 'G': 368.0, 'T': 42.0},\n",
       "  9: {'A': 293.0, 'C': 54.0, 'G': 838.0, 'T': 12.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 1237.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 1347.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 898.0, 'G': 164.0, 'T': 1.0},\n",
       "  13: {'A': 651.0, 'C': 107.0, 'G': 180.0, 'T': 227.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16420676',\n",
       "  'uniprot_ids': 'Q92481',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0098.3',\n",
       "  'Matrix_Name': 'ETS1',\n",
       "  1: {'A': 2684.0, 'C': 211.0, 'G': 641.0, 'T': 242.0},\n",
       "  2: {'A': 181.0, 'C': 2684.0, 'G': 298.0, 'T': 23.0},\n",
       "  3: {'A': 426.0, 'C': 2684.0, 'G': 8.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 22.0, 'G': 2684.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 2684.0, 'T': 13.0},\n",
       "  6: {'A': 2684.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 2684.0, 'C': 10.0, 'G': 32.0, 'T': 910.0},\n",
       "  8: {'A': 1103.0, 'C': 22.0, 'G': 1581.0, 'T': 13.0},\n",
       "  9: {'A': 90.0, 'C': 713.0, 'G': 125.0, 'T': 1971.0},\n",
       "  10: {'A': 804.0, 'C': 402.0, 'G': 1084.0, 'T': 397.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'P14921',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0596.1',\n",
       "  'Matrix_Name': 'SREBF2',\n",
       "  1: {'A': 25.0, 'C': 10.0, 'G': 15.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 48.0},\n",
       "  3: {'A': 1.0, 'C': 4.0, 'G': 45.0, 'T': 1.0},\n",
       "  4: {'A': 11.0, 'C': 1.0, 'G': 38.0, 'T': 1.0},\n",
       "  5: {'A': 3.0, 'C': 11.0, 'G': 28.0, 'T': 9.0},\n",
       "  6: {'A': 3.0, 'C': 10.0, 'G': 37.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 48.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 48.0, 'T': 1.0},\n",
       "  9: {'A': 48.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 3.0, 'C': 17.0, 'G': 3.0, 'T': 28.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8156598',\n",
       "  'uniprot_ids': 'Q12772',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0051.1',\n",
       "  'Matrix_Name': 'IRF2',\n",
       "  1: {'A': 1.0, 'C': 5.0, 'G': 8.0, 'T': 2.0},\n",
       "  2: {'A': 3.0, 'C': 1.0, 'G': 11.0, 'T': 1.0},\n",
       "  3: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 12.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 3.0, 'C': 1.0, 'G': 11.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 7.0, 'G': 1.0, 'T': 7.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 13.0, 'T': 1.0},\n",
       "  9: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 7.0, 'G': 7.0, 'T': 1.0},\n",
       "  13: {'A': 1.0, 'C': 8.0, 'G': 3.0, 'T': 4.0},\n",
       "  14: {'A': 6.0, 'C': 3.0, 'G': 4.0, 'T': 3.0},\n",
       "  15: {'A': 7.0, 'C': 1.0, 'G': 3.0, 'T': 5.0},\n",
       "  16: {'A': 7.0, 'C': 2.0, 'G': 2.0, 'T': 5.0},\n",
       "  17: {'A': 6.0, 'C': 3.0, 'G': 2.0, 'T': 5.0},\n",
       "  18: {'A': 4.0, 'C': 7.0, 'G': 2.0, 'T': 3.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '7687740',\n",
       "  'uniprot_ids': 'P14316',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0618.1',\n",
       "  'Matrix_Name': 'LBX1',\n",
       "  1: {'A': 218.0, 'C': 132.0, 'G': 175.0, 'T': 479.0},\n",
       "  2: {'A': 1.0, 'C': 50.0, 'G': 1.0, 'T': 952.0},\n",
       "  3: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 263.0, 'G': 1.0, 'T': 739.0},\n",
       "  6: {'A': 144.0, 'C': 1.0, 'G': 120.0, 'T': 739.0},\n",
       "  7: {'A': 732.0, 'C': 1.0, 'G': 269.0, 'T': 1.0},\n",
       "  8: {'A': 30.0, 'C': 115.0, 'G': 800.0, 'T': 58.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P52954',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0113.3',\n",
       "  'Matrix_Name': 'NR3C1',\n",
       "  1: {'A': 6521.0, 'C': 2275.0, 'G': 9170.0, 'T': 4862.0},\n",
       "  2: {'A': 8886.0, 'C': 108.0, 'G': 12769.0, 'T': 226.0},\n",
       "  3: {'A': 41.0, 'C': 8.0, 'G': 27338.0, 'T': 39.0},\n",
       "  4: {'A': 8933.0, 'C': 600.0, 'G': 2855.0, 'T': 10449.0},\n",
       "  5: {'A': 18208.0, 'C': 9.0, 'G': 10.0, 'T': 11.0},\n",
       "  6: {'A': 1.0, 'C': 24482.0, 'G': 15.0, 'T': 1.0},\n",
       "  7: {'A': 18422.0, 'C': 19.0, 'G': 699.0, 'T': 72.0},\n",
       "  8: {'A': 2471.0, 'C': 9570.0, 'G': 1584.0, 'T': 10712.0},\n",
       "  9: {'A': 7737.0, 'C': 3657.0, 'G': 5590.0, 'T': 5973.0},\n",
       "  10: {'A': 11414.0, 'C': 965.0, 'G': 7862.0, 'T': 1823.0},\n",
       "  11: {'A': 82.0, 'C': 566.0, 'G': 51.0, 'T': 23752.0},\n",
       "  12: {'A': 1.0, 'C': 12.0, 'G': 27161.0, 'T': 6.0},\n",
       "  13: {'A': 9.0, 'C': 10.0, 'G': 51.0, 'T': 24435.0},\n",
       "  14: {'A': 8833.0, 'C': 3150.0, 'G': 649.0, 'T': 10376.0},\n",
       "  15: {'A': 34.0, 'C': 24381.0, 'G': 6.0, 'T': 59.0},\n",
       "  16: {'A': 235.0, 'C': 14666.0, 'G': 189.0, 'T': 9822.0},\n",
       "  17: {'A': 6220.0, 'C': 8913.0, 'G': 3102.0, 'T': 5076.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15563547',\n",
       "  'uniprot_ids': 'P04150',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0775.1',\n",
       "  'Matrix_Name': 'MEIS3',\n",
       "  1: {'A': 1867.0, 'C': 61.0, 'G': 1789.0, 'T': 3452.0},\n",
       "  2: {'A': 466.0, 'C': 384.0, 'G': 261.0, 'T': 7166.0},\n",
       "  3: {'A': 1.0, 'C': 7.0, 'G': 7166.0, 'T': 1.0},\n",
       "  4: {'A': 7166.0, 'C': 21.0, 'G': 456.0, 'T': 54.0},\n",
       "  5: {'A': 54.0, 'C': 7166.0, 'G': 9.0, 'T': 161.0},\n",
       "  6: {'A': 7166.0, 'C': 9.0, 'G': 469.0, 'T': 146.0},\n",
       "  7: {'A': 1691.0, 'C': 1890.0, 'G': 7166.0, 'T': 2095.0},\n",
       "  8: {'A': 961.0, 'C': 2343.0, 'G': 2889.0, 'T': 976.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9405651',\n",
       "  'uniprot_ids': 'Q99687',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0853.1',\n",
       "  'Matrix_Name': 'Alx4',\n",
       "  1: {'A': 17.0, 'C': 60.0, 'G': 16.0, 'T': 12.0},\n",
       "  2: {'A': 11.0, 'C': 25.0, 'G': 57.0, 'T': 11.0},\n",
       "  3: {'A': 20.0, 'C': 45.0, 'G': 17.0, 'T': 22.0},\n",
       "  4: {'A': 41.0, 'C': 16.0, 'G': 30.0, 'T': 17.0},\n",
       "  5: {'A': 3.0, 'C': 35.0, 'G': 2.0, 'T': 63.0},\n",
       "  6: {'A': 2.0, 'C': 9.0, 'G': 1.0, 'T': 93.0},\n",
       "  7: {'A': 100.0, 'C': 2.0, 'G': 2.0, 'T': 1.0},\n",
       "  8: {'A': 99.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 99.0},\n",
       "  10: {'A': 1.0, 'C': 2.0, 'G': 2.0, 'T': 100.0},\n",
       "  11: {'A': 93.0, 'C': 1.0, 'G': 9.0, 'T': 2.0},\n",
       "  12: {'A': 63.0, 'C': 2.0, 'G': 35.0, 'T': 3.0},\n",
       "  13: {'A': 18.0, 'C': 24.0, 'G': 19.0, 'T': 43.0},\n",
       "  14: {'A': 30.0, 'C': 29.0, 'G': 14.0, 'T': 31.0},\n",
       "  15: {'A': 37.0, 'C': 24.0, 'G': 20.0, 'T': 23.0},\n",
       "  16: {'A': 24.0, 'C': 30.0, 'G': 26.0, 'T': 24.0},\n",
       "  17: {'A': 13.0, 'C': 44.0, 'G': 22.0, 'T': 25.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '10542273',\n",
       "  'uniprot_ids': 'O35137',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0756.1',\n",
       "  'Matrix_Name': 'ONECUT2',\n",
       "  1: {'A': 1029.0, 'C': 446.0, 'G': 540.0, 'T': 367.0},\n",
       "  2: {'A': 1364.0, 'C': 193.0, 'G': 516.0, 'T': 308.0},\n",
       "  3: {'A': 2379.0, 'C': 255.0, 'G': 376.0, 'T': 508.0},\n",
       "  4: {'A': 2379.0, 'C': 158.0, 'G': 84.0, 'T': 596.0},\n",
       "  5: {'A': 2379.0, 'C': 13.0, 'G': 115.0, 'T': 21.0},\n",
       "  6: {'A': 2379.0, 'C': 2.0, 'G': 19.0, 'T': 7.0},\n",
       "  7: {'A': 9.0, 'C': 9.0, 'G': 10.0, 'T': 2379.0},\n",
       "  8: {'A': 7.0, 'C': 2379.0, 'G': 1.0, 'T': 6.0},\n",
       "  9: {'A': 894.0, 'C': 17.0, 'G': 1486.0, 'T': 1.0},\n",
       "  10: {'A': 2379.0, 'C': 3.0, 'G': 2.0, 'T': 23.0},\n",
       "  11: {'A': 21.0, 'C': 15.0, 'G': 1.0, 'T': 2379.0},\n",
       "  12: {'A': 2379.0, 'C': 85.0, 'G': 106.0, 'T': 206.0},\n",
       "  13: {'A': 1548.0, 'C': 577.0, 'G': 239.0, 'T': 833.0},\n",
       "  14: {'A': 652.0, 'C': 527.0, 'G': 238.0, 'T': 965.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'HD-CUT',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8887657',\n",
       "  'uniprot_ids': 'O95948',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1421.1',\n",
       "  'Matrix_Name': 'TCF7L1',\n",
       "  1: {'A': 110.0, 'C': 21.0, 'G': 7.0, 'T': 25.0},\n",
       "  2: {'A': 160.0, 'C': 8.0, 'G': 31.0, 'T': 8.0},\n",
       "  3: {'A': 160.0, 'C': 1.0, 'G': 1.0, 'T': 6.0},\n",
       "  4: {'A': 1.0, 'C': 34.0, 'G': 127.0, 'T': 7.0},\n",
       "  5: {'A': 160.0, 'C': 10.0, 'G': 14.0, 'T': 19.0},\n",
       "  6: {'A': 1.0, 'C': 7.0, 'G': 10.0, 'T': 160.0},\n",
       "  7: {'A': 6.0, 'C': 160.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 160.0, 'C': 3.0, 'G': 2.0, 'T': 1.0},\n",
       "  9: {'A': 160.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 160.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 10.0, 'G': 160.0, 'T': 6.0},\n",
       "  12: {'A': 29.0, 'C': 10.0, 'G': 114.0, 'T': 10.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TCF-7-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '27447672',\n",
       "  'uniprot_ids': 'Q9HCS4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1123.2',\n",
       "  'Matrix_Name': 'TWIST1',\n",
       "  1: {'A': 8330.0, 'C': 5762.0, 'G': 6486.0, 'T': 6759.0},\n",
       "  2: {'A': 7823.0, 'C': 5531.0, 'G': 5723.0, 'T': 8260.0},\n",
       "  3: {'A': 7537.0, 'C': 4598.0, 'G': 5773.0, 'T': 9429.0},\n",
       "  4: {'A': 3734.0, 'C': 20413.0, 'G': 2337.0, 'T': 853.0},\n",
       "  5: {'A': 378.0, 'C': 26666.0, 'G': 118.0, 'T': 175.0},\n",
       "  6: {'A': 26714.0, 'C': 137.0, 'G': 211.0, 'T': 275.0},\n",
       "  7: {'A': 441.0, 'C': 1321.0, 'G': 24167.0, 'T': 1408.0},\n",
       "  8: {'A': 24525.0, 'C': 1688.0, 'G': 909.0, 'T': 215.0},\n",
       "  9: {'A': 358.0, 'C': 457.0, 'G': 289.0, 'T': 26233.0},\n",
       "  10: {'A': 381.0, 'C': 397.0, 'G': 25976.0, 'T': 583.0},\n",
       "  11: {'A': 1802.0, 'C': 2920.0, 'G': 4070.0, 'T': 18545.0},\n",
       "  12: {'A': 4071.0, 'C': 7403.0, 'G': 7274.0, 'T': 8589.0},\n",
       "  13: {'A': 6824.0, 'C': 6562.0, 'G': 6214.0, 'T': 7737.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '29453333',\n",
       "  'uniprot_ids': 'Q15672',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0074.1',\n",
       "  'Matrix_Name': 'RXRA::VDR',\n",
       "  1: {'A': 4.0, 'C': 1.0, 'G': 8.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 11.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 10.0, 'T': 2.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 11.0},\n",
       "  5: {'A': 1.0, 'C': 10.0, 'G': 1.0, 'T': 2.0},\n",
       "  6: {'A': 10.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  7: {'A': 5.0, 'C': 3.0, 'G': 1.0, 'T': 5.0},\n",
       "  8: {'A': 3.0, 'C': 5.0, 'G': 3.0, 'T': 3.0},\n",
       "  9: {'A': 3.0, 'C': 1.0, 'G': 9.0, 'T': 1.0},\n",
       "  10: {'A': 6.0, 'C': 1.0, 'G': 6.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 11.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 11.0},\n",
       "  13: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 10.0},\n",
       "  14: {'A': 1.0, 'C': 10.0, 'G': 1.0, 'T': 2.0},\n",
       "  15: {'A': 8.0, 'C': 2.0, 'G': 3.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '8674817',\n",
       "  'uniprot_ids': 'P19793;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1129.1',\n",
       "  'Matrix_Name': 'FOSL1::JUN(var.2)',\n",
       "  1: {'A': 640.0, 'C': 70.0, 'G': 267.0, 'T': 27.0},\n",
       "  2: {'A': 29.0, 'C': 31.0, 'G': 25.0, 'T': 920.0},\n",
       "  3: {'A': 34.0, 'C': 57.0, 'G': 673.0, 'T': 241.0},\n",
       "  4: {'A': 826.0, 'C': 27.0, 'G': 107.0, 'T': 44.0},\n",
       "  5: {'A': 23.0, 'C': 846.0, 'G': 39.0, 'T': 96.0},\n",
       "  6: {'A': 94.0, 'C': 37.0, 'G': 848.0, 'T': 25.0},\n",
       "  7: {'A': 42.0, 'C': 106.0, 'G': 28.0, 'T': 828.0},\n",
       "  8: {'A': 239.0, 'C': 671.0, 'G': 59.0, 'T': 36.0},\n",
       "  9: {'A': 918.0, 'C': 23.0, 'G': 32.0, 'T': 31.0},\n",
       "  10: {'A': 25.0, 'C': 265.0, 'G': 72.0, 'T': 642.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15407;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0718.1',\n",
       "  'Matrix_Name': 'RAX',\n",
       "  1: {'A': 776.0, 'C': 371.0, 'G': 922.0, 'T': 348.0},\n",
       "  2: {'A': 340.0, 'C': 962.0, 'G': 420.0, 'T': 694.0},\n",
       "  3: {'A': 215.0, 'C': 1282.0, 'G': 26.0, 'T': 1132.0},\n",
       "  4: {'A': 2413.0, 'C': 18.0, 'G': 4.0, 'T': 41.0},\n",
       "  5: {'A': 2413.0, 'C': 39.0, 'G': 13.0, 'T': 25.0},\n",
       "  6: {'A': 124.0, 'C': 1.0, 'G': 1.0, 'T': 2413.0},\n",
       "  7: {'A': 72.0, 'C': 72.0, 'G': 50.0, 'T': 2413.0},\n",
       "  8: {'A': 2413.0, 'C': 1.0, 'G': 21.0, 'T': 127.0},\n",
       "  9: {'A': 1052.0, 'C': 285.0, 'G': 702.0, 'T': 376.0},\n",
       "  10: {'A': 540.0, 'C': 839.0, 'G': 384.0, 'T': 653.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q9Y2V3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0901.2',\n",
       "  'Matrix_Name': 'HOXB13',\n",
       "  1: {'A': 31358.0, 'C': 10860.0, 'G': 13067.0, 'T': 23433.0},\n",
       "  2: {'A': 27208.0, 'C': 13276.0, 'G': 15207.0, 'T': 23027.0},\n",
       "  3: {'A': 11897.0, 'C': 47675.0, 'G': 9097.0, 'T': 10049.0},\n",
       "  4: {'A': 4213.0, 'C': 65313.0, 'G': 583.0, 'T': 8609.0},\n",
       "  5: {'A': 66410.0, 'C': 4612.0, 'G': 352.0, 'T': 7344.0},\n",
       "  6: {'A': 73045.0, 'C': 727.0, 'G': 1914.0, 'T': 3032.0},\n",
       "  7: {'A': 422.0, 'C': 551.0, 'G': 587.0, 'T': 77158.0},\n",
       "  8: {'A': 73613.0, 'C': 524.0, 'G': 1084.0, 'T': 3497.0},\n",
       "  9: {'A': 75998.0, 'C': 600.0, 'G': 491.0, 'T': 1629.0},\n",
       "  10: {'A': 76078.0, 'C': 761.0, 'G': 626.0, 'T': 1253.0},\n",
       "  11: {'A': 70231.0, 'C': 2466.0, 'G': 1883.0, 'T': 4138.0},\n",
       "  12: {'A': 22969.0, 'C': 12416.0, 'G': 7377.0, 'T': 35956.0},\n",
       "  13: {'A': 26157.0, 'C': 13153.0, 'G': 11760.0, 'T': 27648.0},\n",
       "  14: {'A': 25172.0, 'C': 13392.0, 'G': 12166.0, 'T': 27988.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q92826',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0888.1',\n",
       "  'Matrix_Name': 'EVX2',\n",
       "  1: {'A': 4412.0, 'C': 5114.0, 'G': 7815.0, 'T': 3083.0},\n",
       "  2: {'A': 5144.0, 'C': 6307.0, 'G': 7042.0, 'T': 1930.0},\n",
       "  3: {'A': 2542.0, 'C': 5670.0, 'G': 970.0, 'T': 20420.0},\n",
       "  4: {'A': 10109.0, 'C': 5019.0, 'G': 4542.0, 'T': 753.0},\n",
       "  5: {'A': 20420.0, 'C': 157.0, 'G': 882.0, 'T': 1.0},\n",
       "  6: {'A': 6.0, 'C': 167.0, 'G': 703.0, 'T': 20420.0},\n",
       "  7: {'A': 304.0, 'C': 864.0, 'G': 446.0, 'T': 20420.0},\n",
       "  8: {'A': 20420.0, 'C': 1.0, 'G': 1560.0, 'T': 447.0},\n",
       "  9: {'A': 4091.0, 'C': 4859.0, 'G': 7790.0, 'T': 3683.0},\n",
       "  10: {'A': 3779.0, 'C': 7716.0, 'G': 4998.0, 'T': 3931.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q03828',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0480.1',\n",
       "  'Matrix_Name': 'Foxo1',\n",
       "  1: {'A': 469.0, 'C': 278.0, 'G': 479.0, 'T': 1268.0},\n",
       "  2: {'A': 128.0, 'C': 1280.0, 'G': 753.0, 'T': 333.0},\n",
       "  3: {'A': 88.0, 'C': 1407.0, 'G': 210.0, 'T': 789.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2491.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 2491.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2491.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2491.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 288.0, 'T': 2204.0},\n",
       "  9: {'A': 1872.0, 'C': 158.0, 'G': 1.0, 'T': 463.0},\n",
       "  10: {'A': 1.0, 'C': 2008.0, 'G': 1.0, 'T': 484.0},\n",
       "  11: {'A': 1073.0, 'C': 574.0, 'G': 72.0, 'T': 775.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '10880363',\n",
       "  'uniprot_ids': 'Q9R1E0',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0755.1',\n",
       "  'Matrix_Name': 'CUX2',\n",
       "  1: {'A': 8112.0, 'C': 4574.0, 'G': 4248.0, 'T': 45570.0},\n",
       "  2: {'A': 45570.0, 'C': 5068.0, 'G': 23535.0, 'T': 12192.0},\n",
       "  3: {'A': 45570.0, 'C': 251.0, 'G': 541.0, 'T': 346.0},\n",
       "  4: {'A': 480.0, 'C': 1842.0, 'G': 486.0, 'T': 45570.0},\n",
       "  5: {'A': 75.0, 'C': 45570.0, 'G': 95.0, 'T': 737.0},\n",
       "  6: {'A': 13768.0, 'C': 209.0, 'G': 31804.0, 'T': 396.0},\n",
       "  7: {'A': 45570.0, 'C': 70.0, 'G': 164.0, 'T': 192.0},\n",
       "  8: {'A': 397.0, 'C': 279.0, 'G': 67.0, 'T': 45570.0},\n",
       "  9: {'A': 45570.0, 'C': 11731.0, 'G': 7588.0, 'T': 12098.0},\n",
       "  10: {'A': 19810.0, 'C': 9535.0, 'G': 5057.0, 'T': 11172.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HD-CUT',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7799919',\n",
       "  'uniprot_ids': 'O14529',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0482.2',\n",
       "  'Matrix_Name': 'GATA4',\n",
       "  1: {'A': 14633.0, 'C': 11762.0, 'G': 11084.0, 'T': 21539.0},\n",
       "  2: {'A': 13535.0, 'C': 12704.0, 'G': 11208.0, 'T': 21571.0},\n",
       "  3: {'A': 3978.0, 'C': 38290.0, 'G': 5100.0, 'T': 11650.0},\n",
       "  4: {'A': 3151.0, 'C': 50063.0, 'G': 1104.0, 'T': 4700.0},\n",
       "  5: {'A': 1587.0, 'C': 1430.0, 'G': 750.0, 'T': 55251.0},\n",
       "  6: {'A': 1651.0, 'C': 1378.0, 'G': 809.0, 'T': 55180.0},\n",
       "  7: {'A': 54421.0, 'C': 1704.0, 'G': 1246.0, 'T': 1647.0},\n",
       "  8: {'A': 771.0, 'C': 583.0, 'G': 540.0, 'T': 57124.0},\n",
       "  9: {'A': 481.0, 'C': 56545.0, 'G': 569.0, 'T': 1423.0},\n",
       "  10: {'A': 7660.0, 'C': 1793.0, 'G': 1377.0, 'T': 48188.0},\n",
       "  11: {'A': 11042.0, 'C': 16060.0, 'G': 15226.0, 'T': 16690.0},\n",
       "  12: {'A': 15350.0, 'C': 14581.0, 'G': 8956.0, 'T': 20131.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '7791790',\n",
       "  'uniprot_ids': 'P43694',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0859.1',\n",
       "  'Matrix_Name': 'Rarg',\n",
       "  1: {'A': 12549.0, 'C': 1656.0, 'G': 11290.0, 'T': 1310.0},\n",
       "  2: {'A': 7221.0, 'C': 148.0, 'G': 3075.0, 'T': 6.0},\n",
       "  3: {'A': 4.0, 'C': 1.0, 'G': 12766.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 4.0, 'G': 11970.0, 'T': 1827.0},\n",
       "  5: {'A': 21.0, 'C': 36.0, 'G': 65.0, 'T': 10031.0},\n",
       "  6: {'A': 4.0, 'C': 12654.0, 'G': 39.0, 'T': 80.0},\n",
       "  7: {'A': 8845.0, 'C': 3.0, 'G': 16.0, 'T': 1.0},\n",
       "  8: {'A': 9170.0, 'C': 50.0, 'G': 115.0, 'T': 258.0},\n",
       "  9: {'A': 8865.0, 'C': 14.0, 'G': 619.0, 'T': 354.0},\n",
       "  10: {'A': 9212.0, 'C': 5.0, 'G': 395.0, 'T': 1.0},\n",
       "  11: {'A': 2.0, 'C': 2.0, 'G': 13268.0, 'T': 4.0},\n",
       "  12: {'A': 1.0, 'C': 3.0, 'G': 9299.0, 'T': 2629.0},\n",
       "  13: {'A': 7.0, 'C': 41.0, 'G': 52.0, 'T': 9775.0},\n",
       "  14: {'A': 1.0, 'C': 11731.0, 'G': 68.0, 'T': 106.0},\n",
       "  15: {'A': 9148.0, 'C': 5.0, 'G': 83.0, 'T': 2.0},\n",
       "  16: {'A': 11045.0, 'C': 4837.0, 'G': 2471.0, 'T': 5567.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P18911',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0845.1',\n",
       "  'Matrix_Name': 'FOXB1',\n",
       "  1: {'A': 2394.0, 'C': 273.0, 'G': 191.0, 'T': 4449.0},\n",
       "  2: {'A': 5344.0, 'C': 264.0, 'G': 1499.0, 'T': 562.0},\n",
       "  3: {'A': 1984.0, 'C': 219.0, 'G': 410.0, 'T': 4859.0},\n",
       "  4: {'A': 1160.0, 'C': 55.0, 'G': 6842.0, 'T': 125.0},\n",
       "  5: {'A': 181.0, 'C': 241.0, 'G': 114.0, 'T': 6842.0},\n",
       "  6: {'A': 4892.0, 'C': 1951.0, 'G': 76.0, 'T': 74.0},\n",
       "  7: {'A': 6842.0, 'C': 214.0, 'G': 1.0, 'T': 382.0},\n",
       "  8: {'A': 6842.0, 'C': 48.0, 'G': 48.0, 'T': 217.0},\n",
       "  9: {'A': 29.0, 'C': 2253.0, 'G': 105.0, 'T': 4590.0},\n",
       "  10: {'A': 6842.0, 'C': 1.0, 'G': 158.0, 'T': 148.0},\n",
       "  11: {'A': 1705.0, 'C': 574.0, 'G': 557.0, 'T': 4009.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '7739038',\n",
       "  'uniprot_ids': 'Q99853',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0886.1',\n",
       "  'Matrix_Name': 'EMX2',\n",
       "  1: {'A': 10480.0, 'C': 8413.0, 'G': 12803.0, 'T': 5570.0},\n",
       "  2: {'A': 5705.0, 'C': 13338.0, 'G': 10583.0, 'T': 7640.0},\n",
       "  3: {'A': 835.0, 'C': 7762.0, 'G': 1.0, 'T': 29502.0},\n",
       "  4: {'A': 37263.0, 'C': 4939.0, 'G': 679.0, 'T': 1237.0},\n",
       "  5: {'A': 37263.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 37263.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 4026.0, 'T': 37263.0},\n",
       "  8: {'A': 37263.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 9140.0, 'C': 7125.0, 'G': 17673.0, 'T': 3329.0},\n",
       "  10: {'A': 5971.0, 'C': 13055.0, 'G': 10196.0, 'T': 8044.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q04743',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0077.1',\n",
       "  'Matrix_Name': 'SOX9',\n",
       "  1: {'A': 4.0, 'C': 56.0, 'G': 10.0, 'T': 10.0},\n",
       "  2: {'A': 9.0, 'C': 39.0, 'G': 7.0, 'T': 25.0},\n",
       "  3: {'A': 72.0, 'C': 2.0, 'G': 1.0, 'T': 5.0},\n",
       "  4: {'A': 3.0, 'C': 2.0, 'G': 3.0, 'T': 72.0},\n",
       "  5: {'A': 8.0, 'C': 1.0, 'G': 5.0, 'T': 66.0},\n",
       "  6: {'A': 3.0, 'C': 3.0, 'G': 73.0, 'T': 1.0},\n",
       "  7: {'A': 14.0, 'C': 1.0, 'G': 5.0, 'T': 60.0},\n",
       "  8: {'A': 10.0, 'C': 8.0, 'G': 7.0, 'T': 55.0},\n",
       "  9: {'A': 15.0, 'C': 32.0, 'G': 8.0, 'T': 25.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '9973626',\n",
       "  'uniprot_ids': 'P48436',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1105.2',\n",
       "  'Matrix_Name': 'GRHL2',\n",
       "  1: {'A': 14484.0, 'C': 9742.0, 'G': 9162.0, 'T': 8494.0},\n",
       "  2: {'A': 17843.0, 'C': 6997.0, 'G': 10777.0, 'T': 6265.0},\n",
       "  3: {'A': 36468.0, 'C': 1148.0, 'G': 2647.0, 'T': 1619.0},\n",
       "  4: {'A': 37066.0, 'C': 713.0, 'G': 2842.0, 'T': 1261.0},\n",
       "  5: {'A': 504.0, 'C': 40447.0, 'G': 610.0, 'T': 321.0},\n",
       "  6: {'A': 34025.0, 'C': 1544.0, 'G': 486.0, 'T': 5827.0},\n",
       "  7: {'A': 2825.0, 'C': 582.0, 'G': 37227.0, 'T': 1248.0},\n",
       "  8: {'A': 643.0, 'C': 988.0, 'G': 39625.0, 'T': 626.0},\n",
       "  9: {'A': 1879.0, 'C': 2332.0, 'G': 749.0, 'T': 36922.0},\n",
       "  10: {'A': 1753.0, 'C': 2001.0, 'G': 1221.0, 'T': 36907.0},\n",
       "  11: {'A': 7062.0, 'C': 9342.0, 'G': 7962.0, 'T': 17516.0},\n",
       "  12: {'A': 8869.0, 'C': 8529.0, 'G': 10537.0, 'T': 13947.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Grainyhead-related',\n",
       "  'tf_class': 'Grainyhead',\n",
       "  'pubmed_ids': '20978075',\n",
       "  'uniprot_ids': 'Q6ISB3',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1641.1',\n",
       "  'Matrix_Name': 'MYF5',\n",
       "  1: {'A': 2699.0, 'C': 2701.0, 'G': 3259.0, 'T': 2555.0},\n",
       "  2: {'A': 3508.0, 'C': 2987.0, 'G': 3146.0, 'T': 1573.0},\n",
       "  3: {'A': 7491.0, 'C': 1369.0, 'G': 1775.0, 'T': 579.0},\n",
       "  4: {'A': 96.0, 'C': 10912.0, 'G': 162.0, 'T': 44.0},\n",
       "  5: {'A': 10966.0, 'C': 48.0, 'G': 70.0, 'T': 130.0},\n",
       "  6: {'A': 101.0, 'C': 231.0, 'G': 10659.0, 'T': 223.0},\n",
       "  7: {'A': 224.0, 'C': 10658.0, 'G': 231.0, 'T': 101.0},\n",
       "  8: {'A': 130.0, 'C': 69.0, 'G': 48.0, 'T': 10967.0},\n",
       "  9: {'A': 43.0, 'C': 162.0, 'G': 10913.0, 'T': 96.0},\n",
       "  10: {'A': 579.0, 'C': 1776.0, 'G': 1369.0, 'T': 7490.0},\n",
       "  11: {'A': 1573.0, 'C': 3147.0, 'G': 2986.0, 'T': 3508.0},\n",
       "  12: {'A': 2555.0, 'C': 3259.0, 'G': 2702.0, 'T': 2698.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '26906734',\n",
       "  'uniprot_ids': 'P13349',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0088.2',\n",
       "  'Matrix_Name': 'ZNF143',\n",
       "  1: {'A': 88.0, 'C': 510.0, 'G': 154.0, 'T': 1286.0},\n",
       "  2: {'A': 950.0, 'C': 1.0, 'G': 14.0, 'T': 653.0},\n",
       "  3: {'A': 22.0, 'C': 1591.0, 'G': 2.0, 'T': 2.0},\n",
       "  4: {'A': 3.0, 'C': 1610.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1610.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1610.0, 'C': 1.0, 'G': 5.0, 'T': 4.0},\n",
       "  7: {'A': 1.0, 'C': 918.0, 'G': 62.0, 'T': 687.0},\n",
       "  8: {'A': 1296.0, 'C': 144.0, 'G': 311.0, 'T': 1.0},\n",
       "  9: {'A': 1573.0, 'C': 1.0, 'G': 68.0, 'T': 2.0},\n",
       "  10: {'A': 7.0, 'C': 1.0, 'G': 2.0, 'T': 1610.0},\n",
       "  11: {'A': 60.0, 'C': 37.0, 'G': 1554.0, 'T': 10.0},\n",
       "  12: {'A': 1.0, 'C': 1610.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 1504.0, 'C': 154.0, 'G': 4.0, 'T': 28.0},\n",
       "  14: {'A': 346.0, 'C': 1068.0, 'G': 16.0, 'T': 1081.0},\n",
       "  15: {'A': 103.0, 'C': 1139.0, 'G': 14.0, 'T': 1200.0},\n",
       "  16: {'A': 383.0, 'C': 22.0, 'G': 1519.0, 'T': 121.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '9009278',\n",
       "  'uniprot_ids': 'P52747',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0521.1',\n",
       "  'Matrix_Name': 'Tcf12',\n",
       "  1: {'A': 6171.0, 'C': 1586.0, 'G': 5124.0, 'T': 18.0},\n",
       "  2: {'A': 8821.0, 'C': 125.0, 'G': 3952.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 12896.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 12896.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 9.0, 'C': 1.0, 'G': 12888.0, 'T': 1.0},\n",
       "  6: {'A': 185.0, 'C': 10823.0, 'G': 1890.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 12896.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 12896.0, 'T': 1.0},\n",
       "  9: {'A': 1876.0, 'C': 5528.0, 'G': 3812.0, 'T': 1683.0},\n",
       "  10: {'A': 3913.0, 'C': 2417.0, 'G': 2926.0, 'T': 3643.0},\n",
       "  11: {'A': 2864.0, 'C': 3006.0, 'G': 4837.0, 'T': 2192.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'E2A-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '1501295',\n",
       "  'uniprot_ids': 'Q61286',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1546.1',\n",
       "  'Matrix_Name': 'PAX3(var.2)',\n",
       "  1: {'A': 204.0, 'C': 516.0, 'G': 217.0, 'T': 412.0},\n",
       "  2: {'A': 113.0, 'C': 881.0, 'G': 465.0, 'T': 168.0},\n",
       "  3: {'A': 2.0, 'C': 3.0, 'G': 1345.0, 'T': 127.0},\n",
       "  4: {'A': 182.0, 'C': 80.0, 'G': 7.0, 'T': 1345.0},\n",
       "  5: {'A': 10.0, 'C': 1345.0, 'G': 24.0, 'T': 264.0},\n",
       "  6: {'A': 1345.0, 'C': 3.0, 'G': 20.0, 'T': 3.0},\n",
       "  7: {'A': 14.0, 'C': 1345.0, 'G': 3.0, 'T': 202.0},\n",
       "  8: {'A': 162.0, 'C': 2.0, 'G': 1345.0, 'T': 1.0},\n",
       "  9: {'A': 4.0, 'C': 837.0, 'G': 509.0, 'T': 7.0},\n",
       "  10: {'A': 198.0, 'C': 415.0, 'G': 128.0, 'T': 607.0},\n",
       "  11: {'A': 513.0, 'C': 229.0, 'G': 139.0, 'T': 468.0},\n",
       "  12: {'A': 1345.0, 'C': 153.0, 'G': 354.0, 'T': 87.0},\n",
       "  13: {'A': 51.0, 'C': 44.0, 'G': 25.0, 'T': 1345.0},\n",
       "  14: {'A': 84.0, 'C': 139.0, 'G': 53.0, 'T': 1345.0},\n",
       "  15: {'A': 1345.0, 'C': 67.0, 'G': 109.0, 'T': 216.0},\n",
       "  16: {'A': 391.0, 'C': 327.0, 'G': 364.0, 'T': 267.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '9632796',\n",
       "  'uniprot_ids': 'P23760',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0494.1',\n",
       "  'Matrix_Name': 'Nr1h3::Rxra',\n",
       "  1: {'A': 1.0, 'C': 51.0, 'G': 29.0, 'T': 1192.0},\n",
       "  2: {'A': 141.0, 'C': 31.0, 'G': 1090.0, 'T': 11.0},\n",
       "  3: {'A': 739.0, 'C': 244.0, 'G': 138.0, 'T': 152.0},\n",
       "  4: {'A': 247.0, 'C': 972.0, 'G': 1.0, 'T': 53.0},\n",
       "  5: {'A': 59.0, 'C': 1081.0, 'G': 4.0, 'T': 129.0},\n",
       "  6: {'A': 79.0, 'C': 314.0, 'G': 129.0, 'T': 751.0},\n",
       "  7: {'A': 359.0, 'C': 319.0, 'G': 344.0, 'T': 251.0},\n",
       "  8: {'A': 375.0, 'C': 258.0, 'G': 337.0, 'T': 303.0},\n",
       "  9: {'A': 795.0, 'C': 1.0, 'G': 322.0, 'T': 155.0},\n",
       "  10: {'A': 279.0, 'C': 20.0, 'G': 904.0, 'T': 70.0},\n",
       "  11: {'A': 78.0, 'C': 1.0, 'G': 5.0, 'T': 1189.0},\n",
       "  12: {'A': 560.0, 'C': 78.0, 'G': 441.0, 'T': 194.0},\n",
       "  13: {'A': 901.0, 'C': 178.0, 'G': 193.0, 'T': 1.0},\n",
       "  14: {'A': 256.0, 'C': 963.0, 'G': 21.0, 'T': 33.0},\n",
       "  15: {'A': 118.0, 'C': 1041.0, 'G': 1.0, 'T': 113.0},\n",
       "  16: {'A': 12.0, 'C': 503.0, 'G': 43.0, 'T': 715.0},\n",
       "  17: {'A': 136.0, 'C': 509.0, 'G': 197.0, 'T': 431.0},\n",
       "  18: {'A': 282.0, 'C': 197.0, 'G': 334.0, 'T': 460.0},\n",
       "  19: {'A': 238.0, 'C': 291.0, 'G': 440.0, 'T': 304.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 19,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '21349840',\n",
       "  'uniprot_ids': 'Q9Z0Y9;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0809.2',\n",
       "  'Matrix_Name': 'TEAD4',\n",
       "  1: {'A': 18933.0, 'C': 22344.0, 'G': 15112.0, 'T': 21701.0},\n",
       "  2: {'A': 17307.0, 'C': 26578.0, 'G': 15428.0, 'T': 18777.0},\n",
       "  3: {'A': 61264.0, 'C': 2415.0, 'G': 11266.0, 'T': 3145.0},\n",
       "  4: {'A': 7544.0, 'C': 66206.0, 'G': 3296.0, 'T': 1044.0},\n",
       "  5: {'A': 74798.0, 'C': 1219.0, 'G': 570.0, 'T': 1503.0},\n",
       "  6: {'A': 966.0, 'C': 1130.0, 'G': 1026.0, 'T': 74968.0},\n",
       "  7: {'A': 2674.0, 'C': 1177.0, 'G': 2101.0, 'T': 72138.0},\n",
       "  8: {'A': 1367.0, 'C': 72556.0, 'G': 1581.0, 'T': 2586.0},\n",
       "  9: {'A': 1315.0, 'C': 75067.0, 'G': 213.0, 'T': 1495.0},\n",
       "  10: {'A': 61803.0, 'C': 3875.0, 'G': 1192.0, 'T': 11220.0},\n",
       "  11: {'A': 16618.0, 'C': 13865.0, 'G': 32221.0, 'T': 15386.0},\n",
       "  12: {'A': 19273.0, 'C': 20302.0, 'G': 22631.0, 'T': 15884.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TEF-1-related',\n",
       "  'tf_class': 'TEA',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q15561',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1571.1',\n",
       "  'Matrix_Name': 'TGIF2LX',\n",
       "  1: {'A': 238.0, 'C': 356.0, 'G': 146.0, 'T': 2540.0},\n",
       "  2: {'A': 53.0, 'C': 188.0, 'G': 2540.0, 'T': 48.0},\n",
       "  3: {'A': 2540.0, 'C': 65.0, 'G': 133.0, 'T': 429.0},\n",
       "  4: {'A': 33.0, 'C': 2540.0, 'G': 30.0, 'T': 60.0},\n",
       "  5: {'A': 2540.0, 'C': 19.0, 'G': 317.0, 'T': 41.0},\n",
       "  6: {'A': 38.0, 'C': 576.0, 'G': 1567.0, 'T': 363.0},\n",
       "  7: {'A': 381.0, 'C': 1569.0, 'G': 555.0, 'T': 39.0},\n",
       "  8: {'A': 38.0, 'C': 319.0, 'G': 24.0, 'T': 2540.0},\n",
       "  9: {'A': 65.0, 'C': 24.0, 'G': 2540.0, 'T': 23.0},\n",
       "  10: {'A': 429.0, 'C': 128.0, 'G': 73.0, 'T': 2540.0},\n",
       "  11: {'A': 46.0, 'C': 2540.0, 'G': 173.0, 'T': 67.0},\n",
       "  12: {'A': 2540.0, 'C': 133.0, 'G': 362.0, 'T': 233.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8537382',\n",
       "  'uniprot_ids': 'Q8IUE1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0913.2',\n",
       "  'Matrix_Name': 'HOXD9',\n",
       "  1: {'A': 1301.0, 'C': 873.0, 'G': 16364.0, 'T': 656.0},\n",
       "  2: {'A': 1.0, 'C': 9774.0, 'G': 43.0, 'T': 6591.0},\n",
       "  3: {'A': 16364.0, 'C': 8644.0, 'G': 676.0, 'T': 928.0},\n",
       "  4: {'A': 16364.0, 'C': 227.0, 'G': 440.0, 'T': 5.0},\n",
       "  5: {'A': 216.0, 'C': 1.0, 'G': 1.0, 'T': 16364.0},\n",
       "  6: {'A': 16364.0, 'C': 9.0, 'G': 1.0, 'T': 5446.0},\n",
       "  7: {'A': 16364.0, 'C': 1.0, 'G': 1.0, 'T': 124.0},\n",
       "  8: {'A': 16364.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 16364.0, 'C': 2203.0, 'G': 1217.0, 'T': 2195.0},\n",
       "  10: {'A': 4150.0, 'C': 5805.0, 'G': 2170.0, 'T': 4242.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '10523646',\n",
       "  'uniprot_ids': 'P28356',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0635.1',\n",
       "  'Matrix_Name': 'BARHL2',\n",
       "  1: {'A': 3319.0, 'C': 3451.0, 'G': 4375.0, 'T': 1714.0},\n",
       "  2: {'A': 3329.0, 'C': 4079.0, 'G': 1757.0, 'T': 3694.0},\n",
       "  3: {'A': 644.0, 'C': 1.0, 'G': 1.0, 'T': 12856.0},\n",
       "  4: {'A': 12856.0, 'C': 1.0, 'G': 1.0, 'T': 1970.0},\n",
       "  5: {'A': 12856.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 12856.0, 'C': 117.0, 'G': 1087.0, 'T': 4033.0},\n",
       "  7: {'A': 444.0, 'C': 12856.0, 'G': 128.0, 'T': 7757.0},\n",
       "  8: {'A': 1571.0, 'C': 1.0, 'G': 12856.0, 'T': 1336.0},\n",
       "  9: {'A': 3607.0, 'C': 2191.0, 'G': 4878.0, 'T': 2183.0},\n",
       "  10: {'A': 2172.0, 'C': 2998.0, 'G': 1277.0, 'T': 6412.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9NY43',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0748.2',\n",
       "  'Matrix_Name': 'YY2',\n",
       "  1: {'A': 1974.0, 'C': 1592.0, 'G': 1069.0, 'T': 1073.0},\n",
       "  2: {'A': 1281.0, 'C': 1608.0, 'G': 2144.0, 'T': 675.0},\n",
       "  3: {'A': 5264.0, 'C': 47.0, 'G': 360.0, 'T': 37.0},\n",
       "  4: {'A': 59.0, 'C': 274.0, 'G': 126.0, 'T': 5249.0},\n",
       "  5: {'A': 49.0, 'C': 121.0, 'G': 5510.0, 'T': 28.0},\n",
       "  6: {'A': 127.0, 'C': 102.0, 'G': 5377.0, 'T': 102.0},\n",
       "  7: {'A': 110.0, 'C': 5385.0, 'G': 26.0, 'T': 187.0},\n",
       "  8: {'A': 65.0, 'C': 331.0, 'G': 5232.0, 'T': 80.0},\n",
       "  9: {'A': 412.0, 'C': 517.0, 'G': 4475.0, 'T': 304.0},\n",
       "  10: {'A': 962.0, 'C': 2877.0, 'G': 921.0, 'T': 948.0},\n",
       "  11: {'A': 940.0, 'C': 1536.0, 'G': 2374.0, 'T': 858.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'O15391',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1145.1',\n",
       "  'Matrix_Name': 'FOSL2::JUND(var.2)',\n",
       "  1: {'A': 199.0, 'C': 178.0, 'G': 287.0, 'T': 340.0},\n",
       "  2: {'A': 220.0, 'C': 190.0, 'G': 341.0, 'T': 253.0},\n",
       "  3: {'A': 560.0, 'C': 46.0, 'G': 341.0, 'T': 58.0},\n",
       "  4: {'A': 27.0, 'C': 28.0, 'G': 11.0, 'T': 938.0},\n",
       "  5: {'A': 8.0, 'C': 47.0, 'G': 679.0, 'T': 269.0},\n",
       "  6: {'A': 944.0, 'C': 7.0, 'G': 32.0, 'T': 21.0},\n",
       "  7: {'A': 13.0, 'C': 912.0, 'G': 36.0, 'T': 42.0},\n",
       "  8: {'A': 136.0, 'C': 16.0, 'G': 824.0, 'T': 29.0},\n",
       "  9: {'A': 29.0, 'C': 13.0, 'G': 11.0, 'T': 951.0},\n",
       "  10: {'A': 34.0, 'C': 943.0, 'G': 16.0, 'T': 12.0},\n",
       "  11: {'A': 973.0, 'C': 7.0, 'G': 12.0, 'T': 12.0},\n",
       "  12: {'A': 24.0, 'C': 346.0, 'G': 56.0, 'T': 578.0},\n",
       "  13: {'A': 218.0, 'C': 435.0, 'G': 105.0, 'T': 246.0},\n",
       "  14: {'A': 163.0, 'C': 223.0, 'G': 417.0, 'T': 201.0},\n",
       "  15: {'A': 249.0, 'C': 289.0, 'G': 242.0, 'T': 224.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15408;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1563.1',\n",
       "  'Matrix_Name': 'SOX18',\n",
       "  1: {'A': 154.0, 'C': 217.0, 'G': 169.0, 'T': 53.0},\n",
       "  2: {'A': 590.0, 'C': 34.0, 'G': 141.0, 'T': 95.0},\n",
       "  3: {'A': 1.0, 'C': 590.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 590.0, 'C': 140.0, 'G': 217.0, 'T': 28.0},\n",
       "  5: {'A': 590.0, 'C': 105.0, 'G': 68.0, 'T': 15.0},\n",
       "  6: {'A': 1.0, 'C': 375.0, 'G': 1.0, 'T': 590.0},\n",
       "  7: {'A': 189.0, 'C': 1.0, 'G': 590.0, 'T': 1.0},\n",
       "  8: {'A': 117.0, 'C': 288.0, 'G': 191.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '7651823',\n",
       "  'uniprot_ids': 'P35713',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0698.1',\n",
       "  'Matrix_Name': 'ZBTB18',\n",
       "  1: {'A': 3562.0, 'C': 4189.0, 'G': 2500.0, 'T': 3424.0},\n",
       "  2: {'A': 7065.0, 'C': 754.0, 'G': 2536.0, 'T': 3320.0},\n",
       "  3: {'A': 1178.0, 'C': 1938.0, 'G': 5213.0, 'T': 13671.0},\n",
       "  4: {'A': 3619.0, 'C': 13671.0, 'G': 1900.0, 'T': 61.0},\n",
       "  5: {'A': 22.0, 'C': 13671.0, 'G': 4.0, 'T': 9.0},\n",
       "  6: {'A': 13671.0, 'C': 6.0, 'G': 34.0, 'T': 207.0},\n",
       "  7: {'A': 4.0, 'C': 246.0, 'G': 13671.0, 'T': 7.0},\n",
       "  8: {'A': 13671.0, 'C': 1049.0, 'G': 279.0, 'T': 298.0},\n",
       "  9: {'A': 15.0, 'C': 16.0, 'G': 95.0, 'T': 13671.0},\n",
       "  10: {'A': 4.0, 'C': 2.0, 'G': 13671.0, 'T': 2.0},\n",
       "  11: {'A': 49.0, 'C': 315.0, 'G': 37.0, 'T': 13671.0},\n",
       "  12: {'A': 211.0, 'C': 178.0, 'G': 5531.0, 'T': 8141.0},\n",
       "  13: {'A': 1726.0, 'C': 4043.0, 'G': 3877.0, 'T': 4028.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '9756912',\n",
       "  'uniprot_ids': 'Q99592',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0802.1',\n",
       "  'Matrix_Name': 'TBR1',\n",
       "  1: {'A': 34510.0, 'C': 360.0, 'G': 3330.0, 'T': 1434.0},\n",
       "  2: {'A': 6881.0, 'C': 3256.0, 'G': 34510.0, 'T': 4481.0},\n",
       "  3: {'A': 71.0, 'C': 255.0, 'G': 34510.0, 'T': 214.0},\n",
       "  4: {'A': 40.0, 'C': 1760.0, 'G': 89.0, 'T': 34510.0},\n",
       "  5: {'A': 1.0, 'C': 7.0, 'G': 34510.0, 'T': 5.0},\n",
       "  6: {'A': 2387.0, 'C': 4743.0, 'G': 158.0, 'T': 34510.0},\n",
       "  7: {'A': 1176.0, 'C': 2552.0, 'G': 34510.0, 'T': 7002.0},\n",
       "  8: {'A': 34510.0, 'C': 91.0, 'G': 585.0, 'T': 206.0},\n",
       "  9: {'A': 25870.0, 'C': 2576.0, 'G': 1489.0, 'T': 4577.0},\n",
       "  10: {'A': 18508.0, 'C': 5620.0, 'G': 4538.0, 'T': 5847.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'TBrain-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'Q16650',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0657.1',\n",
       "  'Matrix_Name': 'KLF13',\n",
       "  1: {'A': 1509.0, 'C': 249.0, 'G': 916.0, 'T': 591.0},\n",
       "  2: {'A': 734.0, 'C': 47.0, 'G': 797.0, 'T': 1767.0},\n",
       "  3: {'A': 1892.0, 'C': 1.0, 'G': 4631.0, 'T': 60.0},\n",
       "  4: {'A': 497.0, 'C': 924.0, 'G': 59.0, 'T': 103.0},\n",
       "  5: {'A': 76.0, 'C': 925.0, 'G': 3.0, 'T': 62.0},\n",
       "  6: {'A': 2385.0, 'C': 1.0, 'G': 12.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1354.0, 'G': 8.0, 'T': 4.0},\n",
       "  8: {'A': 25.0, 'C': 3.0, 'G': 3029.0, 'T': 10.0},\n",
       "  9: {'A': 1.0, 'C': 1597.0, 'G': 1.0, 'T': 8.0},\n",
       "  10: {'A': 5.0, 'C': 1470.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 2.0, 'C': 1761.0, 'G': 1.0, 'T': 17.0},\n",
       "  12: {'A': 510.0, 'C': 1767.0, 'G': 1.0, 'T': 11.0},\n",
       "  13: {'A': 4.0, 'C': 639.0, 'G': 4.0, 'T': 1404.0},\n",
       "  14: {'A': 157.0, 'C': 351.0, 'G': 55.0, 'T': 2041.0},\n",
       "  15: {'A': 248.0, 'C': 236.0, 'G': 271.0, 'T': 2969.0},\n",
       "  16: {'A': 736.0, 'C': 555.0, 'G': 842.0, 'T': 2248.0},\n",
       "  17: {'A': 405.0, 'C': 177.0, 'G': 484.0, 'T': 1673.0},\n",
       "  18: {'A': 286.0, 'C': 222.0, 'G': 1227.0, 'T': 561.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16266294',\n",
       "  'uniprot_ids': 'Q9Y2Y9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0162.4',\n",
       "  'Matrix_Name': 'EGR1',\n",
       "  1: {'A': 7088.0, 'C': 11611.0, 'G': 6622.0, 'T': 5281.0},\n",
       "  2: {'A': 3998.0, 'C': 14750.0, 'G': 7212.0, 'T': 4642.0},\n",
       "  3: {'A': 14509.0, 'C': 10304.0, 'G': 3855.0, 'T': 1934.0},\n",
       "  4: {'A': 643.0, 'C': 28404.0, 'G': 757.0, 'T': 798.0},\n",
       "  5: {'A': 5111.0, 'C': 1132.0, 'G': 22238.0, 'T': 2121.0},\n",
       "  6: {'A': 228.0, 'C': 29339.0, 'G': 666.0, 'T': 369.0},\n",
       "  7: {'A': 1441.0, 'C': 27322.0, 'G': 1456.0, 'T': 383.0},\n",
       "  8: {'A': 315.0, 'C': 28959.0, 'G': 699.0, 'T': 629.0},\n",
       "  9: {'A': 22298.0, 'C': 5421.0, 'G': 2123.0, 'T': 760.0},\n",
       "  10: {'A': 154.0, 'C': 29508.0, 'G': 700.0, 'T': 240.0},\n",
       "  11: {'A': 4697.0, 'C': 2807.0, 'G': 20545.0, 'T': 2553.0},\n",
       "  12: {'A': 970.0, 'C': 25197.0, 'G': 2158.0, 'T': 2277.0},\n",
       "  13: {'A': 8904.0, 'C': 10442.0, 'G': 6346.0, 'T': 4910.0},\n",
       "  14: {'A': 3810.0, 'C': 13687.0, 'G': 7853.0, 'T': 5252.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16041365',\n",
       "  'uniprot_ids': 'P18146',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0464.2',\n",
       "  'Matrix_Name': 'BHLHE40',\n",
       "  1: {'A': 3825.0, 'C': 3026.0, 'G': 2927.0, 'T': 1448.0},\n",
       "  2: {'A': 1392.0, 'C': 482.0, 'G': 5001.0, 'T': 6223.0},\n",
       "  3: {'A': 60.0, 'C': 11223.0, 'G': 1.0, 'T': 15.0},\n",
       "  4: {'A': 11223.0, 'C': 131.0, 'G': 79.0, 'T': 78.0},\n",
       "  5: {'A': 24.0, 'C': 11223.0, 'G': 3.0, 'T': 33.0},\n",
       "  6: {'A': 67.0, 'C': 31.0, 'G': 11223.0, 'T': 1.0},\n",
       "  7: {'A': 168.0, 'C': 249.0, 'G': 88.0, 'T': 11223.0},\n",
       "  8: {'A': 5.0, 'C': 13.0, 'G': 11223.0, 'T': 73.0},\n",
       "  9: {'A': 5928.0, 'C': 5296.0, 'G': 130.0, 'T': 501.0},\n",
       "  10: {'A': 1545.0, 'C': 5707.0, 'G': 1386.0, 'T': 2588.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15186484',\n",
       "  'uniprot_ids': 'O14503',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1601.1',\n",
       "  'Matrix_Name': 'ZNF75D',\n",
       "  1: {'A': 2112.0, 'C': 1574.0, 'G': 1851.0, 'T': 1226.0},\n",
       "  2: {'A': 1058.0, 'C': 716.0, 'G': 1469.0, 'T': 3520.0},\n",
       "  3: {'A': 3.0, 'C': 1.0, 'G': 6758.0, 'T': 1.0},\n",
       "  4: {'A': 11.0, 'C': 60.0, 'G': 5.0, 'T': 6687.0},\n",
       "  5: {'A': 27.0, 'C': 19.0, 'G': 6711.0, 'T': 6.0},\n",
       "  6: {'A': 47.0, 'C': 45.0, 'G': 6546.0, 'T': 125.0},\n",
       "  7: {'A': 25.0, 'C': 30.0, 'G': 6679.0, 'T': 29.0},\n",
       "  8: {'A': 6707.0, 'C': 21.0, 'G': 16.0, 'T': 19.0},\n",
       "  9: {'A': 4095.0, 'C': 577.0, 'G': 1694.0, 'T': 397.0},\n",
       "  10: {'A': 5246.0, 'C': 257.0, 'G': 747.0, 'T': 513.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'P51815',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0160.1',\n",
       "  'Matrix_Name': 'NR4A2',\n",
       "  1: {'A': 9.0, 'C': 2.0, 'G': 4.0, 'T': 2.0},\n",
       "  2: {'A': 14.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 14.0, 'T': 2.0},\n",
       "  4: {'A': 4.0, 'C': 1.0, 'G': 12.0, 'T': 1.0},\n",
       "  5: {'A': 3.0, 'C': 3.0, 'G': 1.0, 'T': 11.0},\n",
       "  6: {'A': 1.0, 'C': 14.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 15.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 4.0, 'C': 9.0, 'G': 3.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NGFI-B-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P43354',\n",
       "  'data_type': 'COMPILED'},\n",
       " {'Matrix_ID': 'MA1515.1',\n",
       "  'Matrix_Name': 'KLF2',\n",
       "  1: {'A': 1055.0, 'C': 1377.0, 'G': 1321.0, 'T': 873.0},\n",
       "  2: {'A': 2511.0, 'C': 1104.0, 'G': 2113.0, 'T': 434.0},\n",
       "  3: {'A': 742.0, 'C': 4623.0, 'G': 496.0, 'T': 347.0},\n",
       "  4: {'A': 404.0, 'C': 4623.0, 'G': 187.0, 'T': 313.0},\n",
       "  5: {'A': 4623.0, 'C': 2266.0, 'G': 1061.0, 'T': 215.0},\n",
       "  6: {'A': 1.0, 'C': 4623.0, 'G': 1.0, 'T': 79.0},\n",
       "  7: {'A': 1703.0, 'C': 275.0, 'G': 2921.0, 'T': 168.0},\n",
       "  8: {'A': 1.0, 'C': 4623.0, 'G': 50.0, 'T': 23.0},\n",
       "  9: {'A': 129.0, 'C': 4623.0, 'G': 196.0, 'T': 10.0},\n",
       "  10: {'A': 69.0, 'C': 4623.0, 'G': 106.0, 'T': 445.0},\n",
       "  11: {'A': 1839.0, 'C': 1143.0, 'G': 390.0, 'T': 1254.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11443140',\n",
       "  'uniprot_ids': 'Q9Y5W3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0497.1',\n",
       "  'Matrix_Name': 'MEF2C',\n",
       "  1: {'A': 706.0, 'C': 322.0, 'G': 677.0, 'T': 508.0},\n",
       "  2: {'A': 734.0, 'C': 152.0, 'G': 574.0, 'T': 753.0},\n",
       "  3: {'A': 432.0, 'C': 197.0, 'G': 823.0, 'T': 761.0},\n",
       "  4: {'A': 383.0, 'C': 1413.0, 'G': 79.0, 'T': 338.0},\n",
       "  5: {'A': 1.0, 'C': 986.0, 'G': 1.0, 'T': 1225.0},\n",
       "  6: {'A': 1617.0, 'C': 257.0, 'G': 75.0, 'T': 264.0},\n",
       "  7: {'A': 1707.0, 'C': 33.0, 'G': 242.0, 'T': 231.0},\n",
       "  8: {'A': 2108.0, 'C': 1.0, 'G': 88.0, 'T': 16.0},\n",
       "  9: {'A': 2132.0, 'C': 1.0, 'G': 3.0, 'T': 77.0},\n",
       "  10: {'A': 2136.0, 'C': 1.0, 'G': 5.0, 'T': 71.0},\n",
       "  11: {'A': 57.0, 'C': 63.0, 'G': 1.0, 'T': 2092.0},\n",
       "  12: {'A': 2178.0, 'C': 1.0, 'G': 33.0, 'T': 1.0},\n",
       "  13: {'A': 390.0, 'C': 121.0, 'G': 1672.0, 'T': 30.0},\n",
       "  14: {'A': 976.0, 'C': 837.0, 'G': 149.0, 'T': 251.0},\n",
       "  15: {'A': 1010.0, 'C': 451.0, 'G': 127.0, 'T': 625.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Regulators',\n",
       "  'tf_class': 'MADS',\n",
       "  'pubmed_ids': '7559475',\n",
       "  'uniprot_ids': 'Q06413',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0636.1',\n",
       "  'Matrix_Name': 'BHLHE41',\n",
       "  1: {'A': 2895.0, 'C': 1047.0, 'G': 5575.0, 'T': 60.0},\n",
       "  2: {'A': 99.0, 'C': 18.0, 'G': 1832.0, 'T': 5575.0},\n",
       "  3: {'A': 3.0, 'C': 5575.0, 'G': 4.0, 'T': 2.0},\n",
       "  4: {'A': 5575.0, 'C': 13.0, 'G': 22.0, 'T': 4.0},\n",
       "  5: {'A': 2.0, 'C': 5575.0, 'G': 2.0, 'T': 6.0},\n",
       "  6: {'A': 3.0, 'C': 2.0, 'G': 5575.0, 'T': 1.0},\n",
       "  7: {'A': 20.0, 'C': 17.0, 'G': 17.0, 'T': 5575.0},\n",
       "  8: {'A': 2.0, 'C': 2.0, 'G': 5575.0, 'T': 1.0},\n",
       "  9: {'A': 5575.0, 'C': 1230.0, 'G': 8.0, 'T': 24.0},\n",
       "  10: {'A': 75.0, 'C': 5575.0, 'G': 986.0, 'T': 1964.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '18345027',\n",
       "  'uniprot_ids': 'Q9C0J9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0891.1',\n",
       "  'Matrix_Name': 'GSC2',\n",
       "  1: {'A': 462.0, 'C': 687.0, 'G': 443.0, 'T': 428.0},\n",
       "  2: {'A': 357.0, 'C': 840.0, 'G': 191.0, 'T': 631.0},\n",
       "  3: {'A': 117.0, 'C': 1.0, 'G': 1.0, 'T': 2017.0},\n",
       "  4: {'A': 2017.0, 'C': 88.0, 'G': 57.0, 'T': 202.0},\n",
       "  5: {'A': 2017.0, 'C': 12.0, 'G': 1.0, 'T': 133.0},\n",
       "  6: {'A': 1.0, 'C': 42.0, 'G': 146.0, 'T': 2017.0},\n",
       "  7: {'A': 256.0, 'C': 2017.0, 'G': 180.0, 'T': 205.0},\n",
       "  8: {'A': 183.0, 'C': 2017.0, 'G': 423.0, 'T': 440.0},\n",
       "  9: {'A': 206.0, 'C': 669.0, 'G': 699.0, 'T': 445.0},\n",
       "  10: {'A': 510.0, 'C': 815.0, 'G': 169.0, 'T': 526.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9441739',\n",
       "  'uniprot_ids': 'O15499',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0861.1',\n",
       "  'Matrix_Name': 'TP73',\n",
       "  1: {'A': 3618.0, 'C': 314.0, 'G': 3892.0, 'T': 3690.0},\n",
       "  2: {'A': 6207.0, 'C': 430.0, 'G': 3305.0, 'T': 209.0},\n",
       "  3: {'A': 1.0, 'C': 10830.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 8761.0, 'C': 125.0, 'G': 558.0, 'T': 199.0},\n",
       "  5: {'A': 3838.0, 'C': 294.0, 'G': 807.0, 'T': 11481.0},\n",
       "  6: {'A': 9.0, 'C': 29.0, 'G': 11788.0, 'T': 2.0},\n",
       "  7: {'A': 131.0, 'C': 2592.0, 'G': 92.0, 'T': 11893.0},\n",
       "  8: {'A': 1378.0, 'C': 8162.0, 'G': 784.0, 'T': 2750.0},\n",
       "  9: {'A': 2167.0, 'C': 2941.0, 'G': 2231.0, 'T': 4243.0},\n",
       "  10: {'A': 3489.0, 'C': 2419.0, 'G': 3559.0, 'T': 2458.0},\n",
       "  11: {'A': 3989.0, 'C': 442.0, 'G': 7798.0, 'T': 1003.0},\n",
       "  12: {'A': 9640.0, 'C': 146.0, 'G': 1415.0, 'T': 443.0},\n",
       "  13: {'A': 1.0, 'C': 11704.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 10126.0, 'C': 144.0, 'G': 280.0, 'T': 2303.0},\n",
       "  15: {'A': 257.0, 'C': 5966.0, 'G': 131.0, 'T': 8477.0},\n",
       "  16: {'A': 17.0, 'C': 171.0, 'G': 13027.0, 'T': 40.0},\n",
       "  17: {'A': 994.0, 'C': 6945.0, 'G': 182.0, 'T': 6996.0},\n",
       "  18: {'A': 4009.0, 'C': 5126.0, 'G': 520.0, 'T': 2233.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'p53-related',\n",
       "  'tf_class': 'p53',\n",
       "  'pubmed_ids': '23243311',\n",
       "  'uniprot_ids': 'O15350',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0513.1',\n",
       "  'Matrix_Name': 'SMAD2::SMAD3::SMAD4',\n",
       "  1: {'A': 167.0, 'C': 433.0, 'G': 169.0, 'T': 134.0},\n",
       "  2: {'A': 197.0, 'C': 76.0, 'G': 35.0, 'T': 595.0},\n",
       "  3: {'A': 15.0, 'C': 9.0, 'G': 878.0, 'T': 1.0},\n",
       "  4: {'A': 34.0, 'C': 46.0, 'G': 104.0, 'T': 719.0},\n",
       "  5: {'A': 1.0, 'C': 900.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 900.0},\n",
       "  7: {'A': 107.0, 'C': 232.0, 'G': 540.0, 'T': 24.0},\n",
       "  8: {'A': 212.0, 'C': 126.0, 'G': 281.0, 'T': 284.0},\n",
       "  9: {'A': 1.0, 'C': 897.0, 'G': 4.0, 'T': 1.0},\n",
       "  10: {'A': 766.0, 'C': 20.0, 'G': 19.0, 'T': 98.0},\n",
       "  11: {'A': 31.0, 'C': 737.0, 'G': 134.0, 'T': 1.0},\n",
       "  12: {'A': 118.0, 'C': 584.0, 'G': 48.0, 'T': 153.0},\n",
       "  13: {'A': 111.0, 'C': 215.0, 'G': 36.0, 'T': 541.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'SMAD',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '9741623',\n",
       "  'uniprot_ids': 'Q15796;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0073.1',\n",
       "  'Matrix_Name': 'RREB1',\n",
       "  1: {'A': 4.0, 'C': 9.0, 'G': 1.0, 'T': 1.0},\n",
       "  2: {'A': 2.0, 'C': 11.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 4.0, 'C': 9.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 12.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 8.0, 'C': 5.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 10.0, 'C': 3.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 9.0, 'C': 4.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 5.0, 'C': 7.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 1.0, 'C': 12.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 12.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 5.0, 'C': 8.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 2.0, 'C': 11.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 4.0, 'C': 9.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 5.0, 'C': 7.0, 'G': 2.0, 'T': 1.0},\n",
       "  15: {'A': 3.0, 'C': 10.0, 'G': 1.0, 'T': 1.0},\n",
       "  16: {'A': 5.0, 'C': 6.0, 'G': 1.0, 'T': 3.0},\n",
       "  17: {'A': 5.0, 'C': 6.0, 'G': 2.0, 'T': 2.0},\n",
       "  18: {'A': 5.0, 'C': 7.0, 'G': 1.0, 'T': 2.0},\n",
       "  19: {'A': 2.0, 'C': 8.0, 'G': 4.0, 'T': 1.0},\n",
       "  20: {'A': 5.0, 'C': 5.0, 'G': 3.0, 'T': 2.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 20,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '8816445',\n",
       "  'uniprot_ids': 'Q92766',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0653.1',\n",
       "  'Matrix_Name': 'IRF9',\n",
       "  1: {'A': 3468.0, 'C': 53.0, 'G': 44.0, 'T': 111.0},\n",
       "  2: {'A': 2393.0, 'C': 499.0, 'G': 205.0, 'T': 1077.0},\n",
       "  3: {'A': 50.0, 'C': 3468.0, 'G': 34.0, 'T': 119.0},\n",
       "  4: {'A': 44.0, 'C': 4.0, 'G': 3468.0, 'T': 6.0},\n",
       "  5: {'A': 3468.0, 'C': 6.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 3468.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  7: {'A': 3468.0, 'C': 3.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 44.0, 'C': 3468.0, 'G': 198.0, 'T': 10.0},\n",
       "  9: {'A': 17.0, 'C': 3468.0, 'G': 1.0, 'T': 832.0},\n",
       "  10: {'A': 3.0, 'C': 1.0, 'G': 3468.0, 'T': 2.0},\n",
       "  11: {'A': 3468.0, 'C': 3.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 3468.0, 'C': 2.0, 'G': 1.0, 'T': 22.0},\n",
       "  13: {'A': 3468.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  14: {'A': 5.0, 'C': 3468.0, 'G': 240.0, 'T': 3.0},\n",
       "  15: {'A': 100.0, 'C': 1130.0, 'G': 8.0, 'T': 2339.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '21731497',\n",
       "  'uniprot_ids': 'Q00978',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0510.2',\n",
       "  'Matrix_Name': 'RFX5',\n",
       "  1: {'A': 278.0, 'C': 1076.0, 'G': 791.0, 'T': 491.0},\n",
       "  2: {'A': 91.0, 'C': 1.0, 'G': 2820.0, 'T': 10.0},\n",
       "  3: {'A': 5.0, 'C': 34.0, 'G': 16.0, 'T': 2903.0},\n",
       "  4: {'A': 48.0, 'C': 99.0, 'G': 11.0, 'T': 2843.0},\n",
       "  5: {'A': 822.0, 'C': 82.0, 'G': 1676.0, 'T': 143.0},\n",
       "  6: {'A': 4.0, 'C': 2379.0, 'G': 39.0, 'T': 70.0},\n",
       "  7: {'A': 2.0, 'C': 2119.0, 'G': 8.0, 'T': 726.0},\n",
       "  8: {'A': 2147.0, 'C': 43.0, 'G': 48.0, 'T': 189.0},\n",
       "  9: {'A': 145.0, 'C': 57.0, 'G': 23.0, 'T': 2873.0},\n",
       "  10: {'A': 761.0, 'C': 12.0, 'G': 2267.0, 'T': 9.0},\n",
       "  11: {'A': 144.0, 'C': 49.0, 'G': 2665.0, 'T': 1.0},\n",
       "  12: {'A': 140.0, 'C': 1614.0, 'G': 62.0, 'T': 883.0},\n",
       "  13: {'A': 2167.0, 'C': 11.0, 'G': 100.0, 'T': 47.0},\n",
       "  14: {'A': 2207.0, 'C': 22.0, 'G': 17.0, 'T': 10.0},\n",
       "  15: {'A': 27.0, 'C': 2380.0, 'G': 13.0, 'T': 68.0},\n",
       "  16: {'A': 463.0, 'C': 891.0, 'G': 906.0, 'T': 287.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'RFX-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '8754849',\n",
       "  'uniprot_ids': 'P48382',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1099.2',\n",
       "  'Matrix_Name': 'HES1',\n",
       "  1: {'A': 3322.0, 'C': 3493.0, 'G': 18527.0, 'T': 4240.0},\n",
       "  2: {'A': 5859.0, 'C': 2073.0, 'G': 18527.0, 'T': 8008.0},\n",
       "  3: {'A': 465.0, 'C': 18527.0, 'G': 1.0, 'T': 19.0},\n",
       "  4: {'A': 9447.0, 'C': 1.0, 'G': 9082.0, 'T': 233.0},\n",
       "  5: {'A': 1.0, 'C': 18527.0, 'G': 1.0, 'T': 792.0},\n",
       "  6: {'A': 22.0, 'C': 1.0, 'G': 18527.0, 'T': 1.0},\n",
       "  7: {'A': 254.0, 'C': 4058.0, 'G': 1.0, 'T': 18527.0},\n",
       "  8: {'A': 2.0, 'C': 1.0, 'G': 18527.0, 'T': 54.0},\n",
       "  9: {'A': 1127.0, 'C': 6204.0, 'G': 6361.0, 'T': 4838.0},\n",
       "  10: {'A': 2743.0, 'C': 12514.0, 'G': 6014.0, 'T': 3217.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15870295',\n",
       "  'uniprot_ids': 'Q14469',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0125.1',\n",
       "  'Matrix_Name': 'Nobox',\n",
       "  1: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 38.0},\n",
       "  2: {'A': 37.0, 'C': 1.0, 'G': 1.0, 'T': 3.0},\n",
       "  3: {'A': 39.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 2.0, 'C': 2.0, 'G': 3.0, 'T': 35.0},\n",
       "  5: {'A': 3.0, 'C': 1.0, 'G': 5.0, 'T': 33.0},\n",
       "  6: {'A': 16.0, 'C': 1.0, 'G': 23.0, 'T': 2.0},\n",
       "  7: {'A': 5.0, 'C': 13.0, 'G': 19.0, 'T': 5.0},\n",
       "  8: {'A': 3.0, 'C': 14.0, 'G': 7.0, 'T': 18.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '16997917',\n",
       "  'uniprot_ids': 'Q8VIH1',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1511.1',\n",
       "  'Matrix_Name': 'KLF10',\n",
       "  1: {'A': 4747.0, 'C': 2436.0, 'G': 14087.0, 'T': 1252.0},\n",
       "  2: {'A': 6634.0, 'C': 12655.0, 'G': 1300.0, 'T': 1374.0},\n",
       "  3: {'A': 634.0, 'C': 19620.0, 'G': 99.0, 'T': 486.0},\n",
       "  4: {'A': 18752.0, 'C': 1664.0, 'G': 1724.0, 'T': 7.0},\n",
       "  5: {'A': 239.0, 'C': 19908.0, 'G': 4.0, 'T': 158.0},\n",
       "  6: {'A': 13507.0, 'C': 12.0, 'G': 6235.0, 'T': 544.0},\n",
       "  7: {'A': 2.0, 'C': 19996.0, 'G': 3.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 19995.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 19245.0, 'G': 1.0, 'T': 2199.0},\n",
       "  10: {'A': 7546.0, 'C': 11404.0, 'G': 588.0, 'T': 2687.0},\n",
       "  11: {'A': 935.0, 'C': 13119.0, 'G': 771.0, 'T': 6275.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11443140',\n",
       "  'uniprot_ids': 'Q13118',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0471.2',\n",
       "  'Matrix_Name': 'E2F6',\n",
       "  1: {'A': 7297.0, 'C': 8076.0, 'G': 12359.0, 'T': 5370.0},\n",
       "  2: {'A': 8380.0, 'C': 7406.0, 'G': 11650.0, 'T': 5666.0},\n",
       "  3: {'A': 9538.0, 'C': 5103.0, 'G': 11563.0, 'T': 6898.0},\n",
       "  4: {'A': 2111.0, 'C': 1149.0, 'G': 29388.0, 'T': 454.0},\n",
       "  5: {'A': 267.0, 'C': 838.0, 'G': 31791.0, 'T': 206.0},\n",
       "  6: {'A': 3857.0, 'C': 26425.0, 'G': 694.0, 'T': 2126.0},\n",
       "  7: {'A': 1174.0, 'C': 540.0, 'G': 30574.0, 'T': 814.0},\n",
       "  8: {'A': 339.0, 'C': 435.0, 'G': 31974.0, 'T': 354.0},\n",
       "  9: {'A': 355.0, 'C': 835.0, 'G': 31615.0, 'T': 297.0},\n",
       "  10: {'A': 29338.0, 'C': 1172.0, 'G': 2323.0, 'T': 269.0},\n",
       "  11: {'A': 25726.0, 'C': 1596.0, 'G': 5022.0, 'T': 758.0},\n",
       "  12: {'A': 8996.0, 'C': 7411.0, 'G': 12531.0, 'T': 4164.0},\n",
       "  13: {'A': 7617.0, 'C': 8478.0, 'G': 11406.0, 'T': 5601.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '17908821',\n",
       "  'uniprot_ids': 'O75461',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1631.1',\n",
       "  'Matrix_Name': 'ASCL1(var.2)',\n",
       "  1: {'A': 6834.0, 'C': 11806.0, 'G': 8372.0, 'T': 7348.0},\n",
       "  2: {'A': 8953.0, 'C': 7999.0, 'G': 8602.0, 'T': 8806.0},\n",
       "  3: {'A': 5837.0, 'C': 6655.0, 'G': 18577.0, 'T': 3291.0},\n",
       "  4: {'A': 185.0, 'C': 33563.0, 'G': 322.0, 'T': 290.0},\n",
       "  5: {'A': 32316.0, 'C': 564.0, 'G': 861.0, 'T': 619.0},\n",
       "  6: {'A': 164.0, 'C': 30170.0, 'G': 3210.0, 'T': 816.0},\n",
       "  7: {'A': 197.0, 'C': 33403.0, 'G': 608.0, 'T': 152.0},\n",
       "  8: {'A': 369.0, 'C': 741.0, 'G': 493.0, 'T': 32757.0},\n",
       "  9: {'A': 252.0, 'C': 446.0, 'G': 33412.0, 'T': 250.0},\n",
       "  10: {'A': 400.0, 'C': 28845.0, 'G': 2720.0, 'T': 2395.0},\n",
       "  11: {'A': 4491.0, 'C': 17884.0, 'G': 2967.0, 'T': 9018.0},\n",
       "  12: {'A': 7084.0, 'C': 10556.0, 'G': 8940.0, 'T': 7780.0},\n",
       "  13: {'A': 5973.0, 'C': 11997.0, 'G': 8357.0, 'T': 8033.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '31086315',\n",
       "  'uniprot_ids': 'P50553',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0688.1',\n",
       "  'Matrix_Name': 'TBX2',\n",
       "  1: {'A': 1259.0, 'C': 403.0, 'G': 718.0, 'T': 805.0},\n",
       "  2: {'A': 3182.0, 'C': 60.0, 'G': 717.0, 'T': 443.0},\n",
       "  3: {'A': 603.0, 'C': 520.0, 'G': 3182.0, 'T': 354.0},\n",
       "  4: {'A': 10.0, 'C': 37.0, 'G': 3182.0, 'T': 15.0},\n",
       "  5: {'A': 1.0, 'C': 330.0, 'G': 22.0, 'T': 3182.0},\n",
       "  6: {'A': 27.0, 'C': 1.0, 'G': 3182.0, 'T': 13.0},\n",
       "  7: {'A': 308.0, 'C': 393.0, 'G': 28.0, 'T': 3182.0},\n",
       "  8: {'A': 311.0, 'C': 1290.0, 'G': 3182.0, 'T': 1410.0},\n",
       "  9: {'A': 3182.0, 'C': 27.0, 'G': 126.0, 'T': 121.0},\n",
       "  10: {'A': 2526.0, 'C': 492.0, 'G': 528.0, 'T': 657.0},\n",
       "  11: {'A': 1478.0, 'C': 447.0, 'G': 591.0, 'T': 668.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'TBX2-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '16285859',\n",
       "  'uniprot_ids': 'Q13207',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0871.2',\n",
       "  'Matrix_Name': 'TFEC',\n",
       "  1: {'A': 3274.0, 'C': 4680.0, 'G': 2356.0, 'T': 2726.0},\n",
       "  2: {'A': 1842.0, 'C': 2888.0, 'G': 2303.0, 'T': 923.0},\n",
       "  3: {'A': 1.0, 'C': 7953.0, 'G': 4.0, 'T': 1.0},\n",
       "  4: {'A': 7953.0, 'C': 1.0, 'G': 146.0, 'T': 375.0},\n",
       "  5: {'A': 1.0, 'C': 7953.0, 'G': 1.0, 'T': 211.0},\n",
       "  6: {'A': 70.0, 'C': 1.0, 'G': 7953.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7953.0},\n",
       "  8: {'A': 1.0, 'C': 5.0, 'G': 7953.0, 'T': 1.0},\n",
       "  9: {'A': 7953.0, 'C': 412.0, 'G': 144.0, 'T': 48.0},\n",
       "  10: {'A': 18.0, 'C': 6901.0, 'G': 59.0, 'T': 1053.0},\n",
       "  11: {'A': 1851.0, 'C': 2829.0, 'G': 1709.0, 'T': 1567.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15186484',\n",
       "  'uniprot_ids': 'O14948',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0863.1',\n",
       "  'Matrix_Name': 'MTF1',\n",
       "  1: {'A': 2.0, 'C': 1.0, 'G': 5.0, 'T': 51.0},\n",
       "  2: {'A': 4.0, 'C': 6.0, 'G': 11.0, 'T': 41.0},\n",
       "  3: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 54.0},\n",
       "  4: {'A': 3.0, 'C': 2.0, 'G': 67.0, 'T': 1.0},\n",
       "  5: {'A': 2.0, 'C': 72.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 54.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  7: {'A': 3.0, 'C': 70.0, 'G': 3.0, 'T': 1.0},\n",
       "  8: {'A': 53.0, 'C': 4.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 1.0, 'C': 68.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 2.0, 'C': 1.0, 'G': 71.0, 'T': 2.0},\n",
       "  11: {'A': 10.0, 'C': 3.0, 'G': 43.0, 'T': 11.0},\n",
       "  12: {'A': 1.0, 'C': 64.0, 'G': 1.0, 'T': 9.0},\n",
       "  13: {'A': 47.0, 'C': 11.0, 'G': 8.0, 'T': 3.0},\n",
       "  14: {'A': 3.0, 'C': 63.0, 'G': 2.0, 'T': 4.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '8265351',\n",
       "  'uniprot_ids': 'Q14872',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0820.1',\n",
       "  'Matrix_Name': 'FIGLA',\n",
       "  1: {'A': 887.0, 'C': 271.0, 'G': 176.0, 'T': 450.0},\n",
       "  2: {'A': 691.0, 'C': 694.0, 'G': 161.0, 'T': 238.0},\n",
       "  3: {'A': 1.0, 'C': 1781.0, 'G': 1.0, 'T': 30.0},\n",
       "  4: {'A': 1781.0, 'C': 47.0, 'G': 1.0, 'T': 15.0},\n",
       "  5: {'A': 1.0, 'C': 1781.0, 'G': 528.0, 'T': 1.0},\n",
       "  6: {'A': 10.0, 'C': 1781.0, 'G': 312.0, 'T': 1.0},\n",
       "  7: {'A': 57.0, 'C': 1.0, 'G': 18.0, 'T': 1781.0},\n",
       "  8: {'A': 37.0, 'C': 34.0, 'G': 1781.0, 'T': 66.0},\n",
       "  9: {'A': 155.0, 'C': 226.0, 'G': 570.0, 'T': 833.0},\n",
       "  10: {'A': 456.0, 'C': 284.0, 'G': 290.0, 'T': 754.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '18499083',\n",
       "  'uniprot_ids': 'Q6QHK4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1621.1',\n",
       "  'Matrix_Name': 'Rbpjl',\n",
       "  1: {'A': 2007.0, 'C': 2882.0, 'G': 2477.0, 'T': 2290.0},\n",
       "  2: {'A': 2467.0, 'C': 2428.0, 'G': 2550.0, 'T': 2211.0},\n",
       "  3: {'A': 3370.0, 'C': 3118.0, 'G': 1803.0, 'T': 1365.0},\n",
       "  4: {'A': 6006.0, 'C': 1374.0, 'G': 1493.0, 'T': 783.0},\n",
       "  5: {'A': 140.0, 'C': 9274.0, 'G': 104.0, 'T': 138.0},\n",
       "  6: {'A': 9387.0, 'C': 71.0, 'G': 111.0, 'T': 87.0},\n",
       "  7: {'A': 85.0, 'C': 8534.0, 'G': 673.0, 'T': 364.0},\n",
       "  8: {'A': 235.0, 'C': 9006.0, 'G': 318.0, 'T': 97.0},\n",
       "  9: {'A': 64.0, 'C': 145.0, 'G': 133.0, 'T': 9314.0},\n",
       "  10: {'A': 103.0, 'C': 132.0, 'G': 9296.0, 'T': 125.0},\n",
       "  11: {'A': 570.0, 'C': 1719.0, 'G': 1582.0, 'T': 5785.0},\n",
       "  12: {'A': 1540.0, 'C': 3248.0, 'G': 2688.0, 'T': 2180.0},\n",
       "  13: {'A': 2274.0, 'C': 2885.0, 'G': 1959.0, 'T': 2538.0},\n",
       "  14: {'A': 2083.0, 'C': 3004.0, 'G': 2282.0, 'T': 2287.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'CSL-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '20398665',\n",
       "  'uniprot_ids': 'O08674',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0520.1',\n",
       "  'Matrix_Name': 'Stat6',\n",
       "  1: {'A': 258.0, 'C': 664.0, 'G': 483.0, 'T': 451.0},\n",
       "  2: {'A': 666.0, 'C': 329.0, 'G': 373.0, 'T': 488.0},\n",
       "  3: {'A': 153.0, 'C': 567.0, 'G': 437.0, 'T': 699.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1853.0},\n",
       "  5: {'A': 14.0, 'C': 1.0, 'G': 1.0, 'T': 1840.0},\n",
       "  6: {'A': 123.0, 'C': 1681.0, 'G': 1.0, 'T': 51.0},\n",
       "  7: {'A': 71.0, 'C': 1205.0, 'G': 1.0, 'T': 579.0},\n",
       "  8: {'A': 665.0, 'C': 22.0, 'G': 220.0, 'T': 949.0},\n",
       "  9: {'A': 329.0, 'C': 535.0, 'G': 899.0, 'T': 93.0},\n",
       "  10: {'A': 1154.0, 'C': 4.0, 'G': 417.0, 'T': 281.0},\n",
       "  11: {'A': 27.0, 'C': 1.0, 'G': 1797.0, 'T': 31.0},\n",
       "  12: {'A': 1853.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 1850.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  14: {'A': 622.0, 'C': 541.0, 'G': 399.0, 'T': 294.0},\n",
       "  15: {'A': 346.0, 'C': 491.0, 'G': 236.0, 'T': 783.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '21828071',\n",
       "  'uniprot_ids': 'P52633',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0732.1',\n",
       "  'Matrix_Name': 'EGR3',\n",
       "  1: {'A': 338.0, 'C': 646.0, 'G': 279.0, 'T': 471.0},\n",
       "  2: {'A': 445.0, 'C': 486.0, 'G': 192.0, 'T': 541.0},\n",
       "  3: {'A': 1032.0, 'C': 702.0, 'G': 11.0, 'T': 21.0},\n",
       "  4: {'A': 3.0, 'C': 1856.0, 'G': 1.0, 'T': 14.0},\n",
       "  5: {'A': 67.0, 'C': 33.0, 'G': 2159.0, 'T': 37.0},\n",
       "  6: {'A': 1.0, 'C': 1857.0, 'G': 3.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1859.0, 'G': 1.0, 'T': 11.0},\n",
       "  8: {'A': 1.0, 'C': 1824.0, 'G': 8.0, 'T': 81.0},\n",
       "  9: {'A': 1160.0, 'C': 456.0, 'G': 12.0, 'T': 5.0},\n",
       "  10: {'A': 1.0, 'C': 1876.0, 'G': 8.0, 'T': 3.0},\n",
       "  11: {'A': 40.0, 'C': 29.0, 'G': 2153.0, 'T': 27.0},\n",
       "  12: {'A': 3.0, 'C': 1900.0, 'G': 2.0, 'T': 22.0},\n",
       "  13: {'A': 1263.0, 'C': 55.0, 'G': 169.0, 'T': 81.0},\n",
       "  14: {'A': 407.0, 'C': 601.0, 'G': 111.0, 'T': 550.0},\n",
       "  15: {'A': 397.0, 'C': 347.0, 'G': 231.0, 'T': 684.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7891721',\n",
       "  'uniprot_ids': 'Q06889',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0723.1',\n",
       "  'Matrix_Name': 'VAX2',\n",
       "  1: {'A': 922.0, 'C': 7817.0, 'G': 1385.0, 'T': 6483.0},\n",
       "  2: {'A': 1411.0, 'C': 2824.0, 'G': 222.0, 'T': 14298.0},\n",
       "  3: {'A': 14298.0, 'C': 2782.0, 'G': 415.0, 'T': 566.0},\n",
       "  4: {'A': 14298.0, 'C': 214.0, 'G': 95.0, 'T': 219.0},\n",
       "  5: {'A': 577.0, 'C': 162.0, 'G': 565.0, 'T': 14298.0},\n",
       "  6: {'A': 1342.0, 'C': 988.0, 'G': 5674.0, 'T': 14298.0},\n",
       "  7: {'A': 14298.0, 'C': 180.0, 'G': 3819.0, 'T': 563.0},\n",
       "  8: {'A': 2712.0, 'C': 5649.0, 'G': 2870.0, 'T': 3070.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q9UIW0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0607.1',\n",
       "  'Matrix_Name': 'Bhlha15',\n",
       "  1: {'A': 144.0, 'C': 572.0, 'G': 287.0, 'T': 2.0},\n",
       "  2: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 201.0, 'T': 800.0},\n",
       "  5: {'A': 800.0, 'C': 201.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1000.0, 'T': 1.0},\n",
       "  8: {'A': 2.0, 'C': 232.0, 'G': 155.0, 'T': 615.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9482738',\n",
       "  'uniprot_ids': 'Q9QYC3',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1100.2',\n",
       "  'Matrix_Name': 'ASCL1',\n",
       "  1: {'A': 1268.0, 'C': 927.0, 'G': 1496.0, 'T': 727.0},\n",
       "  2: {'A': 1335.0, 'C': 522.0, 'G': 3080.0, 'T': 355.0},\n",
       "  3: {'A': 1.0, 'C': 4414.0, 'G': 1.0, 'T': 15.0},\n",
       "  4: {'A': 4414.0, 'C': 158.0, 'G': 61.0, 'T': 219.0},\n",
       "  5: {'A': 108.0, 'C': 1289.0, 'G': 4414.0, 'T': 626.0},\n",
       "  6: {'A': 730.0, 'C': 4414.0, 'G': 1321.0, 'T': 21.0},\n",
       "  7: {'A': 119.0, 'C': 25.0, 'G': 122.0, 'T': 4414.0},\n",
       "  8: {'A': 53.0, 'C': 1.0, 'G': 4414.0, 'T': 30.0},\n",
       "  9: {'A': 178.0, 'C': 3066.0, 'G': 815.0, 'T': 1350.0},\n",
       "  10: {'A': 844.0, 'C': 1488.0, 'G': 890.0, 'T': 1195.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20382226',\n",
       "  'uniprot_ids': 'P50553',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0740.1',\n",
       "  'Matrix_Name': 'KLF14',\n",
       "  1: {'A': 431.0, 'C': 216.0, 'G': 967.0, 'T': 694.0},\n",
       "  2: {'A': 1206.0, 'C': 74.0, 'G': 2281.0, 'T': 72.0},\n",
       "  3: {'A': 527.0, 'C': 1361.0, 'G': 97.0, 'T': 405.0},\n",
       "  4: {'A': 1.0, 'C': 1600.0, 'G': 11.0, 'T': 249.0},\n",
       "  5: {'A': 1709.0, 'C': 137.0, 'G': 59.0, 'T': 6.0},\n",
       "  6: {'A': 3.0, 'C': 1597.0, 'G': 1.0, 'T': 3.0},\n",
       "  7: {'A': 70.0, 'C': 37.0, 'G': 2940.0, 'T': 32.0},\n",
       "  8: {'A': 1.0, 'C': 1603.0, 'G': 4.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1605.0, 'G': 4.0, 'T': 7.0},\n",
       "  10: {'A': 1.0, 'C': 1610.0, 'G': 5.0, 'T': 66.0},\n",
       "  11: {'A': 717.0, 'C': 962.0, 'G': 2.0, 'T': 32.0},\n",
       "  12: {'A': 33.0, 'C': 1464.0, 'G': 16.0, 'T': 560.0},\n",
       "  13: {'A': 281.0, 'C': 595.0, 'G': 90.0, 'T': 1042.0},\n",
       "  14: {'A': 650.0, 'C': 742.0, 'G': 315.0, 'T': 1671.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16266294',\n",
       "  'uniprot_ids': 'Q8TD94',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0826.1',\n",
       "  'Matrix_Name': 'OLIG1',\n",
       "  1: {'A': 11392.0, 'C': 241.0, 'G': 1527.0, 'T': 206.0},\n",
       "  2: {'A': 5932.0, 'C': 4143.0, 'G': 1322.0, 'T': 1.0},\n",
       "  3: {'A': 53.0, 'C': 11392.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 11392.0, 'C': 2.0, 'G': 140.0, 'T': 81.0},\n",
       "  5: {'A': 13.0, 'C': 16.0, 'G': 114.0, 'T': 11392.0},\n",
       "  6: {'A': 11392.0, 'C': 81.0, 'G': 18.0, 'T': 11.0},\n",
       "  7: {'A': 196.0, 'C': 176.0, 'G': 1.0, 'T': 11392.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 11392.0, 'T': 75.0},\n",
       "  9: {'A': 13.0, 'C': 2083.0, 'G': 4569.0, 'T': 4731.0},\n",
       "  10: {'A': 491.0, 'C': 2813.0, 'G': 258.0, 'T': 11392.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20959288',\n",
       "  'uniprot_ids': 'Q8TAK6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0517.1',\n",
       "  'Matrix_Name': 'STAT1::STAT2',\n",
       "  1: {'A': 70.0, 'C': 64.0, 'G': 78.0, 'T': 412.0},\n",
       "  2: {'A': 151.0, 'C': 231.0, 'G': 85.0, 'T': 157.0},\n",
       "  3: {'A': 431.0, 'C': 8.0, 'G': 152.0, 'T': 33.0},\n",
       "  4: {'A': 4.0, 'C': 149.0, 'G': 468.0, 'T': 3.0},\n",
       "  5: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 619.0},\n",
       "  6: {'A': 4.0, 'C': 3.0, 'G': 1.0, 'T': 616.0},\n",
       "  7: {'A': 4.0, 'C': 14.0, 'G': 5.0, 'T': 601.0},\n",
       "  8: {'A': 1.0, 'C': 582.0, 'G': 1.0, 'T': 40.0},\n",
       "  9: {'A': 256.0, 'C': 76.0, 'G': 192.0, 'T': 100.0},\n",
       "  10: {'A': 108.0, 'C': 112.0, 'G': 162.0, 'T': 242.0},\n",
       "  11: {'A': 5.0, 'C': 14.0, 'G': 11.0, 'T': 594.0},\n",
       "  12: {'A': 11.0, 'C': 3.0, 'G': 1.0, 'T': 609.0},\n",
       "  13: {'A': 5.0, 'C': 171.0, 'G': 5.0, 'T': 443.0},\n",
       "  14: {'A': 16.0, 'C': 490.0, 'G': 10.0, 'T': 108.0},\n",
       "  15: {'A': 29.0, 'C': 366.0, 'G': 37.0, 'T': 192.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '16319195',\n",
       "  'uniprot_ids': 'P42224;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1495.1',\n",
       "  'Matrix_Name': 'HOXA1',\n",
       "  1: {'A': 1168.0, 'C': 2245.0, 'G': 2668.0, 'T': 1329.0},\n",
       "  2: {'A': 1.0, 'C': 1547.0, 'G': 1.0, 'T': 7407.0},\n",
       "  3: {'A': 7407.0, 'C': 2084.0, 'G': 1245.0, 'T': 1.0},\n",
       "  4: {'A': 7407.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7407.0},\n",
       "  6: {'A': 1.0, 'C': 1311.0, 'G': 2266.0, 'T': 7407.0},\n",
       "  7: {'A': 7407.0, 'C': 1.0, 'G': 1577.0, 'T': 1.0},\n",
       "  8: {'A': 1176.0, 'C': 2812.0, 'G': 2238.0, 'T': 1183.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '16341070',\n",
       "  'uniprot_ids': 'P49639',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0038.2',\n",
       "  'Matrix_Name': 'GFI1',\n",
       "  1: {'A': 3669.0, 'C': 11576.0, 'G': 7377.0, 'T': 7528.0},\n",
       "  2: {'A': 19147.0, 'C': 10156.0, 'G': 813.0, 'T': 566.0},\n",
       "  3: {'A': 30169.0, 'C': 3.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 30151.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 30162.0},\n",
       "  6: {'A': 16.0, 'C': 30140.0, 'G': 1.0, 'T': 11.0},\n",
       "  7: {'A': 29256.0, 'C': 1.0, 'G': 1.0, 'T': 1595.0},\n",
       "  8: {'A': 839.0, 'C': 27770.0, 'G': 3889.0, 'T': 1.0},\n",
       "  9: {'A': 7749.0, 'C': 55.0, 'G': 8350.0, 'T': 13997.0},\n",
       "  10: {'A': 2295.0, 'C': 267.0, 'G': 27515.0, 'T': 889.0},\n",
       "  11: {'A': 293.0, 'C': 27892.0, 'G': 1.0, 'T': 3703.0},\n",
       "  12: {'A': 16714.0, 'C': 5222.0, 'G': 4080.0, 'T': 4135.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '8754800',\n",
       "  'uniprot_ids': 'Q99684',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0141.3',\n",
       "  'Matrix_Name': 'ESRRB',\n",
       "  1: {'A': 109.0, 'C': 374.0, 'G': 76.0, 'T': 1747.0},\n",
       "  2: {'A': 10.0, 'C': 1747.0, 'G': 256.0, 'T': 8.0},\n",
       "  3: {'A': 1747.0, 'C': 2.0, 'G': 1.0, 'T': 5.0},\n",
       "  4: {'A': 1747.0, 'C': 2.0, 'G': 4.0, 'T': 1.0},\n",
       "  5: {'A': 4.0, 'C': 3.0, 'G': 1747.0, 'T': 2.0},\n",
       "  6: {'A': 2.0, 'C': 5.0, 'G': 1747.0, 'T': 4.0},\n",
       "  7: {'A': 7.0, 'C': 12.0, 'G': 9.0, 'T': 1747.0},\n",
       "  8: {'A': 3.0, 'C': 1747.0, 'G': 2.0, 'T': 16.0},\n",
       "  9: {'A': 1747.0, 'C': 1.0, 'G': 57.0, 'T': 4.0},\n",
       "  10: {'A': 703.0, 'C': 257.0, 'G': 43.0, 'T': 1046.0},\n",
       "  11: {'A': 693.0, 'C': 329.0, 'G': 190.0, 'T': 538.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '18555785',\n",
       "  'uniprot_ids': 'O95718',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1540.1',\n",
       "  'Matrix_Name': 'NR5A1',\n",
       "  1: {'A': 768.0, 'C': 992.0, 'G': 1269.0, 'T': 1094.0},\n",
       "  2: {'A': 907.0, 'C': 1691.0, 'G': 903.0, 'T': 2429.0},\n",
       "  3: {'A': 85.0, 'C': 4119.0, 'G': 717.0, 'T': 136.0},\n",
       "  4: {'A': 4119.0, 'C': 1.0, 'G': 288.0, 'T': 5.0},\n",
       "  5: {'A': 4119.0, 'C': 129.0, 'G': 138.0, 'T': 144.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 4119.0, 'T': 87.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 4119.0, 'T': 1.0},\n",
       "  8: {'A': 952.0, 'C': 2022.0, 'G': 382.0, 'T': 4119.0},\n",
       "  9: {'A': 71.0, 'C': 4119.0, 'G': 219.0, 'T': 652.0},\n",
       "  10: {'A': 4119.0, 'C': 243.0, 'G': 1685.0, 'T': 288.0},\n",
       "  11: {'A': 850.0, 'C': 1324.0, 'G': 821.0, 'T': 1127.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'FTZ-F1-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '10913126',\n",
       "  'uniprot_ids': 'Q13285',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0738.1',\n",
       "  'Matrix_Name': 'HIC2',\n",
       "  1: {'A': 2607.0, 'C': 395.0, 'G': 2286.0, 'T': 368.0},\n",
       "  2: {'A': 1.0, 'C': 118.0, 'G': 1.0, 'T': 4892.0},\n",
       "  3: {'A': 495.0, 'C': 355.0, 'G': 4892.0, 'T': 152.0},\n",
       "  4: {'A': 24.0, 'C': 4892.0, 'G': 15.0, 'T': 1.0},\n",
       "  5: {'A': 436.0, 'C': 4892.0, 'G': 1.0, 'T': 60.0},\n",
       "  6: {'A': 1467.0, 'C': 3426.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 2494.0, 'C': 711.0, 'G': 1291.0, 'T': 400.0},\n",
       "  8: {'A': 771.0, 'C': 2201.0, 'G': 1238.0, 'T': 685.0},\n",
       "  9: {'A': 761.0, 'C': 1733.0, 'G': 1002.0, 'T': 1399.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '15231840',\n",
       "  'uniprot_ids': 'Q96JB3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1104.2',\n",
       "  'Matrix_Name': 'GATA6',\n",
       "  1: {'A': 22321.0, 'C': 16230.0, 'G': 13433.0, 'T': 27464.0},\n",
       "  2: {'A': 20859.0, 'C': 14162.0, 'G': 11895.0, 'T': 32532.0},\n",
       "  3: {'A': 35361.0, 'C': 13348.0, 'G': 10395.0, 'T': 20344.0},\n",
       "  4: {'A': 5913.0, 'C': 11832.0, 'G': 7067.0, 'T': 54636.0},\n",
       "  5: {'A': 4536.0, 'C': 62937.0, 'G': 6460.0, 'T': 5515.0},\n",
       "  6: {'A': 2561.0, 'C': 1440.0, 'G': 581.0, 'T': 74866.0},\n",
       "  7: {'A': 5045.0, 'C': 1394.0, 'G': 616.0, 'T': 72393.0},\n",
       "  8: {'A': 76687.0, 'C': 816.0, 'G': 820.0, 'T': 1125.0},\n",
       "  9: {'A': 1508.0, 'C': 853.0, 'G': 457.0, 'T': 76630.0},\n",
       "  10: {'A': 1097.0, 'C': 75931.0, 'G': 713.0, 'T': 1707.0},\n",
       "  11: {'A': 13150.0, 'C': 3229.0, 'G': 1811.0, 'T': 61258.0},\n",
       "  12: {'A': 18912.0, 'C': 19055.0, 'G': 18154.0, 'T': 23327.0},\n",
       "  13: {'A': 22173.0, 'C': 17970.0, 'G': 11606.0, 'T': 27699.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '9915795',\n",
       "  'uniprot_ids': 'Q92908',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0778.1',\n",
       "  'Matrix_Name': 'NFKB2',\n",
       "  1: {'A': 5042.0, 'C': 2457.0, 'G': 2307.0, 'T': 1981.0},\n",
       "  2: {'A': 102.0, 'C': 98.0, 'G': 16292.0, 'T': 5.0},\n",
       "  3: {'A': 2.0, 'C': 2.0, 'G': 16042.0, 'T': 4.0},\n",
       "  4: {'A': 9.0, 'C': 6.0, 'G': 16056.0, 'T': 566.0},\n",
       "  5: {'A': 2664.0, 'C': 12.0, 'G': 13130.0, 'T': 988.0},\n",
       "  6: {'A': 8381.0, 'C': 978.0, 'G': 857.0, 'T': 637.0},\n",
       "  7: {'A': 5183.0, 'C': 61.0, 'G': 52.0, 'T': 5971.0},\n",
       "  8: {'A': 1174.0, 'C': 1396.0, 'G': 1078.0, 'T': 10977.0},\n",
       "  9: {'A': 853.0, 'C': 11012.0, 'G': 20.0, 'T': 1868.0},\n",
       "  10: {'A': 257.0, 'C': 12674.0, 'G': 2.0, 'T': 45.0},\n",
       "  11: {'A': 3.0, 'C': 12711.0, 'G': 2.0, 'T': 3.0},\n",
       "  12: {'A': 9.0, 'C': 12889.0, 'G': 4.0, 'T': 24.0},\n",
       "  13: {'A': 1528.0, 'C': 3325.0, 'G': 2183.0, 'T': 5986.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'NF-kappaB-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '20066093',\n",
       "  'uniprot_ids': 'Q00653',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1146.1',\n",
       "  'Matrix_Name': 'NR1H4::RXRA',\n",
       "  1: {'A': 224.0, 'C': 173.0, 'G': 452.0, 'T': 155.0},\n",
       "  2: {'A': 654.0, 'C': 56.0, 'G': 281.0, 'T': 12.0},\n",
       "  3: {'A': 30.0, 'C': 16.0, 'G': 938.0, 'T': 20.0},\n",
       "  4: {'A': 23.0, 'C': 10.0, 'G': 647.0, 'T': 325.0},\n",
       "  5: {'A': 25.0, 'C': 50.0, 'G': 102.0, 'T': 827.0},\n",
       "  6: {'A': 12.0, 'C': 876.0, 'G': 14.0, 'T': 101.0},\n",
       "  7: {'A': 721.0, 'C': 14.0, 'G': 250.0, 'T': 18.0},\n",
       "  8: {'A': 145.0, 'C': 122.0, 'G': 76.0, 'T': 661.0},\n",
       "  9: {'A': 24.0, 'C': 93.0, 'G': 49.0, 'T': 839.0},\n",
       "  10: {'A': 109.0, 'C': 103.0, 'G': 754.0, 'T': 38.0},\n",
       "  11: {'A': 764.0, 'C': 113.0, 'G': 79.0, 'T': 49.0},\n",
       "  12: {'A': 135.0, 'C': 774.0, 'G': 10.0, 'T': 84.0},\n",
       "  13: {'A': 114.0, 'C': 799.0, 'G': 39.0, 'T': 52.0},\n",
       "  14: {'A': 26.0, 'C': 384.0, 'G': 72.0, 'T': 522.0},\n",
       "  15: {'A': 96.0, 'C': 314.0, 'G': 213.0, 'T': 381.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '10744760',\n",
       "  'uniprot_ids': 'Q96RI1;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0887.1',\n",
       "  'Matrix_Name': 'EVX1',\n",
       "  1: {'A': 1960.0, 'C': 1807.0, 'G': 3093.0, 'T': 1038.0},\n",
       "  2: {'A': 2046.0, 'C': 2507.0, 'G': 2630.0, 'T': 715.0},\n",
       "  3: {'A': 1197.0, 'C': 2344.0, 'G': 270.0, 'T': 7895.0},\n",
       "  4: {'A': 3900.0, 'C': 1753.0, 'G': 1886.0, 'T': 359.0},\n",
       "  5: {'A': 7895.0, 'C': 30.0, 'G': 237.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 68.0, 'G': 268.0, 'T': 7895.0},\n",
       "  7: {'A': 233.0, 'C': 464.0, 'G': 172.0, 'T': 7895.0},\n",
       "  8: {'A': 7895.0, 'C': 15.0, 'G': 726.0, 'T': 173.0},\n",
       "  9: {'A': 1334.0, 'C': 2099.0, 'G': 2721.0, 'T': 1744.0},\n",
       "  10: {'A': 1433.0, 'C': 3319.0, 'G': 1959.0, 'T': 1187.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P49640',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0894.1',\n",
       "  'Matrix_Name': 'HESX1',\n",
       "  1: {'A': 832.0, 'C': 754.0, 'G': 1744.0, 'T': 745.0},\n",
       "  2: {'A': 880.0, 'C': 1433.0, 'G': 448.0, 'T': 1314.0},\n",
       "  3: {'A': 188.0, 'C': 1.0, 'G': 35.0, 'T': 4073.0},\n",
       "  4: {'A': 4073.0, 'C': 1.0, 'G': 54.0, 'T': 32.0},\n",
       "  5: {'A': 4073.0, 'C': 8.0, 'G': 10.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 16.0, 'T': 4073.0},\n",
       "  7: {'A': 1.0, 'C': 16.0, 'G': 1.0, 'T': 4073.0},\n",
       "  8: {'A': 1893.0, 'C': 39.0, 'G': 2181.0, 'T': 98.0},\n",
       "  9: {'A': 1246.0, 'C': 773.0, 'G': 1682.0, 'T': 374.0},\n",
       "  10: {'A': 802.0, 'C': 1602.0, 'G': 462.0, 'T': 1210.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '11748154',\n",
       "  'uniprot_ids': 'Q9UBX0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1620.1',\n",
       "  'Matrix_Name': 'Ptf1a(var.3)',\n",
       "  1: {'A': 2835.0, 'C': 2470.0, 'G': 2665.0, 'T': 2325.0},\n",
       "  2: {'A': 2614.0, 'C': 3701.0, 'G': 2380.0, 'T': 1600.0},\n",
       "  3: {'A': 6868.0, 'C': 1189.0, 'G': 1411.0, 'T': 827.0},\n",
       "  4: {'A': 185.0, 'C': 9888.0, 'G': 114.0, 'T': 108.0},\n",
       "  5: {'A': 10034.0, 'C': 61.0, 'G': 85.0, 'T': 115.0},\n",
       "  6: {'A': 117.0, 'C': 8568.0, 'G': 1183.0, 'T': 427.0},\n",
       "  7: {'A': 257.0, 'C': 9613.0, 'G': 295.0, 'T': 130.0},\n",
       "  8: {'A': 86.0, 'C': 95.0, 'G': 99.0, 'T': 10015.0},\n",
       "  9: {'A': 69.0, 'C': 110.0, 'G': 9973.0, 'T': 143.0},\n",
       "  10: {'A': 528.0, 'C': 1616.0, 'G': 1517.0, 'T': 6634.0},\n",
       "  11: {'A': 1679.0, 'C': 2528.0, 'G': 3342.0, 'T': 2746.0},\n",
       "  12: {'A': 2294.0, 'C': 3005.0, 'G': 2184.0, 'T': 2812.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23754747',\n",
       "  'uniprot_ids': 'Q9QX98',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1581.1',\n",
       "  'Matrix_Name': 'ZBTB6',\n",
       "  1: {'A': 594.0, 'C': 318.0, 'G': 714.0, 'T': 605.0},\n",
       "  2: {'A': 399.0, 'C': 624.0, 'G': 482.0, 'T': 726.0},\n",
       "  3: {'A': 313.0, 'C': 1158.0, 'G': 594.0, 'T': 166.0},\n",
       "  4: {'A': 95.0, 'C': 2081.0, 'G': 26.0, 'T': 29.0},\n",
       "  5: {'A': 30.0, 'C': 105.0, 'G': 53.0, 'T': 2043.0},\n",
       "  6: {'A': 208.0, 'C': 46.0, 'G': 410.0, 'T': 1567.0},\n",
       "  7: {'A': 14.0, 'C': 60.0, 'G': 2149.0, 'T': 8.0},\n",
       "  8: {'A': 2159.0, 'C': 29.0, 'G': 21.0, 'T': 22.0},\n",
       "  9: {'A': 59.0, 'C': 21.0, 'G': 2136.0, 'T': 15.0},\n",
       "  10: {'A': 57.0, 'C': 2088.0, 'G': 39.0, 'T': 47.0},\n",
       "  11: {'A': 103.0, 'C': 1815.0, 'G': 89.0, 'T': 224.0},\n",
       "  12: {'A': 590.0, 'C': 708.0, 'G': 468.0, 'T': 465.0},\n",
       "  13: {'A': 542.0, 'C': 594.0, 'G': 750.0, 'T': 345.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '16381825',\n",
       "  'uniprot_ids': 'Q15916',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0092.1',\n",
       "  'Matrix_Name': 'Hand1::Tcf3',\n",
       "  1: {'A': 5.0, 'C': 9.0, 'G': 11.0, 'T': 8.0},\n",
       "  2: {'A': 11.0, 'C': 1.0, 'G': 16.0, 'T': 5.0},\n",
       "  3: {'A': 3.0, 'C': 3.0, 'G': 2.0, 'T': 25.0},\n",
       "  4: {'A': 1.0, 'C': 29.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 30.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 30.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 4.0, 'G': 26.0, 'T': 2.0},\n",
       "  8: {'A': 10.0, 'C': 15.0, 'G': 2.0, 'T': 6.0},\n",
       "  9: {'A': 17.0, 'C': 1.0, 'G': 4.0, 'T': 11.0},\n",
       "  10: {'A': 6.0, 'C': 5.0, 'G': 5.0, 'T': 17.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '7791788',\n",
       "  'uniprot_ids': 'Q64279;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0823.1',\n",
       "  'Matrix_Name': 'HEY1',\n",
       "  1: {'A': 438.0, 'C': 762.0, 'G': 1789.0, 'T': 353.0},\n",
       "  2: {'A': 1374.0, 'C': 657.0, 'G': 1200.0, 'T': 111.0},\n",
       "  3: {'A': 2.0, 'C': 3339.0, 'G': 4.0, 'T': 16.0},\n",
       "  4: {'A': 3339.0, 'C': 31.0, 'G': 124.0, 'T': 22.0},\n",
       "  5: {'A': 1.0, 'C': 3339.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 19.0, 'G': 3339.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 475.0, 'G': 11.0, 'T': 3339.0},\n",
       "  8: {'A': 33.0, 'C': 20.0, 'G': 3339.0, 'T': 23.0},\n",
       "  9: {'A': 334.0, 'C': 1763.0, 'G': 381.0, 'T': 864.0},\n",
       "  10: {'A': 476.0, 'C': 2173.0, 'G': 374.0, 'T': 319.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22615585',\n",
       "  'uniprot_ids': 'Q9Y5J3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0731.1',\n",
       "  'Matrix_Name': 'BCL6B',\n",
       "  1: {'A': 1.0, 'C': 20.0, 'G': 1.0, 'T': 208.0},\n",
       "  2: {'A': 4.0, 'C': 1.0, 'G': 187.0, 'T': 2.0},\n",
       "  3: {'A': 1.0, 'C': 368.0, 'G': 1.0, 'T': 3.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 213.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 214.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 3.0, 'T': 213.0},\n",
       "  7: {'A': 1.0, 'C': 385.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 13.0, 'G': 35.0, 'T': 215.0},\n",
       "  9: {'A': 281.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 174.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 172.0, 'T': 1.0},\n",
       "  12: {'A': 290.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  13: {'A': 291.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 216.0},\n",
       "  15: {'A': 1.0, 'C': 17.0, 'G': 1.0, 'T': 212.0},\n",
       "  16: {'A': 114.0, 'C': 244.0, 'G': 10.0, 'T': 1.0},\n",
       "  17: {'A': 184.0, 'C': 147.0, 'G': 8.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11374866',\n",
       "  'uniprot_ids': 'A8KA13',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0058.3',\n",
       "  'Matrix_Name': 'MAX',\n",
       "  1: {'A': 3308.0, 'C': 1557.0, 'G': 1296.0, 'T': 720.0},\n",
       "  2: {'A': 1916.0, 'C': 2810.0, 'G': 1515.0, 'T': 638.0},\n",
       "  3: {'A': 40.0, 'C': 6877.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 6877.0, 'C': 1.0, 'G': 88.0, 'T': 39.0},\n",
       "  5: {'A': 1.0, 'C': 6877.0, 'G': 8.0, 'T': 230.0},\n",
       "  6: {'A': 330.0, 'C': 40.0, 'G': 6877.0, 'T': 20.0},\n",
       "  7: {'A': 70.0, 'C': 160.0, 'G': 21.0, 'T': 6877.0},\n",
       "  8: {'A': 1.0, 'C': 28.0, 'G': 6877.0, 'T': 37.0},\n",
       "  9: {'A': 1342.0, 'C': 2451.0, 'G': 1639.0, 'T': 1447.0},\n",
       "  10: {'A': 1123.0, 'C': 2226.0, 'G': 987.0, 'T': 2544.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22955619',\n",
       "  'uniprot_ids': 'P61244',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0650.2',\n",
       "  'Matrix_Name': 'HOXA13',\n",
       "  1: {'A': 295.0, 'C': 352.0, 'G': 291.0, 'T': 19.0},\n",
       "  2: {'A': 64.0, 'C': 955.0, 'G': 255.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 955.0, 'G': 1.0, 'T': 55.0},\n",
       "  4: {'A': 955.0, 'C': 395.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 955.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 955.0},\n",
       "  7: {'A': 955.0, 'C': 1.0, 'G': 18.0, 'T': 1.0},\n",
       "  8: {'A': 955.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  9: {'A': 955.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 955.0, 'C': 246.0, 'G': 37.0, 'T': 83.0},\n",
       "  11: {'A': 197.0, 'C': 759.0, 'G': 103.0, 'T': 195.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '17200107',\n",
       "  'uniprot_ids': 'P31271',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1624.1',\n",
       "  'Matrix_Name': 'Stat5a',\n",
       "  1: {'A': 1329.0, 'C': 852.0, 'G': 1005.0, 'T': 1332.0},\n",
       "  2: {'A': 315.0, 'C': 373.0, 'G': 208.0, 'T': 3622.0},\n",
       "  3: {'A': 152.0, 'C': 90.0, 'G': 145.0, 'T': 4131.0},\n",
       "  4: {'A': 67.0, 'C': 4382.0, 'G': 27.0, 'T': 42.0},\n",
       "  5: {'A': 72.0, 'C': 4047.0, 'G': 29.0, 'T': 370.0},\n",
       "  6: {'A': 3192.0, 'C': 589.0, 'G': 138.0, 'T': 599.0},\n",
       "  7: {'A': 3352.0, 'C': 81.0, 'G': 981.0, 'T': 104.0},\n",
       "  8: {'A': 20.0, 'C': 23.0, 'G': 4442.0, 'T': 33.0},\n",
       "  9: {'A': 4338.0, 'C': 42.0, 'G': 50.0, 'T': 88.0},\n",
       "  10: {'A': 4275.0, 'C': 67.0, 'G': 102.0, 'T': 74.0},\n",
       "  11: {'A': 1433.0, 'C': 1029.0, 'G': 989.0, 'T': 1067.0},\n",
       "  12: {'A': 1161.0, 'C': 1218.0, 'G': 688.0, 'T': 1451.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '10594041',\n",
       "  'uniprot_ids': 'P42230',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0722.1',\n",
       "  'Matrix_Name': 'VAX1',\n",
       "  1: {'A': 448.0, 'C': 2776.0, 'G': 456.0, 'T': 2480.0},\n",
       "  2: {'A': 680.0, 'C': 996.0, 'G': 102.0, 'T': 5254.0},\n",
       "  3: {'A': 5254.0, 'C': 864.0, 'G': 190.0, 'T': 303.0},\n",
       "  4: {'A': 5254.0, 'C': 154.0, 'G': 47.0, 'T': 134.0},\n",
       "  5: {'A': 298.0, 'C': 87.0, 'G': 214.0, 'T': 5254.0},\n",
       "  6: {'A': 509.0, 'C': 322.0, 'G': 1952.0, 'T': 5254.0},\n",
       "  7: {'A': 5254.0, 'C': 88.0, 'G': 1333.0, 'T': 399.0},\n",
       "  8: {'A': 978.0, 'C': 1912.0, 'G': 934.0, 'T': 1434.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q5SQQ9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0882.1',\n",
       "  'Matrix_Name': 'DLX6',\n",
       "  1: {'A': 603.0, 'C': 926.0, 'G': 681.0, 'T': 416.0},\n",
       "  2: {'A': 138.0, 'C': 1411.0, 'G': 22.0, 'T': 1213.0},\n",
       "  3: {'A': 2624.0, 'C': 274.0, 'G': 144.0, 'T': 185.0},\n",
       "  4: {'A': 2624.0, 'C': 125.0, 'G': 1.0, 'T': 57.0},\n",
       "  5: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 2624.0},\n",
       "  6: {'A': 49.0, 'C': 141.0, 'G': 69.0, 'T': 2624.0},\n",
       "  7: {'A': 2624.0, 'C': 31.0, 'G': 17.0, 'T': 194.0},\n",
       "  8: {'A': 704.0, 'C': 824.0, 'G': 474.0, 'T': 625.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9889271',\n",
       "  'uniprot_ids': 'P56179',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1115.1',\n",
       "  'Matrix_Name': 'POU5F1',\n",
       "  1: {'A': 4983.0, 'C': 1291.0, 'G': 1997.0, 'T': 4847.0},\n",
       "  2: {'A': 3604.0, 'C': 2786.0, 'G': 1909.0, 'T': 4819.0},\n",
       "  3: {'A': 12694.0, 'C': 123.0, 'G': 230.0, 'T': 71.0},\n",
       "  4: {'A': 84.0, 'C': 131.0, 'G': 68.0, 'T': 12835.0},\n",
       "  5: {'A': 718.0, 'C': 193.0, 'G': 11572.0, 'T': 635.0},\n",
       "  6: {'A': 534.0, 'C': 10311.0, 'G': 549.0, 'T': 1724.0},\n",
       "  7: {'A': 11757.0, 'C': 133.0, 'G': 119.0, 'T': 1109.0},\n",
       "  8: {'A': 10561.0, 'C': 691.0, 'G': 1027.0, 'T': 839.0},\n",
       "  9: {'A': 12281.0, 'C': 271.0, 'G': 258.0, 'T': 308.0},\n",
       "  10: {'A': 2557.0, 'C': 2105.0, 'G': 2072.0, 'T': 6384.0},\n",
       "  11: {'A': 3477.0, 'C': 2451.0, 'G': 3552.0, 'T': 3638.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23747866',\n",
       "  'uniprot_ids': 'Q01860',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0105.4',\n",
       "  'Matrix_Name': 'NFKB1',\n",
       "  1: {'A': 8442.0, 'C': 2014.0, 'G': 991.0, 'T': 819.0},\n",
       "  2: {'A': 10.0, 'C': 1.0, 'G': 19048.0, 'T': 6.0},\n",
       "  3: {'A': 1.0, 'C': 4.0, 'G': 19178.0, 'T': 3.0},\n",
       "  4: {'A': 5.0, 'C': 1.0, 'G': 19168.0, 'T': 319.0},\n",
       "  5: {'A': 230.0, 'C': 4.0, 'G': 19459.0, 'T': 28.0},\n",
       "  6: {'A': 12005.0, 'C': 63.0, 'G': 822.0, 'T': 28.0},\n",
       "  7: {'A': 7004.0, 'C': 19.0, 'G': 33.0, 'T': 6745.0},\n",
       "  8: {'A': 23.0, 'C': 1312.0, 'G': 117.0, 'T': 14373.0},\n",
       "  9: {'A': 18.0, 'C': 9575.0, 'G': 5.0, 'T': 131.0},\n",
       "  10: {'A': 498.0, 'C': 9537.0, 'G': 2.0, 'T': 4.0},\n",
       "  11: {'A': 5.0, 'C': 9592.0, 'G': 2.0, 'T': 3.0},\n",
       "  12: {'A': 7.0, 'C': 9643.0, 'G': 3.0, 'T': 9.0},\n",
       "  13: {'A': 1232.0, 'C': 1294.0, 'G': 3349.0, 'T': 9387.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'NF-kappaB-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '20066093',\n",
       "  'uniprot_ids': 'P19838',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1561.1',\n",
       "  'Matrix_Name': 'SOX12',\n",
       "  1: {'A': 855.0, 'C': 268.0, 'G': 110.0, 'T': 139.0},\n",
       "  2: {'A': 154.0, 'C': 855.0, 'G': 59.0, 'T': 79.0},\n",
       "  3: {'A': 49.0, 'C': 855.0, 'G': 35.0, 'T': 57.0},\n",
       "  4: {'A': 48.0, 'C': 27.0, 'G': 855.0, 'T': 49.0},\n",
       "  5: {'A': 855.0, 'C': 27.0, 'G': 3.0, 'T': 2.0},\n",
       "  6: {'A': 855.0, 'C': 3.0, 'G': 5.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 855.0, 'G': 1.0, 'T': 4.0},\n",
       "  8: {'A': 855.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 855.0, 'C': 92.0, 'G': 45.0, 'T': 21.0},\n",
       "  10: {'A': 130.0, 'C': 212.0, 'G': 58.0, 'T': 855.0},\n",
       "  11: {'A': 199.0, 'C': 211.0, 'G': 306.0, 'T': 142.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '18403418',\n",
       "  'uniprot_ids': 'O15370',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0652.1',\n",
       "  'Matrix_Name': 'IRF8',\n",
       "  1: {'A': 1310.0, 'C': 1997.0, 'G': 572.0, 'T': 2003.0},\n",
       "  2: {'A': 64.0, 'C': 5879.0, 'G': 20.0, 'T': 885.0},\n",
       "  3: {'A': 4.0, 'C': 1.0, 'G': 5879.0, 'T': 3.0},\n",
       "  4: {'A': 5879.0, 'C': 3.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 5879.0, 'C': 1.0, 'G': 1.0, 'T': 34.0},\n",
       "  6: {'A': 5879.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  7: {'A': 55.0, 'C': 5879.0, 'G': 581.0, 'T': 7.0},\n",
       "  8: {'A': 7.0, 'C': 5879.0, 'G': 1.0, 'T': 1229.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 5879.0, 'T': 1.0},\n",
       "  10: {'A': 5879.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  11: {'A': 5879.0, 'C': 1.0, 'G': 3.0, 'T': 69.0},\n",
       "  12: {'A': 5879.0, 'C': 6.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 12.0, 'C': 5879.0, 'G': 1062.0, 'T': 16.0},\n",
       "  14: {'A': 322.0, 'C': 1455.0, 'G': 9.0, 'T': 5879.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '21731497',\n",
       "  'uniprot_ids': 'Q02556',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1519.1',\n",
       "  'Matrix_Name': 'LHX5',\n",
       "  1: {'A': 395.0, 'C': 3421.0, 'G': 3722.0, 'T': 1216.0},\n",
       "  2: {'A': 1.0, 'C': 2094.0, 'G': 1.0, 'T': 5050.0},\n",
       "  3: {'A': 7143.0, 'C': 188.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 7143.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7143.0},\n",
       "  6: {'A': 1.0, 'C': 145.0, 'G': 29.0, 'T': 7143.0},\n",
       "  7: {'A': 7143.0, 'C': 1.0, 'G': 161.0, 'T': 1.0},\n",
       "  8: {'A': 2750.0, 'C': 1577.0, 'G': 1484.0, 'T': 1334.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9H2C1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0103.3',\n",
       "  'Matrix_Name': 'ZEB1',\n",
       "  1: {'A': 3212.0, 'C': 7999.0, 'G': 5229.0, 'T': 3693.0},\n",
       "  2: {'A': 4686.0, 'C': 5551.0, 'G': 5180.0, 'T': 4716.0},\n",
       "  3: {'A': 115.0, 'C': 19196.0, 'G': 159.0, 'T': 663.0},\n",
       "  4: {'A': 19906.0, 'C': 1.0, 'G': 224.0, 'T': 2.0},\n",
       "  5: {'A': 33.0, 'C': 19879.0, 'G': 98.0, 'T': 123.0},\n",
       "  6: {'A': 194.0, 'C': 19258.0, 'G': 650.0, 'T': 31.0},\n",
       "  7: {'A': 5.0, 'C': 502.0, 'G': 356.0, 'T': 19270.0},\n",
       "  8: {'A': 442.0, 'C': 179.0, 'G': 19382.0, 'T': 130.0},\n",
       "  9: {'A': 1983.0, 'C': 9562.0, 'G': 5179.0, 'T': 3409.0},\n",
       "  10: {'A': 4259.0, 'C': 6118.0, 'G': 6602.0, 'T': 3154.0},\n",
       "  11: {'A': 3491.0, 'C': 6956.0, 'G': 5829.0, 'T': 3857.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HD-ZF',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8065305',\n",
       "  'uniprot_ids': 'P37275',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0697.1',\n",
       "  'Matrix_Name': 'ZIC3',\n",
       "  1: {'A': 21.0, 'C': 51.0, 'G': 164.0, 'T': 70.0},\n",
       "  2: {'A': 164.0, 'C': 33.0, 'G': 44.0, 'T': 1.0},\n",
       "  3: {'A': 8.0, 'C': 204.0, 'G': 1.0, 'T': 5.0},\n",
       "  4: {'A': 8.0, 'C': 205.0, 'G': 4.0, 'T': 8.0},\n",
       "  5: {'A': 23.0, 'C': 189.0, 'G': 1.0, 'T': 20.0},\n",
       "  6: {'A': 1.0, 'C': 210.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 207.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 143.0, 'G': 1.0, 'T': 62.0},\n",
       "  9: {'A': 23.0, 'C': 6.0, 'G': 218.0, 'T': 3.0},\n",
       "  10: {'A': 5.0, 'C': 179.0, 'G': 7.0, 'T': 30.0},\n",
       "  11: {'A': 19.0, 'C': 27.0, 'G': 54.0, 'T': 164.0},\n",
       "  12: {'A': 4.0, 'C': 1.0, 'G': 226.0, 'T': 4.0},\n",
       "  13: {'A': 25.0, 'C': 90.0, 'G': 14.0, 'T': 75.0},\n",
       "  14: {'A': 4.0, 'C': 3.0, 'G': 215.0, 'T': 24.0},\n",
       "  15: {'A': 57.0, 'C': 80.0, 'G': 14.0, 'T': 54.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'O60481',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0470.2',\n",
       "  'Matrix_Name': 'E2F4',\n",
       "  1: {'A': 4333.0, 'C': 1037.0, 'G': 1773.0, 'T': 9142.0},\n",
       "  2: {'A': 3441.0, 'C': 927.0, 'G': 969.0, 'T': 12852.0},\n",
       "  3: {'A': 2542.0, 'C': 477.0, 'G': 3028.0, 'T': 12537.0},\n",
       "  4: {'A': 947.0, 'C': 490.0, 'G': 1562.0, 'T': 13461.0},\n",
       "  5: {'A': 319.0, 'C': 2177.0, 'G': 13672.0, 'T': 21.0},\n",
       "  6: {'A': 11.0, 'C': 51.0, 'G': 13940.0, 'T': 1.0},\n",
       "  7: {'A': 16.0, 'C': 13955.0, 'G': 3.0, 'T': 12.0},\n",
       "  8: {'A': 10.0, 'C': 8.0, 'G': 13951.0, 'T': 26.0},\n",
       "  9: {'A': 1.0, 'C': 13940.0, 'G': 54.0, 'T': 1.0},\n",
       "  10: {'A': 30.0, 'C': 13731.0, 'G': 3994.0, 'T': 578.0},\n",
       "  11: {'A': 12270.0, 'C': 4541.0, 'G': 1063.0, 'T': 2505.0},\n",
       "  12: {'A': 4887.0, 'C': 2002.0, 'G': 525.0, 'T': 8280.0},\n",
       "  13: {'A': 4628.0, 'C': 907.0, 'G': 847.0, 'T': 8793.0},\n",
       "  14: {'A': 5355.0, 'C': 2864.0, 'G': 2398.0, 'T': 6867.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '18836037',\n",
       "  'uniprot_ids': 'Q16254',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0465.2',\n",
       "  'Matrix_Name': 'CDX2',\n",
       "  1: {'A': 11370.0, 'C': 6381.0, 'G': 7435.0, 'T': 6844.0},\n",
       "  2: {'A': 11736.0, 'C': 2569.0, 'G': 8817.0, 'T': 8908.0},\n",
       "  3: {'A': 3158.0, 'C': 2097.0, 'G': 24215.0, 'T': 2560.0},\n",
       "  4: {'A': 1227.0, 'C': 26588.0, 'G': 608.0, 'T': 3607.0},\n",
       "  5: {'A': 26721.0, 'C': 3313.0, 'G': 185.0, 'T': 1811.0},\n",
       "  6: {'A': 29958.0, 'C': 415.0, 'G': 1197.0, 'T': 460.0},\n",
       "  7: {'A': 275.0, 'C': 392.0, 'G': 387.0, 'T': 30976.0},\n",
       "  8: {'A': 29222.0, 'C': 762.0, 'G': 1000.0, 'T': 1046.0},\n",
       "  9: {'A': 30646.0, 'C': 350.0, 'G': 367.0, 'T': 667.0},\n",
       "  10: {'A': 30126.0, 'C': 596.0, 'G': 503.0, 'T': 805.0},\n",
       "  11: {'A': 13753.0, 'C': 5300.0, 'G': 5885.0, 'T': 7092.0},\n",
       "  12: {'A': 10201.0, 'C': 6906.0, 'G': 5935.0, 'T': 8988.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '20696899',\n",
       "  'uniprot_ids': 'Q99626',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1467.1',\n",
       "  'Matrix_Name': 'ATOH1(var.2)',\n",
       "  1: {'A': 1369.0, 'C': 172.0, 'G': 467.0, 'T': 1.0},\n",
       "  2: {'A': 1009.0, 'C': 180.0, 'G': 183.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1369.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1369.0, 'C': 1.0, 'G': 1.0, 'T': 77.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1369.0, 'T': 182.0},\n",
       "  6: {'A': 205.0, 'C': 1369.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1369.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1369.0, 'T': 1.0},\n",
       "  9: {'A': 94.0, 'C': 297.0, 'G': 36.0, 'T': 945.0},\n",
       "  10: {'A': 1.0, 'C': 944.0, 'G': 1.0, 'T': 427.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22985730',\n",
       "  'uniprot_ids': 'Q92858',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1608.1',\n",
       "  'Matrix_Name': 'Isl1',\n",
       "  1: {'A': 1833.0, 'C': 1058.0, 'G': 1409.0, 'T': 1341.0},\n",
       "  2: {'A': 1077.0, 'C': 1296.0, 'G': 1474.0, 'T': 1794.0},\n",
       "  3: {'A': 357.0, 'C': 4711.0, 'G': 193.0, 'T': 380.0},\n",
       "  4: {'A': 433.0, 'C': 4886.0, 'G': 55.0, 'T': 267.0},\n",
       "  5: {'A': 5477.0, 'C': 31.0, 'G': 100.0, 'T': 33.0},\n",
       "  6: {'A': 68.0, 'C': 55.0, 'G': 35.0, 'T': 5483.0},\n",
       "  7: {'A': 65.0, 'C': 105.0, 'G': 40.0, 'T': 5431.0},\n",
       "  8: {'A': 5463.0, 'C': 46.0, 'G': 21.0, 'T': 111.0},\n",
       "  9: {'A': 1090.0, 'C': 411.0, 'G': 3703.0, 'T': 437.0},\n",
       "  10: {'A': 1090.0, 'C': 1746.0, 'G': 1319.0, 'T': 1486.0},\n",
       "  11: {'A': 1489.0, 'C': 1473.0, 'G': 1393.0, 'T': 1286.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P61372',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0518.1',\n",
       "  'Matrix_Name': 'Stat4',\n",
       "  1: {'A': 212.0, 'C': 1020.0, 'G': 485.0, 'T': 1160.0},\n",
       "  2: {'A': 13.0, 'C': 3.0, 'G': 1.0, 'T': 2860.0},\n",
       "  3: {'A': 27.0, 'C': 1.0, 'G': 530.0, 'T': 2319.0},\n",
       "  4: {'A': 633.0, 'C': 2230.0, 'G': 13.0, 'T': 1.0},\n",
       "  5: {'A': 308.0, 'C': 1492.0, 'G': 73.0, 'T': 1004.0},\n",
       "  6: {'A': 1451.0, 'C': 29.0, 'G': 1224.0, 'T': 173.0},\n",
       "  7: {'A': 734.0, 'C': 1.0, 'G': 2135.0, 'T': 7.0},\n",
       "  8: {'A': 46.0, 'C': 1.0, 'G': 2829.0, 'T': 1.0},\n",
       "  9: {'A': 2874.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 2874.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1574.0, 'C': 280.0, 'G': 834.0, 'T': 189.0},\n",
       "  12: {'A': 487.0, 'C': 786.0, 'G': 686.0, 'T': 918.0},\n",
       "  13: {'A': 633.0, 'C': 722.0, 'G': 1029.0, 'T': 493.0},\n",
       "  14: {'A': 890.0, 'C': 444.0, 'G': 1118.0, 'T': 425.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '19710469',\n",
       "  'uniprot_ids': 'P42228',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0687.1',\n",
       "  'Matrix_Name': 'SPIC',\n",
       "  1: {'A': 145.0, 'C': 84.0, 'G': 69.0, 'T': 73.0},\n",
       "  2: {'A': 163.0, 'C': 32.0, 'G': 71.0, 'T': 41.0},\n",
       "  3: {'A': 187.0, 'C': 26.0, 'G': 14.0, 'T': 19.0},\n",
       "  4: {'A': 195.0, 'C': 10.0, 'G': 1.0, 'T': 5.0},\n",
       "  5: {'A': 172.0, 'C': 1.0, 'G': 1.0, 'T': 76.0},\n",
       "  6: {'A': 31.0, 'C': 76.0, 'G': 182.0, 'T': 14.0},\n",
       "  7: {'A': 89.0, 'C': 60.0, 'G': 49.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 9.0, 'G': 218.0, 'T': 4.0},\n",
       "  9: {'A': 14.0, 'C': 1.0, 'G': 233.0, 'T': 27.0},\n",
       "  10: {'A': 205.0, 'C': 8.0, 'G': 1.0, 'T': 17.0},\n",
       "  11: {'A': 193.0, 'C': 32.0, 'G': 14.0, 'T': 9.0},\n",
       "  12: {'A': 14.0, 'C': 1.0, 'G': 232.0, 'T': 1.0},\n",
       "  13: {'A': 46.0, 'C': 25.0, 'G': 18.0, 'T': 195.0},\n",
       "  14: {'A': 116.0, 'C': 37.0, 'G': 10.0, 'T': 20.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '16887979',\n",
       "  'uniprot_ids': 'Q8N5J4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1545.1',\n",
       "  'Matrix_Name': 'OVOL2',\n",
       "  1: {'A': 4026.0, 'C': 1034.0, 'G': 4557.0, 'T': 2878.0},\n",
       "  2: {'A': 5376.0, 'C': 1850.0, 'G': 4569.0, 'T': 7117.0},\n",
       "  3: {'A': 8886.0, 'C': 1603.0, 'G': 451.0, 'T': 3608.0},\n",
       "  4: {'A': 4.0, 'C': 12492.0, 'G': 25.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 12492.0, 'G': 42.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 12492.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1403.0, 'G': 1.0, 'T': 12492.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 10.0, 'T': 12492.0},\n",
       "  9: {'A': 12492.0, 'C': 14.0, 'G': 152.0, 'T': 1.0},\n",
       "  10: {'A': 1860.0, 'C': 5645.0, 'G': 3449.0, 'T': 6849.0},\n",
       "  11: {'A': 3244.0, 'C': 2608.0, 'G': 3396.0, 'T': 3247.0},\n",
       "  12: {'A': 2783.0, 'C': 5866.0, 'G': 1794.0, 'T': 6627.0},\n",
       "  13: {'A': 2674.0, 'C': 3145.0, 'G': 3873.0, 'T': 2803.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '24735879',\n",
       "  'uniprot_ids': 'Q9BRP0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1628.1',\n",
       "  'Matrix_Name': 'Zic1::Zic2',\n",
       "  1: {'A': 2073.0, 'C': 4621.0, 'G': 2207.0, 'T': 995.0},\n",
       "  2: {'A': 3058.0, 'C': 2590.0, 'G': 2774.0, 'T': 1474.0},\n",
       "  3: {'A': 30.0, 'C': 9595.0, 'G': 107.0, 'T': 164.0},\n",
       "  4: {'A': 9239.0, 'C': 385.0, 'G': 267.0, 'T': 5.0},\n",
       "  5: {'A': 97.0, 'C': 196.0, 'G': 9528.0, 'T': 75.0},\n",
       "  6: {'A': 76.0, 'C': 9294.0, 'G': 97.0, 'T': 429.0},\n",
       "  7: {'A': 8951.0, 'C': 312.0, 'G': 628.0, 'T': 5.0},\n",
       "  8: {'A': 12.0, 'C': 513.0, 'G': 9280.0, 'T': 91.0},\n",
       "  9: {'A': 47.0, 'C': 92.0, 'G': 9632.0, 'T': 125.0},\n",
       "  10: {'A': 2430.0, 'C': 2517.0, 'G': 2776.0, 'T': 2173.0},\n",
       "  11: {'A': 2525.0, 'C': 2155.0, 'G': 3507.0, 'T': 1709.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11038359',\n",
       "  'uniprot_ids': 'P46684;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1521.1',\n",
       "  'Matrix_Name': 'MAFA',\n",
       "  1: {'A': 2274.0, 'C': 1564.0, 'G': 2153.0, 'T': 2476.0},\n",
       "  2: {'A': 674.0, 'C': 1858.0, 'G': 317.0, 'T': 6278.0},\n",
       "  3: {'A': 305.0, 'C': 207.0, 'G': 8344.0, 'T': 120.0},\n",
       "  4: {'A': 971.0, 'C': 7940.0, 'G': 186.0, 'T': 541.0},\n",
       "  5: {'A': 974.0, 'C': 987.0, 'G': 452.0, 'T': 8139.0},\n",
       "  6: {'A': 669.0, 'C': 476.0, 'G': 8029.0, 'T': 1292.0},\n",
       "  7: {'A': 7499.0, 'C': 1616.0, 'G': 318.0, 'T': 1232.0},\n",
       "  8: {'A': 603.0, 'C': 3986.0, 'G': 4025.0, 'T': 658.0},\n",
       "  9: {'A': 1282.0, 'C': 241.0, 'G': 1782.0, 'T': 7525.0},\n",
       "  10: {'A': 1205.0, 'C': 7986.0, 'G': 423.0, 'T': 716.0},\n",
       "  11: {'A': 8129.0, 'C': 454.0, 'G': 890.0, 'T': 1124.0},\n",
       "  12: {'A': 474.0, 'C': 105.0, 'G': 7958.0, 'T': 974.0},\n",
       "  13: {'A': 27.0, 'C': 8372.0, 'G': 222.0, 'T': 259.0},\n",
       "  14: {'A': 6133.0, 'C': 224.0, 'G': 1957.0, 'T': 803.0},\n",
       "  15: {'A': 2466.0, 'C': 2127.0, 'G': 1533.0, 'T': 2341.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9571165',\n",
       "  'uniprot_ids': 'Q8NHW3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0039.4',\n",
       "  'Matrix_Name': 'KLF4',\n",
       "  1: {'A': 12571.0, 'C': 17521.0, 'G': 14244.0, 'T': 11485.0},\n",
       "  2: {'A': 11488.0, 'C': 14039.0, 'G': 18474.0, 'T': 11820.0},\n",
       "  3: {'A': 2466.0, 'C': 49210.0, 'G': 1584.0, 'T': 2561.0},\n",
       "  4: {'A': 2106.0, 'C': 47866.0, 'G': 1215.0, 'T': 4634.0},\n",
       "  5: {'A': 7022.0, 'C': 45406.0, 'G': 1423.0, 'T': 1970.0},\n",
       "  6: {'A': 1174.0, 'C': 52876.0, 'G': 794.0, 'T': 977.0},\n",
       "  7: {'A': 45603.0, 'C': 162.0, 'G': 6599.0, 'T': 3457.0},\n",
       "  8: {'A': 853.0, 'C': 52367.0, 'G': 1471.0, 'T': 1130.0},\n",
       "  9: {'A': 1618.0, 'C': 51113.0, 'G': 1871.0, 'T': 1219.0},\n",
       "  10: {'A': 1203.0, 'C': 51046.0, 'G': 1006.0, 'T': 2566.0},\n",
       "  11: {'A': 16265.0, 'C': 18250.0, 'G': 6293.0, 'T': 15013.0},\n",
       "  12: {'A': 9960.0, 'C': 19218.0, 'G': 14423.0, 'T': 12220.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'O43474',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0896.1',\n",
       "  'Matrix_Name': 'Hmx1',\n",
       "  1: {'A': 35.0, 'C': 30.0, 'G': 25.0, 'T': 14.0},\n",
       "  2: {'A': 18.0, 'C': 44.0, 'G': 28.0, 'T': 14.0},\n",
       "  3: {'A': 37.0, 'C': 30.0, 'G': 27.0, 'T': 10.0},\n",
       "  4: {'A': 70.0, 'C': 7.0, 'G': 17.0, 'T': 10.0},\n",
       "  5: {'A': 14.0, 'C': 9.0, 'G': 73.0, 'T': 8.0},\n",
       "  6: {'A': 2.0, 'C': 96.0, 'G': 1.0, 'T': 5.0},\n",
       "  7: {'A': 88.0, 'C': 1.0, 'G': 13.0, 'T': 2.0},\n",
       "  8: {'A': 89.0, 'C': 12.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 100.0},\n",
       "  10: {'A': 3.0, 'C': 7.0, 'G': 1.0, 'T': 94.0},\n",
       "  11: {'A': 96.0, 'C': 1.0, 'G': 2.0, 'T': 4.0},\n",
       "  12: {'A': 89.0, 'C': 2.0, 'G': 5.0, 'T': 7.0},\n",
       "  13: {'A': 26.0, 'C': 16.0, 'G': 18.0, 'T': 44.0},\n",
       "  14: {'A': 15.0, 'C': 23.0, 'G': 51.0, 'T': 14.0},\n",
       "  15: {'A': 39.0, 'C': 30.0, 'G': 29.0, 'T': 5.0},\n",
       "  16: {'A': 34.0, 'C': 14.0, 'G': 25.0, 'T': 32.0},\n",
       "  17: {'A': 19.0, 'C': 21.0, 'G': 23.0, 'T': 41.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'O70218',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0654.1',\n",
       "  'Matrix_Name': 'ISX',\n",
       "  1: {'A': 319.0, 'C': 864.0, 'G': 365.0, 'T': 521.0},\n",
       "  2: {'A': 11.0, 'C': 1004.0, 'G': 1.0, 'T': 1062.0},\n",
       "  3: {'A': 2066.0, 'C': 49.0, 'G': 6.0, 'T': 25.0},\n",
       "  4: {'A': 2066.0, 'C': 17.0, 'G': 1.0, 'T': 11.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2066.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 25.0, 'T': 2066.0},\n",
       "  7: {'A': 2066.0, 'C': 1.0, 'G': 1.0, 'T': 50.0},\n",
       "  8: {'A': 771.0, 'C': 308.0, 'G': 705.0, 'T': 284.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q2M1V0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0478.1',\n",
       "  'Matrix_Name': 'FOSL2',\n",
       "  1: {'A': 834.0, 'C': 995.0, 'G': 2087.0, 'T': 1406.0},\n",
       "  2: {'A': 1526.0, 'C': 593.0, 'G': 2708.0, 'T': 495.0},\n",
       "  3: {'A': 2862.0, 'C': 56.0, 'G': 2329.0, 'T': 75.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 5319.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 5319.0, 'T': 1.0},\n",
       "  6: {'A': 5076.0, 'C': 52.0, 'G': 193.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 3287.0, 'G': 1839.0, 'T': 195.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 5319.0},\n",
       "  9: {'A': 206.0, 'C': 5114.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 5319.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1400.0, 'G': 1934.0, 'T': 1987.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P15408',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0840.1',\n",
       "  'Matrix_Name': 'Creb5',\n",
       "  1: {'A': 1203.0, 'C': 741.0, 'G': 1067.0, 'T': 696.0},\n",
       "  2: {'A': 3705.0, 'C': 228.0, 'G': 257.0, 'T': 41.0},\n",
       "  3: {'A': 7.0, 'C': 17.0, 'G': 1.0, 'T': 3705.0},\n",
       "  4: {'A': 14.0, 'C': 7.0, 'G': 3705.0, 'T': 919.0},\n",
       "  5: {'A': 3705.0, 'C': 1.0, 'G': 2.0, 'T': 11.0},\n",
       "  6: {'A': 1.0, 'C': 3705.0, 'G': 1.0, 'T': 58.0},\n",
       "  7: {'A': 152.0, 'C': 1.0, 'G': 3705.0, 'T': 1.0},\n",
       "  8: {'A': 19.0, 'C': 7.0, 'G': 1.0, 'T': 3705.0},\n",
       "  9: {'A': 918.0, 'C': 3705.0, 'G': 1.0, 'T': 5.0},\n",
       "  10: {'A': 3705.0, 'C': 1.0, 'G': 1.0, 'T': 3.0},\n",
       "  11: {'A': 26.0, 'C': 1862.0, 'G': 78.0, 'T': 1844.0},\n",
       "  12: {'A': 741.0, 'C': 1467.0, 'G': 373.0, 'T': 1127.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11134034',\n",
       "  'uniprot_ids': 'Q8K1L0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0699.1',\n",
       "  'Matrix_Name': 'LBX2',\n",
       "  1: {'A': 810.0, 'C': 740.0, 'G': 833.0, 'T': 617.0},\n",
       "  2: {'A': 367.0, 'C': 1438.0, 'G': 478.0, 'T': 716.0},\n",
       "  3: {'A': 158.0, 'C': 1635.0, 'G': 110.0, 'T': 1363.0},\n",
       "  4: {'A': 2997.0, 'C': 339.0, 'G': 326.0, 'T': 58.0},\n",
       "  5: {'A': 2997.0, 'C': 291.0, 'G': 131.0, 'T': 90.0},\n",
       "  6: {'A': 1.0, 'C': 30.0, 'G': 119.0, 'T': 2997.0},\n",
       "  7: {'A': 151.0, 'C': 259.0, 'G': 294.0, 'T': 2997.0},\n",
       "  8: {'A': 2997.0, 'C': 35.0, 'G': 107.0, 'T': 149.0},\n",
       "  9: {'A': 991.0, 'C': 435.0, 'G': 1136.0, 'T': 439.0},\n",
       "  10: {'A': 723.0, 'C': 978.0, 'G': 738.0, 'T': 561.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q6XYB7',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0883.1',\n",
       "  'Matrix_Name': 'Dmbx1',\n",
       "  1: {'A': 35.0, 'C': 14.0, 'G': 13.0, 'T': 42.0},\n",
       "  2: {'A': 29.0, 'C': 9.0, 'G': 38.0, 'T': 28.0},\n",
       "  3: {'A': 50.0, 'C': 9.0, 'G': 32.0, 'T': 13.0},\n",
       "  4: {'A': 50.0, 'C': 9.0, 'G': 14.0, 'T': 31.0},\n",
       "  5: {'A': 26.0, 'C': 37.0, 'G': 19.0, 'T': 22.0},\n",
       "  6: {'A': 29.0, 'C': 51.0, 'G': 21.0, 'T': 4.0},\n",
       "  7: {'A': 7.0, 'C': 2.0, 'G': 94.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 2.0, 'G': 100.0, 'T': 1.0},\n",
       "  9: {'A': 95.0, 'C': 7.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 100.0},\n",
       "  11: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 99.0},\n",
       "  12: {'A': 98.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  13: {'A': 41.0, 'C': 6.0, 'G': 36.0, 'T': 21.0},\n",
       "  14: {'A': 10.0, 'C': 12.0, 'G': 38.0, 'T': 43.0},\n",
       "  15: {'A': 16.0, 'C': 29.0, 'G': 31.0, 'T': 28.0},\n",
       "  16: {'A': 31.0, 'C': 21.0, 'G': 20.0, 'T': 31.0},\n",
       "  17: {'A': 38.0, 'C': 13.0, 'G': 16.0, 'T': 37.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q91ZK4',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0101.1',\n",
       "  'Matrix_Name': 'REL',\n",
       "  1: {'A': 1.0, 'C': 6.0, 'G': 9.0, 'T': 5.0},\n",
       "  2: {'A': 1.0, 'C': 2.0, 'G': 16.0, 'T': 2.0},\n",
       "  3: {'A': 2.0, 'C': 1.0, 'G': 16.0, 'T': 2.0},\n",
       "  4: {'A': 6.0, 'C': 2.0, 'G': 10.0, 'T': 3.0},\n",
       "  5: {'A': 7.0, 'C': 6.0, 'G': 4.0, 'T': 4.0},\n",
       "  6: {'A': 6.0, 'C': 2.0, 'G': 2.0, 'T': 11.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 1.0, 'T': 17.0},\n",
       "  8: {'A': 3.0, 'C': 1.0, 'G': 1.0, 'T': 16.0},\n",
       "  9: {'A': 1.0, 'C': 16.0, 'G': 1.0, 'T': 3.0},\n",
       "  10: {'A': 2.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NF-kappaB-related',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '1406630',\n",
       "  'uniprot_ids': 'Q04864',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1151.1',\n",
       "  'Matrix_Name': 'RORC',\n",
       "  1: {'A': 307.0, 'C': 180.0, 'G': 261.0, 'T': 256.0},\n",
       "  2: {'A': 330.0, 'C': 115.0, 'G': 142.0, 'T': 417.0},\n",
       "  3: {'A': 856.0, 'C': 15.0, 'G': 12.0, 'T': 121.0},\n",
       "  4: {'A': 528.0, 'C': 11.0, 'G': 33.0, 'T': 432.0},\n",
       "  5: {'A': 226.0, 'C': 250.0, 'G': 278.0, 'T': 250.0},\n",
       "  6: {'A': 44.0, 'C': 88.0, 'G': 117.0, 'T': 754.0},\n",
       "  7: {'A': 624.0, 'C': 25.0, 'G': 344.0, 'T': 12.0},\n",
       "  8: {'A': 6.0, 'C': 3.0, 'G': 920.0, 'T': 75.0},\n",
       "  9: {'A': 37.0, 'C': 4.0, 'G': 953.0, 'T': 9.0},\n",
       "  10: {'A': 60.0, 'C': 113.0, 'G': 169.0, 'T': 662.0},\n",
       "  11: {'A': 7.0, 'C': 823.0, 'G': 20.0, 'T': 154.0},\n",
       "  12: {'A': 786.0, 'C': 13.0, 'G': 162.0, 'T': 43.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '25500868',\n",
       "  'uniprot_ids': 'P51449',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0870.1',\n",
       "  'Matrix_Name': 'Sox1',\n",
       "  1: {'A': 1370.0, 'C': 397.0, 'G': 357.0, 'T': 196.0},\n",
       "  2: {'A': 2317.0, 'C': 52.0, 'G': 37.0, 'T': 86.0},\n",
       "  3: {'A': 1.0, 'C': 2317.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 2317.0, 'C': 13.0, 'G': 1.0, 'T': 8.0},\n",
       "  5: {'A': 2317.0, 'C': 326.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 14.0, 'C': 1.0, 'G': 1.0, 'T': 2317.0},\n",
       "  7: {'A': 1683.0, 'C': 6.0, 'G': 360.0, 'T': 635.0},\n",
       "  8: {'A': 783.0, 'C': 509.0, 'G': 526.0, 'T': 503.0},\n",
       "  9: {'A': 5.0, 'C': 2317.0, 'G': 52.0, 'T': 34.0},\n",
       "  10: {'A': 2317.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 245.0, 'T': 2317.0},\n",
       "  12: {'A': 1.0, 'C': 15.0, 'G': 1.0, 'T': 2317.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 2317.0, 'T': 1.0},\n",
       "  14: {'A': 80.0, 'C': 261.0, 'G': 59.0, 'T': 2317.0},\n",
       "  15: {'A': 223.0, 'C': 330.0, 'G': 266.0, 'T': 1500.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '12920151',\n",
       "  'uniprot_ids': 'P53783',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0858.1',\n",
       "  'Matrix_Name': 'Rarb(var.2)',\n",
       "  1: {'A': 611.0, 'C': 4.0, 'G': 40.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 2.0, 'G': 714.0, 'T': 1.0},\n",
       "  3: {'A': 3.0, 'C': 2.0, 'G': 664.0, 'T': 23.0},\n",
       "  4: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 695.0},\n",
       "  5: {'A': 2.0, 'C': 687.0, 'G': 3.0, 'T': 2.0},\n",
       "  6: {'A': 729.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  7: {'A': 294.0, 'C': 211.0, 'G': 16.0, 'T': 205.0},\n",
       "  8: {'A': 250.0, 'C': 253.0, 'G': 141.0, 'T': 65.0},\n",
       "  9: {'A': 39.0, 'C': 240.0, 'G': 117.0, 'T': 282.0},\n",
       "  10: {'A': 439.0, 'C': 35.0, 'G': 42.0, 'T': 159.0},\n",
       "  11: {'A': 399.0, 'C': 5.0, 'G': 219.0, 'T': 5.0},\n",
       "  12: {'A': 640.0, 'C': 1.0, 'G': 38.0, 'T': 2.0},\n",
       "  13: {'A': 2.0, 'C': 2.0, 'G': 683.0, 'T': 1.0},\n",
       "  14: {'A': 1.0, 'C': 3.0, 'G': 664.0, 'T': 27.0},\n",
       "  15: {'A': 1.0, 'C': 4.0, 'G': 5.0, 'T': 675.0},\n",
       "  16: {'A': 1.0, 'C': 766.0, 'G': 1.0, 'T': 5.0},\n",
       "  17: {'A': 694.0, 'C': 1.0, 'G': 3.0, 'T': 2.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P22605',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1619.1',\n",
       "  'Matrix_Name': 'Ptf1a(var.2)',\n",
       "  1: {'A': 1870.0, 'C': 1804.0, 'G': 1975.0, 'T': 1608.0},\n",
       "  2: {'A': 1979.0, 'C': 2431.0, 'G': 1630.0, 'T': 1217.0},\n",
       "  3: {'A': 4347.0, 'C': 1071.0, 'G': 1309.0, 'T': 530.0},\n",
       "  4: {'A': 50.0, 'C': 7150.0, 'G': 29.0, 'T': 28.0},\n",
       "  5: {'A': 7120.0, 'C': 42.0, 'G': 66.0, 'T': 29.0},\n",
       "  6: {'A': 35.0, 'C': 308.0, 'G': 6690.0, 'T': 224.0},\n",
       "  7: {'A': 225.0, 'C': 6698.0, 'G': 300.0, 'T': 34.0},\n",
       "  8: {'A': 27.0, 'C': 67.0, 'G': 41.0, 'T': 7122.0},\n",
       "  9: {'A': 28.0, 'C': 28.0, 'G': 7151.0, 'T': 50.0},\n",
       "  10: {'A': 536.0, 'C': 1313.0, 'G': 1066.0, 'T': 4342.0},\n",
       "  11: {'A': 1222.0, 'C': 1633.0, 'G': 2427.0, 'T': 1975.0},\n",
       "  12: {'A': 1603.0, 'C': 1978.0, 'G': 1804.0, 'T': 1872.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23754747',\n",
       "  'uniprot_ids': 'Q9QX98',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0788.1',\n",
       "  'Matrix_Name': 'POU3F3',\n",
       "  1: {'A': 577.0, 'C': 76.0, 'G': 138.0, 'T': 291.0},\n",
       "  2: {'A': 441.0, 'C': 42.0, 'G': 132.0, 'T': 638.0},\n",
       "  3: {'A': 130.0, 'C': 58.0, 'G': 65.0, 'T': 1079.0},\n",
       "  4: {'A': 1079.0, 'C': 12.0, 'G': 13.0, 'T': 31.0},\n",
       "  5: {'A': 26.0, 'C': 34.0, 'G': 50.0, 'T': 1079.0},\n",
       "  6: {'A': 23.0, 'C': 11.0, 'G': 1079.0, 'T': 87.0},\n",
       "  7: {'A': 29.0, 'C': 1079.0, 'G': 119.0, 'T': 344.0},\n",
       "  8: {'A': 507.0, 'C': 5.0, 'G': 2.0, 'T': 573.0},\n",
       "  9: {'A': 1079.0, 'C': 12.0, 'G': 78.0, 'T': 80.0},\n",
       "  10: {'A': 1079.0, 'C': 7.0, 'G': 16.0, 'T': 15.0},\n",
       "  11: {'A': 54.0, 'C': 21.0, 'G': 13.0, 'T': 1079.0},\n",
       "  12: {'A': 241.0, 'C': 142.0, 'G': 238.0, 'T': 1079.0},\n",
       "  13: {'A': 537.0, 'C': 86.0, 'G': 116.0, 'T': 543.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9852081',\n",
       "  'uniprot_ids': 'P20264',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0159.1',\n",
       "  'Matrix_Name': 'RARA::RXRA',\n",
       "  1: {'A': 13.0, 'C': 1.0, 'G': 12.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 24.0, 'T': 1.0},\n",
       "  3: {'A': 2.0, 'C': 1.0, 'G': 14.0, 'T': 10.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 23.0},\n",
       "  5: {'A': 1.0, 'C': 19.0, 'G': 4.0, 'T': 3.0},\n",
       "  6: {'A': 23.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  7: {'A': 5.0, 'C': 8.0, 'G': 6.0, 'T': 8.0},\n",
       "  8: {'A': 6.0, 'C': 9.0, 'G': 10.0, 'T': 2.0},\n",
       "  9: {'A': 6.0, 'C': 5.0, 'G': 12.0, 'T': 4.0},\n",
       "  10: {'A': 14.0, 'C': 2.0, 'G': 8.0, 'T': 3.0},\n",
       "  11: {'A': 6.0, 'C': 7.0, 'G': 13.0, 'T': 1.0},\n",
       "  12: {'A': 18.0, 'C': 4.0, 'G': 4.0, 'T': 1.0},\n",
       "  13: {'A': 2.0, 'C': 2.0, 'G': 21.0, 'T': 2.0},\n",
       "  14: {'A': 1.0, 'C': 2.0, 'G': 17.0, 'T': 7.0},\n",
       "  15: {'A': 3.0, 'C': 2.0, 'G': 4.0, 'T': 18.0},\n",
       "  16: {'A': 2.0, 'C': 18.0, 'G': 4.0, 'T': 3.0},\n",
       "  17: {'A': 22.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P10276;',\n",
       "  'data_type': 'COMPILED'},\n",
       " {'Matrix_ID': 'MA1516.1',\n",
       "  'Matrix_Name': 'KLF3',\n",
       "  1: {'A': 7718.0, 'C': 2656.0, 'G': 26569.0, 'T': 6320.0},\n",
       "  2: {'A': 24772.0, 'C': 616.0, 'G': 18490.0, 'T': 208.0},\n",
       "  3: {'A': 1443.0, 'C': 43260.0, 'G': 1955.0, 'T': 706.0},\n",
       "  4: {'A': 299.0, 'C': 43260.0, 'G': 249.0, 'T': 336.0},\n",
       "  5: {'A': 26758.0, 'C': 708.0, 'G': 16504.0, 'T': 420.0},\n",
       "  6: {'A': 2262.0, 'C': 43260.0, 'G': 22.0, 'T': 1410.0},\n",
       "  7: {'A': 4246.0, 'C': 335.0, 'G': 43260.0, 'T': 1213.0},\n",
       "  8: {'A': 203.0, 'C': 43260.0, 'G': 32.0, 'T': 1.0},\n",
       "  9: {'A': 84.0, 'C': 43260.0, 'G': 30.0, 'T': 1.0},\n",
       "  10: {'A': 78.0, 'C': 43260.0, 'G': 49.0, 'T': 1219.0},\n",
       "  11: {'A': 17030.0, 'C': 10100.0, 'G': 1515.0, 'T': 14618.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11443140',\n",
       "  'uniprot_ids': 'P57682',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0784.1',\n",
       "  'Matrix_Name': 'POU1F1',\n",
       "  1: {'A': 1212.0, 'C': 248.0, 'G': 274.0, 'T': 398.0},\n",
       "  2: {'A': 1129.0, 'C': 124.0, 'G': 188.0, 'T': 1001.0},\n",
       "  3: {'A': 277.0, 'C': 84.0, 'G': 109.0, 'T': 2129.0},\n",
       "  4: {'A': 2129.0, 'C': 15.0, 'G': 33.0, 'T': 20.0},\n",
       "  5: {'A': 46.0, 'C': 75.0, 'G': 27.0, 'T': 2129.0},\n",
       "  6: {'A': 29.0, 'C': 12.0, 'G': 2129.0, 'T': 177.0},\n",
       "  7: {'A': 99.0, 'C': 2129.0, 'G': 165.0, 'T': 537.0},\n",
       "  8: {'A': 1268.0, 'C': 13.0, 'G': 8.0, 'T': 862.0},\n",
       "  9: {'A': 2129.0, 'C': 12.0, 'G': 83.0, 'T': 86.0},\n",
       "  10: {'A': 2129.0, 'C': 11.0, 'G': 1.0, 'T': 4.0},\n",
       "  11: {'A': 77.0, 'C': 38.0, 'G': 41.0, 'T': 2129.0},\n",
       "  12: {'A': 223.0, 'C': 242.0, 'G': 841.0, 'T': 1288.0},\n",
       "  13: {'A': 2129.0, 'C': 175.0, 'G': 195.0, 'T': 274.0},\n",
       "  14: {'A': 737.0, 'C': 480.0, 'G': 2129.0, 'T': 548.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '2350782',\n",
       "  'uniprot_ids': 'P28069',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1683.1',\n",
       "  'Matrix_Name': 'FOXA3',\n",
       "  1: {'A': 18376.0, 'C': 8521.0, 'G': 12745.0, 'T': 9299.0},\n",
       "  2: {'A': 16694.0, 'C': 3628.0, 'G': 9297.0, 'T': 19322.0},\n",
       "  3: {'A': 4482.0, 'C': 1465.0, 'G': 40397.0, 'T': 2597.0},\n",
       "  4: {'A': 759.0, 'C': 904.0, 'G': 691.0, 'T': 46587.0},\n",
       "  5: {'A': 45179.0, 'C': 1981.0, 'G': 637.0, 'T': 1144.0},\n",
       "  6: {'A': 43100.0, 'C': 3884.0, 'G': 978.0, 'T': 979.0},\n",
       "  7: {'A': 46047.0, 'C': 974.0, 'G': 935.0, 'T': 985.0},\n",
       "  8: {'A': 952.0, 'C': 39510.0, 'G': 1206.0, 'T': 7273.0},\n",
       "  9: {'A': 45553.0, 'C': 934.0, 'G': 615.0, 'T': 1839.0},\n",
       "  10: {'A': 13443.0, 'C': 9976.0, 'G': 11452.0, 'T': 14070.0},\n",
       "  11: {'A': 14903.0, 'C': 9885.0, 'G': 11104.0, 'T': 13049.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '19919681',\n",
       "  'uniprot_ids': 'P55318',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1479.1',\n",
       "  'Matrix_Name': 'DMRTC2',\n",
       "  1: {'A': 655.0, 'C': 103.0, 'G': 84.0, 'T': 220.0},\n",
       "  2: {'A': 644.0, 'C': 61.0, 'G': 108.0, 'T': 231.0},\n",
       "  3: {'A': 186.0, 'C': 79.0, 'G': 37.0, 'T': 874.0},\n",
       "  4: {'A': 49.0, 'C': 91.0, 'G': 135.0, 'T': 874.0},\n",
       "  5: {'A': 1.0, 'C': 13.0, 'G': 874.0, 'T': 1.0},\n",
       "  6: {'A': 460.0, 'C': 268.0, 'G': 2.0, 'T': 148.0},\n",
       "  7: {'A': 101.0, 'C': 81.0, 'G': 8.0, 'T': 874.0},\n",
       "  8: {'A': 874.0, 'C': 7.0, 'G': 1.0, 'T': 6.0},\n",
       "  9: {'A': 1.0, 'C': 874.0, 'G': 3.0, 'T': 1.0},\n",
       "  10: {'A': 874.0, 'C': 1.0, 'G': 41.0, 'T': 74.0},\n",
       "  11: {'A': 243.0, 'C': 166.0, 'G': 81.0, 'T': 874.0},\n",
       "  12: {'A': 143.0, 'C': 142.0, 'G': 154.0, 'T': 438.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'DMRT',\n",
       "  'tf_class': 'DM-type',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q8IXT2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0639.1',\n",
       "  'Matrix_Name': 'DBP',\n",
       "  1: {'A': 1553.0, 'C': 1715.0, 'G': 1904.0, 'T': 1931.0},\n",
       "  2: {'A': 3707.0, 'C': 1131.0, 'G': 3394.0, 'T': 94.0},\n",
       "  3: {'A': 6.0, 'C': 5.0, 'G': 3.0, 'T': 7101.0},\n",
       "  4: {'A': 5.0, 'C': 7.0, 'G': 603.0, 'T': 7101.0},\n",
       "  5: {'A': 7101.0, 'C': 1.0, 'G': 499.0, 'T': 5.0},\n",
       "  6: {'A': 3.0, 'C': 7101.0, 'G': 22.0, 'T': 2665.0},\n",
       "  7: {'A': 3181.0, 'C': 18.0, 'G': 7101.0, 'T': 2.0},\n",
       "  8: {'A': 4.0, 'C': 542.0, 'G': 8.0, 'T': 7101.0},\n",
       "  9: {'A': 7101.0, 'C': 289.0, 'G': 8.0, 'T': 7.0},\n",
       "  10: {'A': 7101.0, 'C': 2.0, 'G': 3.0, 'T': 3.0},\n",
       "  11: {'A': 159.0, 'C': 3708.0, 'G': 1544.0, 'T': 3394.0},\n",
       "  12: {'A': 2575.0, 'C': 1658.0, 'G': 1748.0, 'T': 1123.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21458520',\n",
       "  'uniprot_ids': 'Q10586',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0130.1',\n",
       "  'Matrix_Name': 'ZNF354C',\n",
       "  1: {'A': 8.0, 'C': 7.0, 'G': 4.0, 'T': 1.0},\n",
       "  2: {'A': 4.0, 'C': 3.0, 'G': 1.0, 'T': 12.0},\n",
       "  3: {'A': 1.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 17.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 16.0, 'G': 2.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 6,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '15555547',\n",
       "  'uniprot_ids': 'Q86Y25',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0805.1',\n",
       "  'Matrix_Name': 'TBX1',\n",
       "  1: {'A': 11932.0, 'C': 114.0, 'G': 1008.0, 'T': 565.0},\n",
       "  2: {'A': 1561.0, 'C': 712.0, 'G': 11932.0, 'T': 670.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 11932.0, 'T': 1.0},\n",
       "  4: {'A': 29.0, 'C': 215.0, 'G': 142.0, 'T': 11932.0},\n",
       "  5: {'A': 1.0, 'C': 6.0, 'G': 11932.0, 'T': 1.0},\n",
       "  6: {'A': 143.0, 'C': 614.0, 'G': 110.0, 'T': 11932.0},\n",
       "  7: {'A': 142.0, 'C': 721.0, 'G': 11932.0, 'T': 1040.0},\n",
       "  8: {'A': 11932.0, 'C': 34.0, 'G': 19.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TBX1-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'O43435',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1153.1',\n",
       "  'Matrix_Name': 'Smad4',\n",
       "  1: {'A': 89.235, 'C': 251.0, 'G': 81.882, 'T': 581.882},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1001.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  4: {'A': 1.0, 'C': 1001.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  6: {'A': 596.588, 'C': 1.0, 'G': 405.412, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1001.0, 'T': 1.0},\n",
       "  8: {'A': 493.647, 'C': 273.059, 'G': 126.0, 'T': 111.294},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'SMAD',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '15329343',\n",
       "  'uniprot_ids': 'P97471',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0114.4',\n",
       "  'Matrix_Name': 'HNF4A',\n",
       "  1: {'A': 10047.0, 'C': 7137.0, 'G': 13906.0, 'T': 15418.0},\n",
       "  2: {'A': 9789.0, 'C': 14436.0, 'G': 9295.0, 'T': 12988.0},\n",
       "  3: {'A': 1469.0, 'C': 41444.0, 'G': 1238.0, 'T': 2357.0},\n",
       "  4: {'A': 43322.0, 'C': 1258.0, 'G': 1062.0, 'T': 866.0},\n",
       "  5: {'A': 40218.0, 'C': 1371.0, 'G': 3616.0, 'T': 1303.0},\n",
       "  6: {'A': 43265.0, 'C': 545.0, 'G': 1898.0, 'T': 800.0},\n",
       "  7: {'A': 717.0, 'C': 917.0, 'G': 43529.0, 'T': 1345.0},\n",
       "  8: {'A': 928.0, 'C': 696.0, 'G': 2185.0, 'T': 42699.0},\n",
       "  9: {'A': 2771.0, 'C': 37805.0, 'G': 1827.0, 'T': 4105.0},\n",
       "  10: {'A': 1364.0, 'C': 42342.0, 'G': 644.0, 'T': 2158.0},\n",
       "  11: {'A': 36655.0, 'C': 2783.0, 'G': 2937.0, 'T': 4133.0},\n",
       "  12: {'A': 13575.0, 'C': 9864.0, 'G': 13015.0, 'T': 10054.0},\n",
       "  13: {'A': 15280.0, 'C': 9071.0, 'G': 11715.0, 'T': 10442.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '22383578',\n",
       "  'uniprot_ids': 'P41235',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0467.1',\n",
       "  'Matrix_Name': 'Crx',\n",
       "  1: {'A': 1208.0, 'C': 298.0, 'G': 459.0, 'T': 136.0},\n",
       "  2: {'A': 1455.0, 'C': 44.0, 'G': 418.0, 'T': 184.0},\n",
       "  3: {'A': 348.0, 'C': 56.0, 'G': 1471.0, 'T': 226.0},\n",
       "  4: {'A': 944.0, 'C': 223.0, 'G': 824.0, 'T': 110.0},\n",
       "  5: {'A': 516.0, 'C': 1.0, 'G': 1575.0, 'T': 9.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 2098.0, 'T': 1.0},\n",
       "  7: {'A': 1633.0, 'C': 466.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2098.0},\n",
       "  9: {'A': 1.0, 'C': 30.0, 'G': 1.0, 'T': 2069.0},\n",
       "  10: {'A': 1996.0, 'C': 1.0, 'G': 1.0, 'T': 103.0},\n",
       "  11: {'A': 545.0, 'C': 53.0, 'G': 1327.0, 'T': 176.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '20463752',\n",
       "  'uniprot_ids': 'O54751',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1535.1',\n",
       "  'Matrix_Name': 'NR2C1',\n",
       "  1: {'A': 2553.0, 'C': 2849.0, 'G': 1998.0, 'T': 2226.0},\n",
       "  2: {'A': 4036.0, 'C': 1226.0, 'G': 5588.0, 'T': 1317.0},\n",
       "  3: {'A': 4867.0, 'C': 228.0, 'G': 4757.0, 'T': 56.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 9623.0, 'T': 298.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 9623.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 272.0, 'T': 9623.0},\n",
       "  7: {'A': 1.0, 'C': 9623.0, 'G': 322.0, 'T': 1.0},\n",
       "  8: {'A': 9623.0, 'C': 1.0, 'G': 1717.0, 'T': 1.0},\n",
       "  9: {'A': 2299.0, 'C': 2787.0, 'G': 2040.0, 'T': 2500.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'P13056',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0609.2',\n",
       "  'Matrix_Name': 'CREM',\n",
       "  1: {'A': 3866.0, 'C': 4231.0, 'G': 5133.0, 'T': 3801.0},\n",
       "  2: {'A': 3967.0, 'C': 5522.0, 'G': 3957.0, 'T': 3585.0},\n",
       "  3: {'A': 5882.0, 'C': 2909.0, 'G': 4475.0, 'T': 3765.0},\n",
       "  4: {'A': 2937.0, 'C': 2160.0, 'G': 11145.0, 'T': 789.0},\n",
       "  5: {'A': 560.0, 'C': 731.0, 'G': 507.0, 'T': 15233.0},\n",
       "  6: {'A': 435.0, 'C': 1039.0, 'G': 14884.0, 'T': 673.0},\n",
       "  7: {'A': 15695.0, 'C': 314.0, 'G': 620.0, 'T': 402.0},\n",
       "  8: {'A': 518.0, 'C': 14032.0, 'G': 898.0, 'T': 1583.0},\n",
       "  9: {'A': 1583.0, 'C': 902.0, 'G': 14027.0, 'T': 519.0},\n",
       "  10: {'A': 401.0, 'C': 623.0, 'G': 315.0, 'T': 15692.0},\n",
       "  11: {'A': 673.0, 'C': 14884.0, 'G': 1040.0, 'T': 434.0},\n",
       "  12: {'A': 15229.0, 'C': 509.0, 'G': 731.0, 'T': 562.0},\n",
       "  13: {'A': 790.0, 'C': 11147.0, 'G': 2157.0, 'T': 2937.0},\n",
       "  14: {'A': 3758.0, 'C': 4474.0, 'G': 2911.0, 'T': 5888.0},\n",
       "  15: {'A': 3583.0, 'C': 3958.0, 'G': 5524.0, 'T': 3966.0},\n",
       "  16: {'A': 3802.0, 'C': 5133.0, 'G': 4231.0, 'T': 3865.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20920259',\n",
       "  'uniprot_ids': 'Q03060',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0903.1',\n",
       "  'Matrix_Name': 'HOXB3',\n",
       "  1: {'A': 4501.0, 'C': 2527.0, 'G': 3152.0, 'T': 2723.0},\n",
       "  2: {'A': 2555.0, 'C': 4146.0, 'G': 4006.0, 'T': 2195.0},\n",
       "  3: {'A': 1745.0, 'C': 4980.0, 'G': 605.0, 'T': 7921.0},\n",
       "  4: {'A': 7794.0, 'C': 5107.0, 'G': 839.0, 'T': 698.0},\n",
       "  5: {'A': 12900.0, 'C': 546.0, 'G': 641.0, 'T': 243.0},\n",
       "  6: {'A': 1.0, 'C': 151.0, 'G': 1.0, 'T': 12900.0},\n",
       "  7: {'A': 442.0, 'C': 840.0, 'G': 1.0, 'T': 12900.0},\n",
       "  8: {'A': 12900.0, 'C': 120.0, 'G': 1.0, 'T': 575.0},\n",
       "  9: {'A': 4543.0, 'C': 2120.0, 'G': 4733.0, 'T': 1507.0},\n",
       "  10: {'A': 3200.0, 'C': 3996.0, 'G': 2982.0, 'T': 2725.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P14651',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0905.1',\n",
       "  'Matrix_Name': 'HOXC10',\n",
       "  1: {'A': 1496.0, 'C': 1045.0, 'G': 3212.0, 'T': 16.0},\n",
       "  2: {'A': 27.0, 'C': 246.0, 'G': 1.0, 'T': 3212.0},\n",
       "  3: {'A': 316.0, 'C': 3212.0, 'G': 13.0, 'T': 153.0},\n",
       "  4: {'A': 2639.0, 'C': 7.0, 'G': 3212.0, 'T': 25.0},\n",
       "  5: {'A': 28.0, 'C': 1.0, 'G': 1.0, 'T': 3212.0},\n",
       "  6: {'A': 2166.0, 'C': 6.0, 'G': 5.0, 'T': 1048.0},\n",
       "  7: {'A': 3212.0, 'C': 4.0, 'G': 6.0, 'T': 103.0},\n",
       "  8: {'A': 3212.0, 'C': 77.0, 'G': 87.0, 'T': 188.0},\n",
       "  9: {'A': 3212.0, 'C': 552.0, 'G': 533.0, 'T': 1070.0},\n",
       "  10: {'A': 989.0, 'C': 680.0, 'G': 454.0, 'T': 1093.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9NYD6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0773.1',\n",
       "  'Matrix_Name': 'MEF2D',\n",
       "  1: {'A': 2121.0, 'C': 110.0, 'G': 1934.0, 'T': 1643.0},\n",
       "  2: {'A': 29.0, 'C': 5804.0, 'G': 1.0, 'T': 54.0},\n",
       "  3: {'A': 4.0, 'C': 23.0, 'G': 1.0, 'T': 5804.0},\n",
       "  4: {'A': 5804.0, 'C': 9.0, 'G': 3.0, 'T': 7.0},\n",
       "  5: {'A': 2783.0, 'C': 1.0, 'G': 3.0, 'T': 3022.0},\n",
       "  6: {'A': 5804.0, 'C': 5.0, 'G': 1.0, 'T': 779.0},\n",
       "  7: {'A': 5804.0, 'C': 1.0, 'G': 3.0, 'T': 218.0},\n",
       "  8: {'A': 5804.0, 'C': 3.0, 'G': 2.0, 'T': 70.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 5804.0},\n",
       "  10: {'A': 5804.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  11: {'A': 340.0, 'C': 18.0, 'G': 5804.0, 'T': 6.0},\n",
       "  12: {'A': 3207.0, 'C': 2599.0, 'G': 71.0, 'T': 586.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Regulators',\n",
       "  'tf_class': 'MADS',\n",
       "  'pubmed_ids': '25217591',\n",
       "  'uniprot_ids': 'Q05BX2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0522.3',\n",
       "  'Matrix_Name': 'TCF3',\n",
       "  1: {'A': 6271.0, 'C': 10213.0, 'G': 8540.0, 'T': 6241.0},\n",
       "  2: {'A': 8033.0, 'C': 9270.0, 'G': 9433.0, 'T': 4529.0},\n",
       "  3: {'A': 470.0, 'C': 29380.0, 'G': 709.0, 'T': 706.0},\n",
       "  4: {'A': 28917.0, 'C': 397.0, 'G': 1575.0, 'T': 376.0},\n",
       "  5: {'A': 310.0, 'C': 28641.0, 'G': 1363.0, 'T': 951.0},\n",
       "  6: {'A': 1128.0, 'C': 27200.0, 'G': 2653.0, 'T': 284.0},\n",
       "  7: {'A': 381.0, 'C': 2004.0, 'G': 1340.0, 'T': 27540.0},\n",
       "  8: {'A': 1090.0, 'C': 1039.0, 'G': 28643.0, 'T': 493.0},\n",
       "  9: {'A': 296.0, 'C': 28743.0, 'G': 988.0, 'T': 1238.0},\n",
       "  10: {'A': 7277.0, 'C': 8728.0, 'G': 8026.0, 'T': 7234.0},\n",
       "  11: {'A': 5486.0, 'C': 10123.0, 'G': 9223.0, 'T': 6433.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'E2A-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '10594029',\n",
       "  'uniprot_ids': 'P15923',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0499.2',\n",
       "  'Matrix_Name': 'MYOD1',\n",
       "  1: {'A': 7176.0, 'C': 11280.0, 'G': 8119.0, 'T': 7729.0},\n",
       "  2: {'A': 9463.0, 'C': 7104.0, 'G': 8445.0, 'T': 9292.0},\n",
       "  3: {'A': 4550.0, 'C': 5623.0, 'G': 21885.0, 'T': 2246.0},\n",
       "  4: {'A': 350.0, 'C': 33329.0, 'G': 355.0, 'T': 270.0},\n",
       "  5: {'A': 32282.0, 'C': 584.0, 'G': 850.0, 'T': 588.0},\n",
       "  6: {'A': 464.0, 'C': 25276.0, 'G': 7337.0, 'T': 1227.0},\n",
       "  7: {'A': 535.0, 'C': 32505.0, 'G': 933.0, 'T': 331.0},\n",
       "  8: {'A': 242.0, 'C': 564.0, 'G': 367.0, 'T': 33131.0},\n",
       "  9: {'A': 209.0, 'C': 387.0, 'G': 33469.0, 'T': 239.0},\n",
       "  10: {'A': 403.0, 'C': 2556.0, 'G': 1700.0, 'T': 29645.0},\n",
       "  11: {'A': 1809.0, 'C': 19053.0, 'G': 4441.0, 'T': 9001.0},\n",
       "  12: {'A': 9463.0, 'C': 10486.0, 'G': 6203.0, 'T': 8152.0},\n",
       "  13: {'A': 5688.0, 'C': 11761.0, 'G': 7496.0, 'T': 9359.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '20412780',\n",
       "  'uniprot_ids': 'P15172',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0717.1',\n",
       "  'Matrix_Name': 'RAX2',\n",
       "  1: {'A': 5335.0, 'C': 15811.0, 'G': 7050.0, 'T': 10667.0},\n",
       "  2: {'A': 3559.0, 'C': 19646.0, 'G': 1689.0, 'T': 19216.0},\n",
       "  3: {'A': 38861.0, 'C': 3061.0, 'G': 4491.0, 'T': 2422.0},\n",
       "  4: {'A': 38861.0, 'C': 1524.0, 'G': 758.0, 'T': 1031.0},\n",
       "  5: {'A': 271.0, 'C': 739.0, 'G': 126.0, 'T': 38861.0},\n",
       "  6: {'A': 2168.0, 'C': 4233.0, 'G': 2514.0, 'T': 38861.0},\n",
       "  7: {'A': 38861.0, 'C': 2645.0, 'G': 361.0, 'T': 4259.0},\n",
       "  8: {'A': 13412.0, 'C': 7488.0, 'G': 12363.0, 'T': 5602.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q96IS3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1564.1',\n",
       "  'Matrix_Name': 'SP9',\n",
       "  1: {'A': 835.0, 'C': 441.0, 'G': 953.0, 'T': 227.0},\n",
       "  2: {'A': 567.0, 'C': 2452.0, 'G': 310.0, 'T': 519.0},\n",
       "  3: {'A': 322.0, 'C': 2452.0, 'G': 114.0, 'T': 285.0},\n",
       "  4: {'A': 2452.0, 'C': 1128.0, 'G': 57.0, 'T': 97.0},\n",
       "  5: {'A': 3.0, 'C': 2452.0, 'G': 18.0, 'T': 10.0},\n",
       "  6: {'A': 212.0, 'C': 201.0, 'G': 2452.0, 'T': 174.0},\n",
       "  7: {'A': 23.0, 'C': 2452.0, 'G': 1.0, 'T': 15.0},\n",
       "  8: {'A': 1.0, 'C': 2452.0, 'G': 1.0, 'T': 31.0},\n",
       "  9: {'A': 40.0, 'C': 2452.0, 'G': 25.0, 'T': 413.0},\n",
       "  10: {'A': 1193.0, 'C': 1260.0, 'G': 39.0, 'T': 354.0},\n",
       "  11: {'A': 189.0, 'C': 1864.0, 'G': 192.0, 'T': 589.0},\n",
       "  12: {'A': 421.0, 'C': 1168.0, 'G': 205.0, 'T': 661.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '26187067',\n",
       "  'uniprot_ids': 'P0CG40',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1491.1',\n",
       "  'Matrix_Name': 'GLI3',\n",
       "  1: {'A': 9266.0, 'C': 7914.0, 'G': 7703.0, 'T': 2623.0},\n",
       "  2: {'A': 178.0, 'C': 233.0, 'G': 18755.0, 'T': 1324.0},\n",
       "  3: {'A': 17349.0, 'C': 3973.0, 'G': 2833.0, 'T': 4.0},\n",
       "  4: {'A': 69.0, 'C': 19264.0, 'G': 8.0, 'T': 1.0},\n",
       "  5: {'A': 8.0, 'C': 19330.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 18185.0, 'C': 2212.0, 'G': 195.0, 'T': 700.0},\n",
       "  7: {'A': 4.0, 'C': 19327.0, 'G': 5.0, 'T': 2.0},\n",
       "  8: {'A': 4985.0, 'C': 18085.0, 'G': 2.0, 'T': 3.0},\n",
       "  9: {'A': 529.0, 'C': 18880.0, 'G': 16.0, 'T': 324.0},\n",
       "  10: {'A': 18710.0, 'C': 1.0, 'G': 1487.0, 'T': 1.0},\n",
       "  11: {'A': 2187.0, 'C': 17928.0, 'G': 1361.0, 'T': 1507.0},\n",
       "  12: {'A': 4644.0, 'C': 5420.0, 'G': 12722.0, 'T': 1616.0},\n",
       "  13: {'A': 8398.0, 'C': 3327.0, 'G': 5325.0, 'T': 9057.0},\n",
       "  14: {'A': 6507.0, 'C': 11364.0, 'G': 2364.0, 'T': 6603.0},\n",
       "  15: {'A': 954.0, 'C': 2626.0, 'G': 17622.0, 'T': 1732.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '2105456',\n",
       "  'uniprot_ids': 'P10071',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0659.2',\n",
       "  'Matrix_Name': 'MAFG',\n",
       "  1: {'A': 5652.0, 'C': 2486.0, 'G': 8587.0, 'T': 4680.0},\n",
       "  2: {'A': 12162.0, 'C': 2068.0, 'G': 2183.0, 'T': 4992.0},\n",
       "  3: {'A': 2201.0, 'C': 3629.0, 'G': 14149.0, 'T': 1426.0},\n",
       "  4: {'A': 274.0, 'C': 167.0, 'G': 199.0, 'T': 20765.0},\n",
       "  5: {'A': 697.0, 'C': 20233.0, 'G': 131.0, 'T': 344.0},\n",
       "  6: {'A': 20839.0, 'C': 72.0, 'G': 175.0, 'T': 319.0},\n",
       "  7: {'A': 265.0, 'C': 271.0, 'G': 19578.0, 'T': 1291.0},\n",
       "  8: {'A': 213.0, 'C': 20694.0, 'G': 201.0, 'T': 297.0},\n",
       "  9: {'A': 19624.0, 'C': 303.0, 'G': 640.0, 'T': 838.0},\n",
       "  10: {'A': 4914.0, 'C': 2567.0, 'G': 2849.0, 'T': 11075.0},\n",
       "  11: {'A': 4022.0, 'C': 1437.0, 'G': 1286.0, 'T': 14660.0},\n",
       "  12: {'A': 3440.0, 'C': 1449.0, 'G': 1052.0, 'T': 15464.0},\n",
       "  13: {'A': 4249.0, 'C': 5545.0, 'G': 1783.0, 'T': 9828.0},\n",
       "  14: {'A': 4698.0, 'C': 4853.0, 'G': 2713.0, 'T': 9141.0},\n",
       "  15: {'A': 5906.0, 'C': 4264.0, 'G': 3924.0, 'T': 7311.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8264639',\n",
       "  'uniprot_ids': 'O15525',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0819.1',\n",
       "  'Matrix_Name': 'CLOCK',\n",
       "  1: {'A': 227.0, 'C': 36.0, 'G': 94.0, 'T': 76.0},\n",
       "  2: {'A': 430.0, 'C': 86.0, 'G': 90.0, 'T': 14.0},\n",
       "  3: {'A': 12.0, 'C': 430.0, 'G': 3.0, 'T': 3.0},\n",
       "  4: {'A': 430.0, 'C': 6.0, 'G': 16.0, 'T': 1.0},\n",
       "  5: {'A': 17.0, 'C': 430.0, 'G': 1.0, 'T': 29.0},\n",
       "  6: {'A': 41.0, 'C': 1.0, 'G': 430.0, 'T': 1.0},\n",
       "  7: {'A': 4.0, 'C': 6.0, 'G': 6.0, 'T': 430.0},\n",
       "  8: {'A': 1.0, 'C': 4.0, 'G': 430.0, 'T': 10.0},\n",
       "  9: {'A': 2.0, 'C': 153.0, 'G': 111.0, 'T': 430.0},\n",
       "  10: {'A': 90.0, 'C': 129.0, 'G': 40.0, 'T': 175.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '24591654',\n",
       "  'uniprot_ids': 'O15516',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0598.3',\n",
       "  'Matrix_Name': 'EHF',\n",
       "  1: {'A': 3261.0, 'C': 4342.0, 'G': 4103.0, 'T': 3666.0},\n",
       "  2: {'A': 2703.0, 'C': 4629.0, 'G': 3356.0, 'T': 4684.0},\n",
       "  3: {'A': 2115.0, 'C': 8888.0, 'G': 1541.0, 'T': 2828.0},\n",
       "  4: {'A': 13415.0, 'C': 457.0, 'G': 776.0, 'T': 724.0},\n",
       "  5: {'A': 170.0, 'C': 13117.0, 'G': 378.0, 'T': 1707.0},\n",
       "  6: {'A': 428.0, 'C': 168.0, 'G': 153.0, 'T': 14623.0},\n",
       "  7: {'A': 141.0, 'C': 165.0, 'G': 200.0, 'T': 14866.0},\n",
       "  8: {'A': 196.0, 'C': 14834.0, 'G': 140.0, 'T': 202.0},\n",
       "  9: {'A': 159.0, 'C': 14998.0, 'G': 102.0, 'T': 113.0},\n",
       "  10: {'A': 429.0, 'C': 609.0, 'G': 1539.0, 'T': 12795.0},\n",
       "  11: {'A': 696.0, 'C': 2984.0, 'G': 9062.0, 'T': 2630.0},\n",
       "  12: {'A': 2316.0, 'C': 3030.0, 'G': 3300.0, 'T': 6726.0},\n",
       "  13: {'A': 2248.0, 'C': 1763.0, 'G': 1265.0, 'T': 10096.0},\n",
       "  14: {'A': 3217.0, 'C': 3672.0, 'G': 3089.0, 'T': 5394.0},\n",
       "  15: {'A': 3084.0, 'C': 4830.0, 'G': 3195.0, 'T': 4263.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '11259407',\n",
       "  'uniprot_ids': 'Q9NZC4',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0047.3',\n",
       "  'Matrix_Name': 'FOXA2',\n",
       "  1: {'A': 112242.0, 'C': 34081.0, 'G': 63807.0, 'T': 52027.0},\n",
       "  2: {'A': 93741.0, 'C': 12433.0, 'G': 38539.0, 'T': 117444.0},\n",
       "  3: {'A': 18771.0, 'C': 2929.0, 'G': 236312.0, 'T': 4145.0},\n",
       "  4: {'A': 3738.0, 'C': 2038.0, 'G': 1891.0, 'T': 254490.0},\n",
       "  5: {'A': 243806.0, 'C': 9425.0, 'G': 2642.0, 'T': 6284.0},\n",
       "  6: {'A': 223062.0, 'C': 22430.0, 'G': 6485.0, 'T': 10180.0},\n",
       "  7: {'A': 248215.0, 'C': 3330.0, 'G': 2806.0, 'T': 7806.0},\n",
       "  8: {'A': 2412.0, 'C': 196970.0, 'G': 3604.0, 'T': 59171.0},\n",
       "  9: {'A': 246666.0, 'C': 3087.0, 'G': 2985.0, 'T': 9419.0},\n",
       "  10: {'A': 76951.0, 'C': 50577.0, 'G': 50528.0, 'T': 84101.0},\n",
       "  11: {'A': 85411.0, 'C': 53634.0, 'G': 56633.0, 'T': 66479.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '28644006',\n",
       "  'uniprot_ids': 'Q9Y261',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0611.1',\n",
       "  'Matrix_Name': 'Dux',\n",
       "  1: {'A': 220.0, 'C': 313.0, 'G': 251.0, 'T': 220.0},\n",
       "  2: {'A': 1.0, 'C': 870.0, 'G': 1.0, 'T': 132.0},\n",
       "  3: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  6: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 486.0, 'C': 32.0, 'G': 243.0, 'T': 243.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '24278031',\n",
       "  'uniprot_ids': 'A1JVI8',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0869.1',\n",
       "  'Matrix_Name': 'Sox11',\n",
       "  1: {'A': 306.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  2: {'A': 306.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 306.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 306.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 306.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 306.0},\n",
       "  7: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 306.0},\n",
       "  8: {'A': 42.0, 'C': 40.0, 'G': 306.0, 'T': 426.0},\n",
       "  9: {'A': 1.0, 'C': 306.0, 'G': 1.0, 'T': 2.0},\n",
       "  10: {'A': 306.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 306.0, 'T': 100.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 3.0, 'T': 306.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 306.0, 'T': 1.0},\n",
       "  14: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 306.0},\n",
       "  15: {'A': 2.0, 'C': 15.0, 'G': 6.0, 'T': 306.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '12920151',\n",
       "  'uniprot_ids': 'Q7M6Y2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0620.3',\n",
       "  'Matrix_Name': 'MITF',\n",
       "  1: {'A': 16904.0, 'C': 14428.0, 'G': 14388.0, 'T': 17811.0},\n",
       "  2: {'A': 17917.0, 'C': 14093.0, 'G': 15141.0, 'T': 16380.0},\n",
       "  3: {'A': 20980.0, 'C': 11588.0, 'G': 14439.0, 'T': 16524.0},\n",
       "  4: {'A': 23674.0, 'C': 7203.0, 'G': 18499.0, 'T': 14155.0},\n",
       "  5: {'A': 19442.0, 'C': 5206.0, 'G': 36762.0, 'T': 2121.0},\n",
       "  6: {'A': 2417.0, 'C': 4998.0, 'G': 2992.0, 'T': 53124.0},\n",
       "  7: {'A': 238.0, 'C': 62667.0, 'G': 417.0, 'T': 209.0},\n",
       "  8: {'A': 61548.0, 'C': 752.0, 'G': 470.0, 'T': 761.0},\n",
       "  9: {'A': 314.0, 'C': 47773.0, 'G': 1141.0, 'T': 14303.0},\n",
       "  10: {'A': 14307.0, 'C': 1141.0, 'G': 47769.0, 'T': 314.0},\n",
       "  11: {'A': 760.0, 'C': 470.0, 'G': 752.0, 'T': 61549.0},\n",
       "  12: {'A': 209.0, 'C': 417.0, 'G': 62667.0, 'T': 238.0},\n",
       "  13: {'A': 53125.0, 'C': 2991.0, 'G': 5000.0, 'T': 2415.0},\n",
       "  14: {'A': 2120.0, 'C': 36766.0, 'G': 5205.0, 'T': 19440.0},\n",
       "  15: {'A': 14150.0, 'C': 18499.0, 'G': 7208.0, 'T': 23674.0},\n",
       "  16: {'A': 16528.0, 'C': 14438.0, 'G': 11587.0, 'T': 20978.0},\n",
       "  17: {'A': 16379.0, 'C': 15136.0, 'G': 14098.0, 'T': 17918.0},\n",
       "  18: {'A': 17808.0, 'C': 14384.0, 'G': 14434.0, 'T': 16905.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '12086670',\n",
       "  'uniprot_ids': 'O75030',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1466.1',\n",
       "  'Matrix_Name': 'ATF6',\n",
       "  1: {'A': 88.0, 'C': 235.0, 'G': 115.0, 'T': 724.0},\n",
       "  2: {'A': 149.0, 'C': 235.0, 'G': 1159.0, 'T': 278.0},\n",
       "  3: {'A': 744.0, 'C': 193.0, 'G': 416.0, 'T': 5.0},\n",
       "  4: {'A': 1.0, 'C': 3.0, 'G': 1.0, 'T': 1159.0},\n",
       "  5: {'A': 4.0, 'C': 4.0, 'G': 1159.0, 'T': 5.0},\n",
       "  6: {'A': 1159.0, 'C': 2.0, 'G': 12.0, 'T': 3.0},\n",
       "  7: {'A': 1.0, 'C': 1159.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 3.0, 'C': 1.0, 'G': 1159.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 4.0, 'G': 1.0, 'T': 1159.0},\n",
       "  10: {'A': 26.0, 'C': 1.0, 'G': 1159.0, 'T': 16.0},\n",
       "  11: {'A': 8.0, 'C': 2.0, 'G': 1159.0, 'T': 197.0},\n",
       "  12: {'A': 31.0, 'C': 1159.0, 'G': 1.0, 'T': 13.0},\n",
       "  13: {'A': 1159.0, 'C': 57.0, 'G': 275.0, 'T': 58.0},\n",
       "  14: {'A': 241.0, 'C': 302.0, 'G': 347.0, 'T': 272.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '10856300',\n",
       "  'uniprot_ids': 'P18850',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0514.1',\n",
       "  'Matrix_Name': 'Sox3',\n",
       "  1: {'A': 1.0, 'C': 1488.0, 'G': 331.0, 'T': 251.0},\n",
       "  2: {'A': 1.0, 'C': 1553.0, 'G': 1.0, 'T': 516.0},\n",
       "  3: {'A': 263.0, 'C': 1.0, 'G': 1.0, 'T': 1806.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2068.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2068.0},\n",
       "  6: {'A': 1.0, 'C': 145.0, 'G': 1924.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 2068.0},\n",
       "  8: {'A': 1.0, 'C': 699.0, 'G': 255.0, 'T': 1116.0},\n",
       "  9: {'A': 20.0, 'C': 854.0, 'G': 23.0, 'T': 1174.0},\n",
       "  10: {'A': 43.0, 'C': 655.0, 'G': 383.0, 'T': 990.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '8625802',\n",
       "  'uniprot_ids': 'P53784',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0604.1',\n",
       "  'Matrix_Name': 'Atf1',\n",
       "  1: {'A': 463.0, 'C': 87.0, 'G': 388.0, 'T': 66.0},\n",
       "  2: {'A': 25.0, 'C': 17.0, 'G': 1.0, 'T': 961.0},\n",
       "  3: {'A': 1.0, 'C': 23.0, 'G': 804.0, 'T': 176.0},\n",
       "  4: {'A': 1001.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1001.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 8.0, 'C': 1.0, 'G': 994.0, 'T': 1.0},\n",
       "  7: {'A': 91.0, 'C': 199.0, 'G': 1.0, 'T': 713.0},\n",
       "  8: {'A': 601.0, 'C': 139.0, 'G': 164.0, 'T': 101.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '1655749',\n",
       "  'uniprot_ids': 'P81269',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1549.1',\n",
       "  'Matrix_Name': 'POU6F1(var.2)',\n",
       "  1: {'A': 2757.0, 'C': 417.0, 'G': 1931.0, 'T': 1313.0},\n",
       "  2: {'A': 165.0, 'C': 274.0, 'G': 11.0, 'T': 6414.0},\n",
       "  3: {'A': 6414.0, 'C': 180.0, 'G': 256.0, 'T': 1.0},\n",
       "  4: {'A': 6414.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 299.0, 'T': 6414.0},\n",
       "  6: {'A': 68.0, 'C': 207.0, 'G': 5800.0, 'T': 616.0},\n",
       "  7: {'A': 6414.0, 'C': 97.0, 'G': 1.0, 'T': 412.0},\n",
       "  8: {'A': 315.0, 'C': 852.0, 'G': 6414.0, 'T': 1856.0},\n",
       "  9: {'A': 1973.0, 'C': 3066.0, 'G': 3349.0, 'T': 1463.0},\n",
       "  10: {'A': 1578.0, 'C': 1239.0, 'G': 1535.0, 'T': 2065.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '26265007',\n",
       "  'uniprot_ids': 'Q14863',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0769.2',\n",
       "  'Matrix_Name': 'TCF7',\n",
       "  1: {'A': 3022.0, 'C': 3658.0, 'G': 3982.0, 'T': 4336.0},\n",
       "  2: {'A': 3111.0, 'C': 4605.0, 'G': 3199.0, 'T': 4083.0},\n",
       "  3: {'A': 253.0, 'C': 14380.0, 'G': 207.0, 'T': 158.0},\n",
       "  4: {'A': 247.0, 'C': 126.0, 'G': 68.0, 'T': 14557.0},\n",
       "  5: {'A': 303.0, 'C': 249.0, 'G': 267.0, 'T': 14179.0},\n",
       "  6: {'A': 361.0, 'C': 63.0, 'G': 84.0, 'T': 14490.0},\n",
       "  7: {'A': 267.0, 'C': 279.0, 'G': 13973.0, 'T': 479.0},\n",
       "  8: {'A': 14578.0, 'C': 64.0, 'G': 116.0, 'T': 240.0},\n",
       "  9: {'A': 4422.0, 'C': 856.0, 'G': 1221.0, 'T': 8499.0},\n",
       "  10: {'A': 2914.0, 'C': 4775.0, 'G': 5326.0, 'T': 1983.0},\n",
       "  11: {'A': 3497.0, 'C': 2741.0, 'G': 1689.0, 'T': 7071.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'TCF-7-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '28092692',\n",
       "  'uniprot_ids': 'P36402',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0734.2',\n",
       "  'Matrix_Name': 'GLI2',\n",
       "  1: {'A': 5235.0, 'C': 4902.0, 'G': 4703.0, 'T': 1608.0},\n",
       "  2: {'A': 89.0, 'C': 216.0, 'G': 15958.0, 'T': 729.0},\n",
       "  3: {'A': 14097.0, 'C': 2887.0, 'G': 1759.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 16428.0, 'G': 1.0, 'T': 9.0},\n",
       "  5: {'A': 1.0, 'C': 16443.0, 'G': 1.0, 'T': 2.0},\n",
       "  6: {'A': 15763.0, 'C': 676.0, 'G': 98.0, 'T': 303.0},\n",
       "  7: {'A': 1.0, 'C': 16445.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 3320.0, 'C': 15080.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 168.0, 'C': 16217.0, 'G': 14.0, 'T': 153.0},\n",
       "  10: {'A': 16176.0, 'C': 8.0, 'G': 540.0, 'T': 25.0},\n",
       "  11: {'A': 1208.0, 'C': 15454.0, 'G': 695.0, 'T': 842.0},\n",
       "  12: {'A': 3355.0, 'C': 4114.0, 'G': 13226.0, 'T': 1147.0},\n",
       "  13: {'A': 7614.0, 'C': 1811.0, 'G': 3254.0, 'T': 7648.0},\n",
       "  14: {'A': 4439.0, 'C': 9204.0, 'G': 1369.0, 'T': 5872.0},\n",
       "  15: {'A': 504.0, 'C': 1258.0, 'G': 15482.0, 'T': 778.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '2105456',\n",
       "  'uniprot_ids': 'P10070',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0763.1',\n",
       "  'Matrix_Name': 'ETV3',\n",
       "  1: {'A': 216.0, 'C': 28.0, 'G': 39.0, 'T': 25.0},\n",
       "  2: {'A': 19.0, 'C': 216.0, 'G': 15.0, 'T': 32.0},\n",
       "  3: {'A': 10.0, 'C': 216.0, 'G': 1.0, 'T': 14.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 216.0, 'T': 15.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 216.0, 'T': 17.0},\n",
       "  6: {'A': 216.0, 'C': 9.0, 'G': 1.0, 'T': 5.0},\n",
       "  7: {'A': 216.0, 'C': 1.0, 'G': 1.0, 'T': 72.0},\n",
       "  8: {'A': 59.0, 'C': 15.0, 'G': 216.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 33.0, 'G': 8.0, 'T': 216.0},\n",
       "  10: {'A': 53.0, 'C': 42.0, 'G': 99.0, 'T': 26.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'P41162',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0844.1',\n",
       "  'Matrix_Name': 'XBP1',\n",
       "  1: {'A': 4397.0, 'C': 559.0, 'G': 802.0, 'T': 2468.0},\n",
       "  2: {'A': 3271.0, 'C': 1018.0, 'G': 2414.0, 'T': 1781.0},\n",
       "  3: {'A': 1105.0, 'C': 1192.0, 'G': 2670.0, 'T': 4000.0},\n",
       "  4: {'A': 975.0, 'C': 368.0, 'G': 9169.0, 'T': 2322.0},\n",
       "  5: {'A': 3700.0, 'C': 3810.0, 'G': 60.0, 'T': 248.0},\n",
       "  6: {'A': 168.0, 'C': 7170.0, 'G': 223.0, 'T': 443.0},\n",
       "  7: {'A': 7841.0, 'C': 18.0, 'G': 277.0, 'T': 119.0},\n",
       "  8: {'A': 10.0, 'C': 7357.0, 'G': 3.0, 'T': 11.0},\n",
       "  9: {'A': 53.0, 'C': 1.0, 'G': 11354.0, 'T': 11.0},\n",
       "  10: {'A': 1.0, 'C': 17.0, 'G': 9.0, 'T': 9108.0},\n",
       "  11: {'A': 381.0, 'C': 7265.0, 'G': 48.0, 'T': 36.0},\n",
       "  12: {'A': 8005.0, 'C': 49.0, 'G': 174.0, 'T': 93.0},\n",
       "  13: {'A': 231.0, 'C': 2274.0, 'G': 3158.0, 'T': 6390.0},\n",
       "  14: {'A': 990.0, 'C': 6314.0, 'G': 1218.0, 'T': 1251.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'XBP-1-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15598891',\n",
       "  'uniprot_ids': 'P17861',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0124.2',\n",
       "  'Matrix_Name': 'Nkx3-1',\n",
       "  1: {'A': 2439.0, 'C': 992.0, 'G': 1362.0, 'T': 545.0},\n",
       "  2: {'A': 982.0, 'C': 5336.0, 'G': 1777.0, 'T': 276.0},\n",
       "  3: {'A': 1.0, 'C': 5336.0, 'G': 11.0, 'T': 337.0},\n",
       "  4: {'A': 5336.0, 'C': 29.0, 'G': 11.0, 'T': 89.0},\n",
       "  5: {'A': 60.0, 'C': 5336.0, 'G': 10.0, 'T': 10.0},\n",
       "  6: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 5336.0},\n",
       "  7: {'A': 1.0, 'C': 255.0, 'G': 1.0, 'T': 5336.0},\n",
       "  8: {'A': 5336.0, 'C': 274.0, 'G': 174.0, 'T': 524.0},\n",
       "  9: {'A': 5336.0, 'C': 1408.0, 'G': 2133.0, 'T': 1510.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P97436',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0155.1',\n",
       "  'Matrix_Name': 'INSM1',\n",
       "  1: {'A': 2.0, 'C': 1.0, 'G': 5.0, 'T': 20.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 21.0, 'T': 5.0},\n",
       "  3: {'A': 1.0, 'C': 9.0, 'G': 4.0, 'T': 14.0},\n",
       "  4: {'A': 7.0, 'C': 16.0, 'G': 1.0, 'T': 4.0},\n",
       "  5: {'A': 17.0, 'C': 1.0, 'G': 1.0, 'T': 9.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 25.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 2.0, 'G': 24.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 25.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 25.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 3.0, 'G': 17.0, 'T': 7.0},\n",
       "  11: {'A': 4.0, 'C': 17.0, 'G': 1.0, 'T': 6.0},\n",
       "  12: {'A': 11.0, 'C': 1.0, 'G': 13.0, 'T': 3.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'Q01101',\n",
       "  'data_type': 'COMPILED'},\n",
       " {'Matrix_ID': 'MA1640.1',\n",
       "  'Matrix_Name': 'MEIS2(var.2)',\n",
       "  1: {'A': 6213.0, 'C': 3312.0, 'G': 5769.0, 'T': 4883.0},\n",
       "  2: {'A': 6446.0, 'C': 4042.0, 'G': 4684.0, 'T': 5005.0},\n",
       "  3: {'A': 9930.0, 'C': 2366.0, 'G': 3897.0, 'T': 3984.0},\n",
       "  4: {'A': 3886.0, 'C': 469.0, 'G': 1401.0, 'T': 14421.0},\n",
       "  5: {'A': 1479.0, 'C': 303.0, 'G': 17327.0, 'T': 1068.0},\n",
       "  6: {'A': 19362.0, 'C': 208.0, 'G': 225.0, 'T': 382.0},\n",
       "  7: {'A': 577.0, 'C': 269.0, 'G': 500.0, 'T': 18831.0},\n",
       "  8: {'A': 713.0, 'C': 202.0, 'G': 279.0, 'T': 18983.0},\n",
       "  9: {'A': 1624.0, 'C': 583.0, 'G': 1802.0, 'T': 16168.0},\n",
       "  10: {'A': 19345.0, 'C': 178.0, 'G': 496.0, 'T': 158.0},\n",
       "  11: {'A': 475.0, 'C': 1054.0, 'G': 503.0, 'T': 18145.0},\n",
       "  12: {'A': 2624.0, 'C': 1090.0, 'G': 12863.0, 'T': 3600.0},\n",
       "  13: {'A': 5763.0, 'C': 1610.0, 'G': 8200.0, 'T': 4604.0},\n",
       "  14: {'A': 3883.0, 'C': 6519.0, 'G': 4177.0, 'T': 5598.0},\n",
       "  15: {'A': 4557.0, 'C': 5249.0, 'G': 3362.0, 'T': 7009.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23602564',\n",
       "  'uniprot_ids': 'O14770',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1509.1',\n",
       "  'Matrix_Name': 'IRF6',\n",
       "  1: {'A': 2653.0, 'C': 165.0, 'G': 649.0, 'T': 90.0},\n",
       "  2: {'A': 68.0, 'C': 2653.0, 'G': 63.0, 'T': 562.0},\n",
       "  3: {'A': 1.0, 'C': 2653.0, 'G': 23.0, 'T': 20.0},\n",
       "  4: {'A': 49.0, 'C': 6.0, 'G': 2653.0, 'T': 10.0},\n",
       "  5: {'A': 2653.0, 'C': 1.0, 'G': 1.0, 'T': 27.0},\n",
       "  6: {'A': 2315.0, 'C': 1.0, 'G': 10.0, 'T': 339.0},\n",
       "  7: {'A': 2653.0, 'C': 1.0, 'G': 18.0, 'T': 3.0},\n",
       "  8: {'A': 1.0, 'C': 2653.0, 'G': 13.0, 'T': 1.0},\n",
       "  9: {'A': 99.0, 'C': 369.0, 'G': 8.0, 'T': 2285.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '18604160',\n",
       "  'uniprot_ids': 'O14896',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0868.2',\n",
       "  'Matrix_Name': 'SOX8',\n",
       "  1: {'A': 1828.0, 'C': 1125.0, 'G': 855.0, 'T': 725.0},\n",
       "  2: {'A': 741.0, 'C': 415.0, 'G': 1828.0, 'T': 572.0},\n",
       "  3: {'A': 1828.0, 'C': 152.0, 'G': 264.0, 'T': 223.0},\n",
       "  4: {'A': 1828.0, 'C': 1.0, 'G': 1.0, 'T': 46.0},\n",
       "  5: {'A': 4.0, 'C': 1828.0, 'G': 36.0, 'T': 71.0},\n",
       "  6: {'A': 1828.0, 'C': 1.0, 'G': 1.0, 'T': 17.0},\n",
       "  7: {'A': 1828.0, 'C': 44.0, 'G': 1.0, 'T': 37.0},\n",
       "  8: {'A': 274.0, 'C': 37.0, 'G': 21.0, 'T': 1828.0},\n",
       "  9: {'A': 1357.0, 'C': 209.0, 'G': 1828.0, 'T': 288.0},\n",
       "  10: {'A': 301.0, 'C': 422.0, 'G': 913.0, 'T': 194.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '17599239',\n",
       "  'uniprot_ids': 'P57073',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1483.1',\n",
       "  'Matrix_Name': 'ELF2',\n",
       "  1: {'A': 9935.0, 'C': 2914.0, 'G': 3383.0, 'T': 4124.0},\n",
       "  2: {'A': 20353.0, 'C': 1982.0, 'G': 429.0, 'T': 3942.0},\n",
       "  3: {'A': 6115.0, 'C': 20353.0, 'G': 981.0, 'T': 2373.0},\n",
       "  4: {'A': 545.0, 'C': 20353.0, 'G': 1807.0, 'T': 18.0},\n",
       "  5: {'A': 593.0, 'C': 20353.0, 'G': 6.0, 'T': 8.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 20353.0, 'T': 1.0},\n",
       "  7: {'A': 22.0, 'C': 3.0, 'G': 20353.0, 'T': 9.0},\n",
       "  8: {'A': 20353.0, 'C': 6.0, 'G': 7.0, 'T': 1.0},\n",
       "  9: {'A': 20353.0, 'C': 7.0, 'G': 1.0, 'T': 90.0},\n",
       "  10: {'A': 3124.0, 'C': 384.0, 'G': 20353.0, 'T': 1.0},\n",
       "  11: {'A': 947.0, 'C': 2944.0, 'G': 178.0, 'T': 20353.0},\n",
       "  12: {'A': 8042.0, 'C': 2744.0, 'G': 7279.0, 'T': 2292.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'Q15723',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1625.1',\n",
       "  'Matrix_Name': 'Stat5b',\n",
       "  1: {'A': 615.0, 'C': 763.0, 'G': 683.0, 'T': 703.0},\n",
       "  2: {'A': 909.0, 'C': 577.0, 'G': 648.0, 'T': 630.0},\n",
       "  3: {'A': 701.0, 'C': 608.0, 'G': 538.0, 'T': 917.0},\n",
       "  4: {'A': 52.0, 'C': 55.0, 'G': 52.0, 'T': 2605.0},\n",
       "  5: {'A': 93.0, 'C': 35.0, 'G': 35.0, 'T': 2601.0},\n",
       "  6: {'A': 33.0, 'C': 2667.0, 'G': 24.0, 'T': 40.0},\n",
       "  7: {'A': 75.0, 'C': 2177.0, 'G': 44.0, 'T': 468.0},\n",
       "  8: {'A': 463.0, 'C': 1747.0, 'G': 128.0, 'T': 426.0},\n",
       "  9: {'A': 2475.0, 'C': 61.0, 'G': 151.0, 'T': 77.0},\n",
       "  10: {'A': 11.0, 'C': 17.0, 'G': 2718.0, 'T': 18.0},\n",
       "  11: {'A': 2555.0, 'C': 36.0, 'G': 87.0, 'T': 86.0},\n",
       "  12: {'A': 2665.0, 'C': 26.0, 'G': 51.0, 'T': 22.0},\n",
       "  13: {'A': 902.0, 'C': 616.0, 'G': 712.0, 'T': 534.0},\n",
       "  14: {'A': 687.0, 'C': 681.0, 'G': 543.0, 'T': 853.0},\n",
       "  15: {'A': 664.0, 'C': 769.0, 'G': 730.0, 'T': 601.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '10594041',\n",
       "  'uniprot_ids': 'P42232',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0259.1',\n",
       "  'Matrix_Name': 'ARNT::HIF1A',\n",
       "  1: {'A': 28.0, 'C': 29.0, 'G': 50.0, 'T': 1.0},\n",
       "  2: {'A': 11.0, 'C': 30.0, 'G': 35.0, 'T': 32.0},\n",
       "  3: {'A': 79.0, 'C': 3.0, 'G': 24.0, 'T': 2.0},\n",
       "  4: {'A': 1.0, 'C': 104.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 105.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 105.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 105.0, 'T': 1.0},\n",
       "  8: {'A': 19.0, 'C': 52.0, 'G': 21.0, 'T': 16.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16234508',\n",
       "  'uniprot_ids': 'P27540;',\n",
       "  'data_type': 'COMPILED'},\n",
       " {'Matrix_ID': 'MA0472.2',\n",
       "  'Matrix_Name': 'EGR2',\n",
       "  1: {'A': 1158.0, 'C': 680.0, 'G': 95.0, 'T': 143.0},\n",
       "  2: {'A': 1.0, 'C': 1552.0, 'G': 29.0, 'T': 74.0},\n",
       "  3: {'A': 296.0, 'C': 1.0, 'G': 3732.0, 'T': 223.0},\n",
       "  4: {'A': 7.0, 'C': 1559.0, 'G': 26.0, 'T': 30.0},\n",
       "  5: {'A': 25.0, 'C': 1555.0, 'G': 16.0, 'T': 14.0},\n",
       "  6: {'A': 1.0, 'C': 1469.0, 'G': 39.0, 'T': 278.0},\n",
       "  7: {'A': 2148.0, 'C': 111.0, 'G': 27.0, 'T': 3.0},\n",
       "  8: {'A': 1.0, 'C': 1574.0, 'G': 3.0, 'T': 24.0},\n",
       "  9: {'A': 107.0, 'C': 1.0, 'G': 4035.0, 'T': 172.0},\n",
       "  10: {'A': 5.0, 'C': 1586.0, 'G': 21.0, 'T': 116.0},\n",
       "  11: {'A': 1732.0, 'C': 158.0, 'G': 543.0, 'T': 431.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7891721',\n",
       "  'uniprot_ids': 'P11161',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0108.2',\n",
       "  'Matrix_Name': 'TBP',\n",
       "  1: {'A': 62.0, 'C': 146.0, 'G': 153.0, 'T': 32.0},\n",
       "  2: {'A': 17.0, 'C': 47.0, 'G': 19.0, 'T': 310.0},\n",
       "  3: {'A': 353.0, 'C': 1.0, 'G': 3.0, 'T': 36.0},\n",
       "  4: {'A': 4.0, 'C': 11.0, 'G': 3.0, 'T': 375.0},\n",
       "  5: {'A': 355.0, 'C': 1.0, 'G': 6.0, 'T': 31.0},\n",
       "  6: {'A': 269.0, 'C': 1.0, 'G': 1.0, 'T': 122.0},\n",
       "  7: {'A': 361.0, 'C': 4.0, 'G': 21.0, 'T': 7.0},\n",
       "  8: {'A': 223.0, 'C': 3.0, 'G': 45.0, 'T': 122.0},\n",
       "  9: {'A': 156.0, 'C': 45.0, 'G': 158.0, 'T': 34.0},\n",
       "  10: {'A': 57.0, 'C': 136.0, 'G': 151.0, 'T': 49.0},\n",
       "  11: {'A': 84.0, 'C': 148.0, 'G': 129.0, 'T': 32.0},\n",
       "  12: {'A': 83.0, 'C': 128.0, 'G': 129.0, 'T': 53.0},\n",
       "  13: {'A': 83.0, 'C': 119.0, 'G': 129.0, 'T': 62.0},\n",
       "  14: {'A': 69.0, 'C': 108.0, 'G': 140.0, 'T': 76.0},\n",
       "  15: {'A': 78.0, 'C': 102.0, 'G': 141.0, 'T': 72.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'TBP-related',\n",
       "  'tf_class': 'TATA-binding',\n",
       "  'pubmed_ids': '2329577',\n",
       "  'uniprot_ids': 'P20226',\n",
       "  'data_type': ''},\n",
       " {'Matrix_ID': 'MA1567.1',\n",
       "  'Matrix_Name': 'TBX6',\n",
       "  1: {'A': 2999.0, 'C': 1631.0, 'G': 4004.0, 'T': 2457.0},\n",
       "  2: {'A': 11088.0, 'C': 1174.0, 'G': 3348.0, 'T': 2158.0},\n",
       "  3: {'A': 1883.0, 'C': 2914.0, 'G': 11088.0, 'T': 1059.0},\n",
       "  4: {'A': 59.0, 'C': 64.0, 'G': 11088.0, 'T': 1.0},\n",
       "  5: {'A': 45.0, 'C': 1003.0, 'G': 1.0, 'T': 11088.0},\n",
       "  6: {'A': 15.0, 'C': 1.0, 'G': 11088.0, 'T': 13.0},\n",
       "  7: {'A': 553.0, 'C': 3054.0, 'G': 321.0, 'T': 11088.0},\n",
       "  8: {'A': 1431.0, 'C': 4877.0, 'G': 11088.0, 'T': 5965.0},\n",
       "  9: {'A': 11088.0, 'C': 653.0, 'G': 2252.0, 'T': 714.0},\n",
       "  10: {'A': 4551.0, 'C': 2067.0, 'G': 2142.0, 'T': 2331.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'TBX6-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'O95947',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0914.1',\n",
       "  'Matrix_Name': 'ISL2',\n",
       "  1: {'A': 1586.0, 'C': 2617.0, 'G': 5632.0, 'T': 1103.0},\n",
       "  2: {'A': 161.0, 'C': 5632.0, 'G': 127.0, 'T': 1020.0},\n",
       "  3: {'A': 5632.0, 'C': 163.0, 'G': 135.0, 'T': 249.0},\n",
       "  4: {'A': 3162.0, 'C': 5632.0, 'G': 172.0, 'T': 399.0},\n",
       "  5: {'A': 219.0, 'C': 131.0, 'G': 170.0, 'T': 5632.0},\n",
       "  6: {'A': 279.0, 'C': 719.0, 'G': 14.0, 'T': 5632.0},\n",
       "  7: {'A': 5632.0, 'C': 207.0, 'G': 60.0, 'T': 1355.0},\n",
       "  8: {'A': 5632.0, 'C': 1232.0, 'G': 3259.0, 'T': 1509.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '7999775',\n",
       "  'uniprot_ids': 'Q96A47',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0908.1',\n",
       "  'Matrix_Name': 'HOXD11',\n",
       "  1: {'A': 129.0, 'C': 55.0, 'G': 231.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 231.0},\n",
       "  3: {'A': 17.0, 'C': 231.0, 'G': 1.0, 'T': 24.0},\n",
       "  4: {'A': 69.0, 'C': 1.0, 'G': 231.0, 'T': 1.0},\n",
       "  5: {'A': 9.0, 'C': 1.0, 'G': 1.0, 'T': 231.0},\n",
       "  6: {'A': 176.0, 'C': 1.0, 'G': 1.0, 'T': 57.0},\n",
       "  7: {'A': 231.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  8: {'A': 231.0, 'C': 5.0, 'G': 1.0, 'T': 8.0},\n",
       "  9: {'A': 231.0, 'C': 30.0, 'G': 29.0, 'T': 72.0},\n",
       "  10: {'A': 89.0, 'C': 52.0, 'G': 24.0, 'T': 70.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31277',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0890.1',\n",
       "  'Matrix_Name': 'GBX2',\n",
       "  1: {'A': 10335.0, 'C': 5148.0, 'G': 8146.0, 'T': 3741.0},\n",
       "  2: {'A': 3762.0, 'C': 10518.0, 'G': 8430.0, 'T': 4659.0},\n",
       "  3: {'A': 700.0, 'C': 15512.0, 'G': 107.0, 'T': 11856.0},\n",
       "  4: {'A': 27367.0, 'C': 571.0, 'G': 1114.0, 'T': 228.0},\n",
       "  5: {'A': 27367.0, 'C': 251.0, 'G': 115.0, 'T': 26.0},\n",
       "  6: {'A': 1.0, 'C': 62.0, 'G': 66.0, 'T': 27367.0},\n",
       "  7: {'A': 319.0, 'C': 571.0, 'G': 674.0, 'T': 27367.0},\n",
       "  8: {'A': 27367.0, 'C': 74.0, 'G': 31.0, 'T': 651.0},\n",
       "  9: {'A': 8237.0, 'C': 4503.0, 'G': 10639.0, 'T': 3992.0},\n",
       "  10: {'A': 5390.0, 'C': 9160.0, 'G': 7076.0, 'T': 5744.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P52951',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0628.1',\n",
       "  'Matrix_Name': 'POU6F1',\n",
       "  1: {'A': 508.0, 'C': 194.0, 'G': 212.0, 'T': 90.0},\n",
       "  2: {'A': 152.0, 'C': 352.0, 'G': 146.0, 'T': 354.0},\n",
       "  3: {'A': 33.0, 'C': 11.0, 'G': 9.0, 'T': 951.0},\n",
       "  4: {'A': 906.0, 'C': 71.0, 'G': 9.0, 'T': 19.0},\n",
       "  5: {'A': 939.0, 'C': 35.0, 'G': 15.0, 'T': 15.0},\n",
       "  6: {'A': 15.0, 'C': 15.0, 'G': 35.0, 'T': 939.0},\n",
       "  7: {'A': 19.0, 'C': 9.0, 'G': 71.0, 'T': 906.0},\n",
       "  8: {'A': 951.0, 'C': 9.0, 'G': 11.0, 'T': 33.0},\n",
       "  9: {'A': 354.0, 'C': 146.0, 'G': 352.0, 'T': 152.0},\n",
       "  10: {'A': 90.0, 'C': 212.0, 'G': 194.0, 'T': 508.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9852081',\n",
       "  'uniprot_ids': 'Q14863',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0154.4',\n",
       "  'Matrix_Name': 'EBF1',\n",
       "  1: {'A': 14161.0, 'C': 8869.0, 'G': 10892.0, 'T': 11507.0},\n",
       "  2: {'A': 11404.0, 'C': 9555.0, 'G': 13482.0, 'T': 10988.0},\n",
       "  3: {'A': 2353.0, 'C': 4507.0, 'G': 5498.0, 'T': 33071.0},\n",
       "  4: {'A': 1322.0, 'C': 34666.0, 'G': 1921.0, 'T': 7520.0},\n",
       "  5: {'A': 569.0, 'C': 43257.0, 'G': 372.0, 'T': 1231.0},\n",
       "  6: {'A': 649.0, 'C': 42563.0, 'G': 444.0, 'T': 1773.0},\n",
       "  7: {'A': 2540.0, 'C': 39676.0, 'G': 863.0, 'T': 2350.0},\n",
       "  8: {'A': 34746.0, 'C': 1708.0, 'G': 1724.0, 'T': 7251.0},\n",
       "  9: {'A': 3503.0, 'C': 815.0, 'G': 38729.0, 'T': 2382.0},\n",
       "  10: {'A': 2618.0, 'C': 496.0, 'G': 41652.0, 'T': 663.0},\n",
       "  11: {'A': 1516.0, 'C': 456.0, 'G': 42790.0, 'T': 667.0},\n",
       "  12: {'A': 8857.0, 'C': 2463.0, 'G': 32762.0, 'T': 1347.0},\n",
       "  13: {'A': 31637.0, 'C': 6356.0, 'G': 4958.0, 'T': 2478.0},\n",
       "  14: {'A': 11230.0, 'C': 13300.0, 'G': 9856.0, 'T': 11043.0},\n",
       "  15: {'A': 11904.0, 'C': 10837.0, 'G': 9174.0, 'T': 13514.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Early',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '8497258',\n",
       "  'uniprot_ids': 'Q9UH73',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0708.1',\n",
       "  'Matrix_Name': 'MSX2',\n",
       "  1: {'A': 518.0, 'C': 1689.0, 'G': 836.0, 'T': 361.0},\n",
       "  2: {'A': 106.0, 'C': 2593.0, 'G': 45.0, 'T': 808.0},\n",
       "  3: {'A': 3401.0, 'C': 162.0, 'G': 68.0, 'T': 85.0},\n",
       "  4: {'A': 3401.0, 'C': 29.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 3401.0},\n",
       "  6: {'A': 53.0, 'C': 170.0, 'G': 73.0, 'T': 3401.0},\n",
       "  7: {'A': 3401.0, 'C': 62.0, 'G': 1.0, 'T': 22.0},\n",
       "  8: {'A': 1199.0, 'C': 792.0, 'G': 893.0, 'T': 519.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P35548',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0145.3',\n",
       "  'Matrix_Name': 'TFCP2',\n",
       "  1: {'A': 340.0, 'C': 62.0, 'G': 130.0, 'T': 48.0},\n",
       "  2: {'A': 576.0, 'C': 7.0, 'G': 25.0, 'T': 33.0},\n",
       "  3: {'A': 576.0, 'C': 3.0, 'G': 9.0, 'T': 103.0},\n",
       "  4: {'A': 2.0, 'C': 576.0, 'G': 3.0, 'T': 1.0},\n",
       "  5: {'A': 6.0, 'C': 576.0, 'G': 2.0, 'T': 120.0},\n",
       "  6: {'A': 130.0, 'C': 4.0, 'G': 576.0, 'T': 8.0},\n",
       "  7: {'A': 2.0, 'C': 10.0, 'G': 576.0, 'T': 2.0},\n",
       "  8: {'A': 190.0, 'C': 36.0, 'G': 1.0, 'T': 576.0},\n",
       "  9: {'A': 47.0, 'C': 61.0, 'G': 21.0, 'T': 576.0},\n",
       "  10: {'A': 97.0, 'C': 197.0, 'G': 32.0, 'T': 253.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'CP2-related',\n",
       "  'tf_class': 'Grainyhead',\n",
       "  'pubmed_ids': '9668115',\n",
       "  'uniprot_ids': 'Q12800',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1572.1',\n",
       "  'Matrix_Name': 'TGIF2LY',\n",
       "  1: {'A': 737.0, 'C': 837.0, 'G': 416.0, 'T': 5210.0},\n",
       "  2: {'A': 163.0, 'C': 271.0, 'G': 5210.0, 'T': 105.0},\n",
       "  3: {'A': 5210.0, 'C': 164.0, 'G': 350.0, 'T': 1198.0},\n",
       "  4: {'A': 124.0, 'C': 5210.0, 'G': 67.0, 'T': 145.0},\n",
       "  5: {'A': 5210.0, 'C': 38.0, 'G': 834.0, 'T': 111.0},\n",
       "  6: {'A': 131.0, 'C': 1268.0, 'G': 2795.0, 'T': 1019.0},\n",
       "  7: {'A': 1034.0, 'C': 2851.0, 'G': 1189.0, 'T': 139.0},\n",
       "  8: {'A': 141.0, 'C': 849.0, 'G': 74.0, 'T': 5210.0},\n",
       "  9: {'A': 176.0, 'C': 62.0, 'G': 5210.0, 'T': 122.0},\n",
       "  10: {'A': 1085.0, 'C': 372.0, 'G': 146.0, 'T': 5210.0},\n",
       "  11: {'A': 136.0, 'C': 5210.0, 'G': 260.0, 'T': 173.0},\n",
       "  12: {'A': 5210.0, 'C': 363.0, 'G': 808.0, 'T': 680.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8537382',\n",
       "  'uniprot_ids': 'Q8IUE0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0093.3',\n",
       "  'Matrix_Name': 'USF1',\n",
       "  1: {'A': 14421.0, 'C': 10467.0, 'G': 14649.0, 'T': 15801.0},\n",
       "  2: {'A': 16096.0, 'C': 6277.0, 'G': 20789.0, 'T': 12176.0},\n",
       "  3: {'A': 7575.0, 'C': 3031.0, 'G': 43095.0, 'T': 1637.0},\n",
       "  4: {'A': 3842.0, 'C': 9544.0, 'G': 3913.0, 'T': 38039.0},\n",
       "  5: {'A': 534.0, 'C': 53537.0, 'G': 767.0, 'T': 500.0},\n",
       "  6: {'A': 53655.0, 'C': 655.0, 'G': 478.0, 'T': 550.0},\n",
       "  7: {'A': 419.0, 'C': 13186.0, 'G': 3667.0, 'T': 38066.0},\n",
       "  8: {'A': 3454.0, 'C': 1388.0, 'G': 50077.0, 'T': 419.0},\n",
       "  9: {'A': 449.0, 'C': 265.0, 'G': 602.0, 'T': 54022.0},\n",
       "  10: {'A': 404.0, 'C': 818.0, 'G': 53635.0, 'T': 481.0},\n",
       "  11: {'A': 45588.0, 'C': 3046.0, 'G': 4680.0, 'T': 2024.0},\n",
       "  12: {'A': 1846.0, 'C': 41451.0, 'G': 3019.0, 'T': 9022.0},\n",
       "  13: {'A': 11281.0, 'C': 19457.0, 'G': 6944.0, 'T': 17656.0},\n",
       "  14: {'A': 14088.0, 'C': 14476.0, 'G': 10334.0, 'T': 16440.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P22415',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0065.2',\n",
       "  'Matrix_Name': 'Pparg::Rxra',\n",
       "  1: {'A': 95.0, 'C': 318.0, 'G': 321.0, 'T': 127.0},\n",
       "  2: {'A': 102.0, 'C': 167.0, 'G': 128.0, 'T': 465.0},\n",
       "  3: {'A': 391.0, 'C': 24.0, 'G': 369.0, 'T': 80.0},\n",
       "  4: {'A': 101.0, 'C': 4.0, 'G': 672.0, 'T': 88.0},\n",
       "  5: {'A': 140.0, 'C': 16.0, 'G': 675.0, 'T': 35.0},\n",
       "  6: {'A': 146.0, 'C': 130.0, 'G': 396.0, 'T': 194.0},\n",
       "  7: {'A': 72.0, 'C': 548.0, 'G': 180.0, 'T': 67.0},\n",
       "  8: {'A': 820.0, 'C': 22.0, 'G': 16.0, 'T': 9.0},\n",
       "  9: {'A': 523.0, 'C': 49.0, 'G': 271.0, 'T': 24.0},\n",
       "  10: {'A': 714.0, 'C': 6.0, 'G': 138.0, 'T': 10.0},\n",
       "  11: {'A': 83.0, 'C': 3.0, 'G': 768.0, 'T': 13.0},\n",
       "  12: {'A': 42.0, 'C': 10.0, 'G': 694.0, 'T': 121.0},\n",
       "  13: {'A': 23.0, 'C': 100.0, 'G': 264.0, 'T': 480.0},\n",
       "  14: {'A': 55.0, 'C': 556.0, 'G': 145.0, 'T': 110.0},\n",
       "  15: {'A': 677.0, 'C': 59.0, 'G': 48.0, 'T': 82.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '18981474',\n",
       "  'uniprot_ids': 'P37238;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0007.3',\n",
       "  'Matrix_Name': 'Ar',\n",
       "  1: {'A': 1095.0, 'C': 168.0, 'G': 1556.0, 'T': 287.0},\n",
       "  2: {'A': 940.0, 'C': 18.0, 'G': 2222.0, 'T': 20.0},\n",
       "  3: {'A': 9.0, 'C': 1.0, 'G': 3838.0, 'T': 7.0},\n",
       "  4: {'A': 1198.0, 'C': 292.0, 'G': 185.0, 'T': 1028.0},\n",
       "  5: {'A': 2168.0, 'C': 6.0, 'G': 4.0, 'T': 4.0},\n",
       "  6: {'A': 5.0, 'C': 2190.0, 'G': 6.0, 'T': 3.0},\n",
       "  7: {'A': 1998.0, 'C': 21.0, 'G': 274.0, 'T': 12.0},\n",
       "  8: {'A': 200.0, 'C': 1229.0, 'G': 329.0, 'T': 713.0},\n",
       "  9: {'A': 562.0, 'C': 612.0, 'G': 992.0, 'T': 510.0},\n",
       "  10: {'A': 756.0, 'C': 251.0, 'G': 1674.0, 'T': 208.0},\n",
       "  11: {'A': 34.0, 'C': 278.0, 'G': 9.0, 'T': 2530.0},\n",
       "  12: {'A': 1.0, 'C': 5.0, 'G': 3619.0, 'T': 2.0},\n",
       "  13: {'A': 23.0, 'C': 5.0, 'G': 1.0, 'T': 2769.0},\n",
       "  14: {'A': 1196.0, 'C': 113.0, 'G': 231.0, 'T': 1095.0},\n",
       "  15: {'A': 8.0, 'C': 2167.0, 'G': 1.0, 'T': 9.0},\n",
       "  16: {'A': 33.0, 'C': 1543.0, 'G': 3.0, 'T': 830.0},\n",
       "  17: {'A': 431.0, 'C': 1093.0, 'G': 344.0, 'T': 914.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Steroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '20943813',\n",
       "  'uniprot_ids': 'P19091',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1133.1',\n",
       "  'Matrix_Name': 'JUN::JUNB(var.2)',\n",
       "  1: {'A': 55.0, 'C': 133.0, 'G': 359.0, 'T': 458.0},\n",
       "  2: {'A': 377.0, 'C': 109.0, 'G': 487.0, 'T': 31.0},\n",
       "  3: {'A': 59.0, 'C': 44.0, 'G': 26.0, 'T': 876.0},\n",
       "  4: {'A': 52.0, 'C': 29.0, 'G': 704.0, 'T': 219.0},\n",
       "  5: {'A': 788.0, 'C': 23.0, 'G': 64.0, 'T': 130.0},\n",
       "  6: {'A': 24.0, 'C': 863.0, 'G': 48.0, 'T': 69.0},\n",
       "  7: {'A': 56.0, 'C': 37.0, 'G': 868.0, 'T': 43.0},\n",
       "  8: {'A': 31.0, 'C': 28.0, 'G': 41.0, 'T': 904.0},\n",
       "  9: {'A': 92.0, 'C': 835.0, 'G': 36.0, 'T': 41.0},\n",
       "  10: {'A': 920.0, 'C': 20.0, 'G': 36.0, 'T': 27.0},\n",
       "  11: {'A': 34.0, 'C': 126.0, 'G': 131.0, 'T': 713.0},\n",
       "  12: {'A': 213.0, 'C': 255.0, 'G': 256.0, 'T': 279.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '15240010',\n",
       "  'uniprot_ids': 'P05412;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0641.1',\n",
       "  'Matrix_Name': 'ELF4',\n",
       "  1: {'A': 599.0, 'C': 110.0, 'G': 161.0, 'T': 193.0},\n",
       "  2: {'A': 706.0, 'C': 74.0, 'G': 16.0, 'T': 150.0},\n",
       "  3: {'A': 199.0, 'C': 931.0, 'G': 76.0, 'T': 107.0},\n",
       "  4: {'A': 18.0, 'C': 991.0, 'G': 106.0, 'T': 2.0},\n",
       "  5: {'A': 55.0, 'C': 998.0, 'G': 7.0, 'T': 1.0},\n",
       "  6: {'A': 2.0, 'C': 1.0, 'G': 1227.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1227.0, 'T': 3.0},\n",
       "  8: {'A': 740.0, 'C': 1.0, 'G': 3.0, 'T': 3.0},\n",
       "  9: {'A': 738.0, 'C': 6.0, 'G': 3.0, 'T': 5.0},\n",
       "  10: {'A': 44.0, 'C': 19.0, 'G': 1208.0, 'T': 7.0},\n",
       "  11: {'A': 5.0, 'C': 57.0, 'G': 29.0, 'T': 980.0},\n",
       "  12: {'A': 404.0, 'C': 84.0, 'G': 464.0, 'T': 114.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '20517297',\n",
       "  'uniprot_ids': 'Q99607',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0877.2',\n",
       "  'Matrix_Name': 'BARHL1',\n",
       "  1: {'A': 838.0, 'C': 1890.0, 'G': 1281.0, 'T': 719.0},\n",
       "  2: {'A': 194.0, 'C': 1.0, 'G': 1.0, 'T': 4726.0},\n",
       "  3: {'A': 4726.0, 'C': 66.0, 'G': 276.0, 'T': 1233.0},\n",
       "  4: {'A': 4726.0, 'C': 56.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 4726.0, 'C': 200.0, 'G': 663.0, 'T': 2115.0},\n",
       "  6: {'A': 1.0, 'C': 4726.0, 'G': 1.0, 'T': 2009.0},\n",
       "  7: {'A': 623.0, 'C': 1.0, 'G': 4726.0, 'T': 260.0},\n",
       "  8: {'A': 1066.0, 'C': 1002.0, 'G': 1509.0, 'T': 1152.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9BZE3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0111.1',\n",
       "  'Matrix_Name': 'Spz1',\n",
       "  1: {'A': 10.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  2: {'A': 1.0, 'C': 3.0, 'G': 10.0, 'T': 2.0},\n",
       "  3: {'A': 3.0, 'C': 1.0, 'G': 11.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 12.0, 'T': 2.0},\n",
       "  5: {'A': 2.0, 'C': 1.0, 'G': 2.0, 'T': 11.0},\n",
       "  6: {'A': 9.0, 'C': 1.0, 'G': 1.0, 'T': 5.0},\n",
       "  7: {'A': 7.0, 'C': 1.0, 'G': 3.0, 'T': 5.0},\n",
       "  8: {'A': 2.0, 'C': 10.0, 'G': 3.0, 'T': 1.0},\n",
       "  9: {'A': 11.0, 'C': 1.0, 'G': 3.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 2.0, 'G': 10.0, 'T': 3.0},\n",
       "  11: {'A': 3.0, 'C': 9.0, 'G': 3.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': '',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11165476',\n",
       "  'uniprot_ids': 'Q99MY0',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0860.1',\n",
       "  'Matrix_Name': 'Rarg(var.2)',\n",
       "  1: {'A': 4075.0, 'C': 11.0, 'G': 807.0, 'T': 32.0},\n",
       "  2: {'A': 4544.0, 'C': 8.0, 'G': 268.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 5.0, 'G': 5889.0, 'T': 3.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 5574.0, 'T': 201.0},\n",
       "  5: {'A': 1.0, 'C': 12.0, 'G': 13.0, 'T': 4310.0},\n",
       "  6: {'A': 3.0, 'C': 6257.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 4045.0, 'C': 4.0, 'G': 13.0, 'T': 1.0},\n",
       "  8: {'A': 1395.0, 'C': 2678.0, 'G': 387.0, 'T': 781.0},\n",
       "  9: {'A': 315.0, 'C': 1779.0, 'G': 2443.0, 'T': 596.0},\n",
       "  10: {'A': 3249.0, 'C': 483.0, 'G': 559.0, 'T': 480.0},\n",
       "  11: {'A': 2746.0, 'C': 59.0, 'G': 2134.0, 'T': 58.0},\n",
       "  12: {'A': 3604.0, 'C': 33.0, 'G': 839.0, 'T': 3.0},\n",
       "  13: {'A': 1.0, 'C': 1.0, 'G': 5892.0, 'T': 2.0},\n",
       "  14: {'A': 1.0, 'C': 9.0, 'G': 5041.0, 'T': 320.0},\n",
       "  15: {'A': 15.0, 'C': 3.0, 'G': 1.0, 'T': 4541.0},\n",
       "  16: {'A': 8.0, 'C': 5341.0, 'G': 3.0, 'T': 23.0},\n",
       "  17: {'A': 4765.0, 'C': 6.0, 'G': 7.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '15019979',\n",
       "  'uniprot_ids': 'P18911',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0791.1',\n",
       "  'Matrix_Name': 'POU4F3',\n",
       "  1: {'A': 1664.0, 'C': 636.0, 'G': 868.0, 'T': 762.0},\n",
       "  2: {'A': 293.0, 'C': 322.0, 'G': 304.0, 'T': 4135.0},\n",
       "  3: {'A': 1068.0, 'C': 375.0, 'G': 4724.0, 'T': 1068.0},\n",
       "  4: {'A': 1869.0, 'C': 2015.0, 'G': 25.0, 'T': 70.0},\n",
       "  5: {'A': 3265.0, 'C': 43.0, 'G': 37.0, 'T': 133.0},\n",
       "  6: {'A': 97.0, 'C': 22.0, 'G': 26.0, 'T': 4005.0},\n",
       "  7: {'A': 3042.0, 'C': 127.0, 'G': 134.0, 'T': 1580.0},\n",
       "  8: {'A': 3107.0, 'C': 122.0, 'G': 127.0, 'T': 659.0},\n",
       "  9: {'A': 434.0, 'C': 102.0, 'G': 167.0, 'T': 3806.0},\n",
       "  10: {'A': 480.0, 'C': 67.0, 'G': 195.0, 'T': 3729.0},\n",
       "  11: {'A': 3101.0, 'C': 652.0, 'G': 790.0, 'T': 869.0},\n",
       "  12: {'A': 3448.0, 'C': 20.0, 'G': 26.0, 'T': 23.0},\n",
       "  13: {'A': 213.0, 'C': 356.0, 'G': 181.0, 'T': 3775.0},\n",
       "  14: {'A': 1009.0, 'C': 856.0, 'G': 4059.0, 'T': 1754.0},\n",
       "  15: {'A': 3063.0, 'C': 1367.0, 'G': 897.0, 'T': 982.0},\n",
       "  16: {'A': 1024.0, 'C': 837.0, 'G': 1730.0, 'T': 694.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18077589',\n",
       "  'uniprot_ids': 'Q15319',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0912.2',\n",
       "  'Matrix_Name': 'HOXD3',\n",
       "  1: {'A': 277.0, 'C': 11616.0, 'G': 828.0, 'T': 2948.0},\n",
       "  2: {'A': 1.0, 'C': 95.0, 'G': 1.0, 'T': 14563.0},\n",
       "  3: {'A': 14563.0, 'C': 743.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 14563.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 14563.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 4454.0, 'T': 14563.0},\n",
       "  7: {'A': 14563.0, 'C': 1.0, 'G': 2199.0, 'T': 1.0},\n",
       "  8: {'A': 715.0, 'C': 8988.0, 'G': 3188.0, 'T': 1675.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31249',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1134.1',\n",
       "  'Matrix_Name': 'FOS::JUNB',\n",
       "  1: {'A': 175.0, 'C': 169.0, 'G': 373.0, 'T': 287.0},\n",
       "  2: {'A': 594.0, 'C': 106.0, 'G': 268.0, 'T': 37.0},\n",
       "  3: {'A': 2.0, 'C': 2.0, 'G': 2.0, 'T': 999.0},\n",
       "  4: {'A': 2.0, 'C': 2.0, 'G': 869.0, 'T': 132.0},\n",
       "  5: {'A': 966.0, 'C': 31.0, 'G': 2.0, 'T': 5.0},\n",
       "  6: {'A': 49.0, 'C': 310.0, 'G': 619.0, 'T': 26.0},\n",
       "  7: {'A': 2.0, 'C': 2.0, 'G': 4.0, 'T': 997.0},\n",
       "  8: {'A': 28.0, 'C': 972.0, 'G': 2.0, 'T': 2.0},\n",
       "  9: {'A': 999.0, 'C': 2.0, 'G': 2.0, 'T': 2.0},\n",
       "  10: {'A': 8.0, 'C': 285.0, 'G': 52.0, 'T': 659.0},\n",
       "  11: {'A': 276.0, 'C': 364.0, 'G': 179.0, 'T': 185.0},\n",
       "  12: {'A': 284.0, 'C': 305.0, 'G': 257.0, 'T': 157.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P01100;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0701.2',\n",
       "  'Matrix_Name': 'LHX9',\n",
       "  1: {'A': 391.0, 'C': 6076.0, 'G': 1057.0, 'T': 1220.0},\n",
       "  2: {'A': 1.0, 'C': 2094.0, 'G': 1.0, 'T': 3984.0},\n",
       "  3: {'A': 6076.0, 'C': 506.0, 'G': 1.0, 'T': 21.0},\n",
       "  4: {'A': 6076.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6076.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 171.0, 'T': 6076.0},\n",
       "  7: {'A': 6076.0, 'C': 1.0, 'G': 306.0, 'T': 1.0},\n",
       "  8: {'A': 2262.0, 'C': 1014.0, 'G': 1947.0, 'T': 857.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9NQ69',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0702.2',\n",
       "  'Matrix_Name': 'LMX1A',\n",
       "  1: {'A': 91.0, 'C': 2223.0, 'G': 306.0, 'T': 5134.0},\n",
       "  2: {'A': 1.0, 'C': 32.0, 'G': 20.0, 'T': 7356.0},\n",
       "  3: {'A': 7356.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 7356.0, 'C': 10.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 51.0, 'G': 1.0, 'T': 7356.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7356.0},\n",
       "  7: {'A': 7356.0, 'C': 103.0, 'G': 502.0, 'T': 1.0},\n",
       "  8: {'A': 2813.0, 'C': 2559.0, 'G': 1159.0, 'T': 828.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HD-LIM',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q8TE12',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0052.4',\n",
       "  'Matrix_Name': 'MEF2A',\n",
       "  1: {'A': 5076.0, 'C': 2120.0, 'G': 3652.0, 'T': 5318.0},\n",
       "  2: {'A': 4034.0, 'C': 1961.0, 'G': 4494.0, 'T': 5677.0},\n",
       "  3: {'A': 1985.0, 'C': 10920.0, 'G': 1008.0, 'T': 2253.0},\n",
       "  4: {'A': 628.0, 'C': 2975.0, 'G': 237.0, 'T': 12326.0},\n",
       "  5: {'A': 12438.0, 'C': 1014.0, 'G': 1067.0, 'T': 1647.0},\n",
       "  6: {'A': 13133.0, 'C': 254.0, 'G': 611.0, 'T': 2168.0},\n",
       "  7: {'A': 14681.0, 'C': 142.0, 'G': 507.0, 'T': 836.0},\n",
       "  8: {'A': 14454.0, 'C': 232.0, 'G': 242.0, 'T': 1238.0},\n",
       "  9: {'A': 14957.0, 'C': 174.0, 'G': 203.0, 'T': 832.0},\n",
       "  10: {'A': 442.0, 'C': 350.0, 'G': 216.0, 'T': 15158.0},\n",
       "  11: {'A': 15583.0, 'C': 51.0, 'G': 423.0, 'T': 109.0},\n",
       "  12: {'A': 2567.0, 'C': 1061.0, 'G': 11105.0, 'T': 1433.0},\n",
       "  13: {'A': 7710.0, 'C': 4040.0, 'G': 1606.0, 'T': 2810.0},\n",
       "  14: {'A': 6172.0, 'C': 3524.0, 'G': 1811.0, 'T': 4659.0},\n",
       "  15: {'A': 5255.0, 'C': 3813.0, 'G': 2480.0, 'T': 4618.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Regulators',\n",
       "  'tf_class': 'MADS',\n",
       "  'pubmed_ids': '25217591',\n",
       "  'uniprot_ids': 'Q02078',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0014.3',\n",
       "  'Matrix_Name': 'PAX5',\n",
       "  1: {'A': 541.0, 'C': 180.0, 'G': 579.0, 'T': 116.0},\n",
       "  2: {'A': 626.0, 'C': 149.0, 'G': 355.0, 'T': 286.0},\n",
       "  3: {'A': 19.0, 'C': 152.0, 'G': 1232.0, 'T': 13.0},\n",
       "  4: {'A': 19.0, 'C': 1277.0, 'G': 35.0, 'T': 85.0},\n",
       "  5: {'A': 112.0, 'C': 36.0, 'G': 1244.0, 'T': 24.0},\n",
       "  6: {'A': 95.0, 'C': 89.0, 'G': 31.0, 'T': 1201.0},\n",
       "  7: {'A': 17.0, 'C': 38.0, 'G': 1356.0, 'T': 5.0},\n",
       "  8: {'A': 1244.0, 'C': 40.0, 'G': 71.0, 'T': 61.0},\n",
       "  9: {'A': 33.0, 'C': 1306.0, 'G': 58.0, 'T': 19.0},\n",
       "  10: {'A': 70.0, 'C': 1082.0, 'G': 119.0, 'T': 145.0},\n",
       "  11: {'A': 378.0, 'C': 406.0, 'G': 313.0, 'T': 319.0},\n",
       "  12: {'A': 264.0, 'C': 404.0, 'G': 360.0, 'T': 388.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '8406007',\n",
       "  'uniprot_ids': 'Q02548',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1534.1',\n",
       "  'Matrix_Name': 'NR1I3',\n",
       "  1: {'A': 23588.0, 'C': 4559.0, 'G': 13054.0, 'T': 7227.0},\n",
       "  2: {'A': 163.0, 'C': 11169.0, 'G': 1.0, 'T': 48425.0},\n",
       "  3: {'A': 6627.0, 'C': 101.0, 'G': 48425.0, 'T': 1.0},\n",
       "  4: {'A': 48425.0, 'C': 2614.0, 'G': 343.0, 'T': 792.0},\n",
       "  5: {'A': 48425.0, 'C': 2973.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 48425.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1871.0, 'G': 1.0, 'T': 48425.0},\n",
       "  8: {'A': 14102.0, 'C': 15444.0, 'G': 3268.0, 'T': 34324.0},\n",
       "  9: {'A': 11058.0, 'C': 8164.0, 'G': 4480.0, 'T': 37368.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '12896978',\n",
       "  'uniprot_ids': 'Q14994',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0670.1',\n",
       "  'Matrix_Name': 'NFIA',\n",
       "  1: {'A': 22970.0, 'C': 22348.0, 'G': 24322.0, 'T': 22283.0},\n",
       "  2: {'A': 20707.0, 'C': 14275.0, 'G': 35090.0, 'T': 23020.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 95497.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 98254.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 93598.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 93395.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 89072.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 89419.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 23753.0, 'C': 21853.0, 'G': 29424.0, 'T': 17411.0},\n",
       "  10: {'A': 27710.0, 'C': 14440.0, 'G': 16506.0, 'T': 32053.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '10085123',\n",
       "  'uniprot_ids': 'Q12857',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0764.2',\n",
       "  'Matrix_Name': 'ETV4',\n",
       "  1: {'A': 3120.0, 'C': 1495.0, 'G': 2722.0, 'T': 1081.0},\n",
       "  2: {'A': 1207.0, 'C': 4609.0, 'G': 1937.0, 'T': 665.0},\n",
       "  3: {'A': 8354.0, 'C': 62.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 93.0, 'G': 8309.0, 'T': 15.0},\n",
       "  5: {'A': 2.0, 'C': 2.0, 'G': 8346.0, 'T': 68.0},\n",
       "  6: {'A': 8375.0, 'C': 1.0, 'G': 1.0, 'T': 41.0},\n",
       "  7: {'A': 8334.0, 'C': 74.0, 'G': 1.0, 'T': 9.0},\n",
       "  8: {'A': 12.0, 'C': 29.0, 'G': 8345.0, 'T': 32.0},\n",
       "  9: {'A': 1297.0, 'C': 2195.0, 'G': 1950.0, 'T': 2976.0},\n",
       "  10: {'A': 2337.0, 'C': 1598.0, 'G': 3056.0, 'T': 1427.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '25866208',\n",
       "  'uniprot_ids': 'P43268',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1573.1',\n",
       "  'Matrix_Name': 'THAP11',\n",
       "  1: {'A': 155.0, 'C': 74.0, 'G': 153.0, 'T': 73.0},\n",
       "  2: {'A': 201.0, 'C': 20.0, 'G': 215.0, 'T': 19.0},\n",
       "  3: {'A': 196.0, 'C': 6.0, 'G': 244.0, 'T': 9.0},\n",
       "  4: {'A': 444.0, 'C': 6.0, 'G': 3.0, 'T': 2.0},\n",
       "  5: {'A': 1.0, 'C': 449.0, 'G': 2.0, 'T': 3.0},\n",
       "  6: {'A': 3.0, 'C': 16.0, 'G': 2.0, 'T': 434.0},\n",
       "  7: {'A': 412.0, 'C': 17.0, 'G': 11.0, 'T': 15.0},\n",
       "  8: {'A': 2.0, 'C': 447.0, 'G': 2.0, 'T': 4.0},\n",
       "  9: {'A': 403.0, 'C': 9.0, 'G': 35.0, 'T': 8.0},\n",
       "  10: {'A': 166.0, 'C': 40.0, 'G': 34.0, 'T': 215.0},\n",
       "  11: {'A': 33.0, 'C': 108.0, 'G': 48.0, 'T': 266.0},\n",
       "  12: {'A': 4.0, 'C': 18.0, 'G': 6.0, 'T': 427.0},\n",
       "  13: {'A': 5.0, 'C': 441.0, 'G': 2.0, 'T': 7.0},\n",
       "  14: {'A': 2.0, 'C': 449.0, 'G': 2.0, 'T': 2.0},\n",
       "  15: {'A': 3.0, 'C': 444.0, 'G': 3.0, 'T': 5.0},\n",
       "  16: {'A': 384.0, 'C': 6.0, 'G': 59.0, 'T': 6.0},\n",
       "  17: {'A': 35.0, 'C': 24.0, 'G': 366.0, 'T': 30.0},\n",
       "  18: {'A': 154.0, 'C': 185.0, 'G': 90.0, 'T': 26.0},\n",
       "  19: {'A': 248.0, 'C': 85.0, 'G': 86.0, 'T': 36.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 19,\n",
       "  'tf_family': 'THAP-related',\n",
       "  'tf_class': 'C2CH',\n",
       "  'pubmed_ids': '20581084',\n",
       "  'uniprot_ids': 'Q96EK4',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0909.2',\n",
       "  'Matrix_Name': 'HOXD13',\n",
       "  1: {'A': 46.0, 'C': 183.0, 'G': 130.0, 'T': 13.0},\n",
       "  2: {'A': 66.0, 'C': 369.0, 'G': 330.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 369.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 369.0, 'C': 141.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 369.0, 'C': 1.0, 'G': 6.0, 'T': 1.0},\n",
       "  6: {'A': 7.0, 'C': 1.0, 'G': 1.0, 'T': 369.0},\n",
       "  7: {'A': 369.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 369.0, 'C': 1.0, 'G': 13.0, 'T': 1.0},\n",
       "  9: {'A': 369.0, 'C': 74.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 369.0, 'C': 250.0, 'G': 14.0, 'T': 1.0},\n",
       "  11: {'A': 61.0, 'C': 304.0, 'G': 4.0, 'T': 3.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P35453',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0148.4',\n",
       "  'Matrix_Name': 'FOXA1',\n",
       "  1: {'A': 134111.0, 'C': 53262.0, 'G': 64511.0, 'T': 70923.0},\n",
       "  2: {'A': 115097.0, 'C': 16072.0, 'G': 53683.0, 'T': 137955.0},\n",
       "  3: {'A': 23287.0, 'C': 5936.0, 'G': 281613.0, 'T': 11971.0},\n",
       "  4: {'A': 11062.0, 'C': 13933.0, 'G': 6093.0, 'T': 291719.0},\n",
       "  5: {'A': 297680.0, 'C': 14649.0, 'G': 5244.0, 'T': 5234.0},\n",
       "  6: {'A': 294901.0, 'C': 13411.0, 'G': 3775.0, 'T': 10720.0},\n",
       "  7: {'A': 312907.0, 'C': 2453.0, 'G': 3653.0, 'T': 3794.0},\n",
       "  8: {'A': 7397.0, 'C': 272944.0, 'G': 4806.0, 'T': 37660.0},\n",
       "  9: {'A': 308905.0, 'C': 3675.0, 'G': 3227.0, 'T': 7000.0},\n",
       "  10: {'A': 42300.0, 'C': 23039.0, 'G': 27284.0, 'T': 230184.0},\n",
       "  11: {'A': 82541.0, 'C': 60572.0, 'G': 94377.0, 'T': 85317.0},\n",
       "  12: {'A': 88105.0, 'C': 64038.0, 'G': 70937.0, 'T': 99727.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '18798982',\n",
       "  'uniprot_ids': 'P55317',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0069.1',\n",
       "  'Matrix_Name': 'PAX6',\n",
       "  1: {'A': 3.0, 'C': 5.0, 'G': 5.0, 'T': 34.0},\n",
       "  2: {'A': 3.0, 'C': 3.0, 'G': 1.0, 'T': 40.0},\n",
       "  3: {'A': 5.0, 'C': 27.0, 'G': 2.0, 'T': 13.0},\n",
       "  4: {'A': 40.0, 'C': 3.0, 'G': 2.0, 'T': 2.0},\n",
       "  5: {'A': 4.0, 'C': 35.0, 'G': 2.0, 'T': 6.0},\n",
       "  6: {'A': 2.0, 'C': 1.0, 'G': 42.0, 'T': 2.0},\n",
       "  7: {'A': 2.0, 'C': 38.0, 'G': 5.0, 'T': 2.0},\n",
       "  8: {'A': 22.0, 'C': 3.0, 'G': 3.0, 'T': 19.0},\n",
       "  9: {'A': 2.0, 'C': 5.0, 'G': 2.0, 'T': 38.0},\n",
       "  10: {'A': 3.0, 'C': 15.0, 'G': 26.0, 'T': 3.0},\n",
       "  11: {'A': 37.0, 'C': 1.0, 'G': 7.0, 'T': 2.0},\n",
       "  12: {'A': 12.0, 'C': 12.0, 'G': 14.0, 'T': 9.0},\n",
       "  13: {'A': 2.0, 'C': 6.0, 'G': 4.0, 'T': 35.0},\n",
       "  14: {'A': 2.0, 'C': 1.0, 'G': 18.0, 'T': 26.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '8132558',\n",
       "  'uniprot_ids': 'P26367',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA0473.3',\n",
       "  'Matrix_Name': 'ELF1',\n",
       "  1: {'A': 18679.0, 'C': 8628.0, 'G': 11541.0, 'T': 8926.0},\n",
       "  2: {'A': 18649.0, 'C': 7835.0, 'G': 11306.0, 'T': 9984.0},\n",
       "  3: {'A': 17441.0, 'C': 12700.0, 'G': 10441.0, 'T': 7192.0},\n",
       "  4: {'A': 3095.0, 'C': 37924.0, 'G': 4820.0, 'T': 1935.0},\n",
       "  5: {'A': 42165.0, 'C': 4039.0, 'G': 933.0, 'T': 637.0},\n",
       "  6: {'A': 321.0, 'C': 422.0, 'G': 46643.0, 'T': 388.0},\n",
       "  7: {'A': 650.0, 'C': 506.0, 'G': 45971.0, 'T': 647.0},\n",
       "  8: {'A': 45931.0, 'C': 797.0, 'G': 677.0, 'T': 369.0},\n",
       "  9: {'A': 45244.0, 'C': 706.0, 'G': 780.0, 'T': 1044.0},\n",
       "  10: {'A': 4064.0, 'C': 1571.0, 'G': 41739.0, 'T': 400.0},\n",
       "  11: {'A': 3010.0, 'C': 1994.0, 'G': 2008.0, 'T': 40762.0},\n",
       "  12: {'A': 5689.0, 'C': 4237.0, 'G': 34758.0, 'T': 3090.0},\n",
       "  13: {'A': 12562.0, 'C': 11844.0, 'G': 16921.0, 'T': 6447.0},\n",
       "  14: {'A': 12021.0, 'C': 12682.0, 'G': 13473.0, 'T': 9598.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '8756667',\n",
       "  'uniprot_ids': 'P32519',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0876.1',\n",
       "  'Matrix_Name': 'BSX',\n",
       "  1: {'A': 1450.0, 'C': 3930.0, 'G': 2291.0, 'T': 1165.0},\n",
       "  2: {'A': 689.0, 'C': 5586.0, 'G': 104.0, 'T': 3248.0},\n",
       "  3: {'A': 4037.0, 'C': 2484.0, 'G': 2198.0, 'T': 117.0},\n",
       "  4: {'A': 8833.0, 'C': 195.0, 'G': 1079.0, 'T': 67.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 8833.0},\n",
       "  6: {'A': 97.0, 'C': 1.0, 'G': 1.0, 'T': 8833.0},\n",
       "  7: {'A': 8833.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 2771.0, 'C': 2370.0, 'G': 2418.0, 'T': 1277.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q3C1V8',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0777.1',\n",
       "  'Matrix_Name': 'MYBL2',\n",
       "  1: {'A': 786.0, 'C': 35.0, 'G': 448.0, 'T': 42.0},\n",
       "  2: {'A': 1233.0, 'C': 123.0, 'G': 723.0, 'T': 465.0},\n",
       "  3: {'A': 88.0, 'C': 1233.0, 'G': 86.0, 'T': 40.0},\n",
       "  4: {'A': 108.0, 'C': 1233.0, 'G': 191.0, 'T': 31.0},\n",
       "  5: {'A': 6.0, 'C': 1.0, 'G': 1233.0, 'T': 2.0},\n",
       "  6: {'A': 10.0, 'C': 7.0, 'G': 25.0, 'T': 1233.0},\n",
       "  7: {'A': 1.0, 'C': 10.0, 'G': 1.0, 'T': 1233.0},\n",
       "  8: {'A': 1233.0, 'C': 75.0, 'G': 78.0, 'T': 562.0},\n",
       "  9: {'A': 1233.0, 'C': 32.0, 'G': 17.0, 'T': 5.0},\n",
       "  10: {'A': 1233.0, 'C': 39.0, 'G': 29.0, 'T': 80.0},\n",
       "  11: {'A': 15.0, 'C': 1233.0, 'G': 7.0, 'T': 5.0},\n",
       "  12: {'A': 58.0, 'C': 419.0, 'G': 468.0, 'T': 290.0},\n",
       "  13: {'A': 83.0, 'C': 124.0, 'G': 1233.0, 'T': 96.0},\n",
       "  14: {'A': 229.0, 'C': 467.0, 'G': 96.0, 'T': 767.0},\n",
       "  15: {'A': 41.0, 'C': 656.0, 'G': 19.0, 'T': 578.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Myb/SANT',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '21998171',\n",
       "  'uniprot_ids': 'P10244',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0601.1',\n",
       "  'Matrix_Name': 'Arid3b',\n",
       "  1: {'A': 469.0, 'C': 178.0, 'G': 178.0, 'T': 178.0},\n",
       "  2: {'A': 274.0, 'C': 81.0, 'G': 81.0, 'T': 567.0},\n",
       "  3: {'A': 610.0, 'C': 120.0, 'G': 35.0, 'T': 239.0},\n",
       "  4: {'A': 105.0, 'C': 1.0, 'G': 1.0, 'T': 896.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  6: {'A': 1001.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 912.0, 'C': 1.0, 'G': 1.0, 'T': 90.0},\n",
       "  8: {'A': 90.0, 'C': 1.0, 'G': 1.0, 'T': 912.0},\n",
       "  9: {'A': 263.0, 'C': 169.0, 'G': 74.0, 'T': 499.0},\n",
       "  10: {'A': 491.0, 'C': 171.0, 'G': 171.0, 'T': 171.0},\n",
       "  11: {'A': 352.0, 'C': 217.0, 'G': 217.0, 'T': 217.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'ARID-related',\n",
       "  'tf_class': 'ARID',\n",
       "  'pubmed_ids': '8543152',\n",
       "  'uniprot_ids': 'Q9Z1N7',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0798.2',\n",
       "  'Matrix_Name': 'RFX3',\n",
       "  1: {'A': 561.0, 'C': 1679.0, 'G': 967.0, 'T': 1142.0},\n",
       "  2: {'A': 687.0, 'C': 928.0, 'G': 1371.0, 'T': 1363.0},\n",
       "  3: {'A': 137.0, 'C': 47.0, 'G': 4094.0, 'T': 71.0},\n",
       "  4: {'A': 37.0, 'C': 92.0, 'G': 79.0, 'T': 4141.0},\n",
       "  5: {'A': 130.0, 'C': 264.0, 'G': 79.0, 'T': 3876.0},\n",
       "  6: {'A': 326.0, 'C': 145.0, 'G': 3598.0, 'T': 280.0},\n",
       "  7: {'A': 44.0, 'C': 3941.0, 'G': 42.0, 'T': 322.0},\n",
       "  8: {'A': 51.0, 'C': 2116.0, 'G': 36.0, 'T': 2146.0},\n",
       "  9: {'A': 3962.0, 'C': 101.0, 'G': 223.0, 'T': 63.0},\n",
       "  10: {'A': 317.0, 'C': 151.0, 'G': 415.0, 'T': 3466.0},\n",
       "  11: {'A': 294.0, 'C': 47.0, 'G': 3882.0, 'T': 126.0},\n",
       "  12: {'A': 377.0, 'C': 231.0, 'G': 3575.0, 'T': 166.0},\n",
       "  13: {'A': 1073.0, 'C': 1642.0, 'G': 749.0, 'T': 885.0},\n",
       "  14: {'A': 2245.0, 'C': 433.0, 'G': 1118.0, 'T': 553.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RFX-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '20378718',\n",
       "  'uniprot_ids': 'P48380',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0746.2',\n",
       "  'Matrix_Name': 'SP3',\n",
       "  1: {'A': 1268.0, 'C': 1885.0, 'G': 1896.0, 'T': 1546.0},\n",
       "  2: {'A': 2147.0, 'C': 532.0, 'G': 4446.0, 'T': 422.0},\n",
       "  3: {'A': 665.0, 'C': 6592.0, 'G': 227.0, 'T': 346.0},\n",
       "  4: {'A': 161.0, 'C': 6592.0, 'G': 55.0, 'T': 203.0},\n",
       "  5: {'A': 5254.0, 'C': 1339.0, 'G': 3.0, 'T': 12.0},\n",
       "  6: {'A': 7.0, 'C': 6592.0, 'G': 1.0, 'T': 13.0},\n",
       "  7: {'A': 466.0, 'C': 1.0, 'G': 6592.0, 'T': 505.0},\n",
       "  8: {'A': 1.0, 'C': 6592.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 6592.0, 'G': 1.0, 'T': 10.0},\n",
       "  10: {'A': 3.0, 'C': 6592.0, 'G': 3.0, 'T': 719.0},\n",
       "  11: {'A': 3431.0, 'C': 3161.0, 'G': 40.0, 'T': 553.0},\n",
       "  12: {'A': 557.0, 'C': 5117.0, 'G': 298.0, 'T': 1476.0},\n",
       "  13: {'A': 1517.0, 'C': 2609.0, 'G': 525.0, 'T': 1944.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23019068',\n",
       "  'uniprot_ids': 'Q02447',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0468.1',\n",
       "  'Matrix_Name': 'DUX4',\n",
       "  1: {'A': 7298.0, 'C': 2496.0, 'G': 980.0, 'T': 27447.0},\n",
       "  2: {'A': 37002.0, 'C': 1.0, 'G': 1217.0, 'T': 1.0},\n",
       "  3: {'A': 38218.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 5229.0, 'C': 11676.0, 'G': 2003.0, 'T': 19313.0},\n",
       "  5: {'A': 1.0, 'C': 16479.0, 'G': 4982.0, 'T': 16759.0},\n",
       "  6: {'A': 1.0, 'C': 13952.0, 'G': 6128.0, 'T': 18140.0},\n",
       "  7: {'A': 35279.0, 'C': 2890.0, 'G': 1.0, 'T': 51.0},\n",
       "  8: {'A': 38218.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 38218.0},\n",
       "  10: {'A': 1.0, 'C': 37409.0, 'G': 1.0, 'T': 810.0},\n",
       "  11: {'A': 33799.0, 'C': 1.0, 'G': 1.0, 'T': 4420.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '22209328',\n",
       "  'uniprot_ids': 'Q9UBX2',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0651.1',\n",
       "  'Matrix_Name': 'HOXC11',\n",
       "  1: {'A': 684.0, 'C': 261.0, 'G': 1111.0, 'T': 407.0},\n",
       "  2: {'A': 1041.0, 'C': 480.0, 'G': 2461.0, 'T': 1.0},\n",
       "  3: {'A': 39.0, 'C': 42.0, 'G': 23.0, 'T': 2461.0},\n",
       "  4: {'A': 58.0, 'C': 2461.0, 'G': 13.0, 'T': 99.0},\n",
       "  5: {'A': 570.0, 'C': 1.0, 'G': 2461.0, 'T': 1.0},\n",
       "  6: {'A': 13.0, 'C': 1.0, 'G': 1.0, 'T': 2461.0},\n",
       "  7: {'A': 1785.0, 'C': 7.0, 'G': 26.0, 'T': 677.0},\n",
       "  8: {'A': 2461.0, 'C': 1.0, 'G': 1.0, 'T': 43.0},\n",
       "  9: {'A': 2461.0, 'C': 44.0, 'G': 18.0, 'T': 23.0},\n",
       "  10: {'A': 2461.0, 'C': 432.0, 'G': 280.0, 'T': 777.0},\n",
       "  11: {'A': 708.0, 'C': 629.0, 'G': 385.0, 'T': 742.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'O43248',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0672.1',\n",
       "  'Matrix_Name': 'NKX2-3',\n",
       "  1: {'A': 3573.0, 'C': 2183.0, 'G': 1990.0, 'T': 1536.0},\n",
       "  2: {'A': 2026.0, 'C': 9279.0, 'G': 4500.0, 'T': 141.0},\n",
       "  3: {'A': 39.0, 'C': 9279.0, 'G': 1.0, 'T': 133.0},\n",
       "  4: {'A': 9279.0, 'C': 99.0, 'G': 17.0, 'T': 251.0},\n",
       "  5: {'A': 9.0, 'C': 9279.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 19.0, 'T': 9279.0},\n",
       "  7: {'A': 1.0, 'C': 940.0, 'G': 1.0, 'T': 9279.0},\n",
       "  8: {'A': 4242.0, 'C': 1452.0, 'G': 5038.0, 'T': 344.0},\n",
       "  9: {'A': 9279.0, 'C': 918.0, 'G': 1832.0, 'T': 2438.0},\n",
       "  10: {'A': 3171.0, 'C': 2755.0, 'G': 2789.0, 'T': 567.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '11042197',\n",
       "  'uniprot_ids': 'Q8TAU0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0685.1',\n",
       "  'Matrix_Name': 'SP4',\n",
       "  1: {'A': 622.0, 'C': 1406.0, 'G': 824.0, 'T': 1623.0},\n",
       "  2: {'A': 2174.0, 'C': 157.0, 'G': 66.0, 'T': 1810.0},\n",
       "  3: {'A': 2601.0, 'C': 147.0, 'G': 1640.0, 'T': 200.0},\n",
       "  4: {'A': 940.0, 'C': 7.0, 'G': 4775.0, 'T': 17.0},\n",
       "  5: {'A': 261.0, 'C': 4444.0, 'G': 1.0, 'T': 5.0},\n",
       "  6: {'A': 3.0, 'C': 4493.0, 'G': 6.0, 'T': 136.0},\n",
       "  7: {'A': 3005.0, 'C': 992.0, 'G': 4.0, 'T': 1.0},\n",
       "  8: {'A': 3.0, 'C': 4663.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 11.0, 'C': 15.0, 'G': 5445.0, 'T': 56.0},\n",
       "  10: {'A': 3.0, 'C': 4682.0, 'G': 5.0, 'T': 1.0},\n",
       "  11: {'A': 4.0, 'C': 4733.0, 'G': 3.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 4668.0, 'G': 4.0, 'T': 73.0},\n",
       "  13: {'A': 1841.0, 'C': 2431.0, 'G': 10.0, 'T': 68.0},\n",
       "  14: {'A': 114.0, 'C': 4277.0, 'G': 22.0, 'T': 1068.0},\n",
       "  15: {'A': 1141.0, 'C': 898.0, 'G': 235.0, 'T': 3044.0},\n",
       "  16: {'A': 583.0, 'C': 1207.0, 'G': 364.0, 'T': 2079.0},\n",
       "  17: {'A': 968.0, 'C': 1181.0, 'G': 557.0, 'T': 1510.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23019068',\n",
       "  'uniprot_ids': 'Q02446',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0512.2',\n",
       "  'Matrix_Name': 'Rxra',\n",
       "  1: {'A': 12569.0, 'C': 5032.0, 'G': 30812.0, 'T': 2085.0},\n",
       "  2: {'A': 21346.0, 'C': 201.0, 'G': 30407.0, 'T': 76.0},\n",
       "  3: {'A': 111.0, 'C': 78.0, 'G': 38291.0, 'T': 160.0},\n",
       "  4: {'A': 133.0, 'C': 73.0, 'G': 36415.0, 'T': 4373.0},\n",
       "  5: {'A': 48.0, 'C': 110.0, 'G': 373.0, 'T': 40597.0},\n",
       "  6: {'A': 98.0, 'C': 44343.0, 'G': 578.0, 'T': 2687.0},\n",
       "  7: {'A': 46205.0, 'C': 52.0, 'G': 121.0, 'T': 22.0},\n",
       "  8: {'A': 45276.0, 'C': 90.0, 'G': 831.0, 'T': 369.0},\n",
       "  9: {'A': 45266.0, 'C': 74.0, 'G': 1370.0, 'T': 20.0},\n",
       "  10: {'A': 40.0, 'C': 72.0, 'G': 39799.0, 'T': 52.0},\n",
       "  11: {'A': 64.0, 'C': 86.0, 'G': 38186.0, 'T': 3336.0},\n",
       "  12: {'A': 45.0, 'C': 115.0, 'G': 268.0, 'T': 39868.0},\n",
       "  13: {'A': 64.0, 'C': 43687.0, 'G': 349.0, 'T': 1776.0},\n",
       "  14: {'A': 44404.0, 'C': 159.0, 'G': 2448.0, 'T': 38.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '1323763',\n",
       "  'uniprot_ids': 'P28700',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0737.1',\n",
       "  'Matrix_Name': 'GLIS3',\n",
       "  1: {'A': 9.0, 'C': 42.0, 'G': 675.0, 'T': 259.0},\n",
       "  2: {'A': 423.0, 'C': 3.0, 'G': 14.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 290.0, 'G': 2.0, 'T': 2.0},\n",
       "  4: {'A': 1.0, 'C': 281.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 3.0, 'C': 310.0, 'G': 1.0, 'T': 10.0},\n",
       "  6: {'A': 1.0, 'C': 289.0, 'G': 1.0, 'T': 2.0},\n",
       "  7: {'A': 5.0, 'C': 201.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 212.0, 'G': 1.0, 'T': 21.0},\n",
       "  9: {'A': 887.0, 'C': 2.0, 'G': 68.0, 'T': 10.0},\n",
       "  10: {'A': 2.0, 'C': 205.0, 'G': 9.0, 'T': 18.0},\n",
       "  11: {'A': 227.0, 'C': 18.0, 'G': 649.0, 'T': 79.0},\n",
       "  12: {'A': 530.0, 'C': 3.0, 'G': 32.0, 'T': 88.0},\n",
       "  13: {'A': 219.0, 'C': 57.0, 'G': 5.0, 'T': 38.0},\n",
       "  14: {'A': 1.0, 'C': 17.0, 'G': 609.0, 'T': 5.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '22391303',\n",
       "  'uniprot_ids': 'Q8NEA6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1472.1',\n",
       "  'Matrix_Name': 'BHLHA15(var.2)',\n",
       "  1: {'A': 698.0, 'C': 282.0, 'G': 358.0, 'T': 171.0},\n",
       "  2: {'A': 898.0, 'C': 608.0, 'G': 592.0, 'T': 82.0},\n",
       "  3: {'A': 48.0, 'C': 1505.0, 'G': 7.0, 'T': 34.0},\n",
       "  4: {'A': 1505.0, 'C': 34.0, 'G': 34.0, 'T': 138.0},\n",
       "  5: {'A': 12.0, 'C': 31.0, 'G': 1505.0, 'T': 468.0},\n",
       "  6: {'A': 446.0, 'C': 1505.0, 'G': 73.0, 'T': 12.0},\n",
       "  7: {'A': 55.0, 'C': 5.0, 'G': 1.0, 'T': 1505.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1505.0, 'T': 1.0},\n",
       "  9: {'A': 55.0, 'C': 611.0, 'G': 490.0, 'T': 1016.0},\n",
       "  10: {'A': 203.0, 'C': 411.0, 'G': 317.0, 'T': 578.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '28174210',\n",
       "  'uniprot_ids': 'Q7RTS1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0911.1',\n",
       "  'Matrix_Name': 'Hoxa11',\n",
       "  1: {'A': 179.0, 'C': 109.0, 'G': 308.0, 'T': 112.0},\n",
       "  2: {'A': 297.0, 'C': 256.0, 'G': 705.0, 'T': 4.0},\n",
       "  3: {'A': 2.0, 'C': 29.0, 'G': 17.0, 'T': 705.0},\n",
       "  4: {'A': 31.0, 'C': 705.0, 'G': 1.0, 'T': 37.0},\n",
       "  5: {'A': 251.0, 'C': 3.0, 'G': 705.0, 'T': 8.0},\n",
       "  6: {'A': 15.0, 'C': 2.0, 'G': 4.0, 'T': 705.0},\n",
       "  7: {'A': 490.0, 'C': 4.0, 'G': 14.0, 'T': 216.0},\n",
       "  8: {'A': 705.0, 'C': 1.0, 'G': 3.0, 'T': 32.0},\n",
       "  9: {'A': 705.0, 'C': 28.0, 'G': 4.0, 'T': 41.0},\n",
       "  10: {'A': 705.0, 'C': 160.0, 'G': 118.0, 'T': 198.0},\n",
       "  11: {'A': 180.0, 'C': 165.0, 'G': 151.0, 'T': 212.0},\n",
       "  12: {'A': 140.0, 'C': 111.0, 'G': 113.0, 'T': 344.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31270',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0842.2',\n",
       "  'Matrix_Name': 'NRL',\n",
       "  1: {'A': 14409.0, 'C': 3103.0, 'G': 5589.0, 'T': 4131.0},\n",
       "  2: {'A': 21190.0, 'C': 839.0, 'G': 3686.0, 'T': 6040.0},\n",
       "  3: {'A': 17468.0, 'C': 320.0, 'G': 636.0, 'T': 9762.0},\n",
       "  4: {'A': 14034.0, 'C': 1315.0, 'G': 823.0, 'T': 13195.0},\n",
       "  5: {'A': 6836.0, 'C': 5670.0, 'G': 7984.0, 'T': 6742.0},\n",
       "  6: {'A': 582.0, 'C': 2769.0, 'G': 69.0, 'T': 27229.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 27229.0, 'T': 1.0},\n",
       "  8: {'A': 1009.0, 'C': 27229.0, 'G': 23.0, 'T': 15.0},\n",
       "  9: {'A': 1.0, 'C': 9.0, 'G': 1.0, 'T': 27229.0},\n",
       "  10: {'A': 74.0, 'C': 1.0, 'G': 27229.0, 'T': 1037.0},\n",
       "  11: {'A': 27229.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  12: {'A': 48.0, 'C': 27229.0, 'G': 560.0, 'T': 2395.0},\n",
       "  13: {'A': 6002.0, 'C': 1670.0, 'G': 14529.0, 'T': 5031.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8552602',\n",
       "  'uniprot_ids': 'P54845',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0694.1',\n",
       "  'Matrix_Name': 'ZBTB7B',\n",
       "  1: {'A': 534.0, 'C': 160.0, 'G': 796.0, 'T': 366.0},\n",
       "  2: {'A': 145.0, 'C': 1853.0, 'G': 225.0, 'T': 55.0},\n",
       "  3: {'A': 395.0, 'C': 20.0, 'G': 1853.0, 'T': 79.0},\n",
       "  4: {'A': 1853.0, 'C': 239.0, 'G': 1.0, 'T': 4.0},\n",
       "  5: {'A': 6.0, 'C': 1853.0, 'G': 2.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1853.0, 'G': 3.0, 'T': 1.0},\n",
       "  7: {'A': 1853.0, 'C': 821.0, 'G': 3.0, 'T': 4.0},\n",
       "  8: {'A': 6.0, 'C': 1853.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 1.0, 'C': 1853.0, 'G': 1.0, 'T': 8.0},\n",
       "  10: {'A': 804.0, 'C': 475.0, 'G': 1853.0, 'T': 306.0},\n",
       "  11: {'A': 1853.0, 'C': 296.0, 'G': 222.0, 'T': 366.0},\n",
       "  12: {'A': 997.0, 'C': 310.0, 'G': 246.0, 'T': 303.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'O15156',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0747.1',\n",
       "  'Matrix_Name': 'SP8',\n",
       "  1: {'A': 53.0, 'C': 28.0, 'G': 79.0, 'T': 11.0},\n",
       "  2: {'A': 55.0, 'C': 114.0, 'G': 1.0, 'T': 5.0},\n",
       "  3: {'A': 26.0, 'C': 168.0, 'G': 1.0, 'T': 14.0},\n",
       "  4: {'A': 134.0, 'C': 34.0, 'G': 2.0, 'T': 3.0},\n",
       "  5: {'A': 8.0, 'C': 168.0, 'G': 2.0, 'T': 6.0},\n",
       "  6: {'A': 26.0, 'C': 38.0, 'G': 168.0, 'T': 40.0},\n",
       "  7: {'A': 6.0, 'C': 168.0, 'G': 7.0, 'T': 3.0},\n",
       "  8: {'A': 3.0, 'C': 168.0, 'G': 4.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 168.0, 'G': 5.0, 'T': 16.0},\n",
       "  10: {'A': 91.0, 'C': 78.0, 'G': 7.0, 'T': 4.0},\n",
       "  11: {'A': 14.0, 'C': 168.0, 'G': 22.0, 'T': 40.0},\n",
       "  12: {'A': 32.0, 'C': 53.0, 'G': 2.0, 'T': 83.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '23019068',\n",
       "  'uniprot_ids': 'Q8IXZ3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0649.1',\n",
       "  'Matrix_Name': 'HEY2',\n",
       "  1: {'A': 108.0, 'C': 100.0, 'G': 1801.0, 'T': 102.0},\n",
       "  2: {'A': 652.0, 'C': 306.0, 'G': 621.0, 'T': 225.0},\n",
       "  3: {'A': 85.0, 'C': 1801.0, 'G': 84.0, 'T': 59.0},\n",
       "  4: {'A': 1801.0, 'C': 1.0, 'G': 129.0, 'T': 1.0},\n",
       "  5: {'A': 8.0, 'C': 1801.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 28.0, 'C': 73.0, 'G': 1801.0, 'T': 1.0},\n",
       "  7: {'A': 37.0, 'C': 121.0, 'G': 1.0, 'T': 1801.0},\n",
       "  8: {'A': 1.0, 'C': 35.0, 'G': 1801.0, 'T': 31.0},\n",
       "  9: {'A': 51.0, 'C': 1048.0, 'G': 129.0, 'T': 754.0},\n",
       "  10: {'A': 286.0, 'C': 870.0, 'G': 383.0, 'T': 265.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Hairy-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22615585',\n",
       "  'uniprot_ids': 'Q9UBP5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0003.4',\n",
       "  'Matrix_Name': 'TFAP2A',\n",
       "  1: {'A': 4336.0, 'C': 4093.0, 'G': 4073.0, 'T': 3470.0},\n",
       "  2: {'A': 2764.0, 'C': 4621.0, 'G': 3375.0, 'T': 5212.0},\n",
       "  3: {'A': 2843.0, 'C': 2635.0, 'G': 3792.0, 'T': 6702.0},\n",
       "  4: {'A': 1089.0, 'C': 3616.0, 'G': 9962.0, 'T': 1305.0},\n",
       "  5: {'A': 281.0, 'C': 15322.0, 'G': 188.0, 'T': 181.0},\n",
       "  6: {'A': 117.0, 'C': 15352.0, 'G': 213.0, 'T': 290.0},\n",
       "  7: {'A': 279.0, 'C': 1174.0, 'G': 751.0, 'T': 13768.0},\n",
       "  8: {'A': 531.0, 'C': 13537.0, 'G': 1407.0, 'T': 497.0},\n",
       "  9: {'A': 13568.0, 'C': 740.0, 'G': 1230.0, 'T': 434.0},\n",
       "  10: {'A': 187.0, 'C': 218.0, 'G': 15433.0, 'T': 134.0},\n",
       "  11: {'A': 155.0, 'C': 235.0, 'G': 15423.0, 'T': 159.0},\n",
       "  12: {'A': 1222.0, 'C': 9446.0, 'G': 4099.0, 'T': 1205.0},\n",
       "  13: {'A': 4221.0, 'C': 4784.0, 'G': 3052.0, 'T': 3915.0},\n",
       "  14: {'A': 5205.0, 'C': 3916.0, 'G': 3842.0, 'T': 3009.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P05549',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0626.1',\n",
       "  'Matrix_Name': 'Npas2',\n",
       "  1: {'A': 274.0, 'C': 178.0, 'G': 294.0, 'T': 258.0},\n",
       "  2: {'A': 155.0, 'C': 378.0, 'G': 421.0, 'T': 49.0},\n",
       "  3: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 82.0, 'C': 919.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 82.0, 'C': 1.0, 'G': 919.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  8: {'A': 82.0, 'C': 1.0, 'G': 919.0, 'T': 1.0},\n",
       "  9: {'A': 75.0, 'C': 173.0, 'G': 189.0, 'T': 567.0},\n",
       "  10: {'A': 298.0, 'C': 300.0, 'G': 203.0, 'T': 203.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P97460',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1627.1',\n",
       "  'Matrix_Name': 'Wt1',\n",
       "  1: {'A': 877.0, 'C': 1882.0, 'G': 795.0, 'T': 1141.0},\n",
       "  2: {'A': 668.0, 'C': 1631.0, 'G': 1162.0, 'T': 1234.0},\n",
       "  3: {'A': 840.0, 'C': 2949.0, 'G': 405.0, 'T': 501.0},\n",
       "  4: {'A': 49.0, 'C': 4547.0, 'G': 59.0, 'T': 40.0},\n",
       "  5: {'A': 121.0, 'C': 70.0, 'G': 320.0, 'T': 4184.0},\n",
       "  6: {'A': 22.0, 'C': 4603.0, 'G': 43.0, 'T': 27.0},\n",
       "  7: {'A': 41.0, 'C': 4355.0, 'G': 76.0, 'T': 223.0},\n",
       "  8: {'A': 44.0, 'C': 4253.0, 'G': 121.0, 'T': 277.0},\n",
       "  9: {'A': 851.0, 'C': 3637.0, 'G': 102.0, 'T': 105.0},\n",
       "  10: {'A': 32.0, 'C': 4574.0, 'G': 47.0, 'T': 42.0},\n",
       "  11: {'A': 3295.0, 'C': 54.0, 'G': 991.0, 'T': 355.0},\n",
       "  12: {'A': 159.0, 'C': 3750.0, 'G': 437.0, 'T': 349.0},\n",
       "  13: {'A': 1676.0, 'C': 1351.0, 'G': 929.0, 'T': 739.0},\n",
       "  14: {'A': 521.0, 'C': 2024.0, 'G': 1039.0, 'T': 1111.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7862533',\n",
       "  'uniprot_ids': 'P22561',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0710.1',\n",
       "  'Matrix_Name': 'NOTO',\n",
       "  1: {'A': 5632.0, 'C': 5683.0, 'G': 6530.0, 'T': 3564.0},\n",
       "  2: {'A': 775.0, 'C': 13987.0, 'G': 618.0, 'T': 7420.0},\n",
       "  3: {'A': 7098.0, 'C': 4002.0, 'G': 574.0, 'T': 21406.0},\n",
       "  4: {'A': 15097.0, 'C': 6311.0, 'G': 1.0, 'T': 140.0},\n",
       "  5: {'A': 21406.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 21406.0},\n",
       "  7: {'A': 375.0, 'C': 63.0, 'G': 423.0, 'T': 21406.0},\n",
       "  8: {'A': 21406.0, 'C': 1.0, 'G': 2058.0, 'T': 1905.0},\n",
       "  9: {'A': 7039.0, 'C': 3857.0, 'G': 7680.0, 'T': 2833.0},\n",
       "  10: {'A': 4079.0, 'C': 7712.0, 'G': 5494.0, 'T': 4125.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'A8MTQ0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0137.3',\n",
       "  'Matrix_Name': 'STAT1',\n",
       "  1: {'A': 367.0, 'C': 732.0, 'G': 786.0, 'T': 1748.0},\n",
       "  2: {'A': 1.0, 'C': 53.0, 'G': 1.0, 'T': 3578.0},\n",
       "  3: {'A': 1.0, 'C': 7.0, 'G': 111.0, 'T': 3514.0},\n",
       "  4: {'A': 362.0, 'C': 3263.0, 'G': 7.0, 'T': 1.0},\n",
       "  5: {'A': 110.0, 'C': 2010.0, 'G': 1.0, 'T': 1512.0},\n",
       "  6: {'A': 1740.0, 'C': 1.0, 'G': 1638.0, 'T': 254.0},\n",
       "  7: {'A': 164.0, 'C': 1.0, 'G': 3467.0, 'T': 1.0},\n",
       "  8: {'A': 30.0, 'C': 8.0, 'G': 3427.0, 'T': 168.0},\n",
       "  9: {'A': 3621.0, 'C': 1.0, 'G': 10.0, 'T': 1.0},\n",
       "  10: {'A': 3625.0, 'C': 6.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1880.0, 'C': 312.0, 'G': 726.0, 'T': 715.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '17558387',\n",
       "  'uniprot_ids': 'P42224',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0476.1',\n",
       "  'Matrix_Name': 'FOS',\n",
       "  1: {'A': 7880.0, 'C': 713.0, 'G': 9687.0, 'T': 11120.0},\n",
       "  2: {'A': 7476.0, 'C': 10178.0, 'G': 10842.0, 'T': 904.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 29397.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 27109.0, 'T': 2289.0},\n",
       "  5: {'A': 29397.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 999.0, 'C': 14080.0, 'G': 11207.0, 'T': 3114.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 29397.0},\n",
       "  8: {'A': 1.0, 'C': 29397.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 29397.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 259.0, 'C': 5824.0, 'G': 1539.0, 'T': 21778.0},\n",
       "  11: {'A': 4007.0, 'C': 8237.0, 'G': 7898.0, 'T': 9258.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P01100',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1514.1',\n",
       "  'Matrix_Name': 'KLF17',\n",
       "  1: {'A': 8970.0, 'C': 14367.0, 'G': 3319.0, 'T': 3580.0},\n",
       "  2: {'A': 19058.0, 'C': 9729.0, 'G': 2103.0, 'T': 2133.0},\n",
       "  3: {'A': 222.0, 'C': 30182.0, 'G': 123.0, 'T': 115.0},\n",
       "  4: {'A': 145.0, 'C': 30202.0, 'G': 106.0, 'T': 88.0},\n",
       "  5: {'A': 28923.0, 'C': 198.0, 'G': 85.0, 'T': 2496.0},\n",
       "  6: {'A': 86.0, 'C': 30108.0, 'G': 54.0, 'T': 424.0},\n",
       "  7: {'A': 7237.0, 'C': 6959.0, 'G': 24408.0, 'T': 1244.0},\n",
       "  8: {'A': 4564.0, 'C': 29154.0, 'G': 105.0, 'T': 374.0},\n",
       "  9: {'A': 19143.0, 'C': 1077.0, 'G': 5177.0, 'T': 8374.0},\n",
       "  10: {'A': 828.0, 'C': 29552.0, 'G': 1411.0, 'T': 221.0},\n",
       "  11: {'A': 133.0, 'C': 30201.0, 'G': 202.0, 'T': 58.0},\n",
       "  12: {'A': 222.0, 'C': 30200.0, 'G': 81.0, 'T': 234.0},\n",
       "  13: {'A': 12267.0, 'C': 14826.0, 'G': 3846.0, 'T': 3319.0},\n",
       "  14: {'A': 3893.0, 'C': 4874.0, 'G': 2369.0, 'T': 23589.0},\n",
       "  15: {'A': 4466.0, 'C': 9630.0, 'G': 2852.0, 'T': 17960.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11443140',\n",
       "  'uniprot_ids': 'Q5JT82',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1552.1',\n",
       "  'Matrix_Name': 'RARB(var.3)',\n",
       "  1: {'A': 218.0, 'C': 31.0, 'G': 170.0, 'T': 57.0},\n",
       "  2: {'A': 362.0, 'C': 46.0, 'G': 112.0, 'T': 1.0},\n",
       "  3: {'A': 4.0, 'C': 18.0, 'G': 473.0, 'T': 32.0},\n",
       "  4: {'A': 11.0, 'C': 5.0, 'G': 473.0, 'T': 77.0},\n",
       "  5: {'A': 22.0, 'C': 65.0, 'G': 53.0, 'T': 473.0},\n",
       "  6: {'A': 6.0, 'C': 473.0, 'G': 8.0, 'T': 17.0},\n",
       "  7: {'A': 323.0, 'C': 1.0, 'G': 151.0, 'T': 4.0},\n",
       "  8: {'A': 4.0, 'C': 6.0, 'G': 1.0, 'T': 473.0},\n",
       "  9: {'A': 17.0, 'C': 9.0, 'G': 473.0, 'T': 5.0},\n",
       "  10: {'A': 473.0, 'C': 27.0, 'G': 19.0, 'T': 14.0},\n",
       "  11: {'A': 77.0, 'C': 473.0, 'G': 1.0, 'T': 8.0},\n",
       "  12: {'A': 13.0, 'C': 473.0, 'G': 9.0, 'T': 17.0},\n",
       "  13: {'A': 11.0, 'C': 102.0, 'G': 46.0, 'T': 372.0},\n",
       "  14: {'A': 64.0, 'C': 129.0, 'G': 56.0, 'T': 227.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '1738600',\n",
       "  'uniprot_ids': 'P10826',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0495.3',\n",
       "  'Matrix_Name': 'MAFF',\n",
       "  1: {'A': 11980.0, 'C': 5574.0, 'G': 11123.0, 'T': 9073.0},\n",
       "  2: {'A': 17646.0, 'C': 4127.0, 'G': 5075.0, 'T': 10902.0},\n",
       "  3: {'A': 3944.0, 'C': 4219.0, 'G': 27743.0, 'T': 1844.0},\n",
       "  4: {'A': 576.0, 'C': 370.0, 'G': 410.0, 'T': 36394.0},\n",
       "  5: {'A': 1729.0, 'C': 35021.0, 'G': 331.0, 'T': 669.0},\n",
       "  6: {'A': 36491.0, 'C': 153.0, 'G': 445.0, 'T': 661.0},\n",
       "  7: {'A': 639.0, 'C': 401.0, 'G': 33924.0, 'T': 2786.0},\n",
       "  8: {'A': 407.0, 'C': 36229.0, 'G': 419.0, 'T': 695.0},\n",
       "  9: {'A': 34457.0, 'C': 560.0, 'G': 1042.0, 'T': 1691.0},\n",
       "  10: {'A': 7640.0, 'C': 4361.0, 'G': 4443.0, 'T': 21306.0},\n",
       "  11: {'A': 7160.0, 'C': 2010.0, 'G': 1834.0, 'T': 26746.0},\n",
       "  12: {'A': 5787.0, 'C': 2424.0, 'G': 1499.0, 'T': 28040.0},\n",
       "  13: {'A': 6526.0, 'C': 4820.0, 'G': 2340.0, 'T': 24064.0},\n",
       "  14: {'A': 11673.0, 'C': 7093.0, 'G': 4732.0, 'T': 14252.0},\n",
       "  15: {'A': 9995.0, 'C': 7585.0, 'G': 6712.0, 'T': 13458.0},\n",
       "  16: {'A': 11388.0, 'C': 7343.0, 'G': 7214.0, 'T': 11805.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Maf-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8264639',\n",
       "  'uniprot_ids': 'Q9ULX9',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0072.1',\n",
       "  'Matrix_Name': 'RORA(var.2)',\n",
       "  1: {'A': 10.0, 'C': 9.0, 'G': 9.0, 'T': 12.0},\n",
       "  2: {'A': 18.0, 'C': 3.0, 'G': 8.0, 'T': 11.0},\n",
       "  3: {'A': 16.0, 'C': 1.0, 'G': 4.0, 'T': 19.0},\n",
       "  4: {'A': 36.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 24.0, 'C': 1.0, 'G': 1.0, 'T': 14.0},\n",
       "  6: {'A': 3.0, 'C': 13.0, 'G': 14.0, 'T': 10.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 37.0},\n",
       "  8: {'A': 29.0, 'C': 1.0, 'G': 9.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 37.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 37.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 37.0},\n",
       "  12: {'A': 1.0, 'C': 37.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 37.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 16.0, 'C': 7.0, 'G': 11.0, 'T': 6.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '7926749',\n",
       "  'uniprot_ids': 'P35398',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1592.1',\n",
       "  'Matrix_Name': 'ZNF274',\n",
       "  1: {'A': 7979.0, 'C': 7295.0, 'G': 8316.0, 'T': 7599.0},\n",
       "  2: {'A': 10813.0, 'C': 4820.0, 'G': 17493.0, 'T': 3309.0},\n",
       "  3: {'A': 1042.0, 'C': 9980.0, 'G': 2363.0, 'T': 26042.0},\n",
       "  4: {'A': 16868.0, 'C': 24.0, 'G': 13541.0, 'T': 2656.0},\n",
       "  5: {'A': 140.0, 'C': 301.0, 'G': 4332.0, 'T': 29358.0},\n",
       "  6: {'A': 12.0, 'C': 6.0, 'G': 31186.0, 'T': 1.0},\n",
       "  7: {'A': 30116.0, 'C': 20.0, 'G': 1446.0, 'T': 1243.0},\n",
       "  8: {'A': 1771.0, 'C': 2890.0, 'G': 29910.0, 'T': 165.0},\n",
       "  9: {'A': 72.0, 'C': 21.0, 'G': 105.0, 'T': 31154.0},\n",
       "  10: {'A': 115.0, 'C': 13339.0, 'G': 1.0, 'T': 27471.0},\n",
       "  11: {'A': 16.0, 'C': 31182.0, 'G': 7.0, 'T': 9.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 31181.0},\n",
       "  13: {'A': 6.0, 'C': 29920.0, 'G': 1.0, 'T': 2992.0},\n",
       "  14: {'A': 8630.0, 'C': 908.0, 'G': 20539.0, 'T': 5407.0},\n",
       "  15: {'A': 4051.0, 'C': 15336.0, 'G': 6975.0, 'T': 12411.0},\n",
       "  16: {'A': 7600.0, 'C': 5480.0, 'G': 8878.0, 'T': 9232.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '21170338',\n",
       "  'uniprot_ids': 'Q96GC6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0749.1',\n",
       "  'Matrix_Name': 'ZBED1',\n",
       "  1: {'A': 1154.0, 'C': 2614.0, 'G': 592.0, 'T': 712.0},\n",
       "  2: {'A': 442.0, 'C': 23.0, 'G': 57.0, 'T': 5322.0},\n",
       "  3: {'A': 3842.0, 'C': 16.0, 'G': 2066.0, 'T': 9.0},\n",
       "  4: {'A': 7.0, 'C': 9.0, 'G': 1.0, 'T': 5511.0},\n",
       "  5: {'A': 1.0, 'C': 5374.0, 'G': 7.0, 'T': 19.0},\n",
       "  6: {'A': 128.0, 'C': 4.0, 'G': 5980.0, 'T': 3.0},\n",
       "  7: {'A': 5.0, 'C': 5358.0, 'G': 1.0, 'T': 111.0},\n",
       "  8: {'A': 5.0, 'C': 5.0, 'G': 6014.0, 'T': 1.0},\n",
       "  9: {'A': 5327.0, 'C': 10.0, 'G': 13.0, 'T': 1.0},\n",
       "  10: {'A': 7.0, 'C': 4331.0, 'G': 1.0, 'T': 866.0},\n",
       "  11: {'A': 5277.0, 'C': 12.0, 'G': 11.0, 'T': 145.0},\n",
       "  12: {'A': 119.0, 'C': 274.0, 'G': 636.0, 'T': 5056.0},\n",
       "  13: {'A': 3124.0, 'C': 271.0, 'G': 1877.0, 'T': 609.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'BED',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '12663651',\n",
       "  'uniprot_ids': 'O96006',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1143.1',\n",
       "  'Matrix_Name': 'FOSL1::JUND(var.2)',\n",
       "  1: {'A': 562.0, 'C': 76.0, 'G': 325.0, 'T': 41.0},\n",
       "  2: {'A': 37.0, 'C': 51.0, 'G': 45.0, 'T': 871.0},\n",
       "  3: {'A': 40.0, 'C': 102.0, 'G': 778.0, 'T': 83.0},\n",
       "  4: {'A': 871.0, 'C': 46.0, 'G': 42.0, 'T': 45.0},\n",
       "  5: {'A': 33.0, 'C': 836.0, 'G': 65.0, 'T': 69.0},\n",
       "  6: {'A': 117.0, 'C': 60.0, 'G': 650.0, 'T': 177.0},\n",
       "  7: {'A': 118.0, 'C': 263.0, 'G': 40.0, 'T': 583.0},\n",
       "  8: {'A': 547.0, 'C': 331.0, 'G': 84.0, 'T': 42.0},\n",
       "  9: {'A': 646.0, 'C': 67.0, 'G': 101.0, 'T': 189.0},\n",
       "  10: {'A': 124.0, 'C': 402.0, 'G': 83.0, 'T': 395.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15407;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0498.2',\n",
       "  'Matrix_Name': 'MEIS1',\n",
       "  1: {'A': 23728.0, 'C': 18798.0, 'G': 9715.0, 'T': 24922.0},\n",
       "  2: {'A': 8691.0, 'C': 5576.0, 'G': 3175.0, 'T': 77160.0},\n",
       "  3: {'A': 1.0, 'C': 1250.0, 'G': 77160.0, 'T': 1.0},\n",
       "  4: {'A': 77160.0, 'C': 2904.0, 'G': 673.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 77160.0, 'G': 1.0, 'T': 1156.0},\n",
       "  6: {'A': 77160.0, 'C': 1.0, 'G': 9264.0, 'T': 3862.0},\n",
       "  7: {'A': 17225.0, 'C': 14865.0, 'G': 29966.0, 'T': 15107.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9405651',\n",
       "  'uniprot_ids': 'O00470',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0752.1',\n",
       "  'Matrix_Name': 'ZNF410',\n",
       "  1: {'A': 662.0, 'C': 701.0, 'G': 951.0, 'T': 1678.0},\n",
       "  2: {'A': 1716.0, 'C': 1873.0, 'G': 400.0, 'T': 3.0},\n",
       "  3: {'A': 2.0, 'C': 3642.0, 'G': 1.0, 'T': 371.0},\n",
       "  4: {'A': 3986.0, 'C': 3.0, 'G': 2.0, 'T': 5.0},\n",
       "  5: {'A': 51.0, 'C': 34.0, 'G': 125.0, 'T': 3828.0},\n",
       "  6: {'A': 70.0, 'C': 3950.0, 'G': 2.0, 'T': 5.0},\n",
       "  7: {'A': 13.0, 'C': 3987.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 3987.0, 'G': 2.0, 'T': 1.0},\n",
       "  9: {'A': 3986.0, 'C': 6.0, 'G': 2.0, 'T': 3.0},\n",
       "  10: {'A': 1.0, 'C': 10.0, 'G': 1.0, 'T': 3988.0},\n",
       "  11: {'A': 3988.0, 'C': 9.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 3984.0, 'C': 3.0, 'G': 4.0, 'T': 2.0},\n",
       "  13: {'A': 5.0, 'C': 7.0, 'G': 1.0, 'T': 3985.0},\n",
       "  14: {'A': 3931.0, 'C': 18.0, 'G': 44.0, 'T': 25.0},\n",
       "  15: {'A': 1070.0, 'C': 1286.0, 'G': 620.0, 'T': 1016.0},\n",
       "  16: {'A': 173.0, 'C': 819.0, 'G': 789.0, 'T': 2275.0},\n",
       "  17: {'A': 799.0, 'C': 2629.0, 'G': 94.0, 'T': 521.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q86VK4',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0442.2',\n",
       "  'Matrix_Name': 'SOX10',\n",
       "  1: {'A': 684.0, 'C': 398.0, 'G': 528.0, 'T': 449.0},\n",
       "  2: {'A': 724.0, 'C': 284.0, 'G': 572.0, 'T': 479.0},\n",
       "  3: {'A': 1119.0, 'C': 373.0, 'G': 318.0, 'T': 249.0},\n",
       "  4: {'A': 2024.0, 'C': 5.0, 'G': 4.0, 'T': 26.0},\n",
       "  5: {'A': 4.0, 'C': 1990.0, 'G': 12.0, 'T': 53.0},\n",
       "  6: {'A': 2030.0, 'C': 8.0, 'G': 8.0, 'T': 13.0},\n",
       "  7: {'A': 1969.0, 'C': 29.0, 'G': 21.0, 'T': 40.0},\n",
       "  8: {'A': 1960.0, 'C': 24.0, 'G': 13.0, 'T': 62.0},\n",
       "  9: {'A': 11.0, 'C': 14.0, 'G': 2013.0, 'T': 21.0},\n",
       "  10: {'A': 667.0, 'C': 553.0, 'G': 620.0, 'T': 219.0},\n",
       "  11: {'A': 639.0, 'C': 556.0, 'G': 506.0, 'T': 358.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P56693',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1470.1',\n",
       "  'Matrix_Name': 'BACH2(var.2)',\n",
       "  1: {'A': 179.0, 'C': 33.0, 'G': 51.0, 'T': 62.0},\n",
       "  2: {'A': 203.0, 'C': 12.0, 'G': 27.0, 'T': 91.0},\n",
       "  3: {'A': 251.0, 'C': 12.0, 'G': 28.0, 'T': 47.0},\n",
       "  4: {'A': 51.0, 'C': 86.0, 'G': 94.0, 'T': 94.0},\n",
       "  5: {'A': 16.0, 'C': 197.0, 'G': 115.0, 'T': 1.0},\n",
       "  6: {'A': 325.0, 'C': 1.0, 'G': 1.0, 'T': 4.0},\n",
       "  7: {'A': 5.0, 'C': 1.0, 'G': 1.0, 'T': 318.0},\n",
       "  8: {'A': 97.0, 'C': 1.0, 'G': 327.0, 'T': 1.0},\n",
       "  9: {'A': 324.0, 'C': 1.0, 'G': 2.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 325.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 328.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 324.0},\n",
       "  13: {'A': 1.0, 'C': 321.0, 'G': 117.0, 'T': 1.0},\n",
       "  14: {'A': 323.0, 'C': 16.0, 'G': 1.0, 'T': 1.0},\n",
       "  15: {'A': 1.0, 'C': 64.0, 'G': 1.0, 'T': 260.0},\n",
       "  16: {'A': 3.0, 'C': 101.0, 'G': 190.0, 'T': 32.0},\n",
       "  17: {'A': 107.0, 'C': 95.0, 'G': 69.0, 'T': 55.0},\n",
       "  18: {'A': 66.0, 'C': 26.0, 'G': 2.0, 'T': 237.0},\n",
       "  19: {'A': 80.0, 'C': 10.0, 'G': 4.0, 'T': 225.0},\n",
       "  20: {'A': 77.0, 'C': 34.0, 'G': 47.0, 'T': 167.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 20,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '28461702',\n",
       "  'uniprot_ids': 'Q9BYV9',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1587.1',\n",
       "  'Matrix_Name': 'ZNF135',\n",
       "  1: {'A': 50.0, 'C': 596.0, 'G': 123.0, 'T': 183.0},\n",
       "  2: {'A': 17.0, 'C': 859.0, 'G': 23.0, 'T': 53.0},\n",
       "  3: {'A': 5.0, 'C': 38.0, 'G': 16.0, 'T': 893.0},\n",
       "  4: {'A': 8.0, 'C': 773.0, 'G': 11.0, 'T': 160.0},\n",
       "  5: {'A': 61.0, 'C': 26.0, 'G': 863.0, 'T': 2.0},\n",
       "  6: {'A': 887.0, 'C': 4.0, 'G': 53.0, 'T': 8.0},\n",
       "  7: {'A': 18.0, 'C': 897.0, 'G': 27.0, 'T': 10.0},\n",
       "  8: {'A': 2.0, 'C': 940.0, 'G': 3.0, 'T': 7.0},\n",
       "  9: {'A': 2.0, 'C': 7.0, 'G': 3.0, 'T': 940.0},\n",
       "  10: {'A': 4.0, 'C': 937.0, 'G': 7.0, 'T': 4.0},\n",
       "  11: {'A': 37.0, 'C': 801.0, 'G': 8.0, 'T': 106.0},\n",
       "  12: {'A': 56.0, 'C': 400.0, 'G': 49.0, 'T': 447.0},\n",
       "  13: {'A': 304.0, 'C': 67.0, 'G': 524.0, 'T': 57.0},\n",
       "  14: {'A': 450.0, 'C': 36.0, 'G': 422.0, 'T': 44.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28092692',\n",
       "  'uniprot_ids': 'P52742',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0523.1',\n",
       "  'Matrix_Name': 'TCF7L2',\n",
       "  1: {'A': 1526.0, 'C': 757.0, 'G': 1141.0, 'T': 768.0},\n",
       "  2: {'A': 1809.0, 'C': 841.0, 'G': 1154.0, 'T': 388.0},\n",
       "  3: {'A': 3071.0, 'C': 137.0, 'G': 560.0, 'T': 424.0},\n",
       "  4: {'A': 86.0, 'C': 1948.0, 'G': 2009.0, 'T': 149.0},\n",
       "  5: {'A': 2796.0, 'C': 27.0, 'G': 1.0, 'T': 1368.0},\n",
       "  6: {'A': 139.0, 'C': 1.0, 'G': 1.0, 'T': 4051.0},\n",
       "  7: {'A': 156.0, 'C': 3305.0, 'G': 730.0, 'T': 1.0},\n",
       "  8: {'A': 4189.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 4104.0, 'C': 16.0, 'G': 71.0, 'T': 1.0},\n",
       "  10: {'A': 4170.0, 'C': 1.0, 'G': 20.0, 'T': 1.0},\n",
       "  11: {'A': 304.0, 'C': 38.0, 'G': 3849.0, 'T': 1.0},\n",
       "  12: {'A': 1036.0, 'C': 850.0, 'G': 2009.0, 'T': 297.0},\n",
       "  13: {'A': 1399.0, 'C': 1124.0, 'G': 1211.0, 'T': 458.0},\n",
       "  14: {'A': 1783.0, 'C': 827.0, 'G': 758.0, 'T': 824.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'TCF-7-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '18268006',\n",
       "  'uniprot_ids': 'Q9NQB0',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1113.2',\n",
       "  'Matrix_Name': 'PBX2',\n",
       "  1: {'A': 8364.0, 'C': 5864.0, 'G': 9626.0, 'T': 5746.0},\n",
       "  2: {'A': 8308.0, 'C': 8860.0, 'G': 3048.0, 'T': 9384.0},\n",
       "  3: {'A': 4926.0, 'C': 19646.0, 'G': 1586.0, 'T': 3442.0},\n",
       "  4: {'A': 26621.0, 'C': 630.0, 'G': 1711.0, 'T': 638.0},\n",
       "  5: {'A': 226.0, 'C': 589.0, 'G': 318.0, 'T': 28467.0},\n",
       "  6: {'A': 24369.0, 'C': 2267.0, 'G': 818.0, 'T': 2146.0},\n",
       "  7: {'A': 27246.0, 'C': 575.0, 'G': 344.0, 'T': 1435.0},\n",
       "  8: {'A': 27260.0, 'C': 755.0, 'G': 570.0, 'T': 1015.0},\n",
       "  9: {'A': 705.0, 'C': 454.0, 'G': 328.0, 'T': 28113.0},\n",
       "  10: {'A': 2299.0, 'C': 23895.0, 'G': 556.0, 'T': 2850.0},\n",
       "  11: {'A': 25946.0, 'C': 849.0, 'G': 353.0, 'T': 2452.0},\n",
       "  12: {'A': 6994.0, 'C': 7058.0, 'G': 3953.0, 'T': 11595.0},\n",
       "  13: {'A': 7898.0, 'C': 7087.0, 'G': 5575.0, 'T': 9040.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '12054735',\n",
       "  'uniprot_ids': 'P40425',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0668.1',\n",
       "  'Matrix_Name': 'NEUROD2',\n",
       "  1: {'A': 605.0, 'C': 120.0, 'G': 770.0, 'T': 96.0},\n",
       "  2: {'A': 528.0, 'C': 876.0, 'G': 241.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1588.0, 'G': 96.0, 'T': 1.0},\n",
       "  4: {'A': 1588.0, 'C': 1.0, 'G': 309.0, 'T': 53.0},\n",
       "  5: {'A': 1.0, 'C': 157.0, 'G': 1.0, 'T': 1588.0},\n",
       "  6: {'A': 1588.0, 'C': 121.0, 'G': 39.0, 'T': 5.0},\n",
       "  7: {'A': 1.0, 'C': 329.0, 'G': 1.0, 'T': 1588.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1588.0, 'T': 113.0},\n",
       "  9: {'A': 1.0, 'C': 374.0, 'G': 706.0, 'T': 549.0},\n",
       "  10: {'A': 210.0, 'C': 593.0, 'G': 102.0, 'T': 686.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '22445365',\n",
       "  'uniprot_ids': 'Q15784',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0615.1',\n",
       "  'Matrix_Name': 'Gmeb1',\n",
       "  1: {'A': 18.0, 'C': 27.0, 'G': 36.0, 'T': 23.0},\n",
       "  2: {'A': 35.0, 'C': 32.0, 'G': 16.0, 'T': 21.0},\n",
       "  3: {'A': 12.0, 'C': 24.0, 'G': 36.0, 'T': 32.0},\n",
       "  4: {'A': 14.0, 'C': 23.0, 'G': 30.0, 'T': 37.0},\n",
       "  5: {'A': 14.0, 'C': 8.0, 'G': 41.0, 'T': 41.0},\n",
       "  6: {'A': 6.0, 'C': 5.0, 'G': 34.0, 'T': 60.0},\n",
       "  7: {'A': 72.0, 'C': 2.0, 'G': 29.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 100.0, 'G': 1.0, 'T': 2.0},\n",
       "  9: {'A': 2.0, 'C': 1.0, 'G': 100.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 29.0, 'G': 2.0, 'T': 72.0},\n",
       "  11: {'A': 60.0, 'C': 34.0, 'G': 5.0, 'T': 6.0},\n",
       "  12: {'A': 41.0, 'C': 41.0, 'G': 8.0, 'T': 14.0},\n",
       "  13: {'A': 22.0, 'C': 24.0, 'G': 38.0, 'T': 20.0},\n",
       "  14: {'A': 45.0, 'C': 16.0, 'G': 19.0, 'T': 25.0},\n",
       "  15: {'A': 19.0, 'C': 27.0, 'G': 24.0, 'T': 34.0},\n",
       "  16: {'A': 28.0, 'C': 22.0, 'G': 32.0, 'T': 21.0},\n",
       "  17: {'A': 25.0, 'C': 26.0, 'G': 28.0, 'T': 25.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'GMEB',\n",
       "  'tf_class': 'SAND',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q9JL60',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0500.2',\n",
       "  'Matrix_Name': 'MYOG',\n",
       "  1: {'A': 5311.0, 'C': 6382.0, 'G': 5597.0, 'T': 5071.0},\n",
       "  2: {'A': 8429.0, 'C': 3755.0, 'G': 5484.0, 'T': 4693.0},\n",
       "  3: {'A': 7160.0, 'C': 2731.0, 'G': 10791.0, 'T': 1679.0},\n",
       "  4: {'A': 87.0, 'C': 21989.0, 'G': 102.0, 'T': 183.0},\n",
       "  5: {'A': 21783.0, 'C': 177.0, 'G': 237.0, 'T': 164.0},\n",
       "  6: {'A': 96.0, 'C': 131.0, 'G': 21964.0, 'T': 170.0},\n",
       "  7: {'A': 168.0, 'C': 21965.0, 'G': 132.0, 'T': 96.0},\n",
       "  8: {'A': 164.0, 'C': 239.0, 'G': 180.0, 'T': 21778.0},\n",
       "  9: {'A': 183.0, 'C': 103.0, 'G': 21986.0, 'T': 89.0},\n",
       "  10: {'A': 1668.0, 'C': 10793.0, 'G': 2707.0, 'T': 7193.0},\n",
       "  11: {'A': 4668.0, 'C': 5498.0, 'G': 3719.0, 'T': 8476.0},\n",
       "  12: {'A': 5019.0, 'C': 5661.0, 'G': 6370.0, 'T': 5311.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '16437161',\n",
       "  'uniprot_ids': 'P15173',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1152.1',\n",
       "  'Matrix_Name': 'SOX15',\n",
       "  1: {'A': 33.0, 'C': 555.0, 'G': 148.0, 'T': 268.0},\n",
       "  2: {'A': 63.0, 'C': 413.0, 'G': 27.0, 'T': 502.0},\n",
       "  3: {'A': 632.0, 'C': 4.0, 'G': 6.0, 'T': 363.0},\n",
       "  4: {'A': 4.0, 'C': 4.0, 'G': 7.0, 'T': 989.0},\n",
       "  5: {'A': 4.0, 'C': 3.0, 'G': 3.0, 'T': 995.0},\n",
       "  6: {'A': 84.0, 'C': 22.0, 'G': 891.0, 'T': 6.0},\n",
       "  7: {'A': 25.0, 'C': 3.0, 'G': 4.0, 'T': 972.0},\n",
       "  8: {'A': 181.0, 'C': 189.0, 'G': 126.0, 'T': 507.0},\n",
       "  9: {'A': 216.0, 'C': 254.0, 'G': 115.0, 'T': 419.0},\n",
       "  10: {'A': 242.0, 'C': 189.0, 'G': 166.0, 'T': 408.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '15863505',\n",
       "  'uniprot_ids': 'O60248',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1137.1',\n",
       "  'Matrix_Name': 'FOSL1::JUNB',\n",
       "  1: {'A': 238.0, 'C': 270.0, 'G': 196.0, 'T': 301.0},\n",
       "  2: {'A': 217.0, 'C': 147.0, 'G': 377.0, 'T': 264.0},\n",
       "  3: {'A': 567.0, 'C': 101.0, 'G': 288.0, 'T': 48.0},\n",
       "  4: {'A': 24.0, 'C': 17.0, 'G': 17.0, 'T': 946.0},\n",
       "  5: {'A': 37.0, 'C': 22.0, 'G': 881.0, 'T': 64.0},\n",
       "  6: {'A': 918.0, 'C': 21.0, 'G': 22.0, 'T': 43.0},\n",
       "  7: {'A': 64.0, 'C': 627.0, 'G': 199.0, 'T': 114.0},\n",
       "  8: {'A': 60.0, 'C': 24.0, 'G': 116.0, 'T': 804.0},\n",
       "  9: {'A': 272.0, 'C': 684.0, 'G': 10.0, 'T': 38.0},\n",
       "  10: {'A': 905.0, 'C': 22.0, 'G': 39.0, 'T': 38.0},\n",
       "  11: {'A': 54.0, 'C': 226.0, 'G': 171.0, 'T': 553.0},\n",
       "  12: {'A': 301.0, 'C': 299.0, 'G': 205.0, 'T': 199.0},\n",
       "  13: {'A': 248.0, 'C': 301.0, 'G': 224.0, 'T': 231.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15407;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0488.1',\n",
       "  'Matrix_Name': 'JUN',\n",
       "  1: {'A': 8142.0, 'C': 2793.0, 'G': 4895.0, 'T': 5142.0},\n",
       "  2: {'A': 7117.0, 'C': 2939.0, 'G': 4986.0, 'T': 5930.0},\n",
       "  3: {'A': 6497.0, 'C': 1904.0, 'G': 9683.0, 'T': 2888.0},\n",
       "  4: {'A': 16527.0, 'C': 730.0, 'G': 3714.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 20969.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 20850.0, 'T': 120.0},\n",
       "  7: {'A': 20968.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 1.0, 'C': 3439.0, 'G': 4513.0, 'T': 13019.0},\n",
       "  9: {'A': 1.0, 'C': 609.0, 'G': 20361.0, 'T': 1.0},\n",
       "  10: {'A': 939.0, 'C': 3532.0, 'G': 1.0, 'T': 16500.0},\n",
       "  11: {'A': 6905.0, 'C': 14065.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 20969.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 204.0, 'C': 3705.0, 'G': 1017.0, 'T': 16046.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21703547',\n",
       "  'uniprot_ids': 'P05412',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0143.4',\n",
       "  'Matrix_Name': 'SOX2',\n",
       "  1: {'A': 29106.0, 'C': 11042.0, 'G': 21435.0, 'T': 24175.0},\n",
       "  2: {'A': 30280.0, 'C': 15155.0, 'G': 17158.0, 'T': 23165.0},\n",
       "  3: {'A': 83516.0, 'C': 437.0, 'G': 419.0, 'T': 1386.0},\n",
       "  4: {'A': 1308.0, 'C': 78357.0, 'G': 1244.0, 'T': 4849.0},\n",
       "  5: {'A': 81588.0, 'C': 1206.0, 'G': 551.0, 'T': 2413.0},\n",
       "  6: {'A': 80826.0, 'C': 1743.0, 'G': 2003.0, 'T': 1186.0},\n",
       "  7: {'A': 4816.0, 'C': 733.0, 'G': 931.0, 'T': 79278.0},\n",
       "  8: {'A': 9927.0, 'C': 801.0, 'G': 73373.0, 'T': 1657.0},\n",
       "  9: {'A': 14773.0, 'C': 6265.0, 'G': 54415.0, 'T': 10305.0},\n",
       "  10: {'A': 28861.0, 'C': 17159.0, 'G': 17766.0, 'T': 21972.0},\n",
       "  11: {'A': 28961.0, 'C': 17476.0, 'G': 15999.0, 'T': 23322.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '15863505',\n",
       "  'uniprot_ids': 'P48431',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0768.1',\n",
       "  'Matrix_Name': 'LEF1',\n",
       "  1: {'A': 602.0, 'C': 140.0, 'G': 216.0, 'T': 139.0},\n",
       "  2: {'A': 656.0, 'C': 27.0, 'G': 208.0, 'T': 64.0},\n",
       "  3: {'A': 698.0, 'C': 8.0, 'G': 35.0, 'T': 15.0},\n",
       "  4: {'A': 6.0, 'C': 243.0, 'G': 1054.0, 'T': 4.0},\n",
       "  5: {'A': 700.0, 'C': 1.0, 'G': 5.0, 'T': 16.0},\n",
       "  6: {'A': 10.0, 'C': 4.0, 'G': 1.0, 'T': 899.0},\n",
       "  7: {'A': 15.0, 'C': 936.0, 'G': 50.0, 'T': 6.0},\n",
       "  8: {'A': 712.0, 'C': 1.0, 'G': 1.0, 'T': 5.0},\n",
       "  9: {'A': 710.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  10: {'A': 713.0, 'C': 1.0, 'G': 14.0, 'T': 5.0},\n",
       "  11: {'A': 34.0, 'C': 25.0, 'G': 1069.0, 'T': 1.0},\n",
       "  12: {'A': 198.0, 'C': 185.0, 'G': 852.0, 'T': 125.0},\n",
       "  13: {'A': 310.0, 'C': 165.0, 'G': 560.0, 'T': 122.0},\n",
       "  14: {'A': 368.0, 'C': 64.0, 'G': 89.0, 'T': 373.0},\n",
       "  15: {'A': 367.0, 'C': 96.0, 'G': 78.0, 'T': 423.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'TCF-7-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q9UJU2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0803.1',\n",
       "  'Matrix_Name': 'TBX15',\n",
       "  1: {'A': 10503.0, 'C': 75.0, 'G': 665.0, 'T': 113.0},\n",
       "  2: {'A': 459.0, 'C': 102.0, 'G': 10503.0, 'T': 104.0},\n",
       "  3: {'A': 20.0, 'C': 8.0, 'G': 10503.0, 'T': 1.0},\n",
       "  4: {'A': 62.0, 'C': 176.0, 'G': 246.0, 'T': 10503.0},\n",
       "  5: {'A': 1.0, 'C': 9.0, 'G': 10503.0, 'T': 1.0},\n",
       "  6: {'A': 141.0, 'C': 436.0, 'G': 325.0, 'T': 10503.0},\n",
       "  7: {'A': 45.0, 'C': 455.0, 'G': 10503.0, 'T': 938.0},\n",
       "  8: {'A': 10503.0, 'C': 72.0, 'G': 307.0, 'T': 97.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TBX1-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '12093383',\n",
       "  'uniprot_ids': 'Q96SF7',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0132.2',\n",
       "  'Matrix_Name': 'PDX1',\n",
       "  1: {'A': 1484.0, 'C': 2207.0, 'G': 2442.0, 'T': 1000.0},\n",
       "  2: {'A': 293.0, 'C': 2470.0, 'G': 1.0, 'T': 4662.0},\n",
       "  3: {'A': 4840.0, 'C': 2292.0, 'G': 186.0, 'T': 79.0},\n",
       "  4: {'A': 7131.0, 'C': 46.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7131.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 7131.0},\n",
       "  7: {'A': 7131.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 2182.0, 'C': 1217.0, 'G': 2882.0, 'T': 853.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'P52945',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0837.1',\n",
       "  'Matrix_Name': 'CEBPE',\n",
       "  1: {'A': 1786.0, 'C': 905.0, 'G': 1080.0, 'T': 207.0},\n",
       "  2: {'A': 29.0, 'C': 51.0, 'G': 16.0, 'T': 3975.0},\n",
       "  3: {'A': 22.0, 'C': 10.0, 'G': 545.0, 'T': 3975.0},\n",
       "  4: {'A': 1820.0, 'C': 8.0, 'G': 2156.0, 'T': 479.0},\n",
       "  5: {'A': 120.0, 'C': 3975.0, 'G': 76.0, 'T': 575.0},\n",
       "  6: {'A': 765.0, 'C': 175.0, 'G': 3975.0, 'T': 142.0},\n",
       "  7: {'A': 627.0, 'C': 3975.0, 'G': 14.0, 'T': 460.0},\n",
       "  8: {'A': 3975.0, 'C': 201.0, 'G': 1.0, 'T': 5.0},\n",
       "  9: {'A': 3975.0, 'C': 1.0, 'G': 1.0, 'T': 5.0},\n",
       "  10: {'A': 56.0, 'C': 1851.0, 'G': 209.0, 'T': 2125.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8632009',\n",
       "  'uniprot_ids': 'Q15744',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0779.1',\n",
       "  'Matrix_Name': 'PAX1',\n",
       "  1: {'A': 315.0, 'C': 3158.0, 'G': 1424.0, 'T': 328.0},\n",
       "  2: {'A': 4.0, 'C': 1.0, 'G': 5267.0, 'T': 80.0},\n",
       "  3: {'A': 235.0, 'C': 81.0, 'G': 1.0, 'T': 4397.0},\n",
       "  4: {'A': 4.0, 'C': 4491.0, 'G': 2.0, 'T': 476.0},\n",
       "  5: {'A': 3575.0, 'C': 214.0, 'G': 3.0, 'T': 1.0},\n",
       "  6: {'A': 3.0, 'C': 4490.0, 'G': 1.0, 'T': 652.0},\n",
       "  7: {'A': 4.0, 'C': 3.0, 'G': 5305.0, 'T': 4.0},\n",
       "  8: {'A': 1.0, 'C': 4405.0, 'G': 1018.0, 'T': 1.0},\n",
       "  9: {'A': 2216.0, 'C': 82.0, 'G': 44.0, 'T': 1530.0},\n",
       "  10: {'A': 12.0, 'C': 426.0, 'G': 3.0, 'T': 4291.0},\n",
       "  11: {'A': 2.0, 'C': 1770.0, 'G': 4653.0, 'T': 55.0},\n",
       "  12: {'A': 3580.0, 'C': 3.0, 'G': 596.0, 'T': 1.0},\n",
       "  13: {'A': 825.0, 'C': 1630.0, 'G': 1231.0, 'T': 745.0},\n",
       "  14: {'A': 75.0, 'C': 613.0, 'G': 33.0, 'T': 4186.0},\n",
       "  15: {'A': 850.0, 'C': 12.0, 'G': 4290.0, 'T': 881.0},\n",
       "  16: {'A': 1454.0, 'C': 2562.0, 'G': 1148.0, 'T': 1.0},\n",
       "  17: {'A': 2644.0, 'C': 2137.0, 'G': 781.0, 'T': 1291.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired',\n",
       "  'tf_class': 'Paired',\n",
       "  'pubmed_ids': '19132093',\n",
       "  'uniprot_ids': 'P15863',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1555.1',\n",
       "  'Matrix_Name': 'RXRB(var.2)',\n",
       "  1: {'A': 615.0, 'C': 191.0, 'G': 852.0, 'T': 149.0},\n",
       "  2: {'A': 935.0, 'C': 60.0, 'G': 871.0, 'T': 16.0},\n",
       "  3: {'A': 60.0, 'C': 30.0, 'G': 1804.0, 'T': 104.0},\n",
       "  4: {'A': 28.0, 'C': 40.0, 'G': 1804.0, 'T': 271.0},\n",
       "  5: {'A': 36.0, 'C': 86.0, 'G': 158.0, 'T': 1804.0},\n",
       "  6: {'A': 24.0, 'C': 1804.0, 'G': 42.0, 'T': 124.0},\n",
       "  7: {'A': 1208.0, 'C': 16.0, 'G': 597.0, 'T': 8.0},\n",
       "  8: {'A': 12.0, 'C': 60.0, 'G': 1.0, 'T': 1804.0},\n",
       "  9: {'A': 93.0, 'C': 36.0, 'G': 1804.0, 'T': 9.0},\n",
       "  10: {'A': 1804.0, 'C': 54.0, 'G': 42.0, 'T': 15.0},\n",
       "  11: {'A': 310.0, 'C': 1804.0, 'G': 12.0, 'T': 24.0},\n",
       "  12: {'A': 59.0, 'C': 1804.0, 'G': 33.0, 'T': 59.0},\n",
       "  13: {'A': 19.0, 'C': 867.0, 'G': 15.0, 'T': 938.0},\n",
       "  14: {'A': 125.0, 'C': 869.0, 'G': 195.0, 'T': 618.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'P28702',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0631.1',\n",
       "  'Matrix_Name': 'Six3',\n",
       "  1: {'A': 34.0, 'C': 10.0, 'G': 37.0, 'T': 22.0},\n",
       "  2: {'A': 73.0, 'C': 9.0, 'G': 9.0, 'T': 13.0},\n",
       "  3: {'A': 24.0, 'C': 6.0, 'G': 18.0, 'T': 56.0},\n",
       "  4: {'A': 48.0, 'C': 11.0, 'G': 33.0, 'T': 11.0},\n",
       "  5: {'A': 8.0, 'C': 8.0, 'G': 83.0, 'T': 5.0},\n",
       "  6: {'A': 3.0, 'C': 3.0, 'G': 91.0, 'T': 8.0},\n",
       "  7: {'A': 17.0, 'C': 2.0, 'G': 84.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 5.0, 'T': 96.0},\n",
       "  9: {'A': 98.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 100.0},\n",
       "  11: {'A': 2.0, 'C': 98.0, 'G': 1.0, 'T': 2.0},\n",
       "  12: {'A': 93.0, 'C': 4.0, 'G': 4.0, 'T': 3.0},\n",
       "  13: {'A': 20.0, 'C': 46.0, 'G': 10.0, 'T': 28.0},\n",
       "  14: {'A': 26.0, 'C': 24.0, 'G': 21.0, 'T': 33.0},\n",
       "  15: {'A': 41.0, 'C': 9.0, 'G': 12.0, 'T': 41.0},\n",
       "  16: {'A': 35.0, 'C': 23.0, 'G': 19.0, 'T': 28.0},\n",
       "  17: {'A': 9.0, 'C': 12.0, 'G': 25.0, 'T': 59.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'HD-SINE',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585360',\n",
       "  'uniprot_ids': 'Q62233',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0593.1',\n",
       "  'Matrix_Name': 'FOXP2',\n",
       "  1: {'A': 331.0, 'C': 114.0, 'G': 191.0, 'T': 134.0},\n",
       "  2: {'A': 341.0, 'C': 8.0, 'G': 143.0, 'T': 278.0},\n",
       "  3: {'A': 161.0, 'C': 1.0, 'G': 607.0, 'T': 1.0},\n",
       "  4: {'A': 60.0, 'C': 1.0, 'G': 1.0, 'T': 708.0},\n",
       "  5: {'A': 767.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 767.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 767.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 758.0, 'G': 1.0, 'T': 10.0},\n",
       "  9: {'A': 767.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 298.0, 'C': 155.0, 'G': 294.0, 'T': 23.0},\n",
       "  11: {'A': 333.0, 'C': 125.0, 'G': 185.0, 'T': 127.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23625967',\n",
       "  'uniprot_ids': 'O15409',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0599.1',\n",
       "  'Matrix_Name': 'KLF5',\n",
       "  1: {'A': 1430.0, 'C': 2024.0, 'G': 7573.0, 'T': 2588.0},\n",
       "  2: {'A': 1.0, 'C': 11901.0, 'G': 1.0, 'T': 1712.0},\n",
       "  3: {'A': 1.0, 'C': 12009.0, 'G': 1.0, 'T': 1604.0},\n",
       "  4: {'A': 3478.0, 'C': 9570.0, 'G': 1.0, 'T': 566.0},\n",
       "  5: {'A': 1.0, 'C': 13612.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 5052.0, 'C': 1.0, 'G': 5183.0, 'T': 3379.0},\n",
       "  7: {'A': 1.0, 'C': 13612.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 13612.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 13136.0, 'G': 1.0, 'T': 477.0},\n",
       "  10: {'A': 3916.0, 'C': 5596.0, 'G': 1.0, 'T': 4102.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '15740636',\n",
       "  'uniprot_ids': 'Q13887',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0623.2',\n",
       "  'Matrix_Name': 'NEUROG1',\n",
       "  1: {'A': 1659.0, 'C': 119.0, 'G': 1861.0, 'T': 34.0},\n",
       "  2: {'A': 1899.0, 'C': 877.0, 'G': 730.0, 'T': 15.0},\n",
       "  3: {'A': 1.0, 'C': 3519.0, 'G': 9.0, 'T': 1.0},\n",
       "  4: {'A': 3519.0, 'C': 1.0, 'G': 64.0, 'T': 1.0},\n",
       "  5: {'A': 5.0, 'C': 115.0, 'G': 1.0, 'T': 3519.0},\n",
       "  6: {'A': 3519.0, 'C': 1.0, 'G': 81.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 87.0, 'G': 1.0, 'T': 3519.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 3519.0, 'T': 1.0},\n",
       "  9: {'A': 24.0, 'C': 656.0, 'G': 908.0, 'T': 1934.0},\n",
       "  10: {'A': 69.0, 'C': 1845.0, 'G': 188.0, 'T': 1674.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q92886',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1634.1',\n",
       "  'Matrix_Name': 'BATF',\n",
       "  1: {'A': 13556.0, 'C': 5743.0, 'G': 11493.0, 'T': 16220.0},\n",
       "  2: {'A': 22576.0, 'C': 8932.0, 'G': 8628.0, 'T': 6876.0},\n",
       "  3: {'A': 484.0, 'C': 356.0, 'G': 284.0, 'T': 45888.0},\n",
       "  4: {'A': 1140.0, 'C': 571.0, 'G': 40618.0, 'T': 4683.0},\n",
       "  5: {'A': 45947.0, 'C': 251.0, 'G': 342.0, 'T': 472.0},\n",
       "  6: {'A': 2052.0, 'C': 37334.0, 'G': 5574.0, 'T': 2052.0},\n",
       "  7: {'A': 1131.0, 'C': 307.0, 'G': 837.0, 'T': 44737.0},\n",
       "  8: {'A': 4743.0, 'C': 40280.0, 'G': 712.0, 'T': 1277.0},\n",
       "  9: {'A': 45317.0, 'C': 432.0, 'G': 508.0, 'T': 755.0},\n",
       "  10: {'A': 8147.0, 'C': 7567.0, 'G': 9682.0, 'T': 21616.0},\n",
       "  11: {'A': 16140.0, 'C': 12049.0, 'G': 5920.0, 'T': 12903.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'B-ATF-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '29588546',\n",
       "  'uniprot_ids': 'Q16520',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0481.3',\n",
       "  'Matrix_Name': 'FOXP1',\n",
       "  1: {'A': 26983.0, 'C': 13909.0, 'G': 16636.0, 'T': 17194.0},\n",
       "  2: {'A': 26880.0, 'C': 6210.0, 'G': 14137.0, 'T': 27495.0},\n",
       "  3: {'A': 10464.0, 'C': 1574.0, 'G': 59572.0, 'T': 3112.0},\n",
       "  4: {'A': 1210.0, 'C': 728.0, 'G': 921.0, 'T': 71863.0},\n",
       "  5: {'A': 69604.0, 'C': 2096.0, 'G': 1074.0, 'T': 1948.0},\n",
       "  6: {'A': 65237.0, 'C': 5068.0, 'G': 1831.0, 'T': 2586.0},\n",
       "  7: {'A': 70031.0, 'C': 1488.0, 'G': 1133.0, 'T': 2070.0},\n",
       "  8: {'A': 1337.0, 'C': 61095.0, 'G': 1494.0, 'T': 10796.0},\n",
       "  9: {'A': 68457.0, 'C': 1905.0, 'G': 750.0, 'T': 3610.0},\n",
       "  10: {'A': 20211.0, 'C': 17236.0, 'G': 15847.0, 'T': 21428.0},\n",
       "  11: {'A': 25975.0, 'C': 14313.0, 'G': 12810.0, 'T': 21624.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '21924763',\n",
       "  'uniprot_ids': 'Q9H334',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1575.1',\n",
       "  'Matrix_Name': 'THRB(var.2)',\n",
       "  1: {'A': 1011.0, 'C': 625.0, 'G': 1641.0, 'T': 838.0},\n",
       "  2: {'A': 27.0, 'C': 619.0, 'G': 16.0, 'T': 4112.0},\n",
       "  3: {'A': 253.0, 'C': 643.0, 'G': 4112.0, 'T': 231.0},\n",
       "  4: {'A': 4112.0, 'C': 352.0, 'G': 433.0, 'T': 1788.0},\n",
       "  5: {'A': 30.0, 'C': 4112.0, 'G': 3.0, 'T': 1.0},\n",
       "  6: {'A': 2.0, 'C': 4112.0, 'G': 3.0, 'T': 9.0},\n",
       "  7: {'A': 1.0, 'C': 1179.0, 'G': 78.0, 'T': 4112.0},\n",
       "  8: {'A': 287.0, 'C': 1513.0, 'G': 873.0, 'T': 1442.0},\n",
       "  9: {'A': 2455.0, 'C': 242.0, 'G': 1354.0, 'T': 64.0},\n",
       "  10: {'A': 1061.0, 'C': 1034.0, 'G': 940.0, 'T': 1080.0},\n",
       "  11: {'A': 64.0, 'C': 1267.0, 'G': 313.0, 'T': 2471.0},\n",
       "  12: {'A': 1470.0, 'C': 905.0, 'G': 1489.0, 'T': 252.0},\n",
       "  13: {'A': 4112.0, 'C': 107.0, 'G': 1234.0, 'T': 3.0},\n",
       "  14: {'A': 1.0, 'C': 4.0, 'G': 4112.0, 'T': 1.0},\n",
       "  15: {'A': 1.0, 'C': 1.0, 'G': 4112.0, 'T': 31.0},\n",
       "  16: {'A': 1689.0, 'C': 471.0, 'G': 304.0, 'T': 4112.0},\n",
       "  17: {'A': 200.0, 'C': 4112.0, 'G': 687.0, 'T': 292.0},\n",
       "  18: {'A': 4112.0, 'C': 6.0, 'G': 630.0, 'T': 6.0},\n",
       "  19: {'A': 836.0, 'C': 1650.0, 'G': 611.0, 'T': 1018.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 19,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '19741045',\n",
       "  'uniprot_ids': 'P10828',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1585.1',\n",
       "  'Matrix_Name': 'ZKSCAN1',\n",
       "  1: {'A': 4634.0, 'C': 457.0, 'G': 585.0, 'T': 455.0},\n",
       "  2: {'A': 188.0, 'C': 2269.0, 'G': 365.0, 'T': 3309.0},\n",
       "  3: {'A': 6055.0, 'C': 54.0, 'G': 21.0, 'T': 1.0},\n",
       "  4: {'A': 19.0, 'C': 6.0, 'G': 6093.0, 'T': 13.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6128.0},\n",
       "  6: {'A': 6080.0, 'C': 49.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 13.0, 'C': 102.0, 'G': 5981.0, 'T': 35.0},\n",
       "  8: {'A': 14.0, 'C': 13.0, 'G': 6076.0, 'T': 28.0},\n",
       "  9: {'A': 606.0, 'C': 1304.0, 'G': 751.0, 'T': 3470.0},\n",
       "  10: {'A': 1274.0, 'C': 1481.0, 'G': 2881.0, 'T': 495.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28092692',\n",
       "  'uniprot_ids': 'P17029',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0646.1',\n",
       "  'Matrix_Name': 'GCM1',\n",
       "  1: {'A': 2523.0, 'C': 8554.0, 'G': 5155.0, 'T': 5211.0},\n",
       "  2: {'A': 21440.0, 'C': 1812.0, 'G': 4011.0, 'T': 262.0},\n",
       "  3: {'A': 162.0, 'C': 155.0, 'G': 97.0, 'T': 21440.0},\n",
       "  4: {'A': 4372.0, 'C': 275.0, 'G': 21440.0, 'T': 16.0},\n",
       "  5: {'A': 1003.0, 'C': 21440.0, 'G': 573.0, 'T': 881.0},\n",
       "  6: {'A': 595.0, 'C': 256.0, 'G': 21440.0, 'T': 3816.0},\n",
       "  7: {'A': 1.0, 'C': 5.0, 'G': 21440.0, 'T': 36.0},\n",
       "  8: {'A': 1.0, 'C': 19.0, 'G': 21440.0, 'T': 11.0},\n",
       "  9: {'A': 147.0, 'C': 4993.0, 'G': 148.0, 'T': 21440.0},\n",
       "  10: {'A': 21440.0, 'C': 2906.0, 'G': 7972.0, 'T': 2015.0},\n",
       "  11: {'A': 790.0, 'C': 13413.0, 'G': 4830.0, 'T': 2410.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'GCM',\n",
       "  'tf_class': 'GCM',\n",
       "  'pubmed_ids': '8962155',\n",
       "  'uniprot_ids': 'Q9NP62',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1637.1',\n",
       "  'Matrix_Name': 'EBF3',\n",
       "  1: {'A': 3819.0, 'C': 5156.0, 'G': 6103.0, 'T': 6988.0},\n",
       "  2: {'A': 2543.0, 'C': 6009.0, 'G': 3049.0, 'T': 10465.0},\n",
       "  3: {'A': 220.0, 'C': 21298.0, 'G': 362.0, 'T': 186.0},\n",
       "  4: {'A': 383.0, 'C': 20682.0, 'G': 262.0, 'T': 739.0},\n",
       "  5: {'A': 337.0, 'C': 21018.0, 'G': 325.0, 'T': 386.0},\n",
       "  6: {'A': 15090.0, 'C': 3635.0, 'G': 1437.0, 'T': 1904.0},\n",
       "  7: {'A': 15927.0, 'C': 1151.0, 'G': 3375.0, 'T': 1613.0},\n",
       "  8: {'A': 516.0, 'C': 261.0, 'G': 21189.0, 'T': 100.0},\n",
       "  9: {'A': 882.0, 'C': 231.0, 'G': 20590.0, 'T': 363.0},\n",
       "  10: {'A': 570.0, 'C': 516.0, 'G': 20516.0, 'T': 464.0},\n",
       "  11: {'A': 16663.0, 'C': 1820.0, 'G': 1986.0, 'T': 1597.0},\n",
       "  12: {'A': 6588.0, 'C': 6115.0, 'G': 5790.0, 'T': 3573.0},\n",
       "  13: {'A': 5693.0, 'C': 5488.0, 'G': 4567.0, 'T': 6318.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Early',\n",
       "  'tf_class': 'Rel',\n",
       "  'pubmed_ids': '20592035',\n",
       "  'uniprot_ids': 'Q9H4W6',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1142.1',\n",
       "  'Matrix_Name': 'FOSL1::JUND',\n",
       "  1: {'A': 266.0, 'C': 222.0, 'G': 280.0, 'T': 236.0},\n",
       "  2: {'A': 552.0, 'C': 128.0, 'G': 283.0, 'T': 41.0},\n",
       "  3: {'A': 31.0, 'C': 24.0, 'G': 35.0, 'T': 914.0},\n",
       "  4: {'A': 23.0, 'C': 62.0, 'G': 778.0, 'T': 141.0},\n",
       "  5: {'A': 865.0, 'C': 13.0, 'G': 83.0, 'T': 42.0},\n",
       "  6: {'A': 50.0, 'C': 582.0, 'G': 192.0, 'T': 181.0},\n",
       "  7: {'A': 99.0, 'C': 46.0, 'G': 188.0, 'T': 670.0},\n",
       "  8: {'A': 271.0, 'C': 642.0, 'G': 26.0, 'T': 65.0},\n",
       "  9: {'A': 913.0, 'C': 41.0, 'G': 18.0, 'T': 32.0},\n",
       "  10: {'A': 175.0, 'C': 213.0, 'G': 145.0, 'T': 471.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15407;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1574.1',\n",
       "  'Matrix_Name': 'THRB',\n",
       "  1: {'A': 2307.0, 'C': 1398.0, 'G': 3706.0, 'T': 1220.0},\n",
       "  2: {'A': 4048.0, 'C': 235.0, 'G': 4580.0, 'T': 75.0},\n",
       "  3: {'A': 168.0, 'C': 53.0, 'G': 8627.0, 'T': 393.0},\n",
       "  4: {'A': 105.0, 'C': 15.0, 'G': 8627.0, 'T': 1893.0},\n",
       "  5: {'A': 21.0, 'C': 133.0, 'G': 727.0, 'T': 8627.0},\n",
       "  6: {'A': 36.0, 'C': 8627.0, 'G': 825.0, 'T': 1852.0},\n",
       "  7: {'A': 8627.0, 'C': 17.0, 'G': 818.0, 'T': 14.0},\n",
       "  8: {'A': 8627.0, 'C': 166.0, 'G': 2026.0, 'T': 1588.0},\n",
       "  9: {'A': 8627.0, 'C': 44.0, 'G': 1901.0, 'T': 12.0},\n",
       "  10: {'A': 25.0, 'C': 7.0, 'G': 8627.0, 'T': 62.0},\n",
       "  11: {'A': 37.0, 'C': 15.0, 'G': 8627.0, 'T': 2011.0},\n",
       "  12: {'A': 24.0, 'C': 247.0, 'G': 1401.0, 'T': 8627.0},\n",
       "  13: {'A': 137.0, 'C': 8627.0, 'G': 727.0, 'T': 2556.0},\n",
       "  14: {'A': 8627.0, 'C': 149.0, 'G': 3669.0, 'T': 226.0},\n",
       "  15: {'A': 1937.0, 'C': 2191.0, 'G': 2161.0, 'T': 2341.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '19741045',\n",
       "  'uniprot_ids': 'P10828',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0864.2',\n",
       "  'Matrix_Name': 'E2F2',\n",
       "  1: {'A': 1372.0, 'C': 682.0, 'G': 1952.0, 'T': 672.0},\n",
       "  2: {'A': 1927.0, 'C': 317.0, 'G': 812.0, 'T': 2568.0},\n",
       "  3: {'A': 1330.0, 'C': 529.0, 'G': 1083.0, 'T': 3642.0},\n",
       "  4: {'A': 1252.0, 'C': 315.0, 'G': 877.0, 'T': 3873.0},\n",
       "  5: {'A': 1130.0, 'C': 164.0, 'G': 928.0, 'T': 4387.0},\n",
       "  6: {'A': 283.0, 'C': 1371.0, 'G': 4566.0, 'T': 34.0},\n",
       "  7: {'A': 3.0, 'C': 33.0, 'G': 4654.0, 'T': 2.0},\n",
       "  8: {'A': 21.0, 'C': 4670.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 18.0, 'C': 1.0, 'G': 4674.0, 'T': 5.0},\n",
       "  10: {'A': 4.0, 'C': 4646.0, 'G': 27.0, 'T': 2.0},\n",
       "  11: {'A': 349.0, 'C': 4435.0, 'G': 1103.0, 'T': 230.0},\n",
       "  12: {'A': 4320.0, 'C': 1193.0, 'G': 272.0, 'T': 1124.0},\n",
       "  13: {'A': 1765.0, 'C': 580.0, 'G': 119.0, 'T': 2760.0},\n",
       "  14: {'A': 1646.0, 'C': 517.0, 'G': 529.0, 'T': 2679.0},\n",
       "  15: {'A': 2184.0, 'C': 1017.0, 'G': 576.0, 'T': 2188.0},\n",
       "  16: {'A': 727.0, 'C': 1711.0, 'G': 804.0, 'T': 1436.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '18836037',\n",
       "  'uniprot_ids': 'Q14209',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0867.2',\n",
       "  'Matrix_Name': 'SOX4',\n",
       "  1: {'A': 3191.0, 'C': 1772.0, 'G': 5583.0, 'T': 2287.0},\n",
       "  2: {'A': 5583.0, 'C': 952.0, 'G': 2048.0, 'T': 432.0},\n",
       "  3: {'A': 5583.0, 'C': 1.0, 'G': 1.0, 'T': 10.0},\n",
       "  4: {'A': 1.0, 'C': 5583.0, 'G': 38.0, 'T': 34.0},\n",
       "  5: {'A': 5583.0, 'C': 2.0, 'G': 7.0, 'T': 1.0},\n",
       "  6: {'A': 5583.0, 'C': 7.0, 'G': 12.0, 'T': 1.0},\n",
       "  7: {'A': 5583.0, 'C': 121.0, 'G': 51.0, 'T': 2720.0},\n",
       "  8: {'A': 1019.0, 'C': 65.0, 'G': 5583.0, 'T': 84.0},\n",
       "  9: {'A': 2094.0, 'C': 1423.0, 'G': 3490.0, 'T': 175.0},\n",
       "  10: {'A': 1776.0, 'C': 1439.0, 'G': 1890.0, 'T': 481.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SOX-related',\n",
       "  'tf_class': 'High-mobility',\n",
       "  'pubmed_ids': '17599239',\n",
       "  'uniprot_ids': 'Q06945',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1615.1',\n",
       "  'Matrix_Name': 'Plagl1',\n",
       "  1: {'A': 2781.0, 'C': 4516.0, 'G': 4261.0, 'T': 2599.0},\n",
       "  2: {'A': 2889.0, 'C': 4402.0, 'G': 4031.0, 'T': 2835.0},\n",
       "  3: {'A': 2139.0, 'C': 7310.0, 'G': 3027.0, 'T': 1681.0},\n",
       "  4: {'A': 1920.0, 'C': 1472.0, 'G': 1339.0, 'T': 9426.0},\n",
       "  5: {'A': 198.0, 'C': 244.0, 'G': 13596.0, 'T': 119.0},\n",
       "  6: {'A': 1214.0, 'C': 357.0, 'G': 12262.0, 'T': 324.0},\n",
       "  7: {'A': 188.0, 'C': 267.0, 'G': 13518.0, 'T': 184.0},\n",
       "  8: {'A': 181.0, 'C': 337.0, 'G': 13496.0, 'T': 143.0},\n",
       "  9: {'A': 783.0, 'C': 12738.0, 'G': 314.0, 'T': 322.0},\n",
       "  10: {'A': 182.0, 'C': 13153.0, 'G': 423.0, 'T': 399.0},\n",
       "  11: {'A': 12065.0, 'C': 687.0, 'G': 1033.0, 'T': 372.0},\n",
       "  12: {'A': 2094.0, 'C': 4619.0, 'G': 4853.0, 'T': 2591.0},\n",
       "  13: {'A': 3415.0, 'C': 3615.0, 'G': 4568.0, 'T': 2559.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28985358',\n",
       "  'uniprot_ids': 'O35745',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1481.1',\n",
       "  'Matrix_Name': 'DRGX',\n",
       "  1: {'A': 1173.0, 'C': 8181.0, 'G': 2693.0, 'T': 6688.0},\n",
       "  2: {'A': 1080.0, 'C': 5998.0, 'G': 1112.0, 'T': 8871.0},\n",
       "  3: {'A': 14868.0, 'C': 502.0, 'G': 2038.0, 'T': 1.0},\n",
       "  4: {'A': 14868.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 14868.0},\n",
       "  6: {'A': 1.0, 'C': 814.0, 'G': 210.0, 'T': 14868.0},\n",
       "  7: {'A': 14868.0, 'C': 18.0, 'G': 963.0, 'T': 145.0},\n",
       "  8: {'A': 3118.0, 'C': 4652.0, 'G': 3996.0, 'T': 3106.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '25131932',\n",
       "  'uniprot_ids': 'A6NNA5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1138.1',\n",
       "  'Matrix_Name': 'FOSL2::JUNB',\n",
       "  1: {'A': 152.0, 'C': 165.0, 'G': 408.0, 'T': 278.0},\n",
       "  2: {'A': 622.0, 'C': 54.0, 'G': 315.0, 'T': 14.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 986.0, 'T': 15.0},\n",
       "  5: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 55.0, 'C': 569.0, 'G': 293.0, 'T': 86.0},\n",
       "  7: {'A': 2.0, 'C': 1.0, 'G': 27.0, 'T': 974.0},\n",
       "  8: {'A': 169.0, 'C': 832.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 36.0, 'C': 252.0, 'G': 113.0, 'T': 603.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15408;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1643.1',\n",
       "  'Matrix_Name': 'NFIB',\n",
       "  1: {'A': 1056.0, 'C': 947.0, 'G': 1024.0, 'T': 1393.0},\n",
       "  2: {'A': 1182.0, 'C': 840.0, 'G': 1299.0, 'T': 1099.0},\n",
       "  3: {'A': 425.0, 'C': 2912.0, 'G': 527.0, 'T': 556.0},\n",
       "  4: {'A': 274.0, 'C': 3143.0, 'G': 194.0, 'T': 809.0},\n",
       "  5: {'A': 134.0, 'C': 21.0, 'G': 163.0, 'T': 4102.0},\n",
       "  6: {'A': 6.0, 'C': 12.0, 'G': 4390.0, 'T': 12.0},\n",
       "  7: {'A': 17.0, 'C': 13.0, 'G': 4373.0, 'T': 17.0},\n",
       "  8: {'A': 149.0, 'C': 4224.0, 'G': 28.0, 'T': 19.0},\n",
       "  9: {'A': 3179.0, 'C': 520.0, 'G': 176.0, 'T': 545.0},\n",
       "  10: {'A': 444.0, 'C': 1191.0, 'G': 609.0, 'T': 2176.0},\n",
       "  11: {'A': 1088.0, 'C': 793.0, 'G': 1150.0, 'T': 1389.0},\n",
       "  12: {'A': 595.0, 'C': 533.0, 'G': 2823.0, 'T': 469.0},\n",
       "  13: {'A': 702.0, 'C': 182.0, 'G': 452.0, 'T': 3084.0},\n",
       "  14: {'A': 15.0, 'C': 32.0, 'G': 4192.0, 'T': 181.0},\n",
       "  15: {'A': 17.0, 'C': 4358.0, 'G': 24.0, 'T': 21.0},\n",
       "  16: {'A': 9.0, 'C': 4383.0, 'G': 12.0, 'T': 16.0},\n",
       "  17: {'A': 3938.0, 'C': 374.0, 'G': 37.0, 'T': 71.0},\n",
       "  18: {'A': 2990.0, 'C': 162.0, 'G': 1056.0, 'T': 212.0},\n",
       "  19: {'A': 553.0, 'C': 468.0, 'G': 2970.0, 'T': 429.0},\n",
       "  20: {'A': 1157.0, 'C': 1194.0, 'G': 875.0, 'T': 1194.0},\n",
       "  21: {'A': 1368.0, 'C': 1016.0, 'G': 985.0, 'T': 1051.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 21,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'O00712',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0629.1',\n",
       "  'Matrix_Name': 'Rhox11',\n",
       "  1: {'A': 55.0, 'C': 14.0, 'G': 16.0, 'T': 19.0},\n",
       "  2: {'A': 34.0, 'C': 29.0, 'G': 24.0, 'T': 18.0},\n",
       "  3: {'A': 18.0, 'C': 12.0, 'G': 39.0, 'T': 34.0},\n",
       "  4: {'A': 31.0, 'C': 29.0, 'G': 24.0, 'T': 20.0},\n",
       "  5: {'A': 14.0, 'C': 57.0, 'G': 7.0, 'T': 26.0},\n",
       "  6: {'A': 4.0, 'C': 2.0, 'G': 96.0, 'T': 2.0},\n",
       "  7: {'A': 10.0, 'C': 77.0, 'G': 16.0, 'T': 1.0},\n",
       "  8: {'A': 2.0, 'C': 1.0, 'G': 2.0, 'T': 100.0},\n",
       "  9: {'A': 5.0, 'C': 1.0, 'G': 95.0, 'T': 3.0},\n",
       "  10: {'A': 2.0, 'C': 2.0, 'G': 1.0, 'T': 100.0},\n",
       "  11: {'A': 57.0, 'C': 1.0, 'G': 1.0, 'T': 44.0},\n",
       "  12: {'A': 79.0, 'C': 5.0, 'G': 4.0, 'T': 16.0},\n",
       "  13: {'A': 53.0, 'C': 11.0, 'G': 2.0, 'T': 37.0},\n",
       "  14: {'A': 32.0, 'C': 12.0, 'G': 34.0, 'T': 25.0},\n",
       "  15: {'A': 25.0, 'C': 37.0, 'G': 29.0, 'T': 13.0},\n",
       "  16: {'A': 32.0, 'C': 17.0, 'G': 44.0, 'T': 11.0},\n",
       "  17: {'A': 42.0, 'C': 14.0, 'G': 11.0, 'T': 36.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q810N8',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA1131.1',\n",
       "  'Matrix_Name': 'FOSL2::JUN(var.2)',\n",
       "  1: {'A': 223.0, 'C': 95.0, 'G': 460.0, 'T': 225.0},\n",
       "  2: {'A': 634.0, 'C': 31.0, 'G': 324.0, 'T': 16.0},\n",
       "  3: {'A': 7.0, 'C': 9.0, 'G': 9.0, 'T': 979.0},\n",
       "  4: {'A': 8.0, 'C': 15.0, 'G': 918.0, 'T': 63.0},\n",
       "  5: {'A': 964.0, 'C': 7.0, 'G': 10.0, 'T': 24.0},\n",
       "  6: {'A': 12.0, 'C': 907.0, 'G': 23.0, 'T': 62.0},\n",
       "  7: {'A': 93.0, 'C': 46.0, 'G': 846.0, 'T': 18.0},\n",
       "  8: {'A': 15.0, 'C': 71.0, 'G': 10.0, 'T': 909.0},\n",
       "  9: {'A': 352.0, 'C': 626.0, 'G': 17.0, 'T': 9.0},\n",
       "  10: {'A': 962.0, 'C': 11.0, 'G': 14.0, 'T': 16.0},\n",
       "  11: {'A': 39.0, 'C': 219.0, 'G': 167.0, 'T': 579.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P15408;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0669.1',\n",
       "  'Matrix_Name': 'NEUROG2',\n",
       "  1: {'A': 247.0, 'C': 3.0, 'G': 181.0, 'T': 1.0},\n",
       "  2: {'A': 427.0, 'C': 151.0, 'G': 66.0, 'T': 1.0},\n",
       "  3: {'A': 2.0, 'C': 427.0, 'G': 1.0, 'T': 6.0},\n",
       "  4: {'A': 427.0, 'C': 1.0, 'G': 4.0, 'T': 1.0},\n",
       "  5: {'A': 5.0, 'C': 6.0, 'G': 21.0, 'T': 427.0},\n",
       "  6: {'A': 427.0, 'C': 9.0, 'G': 6.0, 'T': 1.0},\n",
       "  7: {'A': 6.0, 'C': 4.0, 'G': 1.0, 'T': 427.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 427.0, 'T': 1.0},\n",
       "  9: {'A': 4.0, 'C': 171.0, 'G': 148.0, 'T': 427.0},\n",
       "  10: {'A': 1.0, 'C': 308.0, 'G': 1.0, 'T': 120.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '19389376',\n",
       "  'uniprot_ids': 'Q9H2A3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0793.1',\n",
       "  'Matrix_Name': 'POU6F2',\n",
       "  1: {'A': 1849.0, 'C': 322.0, 'G': 583.0, 'T': 547.0},\n",
       "  2: {'A': 715.0, 'C': 1182.0, 'G': 2117.0, 'T': 267.0},\n",
       "  3: {'A': 240.0, 'C': 3298.0, 'G': 220.0, 'T': 258.0},\n",
       "  4: {'A': 76.0, 'C': 65.0, 'G': 115.0, 'T': 3298.0},\n",
       "  5: {'A': 1193.0, 'C': 2106.0, 'G': 51.0, 'T': 82.0},\n",
       "  6: {'A': 3298.0, 'C': 76.0, 'G': 1.0, 'T': 22.0},\n",
       "  7: {'A': 1.0, 'C': 5.0, 'G': 1.0, 'T': 3298.0},\n",
       "  8: {'A': 1.0, 'C': 13.0, 'G': 42.0, 'T': 3298.0},\n",
       "  9: {'A': 3298.0, 'C': 28.0, 'G': 23.0, 'T': 54.0},\n",
       "  10: {'A': 1269.0, 'C': 547.0, 'G': 206.0, 'T': 1279.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '9852081',\n",
       "  'uniprot_ids': 'P78424',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1512.1',\n",
       "  'Matrix_Name': 'KLF11',\n",
       "  1: {'A': 1883.0, 'C': 2055.0, 'G': 3931.0, 'T': 1300.0},\n",
       "  2: {'A': 2023.0, 'C': 5550.0, 'G': 402.0, 'T': 1193.0},\n",
       "  3: {'A': 651.0, 'C': 8663.0, 'G': 157.0, 'T': 507.0},\n",
       "  4: {'A': 8656.0, 'C': 1290.0, 'G': 116.0, 'T': 28.0},\n",
       "  5: {'A': 1.0, 'C': 9143.0, 'G': 1.0, 'T': 39.0},\n",
       "  6: {'A': 2427.0, 'C': 53.0, 'G': 8292.0, 'T': 350.0},\n",
       "  7: {'A': 12.0, 'C': 9162.0, 'G': 19.0, 'T': 24.0},\n",
       "  8: {'A': 2.0, 'C': 9167.0, 'G': 21.0, 'T': 8.0},\n",
       "  9: {'A': 1.0, 'C': 8942.0, 'G': 5.0, 'T': 886.0},\n",
       "  10: {'A': 4690.0, 'C': 4227.0, 'G': 70.0, 'T': 762.0},\n",
       "  11: {'A': 582.0, 'C': 8404.0, 'G': 359.0, 'T': 1282.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11443140',\n",
       "  'uniprot_ids': 'O14901',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0621.1',\n",
       "  'Matrix_Name': 'mix-a',\n",
       "  1: {'A': 313.0, 'C': 230.0, 'G': 230.0, 'T': 230.0},\n",
       "  2: {'A': 346.0, 'C': 151.0, 'G': 262.0, 'T': 245.0},\n",
       "  3: {'A': 131.0, 'C': 147.0, 'G': 47.0, 'T': 680.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  5: {'A': 1001.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 1001.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1001.0},\n",
       "  9: {'A': 939.0, 'C': 22.0, 'G': 22.0, 'T': 22.0},\n",
       "  10: {'A': 229.0, 'C': 101.0, 'G': 489.0, 'T': 185.0},\n",
       "  11: {'A': 206.0, 'C': 206.0, 'G': 291.0, 'T': 302.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'P21711',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1505.1',\n",
       "  'Matrix_Name': 'HOXC8',\n",
       "  1: {'A': 1767.0, 'C': 1553.0, 'G': 3725.0, 'T': 2231.0},\n",
       "  2: {'A': 1.0, 'C': 4556.0, 'G': 1.0, 'T': 9274.0},\n",
       "  3: {'A': 9274.0, 'C': 2728.0, 'G': 20.0, 'T': 984.0},\n",
       "  4: {'A': 9274.0, 'C': 21.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 212.0, 'T': 9274.0},\n",
       "  6: {'A': 962.0, 'C': 100.0, 'G': 2657.0, 'T': 9274.0},\n",
       "  7: {'A': 9274.0, 'C': 1.0, 'G': 4326.0, 'T': 1.0},\n",
       "  8: {'A': 2266.0, 'C': 3852.0, 'G': 1470.0, 'T': 1688.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31273',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1150.1',\n",
       "  'Matrix_Name': 'RORB',\n",
       "  1: {'A': 786.0, 'C': 15.0, 'G': 30.0, 'T': 173.0},\n",
       "  2: {'A': 527.0, 'C': 27.0, 'G': 25.0, 'T': 426.0},\n",
       "  3: {'A': 234.0, 'C': 163.0, 'G': 202.0, 'T': 404.0},\n",
       "  4: {'A': 54.0, 'C': 166.0, 'G': 236.0, 'T': 548.0},\n",
       "  5: {'A': 603.0, 'C': 15.0, 'G': 359.0, 'T': 27.0},\n",
       "  6: {'A': 17.0, 'C': 26.0, 'G': 908.0, 'T': 53.0},\n",
       "  7: {'A': 33.0, 'C': 14.0, 'G': 929.0, 'T': 28.0},\n",
       "  8: {'A': 29.0, 'C': 147.0, 'G': 51.0, 'T': 777.0},\n",
       "  9: {'A': 48.0, 'C': 782.0, 'G': 18.0, 'T': 156.0},\n",
       "  10: {'A': 897.0, 'C': 16.0, 'G': 64.0, 'T': 27.0},\n",
       "  11: {'A': 206.0, 'C': 327.0, 'G': 146.0, 'T': 325.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '25500868',\n",
       "  'uniprot_ids': 'Q92753',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA1536.1',\n",
       "  'Matrix_Name': 'NR2C2(var.2)',\n",
       "  1: {'A': 6237.0, 'C': 3112.0, 'G': 6421.0, 'T': 2862.0},\n",
       "  2: {'A': 10233.0, 'C': 720.0, 'G': 8396.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 237.0, 'G': 18629.0, 'T': 1342.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 18629.0, 'T': 4791.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1589.0, 'T': 18629.0},\n",
       "  6: {'A': 1.0, 'C': 18629.0, 'G': 1644.0, 'T': 4407.0},\n",
       "  7: {'A': 18629.0, 'C': 1.0, 'G': 6640.0, 'T': 1.0},\n",
       "  8: {'A': 5066.0, 'C': 4758.0, 'G': 3685.0, 'T': 5123.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '11478808',\n",
       "  'uniprot_ids': 'P49116',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0835.2',\n",
       "  'Matrix_Name': 'BATF3',\n",
       "  1: {'A': 3610.0, 'C': 1246.0, 'G': 2609.0, 'T': 4504.0},\n",
       "  2: {'A': 6138.0, 'C': 2107.0, 'G': 1832.0, 'T': 1892.0},\n",
       "  3: {'A': 133.0, 'C': 60.0, 'G': 38.0, 'T': 11738.0},\n",
       "  4: {'A': 246.0, 'C': 63.0, 'G': 10421.0, 'T': 1239.0},\n",
       "  5: {'A': 11671.0, 'C': 93.0, 'G': 88.0, 'T': 117.0},\n",
       "  6: {'A': 409.0, 'C': 10590.0, 'G': 561.0, 'T': 409.0},\n",
       "  7: {'A': 268.0, 'C': 80.0, 'G': 118.0, 'T': 11503.0},\n",
       "  8: {'A': 1249.0, 'C': 10348.0, 'G': 89.0, 'T': 283.0},\n",
       "  9: {'A': 11530.0, 'C': 85.0, 'G': 118.0, 'T': 236.0},\n",
       "  10: {'A': 2335.0, 'C': 1665.0, 'G': 2250.0, 'T': 5719.0},\n",
       "  11: {'A': 4522.0, 'C': 2734.0, 'G': 1256.0, 'T': 3457.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'B-ATF-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '29588546',\n",
       "  'uniprot_ids': 'Q9NR55',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1602.1',\n",
       "  'Matrix_Name': 'ZSCAN29',\n",
       "  1: {'A': 69.0, 'C': 498.0, 'G': 125.0, 'T': 311.0},\n",
       "  2: {'A': 21.0, 'C': 1.0, 'G': 978.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 17.0, 'G': 5.0, 'T': 978.0},\n",
       "  4: {'A': 5.0, 'C': 994.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 49.0, 'G': 1.0, 'T': 950.0},\n",
       "  6: {'A': 990.0, 'C': 5.0, 'G': 1.0, 'T': 5.0},\n",
       "  7: {'A': 5.0, 'C': 839.0, 'G': 1.0, 'T': 156.0},\n",
       "  8: {'A': 648.0, 'C': 1.0, 'G': 351.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 962.0, 'G': 25.0, 'T': 13.0},\n",
       "  10: {'A': 307.0, 'C': 208.0, 'G': 311.0, 'T': 176.0},\n",
       "  11: {'A': 128.0, 'C': 41.0, 'G': 787.0, 'T': 45.0},\n",
       "  12: {'A': 200.0, 'C': 295.0, 'G': 327.0, 'T': 180.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q8IWY8',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1594.1',\n",
       "  'Matrix_Name': 'ZNF382',\n",
       "  1: {'A': 3567.0, 'C': 120.0, 'G': 343.0, 'T': 226.0},\n",
       "  2: {'A': 621.0, 'C': 121.0, 'G': 2798.0, 'T': 716.0},\n",
       "  3: {'A': 306.0, 'C': 333.0, 'G': 3519.0, 'T': 98.0},\n",
       "  4: {'A': 1785.0, 'C': 48.0, 'G': 2132.0, 'T': 291.0},\n",
       "  5: {'A': 277.0, 'C': 42.0, 'G': 3823.0, 'T': 114.0},\n",
       "  6: {'A': 3588.0, 'C': 247.0, 'G': 308.0, 'T': 113.0},\n",
       "  7: {'A': 156.0, 'C': 2050.0, 'G': 306.0, 'T': 1744.0},\n",
       "  8: {'A': 4178.0, 'C': 17.0, 'G': 45.0, 'T': 16.0},\n",
       "  9: {'A': 421.0, 'C': 233.0, 'G': 164.0, 'T': 3438.0},\n",
       "  10: {'A': 94.0, 'C': 2763.0, 'G': 37.0, 'T': 1362.0},\n",
       "  11: {'A': 4193.0, 'C': 13.0, 'G': 26.0, 'T': 24.0},\n",
       "  12: {'A': 59.0, 'C': 3883.0, 'G': 171.0, 'T': 143.0},\n",
       "  13: {'A': 601.0, 'C': 1694.0, 'G': 32.0, 'T': 1929.0},\n",
       "  14: {'A': 4205.0, 'C': 17.0, 'G': 25.0, 'T': 9.0},\n",
       "  15: {'A': 15.0, 'C': 4074.0, 'G': 40.0, 'T': 127.0},\n",
       "  16: {'A': 1958.0, 'C': 1447.0, 'G': 92.0, 'T': 759.0},\n",
       "  17: {'A': 559.0, 'C': 44.0, 'G': 3579.0, 'T': 74.0},\n",
       "  18: {'A': 3944.0, 'C': 63.0, 'G': 168.0, 'T': 81.0},\n",
       "  19: {'A': 123.0, 'C': 1606.0, 'G': 96.0, 'T': 2431.0},\n",
       "  20: {'A': 916.0, 'C': 2983.0, 'G': 144.0, 'T': 213.0},\n",
       "  21: {'A': 185.0, 'C': 3608.0, 'G': 62.0, 'T': 401.0},\n",
       "  22: {'A': 361.0, 'C': 2616.0, 'G': 89.0, 'T': 1190.0},\n",
       "  23: {'A': 3405.0, 'C': 154.0, 'G': 250.0, 'T': 447.0},\n",
       "  24: {'A': 172.0, 'C': 3416.0, 'G': 144.0, 'T': 524.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 24,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '27852650',\n",
       "  'uniprot_ids': 'Q96SR6',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0630.1',\n",
       "  'Matrix_Name': 'SHOX',\n",
       "  1: {'A': 180.0, 'C': 302.0, 'G': 163.0, 'T': 358.0},\n",
       "  2: {'A': 75.0, 'C': 31.0, 'G': 10.0, 'T': 887.0},\n",
       "  3: {'A': 853.0, 'C': 49.0, 'G': 56.0, 'T': 46.0},\n",
       "  4: {'A': 992.0, 'C': 3.0, 'G': 6.0, 'T': 3.0},\n",
       "  5: {'A': 16.0, 'C': 16.0, 'G': 28.0, 'T': 944.0},\n",
       "  6: {'A': 99.0, 'C': 51.0, 'G': 85.0, 'T': 769.0},\n",
       "  7: {'A': 417.0, 'C': 49.0, 'G': 425.0, 'T': 112.0},\n",
       "  8: {'A': 302.0, 'C': 188.0, 'G': 380.0, 'T': 134.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'O15266',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA0461.2',\n",
       "  'Matrix_Name': 'Atoh1',\n",
       "  1: {'A': 656.0, 'C': 109.0, 'G': 373.0, 'T': 60.0},\n",
       "  2: {'A': 524.0, 'C': 335.0, 'G': 150.0, 'T': 22.0},\n",
       "  3: {'A': 43.0, 'C': 1028.0, 'G': 9.0, 'T': 1.0},\n",
       "  4: {'A': 1028.0, 'C': 21.0, 'G': 35.0, 'T': 1.0},\n",
       "  5: {'A': 18.0, 'C': 11.0, 'G': 137.0, 'T': 1028.0},\n",
       "  6: {'A': 1028.0, 'C': 75.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 23.0, 'C': 5.0, 'G': 6.0, 'T': 1028.0},\n",
       "  8: {'A': 3.0, 'C': 2.0, 'G': 1028.0, 'T': 32.0},\n",
       "  9: {'A': 35.0, 'C': 235.0, 'G': 377.0, 'T': 385.0},\n",
       "  10: {'A': 129.0, 'C': 495.0, 'G': 127.0, 'T': 534.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21300888',\n",
       "  'uniprot_ids': 'P48985',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0676.1',\n",
       "  'Matrix_Name': 'Nr2e1',\n",
       "  1: {'A': 1119.0, 'C': 245.0, 'G': 192.0, 'T': 402.0},\n",
       "  2: {'A': 1119.0, 'C': 90.0, 'G': 449.0, 'T': 370.0},\n",
       "  3: {'A': 1119.0, 'C': 2.0, 'G': 102.0, 'T': 20.0},\n",
       "  4: {'A': 1119.0, 'C': 1.0, 'G': 75.0, 'T': 6.0},\n",
       "  5: {'A': 41.0, 'C': 1.0, 'G': 1119.0, 'T': 9.0},\n",
       "  6: {'A': 1.0, 'C': 59.0, 'G': 1.0, 'T': 1119.0},\n",
       "  7: {'A': 12.0, 'C': 1119.0, 'G': 13.0, 'T': 113.0},\n",
       "  8: {'A': 1119.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 1119.0, 'C': 132.0, 'G': 118.0, 'T': 248.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 9,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '10706625',\n",
       "  'uniprot_ids': 'Q9Y466',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0808.1',\n",
       "  'Matrix_Name': 'TEAD3',\n",
       "  1: {'A': 12531.0, 'C': 67.0, 'G': 8914.0, 'T': 1110.0},\n",
       "  2: {'A': 2525.0, 'C': 18921.0, 'G': 442.0, 'T': 1.0},\n",
       "  3: {'A': 21445.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 21445.0},\n",
       "  5: {'A': 4567.0, 'C': 24.0, 'G': 970.0, 'T': 16879.0},\n",
       "  6: {'A': 1.0, 'C': 21445.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 21445.0, 'G': 1.0, 'T': 5.0},\n",
       "  8: {'A': 21445.0, 'C': 1040.0, 'G': 8655.0, 'T': 13664.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'TEF-1-related',\n",
       "  'tf_class': 'TEA',\n",
       "  'pubmed_ids': '9571041',\n",
       "  'uniprot_ids': 'Q99594',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1538.1',\n",
       "  'Matrix_Name': 'NR2F1(var.3)',\n",
       "  1: {'A': 1652.0, 'C': 281.0, 'G': 1272.0, 'T': 313.0},\n",
       "  2: {'A': 2467.0, 'C': 76.0, 'G': 1050.0, 'T': 80.0},\n",
       "  3: {'A': 45.0, 'C': 122.0, 'G': 3516.0, 'T': 357.0},\n",
       "  4: {'A': 76.0, 'C': 46.0, 'G': 3516.0, 'T': 68.0},\n",
       "  5: {'A': 23.0, 'C': 88.0, 'G': 233.0, 'T': 3516.0},\n",
       "  6: {'A': 21.0, 'C': 3516.0, 'G': 59.0, 'T': 172.0},\n",
       "  7: {'A': 2474.0, 'C': 59.0, 'G': 1042.0, 'T': 5.0},\n",
       "  8: {'A': 814.0, 'C': 856.0, 'G': 859.0, 'T': 991.0},\n",
       "  9: {'A': 1.0, 'C': 90.0, 'G': 53.0, 'T': 3516.0},\n",
       "  10: {'A': 82.0, 'C': 16.0, 'G': 3516.0, 'T': 13.0},\n",
       "  11: {'A': 3516.0, 'C': 96.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 3516.0, 'G': 2.0, 'T': 46.0},\n",
       "  13: {'A': 132.0, 'C': 3516.0, 'G': 76.0, 'T': 51.0},\n",
       "  14: {'A': 58.0, 'C': 950.0, 'G': 33.0, 'T': 2567.0},\n",
       "  15: {'A': 309.0, 'C': 1178.0, 'G': 206.0, 'T': 1826.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '14559106',\n",
       "  'uniprot_ids': 'P10589',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1623.1',\n",
       "  'Matrix_Name': 'Stat2',\n",
       "  1: {'A': 1606.0, 'C': 539.0, 'G': 1126.0, 'T': 692.0},\n",
       "  2: {'A': 970.0, 'C': 307.0, 'G': 2288.0, 'T': 398.0},\n",
       "  3: {'A': 3555.0, 'C': 77.0, 'G': 169.0, 'T': 162.0},\n",
       "  4: {'A': 3737.0, 'C': 76.0, 'G': 72.0, 'T': 78.0},\n",
       "  5: {'A': 3739.0, 'C': 73.0, 'G': 88.0, 'T': 63.0},\n",
       "  6: {'A': 287.0, 'C': 2908.0, 'G': 525.0, 'T': 243.0},\n",
       "  7: {'A': 3147.0, 'C': 152.0, 'G': 75.0, 'T': 589.0},\n",
       "  8: {'A': 153.0, 'C': 46.0, 'G': 3708.0, 'T': 56.0},\n",
       "  9: {'A': 3701.0, 'C': 50.0, 'G': 164.0, 'T': 48.0},\n",
       "  10: {'A': 3748.0, 'C': 46.0, 'G': 99.0, 'T': 70.0},\n",
       "  11: {'A': 3515.0, 'C': 148.0, 'G': 158.0, 'T': 142.0},\n",
       "  12: {'A': 741.0, 'C': 1503.0, 'G': 1181.0, 'T': 538.0},\n",
       "  13: {'A': 1263.0, 'C': 759.0, 'G': 592.0, 'T': 1349.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '9242679',\n",
       "  'uniprot_ids': 'Q9WVL2',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0736.1',\n",
       "  'Matrix_Name': 'GLIS2',\n",
       "  1: {'A': 75.0, 'C': 180.0, 'G': 930.0, 'T': 440.0},\n",
       "  2: {'A': 814.0, 'C': 95.0, 'G': 131.0, 'T': 2.0},\n",
       "  3: {'A': 4.0, 'C': 1073.0, 'G': 4.0, 'T': 6.0},\n",
       "  4: {'A': 5.0, 'C': 1075.0, 'G': 1.0, 'T': 2.0},\n",
       "  5: {'A': 8.0, 'C': 1075.0, 'G': 1.0, 'T': 15.0},\n",
       "  6: {'A': 3.0, 'C': 1088.0, 'G': 1.0, 'T': 1.0},\n",
       "  7: {'A': 57.0, 'C': 1062.0, 'G': 1.0, 'T': 5.0},\n",
       "  8: {'A': 53.0, 'C': 1057.0, 'G': 2.0, 'T': 100.0},\n",
       "  9: {'A': 413.0, 'C': 13.0, 'G': 916.0, 'T': 51.0},\n",
       "  10: {'A': 4.0, 'C': 1081.0, 'G': 29.0, 'T': 70.0},\n",
       "  11: {'A': 460.0, 'C': 37.0, 'G': 910.0, 'T': 65.0},\n",
       "  12: {'A': 485.0, 'C': 17.0, 'G': 235.0, 'T': 229.0},\n",
       "  13: {'A': 408.0, 'C': 266.0, 'G': 113.0, 'T': 198.0},\n",
       "  14: {'A': 16.0, 'C': 188.0, 'G': 1052.0, 'T': 64.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '22391303',\n",
       "  'uniprot_ids': 'Q9BZE0',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1600.1',\n",
       "  'Matrix_Name': 'ZNF684',\n",
       "  1: {'A': 525.0, 'C': 169.0, 'G': 363.0, 'T': 207.0},\n",
       "  2: {'A': 83.0, 'C': 404.0, 'G': 315.0, 'T': 722.0},\n",
       "  3: {'A': 1239.0, 'C': 25.0, 'G': 69.0, 'T': 20.0},\n",
       "  4: {'A': 10.0, 'C': 1254.0, 'G': 10.0, 'T': 20.0},\n",
       "  5: {'A': 1185.0, 'C': 12.0, 'G': 111.0, 'T': 132.0},\n",
       "  6: {'A': 61.0, 'C': 217.0, 'G': 1136.0, 'T': 75.0},\n",
       "  7: {'A': 10.0, 'C': 15.0, 'G': 60.0, 'T': 1260.0},\n",
       "  8: {'A': 14.0, 'C': 1237.0, 'G': 26.0, 'T': 11.0},\n",
       "  9: {'A': 41.0, 'C': 1222.0, 'G': 3.0, 'T': 68.0},\n",
       "  10: {'A': 1013.0, 'C': 10.0, 'G': 58.0, 'T': 221.0},\n",
       "  11: {'A': 4.0, 'C': 1227.0, 'G': 64.0, 'T': 62.0},\n",
       "  12: {'A': 45.0, 'C': 1254.0, 'G': 11.0, 'T': 7.0},\n",
       "  13: {'A': 8.0, 'C': 1230.0, 'G': 8.0, 'T': 93.0},\n",
       "  14: {'A': 1.0, 'C': 1264.0, 'G': 1.0, 'T': 4.0},\n",
       "  15: {'A': 250.0, 'C': 130.0, 'G': 19.0, 'T': 1108.0},\n",
       "  16: {'A': 57.0, 'C': 60.0, 'G': 15.0, 'T': 1225.0},\n",
       "  17: {'A': 217.0, 'C': 169.0, 'G': 515.0, 'T': 620.0},\n",
       "  18: {'A': 607.0, 'C': 202.0, 'G': 327.0, 'T': 128.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '25690854',\n",
       "  'uniprot_ids': 'Q5T5D7',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0037.3',\n",
       "  'Matrix_Name': 'GATA3',\n",
       "  1: {'A': 503.0, 'C': 227.0, 'G': 2.0, 'T': 272.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 1000.0, 'T': 1.0},\n",
       "  3: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  5: {'A': 915.0, 'C': 3.0, 'G': 1.0, 'T': 84.0},\n",
       "  6: {'A': 863.0, 'C': 16.0, 'G': 77.0, 'T': 48.0},\n",
       "  7: {'A': 115.0, 'C': 302.0, 'G': 477.0, 'T': 110.0},\n",
       "  8: {'A': 424.0, 'C': 189.0, 'G': 322.0, 'T': 69.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '8321207',\n",
       "  'uniprot_ids': 'P23771',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0772.1',\n",
       "  'Matrix_Name': 'IRF7',\n",
       "  1: {'A': 374.0, 'C': 324.0, 'G': 72.0, 'T': 192.0},\n",
       "  2: {'A': 98.0, 'C': 958.0, 'G': 51.0, 'T': 47.0},\n",
       "  3: {'A': 37.0, 'C': 68.0, 'G': 958.0, 'T': 91.0},\n",
       "  4: {'A': 958.0, 'C': 4.0, 'G': 2.0, 'T': 1.0},\n",
       "  5: {'A': 958.0, 'C': 1.0, 'G': 1.0, 'T': 2.0},\n",
       "  6: {'A': 958.0, 'C': 3.0, 'G': 2.0, 'T': 2.0},\n",
       "  7: {'A': 280.0, 'C': 157.0, 'G': 506.0, 'T': 18.0},\n",
       "  8: {'A': 9.0, 'C': 583.0, 'G': 3.0, 'T': 376.0},\n",
       "  9: {'A': 31.0, 'C': 4.0, 'G': 958.0, 'T': 8.0},\n",
       "  10: {'A': 958.0, 'C': 17.0, 'G': 5.0, 'T': 6.0},\n",
       "  11: {'A': 958.0, 'C': 4.0, 'G': 4.0, 'T': 2.0},\n",
       "  12: {'A': 958.0, 'C': 11.0, 'G': 8.0, 'T': 42.0},\n",
       "  13: {'A': 1185.0, 'C': 958.0, 'G': 1236.0, 'T': 302.0},\n",
       "  14: {'A': 161.0, 'C': 154.0, 'G': 32.0, 'T': 805.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '10938111',\n",
       "  'uniprot_ids': 'Q92985',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0766.2',\n",
       "  'Matrix_Name': 'GATA5',\n",
       "  1: {'A': 2499.0, 'C': 6616.0, 'G': 3536.0, 'T': 2723.0},\n",
       "  2: {'A': 8022.0, 'C': 2329.0, 'G': 53.0, 'T': 7349.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 15370.0, 'T': 1.0},\n",
       "  4: {'A': 15370.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 15370.0},\n",
       "  6: {'A': 15370.0, 'C': 1.0, 'G': 1.0, 'T': 332.0},\n",
       "  7: {'A': 15370.0, 'C': 1.0, 'G': 201.0, 'T': 205.0},\n",
       "  8: {'A': 213.0, 'C': 7595.0, 'G': 7775.0, 'T': 587.0},\n",
       "  9: {'A': 7191.0, 'C': 3216.0, 'G': 8180.0, 'T': 669.0},\n",
       "  10: {'A': 5020.0, 'C': 4580.0, 'G': 3145.0, 'T': 2627.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'GATA-type',\n",
       "  'tf_class': 'Other',\n",
       "  'pubmed_ids': '14612389',\n",
       "  'uniprot_ids': 'Q9BWX5',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1639.1',\n",
       "  'Matrix_Name': 'MEIS1(var.2)',\n",
       "  1: {'A': 1825.0, 'C': 1426.0, 'G': 1903.0, 'T': 1586.0},\n",
       "  2: {'A': 1865.0, 'C': 1980.0, 'G': 763.0, 'T': 2132.0},\n",
       "  3: {'A': 971.0, 'C': 4572.0, 'G': 184.0, 'T': 1013.0},\n",
       "  4: {'A': 6384.0, 'C': 62.0, 'G': 219.0, 'T': 75.0},\n",
       "  5: {'A': 81.0, 'C': 70.0, 'G': 84.0, 'T': 6505.0},\n",
       "  6: {'A': 2296.0, 'C': 3581.0, 'G': 253.0, 'T': 610.0},\n",
       "  7: {'A': 6064.0, 'C': 337.0, 'G': 34.0, 'T': 305.0},\n",
       "  8: {'A': 6113.0, 'C': 215.0, 'G': 193.0, 'T': 219.0},\n",
       "  9: {'A': 128.0, 'C': 88.0, 'G': 64.0, 'T': 6460.0},\n",
       "  10: {'A': 157.0, 'C': 6368.0, 'G': 62.0, 'T': 153.0},\n",
       "  11: {'A': 5434.0, 'C': 280.0, 'G': 153.0, 'T': 873.0},\n",
       "  12: {'A': 1736.0, 'C': 1505.0, 'G': 935.0, 'T': 2564.0},\n",
       "  13: {'A': 1873.0, 'C': 1543.0, 'G': 1262.0, 'T': 2062.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'TALE-type',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '23602564',\n",
       "  'uniprot_ids': 'O00470',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0142.1',\n",
       "  'Matrix_Name': 'Pou5f1::Sox2',\n",
       "  1: {'A': 64.0, 'C': 847.0, 'G': 67.0, 'T': 390.0},\n",
       "  2: {'A': 580.0, 'C': 59.0, 'G': 29.0, 'T': 702.0},\n",
       "  3: {'A': 12.0, 'C': 51.0, 'G': 37.0, 'T': 1271.0},\n",
       "  4: {'A': 48.0, 'C': 13.0, 'G': 80.0, 'T': 1232.0},\n",
       "  5: {'A': 119.0, 'C': 364.0, 'G': 826.0, 'T': 64.0},\n",
       "  6: {'A': 416.0, 'C': 19.0, 'G': 30.0, 'T': 908.0},\n",
       "  7: {'A': 207.0, 'C': 366.0, 'G': 187.0, 'T': 613.0},\n",
       "  8: {'A': 1236.0, 'C': 31.0, 'G': 25.0, 'T': 81.0},\n",
       "  9: {'A': 18.0, 'C': 6.0, 'G': 16.0, 'T': 1333.0},\n",
       "  10: {'A': 11.0, 'C': 17.0, 'G': 1240.0, 'T': 105.0},\n",
       "  11: {'A': 15.0, 'C': 1030.0, 'G': 130.0, 'T': 197.0},\n",
       "  12: {'A': 1051.0, 'C': 17.0, 'G': 31.0, 'T': 270.0},\n",
       "  13: {'A': 888.0, 'C': 69.0, 'G': 316.0, 'T': 93.0},\n",
       "  14: {'A': 1201.0, 'C': 34.0, 'G': 76.0, 'T': 54.0},\n",
       "  15: {'A': 199.0, 'C': 120.0, 'G': 208.0, 'T': 838.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'POU',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18555785',\n",
       "  'uniprot_ids': 'P20263;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0814.2',\n",
       "  'Matrix_Name': 'TFAP2C(var.2)',\n",
       "  1: {'A': 11315.0, 'C': 15859.0, 'G': 17039.0, 'T': 16610.0},\n",
       "  2: {'A': 13629.0, 'C': 12566.0, 'G': 20130.0, 'T': 14498.0},\n",
       "  3: {'A': 3783.0, 'C': 14687.0, 'G': 37730.0, 'T': 4623.0},\n",
       "  4: {'A': 737.0, 'C': 57109.0, 'G': 2274.0, 'T': 703.0},\n",
       "  5: {'A': 483.0, 'C': 57618.0, 'G': 2061.0, 'T': 661.0},\n",
       "  6: {'A': 1368.0, 'C': 5540.0, 'G': 4911.0, 'T': 49004.0},\n",
       "  7: {'A': 1324.0, 'C': 3037.0, 'G': 55223.0, 'T': 1239.0},\n",
       "  8: {'A': 51059.0, 'C': 4734.0, 'G': 3945.0, 'T': 1085.0},\n",
       "  9: {'A': 875.0, 'C': 1487.0, 'G': 57996.0, 'T': 465.0},\n",
       "  10: {'A': 715.0, 'C': 1576.0, 'G': 57917.0, 'T': 615.0},\n",
       "  11: {'A': 4314.0, 'C': 42759.0, 'G': 9868.0, 'T': 3882.0},\n",
       "  12: {'A': 10187.0, 'C': 12554.0, 'G': 27374.0, 'T': 10708.0},\n",
       "  13: {'A': 16183.0, 'C': 15253.0, 'G': 18681.0, 'T': 10706.0},\n",
       "  14: {'A': 12087.0, 'C': 16469.0, 'G': 17368.0, 'T': 14899.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'AP-2',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23332764',\n",
       "  'uniprot_ids': 'Q92754',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1636.1',\n",
       "  'Matrix_Name': 'CEBPG(var.2)',\n",
       "  1: {'A': 12056.0, 'C': 7639.0, 'G': 8656.0, 'T': 8969.0},\n",
       "  2: {'A': 11640.0, 'C': 4689.0, 'G': 10137.0, 'T': 10854.0},\n",
       "  3: {'A': 25957.0, 'C': 3081.0, 'G': 7548.0, 'T': 734.0},\n",
       "  4: {'A': 319.0, 'C': 250.0, 'G': 307.0, 'T': 36444.0},\n",
       "  5: {'A': 660.0, 'C': 352.0, 'G': 29948.0, 'T': 6360.0},\n",
       "  6: {'A': 31892.0, 'C': 482.0, 'G': 2566.0, 'T': 2380.0},\n",
       "  7: {'A': 2083.0, 'C': 2720.0, 'G': 1102.0, 'T': 31415.0},\n",
       "  8: {'A': 523.0, 'C': 216.0, 'G': 36190.0, 'T': 391.0},\n",
       "  9: {'A': 4375.0, 'C': 28898.0, 'G': 310.0, 'T': 3737.0},\n",
       "  10: {'A': 36287.0, 'C': 583.0, 'G': 130.0, 'T': 320.0},\n",
       "  11: {'A': 36429.0, 'C': 192.0, 'G': 305.0, 'T': 394.0},\n",
       "  12: {'A': 1082.0, 'C': 4762.0, 'G': 2292.0, 'T': 29184.0},\n",
       "  13: {'A': 12950.0, 'C': 15061.0, 'G': 4028.0, 'T': 5281.0},\n",
       "  14: {'A': 13000.0, 'C': 7053.0, 'G': 5335.0, 'T': 11932.0},\n",
       "  15: {'A': 8848.0, 'C': 8979.0, 'G': 5958.0, 'T': 13535.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '21543584',\n",
       "  'uniprot_ids': 'P53567',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1557.1',\n",
       "  'Matrix_Name': 'SMAD5',\n",
       "  1: {'A': 767.0, 'C': 2396.0, 'G': 338.0, 'T': 2530.0},\n",
       "  2: {'A': 35.0, 'C': 17.0, 'G': 4925.0, 'T': 6.0},\n",
       "  3: {'A': 316.0, 'C': 109.0, 'G': 448.0, 'T': 4925.0},\n",
       "  4: {'A': 42.0, 'C': 4925.0, 'G': 40.0, 'T': 6.0},\n",
       "  5: {'A': 2.0, 'C': 227.0, 'G': 586.0, 'T': 4925.0},\n",
       "  6: {'A': 4925.0, 'C': 544.0, 'G': 135.0, 'T': 9.0},\n",
       "  7: {'A': 22.0, 'C': 34.0, 'G': 4925.0, 'T': 7.0},\n",
       "  8: {'A': 4925.0, 'C': 483.0, 'G': 21.0, 'T': 282.0},\n",
       "  9: {'A': 23.0, 'C': 4925.0, 'G': 6.0, 'T': 25.0},\n",
       "  10: {'A': 4925.0, 'C': 785.0, 'G': 483.0, 'T': 1409.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'SMAD',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '11527422',\n",
       "  'uniprot_ids': 'Q99717',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1533.1',\n",
       "  'Matrix_Name': 'NR1I2',\n",
       "  1: {'A': 59.0, 'C': 19.0, 'G': 56.0, 'T': 34.0},\n",
       "  2: {'A': 5.0, 'C': 61.0, 'G': 10.0, 'T': 104.0},\n",
       "  3: {'A': 10.0, 'C': 1.0, 'G': 165.0, 'T': 3.0},\n",
       "  4: {'A': 165.0, 'C': 41.0, 'G': 25.0, 'T': 14.0},\n",
       "  5: {'A': 165.0, 'C': 62.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 4.0, 'C': 165.0, 'G': 2.0, 'T': 1.0},\n",
       "  7: {'A': 8.0, 'C': 51.0, 'G': 4.0, 'T': 115.0},\n",
       "  8: {'A': 10.0, 'C': 87.0, 'G': 52.0, 'T': 19.0},\n",
       "  9: {'A': 52.0, 'C': 31.0, 'G': 57.0, 'T': 28.0},\n",
       "  10: {'A': 67.0, 'C': 24.0, 'G': 41.0, 'T': 37.0},\n",
       "  11: {'A': 1.0, 'C': 47.0, 'G': 1.0, 'T': 118.0},\n",
       "  12: {'A': 22.0, 'C': 3.0, 'G': 165.0, 'T': 10.0},\n",
       "  13: {'A': 165.0, 'C': 42.0, 'G': 13.0, 'T': 6.0},\n",
       "  14: {'A': 165.0, 'C': 52.0, 'G': 3.0, 'T': 1.0},\n",
       "  15: {'A': 1.0, 'C': 165.0, 'G': 1.0, 'T': 4.0},\n",
       "  16: {'A': 2.0, 'C': 55.0, 'G': 6.0, 'T': 110.0},\n",
       "  17: {'A': 12.0, 'C': 84.0, 'G': 34.0, 'T': 38.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '12016543',\n",
       "  'uniprot_ids': 'O75469',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0603.1',\n",
       "  'Matrix_Name': 'Arntl',\n",
       "  1: {'A': 252.0, 'C': 230.0, 'G': 262.0, 'T': 260.0},\n",
       "  2: {'A': 247.0, 'C': 126.0, 'G': 559.0, 'T': 72.0},\n",
       "  3: {'A': 63.0, 'C': 45.0, 'G': 144.0, 'T': 753.0},\n",
       "  4: {'A': 4.0, 'C': 998.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 954.0, 'C': 17.0, 'G': 11.0, 'T': 22.0},\n",
       "  6: {'A': 3.0, 'C': 966.0, 'G': 10.0, 'T': 25.0},\n",
       "  7: {'A': 49.0, 'C': 4.0, 'G': 947.0, 'T': 4.0},\n",
       "  8: {'A': 55.0, 'C': 59.0, 'G': 52.0, 'T': 838.0},\n",
       "  9: {'A': 10.0, 'C': 5.0, 'G': 939.0, 'T': 49.0},\n",
       "  10: {'A': 256.0, 'C': 332.0, 'G': 147.0, 'T': 269.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'PAS',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '7592839',\n",
       "  'uniprot_ids': 'Q9WTL8',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1485.1',\n",
       "  'Matrix_Name': 'FERD3L',\n",
       "  1: {'A': 2159.0, 'C': 69.0, 'G': 29606.0, 'T': 226.0},\n",
       "  2: {'A': 451.0, 'C': 20050.0, 'G': 349.0, 'T': 9558.0},\n",
       "  3: {'A': 14385.0, 'C': 2444.0, 'G': 12704.0, 'T': 75.0},\n",
       "  4: {'A': 15391.0, 'C': 9364.0, 'G': 3143.0, 'T': 1711.0},\n",
       "  5: {'A': 37.0, 'C': 29606.0, 'G': 1.0, 'T': 1.0},\n",
       "  6: {'A': 29606.0, 'C': 127.0, 'G': 119.0, 'T': 145.0},\n",
       "  7: {'A': 1.0, 'C': 842.0, 'G': 29606.0, 'T': 4692.0},\n",
       "  8: {'A': 3650.0, 'C': 29606.0, 'G': 472.0, 'T': 1.0},\n",
       "  9: {'A': 82.0, 'C': 130.0, 'G': 211.0, 'T': 29606.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 29606.0, 'T': 1.0},\n",
       "  11: {'A': 1269.0, 'C': 3879.0, 'G': 3396.0, 'T': 21066.0},\n",
       "  12: {'A': 104.0, 'C': 8665.0, 'G': 5921.0, 'T': 14920.0},\n",
       "  13: {'A': 29606.0, 'C': 810.0, 'G': 9816.0, 'T': 959.0},\n",
       "  14: {'A': 1.0, 'C': 29606.0, 'G': 1.0, 'T': 49.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '17027923',\n",
       "  'uniprot_ids': 'Q96RJ6',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0032.2',\n",
       "  'Matrix_Name': 'FOXC1',\n",
       "  1: {'A': 6453.0, 'C': 773.0, 'G': 305.0, 'T': 12039.0},\n",
       "  2: {'A': 14325.0, 'C': 729.0, 'G': 4166.0, 'T': 1387.0},\n",
       "  3: {'A': 7894.0, 'C': 2009.0, 'G': 2423.0, 'T': 10597.0},\n",
       "  4: {'A': 4859.0, 'C': 337.0, 'G': 13632.0, 'T': 52.0},\n",
       "  5: {'A': 2497.0, 'C': 1653.0, 'G': 391.0, 'T': 18490.0},\n",
       "  6: {'A': 13118.0, 'C': 5373.0, 'G': 22.0, 'T': 46.0},\n",
       "  7: {'A': 18490.0, 'C': 109.0, 'G': 1.0, 'T': 58.0},\n",
       "  8: {'A': 18490.0, 'C': 57.0, 'G': 142.0, 'T': 113.0},\n",
       "  9: {'A': 123.0, 'C': 8783.0, 'G': 166.0, 'T': 9709.0},\n",
       "  10: {'A': 18490.0, 'C': 28.0, 'G': 434.0, 'T': 185.0},\n",
       "  11: {'A': 6703.0, 'C': 1559.0, 'G': 1108.0, 'T': 11788.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '17993506',\n",
       "  'uniprot_ids': 'Q12948',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0071.1',\n",
       "  'Matrix_Name': 'RORA',\n",
       "  1: {'A': 16.0, 'C': 2.0, 'G': 3.0, 'T': 8.0},\n",
       "  2: {'A': 10.0, 'C': 2.0, 'G': 1.0, 'T': 16.0},\n",
       "  3: {'A': 7.0, 'C': 13.0, 'G': 5.0, 'T': 4.0},\n",
       "  4: {'A': 12.0, 'C': 3.0, 'G': 6.0, 'T': 8.0},\n",
       "  5: {'A': 22.0, 'C': 1.0, 'G': 5.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 26.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 26.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 26.0},\n",
       "  9: {'A': 1.0, 'C': 26.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 26.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '7926749',\n",
       "  'uniprot_ids': 'P35398',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1103.2',\n",
       "  'Matrix_Name': 'FOXK2',\n",
       "  1: {'A': 6549.0, 'C': 2873.0, 'G': 3472.0, 'T': 3789.0},\n",
       "  2: {'A': 4862.0, 'C': 2199.0, 'G': 3424.0, 'T': 6198.0},\n",
       "  3: {'A': 2698.0, 'C': 552.0, 'G': 12738.0, 'T': 695.0},\n",
       "  4: {'A': 361.0, 'C': 211.0, 'G': 328.0, 'T': 15783.0},\n",
       "  5: {'A': 15914.0, 'C': 278.0, 'G': 149.0, 'T': 342.0},\n",
       "  6: {'A': 14388.0, 'C': 1292.0, 'G': 437.0, 'T': 566.0},\n",
       "  7: {'A': 15594.0, 'C': 331.0, 'G': 392.0, 'T': 366.0},\n",
       "  8: {'A': 500.0, 'C': 14283.0, 'G': 427.0, 'T': 1473.0},\n",
       "  9: {'A': 15710.0, 'C': 296.0, 'G': 171.0, 'T': 506.0},\n",
       "  10: {'A': 4837.0, 'C': 3878.0, 'G': 3951.0, 'T': 4017.0},\n",
       "  11: {'A': 4856.0, 'C': 3789.0, 'G': 5013.0, 'T': 3025.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '26578569',\n",
       "  'uniprot_ids': 'Q01167',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0622.1',\n",
       "  'Matrix_Name': 'Mlxip',\n",
       "  1: {'A': 54.0, 'C': 264.0, 'G': 422.0, 'T': 264.0},\n",
       "  2: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 1000.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 1000.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1000.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 1000.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 1000.0, 'T': 1.0},\n",
       "  8: {'A': 189.0, 'C': 64.0, 'G': 251.0, 'T': 500.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '12446771',\n",
       "  'uniprot_ids': 'Q2VPU4',\n",
       "  'data_type': 'PBM'},\n",
       " {'Matrix_ID': 'MA1478.1',\n",
       "  'Matrix_Name': 'DMRTA2',\n",
       "  1: {'A': 766.0, 'C': 321.0, 'G': 277.0, 'T': 349.0},\n",
       "  2: {'A': 840.0, 'C': 184.0, 'G': 293.0, 'T': 396.0},\n",
       "  3: {'A': 823.0, 'C': 434.0, 'G': 381.0, 'T': 1711.0},\n",
       "  4: {'A': 325.0, 'C': 774.0, 'G': 558.0, 'T': 1711.0},\n",
       "  5: {'A': 1.0, 'C': 2.0, 'G': 1711.0, 'T': 7.0},\n",
       "  6: {'A': 728.0, 'C': 528.0, 'G': 11.0, 'T': 984.0},\n",
       "  7: {'A': 579.0, 'C': 148.0, 'G': 30.0, 'T': 1711.0},\n",
       "  8: {'A': 1711.0, 'C': 10.0, 'G': 72.0, 'T': 14.0},\n",
       "  9: {'A': 1.0, 'C': 1711.0, 'G': 1.0, 'T': 4.0},\n",
       "  10: {'A': 1711.0, 'C': 24.0, 'G': 79.0, 'T': 262.0},\n",
       "  11: {'A': 932.0, 'C': 379.0, 'G': 663.0, 'T': 1711.0},\n",
       "  12: {'A': 358.0, 'C': 400.0, 'G': 433.0, 'T': 1711.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'DMRT',\n",
       "  'tf_class': 'DM-type',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q96SC8',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0667.1',\n",
       "  'Matrix_Name': 'MYF6',\n",
       "  1: {'A': 445.0, 'C': 13.0, 'G': 71.0, 'T': 8.0},\n",
       "  2: {'A': 445.0, 'C': 19.0, 'G': 25.0, 'T': 2.0},\n",
       "  3: {'A': 1.0, 'C': 445.0, 'G': 1.0, 'T': 1.0},\n",
       "  4: {'A': 445.0, 'C': 4.0, 'G': 8.0, 'T': 1.0},\n",
       "  5: {'A': 160.0, 'C': 31.0, 'G': 257.0, 'T': 13.0},\n",
       "  6: {'A': 13.0, 'C': 354.0, 'G': 36.0, 'T': 57.0},\n",
       "  7: {'A': 12.0, 'C': 1.0, 'G': 1.0, 'T': 445.0},\n",
       "  8: {'A': 1.0, 'C': 3.0, 'G': 445.0, 'T': 1.0},\n",
       "  9: {'A': 3.0, 'C': 23.0, 'G': 1.0, 'T': 445.0},\n",
       "  10: {'A': 1.0, 'C': 160.0, 'G': 8.0, 'T': 445.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'MyoD',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'P23409',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0885.1',\n",
       "  'Matrix_Name': 'Dlx2',\n",
       "  1: {'A': 3171.0, 'C': 3566.0, 'G': 4740.0, 'T': 2276.0},\n",
       "  2: {'A': 544.0, 'C': 7410.0, 'G': 136.0, 'T': 6341.0},\n",
       "  3: {'A': 13750.0, 'C': 685.0, 'G': 489.0, 'T': 525.0},\n",
       "  4: {'A': 13750.0, 'C': 290.0, 'G': 16.0, 'T': 180.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 13750.0},\n",
       "  6: {'A': 261.0, 'C': 394.0, 'G': 566.0, 'T': 13750.0},\n",
       "  7: {'A': 13750.0, 'C': 47.0, 'G': 56.0, 'T': 580.0},\n",
       "  8: {'A': 3711.0, 'C': 3278.0, 'G': 3066.0, 'T': 3698.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P40764',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0506.1',\n",
       "  'Matrix_Name': 'NRF1',\n",
       "  1: {'A': 603.0, 'C': 376.0, 'G': 3648.0, 'T': 1.0},\n",
       "  2: {'A': 624.0, 'C': 4002.0, 'G': 1.0, 'T': 1.0},\n",
       "  3: {'A': 197.0, 'C': 190.0, 'G': 4240.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 4017.0, 'G': 469.0, 'T': 141.0},\n",
       "  5: {'A': 1515.0, 'C': 1863.0, 'G': 936.0, 'T': 314.0},\n",
       "  6: {'A': 1.0, 'C': 404.0, 'G': 529.0, 'T': 3694.0},\n",
       "  7: {'A': 1.0, 'C': 1.0, 'G': 4625.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 4339.0, 'G': 1.0, 'T': 287.0},\n",
       "  9: {'A': 1.0, 'C': 350.0, 'G': 4276.0, 'T': 1.0},\n",
       "  10: {'A': 75.0, 'C': 4551.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 2246.0, 'C': 805.0, 'G': 1576.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '12533512',\n",
       "  'uniprot_ids': 'Q16656',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0028.2',\n",
       "  'Matrix_Name': 'ELK1',\n",
       "  1: {'A': 45808.0, 'C': 2751.0, 'G': 12960.0, 'T': 5719.0},\n",
       "  2: {'A': 1218.0, 'C': 45808.0, 'G': 1479.0, 'T': 237.0},\n",
       "  3: {'A': 360.0, 'C': 45808.0, 'G': 60.0, 'T': 22.0},\n",
       "  4: {'A': 1.0, 'C': 22.0, 'G': 45808.0, 'T': 27.0},\n",
       "  5: {'A': 5.0, 'C': 42.0, 'G': 45808.0, 'T': 714.0},\n",
       "  6: {'A': 45808.0, 'C': 234.0, 'G': 63.0, 'T': 26.0},\n",
       "  7: {'A': 45808.0, 'C': 334.0, 'G': 61.0, 'T': 2166.0},\n",
       "  8: {'A': 3017.0, 'C': 2362.0, 'G': 45808.0, 'T': 295.0},\n",
       "  9: {'A': 917.0, 'C': 6310.0, 'G': 1124.0, 'T': 45808.0},\n",
       "  10: {'A': 13275.0, 'C': 6674.0, 'G': 17274.0, 'T': 8588.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '1425594',\n",
       "  'uniprot_ids': 'P19419',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1127.1',\n",
       "  'Matrix_Name': 'FOSB::JUN',\n",
       "  1: {'A': 194.0, 'C': 71.0, 'G': 576.0, 'T': 163.0},\n",
       "  2: {'A': 650.0, 'C': 88.0, 'G': 217.0, 'T': 49.0},\n",
       "  3: {'A': 13.0, 'C': 11.0, 'G': 13.0, 'T': 966.0},\n",
       "  4: {'A': 18.0, 'C': 11.0, 'G': 873.0, 'T': 101.0},\n",
       "  5: {'A': 955.0, 'C': 9.0, 'G': 18.0, 'T': 21.0},\n",
       "  6: {'A': 49.0, 'C': 887.0, 'G': 26.0, 'T': 43.0},\n",
       "  7: {'A': 76.0, 'C': 15.0, 'G': 898.0, 'T': 15.0},\n",
       "  8: {'A': 26.0, 'C': 58.0, 'G': 15.0, 'T': 905.0},\n",
       "  9: {'A': 248.0, 'C': 698.0, 'G': 21.0, 'T': 37.0},\n",
       "  10: {'A': 950.0, 'C': 17.0, 'G': 15.0, 'T': 22.0},\n",
       "  11: {'A': 93.0, 'C': 262.0, 'G': 54.0, 'T': 594.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Fos-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '11988758',\n",
       "  'uniprot_ids': 'P53539;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0684.2',\n",
       "  'Matrix_Name': 'RUNX3',\n",
       "  1: {'A': 7769.0, 'C': 6084.0, 'G': 6686.0, 'T': 6842.0},\n",
       "  2: {'A': 9559.0, 'C': 5008.0, 'G': 3548.0, 'T': 9266.0},\n",
       "  3: {'A': 19137.0, 'C': 2663.0, 'G': 3501.0, 'T': 2080.0},\n",
       "  4: {'A': 26244.0, 'C': 316.0, 'G': 432.0, 'T': 389.0},\n",
       "  5: {'A': 428.0, 'C': 26294.0, 'G': 199.0, 'T': 460.0},\n",
       "  6: {'A': 496.0, 'C': 26209.0, 'G': 386.0, 'T': 290.0},\n",
       "  7: {'A': 6358.0, 'C': 420.0, 'G': 1912.0, 'T': 18691.0},\n",
       "  8: {'A': 612.0, 'C': 26016.0, 'G': 391.0, 'T': 362.0},\n",
       "  9: {'A': 26122.0, 'C': 415.0, 'G': 339.0, 'T': 505.0},\n",
       "  10: {'A': 16138.0, 'C': 3078.0, 'G': 4697.0, 'T': 3468.0},\n",
       "  11: {'A': 8779.0, 'C': 5995.0, 'G': 5643.0, 'T': 6964.0},\n",
       "  12: {'A': 6880.0, 'C': 6901.0, 'G': 6373.0, 'T': 7227.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Runt-related',\n",
       "  'tf_class': 'Runt',\n",
       "  'pubmed_ids': '20378718',\n",
       "  'uniprot_ids': 'Q13761',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0893.2',\n",
       "  'Matrix_Name': 'GSX2',\n",
       "  1: {'A': 817.0, 'C': 2278.0, 'G': 2722.0, 'T': 1.0},\n",
       "  2: {'A': 504.0, 'C': 2612.0, 'G': 401.0, 'T': 4999.0},\n",
       "  3: {'A': 2776.0, 'C': 2224.0, 'G': 96.0, 'T': 30.0},\n",
       "  4: {'A': 4999.0, 'C': 1.0, 'G': 709.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 122.0, 'G': 1.0, 'T': 4999.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 78.0, 'T': 4999.0},\n",
       "  7: {'A': 4999.0, 'C': 1.0, 'G': 1006.0, 'T': 21.0},\n",
       "  8: {'A': 2159.0, 'C': 660.0, 'G': 2841.0, 'T': 919.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9BZM3',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0080.5',\n",
       "  'Matrix_Name': 'SPI1',\n",
       "  1: {'A': 42202.0, 'C': 22588.0, 'G': 38406.0, 'T': 30011.0},\n",
       "  2: {'A': 48241.0, 'C': 21263.0, 'G': 34278.0, 'T': 29425.0},\n",
       "  3: {'A': 54155.0, 'C': 20184.0, 'G': 37342.0, 'T': 21526.0},\n",
       "  4: {'A': 78832.0, 'C': 11425.0, 'G': 25894.0, 'T': 17056.0},\n",
       "  5: {'A': 81905.0, 'C': 12270.0, 'G': 25581.0, 'T': 13451.0},\n",
       "  6: {'A': 99740.0, 'C': 2915.0, 'G': 13480.0, 'T': 17072.0},\n",
       "  7: {'A': 15302.0, 'C': 10959.0, 'G': 100826.0, 'T': 6120.0},\n",
       "  8: {'A': 113088.0, 'C': 3426.0, 'G': 12545.0, 'T': 4148.0},\n",
       "  9: {'A': 5231.0, 'C': 569.0, 'G': 125704.0, 'T': 1703.0},\n",
       "  10: {'A': 3324.0, 'C': 955.0, 'G': 127640.0, 'T': 1288.0},\n",
       "  11: {'A': 129178.0, 'C': 1340.0, 'G': 1408.0, 'T': 1281.0},\n",
       "  12: {'A': 127889.0, 'C': 793.0, 'G': 1309.0, 'T': 3216.0},\n",
       "  13: {'A': 11280.0, 'C': 9932.0, 'G': 109803.0, 'T': 2192.0},\n",
       "  14: {'A': 13932.0, 'C': 3718.0, 'G': 7232.0, 'T': 108325.0},\n",
       "  15: {'A': 11029.0, 'C': 12698.0, 'G': 103486.0, 'T': 5994.0},\n",
       "  16: {'A': 90926.0, 'C': 11317.0, 'G': 20648.0, 'T': 10316.0},\n",
       "  17: {'A': 60832.0, 'C': 20114.0, 'G': 30578.0, 'T': 21683.0},\n",
       "  18: {'A': 59461.0, 'C': 16928.0, 'G': 29189.0, 'T': 27629.0},\n",
       "  19: {'A': 39412.0, 'C': 26247.0, 'G': 34830.0, 'T': 32718.0},\n",
       "  20: {'A': 42839.0, 'C': 24885.0, 'G': 28336.0, 'T': 37147.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 20,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '7624145',\n",
       "  'uniprot_ids': 'P17947',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0104.4',\n",
       "  'Matrix_Name': 'MYCN',\n",
       "  1: {'A': 1489.0, 'C': 1895.0, 'G': 2455.0, 'T': 1149.0},\n",
       "  2: {'A': 1735.0, 'C': 1551.0, 'G': 2600.0, 'T': 1102.0},\n",
       "  3: {'A': 481.0, 'C': 5342.0, 'G': 744.0, 'T': 421.0},\n",
       "  4: {'A': 35.0, 'C': 6930.0, 'G': 15.0, 'T': 8.0},\n",
       "  5: {'A': 6370.0, 'C': 57.0, 'G': 286.0, 'T': 275.0},\n",
       "  6: {'A': 23.0, 'C': 6518.0, 'G': 123.0, 'T': 324.0},\n",
       "  7: {'A': 324.0, 'C': 123.0, 'G': 6518.0, 'T': 23.0},\n",
       "  8: {'A': 275.0, 'C': 286.0, 'G': 57.0, 'T': 6370.0},\n",
       "  9: {'A': 8.0, 'C': 15.0, 'G': 6930.0, 'T': 35.0},\n",
       "  10: {'A': 421.0, 'C': 744.0, 'G': 5342.0, 'T': 481.0},\n",
       "  11: {'A': 1102.0, 'C': 2600.0, 'G': 1551.0, 'T': 1735.0},\n",
       "  12: {'A': 1149.0, 'C': 2455.0, 'G': 1895.0, 'T': 1489.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '18555785',\n",
       "  'uniprot_ids': 'P04198',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1517.1',\n",
       "  'Matrix_Name': 'KLF6',\n",
       "  1: {'A': 444.0, 'C': 625.0, 'G': 627.0, 'T': 500.0},\n",
       "  2: {'A': 1036.0, 'C': 491.0, 'G': 1158.0, 'T': 137.0},\n",
       "  3: {'A': 570.0, 'C': 2193.0, 'G': 333.0, 'T': 220.0},\n",
       "  4: {'A': 433.0, 'C': 2193.0, 'G': 299.0, 'T': 141.0},\n",
       "  5: {'A': 2193.0, 'C': 285.0, 'G': 61.0, 'T': 61.0},\n",
       "  6: {'A': 1.0, 'C': 2193.0, 'G': 1.0, 'T': 25.0},\n",
       "  7: {'A': 1025.0, 'C': 1.0, 'G': 2193.0, 'T': 1.0},\n",
       "  8: {'A': 1.0, 'C': 2193.0, 'G': 1.0, 'T': 1.0},\n",
       "  9: {'A': 26.0, 'C': 2193.0, 'G': 61.0, 'T': 1.0},\n",
       "  10: {'A': 72.0, 'C': 2193.0, 'G': 13.0, 'T': 65.0},\n",
       "  11: {'A': 956.0, 'C': 666.0, 'G': 53.0, 'T': 521.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '11443140',\n",
       "  'uniprot_ids': 'Q99612',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0758.1',\n",
       "  'Matrix_Name': 'E2F7',\n",
       "  1: {'A': 223.0, 'C': 73.0, 'G': 183.0, 'T': 578.0},\n",
       "  2: {'A': 39.0, 'C': 8.0, 'G': 19.0, 'T': 892.0},\n",
       "  3: {'A': 1.0, 'C': 4.0, 'G': 1.0, 'T': 964.0},\n",
       "  4: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 958.0},\n",
       "  5: {'A': 1.0, 'C': 961.0, 'G': 3.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 962.0, 'G': 11.0, 'T': 1.0},\n",
       "  7: {'A': 1.0, 'C': 966.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 3.0, 'C': 2.0, 'G': 996.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 987.0, 'G': 1.0, 'T': 1.0},\n",
       "  10: {'A': 1.0, 'C': 978.0, 'G': 18.0, 'T': 2.0},\n",
       "  11: {'A': 913.0, 'C': 1.0, 'G': 1.0, 'T': 6.0},\n",
       "  12: {'A': 931.0, 'C': 2.0, 'G': 4.0, 'T': 7.0},\n",
       "  13: {'A': 860.0, 'C': 3.0, 'G': 1.0, 'T': 44.0},\n",
       "  14: {'A': 609.0, 'C': 3.0, 'G': 3.0, 'T': 293.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'E2F-related',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '18836037',\n",
       "  'uniprot_ids': 'Q96AV8',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1522.1',\n",
       "  'Matrix_Name': 'MAZ',\n",
       "  1: {'A': 2481.0, 'C': 5199.0, 'G': 4278.0, 'T': 3109.0},\n",
       "  2: {'A': 1820.0, 'C': 5069.0, 'G': 5164.0, 'T': 3014.0},\n",
       "  3: {'A': 18.0, 'C': 14946.0, 'G': 76.0, 'T': 27.0},\n",
       "  4: {'A': 50.0, 'C': 14631.0, 'G': 385.0, 'T': 1.0},\n",
       "  5: {'A': 37.0, 'C': 14732.0, 'G': 291.0, 'T': 7.0},\n",
       "  6: {'A': 40.0, 'C': 14677.0, 'G': 348.0, 'T': 2.0},\n",
       "  7: {'A': 7.0, 'C': 1.0, 'G': 131.0, 'T': 14928.0},\n",
       "  8: {'A': 91.0, 'C': 14962.0, 'G': 9.0, 'T': 5.0},\n",
       "  9: {'A': 24.0, 'C': 14945.0, 'G': 97.0, 'T': 1.0},\n",
       "  10: {'A': 1110.0, 'C': 9373.0, 'G': 1769.0, 'T': 2815.0},\n",
       "  11: {'A': 2017.0, 'C': 6776.0, 'G': 2540.0, 'T': 3734.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '7760814',\n",
       "  'uniprot_ids': 'P56270',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0018.4',\n",
       "  'Matrix_Name': 'CREB1',\n",
       "  1: {'A': 21396.0, 'C': 17830.0, 'G': 15571.0, 'T': 24751.0},\n",
       "  2: {'A': 21089.0, 'C': 15047.0, 'G': 21105.0, 'T': 22307.0},\n",
       "  3: {'A': 28568.0, 'C': 13033.0, 'G': 27797.0, 'T': 10150.0},\n",
       "  4: {'A': 1912.0, 'C': 971.0, 'G': 1110.0, 'T': 75555.0},\n",
       "  5: {'A': 1729.0, 'C': 2299.0, 'G': 72556.0, 'T': 2964.0},\n",
       "  6: {'A': 73506.0, 'C': 1749.0, 'G': 2013.0, 'T': 2280.0},\n",
       "  7: {'A': 2187.0, 'C': 14145.0, 'G': 6340.0, 'T': 56876.0},\n",
       "  8: {'A': 1418.0, 'C': 1896.0, 'G': 74402.0, 'T': 1832.0},\n",
       "  9: {'A': 8890.0, 'C': 1232.0, 'G': 1208.0, 'T': 68218.0},\n",
       "  10: {'A': 3534.0, 'C': 71983.0, 'G': 2178.0, 'T': 1853.0},\n",
       "  11: {'A': 75820.0, 'C': 1019.0, 'G': 1223.0, 'T': 1486.0},\n",
       "  12: {'A': 10288.0, 'C': 21583.0, 'G': 14803.0, 'T': 32874.0},\n",
       "  13: {'A': 22489.0, 'C': 21877.0, 'G': 13551.0, 'T': 21631.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'CREB-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '17916232',\n",
       "  'uniprot_ids': 'P16220',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0662.1',\n",
       "  'Matrix_Name': 'MIXL1',\n",
       "  1: {'A': 6775.0, 'C': 6038.0, 'G': 6647.0, 'T': 7586.0},\n",
       "  2: {'A': 3982.0, 'C': 9031.0, 'G': 5462.0, 'T': 8571.0},\n",
       "  3: {'A': 2636.0, 'C': 11640.0, 'G': 1246.0, 'T': 15404.0},\n",
       "  4: {'A': 27043.0, 'C': 1912.0, 'G': 3661.0, 'T': 1039.0},\n",
       "  5: {'A': 27043.0, 'C': 663.0, 'G': 557.0, 'T': 737.0},\n",
       "  6: {'A': 1.0, 'C': 511.0, 'G': 182.0, 'T': 27043.0},\n",
       "  7: {'A': 1386.0, 'C': 4271.0, 'G': 2032.0, 'T': 27043.0},\n",
       "  8: {'A': 27043.0, 'C': 1904.0, 'G': 116.0, 'T': 2896.0},\n",
       "  9: {'A': 9689.0, 'C': 4895.0, 'G': 8530.0, 'T': 3932.0},\n",
       "  10: {'A': 8055.0, 'C': 8463.0, 'G': 6130.0, 'T': 4398.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'Paired-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '19711456',\n",
       "  'uniprot_ids': 'Q9H2W2',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1606.1',\n",
       "  'Matrix_Name': 'Foxf1',\n",
       "  1: {'A': 23160.0, 'C': 10754.0, 'G': 12463.0, 'T': 13167.0},\n",
       "  2: {'A': 17276.0, 'C': 7561.0, 'G': 8288.0, 'T': 26419.0},\n",
       "  3: {'A': 18535.0, 'C': 3755.0, 'G': 33600.0, 'T': 3654.0},\n",
       "  4: {'A': 569.0, 'C': 447.0, 'G': 1159.0, 'T': 57369.0},\n",
       "  5: {'A': 57294.0, 'C': 900.0, 'G': 446.0, 'T': 904.0},\n",
       "  6: {'A': 53943.0, 'C': 2250.0, 'G': 812.0, 'T': 2539.0},\n",
       "  7: {'A': 57168.0, 'C': 476.0, 'G': 1068.0, 'T': 832.0},\n",
       "  8: {'A': 438.0, 'C': 54395.0, 'G': 523.0, 'T': 4188.0},\n",
       "  9: {'A': 56437.0, 'C': 746.0, 'G': 444.0, 'T': 1917.0},\n",
       "  10: {'A': 18674.0, 'C': 12644.0, 'G': 14295.0, 'T': 13931.0},\n",
       "  11: {'A': 20324.0, 'C': 11716.0, 'G': 12845.0, 'T': 14659.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q61080',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0846.1',\n",
       "  'Matrix_Name': 'FOXC2',\n",
       "  1: {'A': 8940.0, 'C': 1280.0, 'G': 455.0, 'T': 11518.0},\n",
       "  2: {'A': 20457.0, 'C': 1653.0, 'G': 5094.0, 'T': 2081.0},\n",
       "  3: {'A': 11302.0, 'C': 3477.0, 'G': 3130.0, 'T': 9156.0},\n",
       "  4: {'A': 6419.0, 'C': 441.0, 'G': 14039.0, 'T': 137.0},\n",
       "  5: {'A': 2006.0, 'C': 3774.0, 'G': 322.0, 'T': 20457.0},\n",
       "  6: {'A': 20457.0, 'C': 6760.0, 'G': 88.0, 'T': 123.0},\n",
       "  7: {'A': 20457.0, 'C': 345.0, 'G': 82.0, 'T': 108.0},\n",
       "  8: {'A': 20457.0, 'C': 106.0, 'G': 280.0, 'T': 243.0},\n",
       "  9: {'A': 32.0, 'C': 14436.0, 'G': 69.0, 'T': 6022.0},\n",
       "  10: {'A': 20457.0, 'C': 102.0, 'G': 329.0, 'T': 216.0},\n",
       "  11: {'A': 13297.0, 'C': 2088.0, 'G': 1711.0, 'T': 7161.0},\n",
       "  12: {'A': 10996.0, 'C': 3081.0, 'G': 2650.0, 'T': 9462.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'Q99958',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0062.3',\n",
       "  'Matrix_Name': 'GABPA',\n",
       "  1: {'A': 15888.0, 'C': 16923.0, 'G': 15946.0, 'T': 16726.0},\n",
       "  2: {'A': 10982.0, 'C': 21680.0, 'G': 13752.0, 'T': 19069.0},\n",
       "  3: {'A': 7640.0, 'C': 41399.0, 'G': 6990.0, 'T': 9454.0},\n",
       "  4: {'A': 52261.0, 'C': 3989.0, 'G': 4874.0, 'T': 4359.0},\n",
       "  5: {'A': 930.0, 'C': 55665.0, 'G': 2170.0, 'T': 6718.0},\n",
       "  6: {'A': 3886.0, 'C': 1444.0, 'G': 966.0, 'T': 59187.0},\n",
       "  7: {'A': 863.0, 'C': 1231.0, 'G': 1634.0, 'T': 61755.0},\n",
       "  8: {'A': 1399.0, 'C': 61374.0, 'G': 1151.0, 'T': 1559.0},\n",
       "  9: {'A': 994.0, 'C': 63108.0, 'G': 521.0, 'T': 860.0},\n",
       "  10: {'A': 1723.0, 'C': 1543.0, 'G': 4562.0, 'T': 57655.0},\n",
       "  11: {'A': 2220.0, 'C': 6329.0, 'G': 51921.0, 'T': 5013.0},\n",
       "  12: {'A': 5345.0, 'C': 12007.0, 'G': 7721.0, 'T': 40410.0},\n",
       "  13: {'A': 11904.0, 'C': 18155.0, 'G': 15082.0, 'T': 20342.0},\n",
       "  14: {'A': 13523.0, 'C': 17383.0, 'G': 13188.0, 'T': 21389.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Ets-related',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '9461436',\n",
       "  'uniprot_ids': 'Q06546',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0831.2',\n",
       "  'Matrix_Name': 'TFE3',\n",
       "  1: {'A': 2.0, 'C': 997.0, 'G': 2.0, 'T': 2.0},\n",
       "  2: {'A': 720.0, 'C': 35.0, 'G': 110.0, 'T': 139.0},\n",
       "  3: {'A': 2.0, 'C': 831.0, 'G': 6.0, 'T': 165.0},\n",
       "  4: {'A': 130.0, 'C': 10.0, 'G': 862.0, 'T': 2.0},\n",
       "  5: {'A': 5.0, 'C': 5.0, 'G': 5.0, 'T': 989.0},\n",
       "  6: {'A': 2.0, 'C': 2.0, 'G': 998.0, 'T': 2.0},\n",
       "  7: {'A': 816.0, 'C': 78.0, 'G': 71.0, 'T': 39.0},\n",
       "  8: {'A': 4.0, 'C': 523.0, 'G': 38.0, 'T': 439.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'bHLH-ZIP',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23933270',\n",
       "  'uniprot_ids': 'P19532',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0516.2',\n",
       "  'Matrix_Name': 'SP2',\n",
       "  1: {'A': 257.0, 'C': 309.0, 'G': 124.0, 'T': 107.0},\n",
       "  2: {'A': 54.0, 'C': 172.0, 'G': 58.0, 'T': 609.0},\n",
       "  3: {'A': 614.0, 'C': 77.0, 'G': 1.0, 'T': 181.0},\n",
       "  4: {'A': 754.0, 'C': 85.0, 'G': 34.0, 'T': 68.0},\n",
       "  5: {'A': 55.0, 'C': 128.0, 'G': 792.0, 'T': 44.0},\n",
       "  6: {'A': 35.0, 'C': 457.0, 'G': 9.0, 'T': 328.0},\n",
       "  7: {'A': 15.0, 'C': 775.0, 'G': 2.0, 'T': 1.0},\n",
       "  8: {'A': 17.0, 'C': 787.0, 'G': 14.0, 'T': 9.0},\n",
       "  9: {'A': 5.0, 'C': 789.0, 'G': 23.0, 'T': 13.0},\n",
       "  10: {'A': 83.0, 'C': 53.0, 'G': 781.0, 'T': 46.0},\n",
       "  11: {'A': 1.0, 'C': 796.0, 'G': 1.0, 'T': 19.0},\n",
       "  12: {'A': 6.0, 'C': 794.0, 'G': 27.0, 'T': 8.0},\n",
       "  13: {'A': 24.0, 'C': 775.0, 'G': 1.0, 'T': 28.0},\n",
       "  14: {'A': 403.0, 'C': 391.0, 'G': 23.0, 'T': 18.0},\n",
       "  15: {'A': 23.0, 'C': 680.0, 'G': 2.0, 'T': 102.0},\n",
       "  16: {'A': 110.0, 'C': 222.0, 'G': 1.0, 'T': 504.0},\n",
       "  17: {'A': 151.0, 'C': 204.0, 'G': 89.0, 'T': 353.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 17,\n",
       "  'tf_family': 'Three-zinc',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '14726517',\n",
       "  'uniprot_ids': 'Q02086',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1119.1',\n",
       "  'Matrix_Name': 'SIX2',\n",
       "  1: {'A': 1674.0, 'C': 1235.0, 'G': 1146.0, 'T': 1402.0},\n",
       "  2: {'A': 1608.0, 'C': 1243.0, 'G': 1227.0, 'T': 1379.0},\n",
       "  3: {'A': 964.0, 'C': 2911.0, 'G': 829.0, 'T': 753.0},\n",
       "  4: {'A': 841.0, 'C': 216.0, 'G': 562.0, 'T': 3838.0},\n",
       "  5: {'A': 166.0, 'C': 57.0, 'G': 5066.0, 'T': 168.0},\n",
       "  6: {'A': 3579.0, 'C': 320.0, 'G': 274.0, 'T': 1284.0},\n",
       "  7: {'A': 4886.0, 'C': 265.0, 'G': 67.0, 'T': 239.0},\n",
       "  8: {'A': 5276.0, 'C': 40.0, 'G': 75.0, 'T': 66.0},\n",
       "  9: {'A': 191.0, 'C': 4368.0, 'G': 171.0, 'T': 727.0},\n",
       "  10: {'A': 587.0, 'C': 4170.0, 'G': 197.0, 'T': 503.0},\n",
       "  11: {'A': 326.0, 'C': 296.0, 'G': 176.0, 'T': 4659.0},\n",
       "  12: {'A': 322.0, 'C': 115.0, 'G': 4831.0, 'T': 189.0},\n",
       "  13: {'A': 5203.0, 'C': 76.0, 'G': 86.0, 'T': 92.0},\n",
       "  14: {'A': 554.0, 'C': 616.0, 'G': 1194.0, 'T': 3093.0},\n",
       "  15: {'A': 2416.0, 'C': 1813.0, 'G': 564.0, 'T': 664.0},\n",
       "  16: {'A': 859.0, 'C': 2524.0, 'G': 512.0, 'T': 1562.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'HD-SINE',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '15327782',\n",
       "  'uniprot_ids': 'Q9NPC8',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1147.1',\n",
       "  'Matrix_Name': 'NR4A2::RXRA',\n",
       "  1: {'A': 289.0, 'C': 213.0, 'G': 372.0, 'T': 131.0},\n",
       "  2: {'A': 499.0, 'C': 125.0, 'G': 349.0, 'T': 30.0},\n",
       "  3: {'A': 21.0, 'C': 12.0, 'G': 925.0, 'T': 47.0},\n",
       "  4: {'A': 52.0, 'C': 8.0, 'G': 805.0, 'T': 140.0},\n",
       "  5: {'A': 47.0, 'C': 135.0, 'G': 129.0, 'T': 692.0},\n",
       "  6: {'A': 8.0, 'C': 825.0, 'G': 32.0, 'T': 139.0},\n",
       "  7: {'A': 593.0, 'C': 39.0, 'G': 344.0, 'T': 28.0},\n",
       "  8: {'A': 249.0, 'C': 61.0, 'G': 88.0, 'T': 605.0},\n",
       "  9: {'A': 3.0, 'C': 70.0, 'G': 6.0, 'T': 925.0},\n",
       "  10: {'A': 71.0, 'C': 83.0, 'G': 837.0, 'T': 13.0},\n",
       "  11: {'A': 927.0, 'C': 55.0, 'G': 18.0, 'T': 4.0},\n",
       "  12: {'A': 158.0, 'C': 831.0, 'G': 3.0, 'T': 12.0},\n",
       "  13: {'A': 41.0, 'C': 938.0, 'G': 5.0, 'T': 21.0},\n",
       "  14: {'A': 6.0, 'C': 536.0, 'G': 24.0, 'T': 438.0},\n",
       "  15: {'A': 105.0, 'C': 474.0, 'G': 181.0, 'T': 245.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'NGFI-B-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '7705655',\n",
       "  'uniprot_ids': 'P43354;',\n",
       "  'data_type': 'SMiLE-seq'},\n",
       " {'Matrix_ID': 'MA0875.1',\n",
       "  'Matrix_Name': 'BARX1',\n",
       "  1: {'A': 1873.0, 'C': 1822.0, 'G': 2237.0, 'T': 878.0},\n",
       "  2: {'A': 591.0, 'C': 5139.0, 'G': 247.0, 'T': 1668.0},\n",
       "  3: {'A': 3340.0, 'C': 1993.0, 'G': 1387.0, 'T': 88.0},\n",
       "  4: {'A': 6806.0, 'C': 208.0, 'G': 193.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6806.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 6806.0},\n",
       "  7: {'A': 6806.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 1918.0, 'C': 1392.0, 'G': 2421.0, 'T': 1078.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 8,\n",
       "  'tf_family': 'NK-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'Q9HBU1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0119.1',\n",
       "  'Matrix_Name': 'NFIC::TLX1',\n",
       "  1: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 17.0},\n",
       "  2: {'A': 1.0, 'C': 1.0, 'G': 17.0, 'T': 1.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 17.0, 'T': 1.0},\n",
       "  4: {'A': 1.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  5: {'A': 15.0, 'C': 2.0, 'G': 1.0, 'T': 2.0},\n",
       "  6: {'A': 3.0, 'C': 9.0, 'G': 6.0, 'T': 2.0},\n",
       "  7: {'A': 3.0, 'C': 9.0, 'G': 5.0, 'T': 3.0},\n",
       "  8: {'A': 8.0, 'C': 2.0, 'G': 6.0, 'T': 4.0},\n",
       "  9: {'A': 5.0, 'C': 4.0, 'G': 3.0, 'T': 8.0},\n",
       "  10: {'A': 1.0, 'C': 1.0, 'G': 17.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 17.0, 'G': 1.0, 'T': 1.0},\n",
       "  13: {'A': 17.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  14: {'A': 15.0, 'C': 1.0, 'G': 2.0, 'T': 2.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '10327073',\n",
       "  'uniprot_ids': 'P08651;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1566.1',\n",
       "  'Matrix_Name': 'TBX3',\n",
       "  1: {'A': 645.0, 'C': 400.0, 'G': 1004.0, 'T': 498.0},\n",
       "  2: {'A': 2545.0, 'C': 482.0, 'G': 1208.0, 'T': 761.0},\n",
       "  3: {'A': 428.0, 'C': 863.0, 'G': 2545.0, 'T': 303.0},\n",
       "  4: {'A': 7.0, 'C': 46.0, 'G': 2545.0, 'T': 1.0},\n",
       "  5: {'A': 17.0, 'C': 736.0, 'G': 1.0, 'T': 2545.0},\n",
       "  6: {'A': 1.0, 'C': 1.0, 'G': 2545.0, 'T': 1.0},\n",
       "  7: {'A': 116.0, 'C': 635.0, 'G': 156.0, 'T': 2545.0},\n",
       "  8: {'A': 171.0, 'C': 1472.0, 'G': 2545.0, 'T': 1113.0},\n",
       "  9: {'A': 2545.0, 'C': 411.0, 'G': 948.0, 'T': 485.0},\n",
       "  10: {'A': 761.0, 'C': 587.0, 'G': 621.0, 'T': 579.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 10,\n",
       "  'tf_family': 'TBX2-related',\n",
       "  'tf_class': 'T-Box',\n",
       "  'pubmed_ids': '18829543',\n",
       "  'uniprot_ids': 'O15119',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0906.1',\n",
       "  'Matrix_Name': 'HOXC12',\n",
       "  1: {'A': 1451.0, 'C': 260.0, 'G': 2173.0, 'T': 418.0},\n",
       "  2: {'A': 1261.0, 'C': 709.0, 'G': 4298.0, 'T': 2.0},\n",
       "  3: {'A': 12.0, 'C': 69.0, 'G': 3.0, 'T': 4298.0},\n",
       "  4: {'A': 200.0, 'C': 4298.0, 'G': 13.0, 'T': 164.0},\n",
       "  5: {'A': 377.0, 'C': 3.0, 'G': 4298.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 3.0, 'G': 2.0, 'T': 4298.0},\n",
       "  7: {'A': 4298.0, 'C': 15.0, 'G': 6.0, 'T': 554.0},\n",
       "  8: {'A': 4298.0, 'C': 1.0, 'G': 3.0, 'T': 26.0},\n",
       "  9: {'A': 4298.0, 'C': 14.0, 'G': 1.0, 'T': 30.0},\n",
       "  10: {'A': 4298.0, 'C': 637.0, 'G': 270.0, 'T': 516.0},\n",
       "  11: {'A': 1443.0, 'C': 1140.0, 'G': 373.0, 'T': 1345.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'HOX-related',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '18585359',\n",
       "  'uniprot_ids': 'P31275',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1419.1',\n",
       "  'Matrix_Name': 'IRF4',\n",
       "  1: {'A': 10449.0, 'C': 18877.0, 'G': 2106.0, 'T': 13902.0},\n",
       "  2: {'A': 365.0, 'C': 45332.0, 'G': 88.0, 'T': 5938.0},\n",
       "  3: {'A': 351.0, 'C': 67.0, 'G': 45332.0, 'T': 32.0},\n",
       "  4: {'A': 45332.0, 'C': 125.0, 'G': 92.0, 'T': 113.0},\n",
       "  5: {'A': 45332.0, 'C': 9.0, 'G': 26.0, 'T': 445.0},\n",
       "  6: {'A': 45332.0, 'C': 7.0, 'G': 5.0, 'T': 9.0},\n",
       "  7: {'A': 188.0, 'C': 45332.0, 'G': 594.0, 'T': 498.0},\n",
       "  8: {'A': 28.0, 'C': 45332.0, 'G': 5.0, 'T': 3802.0},\n",
       "  9: {'A': 58.0, 'C': 7.0, 'G': 45332.0, 'T': 4.0},\n",
       "  10: {'A': 45332.0, 'C': 47.0, 'G': 8.0, 'T': 6.0},\n",
       "  11: {'A': 45332.0, 'C': 12.0, 'G': 24.0, 'T': 1003.0},\n",
       "  12: {'A': 45332.0, 'C': 16.0, 'G': 5.0, 'T': 14.0},\n",
       "  13: {'A': 830.0, 'C': 45332.0, 'G': 4252.0, 'T': 81.0},\n",
       "  14: {'A': 1378.0, 'C': 17590.0, 'G': 254.0, 'T': 27743.0},\n",
       "  15: {'A': 25203.0, 'C': 4641.0, 'G': 5645.0, 'T': 9847.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Interferon-regulatory',\n",
       "  'tf_class': 'Tryptophan',\n",
       "  'pubmed_ids': '25896227',\n",
       "  'uniprot_ids': 'Q15306',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0161.2',\n",
       "  'Matrix_Name': 'NFIC',\n",
       "  1: {'A': 2616.0, 'C': 2648.0, 'G': 2479.0, 'T': 3868.0},\n",
       "  2: {'A': 3897.0, 'C': 2138.0, 'G': 2087.0, 'T': 3489.0},\n",
       "  3: {'A': 413.0, 'C': 10543.0, 'G': 330.0, 'T': 325.0},\n",
       "  4: {'A': 155.0, 'C': 304.0, 'G': 370.0, 'T': 10782.0},\n",
       "  5: {'A': 353.0, 'C': 334.0, 'G': 178.0, 'T': 10746.0},\n",
       "  6: {'A': 311.0, 'C': 130.0, 'G': 11088.0, 'T': 82.0},\n",
       "  7: {'A': 224.0, 'C': 122.0, 'G': 10974.0, 'T': 291.0},\n",
       "  8: {'A': 265.0, 'C': 10726.0, 'G': 89.0, 'T': 531.0},\n",
       "  9: {'A': 10159.0, 'C': 398.0, 'G': 479.0, 'T': 575.0},\n",
       "  10: {'A': 2627.0, 'C': 2952.0, 'G': 3187.0, 'T': 2845.0},\n",
       "  11: {'A': 3345.0, 'C': 2713.0, 'G': 2761.0, 'T': 2792.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Nuclear',\n",
       "  'tf_class': 'SMAD/NF-1',\n",
       "  'pubmed_ids': '12101405',\n",
       "  'uniprot_ids': 'P08651',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1529.1',\n",
       "  'Matrix_Name': 'NHLH2',\n",
       "  1: {'A': 4835.0, 'C': 2656.0, 'G': 9733.0, 'T': 4144.0},\n",
       "  2: {'A': 4313.0, 'C': 2220.0, 'G': 13904.0, 'T': 5199.0},\n",
       "  3: {'A': 3742.0, 'C': 1066.0, 'G': 18451.0, 'T': 2764.0},\n",
       "  4: {'A': 4922.0, 'C': 6142.0, 'G': 4909.0, 'T': 5395.0},\n",
       "  5: {'A': 5604.0, 'C': 15452.0, 'G': 672.0, 'T': 152.0},\n",
       "  6: {'A': 2379.0, 'C': 2325.0, 'G': 18931.0, 'T': 2635.0},\n",
       "  7: {'A': 3.0, 'C': 21360.0, 'G': 3.0, 'T': 6.0},\n",
       "  8: {'A': 21365.0, 'C': 1.0, 'G': 27.0, 'T': 2.0},\n",
       "  9: {'A': 7.0, 'C': 2460.0, 'G': 20097.0, 'T': 1055.0},\n",
       "  10: {'A': 526.0, 'C': 20269.0, 'G': 2267.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 15.0, 'G': 4.0, 'T': 21359.0},\n",
       "  12: {'A': 1.0, 'C': 3.0, 'G': 21364.0, 'T': 1.0},\n",
       "  13: {'A': 694.0, 'C': 20612.0, 'G': 276.0, 'T': 1231.0},\n",
       "  14: {'A': 211.0, 'C': 205.0, 'G': 20978.0, 'T': 741.0},\n",
       "  15: {'A': 1185.0, 'C': 7208.0, 'G': 1521.0, 'T': 13504.0},\n",
       "  16: {'A': 3767.0, 'C': 18352.0, 'G': 1184.0, 'T': 2534.0},\n",
       "  17: {'A': 6393.0, 'C': 13463.0, 'G': 1580.0, 'T': 3331.0},\n",
       "  18: {'A': 3602.0, 'C': 11272.0, 'G': 2265.0, 'T': 4229.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 18,\n",
       "  'tf_family': 'Tal-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '23354101',\n",
       "  'uniprot_ids': 'Q02577',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0528.2',\n",
       "  'Matrix_Name': 'ZNF263',\n",
       "  1: {'A': 13288.0, 'C': 13041.0, 'G': 20284.0, 'T': 9734.0},\n",
       "  2: {'A': 11017.0, 'C': 14910.0, 'G': 17884.0, 'T': 12536.0},\n",
       "  3: {'A': 13002.0, 'C': 6184.0, 'G': 28913.0, 'T': 8248.0},\n",
       "  4: {'A': 1611.0, 'C': 1660.0, 'G': 51805.0, 'T': 1271.0},\n",
       "  5: {'A': 790.0, 'C': 1545.0, 'G': 52819.0, 'T': 1193.0},\n",
       "  6: {'A': 951.0, 'C': 973.0, 'G': 53436.0, 'T': 987.0},\n",
       "  7: {'A': 53901.0, 'C': 1661.0, 'G': 164.0, 'T': 621.0},\n",
       "  8: {'A': 831.0, 'C': 967.0, 'G': 52876.0, 'T': 1673.0},\n",
       "  9: {'A': 696.0, 'C': 2191.0, 'G': 52742.0, 'T': 718.0},\n",
       "  10: {'A': 52500.0, 'C': 1759.0, 'G': 1449.0, 'T': 639.0},\n",
       "  11: {'A': 9540.0, 'C': 15414.0, 'G': 24826.0, 'T': 6567.0},\n",
       "  12: {'A': 8437.0, 'C': 11690.0, 'G': 25116.0, 'T': 11104.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19887448',\n",
       "  'uniprot_ids': 'O14978',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1583.1',\n",
       "  'Matrix_Name': 'ZFP57',\n",
       "  1: {'A': 1157.0, 'C': 1722.0, 'G': 1830.0, 'T': 1627.0},\n",
       "  2: {'A': 1554.0, 'C': 1612.0, 'G': 1599.0, 'T': 1571.0},\n",
       "  3: {'A': 2237.0, 'C': 1515.0, 'G': 1644.0, 'T': 940.0},\n",
       "  4: {'A': 640.0, 'C': 1062.0, 'G': 1256.0, 'T': 3378.0},\n",
       "  5: {'A': 13.0, 'C': 38.0, 'G': 141.0, 'T': 6144.0},\n",
       "  6: {'A': 25.0, 'C': 72.0, 'G': 6214.0, 'T': 25.0},\n",
       "  7: {'A': 27.0, 'C': 6255.0, 'G': 17.0, 'T': 37.0},\n",
       "  8: {'A': 12.0, 'C': 5596.0, 'G': 720.0, 'T': 8.0},\n",
       "  9: {'A': 92.0, 'C': 17.0, 'G': 6167.0, 'T': 60.0},\n",
       "  10: {'A': 39.0, 'C': 5829.0, 'G': 280.0, 'T': 188.0},\n",
       "  11: {'A': 5117.0, 'C': 332.0, 'G': 730.0, 'T': 157.0},\n",
       "  12: {'A': 1451.0, 'C': 1700.0, 'G': 1897.0, 'T': 1288.0},\n",
       "  13: {'A': 1352.0, 'C': 1666.0, 'G': 1628.0, 'T': 1690.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 13,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '30602440',\n",
       "  'uniprot_ids': 'Q9NU63',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0019.1',\n",
       "  'Matrix_Name': 'Ddit3::Cebpa',\n",
       "  1: {'A': 15.0, 'C': 8.0, 'G': 13.0, 'T': 7.0},\n",
       "  2: {'A': 12.0, 'C': 8.0, 'G': 15.0, 'T': 8.0},\n",
       "  3: {'A': 19.0, 'C': 4.0, 'G': 16.0, 'T': 4.0},\n",
       "  4: {'A': 1.0, 'C': 2.0, 'G': 1.0, 'T': 39.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 39.0, 'T': 2.0},\n",
       "  6: {'A': 5.0, 'C': 34.0, 'G': 1.0, 'T': 3.0},\n",
       "  7: {'A': 39.0, 'C': 2.0, 'G': 1.0, 'T': 1.0},\n",
       "  8: {'A': 37.0, 'C': 3.0, 'G': 2.0, 'T': 1.0},\n",
       "  9: {'A': 1.0, 'C': 7.0, 'G': 1.0, 'T': 34.0},\n",
       "  10: {'A': 15.0, 'C': 18.0, 'G': 6.0, 'T': 4.0},\n",
       "  11: {'A': 5.0, 'C': 24.0, 'G': 10.0, 'T': 4.0},\n",
       "  12: {'A': 1.0, 'C': 27.0, 'G': 7.0, 'T': 8.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'C/EBP-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '8657121',\n",
       "  'uniprot_ids': 'Q62857;',\n",
       "  'data_type': 'SELEX'},\n",
       " {'Matrix_ID': 'MA1110.1',\n",
       "  'Matrix_Name': 'NR1H4',\n",
       "  1: {'A': 77.0, 'C': 84.0, 'G': 118.0, 'T': 230.0},\n",
       "  2: {'A': 69.0, 'C': 241.0, 'G': 88.0, 'T': 111.0},\n",
       "  3: {'A': 462.0, 'C': 11.0, 'G': 13.0, 'T': 23.0},\n",
       "  4: {'A': 414.0, 'C': 29.0, 'G': 37.0, 'T': 29.0},\n",
       "  5: {'A': 4.0, 'C': 16.0, 'G': 7.0, 'T': 482.0},\n",
       "  6: {'A': 25.0, 'C': 8.0, 'G': 463.0, 'T': 13.0},\n",
       "  7: {'A': 456.0, 'C': 23.0, 'G': 15.0, 'T': 15.0},\n",
       "  8: {'A': 27.0, 'C': 469.0, 'G': 1.0, 'T': 12.0},\n",
       "  9: {'A': 13.0, 'C': 477.0, 'G': 7.0, 'T': 12.0},\n",
       "  10: {'A': 73.0, 'C': 136.0, 'G': 82.0, 'T': 218.0},\n",
       "  11: {'A': 136.0, 'C': 132.0, 'G': 106.0, 'T': 135.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'Thyroid',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '24498423',\n",
       "  'uniprot_ids': 'Q96RI1',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1616.1',\n",
       "  'Matrix_Name': 'Prdm15',\n",
       "  1: {'A': 639.0, 'C': 487.0, 'G': 561.0, 'T': 554.0},\n",
       "  2: {'A': 625.0, 'C': 330.0, 'G': 921.0, 'T': 365.0},\n",
       "  3: {'A': 253.0, 'C': 326.0, 'G': 1334.0, 'T': 328.0},\n",
       "  4: {'A': 2012.0, 'C': 61.0, 'G': 136.0, 'T': 32.0},\n",
       "  5: {'A': 2122.0, 'C': 35.0, 'G': 59.0, 'T': 25.0},\n",
       "  6: {'A': 1943.0, 'C': 48.0, 'G': 201.0, 'T': 49.0},\n",
       "  7: {'A': 1802.0, 'C': 139.0, 'G': 153.0, 'T': 147.0},\n",
       "  8: {'A': 68.0, 'C': 1942.0, 'G': 102.0, 'T': 129.0},\n",
       "  9: {'A': 223.0, 'C': 1887.0, 'G': 108.0, 'T': 23.0},\n",
       "  10: {'A': 221.0, 'C': 362.0, 'G': 29.0, 'T': 1629.0},\n",
       "  11: {'A': 20.0, 'C': 8.0, 'G': 2200.0, 'T': 13.0},\n",
       "  12: {'A': 65.0, 'C': 24.0, 'G': 2127.0, 'T': 25.0},\n",
       "  13: {'A': 1532.0, 'C': 315.0, 'G': 221.0, 'T': 173.0},\n",
       "  14: {'A': 688.0, 'C': 439.0, 'G': 749.0, 'T': 365.0},\n",
       "  15: {'A': 541.0, 'C': 636.0, 'G': 461.0, 'T': 603.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28740264',\n",
       "  'uniprot_ids': 'E9Q8T2',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1655.1',\n",
       "  'Matrix_Name': 'ZNF341',\n",
       "  1: {'A': 4267.0, 'C': 4257.0, 'G': 5799.0, 'T': 4409.0},\n",
       "  2: {'A': 5118.0, 'C': 2181.0, 'G': 8833.0, 'T': 2600.0},\n",
       "  3: {'A': 1505.0, 'C': 2044.0, 'G': 13740.0, 'T': 1443.0},\n",
       "  4: {'A': 17075.0, 'C': 956.0, 'G': 514.0, 'T': 187.0},\n",
       "  5: {'A': 17701.0, 'C': 212.0, 'G': 577.0, 'T': 242.0},\n",
       "  6: {'A': 302.0, 'C': 16814.0, 'G': 1034.0, 'T': 582.0},\n",
       "  7: {'A': 17628.0, 'C': 400.0, 'G': 469.0, 'T': 235.0},\n",
       "  8: {'A': 169.0, 'C': 360.0, 'G': 17973.0, 'T': 230.0},\n",
       "  9: {'A': 228.0, 'C': 18004.0, 'G': 271.0, 'T': 229.0},\n",
       "  10: {'A': 3083.0, 'C': 10947.0, 'G': 2290.0, 'T': 2412.0},\n",
       "  11: {'A': 5306.0, 'C': 4432.0, 'G': 4673.0, 'T': 4321.0},\n",
       "  12: {'A': 3757.0, 'C': 5746.0, 'G': 5692.0, 'T': 3537.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'Factors',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '29907691',\n",
       "  'uniprot_ids': 'Q9BYN7',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA1579.1',\n",
       "  'Matrix_Name': 'ZBTB26',\n",
       "  1: {'A': 400.0, 'C': 387.0, 'G': 403.0, 'T': 441.0},\n",
       "  2: {'A': 450.0, 'C': 343.0, 'G': 413.0, 'T': 425.0},\n",
       "  3: {'A': 275.0, 'C': 873.0, 'G': 124.0, 'T': 359.0},\n",
       "  4: {'A': 10.0, 'C': 6.0, 'G': 6.0, 'T': 1609.0},\n",
       "  5: {'A': 21.0, 'C': 1593.0, 'G': 13.0, 'T': 4.0},\n",
       "  6: {'A': 6.0, 'C': 1504.0, 'G': 14.0, 'T': 107.0},\n",
       "  7: {'A': 1536.0, 'C': 18.0, 'G': 50.0, 'T': 27.0},\n",
       "  8: {'A': 28.0, 'C': 4.0, 'G': 1582.0, 'T': 17.0},\n",
       "  9: {'A': 1583.0, 'C': 23.0, 'G': 10.0, 'T': 15.0},\n",
       "  10: {'A': 1135.0, 'C': 139.0, 'G': 224.0, 'T': 133.0},\n",
       "  11: {'A': 670.0, 'C': 328.0, 'G': 323.0, 'T': 310.0},\n",
       "  12: {'A': 584.0, 'C': 379.0, 'G': 339.0, 'T': 329.0},\n",
       "  13: {'A': 433.0, 'C': 380.0, 'G': 484.0, 'T': 334.0},\n",
       "  14: {'A': 426.0, 'C': 410.0, 'G': 393.0, 'T': 402.0},\n",
       "  15: {'A': 382.0, 'C': 450.0, 'G': 328.0, 'T': 471.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '28473536',\n",
       "  'uniprot_ids': 'Q9HCK0',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0852.2',\n",
       "  'Matrix_Name': 'FOXK1',\n",
       "  1: {'A': 330.0, 'C': 185.0, 'G': 259.0, 'T': 286.0},\n",
       "  2: {'A': 417.0, 'C': 176.0, 'G': 271.0, 'T': 196.0},\n",
       "  3: {'A': 248.0, 'C': 143.0, 'G': 273.0, 'T': 396.0},\n",
       "  4: {'A': 144.0, 'C': 29.0, 'G': 869.0, 'T': 18.0},\n",
       "  5: {'A': 61.0, 'C': 6.0, 'G': 45.0, 'T': 948.0},\n",
       "  6: {'A': 1004.0, 'C': 38.0, 'G': 11.0, 'T': 7.0},\n",
       "  7: {'A': 1000.0, 'C': 38.0, 'G': 12.0, 'T': 10.0},\n",
       "  8: {'A': 1023.0, 'C': 14.0, 'G': 11.0, 'T': 12.0},\n",
       "  9: {'A': 11.0, 'C': 959.0, 'G': 13.0, 'T': 77.0},\n",
       "  10: {'A': 1016.0, 'C': 17.0, 'G': 13.0, 'T': 14.0},\n",
       "  11: {'A': 513.0, 'C': 200.0, 'G': 205.0, 'T': 142.0},\n",
       "  12: {'A': 186.0, 'C': 202.0, 'G': 569.0, 'T': 103.0},\n",
       "  13: {'A': 243.0, 'C': 270.0, 'G': 363.0, 'T': 184.0},\n",
       "  14: {'A': 284.0, 'C': 289.0, 'G': 271.0, 'T': 216.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '16624804',\n",
       "  'uniprot_ids': 'P85037',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0519.1',\n",
       "  'Matrix_Name': 'Stat5a::Stat5b',\n",
       "  1: {'A': 6173.0, 'C': 2327.0, 'G': 4426.0, 'T': 3585.0},\n",
       "  2: {'A': 2074.0, 'C': 3597.0, 'G': 2902.0, 'T': 7938.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 1.0, 'T': 16508.0},\n",
       "  4: {'A': 1.0, 'C': 1601.0, 'G': 1.0, 'T': 14908.0},\n",
       "  5: {'A': 1.0, 'C': 16342.0, 'G': 1.0, 'T': 167.0},\n",
       "  6: {'A': 264.0, 'C': 12796.0, 'G': 1.0, 'T': 3450.0},\n",
       "  7: {'A': 7045.0, 'C': 6138.0, 'G': 1.0, 'T': 3327.0},\n",
       "  8: {'A': 11544.0, 'C': 1.0, 'G': 4141.0, 'T': 825.0},\n",
       "  9: {'A': 1.0, 'C': 1.0, 'G': 16508.0, 'T': 1.0},\n",
       "  10: {'A': 14889.0, 'C': 94.0, 'G': 1.0, 'T': 1527.0},\n",
       "  11: {'A': 14887.0, 'C': 1.0, 'G': 1622.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'STAT',\n",
       "  'tf_class': 'STAT',\n",
       "  'pubmed_ids': '22158971',\n",
       "  'uniprot_ids': 'P42230;',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0089.2',\n",
       "  'Matrix_Name': 'NFE2L1',\n",
       "  1: {'A': 1245.0, 'C': 704.0, 'G': 672.0, 'T': 739.0},\n",
       "  2: {'A': 879.0, 'C': 947.0, 'G': 1051.0, 'T': 483.0},\n",
       "  3: {'A': 993.0, 'C': 854.0, 'G': 743.0, 'T': 770.0},\n",
       "  4: {'A': 2583.0, 'C': 58.0, 'G': 707.0, 'T': 12.0},\n",
       "  5: {'A': 13.0, 'C': 6.0, 'G': 3.0, 'T': 3338.0},\n",
       "  6: {'A': 18.0, 'C': 8.0, 'G': 3309.0, 'T': 25.0},\n",
       "  7: {'A': 3337.0, 'C': 5.0, 'G': 11.0, 'T': 7.0},\n",
       "  8: {'A': 52.0, 'C': 2865.0, 'G': 359.0, 'T': 84.0},\n",
       "  9: {'A': 219.0, 'C': 13.0, 'G': 36.0, 'T': 3092.0},\n",
       "  10: {'A': 186.0, 'C': 3050.0, 'G': 26.0, 'T': 98.0},\n",
       "  11: {'A': 3172.0, 'C': 10.0, 'G': 79.0, 'T': 99.0},\n",
       "  12: {'A': 58.0, 'C': 19.0, 'G': 3019.0, 'T': 264.0},\n",
       "  13: {'A': 17.0, 'C': 3282.0, 'G': 34.0, 'T': 27.0},\n",
       "  14: {'A': 2858.0, 'C': 65.0, 'G': 143.0, 'T': 294.0},\n",
       "  15: {'A': 1363.0, 'C': 518.0, 'G': 695.0, 'T': 784.0},\n",
       "  16: {'A': 1146.0, 'C': 225.0, 'G': 306.0, 'T': 1683.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 16,\n",
       "  'tf_family': 'Jun-related',\n",
       "  'tf_class': 'Basic',\n",
       "  'pubmed_ids': '9421508',\n",
       "  'uniprot_ids': 'Q14494',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0850.1',\n",
       "  'Matrix_Name': 'FOXP3',\n",
       "  1: {'A': 75.0, 'C': 1.0, 'G': 138.0, 'T': 1.0},\n",
       "  2: {'A': 21.0, 'C': 53.0, 'G': 1.0, 'T': 212.0},\n",
       "  3: {'A': 212.0, 'C': 1.0, 'G': 1.0, 'T': 32.0},\n",
       "  4: {'A': 212.0, 'C': 1.0, 'G': 10.0, 'T': 9.0},\n",
       "  5: {'A': 212.0, 'C': 16.0, 'G': 50.0, 'T': 1.0},\n",
       "  6: {'A': 1.0, 'C': 212.0, 'G': 33.0, 'T': 40.0},\n",
       "  7: {'A': 212.0, 'C': 1.0, 'G': 19.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '20554955',\n",
       "  'uniprot_ids': 'B7ZLG1',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1650.1',\n",
       "  'Matrix_Name': 'ZBTB14',\n",
       "  1: {'A': 571.0, 'C': 1780.0, 'G': 1669.0, 'T': 531.0},\n",
       "  2: {'A': 502.0, 'C': 1973.0, 'G': 1556.0, 'T': 520.0},\n",
       "  3: {'A': 336.0, 'C': 3526.0, 'G': 615.0, 'T': 74.0},\n",
       "  4: {'A': 42.0, 'C': 4293.0, 'G': 156.0, 'T': 60.0},\n",
       "  5: {'A': 94.0, 'C': 295.0, 'G': 4099.0, 'T': 63.0},\n",
       "  6: {'A': 69.0, 'C': 4079.0, 'G': 320.0, 'T': 83.0},\n",
       "  7: {'A': 133.0, 'C': 306.0, 'G': 4044.0, 'T': 68.0},\n",
       "  8: {'A': 57.0, 'C': 4179.0, 'G': 262.0, 'T': 53.0},\n",
       "  9: {'A': 4501.0, 'C': 38.0, 'G': 11.0, 'T': 1.0},\n",
       "  10: {'A': 100.0, 'C': 3675.0, 'G': 652.0, 'T': 124.0},\n",
       "  11: {'A': 971.0, 'C': 1619.0, 'G': 1452.0, 'T': 509.0},\n",
       "  12: {'A': 443.0, 'C': 2261.0, 'G': 1240.0, 'T': 607.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 12,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'O43829',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0602.1',\n",
       "  'Matrix_Name': 'Arid5a',\n",
       "  1: {'A': 19.0, 'C': 44.0, 'G': 24.0, 'T': 18.0},\n",
       "  2: {'A': 17.0, 'C': 33.0, 'G': 4.0, 'T': 49.0},\n",
       "  3: {'A': 86.0, 'C': 4.0, 'G': 8.0, 'T': 6.0},\n",
       "  4: {'A': 97.0, 'C': 1.0, 'G': 2.0, 'T': 3.0},\n",
       "  5: {'A': 7.0, 'C': 1.0, 'G': 2.0, 'T': 94.0},\n",
       "  6: {'A': 94.0, 'C': 2.0, 'G': 2.0, 'T': 7.0},\n",
       "  7: {'A': 3.0, 'C': 2.0, 'G': 2.0, 'T': 97.0},\n",
       "  8: {'A': 5.0, 'C': 10.0, 'G': 5.0, 'T': 84.0},\n",
       "  9: {'A': 24.0, 'C': 4.0, 'G': 53.0, 'T': 23.0},\n",
       "  10: {'A': 35.0, 'C': 36.0, 'G': 19.0, 'T': 13.0},\n",
       "  11: {'A': 30.0, 'C': 14.0, 'G': 28.0, 'T': 32.0},\n",
       "  12: {'A': 58.0, 'C': 9.0, 'G': 20.0, 'T': 17.0},\n",
       "  13: {'A': 30.0, 'C': 19.0, 'G': 27.0, 'T': 28.0},\n",
       "  14: {'A': 35.0, 'C': 24.0, 'G': 16.0, 'T': 28.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'ARID-related',\n",
       "  'tf_class': 'ARID',\n",
       "  'pubmed_ids': '25215497',\n",
       "  'uniprot_ids': 'Q3U108',\n",
       "  'data_type': 'universal'},\n",
       " {'Matrix_ID': 'MA0757.1',\n",
       "  'Matrix_Name': 'ONECUT3',\n",
       "  1: {'A': 2840.0, 'C': 1023.0, 'G': 1389.0, 'T': 832.0},\n",
       "  2: {'A': 3852.0, 'C': 380.0, 'G': 1142.0, 'T': 710.0},\n",
       "  3: {'A': 6081.0, 'C': 275.0, 'G': 499.0, 'T': 879.0},\n",
       "  4: {'A': 6081.0, 'C': 174.0, 'G': 113.0, 'T': 1222.0},\n",
       "  5: {'A': 6081.0, 'C': 29.0, 'G': 103.0, 'T': 55.0},\n",
       "  6: {'A': 6081.0, 'C': 6.0, 'G': 42.0, 'T': 12.0},\n",
       "  7: {'A': 33.0, 'C': 104.0, 'G': 15.0, 'T': 6081.0},\n",
       "  8: {'A': 24.0, 'C': 6081.0, 'G': 4.0, 'T': 6.0},\n",
       "  9: {'A': 3951.0, 'C': 21.0, 'G': 2131.0, 'T': 12.0},\n",
       "  10: {'A': 6081.0, 'C': 7.0, 'G': 10.0, 'T': 23.0},\n",
       "  11: {'A': 44.0, 'C': 7.0, 'G': 8.0, 'T': 6081.0},\n",
       "  12: {'A': 6081.0, 'C': 106.0, 'G': 188.0, 'T': 346.0},\n",
       "  13: {'A': 3621.0, 'C': 862.0, 'G': 418.0, 'T': 1183.0},\n",
       "  14: {'A': 1863.0, 'C': 1272.0, 'G': 634.0, 'T': 2316.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 14,\n",
       "  'tf_family': 'HD-CUT',\n",
       "  'tf_class': 'Homeo',\n",
       "  'pubmed_ids': '8887657',\n",
       "  'uniprot_ids': 'O60422',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0728.1',\n",
       "  'Matrix_Name': 'Nr2f6(var.2)',\n",
       "  1: {'A': 101.0, 'C': 17.0, 'G': 173.0, 'T': 1.0},\n",
       "  2: {'A': 266.0, 'C': 1.0, 'G': 54.0, 'T': 4.0},\n",
       "  3: {'A': 1.0, 'C': 1.0, 'G': 211.0, 'T': 1.0},\n",
       "  4: {'A': 2.0, 'C': 1.0, 'G': 184.0, 'T': 1.0},\n",
       "  5: {'A': 1.0, 'C': 1.0, 'G': 2.0, 'T': 379.0},\n",
       "  6: {'A': 1.0, 'C': 668.0, 'G': 2.0, 'T': 52.0},\n",
       "  7: {'A': 495.0, 'C': 31.0, 'G': 1.0, 'T': 2.0},\n",
       "  8: {'A': 407.0, 'C': 5.0, 'G': 8.0, 'T': 10.0},\n",
       "  9: {'A': 166.0, 'C': 1.0, 'G': 95.0, 'T': 60.0},\n",
       "  10: {'A': 259.0, 'C': 1.0, 'G': 1.0, 'T': 1.0},\n",
       "  11: {'A': 1.0, 'C': 1.0, 'G': 325.0, 'T': 1.0},\n",
       "  12: {'A': 1.0, 'C': 1.0, 'G': 156.0, 'T': 6.0},\n",
       "  13: {'A': 3.0, 'C': 1.0, 'G': 1.0, 'T': 370.0},\n",
       "  14: {'A': 1.0, 'C': 679.0, 'G': 1.0, 'T': 4.0},\n",
       "  15: {'A': 447.0, 'C': 1.0, 'G': 10.0, 'T': 1.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 15,\n",
       "  'tf_family': 'RXR-related',\n",
       "  'tf_class': 'Nuclear',\n",
       "  'pubmed_ids': '12690040',\n",
       "  'uniprot_ids': 'P43136',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA0848.1',\n",
       "  'Matrix_Name': 'FOXO4',\n",
       "  1: {'A': 1670.0, 'C': 14.0, 'G': 5379.0, 'T': 1.0},\n",
       "  2: {'A': 346.0, 'C': 70.0, 'G': 124.0, 'T': 5379.0},\n",
       "  3: {'A': 5379.0, 'C': 1260.0, 'G': 33.0, 'T': 21.0},\n",
       "  4: {'A': 5379.0, 'C': 147.0, 'G': 20.0, 'T': 21.0},\n",
       "  5: {'A': 5379.0, 'C': 1.0, 'G': 91.0, 'T': 11.0},\n",
       "  6: {'A': 59.0, 'C': 5379.0, 'G': 34.0, 'T': 959.0},\n",
       "  7: {'A': 5379.0, 'C': 11.0, 'G': 11.0, 'T': 41.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 7,\n",
       "  'tf_family': 'Forkhead',\n",
       "  'tf_class': 'Fork',\n",
       "  'pubmed_ids': '10880363',\n",
       "  'uniprot_ids': 'P98177',\n",
       "  'data_type': 'HT-SELEX'},\n",
       " {'Matrix_ID': 'MA1630.1',\n",
       "  'Matrix_Name': 'Znf281',\n",
       "  1: {'A': 1198.0, 'C': 879.0, 'G': 3540.0, 'T': 955.0},\n",
       "  2: {'A': 997.0, 'C': 1077.0, 'G': 3752.0, 'T': 746.0},\n",
       "  3: {'A': 9.0, 'C': 234.0, 'G': 30.0, 'T': 6299.0},\n",
       "  4: {'A': 20.0, 'C': 89.0, 'G': 6450.0, 'T': 13.0},\n",
       "  5: {'A': 26.0, 'C': 164.0, 'G': 6357.0, 'T': 25.0},\n",
       "  6: {'A': 37.0, 'C': 132.0, 'G': 6379.0, 'T': 24.0},\n",
       "  7: {'A': 23.0, 'C': 116.0, 'G': 6403.0, 'T': 30.0},\n",
       "  8: {'A': 19.0, 'C': 42.0, 'G': 6458.0, 'T': 53.0},\n",
       "  9: {'A': 6505.0, 'C': 46.0, 'G': 9.0, 'T': 12.0},\n",
       "  10: {'A': 736.0, 'C': 419.0, 'G': 4313.0, 'T': 1104.0},\n",
       "  11: {'A': 375.0, 'C': 357.0, 'G': 5524.0, 'T': 316.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 11,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '19443739',\n",
       "  'uniprot_ids': 'Q99LI5',\n",
       "  'data_type': 'ChIP-seq'},\n",
       " {'Matrix_ID': 'MA0139.1',\n",
       "  'Matrix_Name': 'CTCF',\n",
       "  1: {'A': 88.0, 'C': 292.0, 'G': 77.0, 'T': 460.0},\n",
       "  2: {'A': 168.0, 'C': 146.0, 'G': 415.0, 'T': 188.0},\n",
       "  3: {'A': 282.0, 'C': 50.0, 'G': 450.0, 'T': 135.0},\n",
       "  4: {'A': 57.0, 'C': 801.0, 'G': 22.0, 'T': 37.0},\n",
       "  5: {'A': 9.0, 'C': 904.0, 'G': 1.0, 'T': 3.0},\n",
       "  6: {'A': 745.0, 'C': 14.0, 'G': 66.0, 'T': 92.0},\n",
       "  7: {'A': 41.0, 'C': 529.0, 'G': 335.0, 'T': 12.0},\n",
       "  8: {'A': 108.0, 'C': 434.0, 'G': 49.0, 'T': 325.0},\n",
       "  9: {'A': 852.0, 'C': 12.0, 'G': 33.0, 'T': 19.0},\n",
       "  10: {'A': 6.0, 'C': 1.0, 'G': 904.0, 'T': 4.0},\n",
       "  11: {'A': 334.0, 'C': 4.0, 'G': 567.0, 'T': 10.0},\n",
       "  12: {'A': 55.0, 'C': 13.0, 'G': 505.0, 'T': 342.0},\n",
       "  13: {'A': 13.0, 'C': 1.0, 'G': 891.0, 'T': 9.0},\n",
       "  14: {'A': 57.0, 'C': 9.0, 'G': 776.0, 'T': 72.0},\n",
       "  15: {'A': 105.0, 'C': 734.0, 'G': 6.0, 'T': 68.0},\n",
       "  16: {'A': 373.0, 'C': 14.0, 'G': 508.0, 'T': 18.0},\n",
       "  17: {'A': 83.0, 'C': 483.0, 'G': 308.0, 'T': 38.0},\n",
       "  18: {'A': 118.0, 'C': 323.0, 'G': 74.0, 'T': 397.0},\n",
       "  19: {'A': 403.0, 'C': 182.0, 'G': 267.0, 'T': 60.0},\n",
       "  'tax_group': 'vertebrates',\n",
       "  'TF_len': 19,\n",
       "  'tf_family': 'More',\n",
       "  'tf_class': 'C2H2',\n",
       "  'pubmed_ids': '17512414',\n",
       "  'uniprot_ids': 'P49711',\n",
       "  'data_type': 'ChIP-seq'}]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "infodicts_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_TFNAME = \"TEF\"\n",
    "\n",
    "tfs_to_check = []\n",
    "try:\n",
    "    tf_list_file = open(test_TFNAME,'r')\n",
    "    for line in tf_list_file:\n",
    "        tfs_to_check.append(line.split('\\n'))\n",
    "except FileNotFoundError:\n",
    "    tfs_to_check.append(test_TFNAME)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for i in infodicts_list:\n",
    "#     print(i['Matrix_Name'],i['TF_len'])\n",
    "bg_z_df = read_csv('/local3/jake/TF_binding/backgroundZ_forTFs.100reps.txt',delimiter='\\t',skiprows=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF_name</th>\n",
       "      <th>BG Z score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>134</td>\n",
       "      <td>ZNF148</td>\n",
       "      <td>0.136562</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    TF_name  BG Z score\n",
       "134  ZNF148    0.136562"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bg_z_df.loc[bg_z_df['TF_name'] == 'ZNF148']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([741.,   3.,   1.,   0.,   0.,   0.,   0.,   0.,   0.,   1.]),\n",
       " array([3.67639196e-11, 1.27126115e+03, 2.54252230e+03, 3.81378344e+03,\n",
       "        5.08504459e+03, 6.35630574e+03, 7.62756689e+03, 8.89882803e+03,\n",
       "        1.01700892e+04, 1.14413503e+04, 1.27126115e+04]),\n",
       " <a list of 10 Patch objects>)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXcAAAD4CAYAAAAXUaZHAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAR+0lEQVR4nO3db2xe53nf8e+vUqy0aTNLNSVokjApgJBVHhA7Jbx4GYouamvlDyIPmAEGzapuHjRg3pBsAzppeTH0hQBnG4pu2NxOSNIJrRtVTZNZcLG2gtpgGDBYpRs3sSxrYqLU4qRKbIo2bQN4lXvtxXOreSyR4iORj0je+34A4tznOvd5znUo+sfD8/xxqgpJUl++Y6UbkCQtP8NdkjpkuEtShwx3SeqQ4S5JHVq/0g0APPDAA7Vz586VbkOS1pQXX3zxD6pqYr5tqyLcd+7cyfT09Eq3IUlrSpLfW2ibt2UkqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDq+Idqku189Cvrshxv/70B1fkuJK0GK/cJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nq0KLhnuSdSV4a+vpmko8n2ZTkVJILbblxaJ/DSWaSnE/y2HhPQZJ0s0XDvarOV9VDVfUQ8P3At4AvAIeA01W1Gzjd1kmyB5gCHgT2Ac8kWTem/iVJ87jT2zJ7ga9W1e8B+4FjrX4MeLyN9wPHq+r1qroIzACPLEezkqTR3Gm4TwGfbeMtVXUFoC03t/o24NLQPrOt9iZJDiaZTjI9Nzd3h21Ikm5n5HBPch/wYeCXF5s6T61uKVQdrarJqpqcmJgYtQ1J0gju5Mr9/cDvVNXVtn41yVaAtrzW6rPAjqH9tgOXl9qoJGl0dxLuH+Hbt2QATgIH2vgA8NxQfSrJhiS7gN3AmaU2Kkka3Ugf+Zvku4AfBv7xUPlp4ESSJ4HXgCcAqupskhPAK8B14KmqemNZu5Yk3dZI4V5V3wK+96baNxi8ema++UeAI0vuTpJ0V3yHqiR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktShkcI9yf1JPpfk1STnkjyaZFOSU0kutOXGofmHk8wkOZ/ksfG1L0maz6hX7v8B+LWq+uvAu4BzwCHgdFXtBk63dZLsAaaAB4F9wDNJ1i1345KkhS0a7kneDvwA8GmAqvq/VfVHwH7gWJt2DHi8jfcDx6vq9aq6CMwAjyx345KkhY1y5f4OYA74uSRfSvKpJG8DtlTVFYC23NzmbwMuDe0/22pvkuRgkukk03Nzc0s6CUnSm40S7uuBdwM/U1UPA39GuwWzgMxTq1sKVUerarKqJicmJkZqVpI0mlHCfRaYraoX2vrnGIT91SRbAdry2tD8HUP7bwcuL0+7kqRRLBruVfX7wKUk72ylvcArwEngQKsdAJ5r45PAVJINSXYBu4Ezy9q1JOm21o84758Bzya5D/ga8A8Y/GI4keRJ4DXgCYCqOpvkBINfANeBp6rqjWXvXJK0oJHCvapeAibn2bR3gflHgCNL6EuStAS+Q1WSOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nq0EjhnuTrSb6S5KUk0622KcmpJBfacuPQ/MNJZpKcT/LYuJqXJM3vTq7c/05VPVRVN/5H2YeA01W1Gzjd1kmyB5gCHgT2Ac8kWbeMPUuSFrGU2zL7gWNtfAx4fKh+vKper6qLwAzwyBKOI0m6Q6OGewG/keTFJAdbbUtVXQFoy82tvg24NLTvbKu9SZKDSaaTTM/Nzd1d95Kkea0fcd57q+pyks3AqSSv3mZu5qnVLYWqo8BRgMnJyVu2S5Lu3khX7lV1uS2vAV9gcJvlapKtAG15rU2fBXYM7b4duLxcDUuSFrdouCd5W5LvuTEGfgR4GTgJHGjTDgDPtfFJYCrJhiS7gN3AmeVuXJK0sFFuy2wBvpDkxvxfrKpfS/LbwIkkTwKvAU8AVNXZJCeAV4DrwFNV9cZYupckzWvRcK+qrwHvmqf+DWDvAvscAY4suTtJ0l3xHaqS1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4Z7pLUIcNdkjpkuEtSh0YO9yTrknwpyfNtfVOSU0kutOXGobmHk8wkOZ/ksXE0Lkla2J1cuX8MODe0fgg4XVW7gdNtnSR7gCngQWAf8EySdcvTriRpFCOFe5LtwAeBTw2V9wPH2vgY8PhQ/XhVvV5VF4EZ4JHlaVeSNIpRr9x/GvgJ4C+Galuq6gpAW25u9W3ApaF5s632JkkOJplOMj03N3fHjUuSFrZouCf5EHCtql4c8TEzT61uKVQdrarJqpqcmJgY8aElSaNYP8Kc9wIfTvIB4K3A25P8AnA1ydaqupJkK3CtzZ8Fdgztvx24vJxNS5Jub9Er96o6XFXbq2ongydKf7OqPgqcBA60aQeA59r4JDCVZEOSXcBu4Myydy5JWtAoV+4LeRo4keRJ4DXgCYCqOpvkBPAKcB14qqreWHKnkqSR3VG4V9UXgS+28TeAvQvMOwIcWWJvkqS75DtUJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4tGu5J3prkTJLfTXI2yU+2+qYkp5JcaMuNQ/scTjKT5HySx8Z5ApKkW41y5f468L6qehfwELAvyXuAQ8DpqtoNnG7rJNkDTAEPAvuAZ5KsG0fzkqT5LRruNfCnbfUt7auA/cCxVj8GPN7G+4HjVfV6VV0EZoBHlrVrSdJtjXTPPcm6JC8B14BTVfUCsKWqrgC05eY2fRtwaWj32Va7+TEPJplOMj03N7eUc5Ak3WSkcK+qN6rqIWA78EiSv3Gb6ZnvIeZ5zKNVNVlVkxMTE6N1K0kayR29Wqaq/gj4IoN76VeTbAVoy2tt2iywY2i37cDlJXcqSRrZKK+WmUhyfxt/J/BDwKvASeBAm3YAeK6NTwJTSTYk2QXsBs4sd+OSpIWtH2HOVuBYe8XLdwAnqur5JP8LOJHkSeA14AmAqjqb5ATwCnAdeKqq3hhP+5Kk+Swa7lX1ZeDheerfAPYusM8R4MiSu5Mk3RXfoSpJHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1aNFwT7IjyW8lOZfkbJKPtfqmJKeSXGjLjUP7HE4yk+R8ksfGeQKSpFuNcuV+HfiXVfV9wHuAp5LsAQ4Bp6tqN3C6rdO2TQEPAvuAZ5KsG0fzkqT5LRruVXWlqn6njf8EOAdsA/YDx9q0Y8DjbbwfOF5Vr1fVRWAGeGS5G5ckLeyO7rkn2Qk8DLwAbKmqKzD4BQBsbtO2AZeGdptttZsf62CS6STTc3Nzd965JGlBI4d7ku8GfgX4eFV983ZT56nVLYWqo1U1WVWTExMTo7YhSRrBSOGe5C0Mgv3Zqvp8K19NsrVt3wpca/VZYMfQ7tuBy8vTriRpFKO8WibAp4FzVfVTQ5tOAgfa+ADw3FB9KsmGJLuA3cCZ5WtZkrSY9SPMeS/w94GvJHmp1f418DRwIsmTwGvAEwBVdTbJCeAVBq+0eaqq3lj2ziVJC1o03KvqfzL/fXSAvQvscwQ4soS+JElL4DtUJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdMtwlqUOGuyR1yHCXpA4tGu5JPpPkWpKXh2qbkpxKcqEtNw5tO5xkJsn5JI+Nq3FJ0sJGuXL/r8C+m2qHgNNVtRs43dZJsgeYAh5s+zyTZN2ydStJGsmi4V5V/wP4w5vK+4FjbXwMeHyofryqXq+qi8AM8Mgy9SpJGtHd3nPfUlVXANpyc6tvAy4NzZtttVskOZhkOsn03NzcXbYhSZrPcj+hmnlqNd/EqjpaVZNVNTkxMbHMbUjS/9/uNtyvJtkK0JbXWn0W2DE0bztw+e7bkyTdjbsN95PAgTY+ADw3VJ9KsiHJLmA3cGZpLUqS7tT6xSYk+Szwg8ADSWaBfwM8DZxI8iTwGvAEQFWdTXICeAW4DjxVVW+MqXdJ0gIWDfeq+sgCm/YuMP8IcGQpTUmSlsZ3qEpShwx3SeqQ4S5JHTLcJalDhrskdchwl6QOGe6S1CHDXZI6ZLhLUocMd0nqkOEuSR0y3CWpQ4a7JHXIcJekDhnuktQhw12SOmS4S1KHDHdJ6pDhLkkdGlu4J9mX5HySmSSHxnUcSdKtxhLuSdYB/xl4P7AH+EiSPeM4liTpVuvH9LiPADNV9TWAJMeB/cArYzreith56FdX5Lhff/qDK3JcSWvHuMJ9G3BpaH0W+JvDE5IcBA621T9Ncn4Jx3sA+IMl7L+S7rj3fHJMndydtfy9B/tfafa/NH9toQ3jCvfMU6s3rVQdBY4uy8GS6aqaXI7HutfWcu9g/yvN/lfWau5/XE+ozgI7hta3A5fHdCxJ0k3GFe6/DexOsivJfcAUcHJMx5Ik3WQst2Wq6nqSfwr8OrAO+ExVnR3HsZplub2zQtZy72D/K83+V9aq7T9VtfgsSdKa4jtUJalDhrskdWhNh/tq/IiDJDuS/FaSc0nOJvlYq29KcirJhbbcOLTP4XYO55M8NlT//iRfadv+Y5L5XmI6rvNYl+RLSZ5fa/0nuT/J55K82v4dHl1j/f/z9rPzcpLPJnnrau4/yWeSXEvy8lBt2fpNsiHJL7X6C0l23oP+/137+flyki8kuX+19r+gqlqTXwyeqP0q8A7gPuB3gT2roK+twLvb+HuA/83gIxj+LXCo1Q8Bn2zjPa33DcCudk7r2rYzwKMM3jfw34H338Pz+BfALwLPt/U10z9wDPhHbXwfcP9a6Z/BGwAvAt/Z1k8AP76a+wd+AHg38PJQbdn6Bf4J8LNtPAX80j3o/0eA9W38ydXc/4LndS8OMqYfqEeBXx9aPwwcXum+5unzOeCHgfPA1lbbCpyfr28GrzB6tM15daj+EeC/3KOetwOngffx7XBfE/0Db2cQjrmpvlb6v/Hu7k0MXs32fAuaVd0/sPOmcFy2fm/MaeP1DN4RmnH2f9O2vws8u5r7n+9rLd+Wme8jDratUC/zan9+PQy8AGypqisAbbm5TVvoPLa18c31e+GngZ8A/mKotlb6fwcwB/xcu630qSRvY430X1X/B/j3wGvAFeCPq+o3WCP9D1nOfv9yn6q6Dvwx8L1j6/xW/5DBlfibemlWbf9rOdwX/YiDlZTku4FfAT5eVd+83dR5anWb+lgl+RBwrapeHHWXeWor1j+DK6N3Az9TVQ8Df8bgtsBCVlX/7d70fgZ/8v9V4G1JPnq7XeapreT3fzF30++KnUuSTwDXgWcX6WXV9b+Ww33VfsRBkrcwCPZnq+rzrXw1yda2fStwrdUXOo/ZNr65Pm7vBT6c5OvAceB9SX6BtdP/LDBbVS+09c8xCPu10v8PAReraq6q/hz4PPC3WDv937Cc/f7lPknWA38F+MOxdd4kOQB8CPjRavdUWEP9r+VwX5UfcdCeIf80cK6qfmpo00ngQBsfYHAv/kZ9qj2jvgvYDZxpf8r+SZL3tMf8saF9xqaqDlfV9qrayeB7+ptV9dE11P/vA5eSvLOV9jL4qOk10T+D2zHvSfJd7bh7gXNrqP8blrPf4cf6ewx+Jsd65ZtkH/CvgA9X1beGNq2J/oG1+4Rq+958gMGrUb4KfGKl+2k9/W0Gf3J9GXipfX2AwT2208CFttw0tM8n2jmcZ+gVDcAk8HLb9p+4B0/C3HQuP8i3n1BdM/0DDwHT7d/gvwEb11j/Pwm82o798wxembFq+wc+y+D5gT9ncJX65HL2C7wV+GVghsErUt5xD/qfYXCf/MZ/wz+7Wvtf6MuPH5CkDq3l2zKSpAUY7pLUIcNdkjpkuEtShwx3SeqQ4S5JHTLcJalD/w9rxp6v/BsldwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.hist(bg_z_df['BG Z score'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3.4360282958021792e-06"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate_bindingP(9.85326613035509,bg_z_df,'NFATC1')\n",
    "np."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "too many values to unpack (expected 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-111-9de1fd5dabe2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mcurr_matrix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_byTF\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfodicts_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m#     print(curr_matrix,curr_seq)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m     \u001b[0mcurr_refscore\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcurr_altscore\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_matrix_scores\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtfname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'tf_name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmatrix_dict\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcurr_matrix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mref_al\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'T'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0malt_al\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'G'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mposition\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcurr_relative_pos\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mreference_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcurr_seq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0;31m#     print(g['tf_name'],curr_refscore,curr_altscore)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;31m#     print((g['pos_start']-ref_pos_start),(g['pos_end']-ref_pos_start),\"\"(position-ref_pos_start))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: too many values to unpack (expected 2)"
     ]
    }
   ],
   "source": [
    "for i,g in gene_df.iterrows():\n",
    "    curr_relative_pos = abs(updated_pos-g['relative_start'])\n",
    "    curr_seq = ref_full_seq[g['relative_start']:(g['relative_end']+1)]\n",
    "    \n",
    "    curr_matrix = get_matrix_byTF(g['tf_name'],infodicts_list)\n",
    "#     print(curr_matrix,curr_seq)\n",
    "    curr_refscore,curr_altscore = get_matrix_scores(tfname=g['tf_name'],matrix_dict=curr_matrix,ref_al='T',alt_al='G',position=curr_relative_pos,reference_seq=curr_seq)\n",
    "#     print(g['tf_name'],curr_refscore,curr_altscore)\n",
    "#     print((g['pos_start']-ref_pos_start),(g['pos_end']-ref_pos_start),\"\"(position-ref_pos_start))\n",
    "#     print(g['pos_start'],g['pos_end'],position)\n",
    "#     print(g['relative_start'],g['relative_end'],curr_relative_pos,updated_pos)\n",
    "    print(curr_seq,curr_seq[curr_relative_pos])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OTX2 {'ref_score': 264488.0, 'alt_score': 148783.0}\n",
      "KLF5 {'ref_score': 72298.0, 'alt_score': 72429.0}\n",
      "NR2C2(var.2) {'ref_score': 28556.0, 'alt_score': 47184.0}\n",
      "PITX1 {'ref_score': 41886.0, 'alt_score': 31355.0}\n",
      "RHOXF1 {'ref_score': 40212.0, 'alt_score': 29296.0}\n",
      "NR2C1 {'ref_score': 16249.0, 'alt_score': 25871.0}\n",
      "PITX3 {'ref_score': 27003.0, 'alt_score': 20765.0}\n",
      "THRB {'ref_score': 14938.0, 'alt_score': 15644.0}\n",
      "HOXB6 {'ref_score': 12451.0, 'alt_score': 13489.0}\n",
      "GSC2 {'ref_score': 8967.0, 'alt_score': 6951.0}\n",
      "RARA::RXRG {'ref_score': 3092.0, 'alt_score': 4033.0}\n",
      "VDR {'ref_score': 1707.0, 'alt_score': 1538.0}\n",
      "SOX18 {'ref_score': 1374.0, 'alt_score': 925.0}\n",
      "Dmbx1 {'ref_score': 396.0, 'alt_score': 405.0}\n",
      "MZF1(var.2) {'ref_score': 17.0, 'alt_score': 28.0}\n"
     ]
    }
   ],
   "source": [
    "for tf,scores in sorted(score_dict_bytf.items(), key=lambda k_v: k_v[1]['alt_score'],reverse=True):\n",
    "    print(tf,scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "results_loc = '/local3/jake/TF_binding/tophit_SNPs/output/'\n",
    "results_files_list = os.listdir(results_loc)\n",
    "results_files_bytraitcode = {}\n",
    "for f in results_files_list:\n",
    "    if(f.split('.')[-1] != 'PWM_scores'):\n",
    "        continue\n",
    "    traitcode = int(f.split('.')[0]) #\".\".join(f.split('.')[:2])\n",
    "    if(traitcode not in results_files_bytraitcode.keys()):\n",
    "        results_files_bytraitcode[traitcode] = []\n",
    "    results_files_bytraitcode[traitcode].append(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bgseq_premade(seqinfo_file,reps,slen,fastaloc,chrmsizes):\n",
    "    chrms_touse = list(chrmsizes['chrom'])\n",
    "    bgseq_list = []\n",
    "    genseqs_df = read_csv(seqinfo_file,delimiter='\\t')\n",
    "    for r,seq in genseqs_df.iterrows():\n",
    "        chrmfile = '{0}/chr{1}.fa'.format(fastaloc,seq['Chrm'])\n",
    "        curr_seq = pybedtools.BedTool.seq('chr{0}:{1}-{2}'.format(seq['Chrm'],seq['Start'],seq['End']),chrmfile)\n",
    "        bgseq_list.append([seq['Chrm'],seq['Start'],seq['End'],curr_seq.upper()])\n",
    "    if(len(bgseq_list != reps)):\n",
    "        new_bgseqs = get_random_bgseqs(slen,reps,)\n",
    "    return bgseq_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "genseqs_df = read_csv('/local3/jake/TF_binding/testing/TGIF2LX.10000reps.random_bgseq.info',delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13\n"
     ]
    }
   ],
   "source": [
    "for r,seq in genseqs_df.iterrows():\n",
    "    print(seq['Chrm'])\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genseqs_df.loc[0]['End'] - genseqs_df.loc[0]['Start']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
